
<h1>Some Basic Exercises Using KBase</h1>
<p><strong>Purpose: </strong>Learn more about the Central Data Model
(CDM) Entity-Relationshop Model and
how to run command line tools to expose the data in it. </p>
<p><strong>Required Prerequisite Activities: </strong><a href="/for-users/tutorials/installing-the-kbase-app-dmg/">Download
KBase: Installing the KBase Application DMG</a>, OR <a href="for-users/tutorials/navigating-iris/">Get Started with IRIS:
Navigating IRIS</a> </p>
<p> <a href="/developer-zone/tutorials/the-kbase-central-data-model/the-cdm-represented-as-an-entity-relationship-model/">The
CDM
Represented as an Entity-Relationship Model </a></p>
<h2>Introduction</h2>
<p>KBase includes an integrated database of genomic data called the
Central Store (<i>CS</i>). Extract data
from the CS using a defined Application Programmic Interface (<i>API</i>),
or access the CS via a fairly simple set
of command-line scripts. The most simple way to access the data in the
CS is using command-line scripts that produce tab-separated tables that
can be examined using common tools. We will demonstrate the basic tools
we provide below. We urge you to go slowly through a few examples in
this tutorial, typing the actual commands and verifying that you
understand the details as they are presented.
</p>
<h2>How to Work with the CDM Entity-Relationship Model</h2>
It is important that you gain some familiarity with the ER-model that
describes the contents of the KBase CS. An entity-relationship model is
an abstract description of the architecture of a database. It describes
the types of <i>entities</i> and the <i>relationships</i> between
entities. The data fields associated with each entity type constitute
most of the data in the CS. However, fields can be associated with
relationships, as well. For example, if two entities of type <i>Feature</i>
are related by a relationship of
type <i>IsCoregulatedWith85</i>, then the Pearson correlation
coefficient is an attribute of the
relationship, and not of either of the related entities by themselves.
The entity types and relationship types are mapped into a standard
relational database (in the case of the CS), but it might well be a
good idea to peruse the ER-model and attempt to gain at least a
rudimentary understanding of the types of data we have included.
<p> There are three standard tools for accessing fields in entities and
relationships, and there is also a set of customized tools. We will
illustrate these tools with simple examples. First, we will determine
which features in the CS
are connected to publications.
This activity is useful for gaining insight as to the reliability of
function assignments. Begin by looking at the <a href="http://kbase.us/developer-zone/tutorials/the-kbase-central-data-model/the-cdm-represented-as-an-entity-relationship-model/">
CDM as represented as
an ER Model</a>. This model describes the entities and relationships
that make up the Central Store. If you go to the <a href="http://pubseed.theseed.org/ErdbDocWidget.cgi?xmlFileName=/home/parrello/CdmiData/Published/KSaplingDBD.xml">ER-Model</a>,
look for the path from
the entity <i>Publication</i> to the
entity <i>Feature</i>. </p>
<p>Verify the following: </p>
<ol>
  <li>Publication is an entity that
connects to the entity <i>ProteinSequence</i>
via the relationship <i>Concerns</i>.
It can be thought of as <i>Publication</i> <i>Concerns</i> <i>ProteinSequence</i>.
This describes crossing the relationship from <i>Publication</i> to <i>ProteinSequence</i>.
There is a
relationship going in the other direction, as well -- even though the
diagram only explicitly lists the name of the one going from <i>Publication</i>
to <i>ProteinSequence</i>. If you click on the
entity <i>Publication</i>, it describes
the relationships that entity type participates in. Similarly, if you
click on <i>ProteinSequence</i>, you
will see that the connection going from <i>ProteinSequence</i>
to <i>Publication</i> is named <i>IsATopicOf</i>
(<i>ProteinSequence IsATopicOf Publication</i>). You need to be
comfortable
seeing what fields are in specific entities and relationships, as well
as understanding the names of the relationships connecting entities.
Note that, if you are looking at the description of a single entity or
relationship, and you wish to go back to see the overall diagram, you
need to hit the "Image" button, and not the back arrow on your browser.
    <br>
    <br>
  </li>
  <li> To get from <i>ProteinSequence</i>
to a related <i>Feature</i> linked to the protein,
one crosses the <i>IsProteinFor</i>
relationship (<i>ProteinSequence IsProteinFor Feature</i>). <br>
    <br>
  </li>
  <li> Thus, one interpretation of the original goal of seeing which
features in the CS connect to publications, is
    <blockquote> Start with the list of <i>Publication</i>s.
For each one, locate the <i>ProteinSequence</i>
described in the publication. Then, use the protein sequences to find
the Features and produce a 2-column table [<i>Publication, Feature</i>]
    </blockquote>
    <br>
Note that you could start with Features and trace back to <i>Publication</i>s,
but there are millions of
Features and only thousands of <i>Publication</i>s
in the CS. It will probably be faster to look up the features connected
to the publications, rather than the other way around. <br>
    <br>
  </li>
  <li> Now, to see a table of the <i>Publication</i>
entities just type <br>
    <pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#all_entities_Publication" target="_blank">all_entities_Publication</a> -a | head<br><br></pre>
For every entity type in the CS, a command like this exists to display
the instances of that entity. We encourage you to try it out. (We pipe
the output to "head" which just shows the head of a file.) <br>
    <br>
To determine which genomes are in the CS, use <br>
    <pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#all_entities_Genome" target="_blank">all_entities_Genome</a> -a<br><br></pre>
Try it again leaving the "-a" option off. Then use <br>
    <pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#all_entities_Genome" target="_blank">all_entities_Genome</a> -show-fields<br><br></pre>
to see what fields the <i>Genome</i> entity contains (compare this
against the <i>ER-Model</i>
description of the fields). Finally, note what <br>
    <pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#all_entities_Genome" target="_blank">all_entities_Genome</a> -f 'contigs,domain,scientific_name'<br><br></pre>
produces. The Unix command <code> grep</code> can be used to
filter out lines containing a desired string. Thus, <br>
    <pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#all_entities_Genome" target="_blank">all_entities_Genome</a> -f scientific_name | grep Streptococcus<br><br></pre>
would give you the IDs of the <i>Streptococcus</i> genomes in KBase. <br>
    <br>
  </li>
  <li> Now, run <br>
    <pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#all_entities_Publication" target="_blank">all_entities_Publication</a> | <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_relationship_Concerns" target="_blank">get_relationship_Concerns</a> -to id<br><br></pre>
This illustrates how to "cross a relationship." The <code>all_entities_Publication</code>
produces a 1-column
table containing pubmed IDs. These go into the command <br>
    <pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_relationship_Concerns" target="_blank">get_relationship_Concerns</a> -to id<br><br></pre>
This is an example of the second class of commands. One thinks of a <code>get_relationship</code>
command as taking as input a set of IDs from the <i>from entity</i>,
crossing the relationship to the <i>to entity</i>, and extracting
whatever fields you want from the <i>from entity</i>, the <i>relationship</i>,
and the <i>to entity</i>. The general form is <br>
    <pre>  get_relationship_<b>REL</b> -from Fields1 -rel RelFields -to Fields2<br><br></pre>
The effect of running the command is to append columns to the input
table. The columns correspond to the fields designated in <i>Fields1,
RelFields</i>, and <i>Fields2</i>. <br>
    <br>
Thus, <br>
    <pre>  echo 'kb|g.0' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_relationship_IsOwnerOf" target="_blank">get_relationship_IsOwnerOf</a> -to id<br><br></pre>
would produce a 2-column table containing [GenomeID,FeatureID] where
all of the GenomeID fields would contain kb|g.0, the CS ID for <i><a name="E.coli124">E.coli</a></i>. <br>
    <br>
  </li>
  <li> Finally, <br>
    <pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#all_entities_Publication" target="_blank">all_entities_Publication</a> | <br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_relationship_Concerns" target="_blank">get_relationship_Concerns</a> -to id |<br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_relationship_IsProteinFor" target="_blank">get_relationship_IsProteinFor</a> -to id<br><br></pre>
would produce a 3-column table <br>
    <pre>  [pubmed-id,protein-id,feature-id]	    <br><br></pre>
We originally wanted only the first and last columns, so we could pipe
the output into <br>
    <pre>  cut -f1,3<br><br></pre>
to get just <br>
    <pre>  [pubmed-id,feature-id]<br><br></pre>
  </li>
</ol>
<br>
Now, we suggest that you do a similar exercise to get the <b>Reactions</b>
and <b>Roles</b> (i.e., the functional roles) that can be connected to
<b>Features</b> connected to <b>Publications</b>. To do this you need
to realize that the overall ER-model is large and subsets are normally
examined. You should examine both the "main" and "Models" subsets --
you need to construct a path starting at the entity <a name="Reaction125">Reaction</a> (in the <a name="Model126">Model</a>
subset of the overall diagram) and trace it through to <b><a name="Publication127">Publication</a></b> (in the Main subset). Begin
by making a list of the entities and relationships in the path: <br>
<pre>  Reaction        	IsUsedAs       	ReactionRule<br>  ReactionRule    	IsStepOf       	Complex<br>  Complex         	IsTriggeredBy  	Role<br>  Role            	IsFunctionalIn 	Feature<br>  Feature         	Produces       	ProteinSequence<br>  ProteinSequence 	IsATopicOf     	Publication<br><br></pre>
Implementing this path can be done using <br>
<pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#all_entities_Reaction" target="_blank">all_entities_Reaction</a> |<br>  get_relationship_IsUsedAs -to id |<br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_relationship_IsStepOf" target="_blank">get_relationship_IsStepOf</a> -to id |<br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_relationship_IsTriggeredBy" target="_blank">get_relationship_IsTriggeredBy</a> -to id |<br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_relationship_IsFunctionalIn" target="_blank">get_relationship_IsFunctionalIn</a> -to id |<br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_relationship_Produces" target="_blank">get_relationship_Produces</a> -to id |<br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_relationship_IsATopicOf" target="_blank">get_relationship_IsATopicOf</a> -to id <br><br></pre>
This picks up all of the Features used in models on the way to getting
them connected to <b><a name="Publication128">Publication</a></b>s. It
will probably be more efficient to use <br>
<pre>  Publication        Concerns           ProteinSequence<br>  ProteinSequence    IsProteinFor       Feature<br>  Feature            HasFunctional      Role<br>  Role               Triggers           Complex<br>  Complex            HasStep            ReactionRule<br>  ReactionRule       IsUseOf            Reaction<br><br></pre>
implemented as <br>
<pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#all_entities_Publication" target="_blank">all_entities_Publication</a> |<br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_relationship_Concerns" target="_blank">get_relationship_Concerns</a> -to id |<br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_relationship_IsProteinFor" target="_blank">get_relationship_IsProteinFor</a> -to id |<br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_relationship_HasFunctional" target="_blank">get_relationship_HasFunctional</a> -to id |<br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_relationship_Triggers" target="_blank">get_relationship_Triggers</a> -to id |<br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_relationship_HasStep" target="_blank">get_relationship_HasStep</a> -to id |<br>  get_relationship_IsUseOf -to id<br><br></pre>
Since we wanted only the pubmed IDs, the feature IDs, the roles, and
the reaction, you could pipe the output through <br>
<pre>  cut -f1,3,4,7<br><br></pre>
which extracts the four desired columns of the table. Finally, since
the last column is a reaction ID (and is quite unreadable), you might
pipe it through the command <br>
<pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#reaction_strings" target="_blank">reaction_strings</a>
<br></pre>
which appends a column containing readable version of the reaction.
<h2>Extracting Fields from a Set of Instances of an Entity</h2>
At this point, we have covered the notions of sketching out a path in
the <a name="ER-Model129">ER-Model</a> and collecting fields from the
entities and relationships. We employed two classes of commands <br>
<pre>  all_entities_<b>ENTITY</b>
</pre>
and <br>
<pre>  get_relationship_<b>RELATIONSHIP</b>
</pre>
There is a third general set of commands that are quite useful. They
have the form <br>
<pre>  get_entity_<b>ENTITY</b>
<br></pre>
They should be used when you have a set of IDs of entities, and you
wish to extract fields from the corresponding instances of the <b>ENTITY</b>
. Thus, <br>
<pre>  echo 'kb|g.0' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_entity_Genome" target="_blank">get_entity_Genome</a> -a<br><br></pre>
returns all of the fields from the <b><a name="Genome130">Genome</a></b>
entity representing data about an <i>Escherichia coli</i> genome. If
you wished to see just the name, number of contigs, and the size of the
genome, you might use <br>
<pre>  echo 'kb|g.0' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_entity_Genome" target="_blank">get_entity_Genome</a> -f 'scientific_name,contigs,dna_size'<br><br></pre>
<h2>Scripts Handling Popular Operations</h2>
Besides the three classes of general tools, there are scripts that
handle frequently employed operations.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#atomic_regulons_to_fids" target="_blank">atomic_regulons_to_fids</a></h3>
An atomic <a name="regulon131">regulon</a> is a set of genes that we
believe have identical expression profiles. This command takes as input
a table with a column containing "atomic <a name="regulon132">regulon</a>
" IDs and adds a column for the features that occur in the atomic <a name="regulon133">regulon</a>
. Try <br>
<pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#all_entities_AtomicRegulon" target="_blank">all_entities_AtomicRegulon</a> | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#atomic_regulons_to_fids" target="_blank">atomic_regulons_to_fids</a>
<br></pre>
to get a 2-column table for the atomic <a name="regulons134">regulons</a>
in the current release of the CS.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#contigs_to_lengths" target="_blank">contigs_to_lengths</a></h3>
It is frequently useful to get the lengths of the contigs that make up
a genome. This command list the contigs in a designated genome, along
with their lengths.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#contigs_to_sequences" target="_blank">contigs_to_sequences</a></h3>
You may wish to extract the sequence of specific contigs (or for all
contigs for a given genome). You have the option of outputting an extra
column with the sequence (use -fasta=0 to get sequence added as a
column). Thus, <br>
<pre>  echo 'kb|g.3857' | <br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_relationship_IsComposedOf" target="_blank">get_relationship_IsComposedOf</a> -to id | <br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#contigs_to_sequences" target="_blank">contigs_to_sequences</a>
</pre>
produces a FASTA file of the contigs in genome kb|g.3857.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_annotations" target="_blank">fids_to_annotations</a></h3>
Over the years, annotators have attached comments to some of the
protein-encoding genes maintained in KBase. This command gives you
access to those comments. Thus, <br>
<pre>  echo 'kb|g.0.peg.1089' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_annotations" target="_blank">fids_to_annotations</a>
</pre>
<br>
can be used to produce a list of <a name="annotations135">annotations</a>
(most recent first) separated by "//" lines.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_atomic_regulons" target="_blank">fids_to_atomic_regulons</a></h3>
This command is used to cross from a <a name="Feature136">Feature</a>
to the AtomicRegulons that contain it. It adds two columns: one with
the size of an atomic <a name="regulon137">regulon</a>
, and the other with the atomic <a name="regulon138">regulon</a> ID.
Thus, <br>
<pre>  echo 'kb|g.0.peg.4288' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_atomic_regulons" target="_blank">fids_to_atomic_regulons</a> | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#atomic_regulons_to_fids" target="_blank">atomic_regulons_to_fids</a>
</pre>
<br>
produces a 4-column table <br>
<pre>  [input-fid,sz-of-atomic-regulon,atomic-regulon-ID,fid-in-atomic-regulon]<br><br></pre>
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_coexpressed_fids" target="_blank">fids_to_coexpressed_fids</a></h3>
For some genomes in the CS, we have some expression data. When we have
data, it can be retrieved using this command. For example, <br>
<pre>  echo 'kb|g.0.peg.1457' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_coexpressed_fids" target="_blank">fids_to_coexpressed_fids</a>
</pre>
<br>
produces <br>
<pre>  kb|g.0.peg.1457	0.517	kb|g.0.peg.196<br>  kb|g.0.peg.1457	0.521	kb|g.0.peg.65<br>  kb|g.0.peg.1457	0.562	kb|g.0.peg.207<br>  kb|g.0.peg.1457	0.507	kb|g.0.peg.242<br>  .<br>  .<br><br></pre>
The last column is a co-expressed fid, and the one before it is the
Pearson correlation coefficient.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_co_occurring_fids" target="_blank">fids_to_co_occurring_fids</a></h3>
One of the major clues to function for genes in prokaryotic genomes
relates to the tendency for functionally-related genes to co-occur
close to one another on the <a name="chromosome139">chromosome</a>
(often in operons). This command checks to see if we have detected
co-occurrence. For example, <br>
<pre>  echo 'kb|g.0.peg.4288' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_co_occurring_fids" target="_blank">fids_to_co_occurring_fids</a>
<br></pre>
produces <br>
<pre>  kb|g.0.peg.4288	139	kb|g.0.peg.4288:kb|g.0.peg.2173<br>  kb|g.0.peg.4288	300	kb|g.0.peg.4288:kb|g.0.peg.3016<br>  kb|g.0.peg.4288	20	kb|g.0.peg.4288:kb|g.0.peg.3472<br>  kb|g.0.peg.4288	17	kb|g.0.peg.4288:kb|g.0.peg.3426<br>  kb|g.0.peg.4288	22	kb|g.0.peg.4288:kb|g.0.peg.3703<br>  kb|g.0.peg.4288	14	kb|g.0.peg.4288:kb|g.0.peg.3691<br><br></pre>
The first line says, "The genes looking like the pair of fids" <br>
<pre>  'kb|g.0.peg.4288:kb|g.0.peg.2173'<br><br></pre>
occur in 139 genomes from distinct OTUs. That is, there is substantial
evidence that these genes are functionally related. The last pair only
occur in 14 distinct OTUs, so the evidence here is pretty weak. You
will need to look at many pairs for which you know the functions to get
insight into the significance of different values of the score.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_dna_sequences" target="_blank">fids_to_dna_sequences</a></h3>
You can create either a tab-separated file in which <a name="DNA140">DNA</a>
sequence is the added column, or you can produce a FASTA file
containing the <a name="DNA141">DNA</a> sequence of a set of fids. The
default behavior is just to produce the FASTA file. Try <br>
<pre>  echo 'kb|g.8776' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_fids" target="_blank">genomes_to_fids</a> | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_dna_sequences" target="_blank">fids_to_dna_sequences</a>
<br></pre>
to get the <a name="DNA142">DNA</a> sequences of the genes of <i>Buchnera
aphidicola</i> str. Sg.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_functions" target="_blank">fids_to_functions</a></h3>
This command is used to take an input table in which fids are in a
column (the last column by default) and tacks on an extra column with
the functions of the fids. Thus, <br>
<pre>  echo 'kb|g.8776' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_fids" target="_blank">genomes_to_fids</a> | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_functions" target="_blank">fids_to_functions</a>
<br></pre>
could be used to get the current function assignments for the genes in
the <em>Buchnera</em> genome.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_literature" target="_blank">fids_to_literature</a></h3>
This command can be used to extract the <a name="literature143">literature</a>
references connected to one or more fids. It takes as input a table
with a column containing fids. It outputs a table with an extra column
containing a pubmed ID. Lines without a connection to any papers are
written to stderr. Try <br>
<pre>  echo 'kb|g.0.peg.118' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_literature" target="_blank">fids_to_literature</a>
<br></pre>
to get references attached to the gene encoding LSU ribosomal protein
L32p in <i>E. coli</i>
.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_locations" target="_blank">fids_to_locations</a></h3>
It is occasionally useful to map fid IDs to precise locations on
contigs. If you type <br>
<pre>  echo 'kb|g.0.peg.118' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_locations" target="_blank">fids_to_locations</a>
<br></pre>
you should get back <br>
<pre>  kb|g.0.peg.118	kb|g.0.c.1_1146590+174<br><br></pre>
which illustrates the KBase encoding of gene locations. A gene location
is a sequence of contiguous regions separated by commas. A contiguous
region is represented as a <a name="contig144">contig</a> ID followed
by an underscore, followed by a starting location (from 1), followed by
the strand ('+' or '-'), followed by the length of the region.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_protein_families" target="_blank">fids_to_protein_families</a></h3>
KBase supports a collection of <a name="protein families145">protein
families</a>
. If a fid is in one or more <a name="families146">families</a>
, that can be determined using this command. Try, <br>
<pre>  echo 'kb|g.0.peg.118' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_protein_families" target="_blank">fids_to_protein_families</a>
<br></pre>
which should produce <br>
<pre>  kb|g.0.peg.118	FIG00006632<br><br></pre>
You could use <br>
<pre>  echo 'kb|g.0.peg.118' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_protein_families" target="_blank">fids_to_protein_families</a> | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_functions" target="_blank">protein_families_to_functions</a>
<br></pre>
which should add on the function of the <a name="family147">family</a>
.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_protein_sequences" target="_blank">fids_to_protein_sequences</a></h3>
You can create either a tab-separated file in which protein sequence is
the added column, or you can produce a FASTA file containing the
translations of a set of fids. The default behavior is just to produce
the FASTA file. Try <br>
<pre>  echo 'kb|g.8776' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_fids" target="_blank">genomes_to_fids</a> | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_protein_sequences" target="_blank">fids_to_protein_sequences</a> 2&gt; /dev/null<br><br></pre>
to get the protein sequences of the genes of <em>Buchnera aphidicola</em>
str. Sg. The lines corresponding to Features that do not get translated
(e.g., RNAs) are just written to stderr. This routine is used to get
translations of fids. If you want to get the protein IDs (i.e., the <a name="md5148">md5</a> hashes that we use as keys), then you would use <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_proteins" target="_blank">fids_to_proteins</a>, which always adds a column with
the <a name="md5149">md5</a> values.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_proteins" target="_blank">fids_to_proteins</a></h3>
Protein sequences are represented by fixed-length hash values called <a name="md5150">md5</a> values. To get the <a name="md5151">md5</a>
values corresponding to a fid (or to a set of fids represented as a
column in the input table), use <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_proteins" target="_blank">fids_to_proteins</a>. Thus, <br>
<pre>  echo 'kb|g.0.peg.118' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_proteins" target="_blank">fids_to_proteins</a>
<br></pre>
would produce <br>
<pre>  kb|g.0.peg.118	211eeaac246f953b67ca2ccd0477fe3b<br><br></pre>
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_regulon_data" target="_blank">fids_to_regulon_data</a></h3>
Sets of <a name="co-regulated152">co-regulated</a> genes maintained at
<a name="RegPrecise153">RegPrecise</a> or other similar databases are
maintained. These <a name="regulons154">regulons</a> are sets of genes
having a common regulatory mechanism, but not necessarily a common
expression pattern. The notion is that of a set of genes affected by a
transcriptional regulator. The notion is certainly closely related to
that of atomic <a name="regulon155">regulon</a>
, but not identical. To see which CoregulatedSets we have in the CS,
use <br>
<pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#all_entities_CoregulatedSet" target="_blank">all_entities_CoregulatedSet</a> | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#regulons_to_fids" target="_blank">regulons_to_fids</a>
<br></pre>
To get a table showing the <a name="regulons156">regulons</a> for a
given genome, use something like <br>
<pre>  echo 'kb|g.0.peg.4288' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_regulon_data" target="_blank">fids_to_regulon_data</a> | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#regulons_to_fids" target="_blank">regulons_to_fids</a>
<br></pre>
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_roles" target="_blank">fids_to_roles</a></h3>
Each fid is thought of as implementing one or more "functional roles".
The majority implement a single role, but <a name="multifunctional genes157">multifunctional genes</a> (via the
proteins they encode) implement several. Thus, <br>
<pre>  echo 'kb|g.0.peg.133' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_roles" target="_blank">fids_to_roles</a>
<br></pre>
produces <br>
<pre>  kb|g.0.peg.133	PTS system, glucose-specific IIB component (EC 2.7.1.69)<br>  kb|g.0.peg.133	PTS system, glucose-specific IIC component (EC 2.7.1.69)<br><br></pre>
since the gene encodes two roles that often occur separately.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_subsystem_data" target="_blank">fids_to_subsystem_data</a></h3>
fids_to_subsystems_data was implemented to allow a user to extract all
of the [<a name="subsystem158">subsystem</a>
,variant,<a name="Role159">Role</a>
] tuples representing <a name="subsystems160">subsystems</a>
containing a given fid. Thus, <br>
<pre>  echo 'kb|g.0.peg.1082' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_subsystem_data" target="_blank">fids_to_subsystem_data</a>
<br></pre>
produces <br>
<pre>  kb|g.0.peg.1082	Ribosome LSU bacterial	1	LSU ribosomal protein L35p<br><br></pre>
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_subsystems" target="_blank">fids_to_subsystems</a></h3>
If you just want to get the <a name="subsystems161">subsystems</a> a
fid is part of (and you do not need the variant code and functional
role), use <br>
<pre>  echo 'kb|g.0.peg.1082' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_subsystems" target="_blank">fids_to_subsystems</a>
<br></pre>
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_contigs" target="_blank">genomes_to_contigs</a></h3>
To get the contigs (i.e., the names of the contigs) in a genome, use <br>
<pre>  echo 'kb|g.3857' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_contigs" target="_blank">genomes_to_contigs</a>
<br></pre>
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_fids" target="_blank">genomes_to_fids</a></h3>
To get the fids in a genome (constrained by type, if necessary), use <br>
<pre>  echo 'kb|g.3857' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_fids" target="_blank">genomes_to_fids</a>
<br></pre>
If you wish to restrict the extraction to fids of specific types, use
the types as arguments. Thus, <br>
<pre>  echo 'kb|g.3857' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_contigs" target="_blank">genomes_to_contigs</a> rna<br><br></pre>
would restrict output to fids of type 'rna.'
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_md5s" target="_blank">genomes_to_md5s</a></h3>
<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_md5s" target="_blank">genomes_to_md5s</a> can be used to retrieve the <a name="md5162">md5</a> hash value for any genome in KBase. Then, using
md5_to_genomes you can find the set of genomes with identical <a name="DNA163">DNA</a> sequence (but not necessarily gene calls,
assignments or function, etc.) that are in KBase. Try <br>
<pre>  echo 'kb|g.3857' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_md5s" target="_blank">genomes_to_md5s</a> |  <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#md5s_to_genomes" target="_blank">md5s_to_genomes</a>
<br></pre>
to see how this works.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_subsystems" target="_blank">genomes_to_subsystems</a></h3>
If you want to find out which <a name="subsystems164">subsystems</a>
with active variants have been associated with specific genomes, you
can use <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_subsystems" target="_blank">genomes_to_subsystems</a>. Thus, to see which <a name="subsystems165">subsystems</a> are in <i>E. coli</i>
, you might use <br>
<pre>  echo 'kb|g.0' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_subsystems" target="_blank">genomes_to_subsystems</a>
<br></pre>
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_taxonomies" target="_blank">genomes_to_taxonomies</a></h3>
Occasionally you will want to get the taxonomies for genomes. The
easiest way to do this is to use <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_taxonomies" target="_blank">genomes_to_taxonomies</a>. To see how it works, try <br>
<pre>  echo 'kb|g.0' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_taxonomies" target="_blank">genomes_to_taxonomies</a>
<br></pre>
which should produce a 2-column table with one row. The second field
would be a long string reflecting the following divisions: <br>
<pre>  kb|g.0	Bacteria: <br>		Proteobacteria: <br>		Gammaproteobacteria: <br>		Enterobacteriales: <br>		Enterobacteriaceae: <br>		Escherichia: <br>		Escherichia coli K-12<br><br></pre>
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#locations_to_dna_sequences" target="_blank">locations_to_dna_sequences</a></h3>
<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#locations_to_dna_sequences" target="_blank">locations_to_dna_sequences</a> can be used to extract <a name="DNA166">DNA</a> from a genome. A location is a sequence of
contiguous regions separated by commas. A contiguous region is
represented as a <a name="contig167">contig</a> ID followed by an
underscore, followed by a starting location (from 1), followed by the
strand ('+' or '-'), followed by the length of the region. You need to
note that the genome ID is computable from a <a name="contig168">contig</a>
ID; hence, no explicit genome ID is required. <br>
<pre>  echo 'kb|g.0.peg.118' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_locations" target="_blank">fids_to_locations</a> <br><br></pre>
produces <br>
<pre>  kb|g.0.peg.118	kb|g.0.c.1_1146590+174<br><br></pre>
Hence, kb|g.0.c.1_1146490+177 should produce the upstream 100 bp, as
well as the start codon.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#locations_to_fids" target="_blank">locations_to_fids</a></h3>
It is frequently the case that one wishes to look up the genes that
occur in a given region of a <a name="contig169">contig</a>
. Location_to_fids can be used to extract such sets of genes for each
region in the input set of regions. We define a gene as "occurring" in
a region if the location of the gene overlaps the designated region.
For example, <br>
<pre>  echo 'kb|g.0.c.1_1146490+177' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#locations_to_fids" target="_blank">locations_to_fids</a>
<br></pre>
produces <br>
<pre>  kb|g.0.c.1_1146490+177	kb|g.0.peg.118<br>  kb|g.0.c.1_1146490+177	kb|g.0.peg.77<br><br></pre>
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#md5s_to_genomes" target="_blank">md5s_to_genomes</a></h3>
Use the <a name="md5170">md5</a> hash value for a genome as a
straightforward way to determine which genomes in KBase have identical <a name="DNA171">DNA</a> sequence.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_co_occurring_families" target="_blank">protein_families_to_co_occurring_families</a></h3>
We have tabulated instances in which members of the same two <a name="protein families172">protein families</a> tend to occur close to
one another (evidence of functional coupling). Try <br>
<pre>  echo 'kb|g.0.peg.4288' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_protein_families" target="_blank">fids_to_protein_families</a> | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_co_occurring_families" target="_blank">protein_families_to_co_occurring_families</a>
<br></pre>
Note that <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_protein_families" target="_blank">fids_to_protein_families</a> added one column (the <a name="family173">family</a> containing the input fid) and <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_co_occurring_families" target="_blank">protein_families_to_co_occurring_families</a> added
three columns: the score, the function of the related <a name="family174">family</a>
, and the <a name="family175">family</a> ID.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_fids" target="_blank">protein_families_to_fids</a></h3>
When you work with a <a name="protein family176">protein family</a>
, you will often want just a list of its members. Try <br>
<pre>  echo FIG00000134 | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_fids" target="_blank">protein_families_to_fids</a>
<br></pre>
If you just wanted the size of the <a name="family177">family</a>
, you might use <br>
<pre>  echo FIG00000134 | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_fids" target="_blank">protein_families_to_fids</a> | wc<br><br></pre>
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_functions" target="_blank">protein_families_to_functions</a></h3>
It is often important to be able to get the functions of one or more <a name="protein families178">protein families</a>
. To do so, use <br>
<pre>  echo FIG00000134 | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_functions" target="_blank">protein_families_to_functions</a>
<br></pre>
which should produce <br>
<pre>  FIG00000134	Threonine synthase (EC 4.2.3.1)<br><br></pre>
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_proteins" target="_blank">protein_families_to_proteins</a></h3>
As we get more and more very similar genomes, it becomes useful to
think in terms of what unique sequences are members of a <a name="protein family179">protein family</a>
. We represent protein IDs as <a name="md5180">md5</a> hash values.
So, if you wanted to know how many unique sequences (rather than how
many members) a <a name="protein family181">protein family</a> had,
you could find out using <br>
<pre>  echo FIG00000134 | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_proteins" target="_blank">protein_families_to_proteins</a> | wc<br><br></pre>
As of the time we were writing this, <br>
<pre>  echo FIG00000134 | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_proteins" target="_blank">protein_families_to_proteins</a> | wc<br><br></pre>
produced 2543 as a count, while <br>
<pre>  echo FIG00000134 | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_fids" target="_blank">protein_families_to_fids</a> | wc<br><br></pre>
produced a count of 3903.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#proteins_to_fids" target="_blank">proteins_to_fids</a></h3>
proteins to fids allows you to take one or more proteins (represented
by <a name="md5182">md5</a> hash values) and retrieve the set of known
features that encode that protein. Thus, <br>
<pre>  echo 'kb|g.0.peg.4288' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_proteins" target="_blank">fids_to_proteins</a> | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#proteins_to_fids" target="_blank">proteins_to_fids</a>
<br></pre>
produces <br>
<pre>  kb|g.0.peg.4288  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.0.peg.4288<br>  kb|g.0.peg.4288  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.1870.peg.3069<br>  kb|g.0.peg.4288  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.2136.peg.3135<br>  kb|g.0.peg.4288  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.2590.peg.3210<br>  kb|g.0.peg.4288  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.2891.peg.4350<br>  kb|g.0.peg.4288  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.3507.peg.611<br>  kb|g.0.peg.4288  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.3558.peg.3115<br>  kb|g.0.peg.4288  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.4761.peg.5016<br>  kb|g.0.peg.4288  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.4762.peg.4707<br>  kb|g.0.peg.4288  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.842.peg.3727<br>  kb|g.0.peg.4288  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.844.peg.3184<br>  kb|g.0.peg.4288  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.9268.peg.816<br>  kb|g.0.peg.4288  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.9269.peg.415<br><br></pre>
indicating that KBase has 13 fids that encode the same protein sequence
(which has an <a name="md5183">md5</a> hash value of
3fe5ee9bcebd44132d18e2b50b8b92c3).
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#proteins_to_functions" target="_blank">proteins_to_functions</a></h3>
<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#proteins_to_functions" target="_blank">proteins_to_functions</a> can be used to gather the
functions assigned to features that have the given <a name="md5184">md5</a>
hashes. For example, <br>
<pre>  echo '3fe5ee9bcebd44132d18e2b50b8b92c3' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#proteins_to_functions" target="_blank">proteins_to_functions</a>
<br></pre>
produces <br>
<pre>  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.0.peg.4288	Threonine synthase (EC 4.2.3.1)<br>  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.1870.peg.3069	Threonine synthase (EC 4.2.3.1)<br>  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.2136.peg.3135	Threonine synthase (EC 4.2.3.1)<br>  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.2590.peg.3210	<br>  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.2891.peg.4350	Threonine synthase (EC 4.2.3.1)<br>  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.3507.peg.611	Threonine synthase (EC 4.2.3.1)<br>  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.3558.peg.3115	Threonine synthase (EC 4.2.3.1)<br>  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.842.peg.3727	Threonine synthase (EC 4.2.3.1)<br>  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.844.peg.3184	Threonine synthase (EC 4.2.3.1)<br>  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.9268.peg.816	threonine synthase<br>  3fe5ee9bcebd44132d18e2b50b8b92c3  kb|g.9269.peg.415	threonine synthase<br><br></pre>
I think that should be a bit sobering: 3 of 11 assignments are
inconsistent with the majority view, and if we reconstructed models of
metabolism for the corresponding organisms, gaps would appear in the <a name="metabolic185">metabolic</a> network, requiring manual efforts to
clean up what are obviously trivial inconsistencies.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#proteins_to_literature" target="_blank">proteins_to_literature</a></h3>
In KBase, <a name="literature186">literature</a> references are
attached to proteins, rather than to collection of Features that share
the same sequence. Thus, <br>
<pre>  echo '211eeaac246f953b67ca2ccd0477fe3b' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#proteins_to_literature" target="_blank">proteins_to_literature</a>
<br></pre>
returns at least two pubmed IDs that are relevant to the function of
proteins with the designated <a name="md5187">md5</a> hash.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#proteins_to_protein_families" target="_blank">proteins_to_protein_families</a></h3>
<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#proteins_to_protein_families" target="_blank">proteins_to_protein_families</a> allows you to find
the <a name="protein families188">protein families</a> that contain
proteins with the designated <a name="md5189">md5</a> hash values. We
anticipate that this will become particularly valuable once we have
multiple collections of <a name="protein families190">protein families</a>
accessible via KBase.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#proteins_to_roles" target="_blank">proteins_to_roles</a></h3>
<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#proteins_to_roles" target="_blank">proteins_to_roles</a> can be used to explore
inconsistencies, among other things. Try <br>
<pre>  echo '211eeaac246f953b67ca2ccd0477fe3b' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#proteins_to_roles" target="_blank">proteins_to_roles</a>
<br></pre>
and you should get <br>
<pre>  211eeaac246f953b67ca2ccd0477fe3b  50S ribosomal protein L32<br>  211eeaac246f953b67ca2ccd0477fe3b  50S ribosomal subunit protein L32<br>  211eeaac246f953b67ca2ccd0477fe3b  COG0333: Ribosomal protein L32<br>  211eeaac246f953b67ca2ccd0477fe3b  LSU ribosomal protein L32p<br>  211eeaac246f953b67ca2ccd0477fe3b  ribosomal protein L32<br><br></pre>
Note that this means that a single protein sequence is shared by
annotated fids that have succeeded in saying the same function in five
distinct ways. To successfully automate the generation of accurate
models, we will either have to make these all consistent, or we will
need to be able to rapidly determine that they are all saying the same
thing. We believe that making them consistent is the easier task (and,
furthermore, that may not require a formidable effort).
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#reactions_to_complexes" target="_blank">reactions_to_complexes</a></h3>
Reactions have IDs that are pretty unreadable. Try <br>
<pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#all_entities_Reaction" target="_blank">all_entities_Reaction</a> | head -n 5<br><br></pre>
to see some. Reactions are thought of as being implemented by
Complexes. These have pretty unreadable IDs as well. You might try <br>
<pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#all_entities_Reaction" target="_blank">all_entities_Reaction</a> | <br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#reactions_to_complexes" target="_blank">reactions_to_complexes</a> 2&gt; /dev/null | <br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#get_relationship_IsTriggeredBy" target="_blank">get_relationship_IsTriggeredBy</a> -to id | <br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#reaction_strings" target="_blank">reaction_strings</a> -c 1<br><br></pre>
This outputs a 4-column table <br>
<pre>  [reaction-id,complex-id,role,reaction-string].<br><br></pre>
There are numerous reactions which do not connect to specific
complexes, which is why we directed the stderr to /dev/null.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#reaction_strings" target="_blank">reaction_strings</a></h3>
It is often convenient to see reactions in a readable form. Use <br>
<pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#reaction_strings" target="_blank">reaction_strings</a> &lt; input.table &gt; table.with.reaction.strings.added<br><br></pre>
to get these. For example, <br>
<pre>  all_entities_REaction | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#reaction_strings" target="_blank">reaction_strings</a>
<br></pre>
will give you access to the current library of reactions in a readable
form.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#regulons_to_fids" target="_blank">regulons_to_fids</a></h3>
<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#regulons_to_fids" target="_blank">regulons_to_fids</a> is used to go from a set of <a name="regulon191">regulon</a> IDs to the fids they contain. We think
of a <a name="regulon192">regulon</a> as a set of genes impacted by a
common transcription factor. An entire <a name="regulon193">regulon</a>
may or may not be part of an "atomic <a name="regulon194">regulon</a>
". Try <br>
<pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#all_entities_CoregulatedSet" target="_blank">all_entities_CoregulatedSet</a> | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#regulons_to_fids" target="_blank">regulons_to_fids</a> | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_functions" target="_blank">fids_to_functions</a>
<br></pre>
to get a picture of what <a name="regulons195">regulons</a> have been
installed.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#roles_to_complexes" target="_blank">roles_to_complexes</a></h3>
<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#roles_to_complexes" target="_blank">roles_to_complexes</a> can be used to find out which
complexes (which often implement reactions) a role occurs in. Try <br>
<pre>  echo 'LSU ribosomal protein L24p (L26e)' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#roles_to_complexes" target="_blank">roles_to_complexes</a>
<br></pre>
You should get back <br>
<pre>  LSU ribosomal protein L24p (L26e)	0	B79EF0CC-5392-11E1-8CD2-A9B7B226DF1C<br><br></pre>
Two columns were added. The first is a boolean flag indicating whether
or not the role is an optional participant in the complex (in this case
it is not considered optional). The second is the ID of the complex.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#roles_to_fids" target="_blank">roles_to_fids</a></h3>
<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#roles_to_fids" target="_blank">roles_to_fids</a> can be used to access all of the
fids that implement a functional role. Note that the functions of the
fids need not match the role (i.e., some of the fids may be
multifunctional). Try <br>
<pre>  echo 'LSU ribosomal protein L24p (L26e)' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#roles_to_fids" target="_blank">roles_to_fids</a> | wc<br><br></pre>
to see how many of the fids in KBase implement the given role (note
that using <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#roles_to_proteins" target="_blank">roles_to_proteins</a> would give you the number of
unique protein sequences associated with the function).
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#roles_to_protein_families" target="_blank">roles_to_protein_families</a></h3>
It is occasionally convenient to be able to look up the set of <a name="protein families196">protein families</a> that implement a given
role. Note that this makes sense only if the roles and <a name="family197">family</a> functions conform to the same controlled
vocabulary. We are not completely there yet, but we are getting close.
Try <br>
<pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#all_entities_Role" target="_blank">all_entities_Role</a> | <br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#roles_to_protein_families" target="_blank">roles_to_protein_families</a> |<br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_functions" target="_blank">protein_families_to_functions</a>
<br></pre>
to get an inventory of the <a name="protein families198">protein
families</a> (and their functions) for each functional role. If you
wanted just this for just the roles that are used in model
construction, you would use <br>
<pre>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#all_roles_used_in_models" target="_blank">all_roles_used_in_models</a> | <br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#roles_to_protein_families" target="_blank">roles_to_protein_families</a> |<br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_functions" target="_blank">protein_families_to_functions</a>
<br></pre>
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#roles_to_proteins" target="_blank">roles_to_proteins</a></h3>
You can use this to find out the <a name="md5199">md5</a> hash values
of the proteins that implement this functional role. For example, try <br>
<pre>  echo 'LSU ribosomal protein L24p (L26e)' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#roles_to_proteins" target="_blank">roles_to_proteins</a>
<br></pre>
You should get something like <br>
<pre>  LSU ribosomal protein L24p (L26e)	00635cdcfd4ebeb32de8bcbe96ed693f<br>  LSU ribosomal protein L24p (L26e)	00779a04d0185ffb63d8b337896ab63a<br>  LSU ribosomal protein L24p (L26e)	00cf60187a19f321f02ad485264e1cb1<br>  .<br>  .<br>  .<br><br></pre>
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#roles_to_subsystems" target="_blank">roles_to_subsystems</a></h3>
You may wish locate the set of <a name="subsystems200">subsystems</a>
that contain a given role. Try <br>
<pre>  echo 'Threonine synthase (EC 4.2.3.1)' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#roles_to_subsystems" target="_blank">roles_to_subsystems</a>
<br></pre>
as one way to do it. You will probably get back one <a name="subsystem201">subsystem</a> that you would expect ("Threonine
and Homoserine Biosynthesis"), but you may well get back some others
that you were not expecting. These were done by individuals for a
variety of reasons. Often they were done to consistently annotate
conserved clusters of genes. In any event, look carefully at what you
get back.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#subsystems_to_fids" target="_blank">subsystems_to_fids</a></h3>
One frequently wishes to get the fids in a genome that participate in a
single <a name="subsystem202">subsystem</a>
. Try <br>
<pre>  echo 'Threonine and Homoserine Biosynthesis' | <br>  <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#subsystems_to_fids" target="_blank">subsystems_to_fids</a> "kb|g.0"<br><br></pre>
You should get back a 4-column table containing <br>
<pre>  [subsystem,genome,variant-code,fid]<br><br>  The subsystems_to_fid command takes as command-line arguments<br>  the genomes for which you wish to see fids.  If you leave off<br>  the genomes, you should get back data for all genomes in the<br>  spreadsheet.<br><br></pre>
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#subsystems_to_genomes" target="_blank">subsystems_to_genomes</a></h3>
You may find yourself asking what genomes have been placed into a given
<a name="subsystem203">subsystem</a>
. Try something like <br>
<pre>  echo 'Threonine and Homoserine Biosynthesis' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#subsystems_to_genomes" target="_blank">subsystems_to_genomes</a>
<br></pre>
as an example. You should get back a 3-column table [<a name="subsystem204">subsystem</a>
,variant-code,genome-id] You should be aware that not all genomes have
been added to specific <a name="subsystems205">subsystems</a>
. The construction and maintenance of <a name="subsystems206">subsystems</a>
is an ongoing effort.
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#subsystems_to_roles" target="_blank">subsystems_to_roles</a></h3>
Subsystems are collections of roles that are simultaneously curated. A <a name="subsystem207">subsystem</a> spreadsheet is a spreadsheet that
shows which fids in which genomes map to the <a name="subsystem208">subsystem</a>
roles. The <a name="subsystem209">subsystem</a> plus the <a name="subsystem210">subsystem</a> spreadsheet is referred to as a
populated <a name="subsystem211">subsystem</a>
. To see the roles included in a <a name="subsystem212">subsystem</a>
, try <br>
<pre>  echo 'Threonine and Homoserine Biosynthesis' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#subsystems_to_roles" target="_blank">subsystems_to_roles</a>
<br></pre>
and you should get back something like <br>
<pre>  Threonine and Homoserine Biosynthesis   Aspartokinase (EC 2.7.2.4)<br>  Threonine and Homoserine Biosynthesis   Aspartate-semialdehyde dehydrogenase (EC 1.2.1.11)<br>  Threonine and Homoserine Biosynthesis   Homoserine dehydrogenase (EC 1.1.1.3)<br>  .<br>  .<br>  .<br><br></pre>
<h3><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#subsystems_to_spreadsheets" target="_blank">subsystems_to_spreadsheets</a></h3>
Subsystems are collections of roles that are simultaneously curated. A <a name="subsystem213">subsystem</a> spreadsheet is a spreadsheet that
shows which fids in which genomes map to the <a name="subsystem214">subsystem</a>
roles. The <a name="subsystem215">subsystem</a> plus the <a name="subsystem216">subsystem</a> spreadsheet is referred to as a
populated <a name="subsystem217">subsystem</a>
. To see the <a name="subsystem218">subsystem</a> spreadsheet
associated with a given <a name="subsystem219">subsystem</a>
, try <br>
<pre>  echo 'Threonine and Homoserine Biosynthesis' | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#subsystems_to_spreadsheets" target="_blank">subsystems_to_spreadsheets</a> 'kb|g.0'<br><br></pre>
You should get back a 5-column table <br>
<pre>  [subsystem,genome,variant-code,role,fid]<br><br></pre>
<h2>Summary</h2>
We have attempted to introduce you the <a name="Entity-Relationship Model220">Entity-Relationship Model</a> used
to construct the CS and the command-line tools that you can use to
expose the data in it. Of course, we do offer an <a name="API221">API</a>
to support programmatic access, and a number of user interfaces (UIs)
are being constructed to support access. However, for now we suggest
that you may wish to use these basic tools to extract data.
