<h1>Advanced Methods for Retrieving Data</h1>
<p><strong>Purpose: </strong>To learn the <a name="CDM222">CDM</a> <a href="http://pubseed.theseed.org/ErdbDocWidget.cgi?xmlFileName=/home/parrello/CdmiData/Published/KSaplingDBD.xml" name="Entity-Relationship Model223">Entity-Relationship Model</a> and
  how to run command-line tools to expose data. </p>
<p><strong>Required Prerequisite Activities: </strong><a href="/for-users/tutorials/installing-the-kbase-app-dmg/">Download KBase: Installing the KBase Application DMG</a>, OR <a href="/for-users/tutorials/navigating-iris/">Get Started with Iris: Navigating Iris</a> </p>
<p><strong>Suggested Prerequisite Activities:</strong> <a href="/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/">Some
  Basic Exercises Using KBase</a>. </p>
<p><strong>Related Tutorials:</strong> <a href="/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/">Extending
  the CS Commands with Operators</a> </p>
<h2>Introduction</h2>
The <a name="Central Store224">Central Store</a> (<b>CS</b>) is the KBase integration of the data needed to support the creation
and validation of <a name="metabolic225">metabolic</a> and regulatory
models. It will certainly be used for many other purposes, but its
creation is being driven by the needs of the modelling community. <br>
<br>
We have described many details of how to access the contents of the CS
via command-line tools already in <a href="/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/">Some
Basic Exercises Using KBase</a>. This document is designed to complement that tutorial. We hope to
explain an overview of the approach of piping KBase tools together on
the command-line, as well as some very minimal notes relating to a
subset of the Unix tools. <br>
<br>
To understand the contents of either tutorial you really do need to be
able to bring up the entity-relationship model describing the contents
of the CS and understand the contents of that model, which is often
called the <a name="Central Data Model226">Central Data Model</a> (<b><a name="CDM227">CDM</a></b>). See the start of the companion tutorial to get an overview of the <a name="CDM228">CDM</a> and how to get started. <br>
<br>
We realize that most users will utlimately use user interfaces that
obviate the need to do anything at the command-line. We look forward to
that day. Until then, there is a great deal of use that can be made of
the CS in its present form, using the rather primitive environment
supported by the command-line.
<h2>The Basic Philosophy of the CS Command-line Tools</h2>
We think of most of the KBase command-line tools as taking in a file
containing a tab-separated table and outputting a modified table. The
most common modification is the addition of one or more columns. We
create "pipelines" of these tools to implement fairly complex
transformations leading to the final table containing the desired
output. For example, consider the following little <a name="pipeline229">pipeline</a> : <br>
<pre>  <a href="/developer-zone/documentation/cdm-entity-relationship-command-scripts/#all_entities_Genome" target="_blank">all_entities_Genome</a> -f scientific_name | grep "Streptococcus pneumoniae"<br>
</pre>
The <i><a href="/developer-zone/documentation/cdm-entity-relationship-command-scripts/#all_entities_Genome" target="_blank">all_entities_Genome</a></i> command is thought of as
producing a table in which the first column (by default) is the genome
ID, and any extra columns come from the arguments of the command. In
this case, we get a 2-column table <br>
<pre>  [genome-ID,genome-name]<br>
</pre>
This 2-column table is fed into a Unix command called <i><a name="grep230">grep</a></i> , which keeps lines that "match" its argument. In this case, the <i><a name="grep231">grep</a></i> extracts rows in the table that "match"
the string "Streptococcus pneumoniae." Thus, we get as a result a
2-column table in which each row contains "Streptococcus pneumoniae."
Upon running the command, we got <br>
<pre>  kb|g.1340	Streptococcus pneumoniae SP19-BS75<br>  kb|g.1880	Streptococcus pneumoniae BS457<br>  kb|g.3485	Streptococcus pneumoniae SPN7465<br>  kb|g.9772	Streptococcus pneumoniae SP18-BS74<br>  kb|g.3478	Streptococcus pneumoniae SPN034183<br>  kb|g.1784	Streptococcus pneumoniae JJA<br>  kb|g.9944	Streptococcus pneumoniae CDC1873-00<br>  kb|g.3474	Streptococcus pneumoniae OXC141<br>  kb|g.3484	Streptococcus pneumoniae SPN033038<br>  kb|g.1881	Streptococcus pneumoniae BS458<br>  kb|g.110	Streptococcus pneumoniae OXC141<br>  kb|g.1334	Streptococcus pneumoniae SP3-BS71<br>  .<br>  .<br>  .<br>
</pre>
Transforming tables by extracting rows that contain a given string (or,
more generally, a pattern), extracting columns from a table, or sorting
the rows in a table is the basic style we advocate. For example,
suppose that you wanted to find all features of <i>Streptococcus
pneumoniae</i> that had been assigned a specific function (say, <i>triosephosphate
isomerase</i> ). You might try using (as one long command) <br>
<pre>
<a href="/developer-zone/documentation/cdm-entity-relationship-command-scripts/#all_entities_Genome" target="_blank">all_entities_Genome</a> -f scientific_name |<br>  grep "Streptococcus pneumoniae" |<br>  <a href="/developer-zone/documentation/cdm-command-line-scripts/#genomes_to_fids" target="_blank">genomes_to_fids</a> CDS -c 1 | <br>  <a href="/developer-zone/documentation/cdm-command-line-scripts/#fids_to_functions" target="_blank">fids_to_functions</a> | <br>  grep -i "triosephosphate isomerase"<br>
</pre>
This command may take a few minutes.  You can issue other commands while waiting for this command to finish.  An alternate way to do the above, is to break it into a series of intermediate steps.
<pre>
<a href="/developer-zone/documentation/cdm-entity-relationship-command-scripts/#all_entities_Genome" target="_blank">all_entities_Genome</a> -f scientific_name |  grep "Streptococcus pneumoniae" > Strep_genomes
cat Strep_genomes | <a href="/developer-zone/documentation/cdm-command-line-scripts/#genomes_to_fids" target="_blank">genomes_to_fids</a> CDS -c 1 > Strep_genomes_with_fids 
cat Strep_genomes_with_fids | <a href="/developer-zone/documentation/cdm-command-line-scripts/#fids_to_functions" target="_blank">fids_to_functions</a> > Strep_genomes_with_functions
grep -i "triosephosphate isomerase" Strep_genomes_with_functions<br>
</pre>
Either of these approaches should produce output like <br>
<pre>  kb|g.1340  Streptococcus pneumoniae SP19-BS75  kb|g.1340.peg.783   Triosephosphate isomerase (EC 5.3.1.1)<br>  kb|g.9772  Streptococcus pneumoniae SP18-BS74  kb|g.9772.peg.1261  triosephosphate isomerase<br>  kb|g.9772  Streptococcus pneumoniae SP18-BS74  kb|g.9772.peg.1663  triosephosphate isomerase<br>  kb|g.9772  Streptococcus pneumoniae SP18-BS74  kb|g.9772.peg.2175  triosephosphate isomerase<br>  kb|g.9772  Streptococcus pneumoniae SP18-BS74  kb|g.9772.peg.2207  triosephosphate isomerase<br>  kb|g.3478  Streptococcus pneumoniae SPN034183  kb|g.3478.peg.287   Triosephosphate isomerase (EC 5.3.1.1)<br>  .<br>  .<br>  .<br>
</pre>
Here, some extra comments are necessary.
<ol>
  <li><i><a href="/developer-zone/documentation/cdm-command-line-scripts/#genomes_to_fids" target="_blank">genomes_to_fids</a></i> is a KBase command that takes
    a tab-separated table as input, and one of the columns in the table
    must contain KBase genome IDs. By default, that column is the last
    column in the table (i.e., the rightmost). If the column of genome IDs
    is not the last, use the "-c N" argument to say that the column is the
    Nth. </li>
  <li><i><a href="/developer-zone/documentation/cdm-command-line-scripts/#fids_to_functions" target="_blank">fids_to_functions</a></i> is a KBase command that
    takes a table containing a column of feature ids (i.e., fids), which by
    default is the last column, and extends the table with one more column,
    the function assigned to the feature. </li>
  <li> Finally, the "-i" argument to <i><a name="grep232">grep</a></i> makes the match case insensitive. <i><a name="grep233">grep</a></i> is
    an extremely powerful command with a number of useful options (you can
    select all rows that do not match, rows that match some specified
    pattern, etc.) </li>
</ol>
<h2> A Short Exercise</h2>
Now, as an exercise, try to show the IDs and functions of all fids that
have precisely the same function as <i>kb|g.9772.<a name="peg234">peg</a>.2175</i>. <br>
<br>
We would likely use the following <a name="pipeline235">pipeline</a> : <br>
<pre>  echo 'kb|g.9772.peg.2175' | <a href="/developer-zone/documentation/cdm-command-line-scripts/#fids_to_proteins" target="_blank">fids_to_proteins</a> | <a href="/developer-zone/documentation/cdm-command-line-scripts/#proteins_to_fids" target="_blank">proteins_to_fids</a> | <a href="/developer-zone/documentation/cdm-command-line-scripts/#fids_to_functions" target="_blank">fids_to_functions</a>
<br>
</pre>
In some sense, this is probably not terribly efficient, but you do get
the answer back in a few seconds (and, we think that the answer itself
poses at least one interesting question). <br>
<br>
<h2>The Abstract View</h2>
A <a name="pipeline236">pipeline</a> begins with a <i><a name="generator237">generator</a></i> -- that is, a command that takes
no input, but outputs a table. <i>Generators</i> come in two flavors:
one generates a table for every instance of a designated entity, and
the other a single instance of an entity. For example, <br>
<pre>  <a href="/developer-zone/documentation/cdm-entity-relationship-command-scripts/#all_entities_Genome" target="_blank">all_entities_Genome</a> <br>
</pre>
is a <a name="generator238">generator</a> that outputs a single column
containing the IDs of all genomes in KBase. Similarly, <br>
<pre>  <a href="/developer-zone/documentation/cdm-entity-relationship-command-scripts/#all_entities_Genome" target="_blank">all_entities_Genome</a> -f 'dna_size,contigs,scientific_name'<br>
</pre>
is a <a name="generator239">generator</a> that would output a 4-column
table. (Try it.) The <i>all_entities_ENTITY</i> commands are <a name="generators240">generators</a> that you may depend on being
present (i.e., if you see the entity in the ER-model, the corresponding <a name="generator241">generator</a> may be assumed to be present). <br>
<pre>  echo 'kb|g.9772.peg.2175'<br>
</pre>
was an example of a <a name="generator242">generator</a> that produced
a single instance of an entity. Now, it is true that you might begin a <a name="pipeline243">pipeline</a> wihout a <a name="generator244">generator</a> , using something like <br>
<pre>  <a href="/developer-zone/documentation/cdm-command-line-scripts/#fids_to_proteins" target="_blank">fids_to_proteins</a> &lt; file.of.saved.fid.ids<br>
</pre>
<p>In this case, you might consider this a "restarted <a name="pipeline245">pipeline</a>", or you might amend our previous
  assertion that pipelines begin with <a name="generators246">generators</a>.
  In either case, if you wish to debate the issue, we claim to have
  succeeded in clarifying the concepts. <br>
  <br>
  There is one more type of <a name="generator247">generator</a> that
  you will occasionally find extremely useful. Suppose that you wished to
  find all of the Features (fids) that have been assigned the function <br>
  <br>
</p>
<pre>  SSU ribosomal protein S9p (S16e)<br>
</pre>
Using the tools we have described so far, this is not easy to do (and
the results are quite inefficient). To search for instances of an
Entity that have a desired value of one of the fields, you can use
commands like <br>
<br>
<pre>  <a href="/developer-zone/documentation/cdm-entity-relationship-command-scripts/#query_entity_Feature" target="_blank">query_entity_Feature</a> -is 'function,SSU ribosomal protein S9p (S16e)'<br><br>  <a href="/developer-zone/documentation/cdm-entity-relationship-command-scripts/#query_entity_Feature" target="_blank">query_entity_Feature</a> -like 'function,SSU ribosomal protein%'<br>
</pre>
The first command looks up the fids for features that have been
assigned the given function, while the second locates all SSDU ribsomal
proteins. <br>
<p><br>
  There are two classes of KBase <b>transformation commands</b> that you
  may depend on being present. These transformation commands take as
  input a table and add columns to it: </p>
<ol>
  <li><b>get_entity_ENTITY</b> commands are used to take as input a
    table that contains a column containing <b>ENTITY</b> IDs. These
    commands are used to add columns corresponding to fields from the
    referenced <b>ENTITY</b>. Thus, <br>
    <pre>  echo 'kb|g.9772.peg.2175' | <a href="/developer-zone/documentation/cdm-entity-relationship-command-scripts/#get_entity_Feature" target="_blank">get_entity_Feature</a> -f 'function,source_id'<br>
</pre>
    is an example <i><a href="/developer-zone/documentation/cdm-entity-relationship-command-scripts/#get_entity_Feature" target="_blank">get_entity_Feature</a></i> that adds two columns to
    the (admittedly limited) input stream. <br>
    <br>
  </li>
  <li> The <b>get_relationship_RELATIONSHIP</b> commands are used for
    "crossing" a relationship from one type of entity (called the <i>from
    entity</i>) to another (called the <i>to entity</i>). The input must
    be a table with a column containing IDs of the <i>from entity</i>. The
    output is formed by tacking on columns of data from three sources: the <i>from
    entity</i>, the relationship, and the <i>to entity</i>. For example, <br>
    <pre>  echo 'kb|g.9772.peg.2175' | <a href="/developer-zone/documentation/cdm-entity-relationship-command-scripts/#get_relationship_IsOwnedBy" target="_blank">get_relationship_IsOwnedBy</a> -from source_id -to 'scientific_name,id'<br>
</pre>
    takes the input IDs of Features, crosses the IsOwnedBy relationship,
    and adds three columns (one from the <i>from entity</i> and two from
    the <i>to entity</i>). </li>
</ol>
<br>
We have supplemented the standard <a name="generators248">generators</a> (all_entity_ENTITY routines) and the standard transformation commands
(get_entity_ENTITY and get_relationship_RELATIONSHIP) with a set of
commands representing what we call "well-trodden paths." When we see
these recurring patterns of use, sometimes we can reduce the requirred
effort to extract the desired data. Thus, <br>
<pre>  <a href="/developer-zone/documentation/cdm-command-line-scripts/#all_roles_used_in_models" target="_blank">all_roles_used_in_models</a>
<br>
</pre>
is a customized <a name="generator249">generator</a> that outputs a
table containing roles that are used in building models. If you wanted
to get the number of unique protein sequences that connect to these
roles, you might use <br>
<pre>  <a href="/developer-zone/documentation/cdm-command-line-scripts/#all_roles_used_in_models" target="_blank">all_roles_used_in_models</a> | <a href="/developer-zone/documentation/cdm-command-line-scripts/#roles_to_proteins" target="_blank">roles_to_proteins</a> | cut -f 2 | sort -u | wc<br>
</pre>
<p>(Note that the command above may take between 10 to 20 minutes. If
  preferred, route the output to a file.) </p>
<p>The <i><a href="/developer-zone/documentation/cdm-command-line-scripts/#roles_to_proteins" target="_blank">roles_to_proteins</a></i> transformation command is an
  example of one of these commands added to transformation commands. The
  following Unix commands are worth noting: </p>
<ul>
  <li><b><a name="cut250">cut</a> -f 2</b> says "extract just the
    second column" (the one containing <a name="md5251">md5</a> values
    representing protein sequences). </li>
  <li><b><a name="sort252">sort</a> -u</b> says "<a name="sort253">sort</a> the input table removing duplicate lines", and </li>
  <li><b><a name="wc254">wc</a></b> says "count the lines, word, and
    characters in the input file" </li>
</ul>
<br>
Well, that is basically it. We end by attempting to convey just one
picky detail. The <a name="sort255">sort</a> command allows you to <a name="sort256">sort</a> lines on specific fields using the <b>-k N</b> argument. The problem is that it breaks a line into fields using
transitions between whitespace characters and non-whitespace
characters, and this is not the behavior you always want. If you are
dealing with tab-delimited fields (as we are), then you want the <i><a name="sort257">sort</a></i> command to split the line into fields
properly. This can be done by executing <br>
<pre>  export TAB=`echo -e "\t"`<br>
</pre>
and then using something like <br>
<pre>  sort -t "$TAB" -k 4 <br>
</pre>
to <a name="sort258">sort</a> on the 4th field in a tab-delimited
table. Within our KBase exercises, you will usually need to worry about
this only if you have "role" or "function" columns in your tables (they
have embedded whitespace). 