
<h1>Towards a Controlled Vocabulary Part 3: Making the MOL Translation
to the Abstract Function Vocabulary</h1>
<p><strong>Purpose: </strong> Use KBase to map MicrobesOnLine (MOL)
features to
the abstract vocabulary. </p>
<p><strong>Required Prerequisite Activities: </strong> <a href="/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/">Getting
Started with KBase</a> </p>
<p> <strong>Suggested Prerequisite Activities: </strong> <a href="/developer-zone/tutorials/iris/some-basic-exercises-using-the-kbase-iris-interface/">Some
Basic Exercises using the KBase Iris Interface </a> and other Iris
tutorials. </p>
<p><strong>Related Tutorials:</strong> <a href="/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/">Mapping
Functions Between Sources of Annotations: Take 2 </a> and other
Command Line tutorials. </p>
<h1><strong>Introduction</strong></h1>
<p>We have described how to define an abstract vocabulary of function
using exemplars in <a href="/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/" l="l" name="Exemplars492">Part 1: Defining Exemplars</a>. Next, we
described how to generate the <a name="SEED493">SEED</a> mapping from <a name="SEED494">SEED</a> IDs to the abstract vocabulary imposed by
exemplars <a href="/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/" name="Exemplars495">Part 2: Mapping to Exemplars</a>. We complete this
discussion of how to reach consistent <a name="annotations496">annotations</a>
in an abstract vocabulary by discussing how one might map <a name="MicrobesOnLine497">MicrobesOnLine</a> (MOL) features to the
abstract vocabulary. <br>
<br>
The overall strategy follows two main steps, as explained in the
sections below.
</p>
<h2>Begin with the SEED Translation</h2>
<p>Begin with the <a name="SEED498">SEED</a> translation, represented
as 4-tuples of the form <br>
</p>
<pre>           [source,source-id,fid,exemplar]<br><br><br>  </pre>
where the <i>source</i> is <i><a name="SEED499">SEED</a></i>. Here,
the <i>fid</i> field is the KBase ID of the <a name="SEED500">SEED</a>
<i>source_id</i>. We begin by adding a column containing the <a name="md5501">md5</a> values using <br>
<pre>           <a href="/developer-zone/documentation/cdm-command-line-scripts/#fids_to_proteins" target="_blank">fids_to_proteins</a> -c 3 &lt; seed.translation.table &gt; seed.translation.with.md5<br><br><br><br>  </pre>
and we <a name="sort502">sort</a> them on the <a name="md5503">md5</a>
value using <br>
<pre>            sort -u -k 5 seed.translation.with.md5 &gt; sorted.seed.with.md5<br><br> </pre>
<h2>Create a Table of Data Relating to the MOL Genomes</h2>
<p>Then, we create a table of data relating to the MOL genomes using <br>
</p>
<pre>            echo MOL | <br>	    <a href="/documentation/cdm-entity-relationship-command-scripts/#get_relationship_Submitted" target="_blank">get_relationship_Submitted</a> -to id | <br>	    <a href="/developer-zone/documentation/cdm-command-line-scripts/#genomes_to_fids" target="_blank">genomes_to_fids</a> peg | <br>	    cut -f 3 | <br>	    <a href="/developer-zone/documentation/cdm-entity-relationship-command-scripts/#get_entity_Feature" target="_blank">get_entity_Feature</a> -f source_id |<br>	    <a href="/developer-zone/documentation/cdm-command-line-scripts/#fids_to_proteins" target="_blank">fids_to_proteins</a> -c 1 2&gt; /dev/null | <br>	    <a href="/developer-zone/documentation/cdm-command-line-scripts/#fids_to_functions" target="_blank">fids_to_functions</a> -c 1 |<br>	    sort -k 3 &gt; MOL.fid.source_id.md5.function<br><br></pre>
<p>Now, we have two fairly large files, <i>seed.translation.with.<a name="md5504">md5</a></i> and <i>MOL.fid.<a name="md5505">md5</a>.function</i>,
both sorted on <a name="md5506">md5</a> values. We can now make a
single pass through the two files, compiling the data we will need to
construct an initial translation file for MOL. Essentially, as we pass
through the two files, we detect cases in which an MOL fid had an
identical <a name="md5507">md5</a> to a <a name="SEED508">SEED</a>
fid; we use this to take the corresponding exemplar and the MOL
function and increment a count of the times this function occurred with
that exemplar. After the pass, we simply take, for each exemplar the
string that MOL most often used to describe the abstract function, and
we write out a file of inconsistent sets to support MOL in detecting
and correcting inconsistencies. This is accomplished by the following
short perl program:
</p>
<h2>A Short Perl Program supporting MOL in Correcting Inconsistencies<br>
</h2>
<pre>#<br># make_MOL_translation.pl<br>#<br>open(IN1,"&lt;","MOL.fid.source_id.md5.function") || die "could not open MOL.fid.source_id.md5.function";<br>open(IN2,"&lt;","sorted.seed.with.md5")           || die "could not open sorted.seed.with.md5";<br><br>my %counts;<br>my $row1 = &lt;IN1&gt;;<br>my $row2 = &lt;IN2&gt;;<br>my $n=1;<br>while ($row1 &amp;&amp; $row2)<br>{<br>    my($fid_MOL,$source_id_MOL,$md5_MOL,$function_MOL) <br>	= ($row1 =~ /^(\S+)\t(\S+)\t(\S+)\t(.*)/);<br>    my($source_id_SEED,$exemplar,$md5_SEED) <br>	= ($row2 =~ /^\S+\t(\S+)\t\S+\t(\S+)\t(\S+)/);<br><br>    if (($md5_MOL lt $md5_SEED) || &amp;hypo($function_MOL))<br>    {<br>	$row1 = &lt;IN1&gt;;<br>    }<br>    elsif ($md5_MOL gt $md5_SEED)<br>    {<br>	$row2 = &lt;IN2&gt;;<br>    }<br>    else<br>    {<br>	$counts{$exemplar}-&gt;{$function_MOL}++;<br>	$row1 = &lt;IN1&gt;;<br>    }<br>}<br>close(IN1);<br>close(IN2);<br><br>foreach my $exemplar (keys(%counts))<br>{<br>    my @funcs = sort { $counts{$exemplar}-&gt;{$b} &lt;=&gt; $counts{$exemplar}-&gt;{$a} } keys(%{$counts{$exemplar}});<br>    print join("\t",($exemplar,$funcs[0])),"\n";<br>}<br><br># a modest attempt to catch most hypothetical roles<br>#<br>sub hypo {<br>    my ($func) = @_;<br>    if (! $func)                             { return 1 }<br>    $func =~ s/\s*\#.*$//;<br>    if ($func =~ /lmo\d+ protein/i)          { return 1 }<br>    if ($func =~ /hypoth/i)                  { return 1 }<br>    if ($func =~ /conserved protein/i)       { return 1 }<br>    if ($func =~ /gene product/i)            { return 1 }<br>    if ($func =~ /interpro/i)                { return 1 }<br>    if ($func =~ /B[sl][lr]\d/i)             { return 1 }<br>    if ($func =~ /^U\d/)                     { return 1 }<br>    if ($func =~ /^orf[^_]/i)                { return 1 }<br>    if ($func =~ /uncharacterized/i)         { return 1 }<br>    if ($func =~ /pseudogene/i)              { return 1 }<br>    if ($func =~ /^predicted/i)              { return 1 }<br>    if ($func =~ /AGR_/)                     { return 1 }<br>    if ($func =~ /similar to/i)              { return 1 }<br>    if ($func =~ /similarity/i)              { return 1 }<br>    if ($func =~ /glimmer/i)                 { return 1 }<br>    if ($func =~ /unknown/i)                 { return 1 }<br>    if (($func =~ /domain/i) ||<br>        ($func =~ /^y[a-z]{2,4}\b/i) ||<br>        ($func =~ /complete/i) ||<br>        ($func =~ /ensang/i) ||<br>        ($func =~ /unnamed/i) ||<br>        ($func =~ /EG:/) ||<br>        ($func =~ /orf\d+/i) ||<br>        ($func =~ /RIKEN/) ||<br>        ($func =~ /Expressed/i) ||<br>        ($func =~ /[a-zA-Z]{2,3}\|/) ||<br>        ($func =~ /predicted by Psort/) ||<br>        ($func =~ /^bh\d+/i) ||<br>        ($func =~ /cds_/i) ||<br>        ($func =~ /^[a-z]{2,3}\d+[^:\+\-0-9]/i) ||<br>        ($func =~ /similar to/i) ||<br>        ($func =~ / identi/i) ||<br>        ($func =~ /\bputative\b/i) ||<br>        ($func =~ /ortholog of/i) ||<br>        ($func =~ /structural feature/i))    { return 1 }<br>    return 0;<br>}<br><br><br></pre>
The program outputs a set of 2-tuples: <br>
<pre>           [exemplar,MOL-function]<br><br></pre>
On our initial uses of the program approximately 80% of the functional
roles used in modelling were included in the initial MOL tranlation
table. <br>
<br>
We have attempted to weed out hypothetical assignments from
consideration. We make no attempt to identify inconsistencies in the
MOL assignments. All we give is a table that maps the abstract roles
(exemplars) to an MOL function that is used at least as often as any
other to represent the role.
