<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html><head>
  <b><title>How to find KBase IDs of genes and genomes</title>
  
  </b>
</head><body>
<h1>How to find KBase IDs of genes and genomes<br>
</h1>

<strong>Purpose:</strong> Learn how to find the KBase IDs of genes and genomes<b>.<br>
Required Prerequisite Activities: </b> Getting What You Need from the
CS Using Command-Line Scripts
<b><br>
Suggested Prerequisite Activities: </b>None <b><br>
Related Tutorials: </b> None
<h1> Introduction </h1>

<p>Among other things, the KBase Central Store (CS) contains data on genomes,
genes they encode, and their projected functional roles. <br>
<br>
  This tutorial will teach you:</p>
<ol>

  <li> How to find the <em>KBase ID</em> of a genome.</li>
  <li>How to find the list of genes encoded in this genome.<br>
  </li>

  <li> How to find the <em>KBase ID</em> of a specific gene in this genome by its name or projected functional role.<br>
</li>
</ol>
<h2>Find KBase ID of a genome<br>
 </h2>
The command <code>all_entities_genome</code> lists all genomes currently included in the KBase. To find KBase ID for<em> Escherichia coli </em>K12 genome one can use the
command: <br>
<br>
<pre>all_entities_Genome -f scientific_name |
grep Escherichia </pre>

Among all <em>E. coli </em>strains listed in the output of this command one spots <br>
the KBase ID for <em>Escherichia coli </em>K12.
Appropriately it is&nbsp; 'kb|g.0'. 
<h2>List and count all protein-coding genes in a user-specified genome<br>
</h2>
Annotated genomes in the KBase contains a number of identified features with unique KBase IDs (fids).<br>
The type of feature is encoded in its KBase ID name including:<br>

<br>

<em>g</em> &nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;Genome<br>

<em>mg &nbsp;</em>&nbsp; &nbsp;metaGenome<br>

<em>c</em> &nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;Contig<br>

<em>CDS </em> &nbsp;CDS<br>

<em>peg</em> &nbsp;&nbsp; Protein encoding gene (being phased out to use CDS instead)<br>

<em>rna </em>&nbsp;&nbsp; RNA <br>

<br>
To count the number of protein coding genes (identified by "peg" in their KBase ID) <br>
in<em> E. coli </em>K12 genome (KBase ID "kb|g.0")&nbsp; one can use:<br>
<pre>
echo 'kb|g.0' | genomes_to_fids | grep 'peg'&nbsp; | wc -l<br></pre>
To find all protein coding genes in <em>E. coli </em>K12 genome along with a brief description <br>
of their predicted functions one adds <code>fid_to_function</code> command to this pipeline.<br>



<pre>
echo 'kb|g.0' |
genomes_to_fids | grep 'peg' | fids_to_functions<br>
</pre>
<h2>Find the KBase ID of a gene in a genome<br>
</h2>

To find the gene encoding the RecA protein one can use:<br>
<pre>
echo 'kb|g.0' | genomes_to_fids | grep 'peg' |
fids_to_functions | grep 'RecA'</pre>

The KBase ID of the RecA gene is &#8216;kb|g.0.peg.1356&#8217;<br>

<br>

Another example is a bacterium <em>Shewanella oneidensis</em> MR-1, with KBase
ID:
&#8220;kb|g.20848&#8221;.&nbsp; <br>
<br>
To find the KBase ID of a gene encoding isocitrate lyase, one runs<br>
<pre>
echo 'kb|g.20848' |
genomes_to_fids | fids_to_functions | grep
"isocitrate lyase"
</pre>

which lists it as "kb|g.20848.CDS.1671".
<br>
</body></html>