<?xml version="1.0" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
 <head>
  <title>
  </title>
  <meta content="text/html; charset=utf-8" http-equiv="content-type"/>
  <link href="mailto:root@ubuntu.chicago.kbase.us" rev="made"/>
 </head>
 <body style="background-color: white">
  <h1 id="NAME">
   NAME
  </h1>
  <p>
   Bio::KBase::PlantExpressionService::Client
  </p>
  <h1 id="DESCRIPTION">
   DESCRIPTION
  </h1>
  <p>
   Plant Expression Service APIs
  </p>
  <pre><code> This module provides services for plant expression data in support of the coexpression
 network and ontology driven data needs of the plant sciences community. This version of
 the modules supports retrieval of the following information:
 1. Retrieval of GEO sample ID list for given EO (environmental ontology) and/or PO (plant ontology -plant tissues/organs of interest).
 2. Retrieval of the expression values for given GEO sample ID list.  
 3. For given expression values tables, it computes co-expression clusters or network (CLI only).</code></pre>
  <p>
   It will serve queries for tissue and condition specific gene expression co-expression network for biologically interesting genes/samples. Users can search differentially expressed genes in different tissues or in numerous experimental conditions or treatments (e.g various biotic or abiotic stresses). Currently the metadata annotation is provided for a subset of gene expression experiments from the NCBI GEO microarray experiments for Arabidopsis and Poplar. The samples of these experiments are manually annotated using plant ontology (PO) [http://www.plantontology.org/] and environment ontology (EO) [http://obo.cvs.sourceforge.net/viewvc/obo/obo/ontology/phenotype/environment/environment_ontology.obo]
  </p>
  <h2 id="get_repid_by_sampleid">
   get_repid_by_sampleid
  </h2>
  <pre><code>  $results = $obj-&gt;get_repid_by_sampleid($ids)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$ids is a SampleIDList
$results is a Sample
SampleIDList is a reference to a list where each element is a SampleID
SampleID is a string
Sample is a reference to a hash where the key is a SampleID and the value is a ValueList
ValueList is a reference to a list where each element is a float</pre>
   </dd>
   <dt id="Description">
    Description
   </dt>
   <dd>
    <p>
     This function takes a list a GSM id. The GSM id can be stored as a csv file, containing one line. The output is the corresponding replicate id
    </p>
   </dd>
  </dl>
  <h2 id="get_experiments_by_seriesid">
   get_experiments_by_seriesid
  </h2>
  <pre><code>  $results = $obj-&gt;get_experiments_by_seriesid($ids)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types1">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$ids is a SeriesIDList
$results is an Experiments
SeriesIDList is a reference to a list where each element is a SeriesID
SeriesID is a string
Experiments is a reference to a hash where the key is a SeriesID and the value is an Experiment
Experiment is a reference to a hash where the following keys are defined:
	series has a value which is a Sample
	genes has a value which is a GeneIDList
Sample is a reference to a hash where the key is a SampleID and the value is a ValueList
SampleID is a string
ValueList is a reference to a list where each element is a float
GeneIDList is a reference to a list where each element is a GeneID
GeneID is a string</pre>
   </dd>
   <dt id="Description1">
    Description
   </dt>
   <dd>
    <p>
     This function provides the expression data for each experiment corresponding to the given list of series (GSE#s). It first retrieves the experiments sample ids (GSM#s) for each series and subsequently, it extracts the expressed genes and their corresponding expression values for each experiment. It then returns a table of data containing GSE#, GSM#, Expressed Gene ID, and Expression Value.
    </p>
   </dd>
  </dl>
  <h2 id="get_experiments_by_sampleid">
   get_experiments_by_sampleid
  </h2>
  <pre><code>  $results = $obj-&gt;get_experiments_by_sampleid($ids)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types2">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$ids is a SampleIDList
$results is an Experiment
SampleIDList is a reference to a list where each element is a SampleID
SampleID is a string
Experiment is a reference to a hash where the following keys are defined:
	series has a value which is a Sample
	genes has a value which is a GeneIDList
Sample is a reference to a hash where the key is a SampleID and the value is a ValueList
ValueList is a reference to a list where each element is a float
GeneIDList is a reference to a list where each element is a GeneID
GeneID is a string</pre>
   </dd>
   <dt id="Description2">
    Description
   </dt>
   <dd>
    <p>
     This function provides the expression values corresponding to each given experiment sample in the input list of sample ids (GSM#s).For each sample in the input list of samples, it extracts the expressed genes (kbase gene identifier) and their corresponding expression values.
    </p>
   </dd>
  </dl>
  <h2 id="get_experiments_by_sampleid_geneid">
   get_experiments_by_sampleid_geneid
  </h2>
  <pre><code>  $results = $obj-&gt;get_experiments_by_sampleid_geneid($ids, $gl)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types3">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$ids is a SampleIDList
$gl is a GeneIDList
$results is an Experiment
SampleIDList is a reference to a list where each element is a SampleID
SampleID is a string
GeneIDList is a reference to a list where each element is a GeneID
GeneID is a string
Experiment is a reference to a hash where the following keys are defined:
	series has a value which is a Sample
	genes has a value which is a GeneIDList
Sample is a reference to a hash where the key is a SampleID and the value is a ValueList
ValueList is a reference to a list where each element is a float</pre>
   </dd>
   <dt id="Description3">
    Description
   </dt>
   <dd>
    <p>
     This function provides the expression values corresponding to the given sample and for given list of kbase gene identifiers. Retrieve the expression values corresponding to each given sample in the input list of samples ((typically NCBI GEO series sample ids: GSM#s) for given list of genes (kbase identifier).
    </p>
   </dd>
  </dl>
  <h2 id="get_eo_sampleidlist">
   get_eo_sampleidlist
  </h2>
  <pre><code>  $results = $obj-&gt;get_eo_sampleidlist($lst)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types4">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$lst is an EOIDList
$results is an EOID2Sample
EOIDList is a reference to a list where each element is an EOID
EOID is a string
EOID2Sample is a reference to a hash where the key is an EOID and the value is a SampleIDList
SampleIDList is a reference to a list where each element is a SampleID
SampleID is a string</pre>
   </dd>
   <dt id="Description4">
    Description
   </dt>
   <dd>
    <p>
     Retrieve the list of expression samples (GSM#s) that correspond to one or more of the environmental conditions (EO) of interest.
    </p>
   </dd>
  </dl>
  <h2 id="get_po_sampleidlist">
   get_po_sampleidlist
  </h2>
  <pre><code>  $results = $obj-&gt;get_po_sampleidlist($lst)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types5">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$lst is a POIDList
$results is a POID2Sample
POIDList is a reference to a list where each element is a POID
POID is a string
POID2Sample is a reference to a hash where the key is a POID and the value is a SampleIDList
SampleIDList is a reference to a list where each element is a SampleID
SampleID is a string</pre>
   </dd>
   <dt id="Description5">
    Description
   </dt>
   <dd>
    <p>
     Retrieve the list of expression samples (GSM#s) that corresponds to one or more of the plant tissue/organ (PO) type of interest.
    </p>
   </dd>
  </dl>
  <h2 id="get_all_po">
   get_all_po
  </h2>
  <pre><code>  $results = $obj-&gt;get_all_po()</code></pre>
  <dl>
   <dt id="Parameter-and-return-types6">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$results is a POID2Description
POID2Description is a reference to a hash where the key is a POID and the value is a string
POID is a string</pre>
   </dd>
   <dt id="Description6">
    Description
   </dt>
   <dd>
   </dd>
  </dl>
  <h2 id="getAllPO">
   getAllPO
  </h2>
  <pre><code> Retrieve the list of all plant ontology IDs (POIDs) currently available in the database.</code></pre>
  <h2 id="get_all_eo">
   get_all_eo
  </h2>
  <pre><code>  $results = $obj-&gt;get_all_eo()</code></pre>
  <dl>
   <dt id="Parameter-and-return-types7">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$results is an EOID2Description
EOID2Description is a reference to a hash where the key is an EOID and the value is a string
EOID is a string</pre>
   </dd>
   <dt id="Description7">
    Description
   </dt>
   <dd>
    <p>
     Retrieve the list of all plant environment ontology IDs (EOIDs) currently available in the database.
    </p>
   </dd>
  </dl>
  <h2 id="get_po_descriptions">
   get_po_descriptions
  </h2>
  <pre><code>  $results = $obj-&gt;get_po_descriptions($ids)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types8">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$ids is a POIDList
$results is a POID2Description
POIDList is a reference to a list where each element is a POID
POID is a string
POID2Description is a reference to a hash where the key is a POID and the value is a string</pre>
   </dd>
   <dt id="Description8">
    Description
   </dt>
   <dd>
    <p>
     Retrieve the list of selected plant ontology IDs (POIDs) description corresponding to an input list of POIDs.
    </p>
   </dd>
  </dl>
  <h2 id="get_eo_descriptions">
   get_eo_descriptions
  </h2>
  <pre><code>  $results = $obj-&gt;get_eo_descriptions($ids)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types9">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$ids is an EOIDList
$results is an EOID2Description
EOIDList is a reference to a list where each element is an EOID
EOID is a string
EOID2Description is a reference to a hash where the key is an EOID and the value is a string</pre>
   </dd>
   <dt id="Description9">
    Description
   </dt>
   <dd>
    <p>
     Retrieve the list of selected plant environment ontology IDs (EOIDs) description corresponding to an input list of EOIDs.
    </p>
   </dd>
  </dl>
  <h1 id="TYPES">
   TYPES
  </h1>
  <h2 id="EOID">
   EOID
  </h2>
  <dl>
   <dt id="Description10">
    Description
   </dt>
   <dd>
    <p>
     external environment ontology id
    </p>
    <pre><code>     The supported EO ids are :
       EO:0007001        UV-B light regimen
       EO:0007002        UV-A light regimen
       EO:0007041        antibiotic regimen
       EO:0007049        soil environment
       EO:0007066        cytokinin regimen
       EO:0007067        hydroponic plant culture media
       EO:0007071        suspension cell culture media
       EO:0007075        high light intensity regimen
       EO:0007079        sulfate fertilizer regimen
       EO:0007105        abscisic acid regimen
       EO:0007106        Stramenopiles
       EO:0007107        Ascomycota
       EO:0007108        Proteobacteria
       EO:0007116        Hemiptera
       EO:0007128        intermittent light regimen
       EO:0007144        Pseudomonas spp.
       EO:0007149        chemical mutagen
       EO:0007158        sandy soil
       EO:0007162        continuous light regimen
       EO:0007173        warm/hot temperature regimen
       EO:0007174        cold temperature regimen
       EO:0007175        temperature environment
       EO:0007183        herbicide regimen
       EO:0007185        salt regimen
       EO:0007193        day light intensity
       EO:0007199        long day length regimen
       EO:0007200        short day length regimen
       EO:0007203        far red light regimen
       EO:0007207        red light regimen
       EO:0007218        blue light regimen
       EO:0007221        visible light regimen
       EO:0007233        Fungi
       EO:0007265        liquid growth media
       EO:0007266        tissue culture growth media
       EO:0007270        continuous dark (no light) regimen
       EO:0007271        low light intensity regimen
       EO:0007303        carbon nutrient regimen
       EO:0007373        mechanical damage
       EO:0007404        drought environment
       EO:0007409        brassinosteroid</code></pre>
   </dd>
   <dt id="Definition">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="POID">
   POID
  </h2>
  <dl>
   <dt id="Description11">
    Description
   </dt>
   <dd>
    <p>
     external plant ontology id
    </p>
    <pre><code>     The supported PO ids are :
       PO:0000003        whole plant
       PO:0000005        cultured cell
       PO:0000006        cultured protoplast
       PO:0009005        root
       PO:0009006        shoot
       PO:0009025        leaf
       PO:0009046        flower
       PO:0009049        Inflorescence
       PO:0009001        fruit
       PO:0009010        seed</code></pre>
   </dd>
   <dt id="Definition1">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="GeneID">
   GeneID
  </h2>
  <dl>
   <dt id="Description12">
    Description
   </dt>
   <dd>
    <p>
     external gene id
    </p>
   </dd>
   <dt id="Definition2">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="SampleID">
   SampleID
  </h2>
  <dl>
   <dt id="Description13">
    Description
   </dt>
   <dd>
    <p>
     expression sample id from GEO (GSM#)
    </p>
   </dd>
   <dt id="Definition3">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="SeriesID">
   SeriesID
  </h2>
  <dl>
   <dt id="Description14">
    Description
   </dt>
   <dd>
    <p>
     Series id from GEO (GSE#)
    </p>
   </dd>
   <dt id="Definition4">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="GeneIDList">
   GeneIDList
  </h2>
  <dl>
   <dt id="Description15">
    Description
   </dt>
   <dd>
    <p>
     list of external gene ids from species of interest
    </p>
   </dd>
   <dt id="Definition5">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list where each element is a GeneID
</pre>
   </dd>
  </dl>
  <h2 id="SampleIDList">
   SampleIDList
  </h2>
  <dl>
   <dt id="Description16">
    Description
   </dt>
   <dd>
    <p>
     list of Sample ids
    </p>
   </dd>
   <dt id="Definition6">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list where each element is a SampleID
</pre>
   </dd>
  </dl>
  <h2 id="EOIDList">
   EOIDList
  </h2>
  <dl>
   <dt id="Description17">
    Description
   </dt>
   <dd>
    <p>
     list of environment ontologies
    </p>
   </dd>
   <dt id="Definition7">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list where each element is an EOID
</pre>
   </dd>
  </dl>
  <h2 id="POIDList">
   POIDList
  </h2>
  <dl>
   <dt id="Description18">
    Description
   </dt>
   <dd>
    <p>
     list of plant ontologies
    </p>
   </dd>
   <dt id="Definition8">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list where each element is a POID
</pre>
   </dd>
  </dl>
  <h2 id="SeriesIDList">
   SeriesIDList
  </h2>
  <dl>
   <dt id="Definition9">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list where each element is a SeriesID
</pre>
   </dd>
  </dl>
  <h2 id="ValueList">
   ValueList
  </h2>
  <dl>
   <dt id="Definition10">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list where each element is a float
</pre>
   </dd>
  </dl>
  <h2 id="Sample">
   Sample
  </h2>
  <dl>
   <dt id="Definition11">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the key is a SampleID and the value is a ValueList
</pre>
   </dd>
  </dl>
  <h2 id="SampleList">
   SampleList
  </h2>
  <dl>
   <dt id="Definition12">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list where each element is a Sample
</pre>
   </dd>
  </dl>
  <h2 id="Experiment">
   Experiment
  </h2>
  <dl>
   <dt id="Definition13">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
series has a value which is a Sample
genes has a value which is a GeneIDList</pre>
   </dd>
  </dl>
  <h2 id="Experiments">
   Experiments
  </h2>
  <dl>
   <dt id="Definition14">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the key is a SeriesID and the value is an Experiment
</pre>
   </dd>
  </dl>
  <h2 id="EOID2Sample">
   EOID2Sample
  </h2>
  <dl>
   <dt id="Definition15">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the key is an EOID and the value is a SampleIDList
</pre>
   </dd>
  </dl>
  <h2 id="POID2Sample">
   POID2Sample
  </h2>
  <dl>
   <dt id="Definition16">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the key is a POID and the value is a SampleIDList
</pre>
   </dd>
  </dl>
  <h2 id="POID2Description">
   POID2Description
  </h2>
  <dl>
   <dt id="Definition17">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the key is a POID and the value is a string
</pre>
   </dd>
  </dl>
  <h2 id="EOID2Description">
   EOID2Description
  </h2>
  <dl>
   <dt id="Definition18">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the key is an EOID and the value is a string
</pre>
   </dd>
  </dl>
  <h1 id="POD-ERRORS">
   POD ERRORS
  </h1>
  <p>
   Hey!
   <b>
    The above document had some coding errors, which are explained below:
   </b>
  </p>
  <dl>
   <dt id="Around-line-657:">
    Around line 657:
   </dt>
   <dd>
    <p>
     You forgot a '=back' before '=head2'
    </p>
   </dd>
   <dt id="Around-line-661:">
    Around line 661:
   </dt>
   <dd>
    <p>
     =back without =over
    </p>
   </dd>
  </dl>
 </body>
</html>
