<?xml version="1.0" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<title></title>
<meta http-equiv="content-type" content="text/html; charset=utf-8" />
<link rev="made" href="mailto:root@t1.novalocal" />
</head>

<body style="background-color: white">



<ul id="index">
  <li><a href="#NAME">NAME</a></li>
  <li><a href="#DESCRIPTION">DESCRIPTION</a>
    <ul>
      <li><a href="#rank_pgene">rank_pgene</a></li>
      <li><a href="#get_pfam">get_pfam</a></li>
      <li><a href="#get_neighbor">get_neighbor</a></li>
    </ul>
  </li>
  <li><a href="#TYPES">TYPES</a>
    <ul>
      <li><a href="#wgene">wgene</a></li>
      <li><a href="#pgene">pgene</a></li>
      <li><a href="#pgenes">pgenes</a></li>
      <li><a href="#score">score</a></li>
      <li><a href="#pgene2score">pgene2score</a></li>
      <li><a href="#domain">domain</a></li>
      <li><a href="#domains">domains</a></li>
      <li><a href="#id">id</a></li>
      <li><a href="#ids">ids</a></li>
      <li><a href="#ids2domains">ids2domains</a></li>
      <li><a href="#neighbor">neighbor</a></li>
      <li><a href="#neighbors">neighbors</a></li>
      <li><a href="#ids2neighbors">ids2neighbors</a></li>
    </ul>
  </li>
</ul>

<h1 id="NAME">NAME</h1>

<p>Bio::KBase::FunctionalOrthologPredictor::Client</p>

<h1 id="DESCRIPTION">DESCRIPTION</h1>

<p>This module provides public interface/APIs for KBase functional ortholog predictor. The function is first demonstrated by Mutwil et al., 2011, Plant Cell. Given a well studied gene in Arabidopsis and a list of potential ortholog in Poplar, this module can predict the best functional ortholog based on gene co-expression network. Before using this module, make sure you can use KBase networks API and plant_expression service.</p>

<p>Here is how it is working: Given a well studied gene_A in a model plant(Arabidopsis), we have a list of candidate orthologs in another less-well studied plant(Poplar). This list may be identified by BLAST, or other sequence-based similarity search. Now, we want to know which one of the potential ortholog has the best chance to perform the same biological function as gene_A does.&#39;rank_pgenes&#39;, means rank the list of potential genes. First, get the co-expression neighbors of gene_A and all the potential genes through KBase networks API. (Currently which co-expression network dataset is used in hard coded. Later, this will be selected by the user). Then, the PFAM domain information for all these genes is retrieved.(Since there is not a KBase-wide PFAM domain table, the PFAM domain table for Arabidopsis and Poplar are stored in devdb1.newyork.kbase.us temporarily.). Finally, the shared PFAM domain between gene_A and each of the potential gene is compared. A score is calculated based on: number of shared domain bewteen ((the neighbors of gene_A and the neighbors of potential gene)/the number of domain of the neighbors of gene_A )+((the neighbors of gene_A and the neighbors of potential gene)/the number of domain of the neighbors of potential gene)). In the later version, a p-value will be calculated to evaluate the significance of the best potential ortholog. Currently, this module can be used only for two plant genomes, Arabidopsis and Poplar.</p>

<h2 id="rank_pgene">rank_pgene</h2>

<pre><code>  $results = $obj-&gt;rank_pgene($wgene, $pgenes)</code></pre>

<dl>

<dt id="Parameter-and-return-types">Parameter and return types</dt>
<dd>



<pre>
$wgene is a wgene
$pgenes is a pgenes
$results is a pgene2score
wgene is a string
pgenes is a reference to a list where each element is a pgene
pgene is a string
pgene2score is a reference to a hash where the key is a pgene and the value is a score
score is a float</pre>

</dd>
<dt id="Description">Description</dt>
<dd>

<p>&#39;rank_pgene&#39; receives two parameters:wgene and pgenes. wgene is a well-annotated gene in organism A. pgenes is a list of potential functional orthologs in organism B. This function first retrieves the co-expression neighbors through KBase Networks API. Then compare the similarity of those neighbors between potential ortholog and the well-annotated gene. PFAM domain information is utilized to do the comparsion. Basically, this function count the shared PFAM domains between two groups of neighbors. The more the shared domains, the higher the score is. When we are using the KBase networks API to get the co-expression neighbors for each gene, we have to select the dataset. Currently, the dataset is selected by default. Later, we can give users the right to select their interested dataset.</p>

</dd>
</dl>

<h2 id="get_pfam">get_pfam</h2>

<pre><code>  $results = $obj-&gt;get_pfam($ids)</code></pre>

<dl>

<dt id="Parameter-and-return-types1">Parameter and return types</dt>
<dd>



<pre>
$ids is an ids
$results is an ids2domains
ids is a reference to a list where each element is an id
id is a string
ids2domains is a reference to a hash where the key is an ids and the value is a domains
domains is a reference to a list where each element is a domain
domain is a string</pre>

</dd>
<dt id="Description1">Description</dt>
<dd>

<p>&#39;get_pfam&#39; receives kbase locus id and returns its PFAM domain list. Current only Arabidopsis and Poplar is supported. The PFAM annotation is download from biomart and stored in devdb1.newyork.kbase.us:kbase_plant:pfam_annotation. This function is only used as a support for the rank_pgene.</p>

</dd>
</dl>

<h2 id="get_neighbor">get_neighbor</h2>

<pre><code>  $results = $obj-&gt;get_neighbor($ids)</code></pre>

<dl>

<dt id="Parameter-and-return-types2">Parameter and return types</dt>
<dd>



<pre>
$ids is an ids
$results is an ids2neighbors
ids is a reference to a list where each element is an id
id is a string
ids2neighbors is a reference to a hash where the key is an ids and the value is a neighbors
neighbors is a reference to a list where each element is a neighbor
neighbor is a string</pre>

</dd>
<dt id="Description2">Description</dt>
<dd>

<p>&#39;get_neighor&#39; receives kbase locus id and returns the network neighbor in a given network. This function is only used as a support for the rank_pgene</p>

</dd>
</dl>

<h1 id="TYPES">TYPES</h1>

<h2 id="wgene">wgene</h2>

<dl>

<dt id="Description3">Description</dt>
<dd>

<p>wgene: wgene is a KBase gene locus id. This gene is well-annotated. Users want to identify its functional ortholog in another organism)</p>

</dd>
<dt id="Definition">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="pgene">pgene</h2>

<dl>

<dt id="Description4">Description</dt>
<dd>

<p>pgene: pgene is a KBase gene locus id, which is a potential functional ortholog of wgene.</p>

</dd>
<dt id="Definition1">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="pgenes">pgenes</h2>

<dl>

<dt id="Description5">Description</dt>
<dd>

<p>pgenes: pgenes is a list of pgene. There are more than one ortholog identified by sequence similarity comparsion.</p>

</dd>
<dt id="Definition2">Definition</dt>
<dd>



<pre>
a reference to a list where each element is a pgene
</pre>

</dd>
</dl>

<h2 id="score">score</h2>

<dl>

<dt id="Description6">Description</dt>
<dd>

<p>score: is a number, used to rank the potential functional ortholog</p>

</dd>
<dt id="Definition3">Definition</dt>
<dd>



<pre>
a float
</pre>

</dd>
</dl>

<h2 id="pgene2score">pgene2score</h2>

<dl>

<dt id="Description7">Description</dt>
<dd>

<p>pgene2score is a hash, the key is a pgene and the value is its score</p>

</dd>
<dt id="Definition4">Definition</dt>
<dd>



<pre>
a reference to a hash where the key is a pgene and the value is a score
</pre>

</dd>
</dl>

<h2 id="domain">domain</h2>

<dl>

<dt id="Description8">Description</dt>
<dd>

<p>domain: is a string, which is also a PFAM domain ID</p>

</dd>
<dt id="Definition5">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="domains">domains</h2>

<dl>

<dt id="Description9">Description</dt>
<dd>

<p>domains is a list of domain</p>

</dd>
<dt id="Definition6">Definition</dt>
<dd>



<pre>
a reference to a list where each element is a domain
</pre>

</dd>
</dl>

<h2 id="id">id</h2>

<dl>

<dt id="Description10">Description</dt>
<dd>

<p>id is a KBase gene locus id</p>

</dd>
<dt id="Definition7">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="ids">ids</h2>

<dl>

<dt id="Description11">Description</dt>
<dd>

<p>ids is a list of id</p>

</dd>
<dt id="Definition8">Definition</dt>
<dd>



<pre>
a reference to a list where each element is an id
</pre>

</dd>
</dl>

<h2 id="ids2domains">ids2domains</h2>

<dl>

<dt id="Description12">Description</dt>
<dd>

<p>ids2domains is a hash where the key is kbase gene locus and the value is its pfam domains</p>

</dd>
<dt id="Definition9">Definition</dt>
<dd>



<pre>
a reference to a hash where the key is an ids and the value is a domains
</pre>

</dd>
</dl>

<h2 id="neighbor">neighbor</h2>

<dl>

<dt id="Description13">Description</dt>
<dd>

<p>neighbor is a kbase locus id, which is also a network node from KBase networks service</p>

</dd>
<dt id="Definition10">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="neighbors">neighbors</h2>

<dl>

<dt id="Description14">Description</dt>
<dd>

<p>neighbors is a list of neighbor</p>

</dd>
<dt id="Definition11">Definition</dt>
<dd>



<pre>
a reference to a list where each element is a neighbor
</pre>

</dd>
</dl>

<h2 id="ids2neighbors">ids2neighbors</h2>

<dl>

<dt id="Description15">Description</dt>
<dd>

<p>ids2neighbors is a hash, where the key is the a id and the value is its neighbors</p>

</dd>
<dt id="Definition12">Definition</dt>
<dd>



<pre>
a reference to a hash where the key is an ids and the value is a neighbors
</pre>

</dd>
</dl>


</body>

</html>


