<?xml version="1.0" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
 <head>
  <title>
   fbaModelServices
  </title>
  <meta content="text/html; charset=utf-8" http-equiv="content-type"/>
  <link href="mailto:root@ubuntu.chicago.kbase.us" rev="made"/>
 </head>
 <body style="background-color: white">
  <h1 id="NAME">
   NAME
  </h1>
  <p>
   Bio::KBase::fbaModelServices::Client
  </p>
  <h1 id="DESCRIPTION">
   DESCRIPTION
  </h1>
  <h1 id="fbaModelServices">
   fbaModelServices
  </h1>
  <h2 id="SYNOPSIS">
   SYNOPSIS
  </h2>
  <p>
   The FBA Model Services include support related to the reconstruction, curation, reconciliation, and analysis of metabolic models. This includes commands to:
  </p>
  <p>
   1.) Load genome typed objects into a workspace
  </p>
  <p>
   2.) Build a model from a genome typed object and curate the model
  </p>
  <p>
   3.) Analyze a model with flux balance analysis
  </p>
  <p>
   4.) Simulate and reconcile a model to an imported set of growth phenotype data
  </p>
  <h2 id="EXAMPLE-OF-API-USE-IN-PERL">
   EXAMPLE OF API USE IN PERL
  </h2>
  <p>
   To use the API, first you need to instantiate a fbaModelServices client object:
  </p>
  <p>
   my $client = Bio::KBase::fbaModelServices::Client-&gt;new;
  </p>
  <p>
   Next, you can run API commands on the client object:
  </p>
  <p>
   my $objmeta = $client-&gt;genome_to_workspace({ genome =&gt; "kb|g.0", workspace =&gt; "myWorkspace" }); my $objmeta = $client-&gt;genome_to_fbamodel({ model =&gt; "myModel" workspace =&gt; "myWorkspace" });
  </p>
  <h2 id="AUTHENTICATION">
   AUTHENTICATION
  </h2>
  <p>
   Each and every function in this service takes a hash reference as its single argument. This hash reference may contain a key
   <code>
    auth
   </code>
   whose value is a bearer token for the user making the request. If this is not provided a default user "public" is assumed.
  </p>
  <h2 id="WORKSPACE">
   WORKSPACE
  </h2>
  <p>
   A workspace is a named collection of objects owned by a specific user, that may be viewable or editable by other users.Functions that operate on workspaces take a
   <code>
    workspace_id
   </code>
   , which is an alphanumeric string that uniquely identifies a workspace among all workspaces.
  </p>
  <h2 id="get_models">
   get_models
  </h2>
  <pre><code>  $out_models = $obj-&gt;get_models($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a get_models_params
$out_models is a reference to a list where each element is an FBAModel
get_models_params is a reference to a hash where the following keys are defined:
	models has a value which is a reference to a list where each element is a fbamodel_id
	workspaces has a value which is a reference to a list where each element is a workspace_id
	auth has a value which is a string
	id_type has a value which is a string
fbamodel_id is a string
workspace_id is a string
FBAModel is a reference to a hash where the following keys are defined:
	id has a value which is a fbamodel_id
	workspace has a value which is a workspace_id
	genome has a value which is a genome_id
	genome_workspace has a value which is a workspace_id
	map has a value which is a mapping_id
	map_workspace has a value which is a workspace_id
	biochemistry has a value which is a biochemistry_id
	biochemistry_workspace has a value which is a workspace_id
	name has a value which is a string
	type has a value which is a string
	status has a value which is a string
	biomasses has a value which is a reference to a list where each element is a ModelBiomass
	compartments has a value which is a reference to a list where each element is a ModelCompartment
	reactions has a value which is a reference to a list where each element is a ModelReaction
	compounds has a value which is a reference to a list where each element is a ModelCompound
	fbas has a value which is a reference to a list where each element is an FBAMeta
	integrated_gapfillings has a value which is a reference to a list where each element is a GapFillMeta
	unintegrated_gapfillings has a value which is a reference to a list where each element is a GapFillMeta
	integrated_gapgenerations has a value which is a reference to a list where each element is a GapGenMeta
	unintegrated_gapgenerations has a value which is a reference to a list where each element is a GapGenMeta
	modelSubsystems has a value which is a reference to a list where each element is a Subsystem
genome_id is a string
mapping_id is a string
biochemistry_id is a string
ModelBiomass is a reference to a hash where the following keys are defined:
	id has a value which is a biomass_id
	name has a value which is a string
	definition has a value which is a string
	biomass_compounds has a value which is a reference to a list where each element is a BiomassCompound
biomass_id is a string
BiomassCompound is a reference to a list containing 3 items:
	0: (modelcompound) a modelcompound_id
	1: (coefficient) a float
	2: (name) a string
modelcompound_id is a string
ModelCompartment is a reference to a hash where the following keys are defined:
	id has a value which is a modelcompartment_id
	name has a value which is a string
	pH has a value which is a float
	potential has a value which is a float
	index has a value which is an int
modelcompartment_id is a string
ModelReaction is a reference to a hash where the following keys are defined:
	id has a value which is a modelreaction_id
	reaction has a value which is a reaction_id
	name has a value which is a string
	direction has a value which is a string
	equation has a value which is a string
	definition has a value which is a string
	gapfilled has a value which is a bool
	features has a value which is a reference to a list where each element is a feature_id
	compartment has a value which is a modelcompartment_id
modelreaction_id is a string
reaction_id is a string
feature_id is a string
ModelCompound is a reference to a hash where the following keys are defined:
	id has a value which is a modelcompound_id
	compound has a value which is a compound_id
	name has a value which is a string
	compartment has a value which is a modelcompartment_id
compound_id is a string
FBAMeta is a reference to a list containing 6 items:
	0: (id) a fba_id
	1: (workspace) a workspace_id
	2: (media) a media_id
	3: (media_workspace) a workspace_id
	4: (objective) a float
	5: (ko) a reference to a list where each element is a feature_id
fba_id is a string
media_id is a string
GapFillMeta is a reference to a list containing 6 items:
	0: (id) a gapfill_id
	1: (workspace) a workspace_id
	2: (media) a media_id
	3: (media_workspace) a workspace_id
	4: (done) a bool
	5: (ko) a reference to a list where each element is a feature_id
gapfill_id is a string
GapGenMeta is a reference to a list containing 6 items:
	0: (id) a gapgen_id
	1: (workspace) a workspace_id
	2: (media) a media_id
	3: (media_workspace) a workspace_id
	4: (done) a bool
	5: (ko) a reference to a list where each element is a feature_id
gapgen_id is a string
Subsystem is a reference to a hash where the following keys are defined:
	id has a value which is a subsystem_id
	name has a value which is a string
	class has a value which is a string
	subclass has a value which is a string
	type has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	roles has a value which is a reference to a list where each element is a role_id</pre>
   </dd>
   <dt id="Description">
    Description
   </dt>
   <dd>
    <p>
     Returns model data for input ids
    </p>
   </dd>
  </dl>
  <h2 id="get_fbas">
   get_fbas
  </h2>
  <pre><code>  $out_fbas = $obj-&gt;get_fbas($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types1">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a get_fbas_params
$out_fbas is a reference to a list where each element is an FBA
get_fbas_params is a reference to a hash where the following keys are defined:
	fbas has a value which is a reference to a list where each element is a fba_id
	workspaces has a value which is a reference to a list where each element is a workspace_id
	auth has a value which is a string
	id_type has a value which is a string
fba_id is a string
workspace_id is a string
FBA is a reference to a hash where the following keys are defined:
	id has a value which is a fba_id
	workspace has a value which is a workspace_id
	model has a value which is a fbamodel_id
	model_workspace has a value which is a workspace_id
	objective has a value which is a float
	isComplete has a value which is a bool
	formulation has a value which is an FBAFormulation
	minimalMediaPredictions has a value which is a reference to a list where each element is a MinimalMediaPrediction
	metaboliteProductions has a value which is a reference to a list where each element is a MetaboliteProduction
	reactionFluxes has a value which is a reference to a list where each element is a ReactionFlux
	compoundFluxes has a value which is a reference to a list where each element is a CompoundFlux
	geneAssertions has a value which is a reference to a list where each element is a GeneAssertion
fbamodel_id is a string
FBAFormulation is a reference to a hash where the following keys are defined:
	media has a value which is a media_id
	additionalcpds has a value which is a reference to a list where each element is a compound_id
	prommodel has a value which is a prommodel_id
	prommodel_workspace has a value which is a workspace_id
	media_workspace has a value which is a workspace_id
	objfraction has a value which is a float
	allreversible has a value which is a bool
	maximizeObjective has a value which is a bool
	objectiveTerms has a value which is a reference to a list where each element is a term
	geneko has a value which is a reference to a list where each element is a feature_id
	rxnko has a value which is a reference to a list where each element is a reaction_id
	bounds has a value which is a reference to a list where each element is a bound
	constraints has a value which is a reference to a list where each element is a constraint
	uptakelim has a value which is a reference to a hash where the key is a string and the value is a float
	defaultmaxflux has a value which is a float
	defaultminuptake has a value which is a float
	defaultmaxuptake has a value which is a float
	simplethermoconst has a value which is a bool
	thermoconst has a value which is a bool
	nothermoerror has a value which is a bool
	minthermoerror has a value which is a bool
media_id is a string
compound_id is a string
prommodel_id is a string
term is a reference to a list containing 3 items:
	0: (coefficient) a float
	1: (varType) a string
	2: (variable) a string
feature_id is a string
reaction_id is a string
bound is a reference to a list containing 4 items:
	0: (min) a float
	1: (max) a float
	2: (varType) a string
	3: (variable) a string
constraint is a reference to a list containing 4 items:
	0: (rhs) a float
	1: (sign) a string
	2: (terms) a reference to a list where each element is a term
	3: (name) a string
MinimalMediaPrediction is a reference to a hash where the following keys are defined:
	optionalNutrients has a value which is a reference to a list where each element is a compound_id
	essentialNutrients has a value which is a reference to a list where each element is a compound_id
MetaboliteProduction is a reference to a list containing 3 items:
	0: (maximumProduction) a float
	1: (modelcompound) a modelcompound_id
	2: (name) a string
modelcompound_id is a string
ReactionFlux is a reference to a list containing 8 items:
	0: (reaction) a modelreaction_id
	1: (value) a float
	2: (upperBound) a float
	3: (lowerBound) a float
	4: (max) a float
	5: (min) a float
	6: (type) a string
	7: (definition) a string
modelreaction_id is a string
CompoundFlux is a reference to a list containing 8 items:
	0: (compound) a modelcompound_id
	1: (value) a float
	2: (upperBound) a float
	3: (lowerBound) a float
	4: (max) a float
	5: (min) a float
	6: (type) a string
	7: (name) a string
GeneAssertion is a reference to a list containing 4 items:
	0: (feature) a feature_id
	1: (growthFraction) a float
	2: (growth) a float
	3: (isEssential) a bool</pre>
   </dd>
   <dt id="Description1">
    Description
   </dt>
   <dd>
    <p>
     Returns data for the requested flux balance analysis formulations
    </p>
   </dd>
  </dl>
  <h2 id="get_gapfills">
   get_gapfills
  </h2>
  <pre><code>  $out_gapfills = $obj-&gt;get_gapfills($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types2">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a get_gapfills_params
$out_gapfills is a reference to a list where each element is a GapFill
get_gapfills_params is a reference to a hash where the following keys are defined:
	gapfills has a value which is a reference to a list where each element is a gapfill_id
	workspaces has a value which is a reference to a list where each element is a workspace_id
	auth has a value which is a string
	id_type has a value which is a string
gapfill_id is a string
workspace_id is a string
GapFill is a reference to a hash where the following keys are defined:
	id has a value which is a gapfill_id
	workspace has a value which is a workspace_id
	model has a value which is a fbamodel_id
	model_workspace has a value which is a workspace_id
	isComplete has a value which is a bool
	formulation has a value which is a GapfillingFormulation
	solutions has a value which is a reference to a list where each element is a GapFillSolution
fbamodel_id is a string
GapfillingFormulation is a reference to a hash where the following keys are defined:
	formulation has a value which is an FBAFormulation
	num_solutions has a value which is an int
	nomediahyp has a value which is a bool
	nobiomasshyp has a value which is a bool
	nogprhyp has a value which is a bool
	nopathwayhyp has a value which is a bool
	allowunbalanced has a value which is a bool
	activitybonus has a value which is a float
	drainpen has a value which is a float
	directionpen has a value which is a float
	nostructpen has a value which is a float
	unfavorablepen has a value which is a float
	nodeltagpen has a value which is a float
	biomasstranspen has a value which is a float
	singletranspen has a value which is a float
	transpen has a value which is a float
	blacklistedrxns has a value which is a reference to a list where each element is a reaction_id
	gauranteedrxns has a value which is a reference to a list where each element is a reaction_id
	allowedcmps has a value which is a reference to a list where each element is a compartment_id
	probabilisticAnnotation has a value which is a probanno_id
	probabilisticAnnotation_workspace has a value which is a workspace_id
FBAFormulation is a reference to a hash where the following keys are defined:
	media has a value which is a media_id
	additionalcpds has a value which is a reference to a list where each element is a compound_id
	prommodel has a value which is a prommodel_id
	prommodel_workspace has a value which is a workspace_id
	media_workspace has a value which is a workspace_id
	objfraction has a value which is a float
	allreversible has a value which is a bool
	maximizeObjective has a value which is a bool
	objectiveTerms has a value which is a reference to a list where each element is a term
	geneko has a value which is a reference to a list where each element is a feature_id
	rxnko has a value which is a reference to a list where each element is a reaction_id
	bounds has a value which is a reference to a list where each element is a bound
	constraints has a value which is a reference to a list where each element is a constraint
	uptakelim has a value which is a reference to a hash where the key is a string and the value is a float
	defaultmaxflux has a value which is a float
	defaultminuptake has a value which is a float
	defaultmaxuptake has a value which is a float
	simplethermoconst has a value which is a bool
	thermoconst has a value which is a bool
	nothermoerror has a value which is a bool
	minthermoerror has a value which is a bool
media_id is a string
compound_id is a string
prommodel_id is a string
term is a reference to a list containing 3 items:
	0: (coefficient) a float
	1: (varType) a string
	2: (variable) a string
feature_id is a string
reaction_id is a string
bound is a reference to a list containing 4 items:
	0: (min) a float
	1: (max) a float
	2: (varType) a string
	3: (variable) a string
constraint is a reference to a list containing 4 items:
	0: (rhs) a float
	1: (sign) a string
	2: (terms) a reference to a list where each element is a term
	3: (name) a string
compartment_id is a string
probanno_id is a string
GapFillSolution is a reference to a hash where the following keys are defined:
	id has a value which is a gapfillsolution_id
	objective has a value which is a float
	integrated has a value which is a bool
	biomassRemovals has a value which is a reference to a list where each element is a biomassRemoval
	mediaAdditions has a value which is a reference to a list where each element is a mediaAddition
	reactionAdditions has a value which is a reference to a list where each element is a reactionAddition
gapfillsolution_id is a string
biomassRemoval is a reference to a list containing 2 items:
	0: (compound) a compound_id
	1: (name) a string
mediaAddition is a reference to a list containing 2 items:
	0: (compound) a compound_id
	1: (name) a string
reactionAddition is a reference to a list containing 5 items:
	0: (reaction) a reaction_id
	1: (direction) a string
	2: (compartment_id) a string
	3: (equation) a string
	4: (definition) a string</pre>
   </dd>
   <dt id="Description2">
    Description
   </dt>
   <dd>
    <p>
     Returns data for the requested gap filling simulations
    </p>
   </dd>
  </dl>
  <h2 id="get_gapgens">
   get_gapgens
  </h2>
  <pre><code>  $out_gapgens = $obj-&gt;get_gapgens($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types3">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a get_gapgens_params
$out_gapgens is a reference to a list where each element is a GapGen
get_gapgens_params is a reference to a hash where the following keys are defined:
	gapgens has a value which is a reference to a list where each element is a gapgen_id
	workspaces has a value which is a reference to a list where each element is a workspace_id
	auth has a value which is a string
	id_type has a value which is a string
gapgen_id is a string
workspace_id is a string
GapGen is a reference to a hash where the following keys are defined:
	id has a value which is a gapgen_id
	workspace has a value which is a workspace_id
	model has a value which is a fbamodel_id
	model_workspace has a value which is a workspace_id
	isComplete has a value which is a bool
	formulation has a value which is a GapgenFormulation
	solutions has a value which is a reference to a list where each element is a GapgenSolution
fbamodel_id is a string
GapgenFormulation is a reference to a hash where the following keys are defined:
	formulation has a value which is an FBAFormulation
	refmedia has a value which is a media_id
	refmedia_workspace has a value which is a workspace_id
	num_solutions has a value which is an int
	nomediahyp has a value which is a bool
	nobiomasshyp has a value which is a bool
	nogprhyp has a value which is a bool
	nopathwayhyp has a value which is a bool
FBAFormulation is a reference to a hash where the following keys are defined:
	media has a value which is a media_id
	additionalcpds has a value which is a reference to a list where each element is a compound_id
	prommodel has a value which is a prommodel_id
	prommodel_workspace has a value which is a workspace_id
	media_workspace has a value which is a workspace_id
	objfraction has a value which is a float
	allreversible has a value which is a bool
	maximizeObjective has a value which is a bool
	objectiveTerms has a value which is a reference to a list where each element is a term
	geneko has a value which is a reference to a list where each element is a feature_id
	rxnko has a value which is a reference to a list where each element is a reaction_id
	bounds has a value which is a reference to a list where each element is a bound
	constraints has a value which is a reference to a list where each element is a constraint
	uptakelim has a value which is a reference to a hash where the key is a string and the value is a float
	defaultmaxflux has a value which is a float
	defaultminuptake has a value which is a float
	defaultmaxuptake has a value which is a float
	simplethermoconst has a value which is a bool
	thermoconst has a value which is a bool
	nothermoerror has a value which is a bool
	minthermoerror has a value which is a bool
media_id is a string
compound_id is a string
prommodel_id is a string
term is a reference to a list containing 3 items:
	0: (coefficient) a float
	1: (varType) a string
	2: (variable) a string
feature_id is a string
reaction_id is a string
bound is a reference to a list containing 4 items:
	0: (min) a float
	1: (max) a float
	2: (varType) a string
	3: (variable) a string
constraint is a reference to a list containing 4 items:
	0: (rhs) a float
	1: (sign) a string
	2: (terms) a reference to a list where each element is a term
	3: (name) a string
GapgenSolution is a reference to a hash where the following keys are defined:
	id has a value which is a gapgensolution_id
	objective has a value which is a float
	biomassAdditions has a value which is a reference to a list where each element is a biomassAddition
	mediaRemovals has a value which is a reference to a list where each element is a mediaRemoval
	reactionRemovals has a value which is a reference to a list where each element is a reactionRemoval
gapgensolution_id is a string
biomassAddition is a reference to a list containing 2 items:
	0: (compound) a compound_id
	1: (name) a string
mediaRemoval is a reference to a list containing 2 items:
	0: (compound) a compound_id
	1: (name) a string
reactionRemoval is a reference to a list containing 4 items:
	0: (reaction) a modelreaction_id
	1: (direction) a string
	2: (equation) a string
	3: (definition) a string
modelreaction_id is a string</pre>
   </dd>
   <dt id="Description3">
    Description
   </dt>
   <dd>
    <p>
     Returns data for the requested gap generation simulations
    </p>
   </dd>
  </dl>
  <h2 id="get_reactions">
   get_reactions
  </h2>
  <pre><code>  $out_reactions = $obj-&gt;get_reactions($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types4">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a get_reactions_params
$out_reactions is a reference to a list where each element is a Reaction
get_reactions_params is a reference to a hash where the following keys are defined:
	reactions has a value which is a reference to a list where each element is a reaction_id
	auth has a value which is a string
	id_type has a value which is a string
reaction_id is a string
Reaction is a reference to a hash where the following keys are defined:
	id has a value which is a reaction_id
	name has a value which is a string
	abbrev has a value which is a string
	enzymes has a value which is a reference to a list where each element is a string
	direction has a value which is a string
	reversibility has a value which is a string
	deltaG has a value which is a float
	deltaGErr has a value which is a float
	equation has a value which is a string
	definition has a value which is a string</pre>
   </dd>
   <dt id="Description4">
    Description
   </dt>
   <dd>
    <p>
     Returns data for the requested reactions
    </p>
   </dd>
  </dl>
  <h2 id="get_compounds">
   get_compounds
  </h2>
  <pre><code>  $out_compounds = $obj-&gt;get_compounds($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types5">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a get_compounds_params
$out_compounds is a reference to a list where each element is a Compound
get_compounds_params is a reference to a hash where the following keys are defined:
	compounds has a value which is a reference to a list where each element is a compound_id
	auth has a value which is a string
	id_type has a value which is a string
compound_id is a string
Compound is a reference to a hash where the following keys are defined:
	id has a value which is a compound_id
	abbrev has a value which is a string
	name has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	charge has a value which is a float
	deltaG has a value which is a float
	deltaGErr has a value which is a float
	formula has a value which is a string</pre>
   </dd>
   <dt id="Description5">
    Description
   </dt>
   <dd>
    <p>
     Returns data for the requested compounds
    </p>
   </dd>
  </dl>
  <h2 id="get_alias">
   get_alias
  </h2>
  <pre><code>  $output = $obj-&gt;get_alias($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types6">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a get_alias_params
$output is a reference to a list where each element is a get_alias_outputs
get_alias_params is a reference to a hash where the following keys are defined:
	object_type has a value which is a string
	input_id_type has a value which is a string
	output_id_type has a value which is a string
	input_ids has a value which is a reference to a list where each element is a string
	auth has a value which is a string
get_alias_outputs is a reference to a hash where the following keys are defined:
	original_id has a value which is a string
	aliases has a value which is a reference to a list where each element is a string</pre>
   </dd>
   <dt id="Description6">
    Description
   </dt>
   <dd>
    <p>
     Turns one compound I into another of a different type
    </p>
   </dd>
  </dl>
  <h2 id="get_aliassets">
   get_aliassets
  </h2>
  <pre><code>  $aliassets = $obj-&gt;get_aliassets($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types7">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a get_aliassets_params
$aliassets is a reference to a list where each element is a string
get_aliassets_params is a reference to a hash where the following keys are defined:
	object_type has a value which is a string
	auth has a value which is a string</pre>
   </dd>
   <dt id="Description7">
    Description
   </dt>
   <dd>
    <p>
     Get possible types of aliases (alias sets)
    </p>
   </dd>
  </dl>
  <h2 id="get_media">
   get_media
  </h2>
  <pre><code>  $out_media = $obj-&gt;get_media($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types8">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a get_media_params
$out_media is a reference to a list where each element is a Media
get_media_params is a reference to a hash where the following keys are defined:
	medias has a value which is a reference to a list where each element is a media_id
	workspaces has a value which is a reference to a list where each element is a workspace_id
	auth has a value which is a string
media_id is a string
workspace_id is a string
Media is a reference to a hash where the following keys are defined:
	id has a value which is a media_id
	name has a value which is a string
	media_compounds has a value which is a reference to a list where each element is a MediaCompound
	pH has a value which is a float
	temperature has a value which is a float
MediaCompound is a reference to a hash where the following keys are defined:
	compound has a value which is a compound_id
	name has a value which is a string
	concentration has a value which is a float
	max_flux has a value which is a float
	min_flux has a value which is a float
compound_id is a string</pre>
   </dd>
   <dt id="Description8">
    Description
   </dt>
   <dd>
    <p>
     Returns data for the requested media formulations
    </p>
   </dd>
  </dl>
  <h2 id="get_biochemistry">
   get_biochemistry
  </h2>
  <pre><code>  $out_biochemistry = $obj-&gt;get_biochemistry($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types9">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a get_biochemistry_params
$out_biochemistry is a Biochemistry
get_biochemistry_params is a reference to a hash where the following keys are defined:
	biochemistry has a value which is a biochemistry_id
	biochemistry_workspace has a value which is a workspace_id
	id_type has a value which is a string
	auth has a value which is a string
biochemistry_id is a string
workspace_id is a string
Biochemistry is a reference to a hash where the following keys are defined:
	id has a value which is a biochemistry_id
	name has a value which is a string
	compounds has a value which is a reference to a list where each element is a compound_id
	reactions has a value which is a reference to a list where each element is a reaction_id
	media has a value which is a reference to a list where each element is a media_id
compound_id is a string
reaction_id is a string
media_id is a string</pre>
   </dd>
   <dt id="Description9">
    Description
   </dt>
   <dd>
    <p>
     Returns biochemistry object
    </p>
   </dd>
  </dl>
  <h2 id="get_ETCDiagram">
   get_ETCDiagram
  </h2>
  <pre><code>  $output = $obj-&gt;get_ETCDiagram($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types10">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a get_ETCDiagram_params
$output is an ETCDiagramSpecs
get_ETCDiagram_params is a reference to a hash where the following keys are defined:
	model has a value which is a fbamodel_id
	workspace has a value which is a workspace_id
	media has a value which is a media_id
	mediaws has a value which is a workspace_id
	auth has a value which is a string
fbamodel_id is a string
workspace_id is a string
media_id is a string
ETCDiagramSpecs is a reference to a hash where the following keys are defined:
	nodes has a value which is a reference to a list where each element is an ETCNodes
	media has a value which is a string
	growth has a value which is a string
	organism has a value which is a string
ETCNodes is a reference to a hash where the following keys are defined:
	resp has a value which is a string
	y has a value which is an int
	x has a value which is an int
	width has a value which is an int
	height has a value which is an int
	shape has a value which is a string
	label has a value which is a string</pre>
   </dd>
   <dt id="Description10">
    Description
   </dt>
   <dd>
    <p>
     This function retrieves an ETC diagram for the input model operating in the input media condition The model must grow on the specified media in order to return a working ETC diagram
    </p>
   </dd>
  </dl>
  <h2 id="import_probanno">
   import_probanno
  </h2>
  <pre><code>  $probannoMeta = $obj-&gt;import_probanno($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types11">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is an import_probanno_params
$probannoMeta is an object_metadata
import_probanno_params is a reference to a hash where the following keys are defined:
	probanno has a value which is a probanno_id
	workspace has a value which is a workspace_id
	genome has a value which is a genome_id
	genome_workspace has a value which is a workspace_id
	annotationProbabilities has a value which is a reference to a list where each element is an annotationProbability
	ignore_errors has a value which is a bool
	auth has a value which is a string
	overwrite has a value which is a bool
probanno_id is a string
workspace_id is a string
genome_id is a string
annotationProbability is a reference to a list containing 3 items:
	0: (feature) a feature_id
	1: (function) a string
	2: (probability) a float
feature_id is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_ref is a string</pre>
   </dd>
   <dt id="Description11">
    Description
   </dt>
   <dd>
    <p>
     Loads an input genome object into the workspace.
    </p>
   </dd>
  </dl>
  <h2 id="genome_object_to_workspace">
   genome_object_to_workspace
  </h2>
  <pre><code>  $genomeMeta = $obj-&gt;genome_object_to_workspace($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types12">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a genome_object_to_workspace_params
$genomeMeta is an object_metadata
genome_object_to_workspace_params is a reference to a hash where the following keys are defined:
	genomeobj has a value which is a GenomeObject
	workspace has a value which is a workspace_id
	auth has a value which is a string
	overwrite has a value which is a bool
GenomeObject is a reference to a hash where the following keys are defined:
	id has a value which is a genome_id
	scientific_name has a value which is a string
	domain has a value which is a string
	genetic_code has a value which is an int
	source has a value which is a string
	source_id has a value which is a string
	contigs has a value which is a reference to a list where each element is a contig
	features has a value which is a reference to a list where each element is a feature
genome_id is a string
contig is a reference to a hash where the following keys are defined:
	id has a value which is a contig_id
	dna has a value which is a string
contig_id is a string
feature is a reference to a hash where the following keys are defined:
	id has a value which is a feature_id
	location has a value which is a location
	type has a value which is a feature_type
	function has a value which is a string
	alternative_functions has a value which is a reference to a list where each element is an alt_func
	protein_translation has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	annotations has a value which is a reference to a list where each element is an annotation
feature_id is a string
location is a reference to a list where each element is a region_of_dna
region_of_dna is a reference to a list containing 4 items:
	0: a contig_id
	1: (begin) an int
	2: (strand) a string
	3: (length) an int
feature_type is a string
alt_func is a reference to a list containing 2 items:
	0: (function) a string
	1: (probability) a float
gene_hit is a reference to a list containing 2 items:
	0: (gene) a feature_id
	1: (blast_score) a float
annotation is a reference to a list containing 3 items:
	0: (comment) a string
	1: (annotator) a string
	2: (annotation_time) an int
workspace_id is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_ref is a string</pre>
   </dd>
   <dt id="Description12">
    Description
   </dt>
   <dd>
    <p>
     Loads an input genome object into the workspace.
    </p>
   </dd>
  </dl>
  <h2 id="genome_to_workspace">
   genome_to_workspace
  </h2>
  <pre><code>  $genomeMeta = $obj-&gt;genome_to_workspace($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types13">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a genome_to_workspace_params
$genomeMeta is an object_metadata
genome_to_workspace_params is a reference to a hash where the following keys are defined:
	genome has a value which is a genome_id
	workspace has a value which is a workspace_id
	sourceLogin has a value which is a string
	sourcePassword has a value which is a string
	source has a value which is a string
	auth has a value which is a string
	overwrite has a value which is a bool
genome_id is a string
workspace_id is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_ref is a string</pre>
   </dd>
   <dt id="Description13">
    Description
   </dt>
   <dd>
    <p>
     Retrieves a genome from the CDM and saves it as a genome object in the workspace.
    </p>
   </dd>
  </dl>
  <h2 id="add_feature_translation">
   add_feature_translation
  </h2>
  <pre><code>  $genomeMeta = $obj-&gt;add_feature_translation($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types14">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is an add_feature_translation_params
$genomeMeta is an object_metadata
add_feature_translation_params is a reference to a hash where the following keys are defined:
	genome has a value which is a genome_id
	workspace has a value which is a workspace_id
	translations has a value which is a reference to a list where each element is a translation
	id_type has a value which is a string
	auth has a value which is a string
	overwrite has a value which is a bool
genome_id is a string
workspace_id is a string
translation is a reference to a list containing 2 items:
	0: (foreign_id) a string
	1: (feature) a feature_id
feature_id is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_ref is a string</pre>
   </dd>
   <dt id="Description14">
    Description
   </dt>
   <dd>
    <p>
     Adds a new set of alternative feature IDs to the specified genome typed object
    </p>
   </dd>
  </dl>
  <h2 id="genome_to_fbamodel">
   genome_to_fbamodel
  </h2>
  <pre><code>  $modelMeta = $obj-&gt;genome_to_fbamodel($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types15">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a genome_to_fbamodel_params
$modelMeta is an object_metadata
genome_to_fbamodel_params is a reference to a hash where the following keys are defined:
	genome has a value which is a genome_id
	genome_workspace has a value which is a workspace_id
	templatemodel has a value which is a template_id
	templatemodel_workspace has a value which is a workspace_id
	model has a value which is a fbamodel_id
	coremodel has a value which is a bool
	workspace has a value which is a workspace_id
	auth has a value which is a string
	overwrite has a value which is a bool
genome_id is a string
workspace_id is a string
template_id is a string
fbamodel_id is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_ref is a string</pre>
   </dd>
   <dt id="Description15">
    Description
   </dt>
   <dd>
    <p>
     Build a genome-scale metabolic model based on annotations in an input genome typed object
    </p>
   </dd>
  </dl>
  <h2 id="import_fbamodel">
   import_fbamodel
  </h2>
  <pre><code>  $modelMeta = $obj-&gt;import_fbamodel($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types16">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is an import_fbamodel_params
$modelMeta is an object_metadata
import_fbamodel_params is a reference to a hash where the following keys are defined:
	genome has a value which is a genome_id
	genome_workspace has a value which is a workspace_id
	biomass has a value which is a string
	reactions has a value which is a reference to a list where each element is a reference to a list containing 4 items:
	0: (id) a string
	1: (direction) a string
	2: (compartment) a string
	3: (gpr) a string	model has a value which is a fbamodel_id
	workspace has a value which is a workspace_id
	ignore_errors has a value which is a bool
	auth has a value which is a string
	overwrite has a value which is a bool
genome_id is a string
workspace_id is a string
fbamodel_id is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_ref is a string
</pre>
   </dd>
   <dt id="Description16">
    Description
   </dt>
   <dd>
    <p>
     Import a model from an input table of model and gene IDs
    </p>
   </dd>
  </dl>
  <h2 id="export_fbamodel">
   export_fbamodel
  </h2>
  <pre><code>  $output = $obj-&gt;export_fbamodel($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types17">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is an export_fbamodel_params
$output is a string
export_fbamodel_params is a reference to a hash where the following keys are defined:
	model has a value which is a fbamodel_id
	workspace has a value which is a workspace_id
	format has a value which is a string
	auth has a value which is a string
fbamodel_id is a string
workspace_id is a string</pre>
   </dd>
   <dt id="Description17">
    Description
   </dt>
   <dd>
    <p>
     This function exports the specified FBAModel to a specified format (sbml,html)
    </p>
   </dd>
  </dl>
  <h2 id="export_object">
   export_object
  </h2>
  <pre><code>  $output = $obj-&gt;export_object($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types18">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is an export_object_params
$output is a string
export_object_params is a reference to a hash where the following keys are defined:
	reference has a value which is a workspace_ref
	type has a value which is a string
	format has a value which is a string
	auth has a value which is a string
workspace_ref is a string</pre>
   </dd>
   <dt id="Description18">
    Description
   </dt>
   <dd>
    <p>
     This function prints the object pointed to by the input reference in the specified format
    </p>
   </dd>
  </dl>
  <h2 id="export_genome">
   export_genome
  </h2>
  <pre><code>  $output = $obj-&gt;export_genome($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types19">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is an export_genome_params
$output is a string
export_genome_params is a reference to a hash where the following keys are defined:
	genome has a value which is a genome_id
	workspace has a value which is a workspace_id
	format has a value which is a string
	auth has a value which is a string
genome_id is a string
workspace_id is a string</pre>
   </dd>
   <dt id="Description19">
    Description
   </dt>
   <dd>
    <p>
     This function exports the specified FBAModel to a specified format (sbml,html)
    </p>
   </dd>
  </dl>
  <h2 id="adjust_model_reaction">
   adjust_model_reaction
  </h2>
  <pre><code>  $modelMeta = $obj-&gt;adjust_model_reaction($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types20">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is an adjust_model_reaction_params
$modelMeta is an object_metadata
adjust_model_reaction_params is a reference to a hash where the following keys are defined:
	model has a value which is a fbamodel_id
	workspace has a value which is a workspace_id
	reaction has a value which is a reference to a list where each element is a reaction_id
	direction has a value which is a reference to a list where each element is a string
	compartment has a value which is a reference to a list where each element is a compartment_id
	compartmentIndex has a value which is a reference to a list where each element is an int
	gpr has a value which is a reference to a list where each element is a string
	removeReaction has a value which is a bool
	addReaction has a value which is a bool
	overwrite has a value which is a bool
	auth has a value which is a string
fbamodel_id is a string
workspace_id is a string
reaction_id is a string
compartment_id is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_ref is a string</pre>
   </dd>
   <dt id="Description20">
    Description
   </dt>
   <dd>
    <p>
     Enables the manual addition of a reaction to model
    </p>
   </dd>
  </dl>
  <h2 id="adjust_biomass_reaction">
   adjust_biomass_reaction
  </h2>
  <pre><code>  $modelMeta = $obj-&gt;adjust_biomass_reaction($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types21">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is an adjust_biomass_reaction_params
$modelMeta is an object_metadata
adjust_biomass_reaction_params is a reference to a hash where the following keys are defined:
	model has a value which is a fbamodel_id
	workspace has a value which is a workspace_id
	biomass has a value which is a biomass_id
	coefficient has a value which is a float
	compound has a value which is a compound_id
	compartment has a value which is a compartment_id
	compartmentIndex has a value which is an int
	overwrite has a value which is a bool
	auth has a value which is a string
fbamodel_id is a string
workspace_id is a string
biomass_id is a string
compound_id is a string
compartment_id is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_ref is a string</pre>
   </dd>
   <dt id="Description21">
    Description
   </dt>
   <dd>
    <p>
     Enables the manual adjustment of model biomass reaction
    </p>
   </dd>
  </dl>
  <h2 id="addmedia">
   addmedia
  </h2>
  <pre><code>  $mediaMeta = $obj-&gt;addmedia($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types22">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is an addmedia_params
$mediaMeta is an object_metadata
addmedia_params is a reference to a hash where the following keys are defined:
	media has a value which is a media_id
	workspace has a value which is a workspace_id
	name has a value which is a string
	isDefined has a value which is a bool
	isMinimal has a value which is a bool
	type has a value which is a string
	compounds has a value which is a reference to a list where each element is a string
	concentrations has a value which is a reference to a list where each element is a float
	maxflux has a value which is a reference to a list where each element is a float
	minflux has a value which is a reference to a list where each element is a float
	overwrite has a value which is a bool
	auth has a value which is a string
media_id is a string
workspace_id is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_ref is a string</pre>
   </dd>
   <dt id="Description22">
    Description
   </dt>
   <dd>
    <p>
     Add media condition to workspace
    </p>
   </dd>
  </dl>
  <h2 id="export_media">
   export_media
  </h2>
  <pre><code>  $output = $obj-&gt;export_media($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types23">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is an export_media_params
$output is a string
export_media_params is a reference to a hash where the following keys are defined:
	media has a value which is a media_id
	workspace has a value which is a workspace_id
	format has a value which is a string
	auth has a value which is a string
media_id is a string
workspace_id is a string</pre>
   </dd>
   <dt id="Description23">
    Description
   </dt>
   <dd>
    <p>
     Exports media in specified format (html,readable)
    </p>
   </dd>
  </dl>
  <h2 id="runfba">
   runfba
  </h2>
  <pre><code>  $fbaMeta = $obj-&gt;runfba($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types24">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a runfba_params
$fbaMeta is an object_metadata
runfba_params is a reference to a hash where the following keys are defined:
	model has a value which is a fbamodel_id
	model_workspace has a value which is a workspace_id
	formulation has a value which is an FBAFormulation
	fva has a value which is a bool
	simulateko has a value which is a bool
	minimizeflux has a value which is a bool
	findminmedia has a value which is a bool
	notes has a value which is a string
	fba has a value which is a fba_id
	workspace has a value which is a workspace_id
	auth has a value which is a string
	overwrite has a value which is a bool
	add_to_model has a value which is a bool
fbamodel_id is a string
workspace_id is a string
FBAFormulation is a reference to a hash where the following keys are defined:
	media has a value which is a media_id
	additionalcpds has a value which is a reference to a list where each element is a compound_id
	prommodel has a value which is a prommodel_id
	prommodel_workspace has a value which is a workspace_id
	media_workspace has a value which is a workspace_id
	objfraction has a value which is a float
	allreversible has a value which is a bool
	maximizeObjective has a value which is a bool
	objectiveTerms has a value which is a reference to a list where each element is a term
	geneko has a value which is a reference to a list where each element is a feature_id
	rxnko has a value which is a reference to a list where each element is a reaction_id
	bounds has a value which is a reference to a list where each element is a bound
	constraints has a value which is a reference to a list where each element is a constraint
	uptakelim has a value which is a reference to a hash where the key is a string and the value is a float
	defaultmaxflux has a value which is a float
	defaultminuptake has a value which is a float
	defaultmaxuptake has a value which is a float
	simplethermoconst has a value which is a bool
	thermoconst has a value which is a bool
	nothermoerror has a value which is a bool
	minthermoerror has a value which is a bool
media_id is a string
compound_id is a string
prommodel_id is a string
term is a reference to a list containing 3 items:
	0: (coefficient) a float
	1: (varType) a string
	2: (variable) a string
feature_id is a string
reaction_id is a string
bound is a reference to a list containing 4 items:
	0: (min) a float
	1: (max) a float
	2: (varType) a string
	3: (variable) a string
constraint is a reference to a list containing 4 items:
	0: (rhs) a float
	1: (sign) a string
	2: (terms) a reference to a list where each element is a term
	3: (name) a string
fba_id is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_ref is a string</pre>
   </dd>
   <dt id="Description24">
    Description
   </dt>
   <dd>
    <p>
     Run flux balance analysis and return ID of FBA object with results
    </p>
   </dd>
  </dl>
  <h2 id="export_fba">
   export_fba
  </h2>
  <pre><code>  $output = $obj-&gt;export_fba($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types25">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is an export_fba_params
$output is a string
export_fba_params is a reference to a hash where the following keys are defined:
	fba has a value which is a fba_id
	workspace has a value which is a workspace_id
	format has a value which is a string
	auth has a value which is a string
fba_id is a string
workspace_id is a string</pre>
   </dd>
   <dt id="Description25">
    Description
   </dt>
   <dd>
    <p>
     Export an FBA solution for viewing
    </p>
   </dd>
  </dl>
  <h2 id="import_phenotypes">
   import_phenotypes
  </h2>
  <pre><code>  $output = $obj-&gt;import_phenotypes($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types26">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is an import_phenotypes_params
$output is an object_metadata
import_phenotypes_params is a reference to a hash where the following keys are defined:
	phenotypeSet has a value which is a phenotypeSet_id
	workspace has a value which is a workspace_id
	genome has a value which is a genome_id
	genome_workspace has a value which is a workspace_id
	phenotypes has a value which is a reference to a list where each element is a Phenotype
	ignore_errors has a value which is a bool
	auth has a value which is a string
phenotypeSet_id is a string
workspace_id is a string
genome_id is a string
Phenotype is a reference to a list containing 5 items:
	0: (geneKO) a reference to a list where each element is a feature_id
	1: (baseMedia) a media_id
	2: (media_workspace) a workspace_id
	3: (additionalCpd) a reference to a list where each element is a compound_id
	4: (normalizedGrowth) a float
feature_id is a string
media_id is a string
compound_id is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_ref is a string</pre>
   </dd>
   <dt id="Description26">
    Description
   </dt>
   <dd>
    <p>
     Loads the specified phenotypes into the workspace
    </p>
   </dd>
  </dl>
  <h2 id="simulate_phenotypes">
   simulate_phenotypes
  </h2>
  <pre><code>  $output = $obj-&gt;simulate_phenotypes($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types27">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a simulate_phenotypes_params
$output is an object_metadata
simulate_phenotypes_params is a reference to a hash where the following keys are defined:
	model has a value which is a fbamodel_id
	model_workspace has a value which is a workspace_id
	phenotypeSet has a value which is a phenotypeSet_id
	phenotypeSet_workspace has a value which is a workspace_id
	formulation has a value which is an FBAFormulation
	notes has a value which is a string
	phenotypeSimultationSet has a value which is a phenotypeSimulationSet_id
	workspace has a value which is a workspace_id
	overwrite has a value which is a bool
	auth has a value which is a string
fbamodel_id is a string
workspace_id is a string
phenotypeSet_id is a string
FBAFormulation is a reference to a hash where the following keys are defined:
	media has a value which is a media_id
	additionalcpds has a value which is a reference to a list where each element is a compound_id
	prommodel has a value which is a prommodel_id
	prommodel_workspace has a value which is a workspace_id
	media_workspace has a value which is a workspace_id
	objfraction has a value which is a float
	allreversible has a value which is a bool
	maximizeObjective has a value which is a bool
	objectiveTerms has a value which is a reference to a list where each element is a term
	geneko has a value which is a reference to a list where each element is a feature_id
	rxnko has a value which is a reference to a list where each element is a reaction_id
	bounds has a value which is a reference to a list where each element is a bound
	constraints has a value which is a reference to a list where each element is a constraint
	uptakelim has a value which is a reference to a hash where the key is a string and the value is a float
	defaultmaxflux has a value which is a float
	defaultminuptake has a value which is a float
	defaultmaxuptake has a value which is a float
	simplethermoconst has a value which is a bool
	thermoconst has a value which is a bool
	nothermoerror has a value which is a bool
	minthermoerror has a value which is a bool
media_id is a string
compound_id is a string
prommodel_id is a string
term is a reference to a list containing 3 items:
	0: (coefficient) a float
	1: (varType) a string
	2: (variable) a string
feature_id is a string
reaction_id is a string
bound is a reference to a list containing 4 items:
	0: (min) a float
	1: (max) a float
	2: (varType) a string
	3: (variable) a string
constraint is a reference to a list containing 4 items:
	0: (rhs) a float
	1: (sign) a string
	2: (terms) a reference to a list where each element is a term
	3: (name) a string
phenotypeSimulationSet_id is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_ref is a string</pre>
   </dd>
   <dt id="Description27">
    Description
   </dt>
   <dd>
    <p>
     Simulates the specified phenotype set
    </p>
   </dd>
  </dl>
  <h2 id="export_phenotypeSimulationSet">
   export_phenotypeSimulationSet
  </h2>
  <pre><code>  $output = $obj-&gt;export_phenotypeSimulationSet($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types28">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is an export_phenotypeSimulationSet_params
$output is a string
export_phenotypeSimulationSet_params is a reference to a hash where the following keys are defined:
	phenotypeSimulationSet has a value which is a phenotypeSimulationSet_id
	workspace has a value which is a workspace_id
	format has a value which is a string
	auth has a value which is a string
phenotypeSimulationSet_id is a string
workspace_id is a string</pre>
   </dd>
   <dt id="Description28">
    Description
   </dt>
   <dd>
    <p>
     Export a PhenotypeSimulationSet for viewing
    </p>
   </dd>
  </dl>
  <h2 id="integrate_reconciliation_solutions">
   integrate_reconciliation_solutions
  </h2>
  <pre><code>  $modelMeta = $obj-&gt;integrate_reconciliation_solutions($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types29">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is an integrate_reconciliation_solutions_params
$modelMeta is an object_metadata
integrate_reconciliation_solutions_params is a reference to a hash where the following keys are defined:
	model has a value which is a fbamodel_id
	model_workspace has a value which is a workspace_id
	gapfillSolutions has a value which is a reference to a list where each element is a gapfillsolution_id
	gapgenSolutions has a value which is a reference to a list where each element is a gapgensolution_id
	out_model has a value which is a fbamodel_id
	workspace has a value which is a workspace_id
	auth has a value which is a string
	overwrite has a value which is a bool
fbamodel_id is a string
workspace_id is a string
gapfillsolution_id is a string
gapgensolution_id is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_ref is a string</pre>
   </dd>
   <dt id="Description29">
    Description
   </dt>
   <dd>
    <p>
     Integrates the specified gapfill and gapgen solutions into the specified model
    </p>
   </dd>
  </dl>
  <h2 id="queue_runfba">
   queue_runfba
  </h2>
  <pre><code>  $job = $obj-&gt;queue_runfba($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types30">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a queue_runfba_params
$job is a JobObject
queue_runfba_params is a reference to a hash where the following keys are defined:
	model has a value which is a fbamodel_id
	model_workspace has a value which is a workspace_id
	formulation has a value which is an FBAFormulation
	fva has a value which is a bool
	simulateko has a value which is a bool
	minimizeflux has a value which is a bool
	findminmedia has a value which is a bool
	notes has a value which is a string
	fba has a value which is a fba_id
	workspace has a value which is a workspace_id
	auth has a value which is a string
	overwrite has a value which is a bool
	add_to_model has a value which is a bool
fbamodel_id is a string
workspace_id is a string
FBAFormulation is a reference to a hash where the following keys are defined:
	media has a value which is a media_id
	additionalcpds has a value which is a reference to a list where each element is a compound_id
	prommodel has a value which is a prommodel_id
	prommodel_workspace has a value which is a workspace_id
	media_workspace has a value which is a workspace_id
	objfraction has a value which is a float
	allreversible has a value which is a bool
	maximizeObjective has a value which is a bool
	objectiveTerms has a value which is a reference to a list where each element is a term
	geneko has a value which is a reference to a list where each element is a feature_id
	rxnko has a value which is a reference to a list where each element is a reaction_id
	bounds has a value which is a reference to a list where each element is a bound
	constraints has a value which is a reference to a list where each element is a constraint
	uptakelim has a value which is a reference to a hash where the key is a string and the value is a float
	defaultmaxflux has a value which is a float
	defaultminuptake has a value which is a float
	defaultmaxuptake has a value which is a float
	simplethermoconst has a value which is a bool
	thermoconst has a value which is a bool
	nothermoerror has a value which is a bool
	minthermoerror has a value which is a bool
media_id is a string
compound_id is a string
prommodel_id is a string
term is a reference to a list containing 3 items:
	0: (coefficient) a float
	1: (varType) a string
	2: (variable) a string
feature_id is a string
reaction_id is a string
bound is a reference to a list containing 4 items:
	0: (min) a float
	1: (max) a float
	2: (varType) a string
	3: (variable) a string
constraint is a reference to a list containing 4 items:
	0: (rhs) a float
	1: (sign) a string
	2: (terms) a reference to a list where each element is a term
	3: (name) a string
fba_id is a string
JobObject is a reference to a hash where the following keys are defined:
	id has a value which is a job_id
	type has a value which is a string
	auth has a value which is a string
	status has a value which is a string
	jobdata has a value which is a reference to a hash where the key is a string and the value is a string
	queuetime has a value which is a string
	starttime has a value which is a string
	completetime has a value which is a string
	owner has a value which is a string
	queuecommand has a value which is a string
job_id is a string</pre>
   </dd>
   <dt id="Description30">
    Description
   </dt>
   <dd>
    <p>
     Queues an FBA job in a single media condition
    </p>
   </dd>
  </dl>
  <h2 id="queue_gapfill_model">
   queue_gapfill_model
  </h2>
  <pre><code>  $job = $obj-&gt;queue_gapfill_model($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types31">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a gapfill_model_params
$job is a JobObject
gapfill_model_params is a reference to a hash where the following keys are defined:
	model has a value which is a fbamodel_id
	model_workspace has a value which is a workspace_id
	formulation has a value which is a GapfillingFormulation
	phenotypeSet has a value which is a phenotypeSet_id
	phenotypeSet_workspace has a value which is a workspace_id
	integrate_solution has a value which is a bool
	out_model has a value which is a fbamodel_id
	workspace has a value which is a workspace_id
	gapFill has a value which is a gapfill_id
	timePerSolution has a value which is an int
	totalTimeLimit has a value which is an int
	auth has a value which is a string
	overwrite has a value which is a bool
	completeGapfill has a value which is a bool
fbamodel_id is a string
workspace_id is a string
GapfillingFormulation is a reference to a hash where the following keys are defined:
	formulation has a value which is an FBAFormulation
	num_solutions has a value which is an int
	nomediahyp has a value which is a bool
	nobiomasshyp has a value which is a bool
	nogprhyp has a value which is a bool
	nopathwayhyp has a value which is a bool
	allowunbalanced has a value which is a bool
	activitybonus has a value which is a float
	drainpen has a value which is a float
	directionpen has a value which is a float
	nostructpen has a value which is a float
	unfavorablepen has a value which is a float
	nodeltagpen has a value which is a float
	biomasstranspen has a value which is a float
	singletranspen has a value which is a float
	transpen has a value which is a float
	blacklistedrxns has a value which is a reference to a list where each element is a reaction_id
	gauranteedrxns has a value which is a reference to a list where each element is a reaction_id
	allowedcmps has a value which is a reference to a list where each element is a compartment_id
	probabilisticAnnotation has a value which is a probanno_id
	probabilisticAnnotation_workspace has a value which is a workspace_id
FBAFormulation is a reference to a hash where the following keys are defined:
	media has a value which is a media_id
	additionalcpds has a value which is a reference to a list where each element is a compound_id
	prommodel has a value which is a prommodel_id
	prommodel_workspace has a value which is a workspace_id
	media_workspace has a value which is a workspace_id
	objfraction has a value which is a float
	allreversible has a value which is a bool
	maximizeObjective has a value which is a bool
	objectiveTerms has a value which is a reference to a list where each element is a term
	geneko has a value which is a reference to a list where each element is a feature_id
	rxnko has a value which is a reference to a list where each element is a reaction_id
	bounds has a value which is a reference to a list where each element is a bound
	constraints has a value which is a reference to a list where each element is a constraint
	uptakelim has a value which is a reference to a hash where the key is a string and the value is a float
	defaultmaxflux has a value which is a float
	defaultminuptake has a value which is a float
	defaultmaxuptake has a value which is a float
	simplethermoconst has a value which is a bool
	thermoconst has a value which is a bool
	nothermoerror has a value which is a bool
	minthermoerror has a value which is a bool
media_id is a string
compound_id is a string
prommodel_id is a string
term is a reference to a list containing 3 items:
	0: (coefficient) a float
	1: (varType) a string
	2: (variable) a string
feature_id is a string
reaction_id is a string
bound is a reference to a list containing 4 items:
	0: (min) a float
	1: (max) a float
	2: (varType) a string
	3: (variable) a string
constraint is a reference to a list containing 4 items:
	0: (rhs) a float
	1: (sign) a string
	2: (terms) a reference to a list where each element is a term
	3: (name) a string
compartment_id is a string
probanno_id is a string
phenotypeSet_id is a string
gapfill_id is a string
JobObject is a reference to a hash where the following keys are defined:
	id has a value which is a job_id
	type has a value which is a string
	auth has a value which is a string
	status has a value which is a string
	jobdata has a value which is a reference to a hash where the key is a string and the value is a string
	queuetime has a value which is a string
	starttime has a value which is a string
	completetime has a value which is a string
	owner has a value which is a string
	queuecommand has a value which is a string
job_id is a string</pre>
   </dd>
   <dt id="Description31">
    Description
   </dt>
   <dd>
    <p>
     Queues an FBAModel gapfilling job in single media condition
    </p>
   </dd>
  </dl>
  <h2 id="queue_gapgen_model">
   queue_gapgen_model
  </h2>
  <pre><code>  $job = $obj-&gt;queue_gapgen_model($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types32">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a gapgen_model_params
$job is a JobObject
gapgen_model_params is a reference to a hash where the following keys are defined:
	model has a value which is a fbamodel_id
	model_workspace has a value which is a workspace_id
	formulation has a value which is a GapgenFormulation
	phenotypeSet has a value which is a phenotypeSet_id
	phenotypeSet_workspace has a value which is a workspace_id
	integrate_solution has a value which is a bool
	out_model has a value which is a fbamodel_id
	workspace has a value which is a workspace_id
	gapGen has a value which is a gapgen_id
	auth has a value which is a string
	timePerSolution has a value which is an int
	totalTimeLimit has a value which is an int
	overwrite has a value which is a bool
fbamodel_id is a string
workspace_id is a string
GapgenFormulation is a reference to a hash where the following keys are defined:
	formulation has a value which is an FBAFormulation
	refmedia has a value which is a media_id
	refmedia_workspace has a value which is a workspace_id
	num_solutions has a value which is an int
	nomediahyp has a value which is a bool
	nobiomasshyp has a value which is a bool
	nogprhyp has a value which is a bool
	nopathwayhyp has a value which is a bool
FBAFormulation is a reference to a hash where the following keys are defined:
	media has a value which is a media_id
	additionalcpds has a value which is a reference to a list where each element is a compound_id
	prommodel has a value which is a prommodel_id
	prommodel_workspace has a value which is a workspace_id
	media_workspace has a value which is a workspace_id
	objfraction has a value which is a float
	allreversible has a value which is a bool
	maximizeObjective has a value which is a bool
	objectiveTerms has a value which is a reference to a list where each element is a term
	geneko has a value which is a reference to a list where each element is a feature_id
	rxnko has a value which is a reference to a list where each element is a reaction_id
	bounds has a value which is a reference to a list where each element is a bound
	constraints has a value which is a reference to a list where each element is a constraint
	uptakelim has a value which is a reference to a hash where the key is a string and the value is a float
	defaultmaxflux has a value which is a float
	defaultminuptake has a value which is a float
	defaultmaxuptake has a value which is a float
	simplethermoconst has a value which is a bool
	thermoconst has a value which is a bool
	nothermoerror has a value which is a bool
	minthermoerror has a value which is a bool
media_id is a string
compound_id is a string
prommodel_id is a string
term is a reference to a list containing 3 items:
	0: (coefficient) a float
	1: (varType) a string
	2: (variable) a string
feature_id is a string
reaction_id is a string
bound is a reference to a list containing 4 items:
	0: (min) a float
	1: (max) a float
	2: (varType) a string
	3: (variable) a string
constraint is a reference to a list containing 4 items:
	0: (rhs) a float
	1: (sign) a string
	2: (terms) a reference to a list where each element is a term
	3: (name) a string
phenotypeSet_id is a string
gapgen_id is a string
JobObject is a reference to a hash where the following keys are defined:
	id has a value which is a job_id
	type has a value which is a string
	auth has a value which is a string
	status has a value which is a string
	jobdata has a value which is a reference to a hash where the key is a string and the value is a string
	queuetime has a value which is a string
	starttime has a value which is a string
	completetime has a value which is a string
	owner has a value which is a string
	queuecommand has a value which is a string
job_id is a string</pre>
   </dd>
   <dt id="Description32">
    Description
   </dt>
   <dd>
    <p>
     Queues an FBAModel gapfilling job in single media condition
    </p>
   </dd>
  </dl>
  <h2 id="queue_wildtype_phenotype_reconciliation">
   queue_wildtype_phenotype_reconciliation
  </h2>
  <pre><code>  $job = $obj-&gt;queue_wildtype_phenotype_reconciliation($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types33">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a wildtype_phenotype_reconciliation_params
$job is a JobObject
wildtype_phenotype_reconciliation_params is a reference to a hash where the following keys are defined:
	model has a value which is a fbamodel_id
	model_workspace has a value which is a workspace_id
	fba_formulation has a value which is an FBAFormulation
	gapfill_formulation has a value which is a GapfillingFormulation
	gapgen_formulation has a value which is a GapgenFormulation
	phenotypeSet has a value which is a phenotypeSet_id
	phenotypeSet_workspace has a value which is a workspace_id
	out_model has a value which is a fbamodel_id
	workspace has a value which is a workspace_id
	gapFills has a value which is a reference to a list where each element is a gapfill_id
	gapGens has a value which is a reference to a list where each element is a gapgen_id
	queueSensitivityAnalysis has a value which is a bool
	queueReconciliationCombination has a value which is a bool
	auth has a value which is a string
	overwrite has a value which is a bool
fbamodel_id is a string
workspace_id is a string
FBAFormulation is a reference to a hash where the following keys are defined:
	media has a value which is a media_id
	additionalcpds has a value which is a reference to a list where each element is a compound_id
	prommodel has a value which is a prommodel_id
	prommodel_workspace has a value which is a workspace_id
	media_workspace has a value which is a workspace_id
	objfraction has a value which is a float
	allreversible has a value which is a bool
	maximizeObjective has a value which is a bool
	objectiveTerms has a value which is a reference to a list where each element is a term
	geneko has a value which is a reference to a list where each element is a feature_id
	rxnko has a value which is a reference to a list where each element is a reaction_id
	bounds has a value which is a reference to a list where each element is a bound
	constraints has a value which is a reference to a list where each element is a constraint
	uptakelim has a value which is a reference to a hash where the key is a string and the value is a float
	defaultmaxflux has a value which is a float
	defaultminuptake has a value which is a float
	defaultmaxuptake has a value which is a float
	simplethermoconst has a value which is a bool
	thermoconst has a value which is a bool
	nothermoerror has a value which is a bool
	minthermoerror has a value which is a bool
media_id is a string
compound_id is a string
prommodel_id is a string
term is a reference to a list containing 3 items:
	0: (coefficient) a float
	1: (varType) a string
	2: (variable) a string
feature_id is a string
reaction_id is a string
bound is a reference to a list containing 4 items:
	0: (min) a float
	1: (max) a float
	2: (varType) a string
	3: (variable) a string
constraint is a reference to a list containing 4 items:
	0: (rhs) a float
	1: (sign) a string
	2: (terms) a reference to a list where each element is a term
	3: (name) a string
GapfillingFormulation is a reference to a hash where the following keys are defined:
	formulation has a value which is an FBAFormulation
	num_solutions has a value which is an int
	nomediahyp has a value which is a bool
	nobiomasshyp has a value which is a bool
	nogprhyp has a value which is a bool
	nopathwayhyp has a value which is a bool
	allowunbalanced has a value which is a bool
	activitybonus has a value which is a float
	drainpen has a value which is a float
	directionpen has a value which is a float
	nostructpen has a value which is a float
	unfavorablepen has a value which is a float
	nodeltagpen has a value which is a float
	biomasstranspen has a value which is a float
	singletranspen has a value which is a float
	transpen has a value which is a float
	blacklistedrxns has a value which is a reference to a list where each element is a reaction_id
	gauranteedrxns has a value which is a reference to a list where each element is a reaction_id
	allowedcmps has a value which is a reference to a list where each element is a compartment_id
	probabilisticAnnotation has a value which is a probanno_id
	probabilisticAnnotation_workspace has a value which is a workspace_id
compartment_id is a string
probanno_id is a string
GapgenFormulation is a reference to a hash where the following keys are defined:
	formulation has a value which is an FBAFormulation
	refmedia has a value which is a media_id
	refmedia_workspace has a value which is a workspace_id
	num_solutions has a value which is an int
	nomediahyp has a value which is a bool
	nobiomasshyp has a value which is a bool
	nogprhyp has a value which is a bool
	nopathwayhyp has a value which is a bool
phenotypeSet_id is a string
gapfill_id is a string
gapgen_id is a string
JobObject is a reference to a hash where the following keys are defined:
	id has a value which is a job_id
	type has a value which is a string
	auth has a value which is a string
	status has a value which is a string
	jobdata has a value which is a reference to a hash where the key is a string and the value is a string
	queuetime has a value which is a string
	starttime has a value which is a string
	completetime has a value which is a string
	owner has a value which is a string
	queuecommand has a value which is a string
job_id is a string</pre>
   </dd>
   <dt id="Description33">
    Description
   </dt>
   <dd>
    <p>
     Queues an FBAModel reconciliation job
    </p>
   </dd>
  </dl>
  <h2 id="queue_reconciliation_sensitivity_analysis">
   queue_reconciliation_sensitivity_analysis
  </h2>
  <pre><code>  $job = $obj-&gt;queue_reconciliation_sensitivity_analysis($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types34">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a wildtype_phenotype_reconciliation_params
$job is a JobObject
wildtype_phenotype_reconciliation_params is a reference to a hash where the following keys are defined:
	model has a value which is a fbamodel_id
	model_workspace has a value which is a workspace_id
	fba_formulation has a value which is an FBAFormulation
	gapfill_formulation has a value which is a GapfillingFormulation
	gapgen_formulation has a value which is a GapgenFormulation
	phenotypeSet has a value which is a phenotypeSet_id
	phenotypeSet_workspace has a value which is a workspace_id
	out_model has a value which is a fbamodel_id
	workspace has a value which is a workspace_id
	gapFills has a value which is a reference to a list where each element is a gapfill_id
	gapGens has a value which is a reference to a list where each element is a gapgen_id
	queueSensitivityAnalysis has a value which is a bool
	queueReconciliationCombination has a value which is a bool
	auth has a value which is a string
	overwrite has a value which is a bool
fbamodel_id is a string
workspace_id is a string
FBAFormulation is a reference to a hash where the following keys are defined:
	media has a value which is a media_id
	additionalcpds has a value which is a reference to a list where each element is a compound_id
	prommodel has a value which is a prommodel_id
	prommodel_workspace has a value which is a workspace_id
	media_workspace has a value which is a workspace_id
	objfraction has a value which is a float
	allreversible has a value which is a bool
	maximizeObjective has a value which is a bool
	objectiveTerms has a value which is a reference to a list where each element is a term
	geneko has a value which is a reference to a list where each element is a feature_id
	rxnko has a value which is a reference to a list where each element is a reaction_id
	bounds has a value which is a reference to a list where each element is a bound
	constraints has a value which is a reference to a list where each element is a constraint
	uptakelim has a value which is a reference to a hash where the key is a string and the value is a float
	defaultmaxflux has a value which is a float
	defaultminuptake has a value which is a float
	defaultmaxuptake has a value which is a float
	simplethermoconst has a value which is a bool
	thermoconst has a value which is a bool
	nothermoerror has a value which is a bool
	minthermoerror has a value which is a bool
media_id is a string
compound_id is a string
prommodel_id is a string
term is a reference to a list containing 3 items:
	0: (coefficient) a float
	1: (varType) a string
	2: (variable) a string
feature_id is a string
reaction_id is a string
bound is a reference to a list containing 4 items:
	0: (min) a float
	1: (max) a float
	2: (varType) a string
	3: (variable) a string
constraint is a reference to a list containing 4 items:
	0: (rhs) a float
	1: (sign) a string
	2: (terms) a reference to a list where each element is a term
	3: (name) a string
GapfillingFormulation is a reference to a hash where the following keys are defined:
	formulation has a value which is an FBAFormulation
	num_solutions has a value which is an int
	nomediahyp has a value which is a bool
	nobiomasshyp has a value which is a bool
	nogprhyp has a value which is a bool
	nopathwayhyp has a value which is a bool
	allowunbalanced has a value which is a bool
	activitybonus has a value which is a float
	drainpen has a value which is a float
	directionpen has a value which is a float
	nostructpen has a value which is a float
	unfavorablepen has a value which is a float
	nodeltagpen has a value which is a float
	biomasstranspen has a value which is a float
	singletranspen has a value which is a float
	transpen has a value which is a float
	blacklistedrxns has a value which is a reference to a list where each element is a reaction_id
	gauranteedrxns has a value which is a reference to a list where each element is a reaction_id
	allowedcmps has a value which is a reference to a list where each element is a compartment_id
	probabilisticAnnotation has a value which is a probanno_id
	probabilisticAnnotation_workspace has a value which is a workspace_id
compartment_id is a string
probanno_id is a string
GapgenFormulation is a reference to a hash where the following keys are defined:
	formulation has a value which is an FBAFormulation
	refmedia has a value which is a media_id
	refmedia_workspace has a value which is a workspace_id
	num_solutions has a value which is an int
	nomediahyp has a value which is a bool
	nobiomasshyp has a value which is a bool
	nogprhyp has a value which is a bool
	nopathwayhyp has a value which is a bool
phenotypeSet_id is a string
gapfill_id is a string
gapgen_id is a string
JobObject is a reference to a hash where the following keys are defined:
	id has a value which is a job_id
	type has a value which is a string
	auth has a value which is a string
	status has a value which is a string
	jobdata has a value which is a reference to a hash where the key is a string and the value is a string
	queuetime has a value which is a string
	starttime has a value which is a string
	completetime has a value which is a string
	owner has a value which is a string
	queuecommand has a value which is a string
job_id is a string</pre>
   </dd>
   <dt id="Description34">
    Description
   </dt>
   <dd>
    <p>
     Queues an FBAModel reconciliation job
    </p>
   </dd>
  </dl>
  <h2 id="queue_combine_wildtype_phenotype_reconciliation">
   queue_combine_wildtype_phenotype_reconciliation
  </h2>
  <pre><code>  $job = $obj-&gt;queue_combine_wildtype_phenotype_reconciliation($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types35">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a combine_wildtype_phenotype_reconciliation_params
$job is a JobObject
combine_wildtype_phenotype_reconciliation_params is a reference to a hash where the following keys are defined:
	model has a value which is a fbamodel_id
	model_workspace has a value which is a workspace_id
	fba_formulation has a value which is an FBAFormulation
	gapfill_formulation has a value which is a GapfillingFormulation
	gapgen_formulation has a value which is a GapgenFormulation
	phenotypeSet has a value which is a phenotypeSet_id
	phenotypeSet_workspace has a value which is a workspace_id
	out_model has a value which is a fbamodel_id
	workspace has a value which is a workspace_id
	gapFills has a value which is a reference to a list where each element is a gapfill_id
	gapGens has a value which is a reference to a list where each element is a gapgen_id
	auth has a value which is a string
	overwrite has a value which is a bool
fbamodel_id is a string
workspace_id is a string
FBAFormulation is a reference to a hash where the following keys are defined:
	media has a value which is a media_id
	additionalcpds has a value which is a reference to a list where each element is a compound_id
	prommodel has a value which is a prommodel_id
	prommodel_workspace has a value which is a workspace_id
	media_workspace has a value which is a workspace_id
	objfraction has a value which is a float
	allreversible has a value which is a bool
	maximizeObjective has a value which is a bool
	objectiveTerms has a value which is a reference to a list where each element is a term
	geneko has a value which is a reference to a list where each element is a feature_id
	rxnko has a value which is a reference to a list where each element is a reaction_id
	bounds has a value which is a reference to a list where each element is a bound
	constraints has a value which is a reference to a list where each element is a constraint
	uptakelim has a value which is a reference to a hash where the key is a string and the value is a float
	defaultmaxflux has a value which is a float
	defaultminuptake has a value which is a float
	defaultmaxuptake has a value which is a float
	simplethermoconst has a value which is a bool
	thermoconst has a value which is a bool
	nothermoerror has a value which is a bool
	minthermoerror has a value which is a bool
media_id is a string
compound_id is a string
prommodel_id is a string
term is a reference to a list containing 3 items:
	0: (coefficient) a float
	1: (varType) a string
	2: (variable) a string
feature_id is a string
reaction_id is a string
bound is a reference to a list containing 4 items:
	0: (min) a float
	1: (max) a float
	2: (varType) a string
	3: (variable) a string
constraint is a reference to a list containing 4 items:
	0: (rhs) a float
	1: (sign) a string
	2: (terms) a reference to a list where each element is a term
	3: (name) a string
GapfillingFormulation is a reference to a hash where the following keys are defined:
	formulation has a value which is an FBAFormulation
	num_solutions has a value which is an int
	nomediahyp has a value which is a bool
	nobiomasshyp has a value which is a bool
	nogprhyp has a value which is a bool
	nopathwayhyp has a value which is a bool
	allowunbalanced has a value which is a bool
	activitybonus has a value which is a float
	drainpen has a value which is a float
	directionpen has a value which is a float
	nostructpen has a value which is a float
	unfavorablepen has a value which is a float
	nodeltagpen has a value which is a float
	biomasstranspen has a value which is a float
	singletranspen has a value which is a float
	transpen has a value which is a float
	blacklistedrxns has a value which is a reference to a list where each element is a reaction_id
	gauranteedrxns has a value which is a reference to a list where each element is a reaction_id
	allowedcmps has a value which is a reference to a list where each element is a compartment_id
	probabilisticAnnotation has a value which is a probanno_id
	probabilisticAnnotation_workspace has a value which is a workspace_id
compartment_id is a string
probanno_id is a string
GapgenFormulation is a reference to a hash where the following keys are defined:
	formulation has a value which is an FBAFormulation
	refmedia has a value which is a media_id
	refmedia_workspace has a value which is a workspace_id
	num_solutions has a value which is an int
	nomediahyp has a value which is a bool
	nobiomasshyp has a value which is a bool
	nogprhyp has a value which is a bool
	nopathwayhyp has a value which is a bool
phenotypeSet_id is a string
gapfill_id is a string
gapgen_id is a string
JobObject is a reference to a hash where the following keys are defined:
	id has a value which is a job_id
	type has a value which is a string
	auth has a value which is a string
	status has a value which is a string
	jobdata has a value which is a reference to a hash where the key is a string and the value is a string
	queuetime has a value which is a string
	starttime has a value which is a string
	completetime has a value which is a string
	owner has a value which is a string
	queuecommand has a value which is a string
job_id is a string</pre>
   </dd>
   <dt id="Description35">
    Description
   </dt>
   <dd>
    <p>
     Queues an FBAModel reconciliation job
    </p>
   </dd>
  </dl>
  <h2 id="jobs_done">
   jobs_done
  </h2>
  <pre><code>  $job = $obj-&gt;jobs_done($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types36">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a jobs_done_params
$job is a JobObject
jobs_done_params is a reference to a hash where the following keys are defined:
	job has a value which is a job_id
	auth has a value which is a string
job_id is a string
JobObject is a reference to a hash where the following keys are defined:
	id has a value which is a job_id
	type has a value which is a string
	auth has a value which is a string
	status has a value which is a string
	jobdata has a value which is a reference to a hash where the key is a string and the value is a string
	queuetime has a value which is a string
	starttime has a value which is a string
	completetime has a value which is a string
	owner has a value which is a string
	queuecommand has a value which is a string</pre>
   </dd>
   <dt id="Description36">
    Description
   </dt>
   <dd>
    <p>
     Mark specified job as complete and run postprocessing
    </p>
   </dd>
  </dl>
  <h2 id="run_job">
   run_job
  </h2>
  <pre><code>  $job = $obj-&gt;run_job($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types37">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a run_job_params
$job is a JobObject
run_job_params is a reference to a hash where the following keys are defined:
	job has a value which is a job_id
	auth has a value which is a string
job_id is a string
JobObject is a reference to a hash where the following keys are defined:
	id has a value which is a job_id
	type has a value which is a string
	auth has a value which is a string
	status has a value which is a string
	jobdata has a value which is a reference to a hash where the key is a string and the value is a string
	queuetime has a value which is a string
	starttime has a value which is a string
	completetime has a value which is a string
	owner has a value which is a string
	queuecommand has a value which is a string</pre>
   </dd>
   <dt id="Description37">
    Description
   </dt>
   <dd>
    <p>
     Runs specified job
    </p>
   </dd>
  </dl>
  <h2 id="set_cofactors">
   set_cofactors
  </h2>
  <pre><code>  $output = $obj-&gt;set_cofactors($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types38">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a set_cofactors_params
$output is an object_metadata
set_cofactors_params is a reference to a hash where the following keys are defined:
	cofactors has a value which is a reference to a list where each element is a compound_id
	biochemistry has a value which is a biochemistry_id
	biochemistry_workspace has a value which is a workspace_id
	reset has a value which is a bool
	overwrite has a value which is a bool
	auth has a value which is a string
compound_id is a string
biochemistry_id is a string
workspace_id is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_ref is a string</pre>
   </dd>
   <dt id="Description38">
    Description
   </dt>
   <dd>
   </dd>
  </dl>
  <h2 id="find_reaction_synonyms">
   find_reaction_synonyms
  </h2>
  <pre><code>  $output = $obj-&gt;find_reaction_synonyms($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types39">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is a find_reaction_synonyms_params
$output is an object_metadata
find_reaction_synonyms_params is a reference to a hash where the following keys are defined:
	reaction_synonyms has a value which is a reaction_synonyms_id
	workspace has a value which is a workspace_id
	biochemistry has a value which is a biochemistry_id
	biochemistry_workspace has a value which is a workspace_id
	overwrite has a value which is a bool
	auth has a value which is a string
reaction_synonyms_id is a string
workspace_id is a string
biochemistry_id is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_ref is a string</pre>
   </dd>
   <dt id="Description39">
    Description
   </dt>
   <dd>
   </dd>
  </dl>
  <h2 id="role_to_reactions">
   role_to_reactions
  </h2>
  <pre><code>  $output = $obj-&gt;role_to_reactions($params)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types40">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$params is a role_to_reactions_params
$output is a reference to a list where each element is a RoleComplexReactions
role_to_reactions_params is a reference to a hash where the following keys are defined:
	templateModel has a value which is a template_id
	workspace has a value which is a workspace_id
	auth has a value which is a string
template_id is a string
workspace_id is a string
RoleComplexReactions is a reference to a hash where the following keys are defined:
	role has a value which is a role_id
	name has a value which is a string
	complexes has a value which is a reference to a list where each element is a ComplexReactions
role_id is a string
ComplexReactions is a reference to a hash where the following keys are defined:
	complex has a value which is a complex_id
	name has a value which is a string
	reactions has a value which is a reference to a list where each element is a TemplateReactions
complex_id is a string
TemplateReactions is a reference to a hash where the following keys are defined:
	reaction has a value which is a reaction_id
	direction has a value which is a string
	equation has a value which is a string
	compartment has a value which is a compartment_id
reaction_id is a string
compartment_id is a string</pre>
   </dd>
   <dt id="Description40">
    Description
   </dt>
   <dd>
    <p>
     Retrieves a list of roles mapped to reactions based on input template model
    </p>
   </dd>
  </dl>
  <h2 id="fasta_to_contigs">
   fasta_to_contigs
  </h2>
  <pre><code>  $output = $obj-&gt;fasta_to_contigs($params)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types41">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$params is a fasta_to_contigs_params
$output is an object_metadata
fasta_to_contigs_params is a reference to a hash where the following keys are defined:
	contigid has a value which is a string
	fasta has a value which is a string
	workspace has a value which is a workspace_id
	auth has a value which is a string
	source has a value which is a string
	genetic_code has a value which is a string
	domain has a value which is a string
	scientific_name has a value which is a string
workspace_id is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_ref is a string</pre>
   </dd>
   <dt id="Description41">
    Description
   </dt>
   <dd>
    <p>
     Loads a fasta file as a Contigs object in the workspace
    </p>
   </dd>
  </dl>
  <h2 id="contigs_to_genome">
   contigs_to_genome
  </h2>
  <pre><code>  $job = $obj-&gt;contigs_to_genome($params)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types42">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$params is a contigs_to_genome_params
$job is a JobObject
contigs_to_genome_params is a reference to a hash where the following keys are defined:
	contigid has a value which is a string
	contig_workspace has a value which is a workspace_id
	workspace has a value which is a workspace_id
	genomeid has a value which is a string
	auth has a value which is a string
workspace_id is a string
JobObject is a reference to a hash where the following keys are defined:
	id has a value which is a job_id
	type has a value which is a string
	auth has a value which is a string
	status has a value which is a string
	jobdata has a value which is a reference to a hash where the key is a string and the value is a string
	queuetime has a value which is a string
	starttime has a value which is a string
	completetime has a value which is a string
	owner has a value which is a string
	queuecommand has a value which is a string
job_id is a string</pre>
   </dd>
   <dt id="Description42">
    Description
   </dt>
   <dd>
    <p>
     Annotates contigs object creating a genome object
    </p>
   </dd>
  </dl>
  <h2 id="get_mapping">
   get_mapping
  </h2>
  <pre><code>  $output = $obj-&gt;get_mapping($params)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types43">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$params is a get_mapping_params
$output is a Mapping
get_mapping_params is a reference to a hash where the following keys are defined:
	map has a value which is a mapping_id
	workspace has a value which is a workspace_id
	auth has a value which is a string
mapping_id is a string
workspace_id is a string
Mapping is a reference to a hash where the following keys are defined:
	id has a value which is a mapping_id
	name has a value which is a string
	subsystems has a value which is a reference to a list where each element is a Subsystem
	roles has a value which is a reference to a list where each element is a FunctionalRole
	complexes has a value which is a reference to a list where each element is a Complex
Subsystem is a reference to a hash where the following keys are defined:
	id has a value which is a subsystem_id
	name has a value which is a string
	class has a value which is a string
	subclass has a value which is a string
	type has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	roles has a value which is a reference to a list where each element is a role_id
subsystem_id is a string
role_id is a string
FunctionalRole is a reference to a hash where the following keys are defined:
	id has a value which is a role_id
	name has a value which is a string
	feature has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	complexes has a value which is a reference to a list where each element is a complex_id
complex_id is a string
Complex is a reference to a hash where the following keys are defined:
	id has a value which is a complex_id
	name has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	roles has a value which is a reference to a list where each element is a ComplexRole
ComplexRole is a reference to a list containing 4 items:
	0: (id) a role_id
	1: (roleType) a string
	2: (optional) a bool
	3: (triggering) a bool</pre>
   </dd>
   <dt id="Description43">
    Description
   </dt>
   <dd>
    <p>
     Annotates contigs object creating a genome object
    </p>
   </dd>
  </dl>
  <h2 id="adjust_mapping_role">
   adjust_mapping_role
  </h2>
  <pre><code>  $output = $obj-&gt;adjust_mapping_role($params)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types44">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$params is an adjust_mapping_role_params
$output is a FunctionalRole
adjust_mapping_role_params is a reference to a hash where the following keys are defined:
	map has a value which is a mapping_id
	workspace has a value which is a workspace_id
	role has a value which is a string
	new has a value which is a bool
	name has a value which is a string
	feature has a value which is a string
	aliasesToAdd has a value which is a reference to a list where each element is a string
	aliasesToRemove has a value which is a reference to a list where each element is a string
	delete has a value which is a bool
	auth has a value which is a string
mapping_id is a string
workspace_id is a string
FunctionalRole is a reference to a hash where the following keys are defined:
	id has a value which is a role_id
	name has a value which is a string
	feature has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	complexes has a value which is a reference to a list where each element is a complex_id
role_id is a string
complex_id is a string</pre>
   </dd>
   <dt id="Description44">
    Description
   </dt>
   <dd>
    <p>
     An API function supporting the curation of functional roles in a mapping object
    </p>
   </dd>
  </dl>
  <h2 id="adjust_mapping_complex">
   adjust_mapping_complex
  </h2>
  <pre><code>  $output = $obj-&gt;adjust_mapping_complex($params)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types45">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$params is an adjust_mapping_complex_params
$output is a Complex
adjust_mapping_complex_params is a reference to a hash where the following keys are defined:
	map has a value which is a mapping_id
	workspace has a value which is a workspace_id
	complex has a value which is a string
	new has a value which is a bool
	name has a value which is a string
	rolesToAdd has a value which is a reference to a list where each element is a string
	rolesToRemove has a value which is a reference to a list where each element is a string
	delete has a value which is a bool
	auth has a value which is a string
mapping_id is a string
workspace_id is a string
Complex is a reference to a hash where the following keys are defined:
	id has a value which is a complex_id
	name has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	roles has a value which is a reference to a list where each element is a ComplexRole
complex_id is a string
ComplexRole is a reference to a list containing 4 items:
	0: (id) a role_id
	1: (roleType) a string
	2: (optional) a bool
	3: (triggering) a bool
role_id is a string</pre>
   </dd>
   <dt id="Description45">
    Description
   </dt>
   <dd>
    <p>
     An API function supporting the curation of complexes in a mapping object
    </p>
   </dd>
  </dl>
  <h2 id="adjust_mapping_subsystem">
   adjust_mapping_subsystem
  </h2>
  <pre><code>  $output = $obj-&gt;adjust_mapping_subsystem($params)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types46">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$params is an adjust_mapping_subsystem_params
$output is a Subsystem
adjust_mapping_subsystem_params is a reference to a hash where the following keys are defined:
	map has a value which is a mapping_id
	workspace has a value which is a workspace_id
	subsystem has a value which is a string
	new has a value which is a bool
	name has a value which is a string
	type has a value which is a string
	class has a value which is a string
	subclass has a value which is a string
	rolesToAdd has a value which is a reference to a list where each element is a string
	rolesToRemove has a value which is a reference to a list where each element is a string
	delete has a value which is a bool
	auth has a value which is a string
mapping_id is a string
workspace_id is a string
Subsystem is a reference to a hash where the following keys are defined:
	id has a value which is a subsystem_id
	name has a value which is a string
	class has a value which is a string
	subclass has a value which is a string
	type has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	roles has a value which is a reference to a list where each element is a role_id
subsystem_id is a string
role_id is a string</pre>
   </dd>
   <dt id="Description46">
    Description
   </dt>
   <dd>
    <p>
     An API function supporting the curation of subsystems in a mapping object
    </p>
   </dd>
  </dl>
  <h2 id="get_template_model">
   get_template_model
  </h2>
  <pre><code>  $output = $obj-&gt;get_template_model($params)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types47">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$params is a get_template_model_params
$output is a TemplateModel
get_template_model_params is a reference to a hash where the following keys are defined:
	templateModel has a value which is a template_id
	workspace has a value which is a workspace_id
	auth has a value which is a string
template_id is a string
workspace_id is a string
TemplateModel is a reference to a hash where the following keys are defined:
	id has a value which is a template_id
	name has a value which is a string
	type has a value which is a string
	domain has a value which is a string
	map has a value which is a mapping_id
	mappingws has a value which is a workspace_id
	reactions has a value which is a reference to a list where each element is a TemplateReaction
	biomasses has a value which is a reference to a list where each element is a TemplateBiomass
mapping_id is a string
TemplateReaction is a reference to a hash where the following keys are defined:
	id has a value which is a temprxn_id
	compartment has a value which is a compartment_id
	reaction has a value which is a reaction_id
	complexes has a value which is a reference to a list where each element is a complex_id
	direction has a value which is a string
	type has a value which is a string
temprxn_id is a string
compartment_id is a string
reaction_id is a string
complex_id is a string
TemplateBiomass is a reference to a hash where the following keys are defined:
	id has a value which is a tempbiomass_id
	name has a value which is a string
	type has a value which is a string
	other has a value which is a string
	protein has a value which is a string
	dna has a value which is a string
	rna has a value which is a string
	cofactor has a value which is a string
	energy has a value which is a string
	cellwall has a value which is a string
	lipid has a value which is a string
	compounds has a value which is a reference to a list where each element is a TemplateBiomassCompounds
tempbiomass_id is a string
TemplateBiomassCompounds is a reference to a list containing 7 items:
	0: (compound) a compound_id
	1: (compartment) a compartment_id
	2: (class) a string
	3: (universal) a string
	4: (coefficientType) a string
	5: (coefficient) a string
	6: (linkedCompounds) a reference to a list where each element is a reference to a list containing 2 items:
		0: (coeffficient) a string
		1: (compound) a compound_idcompound_id is a string</pre>
   </dd>
   <dt id="Description47">
    Description
   </dt>
   <dd>
    <p>
     Retrieves the specified template model
    </p>
   </dd>
  </dl>
  <h2 id="import_template_fbamodel">
   import_template_fbamodel
  </h2>
  <pre><code>  $modelMeta = $obj-&gt;import_template_fbamodel($input)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types48">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$input is an import_template_fbamodel_params
$modelMeta is an object_metadata
import_template_fbamodel_params is a reference to a hash where the following keys are defined:
	map has a value which is a mapping_id
	mapping_workspace has a value which is a workspace_id
	templateReactions has a value which is a reference to a list where each element is a reference to a list containing 5 items:
	0: (id) a string
	1: (compartment) a string
	2: (direction) a string
	3: (type) a string
	4: (complexes) a reference to a list where each element is a string	templateBiomass has a value which is a reference to a list where each element is a reference to a list containing 11 items:
	0: (name) a string
	1: (type) a string
	2: (dna) a float
	3: (rna) a float
	4: (protein) a float
	5: (lipid) a float
	6: (cellwall) a float
	7: (cofactor) a float
	8: (energy) a float
	9: (other) a float
	10: (compounds) a reference to a list where each element is a reference to a list containing 6 items:
		0: (id) a string
		1: (compartment) a string
		2: (class) a string
		3: (coefficientType) a string
		4: (coefficient) a float
		5: (conditions) a string


	name has a value which is a string
	modelType has a value which is a string
	domain has a value which is a string
	id has a value which is a template_id
	workspace has a value which is a workspace_id
	ignore_errors has a value which is a bool
	auth has a value which is a string
mapping_id is a string
workspace_id is a string
template_id is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_ref is a string
</pre>
   </dd>
   <dt id="Description48">
    Description
   </dt>
   <dd>
    <p>
     Import a template model from an input table of template reactions and biomass components
    </p>
   </dd>
  </dl>
  <h2 id="adjust_template_reaction">
   adjust_template_reaction
  </h2>
  <pre><code>  $output = $obj-&gt;adjust_template_reaction($params)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types49">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$params is an adjust_template_reaction_params
$output is a TemplateReaction
adjust_template_reaction_params is a reference to a hash where the following keys are defined:
	templateModel has a value which is a template_id
	workspace has a value which is a workspace_id
	reaction has a value which is a string
	clearComplexes has a value which is a bool
	new has a value which is a bool
	delete has a value which is a bool
	compartment has a value which is a compartment_id
	complexesToAdd has a value which is a reference to a list where each element is a complex_id
	complexesToRemove has a value which is a reference to a list where each element is a complex_id
	direction has a value which is a string
	type has a value which is a string
	auth has a value which is a string
template_id is a string
workspace_id is a string
compartment_id is a string
complex_id is a string
TemplateReaction is a reference to a hash where the following keys are defined:
	id has a value which is a temprxn_id
	compartment has a value which is a compartment_id
	reaction has a value which is a reaction_id
	complexes has a value which is a reference to a list where each element is a complex_id
	direction has a value which is a string
	type has a value which is a string
temprxn_id is a string
reaction_id is a string</pre>
   </dd>
   <dt id="Description49">
    Description
   </dt>
   <dd>
    <p>
     Modifies a reaction of a template model
    </p>
   </dd>
  </dl>
  <h2 id="adjust_template_biomass">
   adjust_template_biomass
  </h2>
  <pre><code>  $output = $obj-&gt;adjust_template_biomass($params)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types50">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$params is an adjust_template_biomass_params
$output is a TemplateBiomass
adjust_template_biomass_params is a reference to a hash where the following keys are defined:
	templateModel has a value which is a template_id
	workspace has a value which is a workspace_id
	biomass has a value which is a string
	new has a value which is a bool
	delete has a value which is a bool
	clearBiomassCompounds has a value which is a bool
	name has a value which is a string
	type has a value which is a string
	other has a value which is a string
	protein has a value which is a string
	dna has a value which is a string
	rna has a value which is a string
	cofactor has a value which is a string
	energy has a value which is a string
	cellwall has a value which is a string
	lipid has a value which is a string
	compoundsToRemove has a value which is a reference to a list where each element is a reference to a list containing 2 items:
	0: (compound) a compound_id
	1: (compartment) a compartment_id	compoundsToAdd has a value which is a reference to a list where each element is a reference to a list containing 7 items:
	0: (compound) a compound_id
	1: (compartment) a compartment_id
	2: (class) a string
	3: (universal) a string
	4: (coefficientType) a string
	5: (coefficient) a string
	6: (linkedCompounds) a reference to a list where each element is a reference to a list containing 2 items:
		0: (coeffficient) a string
		1: (compound) a compound_id


	auth has a value which is a string
template_id is a string
workspace_id is a string
compound_id is a string
compartment_id is a string
TemplateBiomass is a reference to a hash where the following keys are defined:
	id has a value which is a tempbiomass_id
	name has a value which is a string
	type has a value which is a string
	other has a value which is a string
	protein has a value which is a string
	dna has a value which is a string
	rna has a value which is a string
	cofactor has a value which is a string
	energy has a value which is a string
	cellwall has a value which is a string
	lipid has a value which is a string
	compounds has a value which is a reference to a list where each element is a TemplateBiomassCompounds
tempbiomass_id is a string
TemplateBiomassCompounds is a reference to a list containing 7 items:
	0: (compound) a compound_id
	1: (compartment) a compartment_id
	2: (class) a string
	3: (universal) a string
	4: (coefficientType) a string
	5: (coefficient) a string
	6: (linkedCompounds) a reference to a list where each element is a reference to a list containing 2 items:
		0: (coeffficient) a string
		1: (compound) a compound_id

</pre>
   </dd>
   <dt id="Description50">
    Description
   </dt>
   <dd>
    <p>
     Modifies the biomass of a template model
    </p>
   </dd>
  </dl>
  <h2 id="add_stimuli">
   add_stimuli
  </h2>
  <pre><code>  $output = $obj-&gt;add_stimuli($params)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types51">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$params is an add_stimuli_params
$output is an object_metadata
add_stimuli_params is a reference to a hash where the following keys are defined:
	biochemid has a value which is a string
	biochem_workspace has a value which is a string
	stimuliid has a value which is a string
	name has a value which is a string
	abbreviation has a value which is a string
	type has a value which is a string
	description has a value which is a string
	compounds has a value which is a reference to a list where each element is a string
	workspace has a value which is a string
	auth has a value which is a string
object_metadata is a reference to a list containing 11 items:
	0: (id) an object_id
	1: (type) an object_type
	2: (moddate) a timestamp
	3: (instance) an int
	4: (command) a string
	5: (lastmodifier) a username
	6: (owner) a username
	7: (workspace) a workspace_id
	8: (ref) a workspace_ref
	9: (chsum) a string
	10: (metadata) a reference to a hash where the key is a string and the value is a string
object_id is a string
object_type is a string
timestamp is a string
username is a string
workspace_id is a string
workspace_ref is a string</pre>
   </dd>
   <dt id="Description51">
    Description
   </dt>
   <dd>
    <p>
     Adds a stimuli either to the central database or as an object in a workspace
    </p>
   </dd>
  </dl>
  <h1 id="TYPES">
   TYPES
  </h1>
  <h2 id="workspace_id">
   workspace_id
  </h2>
  <dl>
   <dt id="Description52">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** Universal simple type definitions ********************************************************************************
    </p>
   </dd>
   <dt id="Definition">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="complex_id">
   complex_id
  </h2>
  <dl>
   <dt id="Description53">
    Description
   </dt>
   <dd>
    <p>
     A string used as an ID for a complex.
    </p>
   </dd>
   <dt id="Definition1">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="template_id">
   template_id
  </h2>
  <dl>
   <dt id="Description54">
    Description
   </dt>
   <dd>
    <p>
     A string used as an ID for a complex.
    </p>
   </dd>
   <dt id="Definition2">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="role_id">
   role_id
  </h2>
  <dl>
   <dt id="Description55">
    Description
   </dt>
   <dd>
    <p>
     A string used as an ID for a complex.
    </p>
   </dd>
   <dt id="Definition3">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="object_type">
   object_type
  </h2>
  <dl>
   <dt id="Description56">
    Description
   </dt>
   <dd>
    <p>
     A string indicating the "type" of an object stored in a workspace. Acceptable types are returned by the "get_types()" command in the workspace_service
    </p>
   </dd>
   <dt id="Definition4">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="object_id">
   object_id
  </h2>
  <dl>
   <dt id="Description57">
    Description
   </dt>
   <dd>
    <p>
     ID of an object stored in the workspace. Any string consisting of alphanumeric characters and "-" is acceptable
    </p>
   </dd>
   <dt id="Definition5">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="username">
   username
  </h2>
  <dl>
   <dt id="Description58">
    Description
   </dt>
   <dd>
    <p>
     Login name of KBase useraccount to which permissions for workspaces are mapped
    </p>
   </dd>
   <dt id="Definition6">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="timestamp">
   timestamp
  </h2>
  <dl>
   <dt id="Description59">
    Description
   </dt>
   <dd>
    <p>
     Exact time for workspace operations. e.g. 2012-12-17T23:24:06
    </p>
   </dd>
   <dt id="Definition7">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="compound_id">
   compound_id
  </h2>
  <dl>
   <dt id="Description60">
    Description
   </dt>
   <dd>
    <p>
     An identifier for compounds in the KBase biochemistry database. e.g. cpd00001
    </p>
   </dd>
   <dt id="Definition8">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="biochemistry_id">
   biochemistry_id
  </h2>
  <dl>
   <dt id="Description61">
    Description
   </dt>
   <dd>
    <p>
     A string used to identify a particular biochemistry database object in KBase. e.g. "default" is the ID of the standard KBase biochemistry
    </p>
   </dd>
   <dt id="Definition9">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="genome_id">
   genome_id
  </h2>
  <dl>
   <dt id="Description62">
    Description
   </dt>
   <dd>
    <p>
     A string identifier for a genome in KBase. e.g. "kb|g.0" is the ID for E. coli
    </p>
   </dd>
   <dt id="Definition10">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="prommodel_id">
   prommodel_id
  </h2>
  <dl>
   <dt id="Description63">
    Description
   </dt>
   <dd>
    <p>
     A string identifier for a prommodel in KBase.
    </p>
   </dd>
   <dt id="Definition11">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="contig_id">
   contig_id
  </h2>
  <dl>
   <dt id="Description64">
    Description
   </dt>
   <dd>
    <p>
     A string identifier for a contiguous piece of DNA in KBase, representing a chromosome or an assembled fragment
    </p>
   </dd>
   <dt id="Definition12">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="feature_type">
   feature_type
  </h2>
  <dl>
   <dt id="Description65">
    Description
   </dt>
   <dd>
    <p>
     A string specifying the type of genome features in KBase
    </p>
   </dd>
   <dt id="Definition13">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="modelcompartment_id">
   modelcompartment_id
  </h2>
  <dl>
   <dt id="Description66">
    Description
   </dt>
   <dd>
    <p>
     A string identifier used for compartments in models in KBase. Compartments could represet organelles in a eukaryotic model, or entire cells in a community model
    </p>
   </dd>
   <dt id="Definition14">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="modelcompound_id">
   modelcompound_id
  </h2>
  <dl>
   <dt id="Description67">
    Description
   </dt>
   <dd>
    <p>
     A string identifier used for compounds in models in KBase.
    </p>
   </dd>
   <dt id="Definition15">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="feature_id">
   feature_id
  </h2>
  <dl>
   <dt id="Description68">
    Description
   </dt>
   <dd>
    <p>
     A string identifier used for a feature in a genome.
    </p>
   </dd>
   <dt id="Definition16">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="reaction_id">
   reaction_id
  </h2>
  <dl>
   <dt id="Description69">
    Description
   </dt>
   <dd>
    <p>
     A string identifier used for a reaction in a KBase biochemistry.
    </p>
   </dd>
   <dt id="Definition17">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="modelreaction_id">
   modelreaction_id
  </h2>
  <dl>
   <dt id="Description70">
    Description
   </dt>
   <dd>
    <p>
     A string identifier used for a reaction in a model in KBase.
    </p>
   </dd>
   <dt id="Definition18">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="biomass_id">
   biomass_id
  </h2>
  <dl>
   <dt id="Description71">
    Description
   </dt>
   <dd>
    <p>
     A string identifier used for a biomass reaction in a KBase model.
    </p>
   </dd>
   <dt id="Definition19">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="media_id">
   media_id
  </h2>
  <dl>
   <dt id="Description72">
    Description
   </dt>
   <dd>
    <p>
     A string identifier used for a media condition in the KBase database.
    </p>
   </dd>
   <dt id="Definition20">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="fba_id">
   fba_id
  </h2>
  <dl>
   <dt id="Description73">
    Description
   </dt>
   <dd>
    <p>
     A string identifier used for a flux balance analysis study in KBase.
    </p>
   </dd>
   <dt id="Definition21">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="gapgen_id">
   gapgen_id
  </h2>
  <dl>
   <dt id="Description74">
    Description
   </dt>
   <dd>
    <p>
     A string identifier for a gap generation study in KBase.
    </p>
   </dd>
   <dt id="Definition22">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="gapfill_id">
   gapfill_id
  </h2>
  <dl>
   <dt id="Description75">
    Description
   </dt>
   <dd>
    <p>
     A string identifier for a gap filling study in KBase.
    </p>
   </dd>
   <dt id="Definition23">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="gapgensolution_id">
   gapgensolution_id
  </h2>
  <dl>
   <dt id="Description76">
    Description
   </dt>
   <dd>
    <p>
     A string identifier for a solution from a gap generation study in KBase.
    </p>
   </dd>
   <dt id="Definition24">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="gapfillsolution_id">
   gapfillsolution_id
  </h2>
  <dl>
   <dt id="Description77">
    Description
   </dt>
   <dd>
    <p>
     A string identifier for a solution from a gap filling study in KBase.
    </p>
   </dd>
   <dt id="Definition25">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="fbamodel_id">
   fbamodel_id
  </h2>
  <dl>
   <dt id="Description78">
    Description
   </dt>
   <dd>
    <p>
     A string identifier for a metabolic model in KBase.
    </p>
   </dd>
   <dt id="Definition26">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="mapping_id">
   mapping_id
  </h2>
  <dl>
   <dt id="Description79">
    Description
   </dt>
   <dd>
    <p>
     A string identifier for a Mapping object in KBase.
    </p>
   </dd>
   <dt id="Definition27">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="regmodel_id">
   regmodel_id
  </h2>
  <dl>
   <dt id="Description80">
    Description
   </dt>
   <dd>
    <p>
     A string identifier for a regulatory model in KBase.
    </p>
   </dd>
   <dt id="Definition28">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="compartment_id">
   compartment_id
  </h2>
  <dl>
   <dt id="Description81">
    Description
   </dt>
   <dd>
    <p>
     A string identifier for a compartment in KBase.
    </p>
   </dd>
   <dt id="Definition29">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="expression_id">
   expression_id
  </h2>
  <dl>
   <dt id="Description82">
    Description
   </dt>
   <dd>
    <p>
     A string identifier for an expression dataset in KBase.
    </p>
   </dd>
   <dt id="Definition30">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="phenotypeSet_id">
   phenotypeSet_id
  </h2>
  <dl>
   <dt id="Description83">
    Description
   </dt>
   <dd>
    <p>
     A string identifier used for a set of phenotype data loaded into KBase.
    </p>
   </dd>
   <dt id="Definition31">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="workspace_ref">
   workspace_ref
  </h2>
  <dl>
   <dt id="Description84">
    Description
   </dt>
   <dd>
    <p>
     A permanent reference to an object in a workspace.
    </p>
   </dd>
   <dt id="Definition32">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="probanno_id">
   probanno_id
  </h2>
  <dl>
   <dt id="Description85">
    Description
   </dt>
   <dd>
    <p>
     A string identifier used for a probabilistic annotation in KBase.
    </p>
   </dd>
   <dt id="Definition33">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="reaction_synonyms_id">
   reaction_synonyms_id
  </h2>
  <dl>
   <dt id="Description86">
    Description
   </dt>
   <dd>
    <p>
     A string identifier for a reaction synonyms in KBase.
    </p>
   </dd>
   <dt id="Definition34">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="object_metadata">
   object_metadata
  </h2>
  <dl>
   <dt id="Description87">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** Object type definition ********************************************************************************
    </p>
   </dd>
   <dt id="Definition35">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 11 items:
0: (id) an object_id
1: (type) an object_type
2: (moddate) a timestamp
3: (instance) an int
4: (command) a string
5: (lastmodifier) a username
6: (owner) a username
7: (workspace) a workspace_id
8: (ref) a workspace_ref
9: (chsum) a string
10: (metadata) a reference to a hash where the key is a string and the value is a string</pre>
   </dd>
  </dl>
  <h2 id="md5">
   md5
  </h2>
  <dl>
   <dt id="Description88">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** Probabilistic Annotation type definition ********************************************************************************
    </p>
   </dd>
   <dt id="Definition36">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="md5s">
   md5s
  </h2>
  <dl>
   <dt id="Definition37">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list where each element is a md5
</pre>
   </dd>
  </dl>
  <h2 id="region_of_dna">
   region_of_dna
  </h2>
  <dl>
   <dt id="Description89">
    Description
   </dt>
   <dd>
    <p>
     A region of DNA is maintained as a tuple of four components:
    </p>
    <pre><code>        the contig
        the beginning position (from 1)
        the strand
        the length

        We often speak of "a region".  By "location", we mean a sequence
        of regions from the same genome (perhaps from distinct contigs).</code></pre>
   </dd>
   <dt id="Definition38">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 4 items:
0: a contig_id
1: (begin) an int
2: (strand) a string
3: (length) an int</pre>
   </dd>
  </dl>
  <h2 id="location">
   location
  </h2>
  <dl>
   <dt id="Description90">
    Description
   </dt>
   <dd>
    <p>
     a "location" refers to a sequence of regions
    </p>
   </dd>
   <dt id="Definition39">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list where each element is a region_of_dna
</pre>
   </dd>
  </dl>
  <h2 id="annotation">
   annotation
  </h2>
  <dl>
   <dt id="Definition40">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 3 items:
0: (comment) a string
1: (annotator) a string
2: (annotation_time) an int</pre>
   </dd>
  </dl>
  <h2 id="gene_hit">
   gene_hit
  </h2>
  <dl>
   <dt id="Definition41">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 2 items:
0: (gene) a feature_id
1: (blast_score) a float</pre>
   </dd>
  </dl>
  <h2 id="alt_func">
   alt_func
  </h2>
  <dl>
   <dt id="Definition42">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 3 items:
0: (function) a string
1: (probability) a float
2: (gene_hits) a reference to a list where each element is a gene_hit</pre>
   </dd>
  </dl>
  <h2 id="feature">
   feature
  </h2>
  <dl>
   <dt id="Definition43">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a feature_id
location has a value which is a location
type has a value which is a feature_type
function has a value which is a string
alternative_functions has a value which is a reference to a list where each element is an alt_func
protein_translation has a value which is a string
aliases has a value which is a reference to a list where each element is a string
annotations has a value which is a reference to a list where each element is an annotation</pre>
   </dd>
  </dl>
  <h2 id="contig">
   contig
  </h2>
  <dl>
   <dt id="Definition44">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a contig_id
dna has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="GenomeObject">
   GenomeObject
  </h2>
  <dl>
   <dt id="Definition45">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a genome_id
scientific_name has a value which is a string
domain has a value which is a string
genetic_code has a value which is an int
source has a value which is a string
source_id has a value which is a string
contigs has a value which is a reference to a list where each element is a contig
features has a value which is a reference to a list where each element is a feature</pre>
   </dd>
  </dl>
  <h2 id="annotationProbability">
   annotationProbability
  </h2>
  <dl>
   <dt id="Description91">
    Description
   </dt>
   <dd>
    <p>
     Data structures to hold a single annotation probability for a single gene
    </p>
    <p>
     feature_id feature - feature the annotation is associated with string function - the name of the functional role being annotated to the feature float probability - the probability that the functional role is associated with the feature
    </p>
   </dd>
   <dt id="Definition46">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 3 items:
0: (feature) a feature_id
1: (function) a string
2: (probability) a float</pre>
   </dd>
  </dl>
  <h2 id="probanno_id1">
   probanno_id
  </h2>
  <dl>
   <dt id="Definition47">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="alt_func1">
   alt_func
  </h2>
  <dl>
   <dt id="Definition48">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 2 items:
0: (function) a string
1: (probability) a float</pre>
   </dd>
  </dl>
  <h2 id="ProbAnnoFeature">
   ProbAnnoFeature
  </h2>
  <dl>
   <dt id="Description92">
    Description
   </dt>
   <dd>
    <p>
     Object to carry alternative functions for each feature
    </p>
    <p>
     feature_id id ID of the feature. Required.
    </p>
    <p>
     string function Primary annotated function of the feature in the genome annotation. Required.
    </p>
    <p>
     list&lt;alt_func&gt; alternative_functions List of tuples containing alternative functions and probabilities. Required.
    </p>
   </dd>
   <dt id="Definition49">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a feature_id
alternative_functions has a value which is a reference to a list where each element is an alt_func</pre>
   </dd>
  </dl>
  <h2 id="ProbabilisticAnnotation">
   ProbabilisticAnnotation
  </h2>
  <dl>
   <dt id="Description93">
    Description
   </dt>
   <dd>
    <p>
     Object to carry alternative functions and probabilities for genes in a genome
    </p>
    <pre><code>    probanno_id id - ID of the probabilistic annotation object. Required.    
    genome_id genome - ID of the genome the probabilistic annotation was built for. Required.
    workspace_ref genome_uuid - Reference to retrieve genome from workspace service. Required.
    list&lt;ProbAnnoFeature&gt; featureAlternativeFunctions - List of ProbAnnoFeature objects holding alternative functions for features. Required.</code></pre>
   </dd>
   <dt id="Definition50">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a probanno_id
genome has a value which is a genome_id
genome_uuid has a value which is a workspace_ref
featureAlternativeFunctions has a value which is a reference to a list where each element is a ProbAnnoFeature</pre>
   </dd>
  </dl>
  <h2 id="ReactionProbability">
   ReactionProbability
  </h2>
  <dl>
   <dt id="Description94">
    Description
   </dt>
   <dd>
    <p>
     Data structure to hold probability of a reaction
    </p>
    <pre><code>        reaction_id reaction - ID of the reaction
        float probability - Probability of the reaction
        string gene_list - List of genes most likely to be attached to reaction</code></pre>
   </dd>
   <dt id="Definition51">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 3 items:
0: (reaction) a reaction_id
1: (probability) a float
2: (gene_list) a string</pre>
   </dd>
  </dl>
  <h2 id="Biochemistry">
   Biochemistry
  </h2>
  <dl>
   <dt id="Description95">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** Biochemistry type definition ********************************************************************************
    </p>
   </dd>
   <dt id="Definition52">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a biochemistry_id
name has a value which is a string
compounds has a value which is a reference to a list where each element is a compound_id
reactions has a value which is a reference to a list where each element is a reaction_id
media has a value which is a reference to a list where each element is a media_id</pre>
   </dd>
  </dl>
  <h2 id="MediaCompound">
   MediaCompound
  </h2>
  <dl>
   <dt id="Description96">
    Description
   </dt>
   <dd>
    <p>
     Data structures for media compound formulation
    </p>
    <p>
     compound_id compound - ID of compound in media string name - name of compound in media float concentration - concentration of compound in media float maxFlux - maximum flux of compound in media float minFlux - minimum flux of compound in media
    </p>
   </dd>
   <dt id="Definition53">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
compound has a value which is a compound_id
name has a value which is a string
concentration has a value which is a float
max_flux has a value which is a float
min_flux has a value which is a float</pre>
   </dd>
  </dl>
  <h2 id="Media">
   Media
  </h2>
  <dl>
   <dt id="Description97">
    Description
   </dt>
   <dd>
    <p>
     Data structures for media formulation
    </p>
    <p>
     media_id id - ID of media formulation string name - name of media formulaiton list&lt;MediaCompound&gt; media_compounds - list of compounds in media formulation float pH - pH of media condition float temperature - temperature of media condition
    </p>
   </dd>
   <dt id="Definition54">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a media_id
name has a value which is a string
media_compounds has a value which is a reference to a list where each element is a MediaCompound
pH has a value which is a float
temperature has a value which is a float</pre>
   </dd>
  </dl>
  <h2 id="Compound">
   Compound
  </h2>
  <dl>
   <dt id="Description98">
    Description
   </dt>
   <dd>
    <p>
     Data structures for media formulation
    </p>
    <p>
     compound_id id - ID of compound string abbrev - abbreviated name of compound string name - primary name of compound list&lt;string&gt; aliases - list of aliases for compound float charge - molecular charge of compound float deltaG - estimated compound delta G float deltaGErr - uncertainty in estimated compound delta G string formula - molecular formula of compound
    </p>
   </dd>
   <dt id="Definition55">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a compound_id
abbrev has a value which is a string
name has a value which is a string
aliases has a value which is a reference to a list where each element is a string
charge has a value which is a float
deltaG has a value which is a float
deltaGErr has a value which is a float
formula has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="Reaction">
   Reaction
  </h2>
  <dl>
   <dt id="Description99">
    Description
   </dt>
   <dd>
    <p>
     Data structures for media formulation
    </p>
    <p>
     reaction_id id - ID of reaction string name - primary name of reaction string abbrev - abbreviated name of reaction list&lt;string&gt; enzymes - list of EC numbers for reaction string direction - directionality of reaction string reversibility - reversibility of reaction float deltaG - estimated delta G of reaction float deltaGErr - uncertainty in estimated delta G of reaction string equation - reaction equation in terms of compound IDs string definition - reaction equation in terms of compound names
    </p>
   </dd>
   <dt id="Definition56">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a reaction_id
name has a value which is a string
abbrev has a value which is a string
enzymes has a value which is a reference to a list where each element is a string
direction has a value which is a string
reversibility has a value which is a string
deltaG has a value which is a float
deltaGErr has a value which is a float
equation has a value which is a string
definition has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="ModelCompartment">
   ModelCompartment
  </h2>
  <dl>
   <dt id="Description100">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** FBAModel type definition ********************************************************************************
    </p>
   </dd>
   <dt id="Definition57">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a modelcompartment_id
name has a value which is a string
pH has a value which is a float
potential has a value which is a float
index has a value which is an int</pre>
   </dd>
  </dl>
  <h2 id="ModelCompound">
   ModelCompound
  </h2>
  <dl>
   <dt id="Description101">
    Description
   </dt>
   <dd>
    <p>
     Data structures for a compound in a model
    </p>
    <p>
     modelcompound_id id - ID of the specific instance of the compound in the model compound_id compound - ID of the compound associated with the model compound string name - name of the compound associated with the model compound modelcompartment_id compartment - ID of the compartment containing the compound
    </p>
   </dd>
   <dt id="Definition58">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a modelcompound_id
compound has a value which is a compound_id
name has a value which is a string
compartment has a value which is a modelcompartment_id</pre>
   </dd>
  </dl>
  <h2 id="ModelReaction">
   ModelReaction
  </h2>
  <dl>
   <dt id="Description102">
    Description
   </dt>
   <dd>
    <p>
     Data structures for a reaction in a model
    </p>
    <p>
     modelreaction_id id - ID of the specific instance of the reaction in the model reaction_id reaction - ID of the reaction string name - name of the reaction string direction - directionality of the reaction string equation - stoichiometric equation of the reaction in terms of compound IDs string definition - stoichiometric equation of the reaction in terms of compound names list&lt;feature_id&gt; features - list of features associated with the reaction modelcompartment_id compartment - ID of the compartment containing the reaction
    </p>
   </dd>
   <dt id="Definition59">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a modelreaction_id
reaction has a value which is a reaction_id
name has a value which is a string
direction has a value which is a string
equation has a value which is a string
definition has a value which is a string
gapfilled has a value which is a bool
features has a value which is a reference to a list where each element is a feature_id
compartment has a value which is a modelcompartment_id</pre>
   </dd>
  </dl>
  <h2 id="BiomassCompound">
   BiomassCompound
  </h2>
  <dl>
   <dt id="Description103">
    Description
   </dt>
   <dd>
    <p>
     Data structures for a reaction in a model
    </p>
    <p>
     modelcompound_id modelcompound - ID of model compound in biomass reaction float coefficient - coefficient of compound in biomass reaction string name - name of compound in biomass reaction
    </p>
   </dd>
   <dt id="Definition60">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 3 items:
0: (modelcompound) a modelcompound_id
1: (coefficient) a float
2: (name) a string</pre>
   </dd>
  </dl>
  <h2 id="ModelBiomass">
   ModelBiomass
  </h2>
  <dl>
   <dt id="Description104">
    Description
   </dt>
   <dd>
    <p>
     Data structures for a reaction in a model
    </p>
    <p>
     biomass_id id - ID of biomass reaction string name - name of biomass reaction string definition - stoichiometric equation of biomass reaction in terms of compound names list&lt;BiomassCompound&gt; biomass_compounds - list of compounds in biomass reaction
    </p>
   </dd>
   <dt id="Definition61">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a biomass_id
name has a value which is a string
definition has a value which is a string
biomass_compounds has a value which is a reference to a list where each element is a BiomassCompound</pre>
   </dd>
  </dl>
  <h2 id="FBAMeta">
   FBAMeta
  </h2>
  <dl>
   <dt id="Description105">
    Description
   </dt>
   <dd>
    <p>
     Data structures for a reaction in a model
    </p>
    <p>
     fba_id id - ID of the FBA object workspace_id workspace - ID of the workspace containing the FBA object media_id media - ID of the media the FBA was performed in workspace_id media_workspace - ID of the workspace containing the media formulation float objective - optimized objective value of the FBA study list&lt;feature_id&gt; ko - list of genes knocked out in the FBA study
    </p>
   </dd>
   <dt id="Definition62">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 6 items:
0: (id) a fba_id
1: (workspace) a workspace_id
2: (media) a media_id
3: (media_workspace) a workspace_id
4: (objective) a float
5: (ko) a reference to a list where each element is a feature_id</pre>
   </dd>
  </dl>
  <h2 id="GapGenMeta">
   GapGenMeta
  </h2>
  <dl>
   <dt id="Description106">
    Description
   </dt>
   <dd>
    <p>
     Metadata object providing a summary of a gapgen simulation
    </p>
    <p>
     gapgen_id id - ID of gapgen study object workspace_id workspace - workspace containing gapgen study media_id media - media formulation for gapgen study workspace_id media_workspace - ID of the workspace containing the media formulation bool done - boolean indicating if gapgen study is complete list&lt;feature_id&gt; ko - list of genes knocked out in gapgen study
    </p>
   </dd>
   <dt id="Definition63">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 6 items:
0: (id) a gapgen_id
1: (workspace) a workspace_id
2: (media) a media_id
3: (media_workspace) a workspace_id
4: (done) a bool
5: (ko) a reference to a list where each element is a feature_id</pre>
   </dd>
  </dl>
  <h2 id="GapFillMeta">
   GapFillMeta
  </h2>
  <dl>
   <dt id="Description107">
    Description
   </dt>
   <dd>
    <p>
     Metadata object providing a summary of a gapfilling simulation
    </p>
    <p>
     gapfill_id id - ID of gapfill study object workspace_id workspace - workspace containing gapfill study media_id media - media formulation for gapfill study workspace_id media_workspace - ID of the workspace containing the media formulation bool done - boolean indicating if gapfill study is complete list&lt;feature_id&gt; ko - list of genes knocked out in gapfill study
    </p>
   </dd>
   <dt id="Definition64">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 6 items:
0: (id) a gapfill_id
1: (workspace) a workspace_id
2: (media) a media_id
3: (media_workspace) a workspace_id
4: (done) a bool
5: (ko) a reference to a list where each element is a feature_id</pre>
   </dd>
  </dl>
  <h2 id="Subsystem">
   Subsystem
  </h2>
  <dl>
   <dt id="Definition65">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
name has a value which is a string
feature has a value which is a reference to a list where each element is a feature_id</pre>
   </dd>
  </dl>
  <h2 id="FBAModel">
   FBAModel
  </h2>
  <dl>
   <dt id="Description108">
    Description
   </dt>
   <dd>
    <p>
     Data structure holding data for metabolic model
    </p>
    <p>
     fbamodel_id id - ID of model workspace_id workspace - workspace containing model genome_id genome - ID of associated genome workspace_id genome_workspace - workspace with associated genome mapping_id map - ID of associated mapping database workspace_id map_workspace - workspace with associated mapping database biochemistry_id biochemistry - ID of associated biochemistry database workspace_id biochemistry_workspace - workspace with associated biochemistry database string name - name of the model string type - type of model (e.g. single genome, community) string status - status of model (e.g. under construction) list&lt;ModelBiomass&gt; biomasses - list of biomass reactions in model list&lt;ModelCompartment&gt; compartments - list of compartments in model list&lt;ModelReaction&gt; reactions - list of reactions in model list&lt;ModelCompound&gt; compounds - list of compounds in model list&lt;FBAMeta&gt; fbas - list of flux balance analysis studies for model list&lt;GapFillMeta&gt; integrated_gapfillings - list of integrated gapfilling solutions list&lt;GapFillMeta&gt; unintegrated_gapfillings - list of unintegrated gapfilling solutions list&lt;GapGenMeta&gt; integrated_gapgenerations - list of integrated gapgen solutions list&lt;GapGenMeta&gt; unintegrated_gapgenerations - list of unintegrated gapgen solutions
    </p>
   </dd>
   <dt id="Definition66">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a fbamodel_id
workspace has a value which is a workspace_id
genome has a value which is a genome_id
genome_workspace has a value which is a workspace_id
map has a value which is a mapping_id
map_workspace has a value which is a workspace_id
biochemistry has a value which is a biochemistry_id
biochemistry_workspace has a value which is a workspace_id
name has a value which is a string
type has a value which is a string
status has a value which is a string
biomasses has a value which is a reference to a list where each element is a ModelBiomass
compartments has a value which is a reference to a list where each element is a ModelCompartment
reactions has a value which is a reference to a list where each element is a ModelReaction
compounds has a value which is a reference to a list where each element is a ModelCompound
fbas has a value which is a reference to a list where each element is an FBAMeta
integrated_gapfillings has a value which is a reference to a list where each element is a GapFillMeta
unintegrated_gapfillings has a value which is a reference to a list where each element is a GapFillMeta
integrated_gapgenerations has a value which is a reference to a list where each element is a GapGenMeta
unintegrated_gapgenerations has a value which is a reference to a list where each element is a GapGenMeta
modelSubsystems has a value which is a reference to a list where each element is a Subsystem</pre>
   </dd>
  </dl>
  <h2 id="GeneAssertion">
   GeneAssertion
  </h2>
  <dl>
   <dt id="Description109">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** Flux Balance Analysis type definition ********************************************************************************
    </p>
   </dd>
   <dt id="Definition67">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 4 items:
0: (feature) a feature_id
1: (growthFraction) a float
2: (growth) a float
3: (isEssential) a bool</pre>
   </dd>
  </dl>
  <h2 id="CompoundFlux">
   CompoundFlux
  </h2>
  <dl>
   <dt id="Description110">
    Description
   </dt>
   <dd>
    <p>
     Compound variable in FBA solution
    </p>
    <p>
     modelcompound_id compound - ID of compound in model in FBA solution float value - flux uptake of compound in FBA solution float upperBound - maximum uptake of compoundin FBA simulation float lowerBound - minimum uptake of compoundin FBA simulation float max - maximum uptake of compoundin FBA simulation float min - minimum uptake of compoundin FBA simulation string type - type of compound variable string name - name of compound
    </p>
   </dd>
   <dt id="Definition68">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 8 items:
0: (compound) a modelcompound_id
1: (value) a float
2: (upperBound) a float
3: (lowerBound) a float
4: (max) a float
5: (min) a float
6: (type) a string
7: (name) a string</pre>
   </dd>
  </dl>
  <h2 id="ReactionFlux">
   ReactionFlux
  </h2>
  <dl>
   <dt id="Description111">
    Description
   </dt>
   <dd>
    <p>
     Reaction variable in FBA solution
    </p>
    <p>
     modelreaction_id reaction - ID of reaction in model in FBA solution float value - flux through reaction in FBA solution float upperBound - maximum flux through reaction in FBA simulation float lowerBound - minimum flux through reaction in FBA simulation float max - maximum flux through reaction in FBA simulation float min - minimum flux through reaction in FBA simulation string type - type of reaction variable string definition - stoichiometry of solution reaction in terms of compound names
    </p>
   </dd>
   <dt id="Definition69">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 8 items:
0: (reaction) a modelreaction_id
1: (value) a float
2: (upperBound) a float
3: (lowerBound) a float
4: (max) a float
5: (min) a float
6: (type) a string
7: (definition) a string</pre>
   </dd>
  </dl>
  <h2 id="MetaboliteProduction">
   MetaboliteProduction
  </h2>
  <dl>
   <dt id="Description112">
    Description
   </dt>
   <dd>
    <p>
     Maximum production of compound in FBA simulation
    </p>
    <p>
     float maximumProduction - maximum production of compound modelcompound_id modelcompound - ID of compound with production maximized string name - name of compound with simulated production
    </p>
   </dd>
   <dt id="Definition70">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 3 items:
0: (maximumProduction) a float
1: (modelcompound) a modelcompound_id
2: (name) a string</pre>
   </dd>
  </dl>
  <h2 id="MinimalMediaPrediction">
   MinimalMediaPrediction
  </h2>
  <dl>
   <dt id="Description113">
    Description
   </dt>
   <dd>
    <p>
     Data structures for gapfilling solution
    </p>
    <p>
     list&lt;compound_id&gt; optionalNutrients - list of optional nutrients list&lt;compound_id&gt; essentialNutrients - list of essential nutrients
    </p>
   </dd>
   <dt id="Definition71">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
optionalNutrients has a value which is a reference to a list where each element is a compound_id
essentialNutrients has a value which is a reference to a list where each element is a compound_id</pre>
   </dd>
  </dl>
  <h2 id="bound">
   bound
  </h2>
  <dl>
   <dt id="Description114">
    Description
   </dt>
   <dd>
    <p>
     Term of constraint or objective in FBA simulation
    </p>
    <p>
     float min - minimum value of custom bound float max - maximum value of custom bound string varType - type of variable for custom bound string variable - variable ID for custom bound
    </p>
   </dd>
   <dt id="Definition72">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 4 items:
0: (min) a float
1: (max) a float
2: (varType) a string
3: (variable) a string</pre>
   </dd>
  </dl>
  <h2 id="term">
   term
  </h2>
  <dl>
   <dt id="Description115">
    Description
   </dt>
   <dd>
    <p>
     Term of constraint or objective in FBA simulation
    </p>
    <p>
     float coefficient - coefficient of term in objective or constraint string varType - type of variable for term in objective or constraint string variable - variable ID for term in objective or constraint
    </p>
   </dd>
   <dt id="Definition73">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 3 items:
0: (coefficient) a float
1: (varType) a string
2: (variable) a string</pre>
   </dd>
  </dl>
  <h2 id="constraint">
   constraint
  </h2>
  <dl>
   <dt id="Description116">
    Description
   </dt>
   <dd>
    <p>
     Custom constraint in FBA simulation
    </p>
    <p>
     float rhs - right hand side of custom constraint string sign - sign of custom constraint (e.g. &lt;, &gt;) list&lt;term&gt; terms - terms in custom constraint string name - name of custom constraint
    </p>
   </dd>
   <dt id="Definition74">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 4 items:
0: (rhs) a float
1: (sign) a string
2: (terms) a reference to a list where each element is a term
3: (name) a string</pre>
   </dd>
  </dl>
  <h2 id="FBAFormulation">
   FBAFormulation
  </h2>
  <dl>
   <dt id="Description117">
    Description
   </dt>
   <dd>
    <p>
     Data structures for gapfilling solution
    </p>
    <p>
     media_id media - ID of media formulation to be used list&lt;compound_id&gt; additionalcpds - list of additional compounds to allow update prommodel_id prommodel - ID of prommodel workspace_id prommodel_workspace - workspace containing prommodel workspace_id media_workspace - workspace containing media for FBA study float objfraction - fraction of objective to use for constraints bool allreversible - flag indicating if all reactions should be reversible bool maximizeObjective - flag indicating if objective should be maximized list&lt;term&gt; objectiveTerms - list of terms of objective function list&lt;feature_id&gt; geneko - list of gene knockouts list&lt;reaction_id&gt; rxnko - list of reaction knockouts list&lt;bound&gt; bounds - list of custom bounds list&lt;constraint&gt; constraints - list of custom constraints mapping&lt;string,float&gt; uptakelim - hash of maximum uptake for elements float defaultmaxflux - default maximum intracellular flux float defaultminuptake - default minimum nutrient uptake float defaultmaxuptake - default maximum nutrient uptake bool simplethermoconst - flag indicating if simple thermodynamic constraints should be used bool thermoconst - flag indicating if thermodynamic constraints should be used bool nothermoerror - flag indicating if no error should be allowed in thermodynamic constraints bool minthermoerror - flag indicating if error should be minimized in thermodynamic constraints
    </p>
   </dd>
   <dt id="Definition75">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
media has a value which is a media_id
additionalcpds has a value which is a reference to a list where each element is a compound_id
prommodel has a value which is a prommodel_id
prommodel_workspace has a value which is a workspace_id
media_workspace has a value which is a workspace_id
objfraction has a value which is a float
allreversible has a value which is a bool
maximizeObjective has a value which is a bool
objectiveTerms has a value which is a reference to a list where each element is a term
geneko has a value which is a reference to a list where each element is a feature_id
rxnko has a value which is a reference to a list where each element is a reaction_id
bounds has a value which is a reference to a list where each element is a bound
constraints has a value which is a reference to a list where each element is a constraint
uptakelim has a value which is a reference to a hash where the key is a string and the value is a float
defaultmaxflux has a value which is a float
defaultminuptake has a value which is a float
defaultmaxuptake has a value which is a float
simplethermoconst has a value which is a bool
thermoconst has a value which is a bool
nothermoerror has a value which is a bool
minthermoerror has a value which is a bool</pre>
   </dd>
  </dl>
  <h2 id="FBA">
   FBA
  </h2>
  <dl>
   <dt id="Description118">
    Description
   </dt>
   <dd>
    <p>
     Data structures for gapfilling solution
    </p>
    <p>
     fba_id id - ID of FBA study workspace_id workspace - workspace containing FBA study fbamodel_id model - ID of model FBA was run on workspace_id model_workspace - workspace with FBA model float objective - objective value of FBA study bool isComplete - flag indicating if job is complete FBAFormulation formulation - specs for FBA study list&lt;MinimalMediaPrediction&gt; minimalMediaPredictions - list of minimal media formulation list&lt;MetaboliteProduction&gt; metaboliteProductions - list of biomass component production list&lt;ReactionFlux&gt; reactionFluxes - list of reaction fluxes list&lt;CompoundFlux&gt; compoundFluxes - list of compound uptake fluxes list&lt;GeneAssertion&gt; geneAssertions - list of gene assertions
    </p>
   </dd>
   <dt id="Definition76">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a fba_id
workspace has a value which is a workspace_id
model has a value which is a fbamodel_id
model_workspace has a value which is a workspace_id
objective has a value which is a float
isComplete has a value which is a bool
formulation has a value which is an FBAFormulation
minimalMediaPredictions has a value which is a reference to a list where each element is a MinimalMediaPrediction
metaboliteProductions has a value which is a reference to a list where each element is a MetaboliteProduction
reactionFluxes has a value which is a reference to a list where each element is a ReactionFlux
compoundFluxes has a value which is a reference to a list where each element is a CompoundFlux
geneAssertions has a value which is a reference to a list where each element is a GeneAssertion</pre>
   </dd>
  </dl>
  <h2 id="GapfillingFormulation">
   GapfillingFormulation
  </h2>
  <dl>
   <dt id="Description119">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** Gapfilling type definition ********************************************************************************
    </p>
   </dd>
   <dt id="Definition77">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
formulation has a value which is an FBAFormulation
num_solutions has a value which is an int
nomediahyp has a value which is a bool
nobiomasshyp has a value which is a bool
nogprhyp has a value which is a bool
nopathwayhyp has a value which is a bool
allowunbalanced has a value which is a bool
activitybonus has a value which is a float
drainpen has a value which is a float
directionpen has a value which is a float
nostructpen has a value which is a float
unfavorablepen has a value which is a float
nodeltagpen has a value which is a float
biomasstranspen has a value which is a float
singletranspen has a value which is a float
transpen has a value which is a float
blacklistedrxns has a value which is a reference to a list where each element is a reaction_id
gauranteedrxns has a value which is a reference to a list where each element is a reaction_id
allowedcmps has a value which is a reference to a list where each element is a compartment_id
probabilisticAnnotation has a value which is a probanno_id
probabilisticAnnotation_workspace has a value which is a workspace_id</pre>
   </dd>
  </dl>
  <h2 id="reactionAddition">
   reactionAddition
  </h2>
  <dl>
   <dt id="Description120">
    Description
   </dt>
   <dd>
    <p>
     Reactions removed in gapgen solution
    </p>
    <p>
     modelreaction_id reaction - ID of the removed reaction string direction - direction of reaction removed in gapgen solution string equation - stoichiometry of removed reaction in terms of compound IDs string definition - stoichiometry of removed reaction in terms of compound names
    </p>
   </dd>
   <dt id="Definition78">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 5 items:
0: (reaction) a reaction_id
1: (direction) a string
2: (compartment_id) a string
3: (equation) a string
4: (definition) a string</pre>
   </dd>
  </dl>
  <h2 id="biomassRemoval">
   biomassRemoval
  </h2>
  <dl>
   <dt id="Description121">
    Description
   </dt>
   <dd>
    <p>
     Biomass component removed in gapfill solution
    </p>
    <p>
     compound_id compound - ID of biomass component removed string name - name of biomass component removed
    </p>
   </dd>
   <dt id="Definition79">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 2 items:
0: (compound) a compound_id
1: (name) a string</pre>
   </dd>
  </dl>
  <h2 id="mediaAddition">
   mediaAddition
  </h2>
  <dl>
   <dt id="Description122">
    Description
   </dt>
   <dd>
    <p>
     Media component added in gapfill solution
    </p>
    <p>
     compound_id compound - ID of media component added string name - name of media component added
    </p>
   </dd>
   <dt id="Definition80">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 2 items:
0: (compound) a compound_id
1: (name) a string</pre>
   </dd>
  </dl>
  <h2 id="GapFillSolution">
   GapFillSolution
  </h2>
  <dl>
   <dt id="Description123">
    Description
   </dt>
   <dd>
    <p>
     Data structures for gapfilling solution
    </p>
    <p>
     gapfillsolution_id id - ID of gapfilling solution float objective - cost of gapfilling solution list&lt;biomassRemoval&gt; biomassRemovals - list of biomass components being removed list&lt;mediaAddition&gt; mediaAdditions - list of media components being added list&lt;reactionAddition&gt; reactionAdditions - list of reactions being added
    </p>
   </dd>
   <dt id="Definition81">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a gapfillsolution_id
objective has a value which is a float
integrated has a value which is a bool
biomassRemovals has a value which is a reference to a list where each element is a biomassRemoval
mediaAdditions has a value which is a reference to a list where each element is a mediaAddition
reactionAdditions has a value which is a reference to a list where each element is a reactionAddition</pre>
   </dd>
  </dl>
  <h2 id="GapFill">
   GapFill
  </h2>
  <dl>
   <dt id="Description124">
    Description
   </dt>
   <dd>
    <p>
     Data structures for gapfilling analysis
    </p>
    <p>
     gapfill_id id - ID of gapfill analysis workspace_id workspace - workspace containing gapfill analysis fbamodel_id model - ID of model being gapfilled workspace_id model_workspace - workspace containing model bool isComplete - indicates if gapfilling is complete GapfillingFormulation formulation - formulation of gapfilling analysis list&lt;GapFillSolution&gt; solutions - list of gapfilling solutions
    </p>
   </dd>
   <dt id="Definition82">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a gapfill_id
workspace has a value which is a workspace_id
model has a value which is a fbamodel_id
model_workspace has a value which is a workspace_id
isComplete has a value which is a bool
formulation has a value which is a GapfillingFormulation
solutions has a value which is a reference to a list where each element is a GapFillSolution</pre>
   </dd>
  </dl>
  <h2 id="GapgenFormulation">
   GapgenFormulation
  </h2>
  <dl>
   <dt id="Description125">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** Gap Generation type definition ********************************************************************************
    </p>
   </dd>
   <dt id="Definition83">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
formulation has a value which is an FBAFormulation
refmedia has a value which is a media_id
refmedia_workspace has a value which is a workspace_id
num_solutions has a value which is an int
nomediahyp has a value which is a bool
nobiomasshyp has a value which is a bool
nogprhyp has a value which is a bool
nopathwayhyp has a value which is a bool</pre>
   </dd>
  </dl>
  <h2 id="reactionRemoval">
   reactionRemoval
  </h2>
  <dl>
   <dt id="Description126">
    Description
   </dt>
   <dd>
    <p>
     Reactions removed in gapgen solution
    </p>
    <p>
     modelreaction_id reaction - ID of the removed reaction string direction - direction of reaction removed in gapgen solution string equation - stoichiometry of removed reaction in terms of compound IDs string definition - stoichiometry of removed reaction in terms of compound names
    </p>
   </dd>
   <dt id="Definition84">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 4 items:
0: (reaction) a modelreaction_id
1: (direction) a string
2: (equation) a string
3: (definition) a string</pre>
   </dd>
  </dl>
  <h2 id="biomassAddition">
   biomassAddition
  </h2>
  <dl>
   <dt id="Description127">
    Description
   </dt>
   <dd>
    <p>
     Compounds added to biomass in gapgen solution
    </p>
    <p>
     compound_id compound - ID of biomass compound added string name - name of biomass compound added
    </p>
   </dd>
   <dt id="Definition85">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 2 items:
0: (compound) a compound_id
1: (name) a string</pre>
   </dd>
  </dl>
  <h2 id="mediaRemoval">
   mediaRemoval
  </h2>
  <dl>
   <dt id="Description128">
    Description
   </dt>
   <dd>
    <p>
     Media components removed in gapgen solution
    </p>
    <p>
     compound_id compound - ID of media component removed string name - name of media component removed
    </p>
   </dd>
   <dt id="Definition86">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 2 items:
0: (compound) a compound_id
1: (name) a string</pre>
   </dd>
  </dl>
  <h2 id="GapgenSolution">
   GapgenSolution
  </h2>
  <dl>
   <dt id="Description129">
    Description
   </dt>
   <dd>
    <p>
     Data structures for gap generation solution
    </p>
    <p>
     gapgensolution_id id - ID of gapgen solution float objective - cost of gapgen solution list&lt;biomassAddition&gt; biomassAdditions - list of components added to biomass list&lt;mediaRemoval&gt; mediaRemovals - list of media components removed list&lt;reactionRemoval&gt; reactionRemovals - list of reactions removed
    </p>
   </dd>
   <dt id="Definition87">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a gapgensolution_id
objective has a value which is a float
biomassAdditions has a value which is a reference to a list where each element is a biomassAddition
mediaRemovals has a value which is a reference to a list where each element is a mediaRemoval
reactionRemovals has a value which is a reference to a list where each element is a reactionRemoval</pre>
   </dd>
  </dl>
  <h2 id="GapGen">
   GapGen
  </h2>
  <dl>
   <dt id="Description130">
    Description
   </dt>
   <dd>
    <p>
     Data structures for gap generation analysis
    </p>
    <p>
     gapgen_id id - ID of gapgen object workspace_id workspace - workspace containing gapgen object fbamodel_id model - ID of model being gap generated workspace_id model_workspace - workspace containing model bool isComplete - flag indicating if gap generation is complete GapgenFormulation formulation - formulation of gap generation analysis list&lt;GapgenSolution&gt; solutions - list of gap generation solutions
    </p>
   </dd>
   <dt id="Definition88">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a gapgen_id
workspace has a value which is a workspace_id
model has a value which is a fbamodel_id
model_workspace has a value which is a workspace_id
isComplete has a value which is a bool
formulation has a value which is a GapgenFormulation
solutions has a value which is a reference to a list where each element is a GapgenSolution</pre>
   </dd>
  </dl>
  <h2 id="Phenotype">
   Phenotype
  </h2>
  <dl>
   <dt id="Description131">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** Phenotype type definitions ********************************************************************************
    </p>
   </dd>
   <dt id="Definition89">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 5 items:
0: (geneKO) a reference to a list where each element is a feature_id
1: (baseMedia) a media_id
2: (media_workspace) a workspace_id
3: (additionalCpd) a reference to a list where each element is a compound_id
4: (normalizedGrowth) a float</pre>
   </dd>
  </dl>
  <h2 id="PhenotypeSet">
   PhenotypeSet
  </h2>
  <dl>
   <dt id="Description132">
    Description
   </dt>
   <dd>
    <p>
     Data structures for set of growth phenotype observations
    </p>
    <p>
     phenotypeSet_id id - ID of the phenotype set genome_id genome - ID of the genome for the strain used with the growth phenotypes workspace_id genome_workspace - workspace containing the genome object list&lt;Phenotype&gt; phenotypes - list of phenotypes included in the phenotype set string importErrors - list of errors encountered during the import of the phenotype set
    </p>
   </dd>
   <dt id="Definition90">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a phenotypeSet_id
genome has a value which is a genome_id
genome_workspace has a value which is a workspace_id
phenotypes has a value which is a reference to a list where each element is a Phenotype
importErrors has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="phenotypeSimulationSet_id">
   phenotypeSimulationSet_id
  </h2>
  <dl>
   <dt id="Description133">
    Description
   </dt>
   <dd>
    <p>
     ID of the phenotype simulation object
    </p>
   </dd>
   <dt id="Definition91">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="PhenotypeSimulation">
   PhenotypeSimulation
  </h2>
  <dl>
   <dt id="Description134">
    Description
   </dt>
   <dd>
    <p>
     Data structures for a phenotype simulation
    </p>
    <p>
     Phenotype phenotypeData - actual phenotype data simulated float simulatedGrowth - actual simulated growth rate float simulatedGrowthFraction - fraction of wildtype simulated growth rate string class - class of the phenotype simulation (i.e. 'CP' - correct positive, 'CN' - correct negative, 'FP' - false positive, 'FN' - false negative)
    </p>
   </dd>
   <dt id="Definition92">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 4 items:
0: (phenotypeData) a Phenotype
1: (simulatedGrowth) a float
2: (simulatedGrowthFraction) a float
3: (class) a string</pre>
   </dd>
  </dl>
  <h2 id="PhenotypeSimulationSet">
   PhenotypeSimulationSet
  </h2>
  <dl>
   <dt id="Description135">
    Description
   </dt>
   <dd>
    <p>
     Data structures for phenotype simulations of a set of phenotype data
    </p>
    <p>
     phenotypeSimulationSet_id id - ID for the phenotype simulation set object fbamodel_id model - ID of the model used to simulate all phenotypes workspace_id model_workspace - workspace containing the model used for the simulation phenotypeSet_id phenotypeSet - set of observed phenotypes that were simulated list&lt;PhenotypeSimulation&gt; phenotypeSimulations - list of simulated phenotypes
    </p>
   </dd>
   <dt id="Definition93">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a phenotypeSimulationSet_id
model has a value which is a fbamodel_id
model_workspace has a value which is a workspace_id
phenotypeSet has a value which is a phenotypeSet_id
phenotypeSimulations has a value which is a reference to a list where each element is a PhenotypeSimulation</pre>
   </dd>
  </dl>
  <h2 id="reactionSpecification">
   reactionSpecification
  </h2>
  <dl>
   <dt id="Description136">
    Description
   </dt>
   <dd>
    <p>
     Data structure for holding gapfill or gapgen solution reaction information
    </p>
    <p>
     string direction - direction of gapfilled or gapgen reaction string reactionID - ID of gapfilled or gapgen reaction
    </p>
   </dd>
   <dt id="Definition94">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 2 items:
0: (direction) a string
1: (reactionID) a string</pre>
   </dd>
  </dl>
  <h2 id="PhenotypeSensitivityAnalysis">
   PhenotypeSensitivityAnalysis
  </h2>
  <dl>
   <dt id="Description137">
    Description
   </dt>
   <dd>
    <p>
     list&lt;string id, string solutionIndex, list&lt;reactionSpecification&gt; reactionList, list&lt;string&gt; biomassEdits,list&lt;tuple&lt;float simulatedGrowth,float simulatedGrowthFraction,string class&gt;&gt; PhenotypeSimulations&gt; reconciliationSolutionSimulations;
    </p>
   </dd>
   <dt id="Definition95">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
phenotypeSet has a value which is a phenotypeSet_id
phenotypeSet_workspace has a value which is a workspace_id
model has a value which is a fbamodel_id
model_workspace has a value which is a workspace_id
phenotypes has a value which is a reference to a list where each element is a Phenotype
wildtypePhenotypeSimulations has a value which is a reference to a list where each element is a reference to a list containing 3 items:
0: (simulatedGrowth) a float
1: (simulatedGrowthFraction) a float
2: (class) a string</pre>
   </dd>
  </dl>
  <h2 id="job_id">
   job_id
  </h2>
  <dl>
   <dt id="Description138">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** Job object type definitions ********************************************************************************
    </p>
   </dd>
   <dt id="Definition96">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="CommandArguments">
   CommandArguments
  </h2>
  <dl>
   <dt id="Description139">
    Description
   </dt>
   <dd>
    <p>
     Object to hold the arguments to be submitted to the post process command
    </p>
   </dd>
   <dt id="Definition97">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="clusterjob">
   clusterjob
  </h2>
  <dl>
   <dt id="Description140">
    Description
   </dt>
   <dd>
    <p>
     Object to hold data required to run cluster job
    </p>
   </dd>
   <dt id="Definition98">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="JobObject">
   JobObject
  </h2>
  <dl>
   <dt id="Description141">
    Description
   </dt>
   <dd>
    <p>
     Data structures for a job object
    </p>
    <p>
     job_id id - ID of the job object string type - type of the job string auth - authentication token of job owner string status - current status of job mapping&lt;string,string&gt; jobdata; string queuetime - time when job was queued string starttime - time when job started running string completetime - time when the job was completed string owner - owner of the job string queuecommand - command used to queue job
    </p>
   </dd>
   <dt id="Definition99">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a job_id
type has a value which is a string
auth has a value which is a string
status has a value which is a string
jobdata has a value which is a reference to a hash where the key is a string and the value is a string
queuetime has a value which is a string
starttime has a value which is a string
completetime has a value which is a string
owner has a value which is a string
queuecommand has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="ETCNodes">
   ETCNodes
  </h2>
  <dl>
   <dt id="Description142">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** ETC object type definitions ********************************************************************************
    </p>
   </dd>
   <dt id="Definition100">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
resp has a value which is a string
y has a value which is an int
x has a value which is an int
width has a value which is an int
height has a value which is an int
shape has a value which is a string
label has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="ETCDiagramSpecs">
   ETCDiagramSpecs
  </h2>
  <dl>
   <dt id="Definition101">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
nodes has a value which is a reference to a list where each element is an ETCNodes
media has a value which is a string
growth has a value which is a string
organism has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="TemplateReactions">
   TemplateReactions
  </h2>
  <dl>
   <dt id="Description143">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** AutoRecon type definitions ********************************************************************************
    </p>
   </dd>
   <dt id="Definition102">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
reaction has a value which is a reaction_id
direction has a value which is a string
equation has a value which is a string
compartment has a value which is a compartment_id</pre>
   </dd>
  </dl>
  <h2 id="ComplexReactions">
   ComplexReactions
  </h2>
  <dl>
   <dt id="Description144">
    Description
   </dt>
   <dd>
    <p>
     Information on complexes in a template model
    </p>
    <pre><code>        complex_id complex - ID of the associated complex
        list&lt;TemplateReactions&gt; reactions - List of template models associated with complex</code></pre>
   </dd>
   <dt id="Definition103">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
complex has a value which is a complex_id
name has a value which is a string
reactions has a value which is a reference to a list where each element is a TemplateReactions</pre>
   </dd>
  </dl>
  <h2 id="RoleComplexReactions">
   RoleComplexReactions
  </h2>
  <dl>
   <dt id="Description145">
    Description
   </dt>
   <dd>
    <p>
     Information on complexes in a template model
    </p>
    <pre><code>        complex_id complex - ID of the associated complex
        list&lt;TemplateReactions&gt; reactions - List of template models associated with complex</code></pre>
   </dd>
   <dt id="Definition104">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
role has a value which is a role_id
name has a value which is a string
complexes has a value which is a reference to a list where each element is a ComplexReactions</pre>
   </dd>
  </dl>
  <h2 id="ReactionDefinition">
   ReactionDefinition
  </h2>
  <dl>
   <dt id="Description146">
    Description
   </dt>
   <dd>
    <p>
     Reaction definition
    </p>
    <pre><code>        reaction_id id - ID of reaction
        string name - name of reaction
        string definition - stoichiometric equation of reaction in terms of compound names</code></pre>
   </dd>
   <dt id="Definition105">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a reaction_id
name has a value which is a string
definition has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="ReactionSynonyms">
   ReactionSynonyms
  </h2>
  <dl>
   <dt id="Description147">
    Description
   </dt>
   <dd>
    <p>
     Reaction synonyms
    </p>
    <pre><code>        reaction_id primary - ID of primary reaction
        list&lt;ReactionDefinition&gt; synonyms - list of synonym reactions to the primary reaction (including itself)</code></pre>
   </dd>
   <dt id="Definition106">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
primary has a value which is a reaction_id
synonyms has a value which is a reference to a list where each element is a ReactionDefinition</pre>
   </dd>
  </dl>
  <h2 id="ReactionSynonymsObject">
   ReactionSynonymsObject
  </h2>
  <dl>
   <dt id="Description148">
    Description
   </dt>
   <dd>
    <p>
     Reaction synonyms object
    </p>
    <pre><code>        int version - version number of object
        biochemistry_id biochemistry - ID of associated biochemistry database
        workspace_id biochemistry_workspace - workspace with associated biochemistry database
        list&lt;ReactionSynonyms&gt; synonym_list - list of all reaction synonyms from a biochemistry database
        list&lt;ReactionDefinition&gt; excluded_list - list of reactions excluded because all compounds are cofactors</code></pre>
   </dd>
   <dt id="Definition107">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
version has a value which is an int
biochemistry has a value which is a biochemistry_id
biochemistry_workspace has a value which is a workspace_id
synonyms_list has a value which is a reference to a list where each element is a ReactionSynonyms
excluded_list has a value which is a reference to a list where each element is a ReactionDefinition</pre>
   </dd>
  </dl>
  <h2 id="get_models_params">
   get_models_params
  </h2>
  <dl>
   <dt id="Description149">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** Function definitions relating to data retrieval for Model Objects ********************************************************************************
    </p>
   </dd>
   <dt id="Definition108">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
models has a value which is a reference to a list where each element is a fbamodel_id
workspaces has a value which is a reference to a list where each element is a workspace_id
auth has a value which is a string
id_type has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="get_fbas_params">
   get_fbas_params
  </h2>
  <dl>
   <dt id="Description150">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "get_fbas" function.
    </p>
    <pre><code>        list&lt;fba_id&gt; fbas - a list of the FBA study IDs for the FBA studies to be returned (a required argument)
        list&lt;workspace_id&gt; workspaces - a list of the workspaces contianing the FBA studies to be returned (a required argument)
string id_type - the type of ID that should be used in the output data (a optional argument; default is 'ModelSEED')
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition109">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
fbas has a value which is a reference to a list where each element is a fba_id
workspaces has a value which is a reference to a list where each element is a workspace_id
auth has a value which is a string
id_type has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="get_gapfills_params">
   get_gapfills_params
  </h2>
  <dl>
   <dt id="Description151">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "get_gapfills" function.
    </p>
    <pre><code>        list&lt;gapfill_id&gt; gapfills - a list of the gapfill study IDs for the gapfill studies to be returned (a required argument)
        list&lt;workspace_id&gt; workspaces - a list of the workspaces contianing the gapfill studies to be returned (a required argument)
string id_type - the type of ID that should be used in the output data (a optional argument; default is 'ModelSEED')
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition110">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
gapfills has a value which is a reference to a list where each element is a gapfill_id
workspaces has a value which is a reference to a list where each element is a workspace_id
auth has a value which is a string
id_type has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="get_gapgens_params">
   get_gapgens_params
  </h2>
  <dl>
   <dt id="Description152">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "get_gapgens" function.
    </p>
    <pre><code>        list&lt;gapgen_id&gt; gapgens - a list of the gapgen study IDs for the gapgen studies to be returned (a required argument)
        list&lt;workspace_id&gt; workspaces - a list of the workspaces contianing the gapgen studies to be returned (a required argument)
string id_type - the type of ID that should be used in the output data (a optional argument; default is 'ModelSEED')
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition111">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
gapgens has a value which is a reference to a list where each element is a gapgen_id
workspaces has a value which is a reference to a list where each element is a workspace_id
auth has a value which is a string
id_type has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="get_reactions_params">
   get_reactions_params
  </h2>
  <dl>
   <dt id="Description153">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "get_reactions" function.
    </p>
    <pre><code>        list&lt;reaction_id&gt; reactions - a list of the reaction IDs for the reactions to be returned (a required argument)
        string id_type - the type of ID that should be used in the output data (a optional argument; default is 'ModelSEED')
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition112">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
reactions has a value which is a reference to a list where each element is a reaction_id
auth has a value which is a string
id_type has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="get_compounds_params">
   get_compounds_params
  </h2>
  <dl>
   <dt id="Description154">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "get_compounds" function. list&lt;compound_id&gt; compounds - a list of the compound IDs for the compounds to be returned (a required argument) string id_type - the type of ID that should be used in the output data (a optional argument; default is 'ModelSEED') string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)
    </p>
   </dd>
   <dt id="Definition113">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
compounds has a value which is a reference to a list where each element is a compound_id
auth has a value which is a string
id_type has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="get_alias_params">
   get_alias_params
  </h2>
  <dl>
   <dt id="Description155">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the get_alias function
    </p>
    <pre><code>                string object_type    - The type of object (e.g. Compound or Reaction)
                string input_id_type - The type (e.g. ModelSEED) of alias to be inputted
                string output_id_type - The type (e.g. KEGG) of alias to be outputted
                list&lt;string&gt; input_ids - A list of input IDs
                string auth; - The authentication token of the KBase account (optional)</code></pre>
   </dd>
   <dt id="Definition114">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
object_type has a value which is a string
input_id_type has a value which is a string
output_id_type has a value which is a string
input_ids has a value which is a reference to a list where each element is a string
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="get_alias_outputs">
   get_alias_outputs
  </h2>
  <dl>
   <dt id="Description156">
    Description
   </dt>
   <dd>
    <p>
     Output for get_alias function
    </p>
    <pre><code>              string original_id - The original ID
              list&lt;string&gt; aliases - Aliases for the original ID in the new format</code></pre>
   </dd>
   <dt id="Definition115">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
original_id has a value which is a string
aliases has a value which is a reference to a list where each element is a string</pre>
   </dd>
  </dl>
  <h2 id="get_aliassets_params">
   get_aliassets_params
  </h2>
  <dl>
   <dt id="Description157">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the get_aliassets function
    </p>
    <pre><code>              string auth; - The authentication token of the KBase account (optional)
              string object_type; - The type of object (e.g. Compound or Reaction)</code></pre>
   </dd>
   <dt id="Definition116">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
object_type has a value which is a string
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="get_media_params">
   get_media_params
  </h2>
  <dl>
   <dt id="Description158">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "get_media" function.
    </p>
    <pre><code>        list&lt;media_id&gt; medias - a list of the media IDs for the media to be returned (a required argument)
        string id_type - the type of ID that should be used in the output data (a optional argument; default is 'ModelSEED')
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition117">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
medias has a value which is a reference to a list where each element is a media_id
workspaces has a value which is a reference to a list where each element is a workspace_id
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="get_biochemistry_params">
   get_biochemistry_params
  </h2>
  <dl>
   <dt id="Description159">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "get_biochemistry" function.
    </p>
    <pre><code>        biochemistry_id biochemistry - ID of the biochemistry database to be returned (a required argument)
        workspace_id biochemistry_workspace - workspace containing the biochemistry database to be returned (a required argument)
        string id_type - the type of ID that should be used in the output data (a optional argument; default is 'ModelSEED')
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition118">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
biochemistry has a value which is a biochemistry_id
biochemistry_workspace has a value which is a workspace_id
id_type has a value which is a string
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="get_ETCDiagram_params">
   get_ETCDiagram_params
  </h2>
  <dl>
   <dt id="Description160">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "genome_to_fbamodel" function.
    </p>
    <pre><code>        model_id model - ID of the model to retrieve ETC for
        workspace_id workspace - ID of the workspace containing the model 
        media_id media - ID of the media to retrieve ETC for
        workspace_id mediaws - workpace containing the specified media
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition119">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
model has a value which is a fbamodel_id
workspace has a value which is a workspace_id
media has a value which is a media_id
mediaws has a value which is a workspace_id
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="import_probanno_params">
   import_probanno_params
  </h2>
  <dl>
   <dt id="Description161">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** Code relating to reconstruction of metabolic models ********************************************************************************
    </p>
   </dd>
   <dt id="Definition120">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
probanno has a value which is a probanno_id
workspace has a value which is a workspace_id
genome has a value which is a genome_id
genome_workspace has a value which is a workspace_id
annotationProbabilities has a value which is a reference to a list where each element is an annotationProbability
ignore_errors has a value which is a bool
auth has a value which is a string
overwrite has a value which is a bool</pre>
   </dd>
  </dl>
  <h2 id="genome_object_to_workspace_params">
   genome_object_to_workspace_params
  </h2>
  <dl>
   <dt id="Description162">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "genome_object_to_workspace" function.
    </p>
    <pre><code>        GenomeObject genomeobj - full genome typed object to be loaded into the workspace (a required argument)
        workspace_id workspace - ID of the workspace into which the genome typed object is to be loaded (a required argument)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition121">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
genomeobj has a value which is a GenomeObject
workspace has a value which is a workspace_id
auth has a value which is a string
overwrite has a value which is a bool</pre>
   </dd>
  </dl>
  <h2 id="genome_to_workspace_params">
   genome_to_workspace_params
  </h2>
  <dl>
   <dt id="Description163">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "genome_to_workspace" function.
    </p>
    <pre><code>        genome_id genome - ID of the CDM genome that is to be loaded into the workspace (a required argument)
        string sourceLogin - login to pull private genome from source database
        string sourcePassword - password to pull private genome from source database
        string source - Source database for genome (i.e. seed, rast, kbase)
        workspace_id workspace - ID of the workspace into which the genome typed object is to be loaded (a required argument)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition122">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
genome has a value which is a genome_id
workspace has a value which is a workspace_id
sourceLogin has a value which is a string
sourcePassword has a value which is a string
source has a value which is a string
auth has a value which is a string
overwrite has a value which is a bool</pre>
   </dd>
  </dl>
  <h2 id="translation">
   translation
  </h2>
  <dl>
   <dt id="Description164">
    Description
   </dt>
   <dd>
    <p>
     A link between a KBase gene ID and the ID for the same gene in another database
    </p>
    <pre><code>        string foreign_id - ID of the gene in another database
        feature_id feature - ID of the gene in KBase</code></pre>
   </dd>
   <dt id="Definition123">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 2 items:
0: (foreign_id) a string
1: (feature) a feature_id</pre>
   </dd>
  </dl>
  <h2 id="add_feature_translation_params">
   add_feature_translation_params
  </h2>
  <dl>
   <dt id="Description165">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "add_feature_translation" function.
    </p>
    <pre><code>        genome_id genome - ID of the genome into which the new aliases are to be loaded (a required argument)
        workspace_id workspace - ID of the workspace containing the target genome (a required argument)
        list&lt;translation&gt; translations - list of translations between KBase gene IDs and gene IDs in another database (a required argument)
        string id_type - type of the IDs being loaded (e.g. KEGG, NCBI) (a required argument)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition124">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
genome has a value which is a genome_id
workspace has a value which is a workspace_id
translations has a value which is a reference to a list where each element is a translation
id_type has a value which is a string
auth has a value which is a string
overwrite has a value which is a bool</pre>
   </dd>
  </dl>
  <h2 id="genome_to_fbamodel_params">
   genome_to_fbamodel_params
  </h2>
  <dl>
   <dt id="Description166">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "genome_to_fbamodel" function.
    </p>
    <pre><code>        genome_id genome - ID of the genome for which a model is to be built (a required argument)
        workspace_id genome_workspace - ID of the workspace containing the target genome (an optional argument; default is the workspace argument)
        template_id templatemodel - 
        workspace_id templatemodel_workspace - 
        bool probannoOnly - a boolean indicating if only the probabilistic annotation should be used in building the model (an optional argument; default is '0')
        fbamodel_id model - ID that should be used for the newly constructed model (an optional argument; default is 'undef')
        bool coremodel - indicates that a core model should be constructed instead of a genome scale model (an optional argument; default is '0')
        workspace_id workspace - ID of the workspace where the newly developed model will be stored; also the default assumed workspace for input objects (a required argument)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition125">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
genome has a value which is a genome_id
genome_workspace has a value which is a workspace_id
templatemodel has a value which is a template_id
templatemodel_workspace has a value which is a workspace_id
model has a value which is a fbamodel_id
coremodel has a value which is a bool
workspace has a value which is a workspace_id
auth has a value which is a string
overwrite has a value which is a bool</pre>
   </dd>
  </dl>
  <h2 id="import_fbamodel_params">
   import_fbamodel_params
  </h2>
  <dl>
   <dt id="Description167">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "import_fbamodel" function.
    </p>
    <pre><code>        genome_id genome - ID of the genome for which a model is to be built (a required argument)
        workspace_id genome_workspace - ID of the workspace containing the target genome (an optional argument; default is the workspace argument)
        string biomass - biomass equation for model (an essential argument)
        list&lt;tuple&lt;string id,string direction,string compartment,string gpr&gt; reactions - list of reactions to appear in imported model (an essential argument)
        fbamodel_id model - ID that should be used for the newly imported model (an optional argument; default is 'undef')
        workspace_id workspace - ID of the workspace where the newly developed model will be stored; also the default assumed workspace for input objects (a required argument)
        bool ignore_errors - ignores missing genes or reactions and imports model anyway
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition126">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
genome has a value which is a genome_id
genome_workspace has a value which is a workspace_id
biomass has a value which is a string
reactions has a value which is a reference to a list where each element is a reference to a list containing 4 items:
0: (id) a string
1: (direction) a string
2: (compartment) a string
3: (gpr) a stringmodel has a value which is a fbamodel_id
workspace has a value which is a workspace_id
ignore_errors has a value which is a bool
auth has a value which is a string
overwrite has a value which is a bool</pre>
   </dd>
  </dl>
  <h2 id="export_fbamodel_params">
   export_fbamodel_params
  </h2>
  <dl>
   <dt id="Description168">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "export_fbamodel" function.
    </p>
    <pre><code>        fbamodel_id model - ID of the model to be exported (a required argument)
        workspace_id workspace - workspace containing the model to be exported (a required argument)
        string format - format to which the model should be exported (sbml, html, json, readable, cytoseed) (a required argument)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition127">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
model has a value which is a fbamodel_id
workspace has a value which is a workspace_id
format has a value which is a string
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="export_object_params">
   export_object_params
  </h2>
  <dl>
   <dt id="Description169">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "export_object" function.
    </p>
    <pre><code>        workspace_ref reference - reference of object to print in html (a required argument)
        string type - type of the object to be exported (a required argument)
        string format - format to which data should be exported (an optional argument; default is html)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition128">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
reference has a value which is a workspace_ref
type has a value which is a string
format has a value which is a string
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="export_genome_params">
   export_genome_params
  </h2>
  <dl>
   <dt id="Description170">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "export_genome" function.
    </p>
    <pre><code>        genome_id genome - ID of the genome to be exported (a required argument)
        workspace_id workspace - workspace containing the model to be exported (a required argument)
        string format - format to which the model should be exported (sbml, html, json, readable, cytoseed) (a required argument)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition129">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
genome has a value which is a genome_id
workspace has a value which is a workspace_id
format has a value which is a string
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="adjust_model_reaction_params">
   adjust_model_reaction_params
  </h2>
  <dl>
   <dt id="Description171">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "adjust_model_reaction" function.
    </p>
    <pre><code>        fbamodel_id model - ID of model to be adjusted
        workspace_id workspace - workspace containing model to be adjusted
        list&lt;reaction_id&gt; reaction - List of IDs of reactions to be added, removed, or adjusted
        list&lt;string&gt; direction - directions to set for reactions being added or adjusted
        list&lt;compartment_id&gt; compartment - IDs of compartment containing reaction being added or adjusted
        list&lt;int&gt; compartmentIndex - indexes of compartment containing reaction being altered or adjusted
        list&lt;string&gt; gpr - array specifying gene-protein-reaction association(s)
        bool removeReaction - boolean indicating listed reaction(s) should be removed
        bool addReaction - boolean indicating reaction(s) should be added
        bool overwrite - boolean indicating whether or not to overwrite model object in the workspace
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)

        For all of the lists above, if only one element is specified it is assumed the user wants to apply the same
        to all the listed reactions.</code></pre>
   </dd>
   <dt id="Definition130">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
model has a value which is a fbamodel_id
workspace has a value which is a workspace_id
reaction has a value which is a reference to a list where each element is a reaction_id
direction has a value which is a reference to a list where each element is a string
compartment has a value which is a reference to a list where each element is a compartment_id
compartmentIndex has a value which is a reference to a list where each element is an int
gpr has a value which is a reference to a list where each element is a string
removeReaction has a value which is a bool
addReaction has a value which is a bool
overwrite has a value which is a bool
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="adjust_biomass_reaction_params">
   adjust_biomass_reaction_params
  </h2>
  <dl>
   <dt id="Description172">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "adjust_biomass_reaction" function.
    </p>
    <pre><code>        fbamodel_id model - ID of model to be adjusted
        workspace_id workspace - workspace containing model to be adjusted
        biomass_id biomass - ID of biomass reaction to adjust
        float coefficient - coefficient of biomass compound
        compound_id compound - ID of biomass compound to adjust in biomass
        compartment_id compartment - ID of compartment containing compound to adjust in biomass
        int compartmentIndex - index of compartment containing compound to adjust in biomass
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition131">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
model has a value which is a fbamodel_id
workspace has a value which is a workspace_id
biomass has a value which is a biomass_id
coefficient has a value which is a float
compound has a value which is a compound_id
compartment has a value which is a compartment_id
compartmentIndex has a value which is an int
overwrite has a value which is a bool
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="addmedia_params">
   addmedia_params
  </h2>
  <dl>
   <dt id="Description173">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** Code relating to flux balance analysis ********************************************************************************
    </p>
   </dd>
   <dt id="Definition132">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
media has a value which is a media_id
workspace has a value which is a workspace_id
name has a value which is a string
isDefined has a value which is a bool
isMinimal has a value which is a bool
type has a value which is a string
compounds has a value which is a reference to a list where each element is a string
concentrations has a value which is a reference to a list where each element is a float
maxflux has a value which is a reference to a list where each element is a float
minflux has a value which is a reference to a list where each element is a float
overwrite has a value which is a bool
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="export_media_params">
   export_media_params
  </h2>
  <dl>
   <dt id="Description174">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "export_media" function.
    </p>
    <pre><code>        media_id media - ID of the media to be exported (a required argument)
        workspace_id workspace - workspace containing the media to be exported (a required argument)
        string format - format to which the media should be exported (html, json, readable) (a required argument)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition133">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
media has a value which is a media_id
workspace has a value which is a workspace_id
format has a value which is a string
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="runfba_params">
   runfba_params
  </h2>
  <dl>
   <dt id="Description175">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "addmedia" function.
    </p>
    <pre><code>        fbamodel_id model - ID of the model that FBA should be run on (a required argument)
        workspace_id model_workspace - workspace where model for FBA should be run (an optional argument; default is the value of the workspace argument)
        FBAFormulation formulation - a hash specifying the parameters for the FBA study (an optional argument)
        bool fva - a flag indicating if flux variability should be run (an optional argument: default is '0')
        bool simulateko - a flag indicating if flux variability should be run (an optional argument: default is '0')
        bool minimizeflux - a flag indicating if flux variability should be run (an optional argument: default is '0')
        bool findminmedia - a flag indicating if flux variability should be run (an optional argument: default is '0')
        string notes - a string of notes to attach to the FBA study (an optional argument; defaul is '')
        fba_id fba - ID under which the FBA results should be saved (an optional argument; defaul is 'undef')
        workspace_id workspace - workspace where FBA results will be saved (a required argument)
        bool add_to_model - a flag indicating if the FBA study should be attached to the model to support viewing results (an optional argument: default is '0')
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition134">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
model has a value which is a fbamodel_id
model_workspace has a value which is a workspace_id
formulation has a value which is an FBAFormulation
fva has a value which is a bool
simulateko has a value which is a bool
minimizeflux has a value which is a bool
findminmedia has a value which is a bool
notes has a value which is a string
fba has a value which is a fba_id
workspace has a value which is a workspace_id
auth has a value which is a string
overwrite has a value which is a bool
add_to_model has a value which is a bool</pre>
   </dd>
  </dl>
  <h2 id="export_fba_params">
   export_fba_params
  </h2>
  <dl>
   <dt id="Description176">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "addmedia" function.
    </p>
    <pre><code>        fba_id fba - ID of the FBA study to be exported (a required argument)
        workspace_id workspace - workspace where FBA study is stored (a required argument)
        string format - format to which the FBA study should be exported (i.e. html, json, readable) (a required argument)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition135">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
fba has a value which is a fba_id
workspace has a value which is a workspace_id
format has a value which is a string
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="import_phenotypes_params">
   import_phenotypes_params
  </h2>
  <dl>
   <dt id="Description177">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** Code relating to phenotype simulation and reconciliation ********************************************************************************
    </p>
   </dd>
   <dt id="Definition136">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
phenotypeSet has a value which is a phenotypeSet_id
workspace has a value which is a workspace_id
genome has a value which is a genome_id
genome_workspace has a value which is a workspace_id
phenotypes has a value which is a reference to a list where each element is a Phenotype
ignore_errors has a value which is a bool
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="simulate_phenotypes_params">
   simulate_phenotypes_params
  </h2>
  <dl>
   <dt id="Description178">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "simulate_phenotypes" function.
    </p>
    <pre><code>        fbamodel_id model - ID of the model to be used for the simulation (a required argument)
        workspace_id model_workspace - workspace containing the model for the simulation (an optional argument: default is value of workspace argument)
        phenotypeSet_id phenotypeSet - ID of the phenotypes set to be simulated (a required argument)
        workspace_id phenotypeSet_workspace - workspace containing the phenotype set to be simulated (an optional argument: default is value of workspace argument)
        FBAFormulation formulation - parameters for the simulation flux balance analysis (an optional argument: default is 'undef')
        string notes - string of notes to associate with the phenotype simulation (an optional argument: default is '')
        phenotypeSimulationSet_id phenotypeSimultationSet - ID of the phenotype simulation set to be generated (an optional argument: default is 'undef')
        workspace_id workspace - workspace where the phenotype simulation set should be saved (a required argument)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition137">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
model has a value which is a fbamodel_id
model_workspace has a value which is a workspace_id
phenotypeSet has a value which is a phenotypeSet_id
phenotypeSet_workspace has a value which is a workspace_id
formulation has a value which is an FBAFormulation
notes has a value which is a string
phenotypeSimultationSet has a value which is a phenotypeSimulationSet_id
workspace has a value which is a workspace_id
overwrite has a value which is a bool
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="export_phenotypeSimulationSet_params">
   export_phenotypeSimulationSet_params
  </h2>
  <dl>
   <dt id="Description179">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "export_phenotypeSimulationSet" function.
    </p>
    <pre><code>        phenotypeSimulationSet_id phenotypeSimultationSet - ID of the phenotype simulation set to be exported (a required argument)
        workspace_id workspace - workspace where the phenotype simulation set is stored (a required argument)
        string format - format to which phenotype simulation set should be exported (html, json)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition138">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
phenotypeSimulationSet has a value which is a phenotypeSimulationSet_id
workspace has a value which is a workspace_id
format has a value which is a string
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="integrate_reconciliation_solutions_params">
   integrate_reconciliation_solutions_params
  </h2>
  <dl>
   <dt id="Description180">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "integrate_reconciliation_solutions" function.
    </p>
    <pre><code>        fbamodel_id model - ID of model for which reconciliation solutions should be integrated (a required argument)
        workspace_id model_workspace - workspace containing model for which solutions should be integrated (an optional argument: default is value of workspace argument)
        list&lt;gapfillsolution_id&gt; gapfillSolutions - list of gapfill solutions to be integrated (a required argument)
        list&lt;gapgensolution_id&gt; gapgenSolutions - list of gapgen solutions to be integrated (a required argument)
        fbamodel_id out_model - ID to which modified model should be saved (an optional argument: default is value of workspace argument)
        workspace_id workspace - workspace where modified model should be saved (a required argument)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition139">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
model has a value which is a fbamodel_id
model_workspace has a value which is a workspace_id
gapfillSolutions has a value which is a reference to a list where each element is a gapfillsolution_id
gapgenSolutions has a value which is a reference to a list where each element is a gapgensolution_id
out_model has a value which is a fbamodel_id
workspace has a value which is a workspace_id
auth has a value which is a string
overwrite has a value which is a bool</pre>
   </dd>
  </dl>
  <h2 id="queue_runfba_params">
   queue_runfba_params
  </h2>
  <dl>
   <dt id="Description181">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** Code relating to queuing long running jobs ********************************************************************************
    </p>
   </dd>
   <dt id="Definition140">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
model has a value which is a fbamodel_id
model_workspace has a value which is a workspace_id
formulation has a value which is an FBAFormulation
fva has a value which is a bool
simulateko has a value which is a bool
minimizeflux has a value which is a bool
findminmedia has a value which is a bool
notes has a value which is a string
fba has a value which is a fba_id
workspace has a value which is a workspace_id
auth has a value which is a string
overwrite has a value which is a bool
add_to_model has a value which is a bool</pre>
   </dd>
  </dl>
  <h2 id="gapfill_model_params">
   gapfill_model_params
  </h2>
  <dl>
   <dt id="Description182">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "queue_gapfill_model" function.
    </p>
    <pre><code>        fbamodel_id model - ID of the model that gapfill should be run on (a required argument)
        workspace_id model_workspace - workspace where model for gapfill should be run (an optional argument; default is the value of the workspace argument)
        GapfillingFormulation formulation - a hash specifying the parameters for the gapfill study (an optional argument)
        phenotypeSet_id phenotypeSet - ID of a phenotype set against which gapfilled model should be simulated (an optional argument: default is 'undef')
        workspace_id phenotypeSet_workspace - workspace containing phenotype set to be simulated (an optional argument; default is the value of the workspace argument)
        bool integrate_solution - a flag indicating if the first solution should be integrated in the model (an optional argument: default is '0')
        fbamodel_id out_model - ID where the gapfilled model will be saved (an optional argument: default is 'undef')
        gapfill_id gapFill - ID to which gapfill solution will be saved (an optional argument: default is 'undef')
        workspace_id workspace - workspace where gapfill results will be saved (a required argument)
        int timePerSolution - maximum time to spend to obtain each solution
        int totalTimeLimit - maximum time to spend to obtain all solutions
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)
        bool completeGapfill - boolean indicating that all inactive reactions should be gapfilled</code></pre>
   </dd>
   <dt id="Definition141">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
model has a value which is a fbamodel_id
model_workspace has a value which is a workspace_id
formulation has a value which is a GapfillingFormulation
phenotypeSet has a value which is a phenotypeSet_id
phenotypeSet_workspace has a value which is a workspace_id
integrate_solution has a value which is a bool
out_model has a value which is a fbamodel_id
workspace has a value which is a workspace_id
gapFill has a value which is a gapfill_id
timePerSolution has a value which is an int
totalTimeLimit has a value which is an int
auth has a value which is a string
overwrite has a value which is a bool
completeGapfill has a value which is a bool</pre>
   </dd>
  </dl>
  <h2 id="gapgen_model_params">
   gapgen_model_params
  </h2>
  <dl>
   <dt id="Description183">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "queue_gapgen_model" function.
    </p>
    <pre><code>        fbamodel_id model - ID of the model that gapgen should be run on (a required argument)
        workspace_id model_workspace - workspace where model for gapgen should be run (an optional argument; default is the value of the workspace argument)
        GapgenFormulation formulation - a hash specifying the parameters for the gapgen study (an optional argument)
        phenotypeSet_id phenotypeSet - ID of a phenotype set against which gapgened model should be simulated (an optional argument: default is 'undef')
        workspace_id phenotypeSet_workspace - workspace containing phenotype set to be simulated (an optional argument; default is the value of the workspace argument)
        bool integrate_solution - a flag indicating if the first solution should be integrated in the model (an optional argument: default is '0')
        fbamodel_id out_model - ID where the gapgened model will be saved (an optional argument: default is 'undef')
        gapgen_id gapGen - ID to which gapgen solution will be saved (an optional argument: default is 'undef')
        workspace_id workspace - workspace where gapgen results will be saved (a required argument)
        int timePerSolution - maximum time to spend to obtain each solution
        int totalTimeLimit - maximum time to spend to obtain all solutions
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition142">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
model has a value which is a fbamodel_id
model_workspace has a value which is a workspace_id
formulation has a value which is a GapgenFormulation
phenotypeSet has a value which is a phenotypeSet_id
phenotypeSet_workspace has a value which is a workspace_id
integrate_solution has a value which is a bool
out_model has a value which is a fbamodel_id
workspace has a value which is a workspace_id
gapGen has a value which is a gapgen_id
auth has a value which is a string
timePerSolution has a value which is an int
totalTimeLimit has a value which is an int
overwrite has a value which is a bool</pre>
   </dd>
  </dl>
  <h2 id="wildtype_phenotype_reconciliation_params">
   wildtype_phenotype_reconciliation_params
  </h2>
  <dl>
   <dt id="Description184">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "queue_wildtype_phenotype_reconciliation" function.
    </p>
    <pre><code>        fbamodel_id model - ID of the model that reconciliation should be run on (a required argument)
        workspace_id model_workspace - workspace where model for reconciliation should be run (an optional argument; default is the value of the workspace argument)
        FBAFormulation formulation - a hash specifying the parameters for the reconciliation study (an optional argument)
        GapfillingFormulation gapfill_formulation - a hash specifying the parameters for the gapfill study (an optional argument)
        GapgenFormulation gapgen_formulation - a hash specifying the parameters for the gapgen study (an optional argument)
        phenotypeSet_id phenotypeSet - ID of a phenotype set against which reconciled model should be simulated (an optional argument: default is 'undef')
        workspace_id phenotypeSet_workspace - workspace containing phenotype set to be simulated (an optional argument; default is the value of the workspace argument)
        fbamodel_id out_model - ID where the reconciled model will be saved (an optional argument: default is 'undef')
        list&lt;gapgen_id&gt; gapGens - IDs of gapgen solutions (an optional argument: default is 'undef')
        list&lt;gapfill_id&gt; gapFills - IDs of gapfill solutions (an optional argument: default is 'undef')
        bool queueSensitivityAnalysis - flag indicating if sensitivity analysis should be queued to run on solutions (an optional argument: default is '0')
        bool queueReconciliationCombination - flag indicating if reconcilication combination should be queued to run on solutions (an optional argument: default is '0')
        workspace_id workspace - workspace where reconciliation results will be saved (a required argument)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition143">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
model has a value which is a fbamodel_id
model_workspace has a value which is a workspace_id
fba_formulation has a value which is an FBAFormulation
gapfill_formulation has a value which is a GapfillingFormulation
gapgen_formulation has a value which is a GapgenFormulation
phenotypeSet has a value which is a phenotypeSet_id
phenotypeSet_workspace has a value which is a workspace_id
out_model has a value which is a fbamodel_id
workspace has a value which is a workspace_id
gapFills has a value which is a reference to a list where each element is a gapfill_id
gapGens has a value which is a reference to a list where each element is a gapgen_id
queueSensitivityAnalysis has a value which is a bool
queueReconciliationCombination has a value which is a bool
auth has a value which is a string
overwrite has a value which is a bool</pre>
   </dd>
  </dl>
  <h2 id="queue_reconciliation_sensitivity_analysis_params">
   queue_reconciliation_sensitivity_analysis_params
  </h2>
  <dl>
   <dt id="Description185">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "queue_reconciliation_sensitivity_analysis" function.
    </p>
    <pre><code>        fbamodel_id model - ID of the model that sensitivity analysis should be run on (a required argument)
        workspace_id model_workspace - workspace where model for sensitivity analysis should be run (an optional argument; default is the value of the workspace argument)
        FBAFormulation formulation - a hash specifying the parameters for the sensitivity analysis study (an optional argument)
        GapfillingFormulation gapfill_formulation - a hash specifying the parameters for the gapfill study (an optional argument)
        GapgenFormulation gapgen_formulation - a hash specifying the parameters for the gapgen study (an optional argument)
        phenotypeSet_id phenotypeSet - ID of a phenotype set against which sensitivity analysis model should be simulated (an optional argument: default is 'undef')
        workspace_id phenotypeSet_workspace - workspace containing phenotype set to be simulated (an optional argument; default is the value of the workspace argument)
        fbamodel_id out_model - ID where the sensitivity analysis model will be saved (an optional argument: default is 'undef')
        list&lt;gapgen_id&gt; gapGens - IDs of gapgen solutions (an optional argument: default is 'undef')
        list&lt;gapfill_id&gt; gapFills - IDs of gapfill solutions (an optional argument: default is 'undef')
        bool queueReconciliationCombination - flag indicating if sensitivity analysis combination should be queued to run on solutions (an optional argument: default is '0')
        workspace_id workspace - workspace where sensitivity analysis results will be saved (a required argument)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition144">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
model has a value which is a fbamodel_id
workspace has a value which is a workspace_id
phenotypeSet has a value which is a phenotypeSet_id
fba_formulation has a value which is an FBAFormulation
model_workspace has a value which is a workspace_id
phenotypeSet_workspace has a value which is a workspace_id
gapFills has a value which is a reference to a list where each element is a gapfill_id
gapGens has a value which is a reference to a list where each element is a gapgen_id
queueReconciliationCombination has a value which is a bool
auth has a value which is a string
overwrite has a value which is a bool</pre>
   </dd>
  </dl>
  <h2 id="combine_wildtype_phenotype_reconciliation_params">
   combine_wildtype_phenotype_reconciliation_params
  </h2>
  <dl>
   <dt id="Description186">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "queue_combine_wildtype_phenotype_reconciliation" function.
    </p>
    <pre><code>        fbamodel_id model - ID of the model that solution combination should be run on (a required argument)
        workspace_id model_workspace - workspace where model for solution combination should be run (an optional argument; default is the value of the workspace argument)
        FBAFormulation formulation - a hash specifying the parameters for the solution combination study (an optional argument)
        GapfillingFormulation gapfill_formulation - a hash specifying the parameters for the gapfill study (an optional argument)
        GapgenFormulation gapgen_formulation - a hash specifying the parameters for the gapgen study (an optional argument)
        phenotypeSet_id phenotypeSet - ID of a phenotype set against which solution combination model should be simulated (an optional argument: default is 'undef')
        workspace_id phenotypeSet_workspace - workspace containing phenotype set to be simulated (an optional argument; default is the value of the workspace argument)
        fbamodel_id out_model - ID where the solution combination model will be saved (an optional argument: default is 'undef')
        list&lt;gapgen_id&gt; gapGens - IDs of gapgen solutions (an optional argument: default is 'undef')
        list&lt;gapfill_id&gt; gapFills - IDs of gapfill solutions (an optional argument: default is 'undef')
        workspace_id workspace - workspace where solution combination results will be saved (a required argument)
        int timePerSolution - maximum time spent per solution
        int totalTimeLimit - maximum time allowed to work on problem
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition145">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
model has a value which is a fbamodel_id
model_workspace has a value which is a workspace_id
fba_formulation has a value which is an FBAFormulation
gapfill_formulation has a value which is a GapfillingFormulation
gapgen_formulation has a value which is a GapgenFormulation
phenotypeSet has a value which is a phenotypeSet_id
phenotypeSet_workspace has a value which is a workspace_id
out_model has a value which is a fbamodel_id
workspace has a value which is a workspace_id
gapFills has a value which is a reference to a list where each element is a gapfill_id
gapGens has a value which is a reference to a list where each element is a gapgen_id
auth has a value which is a string
overwrite has a value which is a bool</pre>
   </dd>
  </dl>
  <h2 id="jobs_done_params">
   jobs_done_params
  </h2>
  <dl>
   <dt id="Description187">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "jobs_done" function.
    </p>
    <pre><code>        job_id job - ID of the job object (a required argument)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition146">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
job has a value which is a job_id
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="run_job_params">
   run_job_params
  </h2>
  <dl>
   <dt id="Description188">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "run_job" function.
    </p>
    <pre><code>        job_id job - ID of the job object (a required argument)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition147">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
job has a value which is a job_id
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="set_cofactors_params">
   set_cofactors_params
  </h2>
  <dl>
   <dt id="Description189">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "set_cofactors" function.
    </p>
    <pre><code>        list&lt;compound_id&gt; cofactors - list of compounds that are universal cofactors (required)
        biochemistry_id biochemistry - ID of biochemistry database (optional, default is "default") 
        workspace_id biochemistry_workspace - ID of workspace containing biochemistry database (optional, default is current workspace)
        bool reset - true to reset (turn off) compounds as universal cofactors (optional, default is false)
        bool overwrite - true to overwrite existing object (optional, default is false)
        string auth - the authentication token of the KBase account (optional, default user is "public")</code></pre>
   </dd>
   <dt id="Definition148">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
cofactors has a value which is a reference to a list where each element is a compound_id
biochemistry has a value which is a biochemistry_id
biochemistry_workspace has a value which is a workspace_id
reset has a value which is a bool
overwrite has a value which is a bool
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="find_reaction_synonyms_params">
   find_reaction_synonyms_params
  </h2>
  <dl>
   <dt id="Description190">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "find_reaction_synonyms" function.
    </p>
    <pre><code>        reaction_synonyms - ID of reaction synonyms object (required argument)
        workspace_id workspace - ID of workspace for storing objects (optional argument, default is current workspace)
        biochemistry_id biochemistry - ID of the biochemistry database (optional argument, default is default)
        workspace_id biochemistry_workspace - ID of workspace containing biochemistry database (optional argument, default is kbase)
        overwrite - True to overwrite existing object (optional argument, default is false)
        string auth - the authentication token of the KBase account (optional argument, default user is "public")</code></pre>
   </dd>
   <dt id="Definition149">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
reaction_synonyms has a value which is a reaction_synonyms_id
workspace has a value which is a workspace_id
biochemistry has a value which is a biochemistry_id
biochemistry_workspace has a value which is a workspace_id
overwrite has a value which is a bool
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="role_to_reactions_params">
   role_to_reactions_params
  </h2>
  <dl>
   <dt id="Description191">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "role_to_reactions" function.
    </p>
    <pre><code>        template_id templateModel - ID of the template model to be used to determine mapping (default is '')
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition150">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
templateModel has a value which is a template_id
workspace has a value which is a workspace_id
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="fasta_to_contigs_params">
   fasta_to_contigs_params
  </h2>
  <dl>
   <dt id="Description192">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "fasta_to_contigs" function.
    </p>
    <pre><code>        string contigid - ID to be assigned to the contigs object created (optional)
        string fasta - string with sequence data from fasta file (required argument)
        workspace_id workspace - ID of workspace for storing objects (optional argument, default is current workspace)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition151">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
contigid has a value which is a string
fasta has a value which is a string
workspace has a value which is a workspace_id
auth has a value which is a string
source has a value which is a string
genetic_code has a value which is a string
domain has a value which is a string
scientific_name has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="contigs_to_genome_params">
   contigs_to_genome_params
  </h2>
  <dl>
   <dt id="Description193">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "contigs_to_genome" function.
    </p>
    <pre><code>        string contigid - ID to be assigned to the contigs object created (optional)
        workspace_id contigws - ID of workspace with contigs (optional argument, default is value of workspace argument)
        workspace_id workspace - ID of workspace for storing objects (optional argument, default is current workspace)
        string genomeid - ID to use for genome object (required argument)
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition152">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
contigid has a value which is a string
contig_workspace has a value which is a workspace_id
workspace has a value which is a workspace_id
genomeid has a value which is a string
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="FunctionalRole">
   FunctionalRole
  </h2>
  <dl>
   <dt id="Description194">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** Code relating to loading, retrieval, and curation of mappings ********************************************************************************
    </p>
   </dd>
   <dt id="Definition153">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a role_id
name has a value which is a string
feature has a value which is a string
aliases has a value which is a reference to a list where each element is a string
complexes has a value which is a reference to a list where each element is a complex_id</pre>
   </dd>
  </dl>
  <h2 id="ComplexRole">
   ComplexRole
  </h2>
  <dl>
   <dt id="Definition154">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 4 items:
0: (id) a role_id
1: (roleType) a string
2: (optional) a bool
3: (triggering) a bool</pre>
   </dd>
  </dl>
  <h2 id="Complex">
   Complex
  </h2>
  <dl>
   <dt id="Definition155">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a complex_id
name has a value which is a string
aliases has a value which is a reference to a list where each element is a string
roles has a value which is a reference to a list where each element is a ComplexRole</pre>
   </dd>
  </dl>
  <h2 id="subsystem_id">
   subsystem_id
  </h2>
  <dl>
   <dt id="Definition156">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="Subsystem1">
   Subsystem
  </h2>
  <dl>
   <dt id="Definition157">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a subsystem_id
name has a value which is a string
class has a value which is a string
subclass has a value which is a string
type has a value which is a string
aliases has a value which is a reference to a list where each element is a string
roles has a value which is a reference to a list where each element is a role_id</pre>
   </dd>
  </dl>
  <h2 id="Mapping">
   Mapping
  </h2>
  <dl>
   <dt id="Definition158">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a mapping_id
name has a value which is a string
subsystems has a value which is a reference to a list where each element is a Subsystem
roles has a value which is a reference to a list where each element is a FunctionalRole
complexes has a value which is a reference to a list where each element is a Complex</pre>
   </dd>
  </dl>
  <h2 id="get_mapping_params">
   get_mapping_params
  </h2>
  <dl>
   <dt id="Definition159">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
map has a value which is a mapping_id
workspace has a value which is a workspace_id
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="adjust_mapping_role_params">
   adjust_mapping_role_params
  </h2>
  <dl>
   <dt id="Description195">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "adjust_mapping_role" function.
    </p>
    <pre><code>        mapping_id map - ID of the mapping object to be edited
        workspace_id workspace - ID of workspace containing mapping to be edited
        string role - identifier for role to be edited
        bool new - boolean indicating that a new role is being added
        string name - new name for the role
        string feature - representative feature MD5
        list&lt;string&gt; aliasesToAdd - list of new aliases for the role
        list&lt;string&gt; aliasesToRemove - list of aliases to remove for role
        bool delete - boolean indicating that role should be deleted
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition160">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
map has a value which is a mapping_id
workspace has a value which is a workspace_id
role has a value which is a string
new has a value which is a bool
name has a value which is a string
feature has a value which is a string
aliasesToAdd has a value which is a reference to a list where each element is a string
aliasesToRemove has a value which is a reference to a list where each element is a string
delete has a value which is a bool
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="adjust_mapping_complex_params">
   adjust_mapping_complex_params
  </h2>
  <dl>
   <dt id="Description196">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "adjust_mapping_complex" function.
    </p>
    <pre><code>        mapping_id map - ID of the mapping object to be edited
        workspace_id workspace - ID of workspace containing mapping to be edited
        string complex - identifier for complex to be edited
        bool new - boolean indicating that a new complex is being added
        string name - new name for the role
        string feature - representative feature MD5
        list&lt;string&gt; rolesToAdd - roles to add to the complex
        list&lt;string&gt; rolesToRemove - roles to remove from the complex
        bool delete - boolean indicating that complex should be deleted
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition161">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
map has a value which is a mapping_id
workspace has a value which is a workspace_id
complex has a value which is a string
new has a value which is a bool
name has a value which is a string
rolesToAdd has a value which is a reference to a list where each element is a string
rolesToRemove has a value which is a reference to a list where each element is a string
delete has a value which is a bool
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="adjust_mapping_subsystem_params">
   adjust_mapping_subsystem_params
  </h2>
  <dl>
   <dt id="Description197">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "adjust_mapping_subsystem" function.
    </p>
    <pre><code>        mapping_id map - ID of the mapping object to be edited
        workspace_id workspace - ID of workspace containing mapping to be edited
        string subsystem - identifier for subsystem to be edited
        bool new - boolean indicating that a new subsystem is being added
        string name - new name for the subsystem
        string type - new type for the subsystem
        string class - new class for the subsystem
        string subclass - new subclass for the subsystem
        list&lt;string&gt; rolesToAdd - roles to add to the subsystem
        list&lt;string&gt; rolesToRemove - roles to remove from the subsystem
        bool delete - boolean indicating that subsystem should be deleted
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition162">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
map has a value which is a mapping_id
workspace has a value which is a workspace_id
subsystem has a value which is a string
new has a value which is a bool
name has a value which is a string
type has a value which is a string
class has a value which is a string
subclass has a value which is a string
rolesToAdd has a value which is a reference to a list where each element is a string
rolesToRemove has a value which is a reference to a list where each element is a string
delete has a value which is a bool
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="temprxn_id">
   temprxn_id
  </h2>
  <dl>
   <dt id="Description198">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** Code relating to loading, retrieval, and curation of template models ********************************************************************************
    </p>
   </dd>
   <dt id="Definition163">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="TemplateReaction">
   TemplateReaction
  </h2>
  <dl>
   <dt id="Definition164">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a temprxn_id
compartment has a value which is a compartment_id
reaction has a value which is a reaction_id
complexes has a value which is a reference to a list where each element is a complex_id
direction has a value which is a string
type has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="TemplateBiomassCompounds">
   TemplateBiomassCompounds
  </h2>
  <dl>
   <dt id="Definition165">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 7 items:
0: (compound) a compound_id
1: (compartment) a compartment_id
2: (class) a string
3: (universal) a string
4: (coefficientType) a string
5: (coefficient) a string
6: (linkedCompounds) a reference to a list where each element is a reference to a list containing 2 items:
	0: (coeffficient) a string
	1: (compound) a compound_id</pre>
   </dd>
  </dl>
  <h2 id="tempbiomass_id">
   tempbiomass_id
  </h2>
  <dl>
   <dt id="Definition166">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="TemplateBiomass">
   TemplateBiomass
  </h2>
  <dl>
   <dt id="Definition167">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a tempbiomass_id
name has a value which is a string
type has a value which is a string
other has a value which is a string
protein has a value which is a string
dna has a value which is a string
rna has a value which is a string
cofactor has a value which is a string
energy has a value which is a string
cellwall has a value which is a string
lipid has a value which is a string
compounds has a value which is a reference to a list where each element is a TemplateBiomassCompounds</pre>
   </dd>
  </dl>
  <h2 id="TemplateModel">
   TemplateModel
  </h2>
  <dl>
   <dt id="Definition168">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a template_id
name has a value which is a string
type has a value which is a string
domain has a value which is a string
map has a value which is a mapping_id
mappingws has a value which is a workspace_id
reactions has a value which is a reference to a list where each element is a TemplateReaction
biomasses has a value which is a reference to a list where each element is a TemplateBiomass</pre>
   </dd>
  </dl>
  <h2 id="get_template_model_params">
   get_template_model_params
  </h2>
  <dl>
   <dt id="Definition169">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
templateModel has a value which is a template_id
workspace has a value which is a workspace_id
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="import_template_fbamodel_params">
   import_template_fbamodel_params
  </h2>
  <dl>
   <dt id="Description199">
    Description
   </dt>
   <dd>
    <p>
     Input parameters for the "import_template_fbamodel" function.
    </p>
    <pre><code>        mapping_id map - ID of the mapping to associate the template model with (an optional argument; default is 'default')
        workspace_id mapping_workspace - ID of the workspace where the associated mapping is found (an optional argument; default is 'kbase')
        list&lt;tuple&lt;string id,string compartment,string direction,string type,list&lt;string complex&gt; complexes&gt;&gt; templateReactions - list of reactions to include in template model
        list&lt;tuple&lt;string name,string type,float dna,float rna,float protein,float lipid,float cellwall,float cofactor,float energy,float other,list&lt;tuple&lt;string id,string compartment,string class,string coefficientType,float coefficient,string conditions&gt;&gt; compounds&gt;&gt; templateBiomass - list of template biomass reactions for template model
        string name - name for template model
        string modelType - type of model constructed by template
        string domain - domain of template model
        template_id id - ID that should be used for the newly imported template model (an optional argument; default is 'undef')
        workspace_id workspace - ID of the workspace where the newly developed template model will be stored; also the default assumed workspace for input objects (a required argument)
        bool ignore_errors - ignores missing roles or reactions and imports template model anyway
        string auth - the authentication token of the KBase account changing workspace permissions; must have 'admin' privelages to workspace (an optional argument; user is "public" if auth is not provided)</code></pre>
   </dd>
   <dt id="Definition170">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
map has a value which is a mapping_id
mapping_workspace has a value which is a workspace_id
templateReactions has a value which is a reference to a list where each element is a reference to a list containing 5 items:
0: (id) a string
1: (compartment) a string
2: (direction) a string
3: (type) a string
4: (complexes) a reference to a list where each element is a stringtemplateBiomass has a value which is a reference to a list where each element is a reference to a list containing 11 items:
0: (name) a string
1: (type) a string
2: (dna) a float
3: (rna) a float
4: (protein) a float
5: (lipid) a float
6: (cellwall) a float
7: (cofactor) a float
8: (energy) a float
9: (other) a float
10: (compounds) a reference to a list where each element is a reference to a list containing 6 items:
	0: (id) a string
	1: (compartment) a string
	2: (class) a string
	3: (coefficientType) a string
	4: (coefficient) a float
	5: (conditions) a stringname has a value which is a string
modelType has a value which is a string
domain has a value which is a string
id has a value which is a template_id
workspace has a value which is a workspace_id
ignore_errors has a value which is a bool
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="adjust_template_reaction_params">
   adjust_template_reaction_params
  </h2>
  <dl>
   <dt id="Definition171">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
templateModel has a value which is a template_id
workspace has a value which is a workspace_id
reaction has a value which is a string
clearComplexes has a value which is a bool
new has a value which is a bool
delete has a value which is a bool
compartment has a value which is a compartment_id
complexesToAdd has a value which is a reference to a list where each element is a complex_id
complexesToRemove has a value which is a reference to a list where each element is a complex_id
direction has a value which is a string
type has a value which is a string
auth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="adjust_template_biomass_params">
   adjust_template_biomass_params
  </h2>
  <dl>
   <dt id="Definition172">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
templateModel has a value which is a template_id
workspace has a value which is a workspace_id
biomass has a value which is a string
new has a value which is a bool
delete has a value which is a bool
clearBiomassCompounds has a value which is a bool
name has a value which is a string
type has a value which is a string
other has a value which is a string
protein has a value which is a string
dna has a value which is a string
rna has a value which is a string
cofactor has a value which is a string
energy has a value which is a string
cellwall has a value which is a string
lipid has a value which is a string
compoundsToRemove has a value which is a reference to a list where each element is a reference to a list containing 2 items:
0: (compound) a compound_id
1: (compartment) a compartment_idcompoundsToAdd has a value which is a reference to a list where each element is a reference to a list containing 7 items:
0: (compound) a compound_id
1: (compartment) a compartment_id
2: (class) a string
3: (universal) a string
4: (coefficientType) a string
5: (coefficient) a string
6: (linkedCompounds) a reference to a list where each element is a reference to a list containing 2 items:
	0: (coeffficient) a string
	1: (compound) a compound_idauth has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="add_stimuli_params">
   add_stimuli_params
  </h2>
  <dl>
   <dt id="Description200">
    Description
   </dt>
   <dd>
    <p>
     ******************************************************************************** Code relating to reconstruction, import, and analysis of regulatory models ********************************************************************************
    </p>
   </dd>
   <dt id="Definition173">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
biochemid has a value which is a string
biochem_workspace has a value which is a string
stimuliid has a value which is a string
name has a value which is a string
abbreviation has a value which is a string
type has a value which is a string
description has a value which is a string
compounds has a value which is a reference to a list where each element is a string
workspace has a value which is a string
auth has a value which is a string</pre>
   </dd>
  </dl>
 </body>
</html>
