<h1>PhiSpy - Finding Prophages in a Genome</h1>

<div>
<p>
<strong>Purpose:</strong> 
How to find prophages in prokaryotic genome by ranking genomic regions enriched in distinctive phage traits, which leads to the successful prediction of 94% of prophages in 50 complete bacterial genomes with a 6% false-negative rate and a 0.66% false-positive rate.

<p>
Like with other services, you can use phispy on your local installation of KBase and IRIS.
<p>
<strong>Required Prerequisite Activities:</strong> 
<a href="/for-users/tutorials/installing-the-kbase-app-dmg/">Download KBase: Installing the KBase Application DMG</a>, OR <a href="for-users/tutorials/navigating-iris/">Get Started with IRIS: Navigating IRIS</a> </p>

</div>
<p>
<div>

<h2>Introduction</h2>
This service can used from the command-line (KBase downloadable application) and our web-based terminal, IRIS.
Prophages are phages in lysogeny that are integrated into, and replicated as part of, the host bacterial genome. These mobile elements can have tremendous impact on their bacterial hostsâ€™ genomes and phenotypes, which may lead to strain emergence and diversification, increased virulence or antibiotic resistance. However, finding prophages in microbial genomes remains a problem with no definitive solution. The majority of existing tools rely on detecting genomic regions enriched in protein-coding genes with known phage homologs, which hinders the de novo discovery of phage regions. 
<p>
PhiSpy is a weighted phage detection algorithm that was developed based on seven distinctive characteristics of prophages, i.e. protein length, transcription strand directionality, customized AT and GC skew, the abundance of unique phage words, phage insertion points and the similarity of phage proteins. The first five characteristics are capable of identifying prophages without any sequence similarity with known phage genes. PhiSpy locates prophages by ranking genomic regions enriched in distinctive phage traits, which leads to the successful prediction of 94% of prophages in 50 complete bacterial genomes with a 6% false-negative rate and a 0.66% false-positive rate.

</div>
<div>
<h2> Using Phispy </h2>

Phispy has two commands to run.
The first is get_all_training_sets. This command lists the available training sets for use in find_prophages. 
To view the available training sets, simply type: 
<pre>
<code title = "iris-tutorials">get_all_training_sets</code>
</pre>
This will return the entire list, like the one below.
<pre>
>/ get_all_training_sets
0 Generic Test Set
1 Bacillus halodurans
2 Bacillus subtilis
3 Bifidobacterium longum
4 Brucella melitensis
5 Caulobacter crescentus
6 Clostridium perfringens
7 Clostridium tetani
8 Deinococcus radiodurans
9 Escherichia coli
10 Haemophilus influenzae
11 Lactococcus lactis
12 Listeria innocua
13 Listeria monocytogenes
14 Mesorhizobium loti
15 Mycobacterium tuberculosis
16 Neisseria meningitidis
17 Pasteurella multocida
18 Pseudomonas aeruginosa
19 Pseudomonas putida
20 Ralstonia solanacearum
21 Salmonella enterica
22 Shewanella oneidensis
23 Shigella flexneri
24 Staphylococcus aureus
25 Streptococcus pyogenes
26 Streptomyces coelicolor
27 Vibrio cholerae
28 Xanthomonas axonopodis
29 Xylella fastidiosa
30 Yersinia pestis 
</pre>

</div>
<div>
To use this with the phage finder, you'll need to identify the training set you would like to use. Pick one that is phylogentically close to your organism. If you are unsure or these do not fit, select training set 0 - Generic Test Set.

<p>
Next you will run the phage finder. This is run using the command find_prophages.
Use the phispy service to find prophages in a given genome. The genome input is a genome typed object as generated by <A HREF="FOR-USERS/TUTORIALS/ANNOTATING-A-GENOME/"><code title = "iris-tutorial">annotate_genome</code></a>. The output is a genome typed object with prophage features added.
<p>
To run <code title = "iris-tutorial">find_prophages</code>:
<pre>
find_prophages training-set-id < genome > genome.with.prophages
</pre>
find prophages is the command that finds the prophages, the training set id is the one you identified using get_all_training_sets, the genome file is the "genome typed object" that you generated from <a href="for-users/tutorials/annotating-a-genome/">annotate_genome</a>, and the output file (in this case we named it genome.with.prophages).
</div>
