<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html><head><title>KBase Networks API 1.0</title></head><body><b>  
  </b>
<h1>KBase Networks API 1.0</h1>

<strong>Purpose: </strong>Learn how to use KBase Networks API 1.0 <b><br>
Required Prerequisite Activities: </b> None
<b><br>
Suggested Prerequisite Activities: </b> Getting What You Need from the
CS Using Command-Line Scripts
<b><br>
Related Tutorials: </b> None
<h2> Introduction </h2>

<p>The KBase Central Store (CS) contains networks data pulled from a variety
  of public databases. This tutorial will teach you:</p>
<ul>
  <li>How to obtain the general information on what networks types,
    datasets, and dataset sources are currently available in KBase CS. </li>
  <li> How to access the KBase CS to extract specific subsets of
    network data using the KBase Networks API 1.0</li>
</ul>
<h2>Learning what network data
are out there </h2>

<h3> Classes of network nodes and edges </h3>

KBase uses a very general definition of a network. We currently allow
three classes of network nodes: <em>gene</em>, <em>protein</em>, and <em>cluster </em>or a group
of gene/proteins such as e.g., a multi-protein complex or a metabolic
pathway. Network edges correspondingly come in five main classes:
<em>gene-gene</em>,<em> protein-protein</em>, <em>gene-cluster</em>, <em>protein-cluster,</em> and
<em>cluster-cluster</em>. In the future, we plan to expand the number of classes
of networks nodes and edges in KBase by adding e.g., genome-genome
nodes for ecosystems deduced from metagenomic data.
<h3> Functional types of networks in KBase </h3>

KBase network data currently come in four distinct types:
<ul>

  <li><em>REGULATORY_NETWORK</em> - a gene-gene edge represents a
transcriptional
regulatory interaction between two genes directed from a transcriptional
regulator to its target. This edge can be positive or negative,
representing activation and repression correspondingly. An alternative
representation of a regulatory network uses gene-cluster edges
representing a gene&#8217;s membership in a regulon (cluster). </li>
  <li><em>PROT_PROT_INTERACTION</em> - an undirected protein-protein edge in
this
network represents a physical interaction (specific binding) between a
pair of proteins. Alternatively, protein-cluster edges represent
membership of a protein in a multi-protein complex (cluster). </li>
  <li><em>METABOLIC_SUBSYSTEM</em> - a directed gene-gene edge in this network
indicates that two genes encoding enzymes are connected to each other
via a metabolite: the direction of this edge is from the enzyme having
this metabolite as a product to the one having it as one of its
substrates. Alternatively, edges of the type gene-cluster represent
membership of a gene in a given metabolic subsystem (cluster) such as
e.g. KEGG or a BioCyc metabolic pathway. </li>
  <li><em>FUNCTIONAL_ASSOCIATION </em>- an edge in this type of network
indicates that
two genes are functionally associated. Functional association is
deduced by integrating various data types including but not limited to
co-expression, co-occurrence and co-localization in sequenced genomes,
and co-occurence in abstracts of published papers, etc. Among examples of
this type of networks are AraNet (http://www.functionalnet.org/aranet/)
for <em>Arabidopsis</em> and StringDB (http://string-db.org) for most sequenced
microbial and plant genomes. </li>
</ul>

To find the current list of network types in KBase you can use
the command<code> allNetworkTypes</code>
<h3> Dataset sources for networks in KBase CS</h3>

Network data in KBase comes from a variety of public sources. The
current list includes:
<ul>

  <li><em>REGPRECISE</em> - Curated database of transcriptional regulons
reconstructed by the comparative genomic approach in a wide variety of
prokaryotic genomes (http://regprecise.lbl.gov)</li>
  <li><em>REGTRANSBASE </em>- Curated database of regulatory interactions in
prokaryotes capturing the knowledge in published scientific literature
(http://regtransbase.lbl.gov)</li>
  <li><em>CMONKEY</em> - Integrated biclustering of heterogeneous genome-wide
datasets for the inference of global regulatory networks
(http://baliga.systemsbiology.net/drupal/content/cmonkey) </li>
  <li><em>MAK_BI_CLUSTER</em> - Massive associative k-biclustering </li>
  <li><em>INTACT</em> - Plants Protein-Protein Interactions
(http://www.ebi.ac.uk/intact) </li>
  <li><em>AGRIS</em> - Database of transcriptional regulation in <em>Arabidopsis</em> collected by the Grotewold lab at Ohio State University
(http://Arabidopsis.med.ohio-state.edu). Their AtRegNet dataset
currently contains 11,355 direct interactions between TFs and target
genes. </li>
  <li><em>PLANTNET </em>- Probabilistic functional association gene network in
several plant species (currently <em>Arabidopsis</em> and Poplar) constructed by
a modified Bayesian integration of 24 types of "omics" data from
multiple organisms, with each data type weighted according to how well
it links genes that are known to function together. It is based on
AraNet (http://www.functionalnet.org/aranet) and RiceNet
(http://www.functionalnet.org/ricenet) developed by collaboration among the
Lee Lab at Yonsei University, Korea; the Marcotte Lab at University of
Texas at Austin, USA; the Rhee Lab at Carnegie Institution for Science,
Stanford, USA; and the Ronald Lab at UC Davis. </li>
  <li><em>MODELSEED</em> - Metabolic subsystems and models (http://www.
theseed.org/models). </li>
  <li><em>MO</em> - A resource for browsing and comparing microbial genomes
(http://www.microbesonline.org). </li>
  <li><em>ECOCYC</em> - EcoCyc is a scientific database for the bacterium
  <em>Escherichia coli </em>K-12 MG1655 (http://www.ecocyc.org). </li>
  <li><em>PPI</em> - Other Protein-Protein Interaction datasets not yet
included in IntAct.</li>
</ul>

To find the current list of dataset sources in the KBase use the
command allDatasetSources.
<h3> Networks datasets in KBase CS</h3>

Each dataset source listed by allDatasetSources may contribute more
than one network dataset to the KBase. Examples include
organism-specific networks extracted from the IntAct, different plant
species in PLANTNET, and different versions of bi-clusters.
To find the current list of datasets in the KBase use the command
allDatasets.
<h2> Learning how to extract specific subsets of network data </h2>

<h3> Finding first neighbors of a node </h3>

The API command buildFirstNeighborNetwork generates the network
connecting user-specified node to its first neighbors in user-specified
list of network datasets and network data types.
To use it efficiently you need to be familiar with how to find the
genes and genomes you are interested in described in <a href="http://www.kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/">Getting
What You
Need from the CS Using Command-Line Scripts</a> tutorial. <br>

<br>

For example to find KBase ID for Escherichia coli K12 genome we used
the
command:
<br>

<br>

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; all_entities_Genome -f scientific_name |
grep Escherichia <br>
<br>
and among all E. coli strains selected the KBase ID for Escherichia coli K12.
Appropriately it is&nbsp; 'kb|g.0'. <br>


<br>

To find the list of genes in this genome we used: <br>

<br>

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; echo 'kb|g.0' |
genomes_to_fids | grep 'peg' | fids_to_functions
<br>

<br>

To find the gene encoding RecA protein we added grep 'RecA' to the
previous comand: <br>

&nbsp;&nbsp;&nbsp;&nbsp; <br>

&nbsp;&nbsp;&nbsp;&nbsp; echo 'kb|g.0' | genomes_to_fids | grep 'peg' |
fids_to_functions | grep 'RecA'
<br>

<br>

The KBase ID of the RecA gene is &#8216;kb|g.0.peg.1356&#8217;<br>

<br>

Another example is a bacterium "Shewanella oneidensis MR-1" with KBase
ID:
&#8220;kb|g.20848&#8221;. To discover what network datasets include genes from this
genome
we run: <br>

<br>

&nbsp;&nbsp; allDatasets | grep "kb|g.20848". <br>

<br>

One of the datasets listed in the output of this command has KBase ID
kb|netdataset.kb|g.20848.regulome.0 <br>

It is a regulatory network from
the REGPRECISE database.
We are interested in a gene encoding isocitrate lyase. <br>
To find its KBase ID we run
<br>

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; echo 'kb|g.20848' |
genomes_to_fids | fids_to_functions | grep
"isocitrate lyase"
<br>

<br>

which shows that in "Shewanella oneidensis MR-1" this gene&#8217;s KBase ID
is "kb|g.20848.CDS.1671".
<br>
<br>
The first neighbor network is found by running: <br>

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; echo
"kb|g.20848.CDS.1671" | buildFirstNeighborNetwork
"kb|netdataset.kb|g.20848.regulome.0" GENE_CLUSTER <br>

<br>

A variant of this
command buildFirstNeighborNetworkLimtedByStrength uses and additional
cutOff parameter setting a lover threshold on the strength of edges to
be considered. This cutOff is a part of some but not all of network
datasets such as e.g. functional association networks in PlantNet, or
co-expression networks.The format to use this command is:<br>

&nbsp;&nbsp;&nbsp;&nbsp; echo KBase_gene_ID1|
buildFirstNeighborNetworkLimtedByStrength KBase_network_dataset_ID1
GENE_GENE cutOff;
<h2> Finding a subnetwork connecting user-specified set of nodes </h2>

The command buildInternalNetwork returns the subnetwork connecting
user-specified list of nodes and constructed
from edges in the user-specified list of network datasets. The format
to use this command for 3 nodes and one
network dataset is:
<br>

<br>

&nbsp;&nbsp;&nbsp;&nbsp; echo KBase_gene_ID1 KBase_gene_ID2
KBase_gene_ID3 buildInternalNetwork
KBase_network_dataset_ID1 GENE_CLUSTER;
<br>

<br>

The command buildInternalNetworkLimitedByStrength adds a numerical
cutOff parameter setting the lower bound on the strength of edges to be
considered.
<br>

It can be invked by using:<br>

<br>

&nbsp;&nbsp;&nbsp; echo KBase_gene_ID1 KBase_gene_ID2 KBase_gene_ID3
buildInternalNetworkLimitedByStrength KBase_network_dataset_ID1
GENE_CLUSTER cutOff;
</body></html>