<h1>Index of Terms in KBase Tutorials</h1>
<b>API</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#API83">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#API221">Summary</a>
</ul>
<li>Tutorial: <i>Comparing the Functional Roles in Two Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/comparing-the-functional-roles-in-two-genomes/#API416">A Short Program for Extracting Data from the CS</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#API606">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#API620">Updating or Defining a CS-API "Well-Trodden Path" Routine</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#API2">Here</a>
</ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#API510">Introduction</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#API527">Introduction</a>
</ul>
</ul>
<b>Atomic Regulons</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Atomic Regulons267">Introduction</a>
</ul>
</ul>
<b>Bio::KBase</b>
<ul>
<li>Tutorial: <i>Developing Software in the KBase Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/developing-software-in-the-kbase-environment/#Bio::KBase723">The KBase Module</a>
</ul>
</ul>
<b>CDM</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#CDM77">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#CDM86">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#CDM222">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#CDM227">Introduction</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#CDM330">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#CDM335">Navigating the Diagram</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#CDM604">Introduction</a>
</ul>
</ul>
<b>CPAN</b>
<ul>
<li>Tutorial: <i>Getting Started with KBase</i>
<ul>
<li><a href="/for-users/get-started#iris#CPAN66">CPAN installation</a>
<li><a href="/for-users/get-started#iris#CPAN75">Prepackaged Installations</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#CPAN14">Here</a>
</ul>
</ul>
<b>CS-API</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#CS-API259">Introduction</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#CS-API603">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#CS-API622">Updating or Defining a CS-API "Well-Trodden Path" Routine</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#CS-API5">Here</a>
</ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#CS-API356">Finding identical genomes</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#CS-API357">A Small Perl Program to Invoke the CS-API Routines</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#CS-API364">Correspondence Between Genes from Two Identical Genomes</a>
</ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#CS-API509">Introduction</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#CS-API560">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>Central Data Model</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Central Data Model79">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Central Data Model87">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#Central Data Model226">Introduction</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#Central Data Model329">Introduction</a>
</ul>
</ul>
<b>Central Store</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#Central Store742">Associating Features with Protein Sequences</a>
</ul>
<li>Tutorial: <i>Annotating a Genome Using KBase Tools</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/annotating-a-genome-using-kbase-tools/#Central Store316">Select a genome</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Central Store82">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Central Store88">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#Central Store224">Introduction</a>
</ul>
<li>Tutorial: <i>Publishing a New Central Store Software Revision</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/publishing-a-new-central-store-software-revision/#Central Store715">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/publishing-a-new-central-store-software-revision/#Central Store716">Introduction</a>
</ul>
<li>Tutorial: <i>HTTP Header Extensions for KBase Messaging</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/http-header-extensions-for-kbase-messaging/#Central Store724">Extending the Request</a>
</ul>
<li>Tutorial: <i>Extracting Genomes, Features, and Sequences from the KBase CS</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/extracting-genomes-features-and-sequences-from-the-kbase-cs/#Central Store348">Here</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#Central Store1">Here</a>
</ul>
</ul>
<b>Co-occurring</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Co-occurring273">Introduction</a>
</ul>
</ul>
<b>Coexpressed</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Coexpressed269">Introduction</a>
</ul>
</ul>
<b>Concerns</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Concerns94">How to Work with the CDM Entity-Relationship Model</a>
</ul>
</ul>
<b>Contig</b>
<ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#Contig366">A Short Program: Two Genes Correspond IFF Locations Are Identical </a>
</ul>
</ul>
<b>ContigSequence</b>
<ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#ContigSequence367">A Short Program: Two Genes Correspond IFF Locations Are Identical </a>
</ul>
</ul>
<b>DNA</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#DNA262">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#DNA140">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Extracting Genomes, Features, and Sequences from the KBase CS</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/extracting-genomes-features-and-sequences-from-the-kbase-cs/#DNA350">Extracting the Genomic Contigs in FASTA Format</a>
<li><a href="http://kbase.us/developer-zone/tutorials/extracting-genomes-features-and-sequences-from-the-kbase-cs/#DNA352">Getting the Sequences of Features</a>
</ul>
<li>Tutorial: <i>Constructing RAST2 in the Iris Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#DNA41"> Tools for Analysis </a>
</ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#DNA511">Introduction</a>
</ul>
</ul>
<b>E.coli</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#E.coli124">How to Work with the CDM Entity-Relationship Model</a>
</ul>
</ul>
<b>ER-Model</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#ER-Model81">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#ER-Model89">How to Work with the CDM Entity-Relationship Model</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#ER-Model129">Extracting Fields from a Set of Instances of an Entity</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#ER-Model333">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#ER-Model346">Navigating the Diagram</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#ER-Model605">Introduction</a>
</ul>
</ul>
<b>Entity-Relationship Model</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Entity-Relationship Model78">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Entity-Relationship Model220">Summary</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#Entity-Relationship Model223">Here</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#Entity-Relationship Model332">Introduction</a>
</ul>
</ul>
<b>Environment</b>
<ul>
<li>Tutorial: <i>Developing Software in the KBase Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/developing-software-in-the-kbase-environment/#Environment721">Here</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#Environment16">Here</a>
</ul>
</ul>
<b>Exemplars</b>
<ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#Exemplars8">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#Exemplars492">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#Exemplars458">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary of Function</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-of-function/#Exemplars424">A Three-Part Tutorial</a>
</ul>
</ul>
<b>Family</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Family271">Introduction</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#Family392">Accessing the FIGfams from within KBase</a>
</ul>
</ul>
<b>Feature</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#Feature739">Getting the Primary Function Associated with a Feature</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#Feature741">Associating Features with Protein Sequences</a>
</ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Feature261">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Feature84">How to Work with the CDM Entity-Relationship Model</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Feature136">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#Feature344">Navigating the Diagram</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#Feature443">The Concept of Exemplars</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#Feature459">Here</a>
</ul>
</ul>
<b>Genome</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Genome260">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Genome121">How to Work with the CDM Entity-Relationship Model</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Genome130">Extracting Fields from a Set of Instances of an Entity</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#Genome336">Navigating the Diagram</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#Genome565">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>Identical Genomes</b>
<ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#Identical Genomes7">Here</a>
</ul>
</ul>
<b>IsCoregulatedWith</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#IsCoregulatedWith85">How to Work with the CDM Entity-Relationship Model</a>
</ul>
</ul>
<b>IsOwnerOf</b>
<ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#IsOwnerOf340">Navigating the Diagram</a>
</ul>
</ul>
<b>IsProteinFor</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#IsProteinFor110">How to Work with the CDM Entity-Relationship Model</a>
</ul>
</ul>
<b>Location</b>
<ul>
<li>Tutorial: <i>HTTP Header Extensions for KBase Messaging</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/http-header-extensions-for-kbase-messaging/#Location725">Extending the Request</a>
<li><a href="http://kbase.us/developer-zone/tutorials/http-header-extensions-for-kbase-messaging/#Location726">Retrieving Backgrounded Results</a>
</ul>
</ul>
<b>MicrobesOnLine</b>
<ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#MicrobesOnLine12">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#MicrobesOnLine497">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#MicrobesOnLine461">Creating a Translation Table</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#MicrobesOnLine530">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>Model</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Model126">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#Model428">Here</a>
</ul>
</ul>
<b>OTU</b>
<ul>
<li>Tutorial: <i>Computing Pan Genomes for OTU's</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/computing-pan-genomes-for-otus/#OTU735">Invoking CS Command scripts from Perl</a>
<li><a href="http://kbase.us/developer-zone/tutorials/computing-pan-genomes-for-otus/#OTU736"> Example 2</a>
</ul>
</ul>
<b>Pearson Correlation Coefficient</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Pearson Correlation Coefficient268">Introduction</a>
</ul>
<li>Tutorial: <i>The Issue of Retrieving Functional Coupling Data</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/the-issue-of-retrieving-functional-coupling-data/#Pearson Correlation Coefficient421">Discussion</a>
</ul>
</ul>
<b>Protein Sequence</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Protein Sequence270">Introduction</a>
</ul>
</ul>
<b>ProteinSequence</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#ProteinSequence743">Associating Features with Protein Sequences</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#ProteinSequence93">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#ProteinSequence442">The Concept of Exemplars</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#ProteinSequence533">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>Protocol</b>
<ul>
<li>Tutorial: <i>HTTP Header Extensions for KBase Messaging</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/http-header-extensions-for-kbase-messaging/#Protocol727">References</a>
</ul>
</ul>
<b>Publication</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Publication90">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#Publication447">The Concept of Exemplars</a>
</ul>
</ul>
<b>Reaction</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Reaction125">How to Work with the CDM Entity-Relationship Model</a>
</ul>
</ul>
<b>RegPrecise</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#RegPrecise153">Scripts Handling Popular Operations</a>
</ul>
</ul>
<b>Role</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Role274">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Role159">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#Role564">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>SEED</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#SEED745">Associating Assignments with Features and Protein Sequences</a>
</ul>
<li>Tutorial: <i>Getting Started with KBase</i>
<ul>
<li><a href="/for-users/get-started#iris#SEED76">Prepackaged Installations</a>
</ul>
<li>Tutorial: <i>Publishing a New Central Store Software Revision</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/publishing-a-new-central-store-software-revision/#SEED717">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/publishing-a-new-central-store-software-revision/#SEED720">Installing the new release into the KBase git</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#SEED601">Introduction</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#SEED15">Here</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#SEED377">What is a Protein Family?</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#SEED427">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#SEED493">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#SEED498">Begin with the SEED Translation</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#SEED508">Create a Table of Data Relating to the MOL Genomes</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#SEED462">Creating a Translation Table</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#SEED483">Generating the SEED Translations</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#SEED529">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>annotations</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#annotations738">Introduction</a>
</ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#annotations286">Introduction</a>
</ul>
<li>Tutorial: <i>Annotating a Genome Using KBase Tools</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/annotating-a-genome-using-kbase-tools/#annotations317">Annotate a genome object</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotating-a-genome-using-kbase-tools/#annotations322">Get roles that might impact metabolic models</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotating-a-genome-using-kbase-tools/#annotations325">Get roles that might have been missed</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#annotations135">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Comparing the Functional Roles in Two Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/comparing-the-functional-roles-in-two-genomes/#annotations413">Introduction</a>
</ul>
<li>Tutorial: <i>Constructing RAST2 in the Iris Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#annotations32">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#annotations38">The Concept of the Genome Typed Object</a>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#annotations39"> Tools for Analysis </a>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#annotations52">Summary</a>
</ul>
<li>Tutorial: <i>Looking for Features with Identical Sequence that have Different Annotations</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/looking-for-features-with-identical-sequence-that-have-different-annotations/#annotations408">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/looking-for-features-with-identical-sequence-that-have-different-annotations/#annotations411">A Program to Look Up Functions Associated with Feature IDs</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#annotations10">Here</a>
</ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#annotations362">Controlling the Output </a>
</ul>
<li>Tutorial: <i>The Issue of Retrieving Functional Coupling Data</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/the-issue-of-retrieving-functional-coupling-data/#annotations419">Introduction</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#annotations457">The Concept of Exemplars</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#annotations496">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#annotations460">Creating a Translation Table</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#annotations490">Generating the SEED Translations</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#annotations526">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#annotations528">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>chromosome</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#chromosome288">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#chromosome139">Scripts Handling Popular Operations</a>
</ul>
</ul>
<b>close_genomes</b>
<ul>
<li>Tutorial: <i>Constructing RAST2 in the Iris Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#close_genomes43"> Tools for Analysis </a>
</ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#close_genomes519">The close_genomes Operator</a>
</ul>
</ul>
<b>co-regulated</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#co-regulated152">Scripts Handling Popular Operations</a>
</ul>
</ul>
<b>conserved neighborhood</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#conserved neighborhood289">Introduction</a>
</ul>
</ul>
<b>contig</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#contig144">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Extracting Genomes, Features, and Sequences from the KBase CS</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/extracting-genomes-features-and-sequences-from-the-kbase-cs/#contig351">Getting the Features for a Genome</a>
</ul>
</ul>
<b>corresponds</b>
<ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#corresponds365">A Short Program: Two Genes Correspond IFF Locations Are Identical </a>
</ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#corresponds512">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#corresponds513">The corresponds Operator</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#corresponds522">The close_genomes Operator</a>
</ul>
</ul>
<b>cut</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#cut250">The Abstract View</a>
</ul>
<li>Tutorial: <i>Getting Started with KBase</i>
<ul>
<li><a href="/for-users/get-started#iris#cut73">CPAN installation</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises using the KBase Iris interface</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/iris/some-basic-exercises-using-the-kbase-iris-interface/#cut64">Iris UNIX command summary;</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#cut621">Updating or Defining a CS-API "Well-Trodden Path" Routine</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#cut440">The Concept of Exemplars</a>
</ul>
</ul>
<b>cvs</b>
<ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#cvs632">Updating or Defining a CS Command-Line Script</a>
</ul>
</ul>
<b>families</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#families283">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#families146">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#families374">What is a Protein Family?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#families387">What Are FIGfams Used for?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#families390">Accessing the FIGfams from within KBase</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#families405">Coming from Roles in Models Instead</a>
</ul>
</ul>
<b>family</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#family147">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Developing Software in the KBase Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/developing-software-in-the-kbase-environment/#family722">A Note on Languages</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#family373">What is a Protein Family?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#family382">What Are FIGfams Used for?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#family396">Accessing the FIGfams from within KBase</a>
</ul>
</ul>
<b>generator</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#generator237">The Abstract View</a>
</ul>
</ul>
<b>generators</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#generators240">The Abstract View</a>
</ul>
</ul>
<b>grep</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#grep123">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#grep230">The Basic Philosophy of the CS Command-line Tools</a>
</ul>
<li>Tutorial: <i>Getting Started with KBase</i>
<ul>
<li><a href="/for-users/get-started#iris#grep74">CPAN installation</a>
</ul>
<li>Tutorial: <i>Extracting Genomes, Features, and Sequences from the KBase CS</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/extracting-genomes-features-and-sequences-from-the-kbase-cs/#grep349">Exploring the Collection of Genomes in the CS</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises using the KBase Iris interface</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/iris/some-basic-exercises-using-the-kbase-iris-interface/#grep54"> Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/iris/some-basic-exercises-using-the-kbase-iris-interface/#grep61">Additional Features of Iris</a>
<li><a href="http://kbase.us/developer-zone/tutorials/iris/some-basic-exercises-using-the-kbase-iris-interface/#grep63">Iris UNIX command summary;</a>
</ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#grep355">Finding identical genomes</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#grep567">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>literature</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#literature284">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#literature143">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#literature435">The Concept of Exemplars</a>
</ul>
</ul>
<b>md5</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#md5299">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#md5148">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#md5251">The Abstract View</a>
</ul>
<li>Tutorial: <i>Looking for Features with Identical Sequence that have Different Annotations</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/looking-for-features-with-identical-sequence-that-have-different-annotations/#md5410">A Program to Look Up Functions Associated with Feature IDs</a>
</ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#md5354">Finding identical genomes</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#md5358">A Small Perl Program to Invoke the CS-API Routines</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#md5363">Correspondence Between Genes from Two Identical Genomes</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#md5368">A Short Program: Two Genes Correspond IFF Locations Are Identical </a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#md5441">The Concept of Exemplars</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#md5501">Begin with the SEED Translation</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#md5504">Create a Table of Data Relating to the MOL Genomes</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#md5467">Creating a Translation Table</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#md5482">Generating the SEED Translations</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#md5532">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>metabolic</b>
<ul>
<li>Tutorial: <i>Annotating a Genome Using KBase Tools</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/annotating-a-genome-using-kbase-tools/#metabolic315">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotating-a-genome-using-kbase-tools/#metabolic318">Build a metabolic reconstruction from an annotated genome</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotating-a-genome-using-kbase-tools/#metabolic323">Get roles that might impact metabolic models</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotating-a-genome-using-kbase-tools/#metabolic326">Create an initial metabolic model</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotating-a-genome-using-kbase-tools/#metabolic328">Export the model to external tools</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#metabolic185">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#metabolic225">Introduction</a>
</ul>
<li>Tutorial: <i>Comparing the Functional Roles in Two Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/comparing-the-functional-roles-in-two-genomes/#metabolic412">Introduction</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#metabolic347">Navigating the Diagram</a>
</ul>
<li>Tutorial: <i>Constructing RAST2 in the Iris Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#metabolic37">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#metabolic51"> Tools for Analysis </a>
</ul>
<li>Tutorial: <i>Reconstruction and Analysis of Metabolic Models in KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/reconstruction-and-analysis-of-metabolic-models-in-kbase/#metabolic708">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/reconstruction-and-analysis-of-metabolic-models-in-kbase/#metabolic714">Summary</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#metabolic4">Here</a>
</ul>
<li>Tutorial: <i>The Issue of Retrieving Functional Coupling Data</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/the-issue-of-retrieving-functional-coupling-data/#metabolic420">Introduction</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#metabolic426">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#metabolic432">Limiting the Scope</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#metabolic540">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>multifunctional genes</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#multifunctional genes157">Scripts Handling Popular Operations</a>
</ul>
</ul>
<b>peg</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#peg740">Getting the Primary Function Associated with a Feature</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#peg744">Associating Assignments with Features and Protein Sequences</a>
</ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#peg290">Introduction</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#peg234"> A Short Exercise</a>
</ul>
<li>Tutorial: <i>Comparing the Functional Roles in Two Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/comparing-the-functional-roles-in-two-genomes/#peg417">A Short Program for Extracting Data from the CS</a>
</ul>
<li>Tutorial: <i>Looking for Features with Identical Sequence that have Different Annotations</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/looking-for-features-with-identical-sequence-that-have-different-annotations/#peg407">Here</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#peg21">Here</a>
</ul>
<li>Tutorial: <i>The Issue of Retrieving Functional Coupling Data</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/the-issue-of-retrieving-functional-coupling-data/#peg422">Discussion</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#peg450">The Concept of Exemplars</a>
</ul>
</ul>
<b>pipeline</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#pipeline229">The Basic Philosophy of the CS Command-line Tools</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#pipeline235"> A Short Exercise</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#pipeline236">The Abstract View</a>
</ul>
<li>Tutorial: <i>Constructing RAST2 in the Iris Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#pipeline50"> Tools for Analysis </a>
</ul>
<li>Tutorial: <i>Some Basic Exercises using the KBase Iris interface</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/iris/some-basic-exercises-using-the-kbase-iris-interface/#pipeline53"> Introduction</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#pipeline391">Accessing the FIGfams from within KBase</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#pipeline563">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>protein families</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#protein families282">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#protein families145">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Constructing RAST2 in the Iris Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#protein families42"> Tools for Analysis </a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein families369">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein families371">What is a Protein Family?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein families380">What Are FIGfams Used for?</a>
</ul>
</ul>
<b>protein family</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#protein family176">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein family370">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein family372">What is a Protein Family?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein family399">Accessing the FIGfams from within KBase</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein family404">Coming from Roles in Models Instead</a>
</ul>
</ul>
<b>regulon</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#regulon298">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#regulon131">Scripts Handling Popular Operations</a>
</ul>
</ul>
<b>regulons</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#regulons296">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#regulons134">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#regulons520">The close_genomes Operator</a>
</ul>
</ul>
<b>sort</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#sort252">The Abstract View</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises using the KBase Iris interface</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/iris/some-basic-exercises-using-the-kbase-iris-interface/#sort62">Iris UNIX command summary;</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#sort393">Accessing the FIGfams from within KBase</a>
</ul>
<li>Tutorial: <i>The Issue of Retrieving Functional Coupling Data</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/the-issue-of-retrieving-functional-coupling-data/#sort423">Discussion</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#sort502">Begin with the SEED Translation</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#sort481">Generating the SEED Translations</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#sort569">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>subsystem</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#subsystem278">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#subsystem158">Scripts Handling Popular Operations</a>
</ul>
</ul>
<b>subsystems</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#subsystems276">Introduction</a>
</ul>
<li>Tutorial: <i>Annotating a Genome Using KBase Tools</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/annotating-a-genome-using-kbase-tools/#subsystems319">Build a metabolic reconstruction from an annotated genome</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#subsystems160">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Constructing RAST2 in the Iris Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#subsystems34">Here</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#subsystems385">What Are FIGfams Used for?</a>
</ul>
</ul>
<b>wc</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#wc254">The Abstract View</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises using the KBase Iris interface</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/iris/some-basic-exercises-using-the-kbase-iris-interface/#wc65">Iris UNIX command summary;</a>
</ul>
</ul>
