<!DOCTYPE HTML>
<HTML>
	<HEAD>
		<TITLE>
			Building A New Genome Scale Metabolic Model Reconstruction
		</TITLE>
	</HEAD>
	<BODY>
		<H1>
			Tutorial: Building a new genome scale metabolic model reconstruction
		</H1>
		<H2>
			Introduction
		</H2>
			This tutorial will teach you how to build a genome scale metabolic model reconstruction starting from an annotated genome object using the DOE Systems Biology Knowledgebase tools. In this example, we will create a draft model for Pseudomonas stutzeri A1501, gapfill the model on complete media and then calculate the fluxes on complete media. 
		<P>
			<B>Prerequisites:</B> IRIS, Genome Annotation Tutorial, Workspaces Tutorial
		</P>
		
		<H2>
			Building a whole genome metabolic model
		</H2>

		<H3>
			Getting Started
		</H3>
		First, you must register for a <A HREF="https://gologin.kbase.us/SignUp">KBase account</A>. The KBase account is necessary to accesss the KBase tools and data stores. You will also need have access to the KBase command-line scripts. This can be done in three ways: installing the KBase scripts on a LINUX machine you have access, downloading the <A HREF="http://www.kbase.us/developer-zone/downloads/">mac DMG</A> or using the <A HREF="http://140.221.84.183/iris/Iris/">IRIS</A> web page. IRIS provides browser based access to a command-line environment with all the required KBase scripts and utilities configured. It is the easiest way to get started, please refer to the IRIS tutorial. 
		<img src="/docs/KBaseFBAModeling/tutorials/img/IRIS.png"> 
		To save your work and your share results, all input and output data objects will be saved to your workspace. Please refer to the Workspace Tutorial for more detailed instructions on how to effectively use the workspaces. In addition to the command-line scripts for accessing the workspaces, there is also a web based graphical <A HREF="http://140.221.92.49/">Workspace Browser</A> that will be used in this tutorial.
		<H3>
			Setting up a workspace
		</H3>
		<P>
			Set up a workspace to store the input and output data objects. We will do this using the <a href="http:140.221.92.49">Workspace Browser</a>. Open the Workspace Browser and sign in by clicking on the "Sign In" link in the upper right courner. You will see a list of workspaces available workspace on the left hand side. Since we want to create a new workspace, scroll to the bottom of this list and click on the "+create" link to create a new workspace for this FBA project. You must pick a workspace name and decide what permissions the public will have for this workspace. The public options are none, read or write. Below, I am creating a workspace name "pseudomonas" that will be publicly readable.
			<img src="/docs/KBaseFBAModeling/tutorials/img/CreateWorkspace.png"> 
			I can also share this workspace with individual users from the Workspace Browser.
		</P>
		<P>
			Now, I can return to the IRIS command-line tool and select the workspace I just created, "pseudomonas" as my active workspace. First, I must login to the KBase system from IRIS. 
			From the IRIS command prompt, use the <i>authenticate</i> command to login. This will bring up a login window where you can enter your KBase username and password.
			<img src="/docs/KBaseFBAModeling/tutorials/img/IRISLogin.png">
			Once logged in, I will set the "pseudomonas" workspace I created as my active workspace. This is done using the <i>kbws-workspace</i> command. By running the command with no argument, I will see my current workspace. By running it with "pseudomonas" as the argument, I will select "pseudomonas" as my workspace.
			<img src="/docs/KBaseFBAModeling/tutorials/img/IRISSetWorkspace.png"> 
			IRIS also contains several commands for the creation and manipulation of workspaces and data objects inside workspaces. a more detailed explaination of those tools can be found in the Workspace tutorial. The workspace commands we will use here are <i>kbws-listobj</i> to view the contents of our workspace and <i>kbws-jobs</i> to see our currently running jobs.

		<H3>
			Selecting a Genome
		</H3>
		<P>
			We must decide which genome we want to use to build a model. We can either select a genome that is already in our Central Data Store (CDS) or we can use a genome that we have annotated ourselves. In this tutorial, I will use the genome of <i>Pseudomonas stutzeri</i> A1501 from the CDS. If you want to use a genome not in the CDS, or alter the annotation of a KBase genome, please refer to the Genome Annotation Tutorial which will guide you through the process of annotating a genome from an assembly fasta file.
		</P>
		<P>
			To import a genome fromt the CDS into our workspace, we use the command <i>kbfba-loadgenome</i>. To use this command, we must give it a valid KBase genome id. The easiest way to find the KBase genome id is to use the <A HREF="http://140.221.92.12/">Genome Browser</A>. Since I know what genome I am looking for, I will type the name into the search box. As I type, the displayed set of genomes will change. I can click on the scientific name of the organism and bring up more information about the genome and make sure the annotation is up to date.
			<img src="/docs/KBaseFBAModeling/tutorials/img/GenomeBrowserPseudomonas.png">
		</P>
		<P>
			We will use the "kb|g.1147" KBase genome id. Now we can return to IRIS and load the genome into our workspace by using the <i>kbfba-loadfbagenome</i> command with "kb|g.1147" as it's argument. Since our workspace is still set to "pseudomonas," our genome object will be created here. 
			<img src="/docs/KBaseFBAModeling/tutorials/img/IRISLoadFBAGenome.png">
			You can use the <i>kbws-listobj</i> command to verify that the "kb|g.1147" genome object has been created in your workspace.
		</P> 
		<H3>
			Draft Model
		</H3>
		<P>
			Now that we have the genome object in our workspace, we can build a draft model using our genome object. The draft model is based only on the existing gene annotations in our genome model. It is relatively simple using the <i>kbfba-buildmodel</i> command with our genome as the argument:
			<img src="/docs/KBaseFBAModeling/tutorials/img/IRISbuildfbamodelrunning.png">
		</P>
		<P>
			Here, you may notice that the <i>kbfba-buildmodel</i> command has not completed. Although the IRIS command prompt has returned, and we can execute more commands. Our <i>kbfba-buildmodel</i> command is still running and can be seen on the left side of the IRIS screen, under the "Running processes" header. When the command completes, it will be removed from under the "Running processes" header and a "Command complete" message will be printed to the IRIS screen.
		</P>
		<H3>
			View Draft Model
		</H3>
		<P>
			Once the command is completed, we can view our draft model. The easiest way to do this is to return to our Workspace Browser. Note: you may have to reload the page to refresh the workspace contents. We should now see that our genome object and model object are in our workspace. 
			<img src="/docs/KBaseFBAModeling/tutorials/img/WBdraftmodel.png">
			To view more information about the model, we can click on the model name to bring up a popup with additional metadata for our model object. To view the actual model data, click on the "View HTML" link in the popup. This will open a new page showing the model contents. 
			<img src="/docs/KBaseFBAModeling/tutorials/img/WBviewdraftmodel.png">
			In this view, we see several tabs with detailed information about the model. Since this is a draft model, the "Gapfilling" and "Gapgen" tabs are empty. 
		</P>
		<P>
			It is also possible to view the model in the graphical metbolic map view, GLAMM. Please refer to the GLAMM tutorial for instructions.
		</P>
		<H2>
			Gapfill Model
		</H2>
		<P>
			Generally, the draft model created by <i>kbfba-buildfbamodel</i> won't grow. This is mainly due to gene annotations missing in the genome object. These annotation include key metabolic genes and transporter necessary for growth in a given media. The gap filling process compares the set of reactions in our model to a database of all known reactions and attempts to find a set of reactions that, when added to our model, allow it to grow. This set of reactions, refered to as the gapfilling solution, can then be integrated into our model, creating a new model capable of growth on the media we used for gapfilling. 
		</P>
		<P>
			The gap filling method implements a cost function associated with each reaction and transporter to help us find the solution with the minimum number of reactions and transporters added. This is because, without addional information about the genome annotation, experimental data or known biochemistry, the solution with the fewest additional reactions is more likely to be correct.
		</P>
		<P>
			The proposed gap filling solutions are just possible solutions based on minimizing the cost function. It is necessary to examine the set of reactions in the solution to judge whether it is plausible that these reactions are found in the organism. In order to create the best model, it is necessary to gap fill on several medias that the organism is known to grow on. Here, we will start with gap filling on complete media.
		</P>
		<h3>
			Running Gapfill
		</h3>
		<p>
			We can gapfill our model using the <i>kbfba-gapfill</i> command with our model as the argument. There are several important options that this program can use. The most significant are those that deal with the methods run time. Because we cannot computationally examine all possible solutions, the method uses hueristics to explore solution space and returns the best (lowest cost) solution it has found in that time.
			<ul>
				<li><b>--numsol</b>: the number of solutions the program should return</li>
				<li><b>--timepersol</b>: time in seconds per solution. We have found that one hour (6000 sec) gives good results</li>
				<li><b>--timelimit</b>: total time to allow program to run</li>
				<li><b>--media</b>: name of media to use in gap filling. Default: Complete media</li>
				<li><b>--mediaws</b>: workspace for media. Default: current workspace</li>
				<li><b>--nosubmit</b>: in the current version of KBase it is necessary to include this option for your command to run. It will be removed in the future.</li>
			</ul>
		</p>
		<p>
			In this case, I am going to gapfill on complete media. In the interest of time, I will only ask for one solution and have the program only run for 10min. It is better to ask for more solutions and the let the program run for at least 1hr per solution.
			<img src="/docs/KBaseFBAModeling/tutorials/img/IRISgapfill.png">
		</p>
		<p>
			Because this is a long running job. The system will return a "job id" so that you can monitor its status. The "job id" is the long string listed as the "Perm ref:"
			<img src="/docs/KBaseFBAModeling/tutorials/img/IRISgapfilljobsubmitted.png">
		</p>
		<h3>
			Jobs
		</h3>
		<p>
			You can monitor your job's status by using the workspace command <i>kbws-jobs</i>. You will see the "status" of your job change from "queued" to "running" and finally to "done." Because these are long running jobs, they are excuted on a KBase compute cluster. Depending on the load on the system it can take sometime for the job go from the queue to running. Once the job is running, it should run uninterrupted until completion. You may run several jobs simultaneously, they will be run as cluster nodes become available.
			<img src="/docs/KBaseFBAModeling/tutorials/img/IRISjobsdone.png">
		</p>
		<h3>
			View Gapfill results
		</h3>
		<p>
			In the case of gap filling, the results are added to our model object. If you run <i>kbws-listobj</i> in your workspace, you will notice that no new files have been created. Again, we can use the Workspace Browser to view the results of our gapfilling.
		</p>
		<p>
			From the Workspace Browser we can click on our model, in this case "kb|g.1147.fbamdl.1", and we will see a popup with the updated metadata for the model. Clicking on the "View HTML" will bring us to the model viewer page. Now the "Gapfilling" tab in our model viewer has been populated with the gap filling solution(s) we requested. In some rare cases, it is possible that the program wasn't able to find a solution in the given amount of time. 
			<img src="/docs/KBaseFBAModeling/tutorials/img/WBviewgapfilledmodel.png">
		</p>
		<p>
			Here we see that a single solution has been returned. This solution adds 12 reactions to our initial model. This gap filling solution has not actually been integrated into the model. We can now examine the list of suggested changes and decide if we believe this solution is plausible. If we choose, we can update our model by integrating this solution, or we can return to the <i>kbfba-gapfill</i> command, change the parameters and find more solutions. In this case, I will accept this solution and integrate it into the model. To do this, I must note the solution id. The solution id is the composed of the "Gapfill simulation" field + "." + "solution" + "." and the "Solution" field: "5271DC9A-7D77-11E2-86C3-05287082D269.solution.0"
		</p>
		<h3>
			Integrate Gapfill solution
		</h3>
		<p>
			With our solution id, we can now return to IRIS and integrate that solution into the model. We will use the <i>kbfba-integratesolution</i> command giving it our model id and our solution id:
			<img src="IRISintegrate.png">
		</p>
		<h3>
			View Integrated Model
		</h3>
		<p>
			To view our integrated model, we can return to the Workspace Browser, open our model and then use the "View HTML" link to get to the Model Viewer. Now when, we look under the "Gapfill" tab, we see that the solution we integrated has had it's status updated to "Integrated" and more reactions have been added to our model. It is important to note that no new genes have been added to our model, only reactions with no associated genes.
			<img src="/docs/KBaseFBAModeling/tutorials/img/WBviewmodelintegrated.png">
		</p>
		<h2>
			Run FBA
		</h2>
		<p>
			Now that we have a gapfilled model on complete media, we can use it in a flux balance analysis to verify that the model will grow and examine the fluxes. In this example, I will be doing this using complete media. Usually, you will want to gapfill your model using a minimal media to improve it before running the FBA.
		</p>
		<p>
			To run the fba, we will return to IRIS and use the <i>kbfba-runfba</i> command giving it our model as the only argument. There are a number of options that you can explore to alter the fba performance or change the growth media. Again, the default media is Complete media.
			<img src="/docs/KBaseFBAModeling/tutorials/img/IRISrunfba.png">
		</p>
		<h3>
			View FBA results
		</h3>
		<p>
			Again, we will return to the Workspace Browser to view the FBA results. In our workspace, there is now a new file "kb|g.1147.fbamdl.1.fba.0". (It maybe necessary to refresh your Workspace Browser window.)
			<img src="/docs/KBaseFBAModeling/tutorials/img/WBfbasolution.png">
			If we click on this FBA, we will first get a popup of the FBA metadata and then again, click on the "View HTML" link. 
			<img src="/docs/KBaseFBAModeling/tutorials/img/WBviewfba.png">
			In this case, you can see that the model grows since it has a significant positive value for the "Objective value". Since we used the default settings, the objective function was for biomass production.

		<H3>
			Exploring fluxes
		</H3>
		<p>
			In the FBA viewer, there are tabs for "Overview", "Reaction Fluxes" and "Compound Fluxes". You can explore the contents of your FBA result by click on the tabs and searching for particular compounds or reactions of interest. It is also possible to view and compare multiple FBA results in the GLAMM viewer. Please refer to the GLAMM tutorial.
			<img src="/docs/KBaseFBAModeling/tutorials/img/IRISviewfbafluxes.png">
			<img src="/docs/KBaseFBAModeling/tutorials/img/IRISviewfbacompounds.png">
		</p>
	</BODY>
</HTML>