<!DOCTYPE HTML>
<HTML>

	<HEAD>

		<TITLE>
			Building A New Genome Scale Metabolic Model Reconstruction</TITLE>
	</HEAD>
	<BODY>
		<H1>
			Tutorial: Building a new genome scale metabolic model reconstruction
		</H1>
		<H2>
			Introduction
		</H2>
		This tutorial will teach you how to build a genome scale metabolic model reconstruction starting from an annotated genome object using the DOE Systems Biology Knowledgebase tools.
		<P>
			<B>Prerequisites:</B> IRIS, Genome Annotation Tutorial, Workspaces Tutorial
		</P>
		
		<H2>
			Building a whole genome metabolic model
		</H2>

		<H3>
			Getting Started
		</H3>
		First, you must have access to the KBase command-line scripts. This can be done in three ways: installing the KBase scripts on a LINUX machine you have access, downloading the <A HREF="http://www.kbase.us/developer-zone/downloads/">mac DMG</A> or using the <A HREF="http://140.221.84.183/iris/Iris/">IRIS</A> web page. IRIS provides browser based access to a command-line environment with all the required KBase scripts and utilities configured. It is the easiest way to get started, please refer to the IRIS tutorial. In order to use the workspace service, you must register for a <A HREF="https://gologin.kbase.us/SignUp">KBase account</A>. The workspaces are necessary to save your work and share results. All input and output data objects will be saved to your workspace. Please refer to the Workspace Tutorial for more detailed instructions on how to effectively use the workspaces. In addition to the command-line scripts for accessing the workspaces, there is also a web based graphical <A HREF="http://140.221.92.49/">Workspace Browser</A> that will be used in this tutorial.
		<H3>
			Setting up a workspace
		</H3>
		<P>
			Set up a workspace to store the input and output data objects. From the IRIS command prompt, use the <i>authenticate</i> command to login. 
			<PRE>>/ authenticate</PRE>
			This will bring up a login window where you can enter your KBase username and password. Once logged in we can either select an existing workspace to use, or create a new one. This can easily be done using the <A HREF="http://140.221.92.49/">Workspace Browser</A>. To do this in IRIS, use the <i>kbws-workspace</i> command with no arguments to view the currently selected workspace or give it a workspace name as an argument to set the workspace. To create a new workspace, use <i>kbws-createws WorkspaceName GlobalPermissions</i>. If you want to list the available workspaces, use the <i>kbws-list</i> command. 
			<PRE>
>/ kbws-createws psd_fba_model r
Workspace created with name:
psd_fba_model
Command completed.

>/ kbws-list
ID                    Owner          Last Mod            Objects User perm Default perm
psd_fba_model         psdehal        2013-02-20T00:24:31    0    a         r  
Command completed.

>/ kbws-workspace psd_fba_model
Current workspace is:
psd_fba_model
Command completed.
			</PRE>
			Here, I have created a new workspace called "psd_fba_model" with the permissions set to "readable" for the public. Then listed the available workspaces and set my current workspace to the "psd_fba_model".
		<H3>
			Selecting a Genome
		</H3>
		<P>
			We must decide which genome we want to use to build a model. We can either select a genome that is already in our Central Data Store (CDS) or we can use a genome that we have annotated ourselves. In this tutorial, we will select a genome from the CDS, but if you want to use a genome not in the CDS, or alter the annotation of a KBase genome, please refer to the Genome Annotation Tutorial which will guide you through the process of annotating a genome from an assembly fasta file.
		</P>
		<P>
			To import a genome fromt the CDS into our workspace, we use the command <i>kbfba-loadgenome</i>. To use this command, we must give it a valid KBase genome id. The easiest way to find the KBase genome id is to use the <A HREF="http://140.221.92.12/">Genome Browser</A>. In this example, we will use the genome of <i>Shewanella oneidensis</i> MR-1. By typing "shewanella" into the search box, we can quickly find the genomes whose scientific name match the search string.
			<img src="GenomeBrowserShewanella.png">
			We will use the "kb|g.20848" KBase genome id. We can now load the genome into our workspace by typing the following command:
			<pre>
>/ kbfba-loadgenome kb|g.20848
Genome successfully loaded to workspace:
Object ID: kb|g.20848
Type: Genome
Workspace: psd_fba_model
Owner: psdehal
Instance: 0
Moddate: 2013-02-20T01:08:35
Last cmd: genome_to_workspace
Modified by: psdehal
Perm ref: 0D013D06-7AFA-11E2-9460-C6157082D269
Checksum: cb2042097cbeff6d4d4c82e4d14997ac
scientific_name: Shewanella oneidensis MR-1
domain: Bacteria
number_features: 4741
id: kb|g.20848
gc: 2354767
size: 5131416
Command completed.
		</pre>
		The "kb|g.20848" genome of <i>Shewanella oneidensis</i> MR-1 has now been loaded into our workspace "psd_fba_model" as a genome object. The command also returns meta-data that has been associated with the object. We can verify that this genome object is in our workspace by using the <i>kbws-listobj</i> command:
		<pre>
>/ kbws-listobj
ID         Type   WS                  Vers Owner   Last modby Cmd                 Moddate
kb|g.20848 Genome 2013-02-20T01:22:06 0    psdehal psdehal    genome_to_workspace psdehal
Command completed.
		</pre>
		</P>
		<H3>Automated Model</H3>
		Now that we have the genome object in our workspace, building an automated model is relatively simple using the <i>kbfba-buildmodel</i> command with our genome as the argument:
		<pre>
>/ kbfba-buildfbamodel kb|g.20848
Model successfully generated in workspace:
Object ID: kb|g.20848.fbamdl.24
Type: Model
Workspace: psd_fba_model
Owner: psdehal
Instance: 0
Moddate: 2013-02-20T01:29:19
Last cmd: genome_to_fbamodel
Modified by: psdehal
Perm ref: F2CCDFA0-7AFC-11E2-A65F-2D017082D269
Checksum: 9ede3f209a2cb696a1ca634a8c62551f
name: Shewanella oneidensis MR-1
number_compartments: 2
number_compounds: 107
number_reactions: 29
number_genes: 0
number_biomasscpd: 60
id: Kbasekb|g.20848
type: Singlegenome
annotation_uuid: 0C62524A-7AFA-11E2-958C-3B1C7082D269
Command completed.
		</pre>
		<H3>Verify model creation</H3>
		Now that the command has completed, we can see using the <i>kbws-listobj</i> command that our model has finished and has the name "kb|g.20848.fbamdl.24". This name is automatically generated by the <i>kbfba-buildmodel</i> script.
		<pre>
>/ kbws-listobj
ID                   Type   WS                  Vers Owner   Last modby Cmd                 Moddate
kb|g.20848           Genome 2013-02-20T02:20:19 0    psdehal psdehal    genome_to_workspace psdehal
kb|g.20848.fbamdl.24 Model  2013-02-20T02:20:19 0    psdehal psdehal    genome_to_fbamodel  psdehal
Command completed.
		</pre>
		If we
		<H3>View model</H3>
		print to html, sbml, export for COBRA? other tools?; view in browser; info on what the model metadata fields; 
		<H3>View model attributes</H3>
		details on model fields (compounds, compartments, biomass, reactions, gapfilling?)
		<H2>FBA</H2>
		Flux balance analysis tools in the runfba command
		<H3>runfba usage</H3>
		getting usage/help/man pages/etc; conventions used by the program
		<H3>finding models in your workspace</H3>
		list command for models (already covered above, but whatever we can repeat it); check metadata to see if it's been gapfilled already
		<H3>running runfba</H3>
		actual command for creating fba output; discussion on some options/defaults; minimum you need to know; media discussion? link to media tutorial?
		<H3>View model</H3>
		same as above mostly; explaination of new metadata fields; viewing new attributes of the model; link to gapfilling tutorial if model didn't grow
		<H3>Exploring fluxes</H3>
		a little discussion; biomass; excreted compounds; input compounds consumed;
		<H3>runfba on different media</H3>
		link to media tutorial; running fba on different media, i.e. exploring carbon sources
		<H3>runfba on gene or reaction knockouts</H3>
		specifiying ko genes --geneko or reactions --rxnko and running runfba
		<H3>changing the objective funcion</H3>
		this is more complicated; assume the user mostly knows what they're doing and understands most of the terms; maybe an "Aside" box with more details

	</BODY>
</HTML>