<html>
<head>
	<title>KBase Genome Information Browser</title>
</head>

<body>
	<h1>Using the KBase Genome Information Browser</h1>
	<p>
		<b>Purpose:</b> This tutorial demonstrates how to browse some basic information regarding genomes stored in KBase.
	</p>
	<p>
		<b>Required Prerequisite Activities:</b> None
	</p>
	<p>
		<b>Suggested Prerequisite Activities:</b> Getting Started with KBase
	</p>
	<p>
		<b>Related Tutorials:</b> Model Builder, Model Viewer, GLAMM
	</p>

	<h2>Splash Page</h2>
	<p>
		On entering the Genome Information Browser from KBase Labs, you will be greeted with a splash page describing the browser and linking to this tutorial. Click on the "Get Started" button to start the info browser.
	</p>

	<h2>Navigation Bar</h2>
	<p>
		The navigation bar exists on all Genome Information Browser pages. 
		<ul>
			<li>Clicking the KBase logo in the upper-left will always return you to the KBase homepage.</li>
			<li>The "Genome Info" link on the left side will always take you to the Genome Information Browser main page.</li>
			<li>The About link takes you to a brief page containing general information about this webpage and KBase Labs.</li>
			<li>The Contact link takes you to a page with ways to contact the author and maintainer of the Genome Information Browser.</li>
		</ul> 
	</p>

	<h2>Main page</h2>
	<p>
		On loading the main page of the Genome Information Browser, you will be greeted with a table listing all genomes stored in KBase. This table lists the KBase ID, scientific name, and biological domain of each genome. Clicking the header for each of these columns will sort the table by that column.
	</p>
	<p>
		The search area on the upper right of the table allows you to view a subset of genomes as you type. For example, typing "arabidop" will show all <i>Arabidopsis thaliana</i> genomes as well as <i>Arabidopsis lyrata</i>. See the image below.
	</p>

	<img src="table_search.png" /><br />

	<p>
		Finally, clicking any of the scientific names of the species will take you to the information page about that genome.
	</p>

	<h2>Genome Information Page</h2>
	<p>
		The Genome Information Page contains a summary of information stored in KBase regarding a selected genome. For this tutorial, we will use the example of <i>Shewanella oneidensis</i> MR-1
	</p> 

	<p>
		As in the image below, every genome information page has 3 main components:
		<ol>
			<li>A title bar with the scientific name and a "return to index" link. The link will take you back to the table of all genomes.</li>
			<li>A table of genome information on the left.</li>
			<li>A contig browser on the right.</li>
		</ol>
	</p>

	<img src="info_example.png" /><br />

	<h2>Genome Information Table</h2>
	<p>
		The general information on each genome is listed in this table, as follows:
		<ul>
			<li><b>ID</b> - the genome's KBase ID.</li>
			<li><b>Scientific Name</b> - the scientific name of the species for this genome.</li>
			<li><b>Domain</b> - the biological domain of this species (eukaryote, bacteria, archaea, virus, etc.).</li>
			<li><b>Complete?</b> - This is marked "Yes" if the genome is completely sequenced and annotated.</li>
			<li><b>Genome Size</b> - The number of nucleotide base pairs in the genome.</li>
			<li><b>GC Content</b> - The combined percentage of guanine and cytosine nucleotides in the genome.</li>
			<li><b>Number of Contigs</b> - The number of contiguous subsequences of which the genome is composed in KBase.</li>
			<li><b>Number of Features</b> - The number of genetic features in the genome (genes, non-coding RNAs, promoter sequences, operons, etc.).</li>
			<li><b>Number of Protein Encoding Genes</b> - The number of features in the genome which are annotated as protein encoding genes.</li>
			<li><b>Number of RNA Features</b> - The number of features in the genome which are annotated as RNA.</li>
			<li><b>Source</b> - The original source of the genome and its ID in that source (e.g. MOL (MicrobesOnline), SEED, iPlant, Ensembl).</li>
			<li><b>Taxonomy</b> - The detailed taxonomic classificaiton of the genome.</li>
		</ul>
	</p>

	<h2>Contig Browser</h2>
	<p>
		The contig browser is a simple widget that allows you to interactively view the feature content of a genome. It is used as follows.
	</p>
	<p>
		First, select a contig from the dropdown menu at the top of the browser. If there is only one contig, then only one element will be present in the menu. Once a contig is selected, the first features present along the first 10,000 bases of the contig will be displayed as red arrows. The direction of the arrow shows the read direction of the feature: features that point to the right are read in the same direction as the genome sequence as stored in KBase, while features that point to the left are read in the opposite direction along the complementary strand of the stored sequence. Features that overlap each other are displayed on separate rows, as in the figure above. Featured composed of multiple regions have their regions linked by a horizontal line.
	</p>
	<p>
		Navigating the contig is done via the six buttons below the browser.
	</p>

	<img src="double_left.png" /> will move the viewer to the first base of the contig.<br />
	<img src="left.png" /> will step the viewer to the left (i.e. upstream).<br />
	<img src="plus.png" /> will zoom the viewer's range in to the contig.<br />
	<img src="minus.png" /> will zoom the viewer's range out from the contig.<br />
	<img src="right.png" /> will step the viewer to the right (i.e. downstream).<br />
	<img src="double_right.png" /> will move the viewer to the last base of the contig.<br />

	<p>
		Clicking on a feature arrow will change that arrow's color from red to yellow, and show some simple information regarding that feature. This includes:
		<ul>
			<li><b>Feature ID</b> - the KBase feature ID. Clicking this ID will take you to the feature information page.</li>
			<li><b>Function</b> - the annotated function of the feature.</li>
			<li><b>Length</b> - the total number of base pairs in the feature.</li>
		</ul>
	</p>

	<img src="feature_track.png" />

	<h2>Unfinished Features</h2>
	<p>
		The remaining links on the left side of this screen - the Genome-Related Data and Genome Models sections - are currently under development. This area will be updated when they are complete.
	</p>


</body>
</html>