<div>
<h1>A Short tutorial on using genome wide association analysis (GWAS) data and performing post GWAS analysis</h1>
<strong>Required Prerequisite Activities:</strong> 
 <a href="/for-users/get-started#iris">Get Started with IRIS</a> </p>

<H2>Introduction</H2>This tutorial uses several KBase data access APIs and underlying
  databases including kb_seed, genotype_phenotype_api, networks, and
  ontology_service. In this short tutorial we will describe how to
  access GWAS data and filter candidate genes by integrating GWAS
  results with functional genomics data including expression profiles,
  co-expression networks, functional networks, gene ontology
  enrichment and pathway enrichment.
<p>


</div><div><h2>Genomes with trait</h2>
To see list of organisms/genomes in CS (KBase central store) with GWAS data, use the following command
     <i><code title = "iris-tutorials">genomes_with_trait</code></i>

This returns list of KBase genome identifiers for genomes with GWAS data.

<br><br><table border=1><tr><td><b>KBase Genome ID</b></td></tr>
<tr><td>kb|g.3899</td></tr>
<tr><td>kb|g.3907</td></tr>
</table><br><br>
where kb|g.3899 is <i>Arabidopsis thaliana</i>  and kb|g.3907 is <i>Populus trichocarpa</i>

</div><div><h2>Experiments</h2> 
We think of the plant GWAS data as being generated by experiments. A GWAS experiment consist of genotype, phenotype, and association study related data of genetically related individuals (observational units).

To get list of experiments that have been loaded into the Central Store, use the following command
     <i><code title = "iris-tutorials">get_experiments</code></i>

This should produce a 3-column table containing:

the experiment id,
design
the originator

<br><br><table border=1>
<tr><td><b>id</b></td><td><b>design</b></td><td><b>originator</b></td></tr>
<tr><td>kb|g.3899.exp.0</td><td>Short-read resequencing Richard Mott;</td><td>WTCHG(Wellcome Trust Centre for Human Genetics)</td></tr>
<tr><td>kb|g.3899.exp.1</td><td>~250K SNPs x 107 phenotpes</td><td>Atwell S, et. al.</td></tr>
<tr><td>kb|g.3907.exp.0</td><td>PoplarConsortium38K</td><td>Jerry Tuskan</td></tr>
</table><br><br>


To get experiments corresponding to a specific genome like <i>Populus trichocarpa</i>, with genome id kb|g.3907 use the following command
     get_experiments --kbgenome_id=kb|g.3907
<br><br><table border=1>
<tr><td><b>id</b></td><td><b>design</b></td><td><b>originator</b></td></tr>
<tr><td>kb|g.3907.exp.0</td><td>PoplarConsortium38K</td><td>Jerry Tuskan</td></tr>
</table><br><br>

</div><div><h2>Traits</h2>
We are interested in specific traits exhibited by the observational units (e.g., drought resistance, glucose release). For each trait of interest, measurements are taken for each of the observational units. To see list of all traits measured in an experiment try
  <code title = "iris-tutorials"> get_traits</code> --kb_experiment_id=kb|g.3907.exp.0
Here we are looking for all traits measure in experiment kb|g.3907.exp.0. This should return a table of 2-columns containing:

id: the ID of the trait,
trait_name: the text name or description of the trait.

<br><br><table border=1>
<tr><td><b>kb_trait_id</b></td><td><b>description</b></td><td><b>unit_of_measure</b></td></tr>
<tr><td>kb|g.3907.trait.3</td><td>Xylose_release</td><td>gram/gram</td></tr>
<tr><td>kb|g.3907.trait.1</td><td>S/G</td><td>ratio</td></tr>
<tr><td>kb|g.3907.trait.0</td><td>Glucose_release</td><td>gram/gram</td></tr>
<tr><td>kb|g.3907.trait.4</td><td>GlucoseXylose_release</td><td>gram/gram</td></tr>
<tr><td>kb|g.3907.trait.2</td><td>Lignin</td><td>percent</td></tr></table><br><br>

</div><div><h2>Variations(SNPs)</h2>
Genotype to phenotype associations are based on statistical analysis. To get list of variations (SNPs) significantly associated with a trait use

    <code title = "iris-tutorials">traits_to_variations</code> --kb_trait_id=kb|g.3907.trait.4 --pvalue_cutoff=0.05 > variation_details.txt

Here we are looking for variations significantly associated (below pvalue of 0.05) with trait kb|g.3907.trait.4. This should return a table of 4-columns containing:

id: KBase contig identifier
position: Base position on the contig
pvalue: pvalue of association with trait
source-contig-id:contig identifier used by source database.
Here we are saving output from this command into a file variation_details.txt. 
<br><br><table border=1>
<tr><td><b>KB Contig ID</b></td><td><b>Position</b></td><td><b>Pvalue</b></td><td><b>Source Contig ID</b></td></tr>
<tr><td>kb|g.3907.c.2535</td><td>937343 </td>	<td>0.000581</td><td>scaffold_6</td></tr>
<tr><td>kb|g.3907.c.2535</td><td>937417</td><td>0.000581</td><td>scaffold_6</td></tr>
<tr><td>kb|g.3907.c.2535</td><td>937437</td><td>0.000878</td><td>scaffold_6</td></tr>
</table><br><br>


</div><div><h2>Genes</h2>
Now to get list of genes in close vicinity of SNPs obtained earlier we need to use variations_to_genes command. This command takes positions_file which is a two column tab delimited chromosomal positions file, where column1 is kb_contig_id and column2 is position. Output obtained from previous command 
traits_to_variations returns four columns. We need to transform this to two column file, using  linux cut command like 

cut -f 1,2 variation_details.txt > positions.txt

This will return 2 column output in file  positions.txt file like : First few lines of the output are shown here.
<br><br><table border=1>
<tr><td><b>KB Contig ID</b></td><td><b>Position</b></td></tr>
<tr><td>kb|g.3907.c.2535</td><td>937343 </td>	</tr>
<tr><td>kb|g.3907.c.2535</td><td>937417</td><tr>
<tr><td>kb|g.3907.c.2535</td><td>937437</td><tr>
</table><br><br>

Now pass this file to variations_to_genes like:
<code title = "iris-tutorials">variations_to_genes</code> --positions_file=positions.txt --distance=1000 > genes.txt

Here we have specified distance as 1000, this command will look  1000 bp on both sides of SNP and returns list of all genes obtained. 

<br><br><table border=1><tr><td><b>Gene ID</b></td></tr>
<tr><td>POPTR_0005s18490</td></tr>
<tr><td>POPTR_0016s04020</td></tr>
<tr><td>POPTR_0012s13300</td></tr>
<tr><td>POPTR_0009s14090</td></tr>
<tr><td>POPTR_0003s13380</td></tr></table><br><br>


</div><div><h2>Trait to Genes</h2>
<i>traits_to_variations</i> and <i>variations_to_genes</i> commands can be combined and run as a single command <i><code title = "iris-tutorials">traits_to_genes</code></i> as:

traits_to_genes --kb_trait_id=kb|g.3907.trait.4 --pvalue_cutoff=0.05 --distance=1000

Here we have passed trait-id, pvalue and distance and the command returns list of genes associated with the trait. This command first obtain list of SNPs significantly associated (below pvalue specified) with the trait and then look for genes in close vicinity (1000 bp) of all SNPs. First few lines of the output are shown here.

<br><br><table border=1><tr><td><b>Gene ID</b></td></tr>
<tr><td>POPTR_0005s18490</td></tr>
<tr><td>POPTR_0016s04020</td></tr>
<tr><td>POPTR_0012s13300</td></tr>
<tr><td>POPTR_0009s14090</td></tr>
<tr><td>POPTR_0003s13380</td></tr></table><br><br>

</div>
