<div>
<h1>Building Combined Regulatory and Metabolic Models with PROM</h1>
<strong>Required Prerequisite Activities:</strong> 
<a href="/for-users/tutorials/installing-the-kbase-app-dmg/">Download KBase: Installing the KBase Application DMG</a>, OR <a href="/for-users/get-started#iris">Get Started with Iris</a> </p>
<p>KBase Interfaces, Iris or the Command-line, use Linux commands. To
  familiarize yourself, check out these quick guides: <a href=
  "https://www.cfa.harvard.edu/~jbattat/computer/linuxReferenceCard.pdf">
  Reference card</a>, <a href=
  "http://mally.stanford.edu/~sr/computing/basic-unix.html">Beginners
  Guide</a></p>
</div>
<div>
<h2>Introduction</h2>
<p>PROM is a method for creating FBA model constraints based on a
regulatory network and gene expression
data. Gene expression data is used to build a probablistic interaction
map of each transcription factor to
each of its target genes. These interaction probabilities are then used
to constrain the flux of reactions
under simulated transcription factor knockout conditions. KBase
provides access to a growing number of
FBA models, expression datasets, and reconstructed regulatory networks,
thus providing a useful platform
for developing FBA models with PROM constraints.</p>
<p>The KBase PROM service is an implementaton of the PROM method
originally described in Chandrasekarana and
Price, 2010. For additional technical details on the method, please
refer to this publication.</p>
  This tutorial covers basic usage of the PROM (Probabilistic Regulation
of Metabolism) service, which
enables the creation of FBA models with regulatory network constraints
inferred from gene expression data [1].
After the tuturial, you will be able to retrieve gene expression and
regulatory network data from KBase,
solve the most common gene ID problems so that all data can be placed
in the same namespace, build a set
of probabilistic constraints, and finally associate the PROM
constraints with an FBA model in order to simulate
transcription factor knockout phenotypes.
<p> [1] Chandrasekarana S. and Price ND. Probabilistic integrative
modeling of genome-scale metabolic and
regulatory networks in Escherichia coli and Mycobacterium tuberculosis.
PNAS (2010) 107:17845-50.
</div>
<div>

<h2>Create an FBA Model</h2>
<p>Before you can create a set of PROM constraints, you need to have a
functional FBA model. KBase
provides a number of options for retrieving existing models or
automatically building your own from
a genomic sequence and annotation. For more details, consult the FBA
modeling services. For the
purposes of this tutorial, we will begin by importing a published model
and genome annotation from
a public workspace. After you have configured your command line
environment, you can do this
by cloning the public workspace named "PublishedFBAModels."</p>
<pre>    &gt; <code title = "iris-tutorial">kbws-clone</code> prom_ws r -w PublishedFBAModels<br></pre>
<p>Here we copy everything in the workspace named PublishedFBAModels
and save it in a new workspace
which is called "prom_ws." We also set a flag named "r" to specify that
other users can have read-only
access to our workspace. Stricter or more open permissions are also
possible.
</p>
<p>You should now interogate your new workspace to determine what was
created. At the time of this writing,
there are two models and two genome objects which provide gene
annotations and sequence information for the
genomes which were used to construct the models. List these objects by
entering the following commands:
</p>
<pre>     &gt;<code title = "iris-tutorial"> kbws-workspace</code> prom_ws<br>     Current workspace is:<br>     prom_ws<br>     &gt; <code title = "iris-tutorial">kbws-listobj</code><br>     ID         Type   WS                  Vers Owner    Last modby Cmd                     Moddate <br>     224308.1   Genome 2013-02-12T20:01:16 1    msneddon msneddon   add_feature_translation msneddon<br>     kb|g.1870  Genome 2013-02-12T20:01:16 1    msneddon msneddon   add_feature_translation msneddon<br>     iJR904     Model  2013-02-12T20:01:16 0    msneddon msneddon   import_fbamodel         msneddon<br>     iBsu1103V2 Model  2013-02-12T20:01:16 0    msneddon msneddon   import_fbamodel         msneddon<br></pre>
<p>
In this tutorial, we will be using the FBA model named iJR904, a
curated <i>E. coli</i> model (Reed et al,
Genome Biol. 2003; 4(9): R54), which has already been linked to a KBase
genome listed here with id 'kb|g.1870'.
The model and genome object are now in our workspace and ready to use.
</p>
</div><div><h2>Load Gene Expression Data</h2>
<p>Next you will need to import gene expression data into your
workspace. The PROM service provides a
simplified script to pull gene expression data into your workspace from
the CDM in a format that is compatible
with PROM. You can try to load all expression data directly linked to
the genome kb|g.1870 with the following
command:</p>
<pre>    &gt; <code title = "iris-tutorial">prom-load-expression-data</code> -g 'kb|g.1870'<br></pre>
<p>However running this command informs us that the KBase CDM does not
currently have gene expression data
associated with this genome. You will see an error message that states:</p>
<pre>    FAILURE - no gene expression experiments found for the specified genome.<br>      -&gt; searching the KBase Central Data Store for expression data for genome: kb|g.1870<br></pre>
<p>It will often be the case that little experimental data exists for a
particular strain or genome in
the database. In this case, no data exists at all. Fortunately, data
for a related strain or genome
often is available. Depending on how similiar the strains are, it may
be valid to attempt to use data
that was originally mapped to a different genome. In our case, we
fortunately know that another <i>E.
coli</i> genome exists in KBase with ID 'kb|g.0', so we can attempt to
use data from that genome instead.
</p>
<pre>    &gt; prom-load-expression-data -g 'kb|g.0'<br></pre>
<p>Indeed, for this genome, a significant amount of gene expression
data exists. Note that this method
can take several minutes or more to complete due to the large amount of
data that is copied to your
workspace. Adding a -v option will allow you to retrieve stats on the
data that was pulled. The
output of this command is simply a workspace ID which identifies your
expression data collection. In our
case, it was named: </p>
<pre>    3CB7B210-7554-11E2-8D7C-2A08F545F304<br></pre>
<p>Since this is an inconvenient name to remember, you can relabel the
object with a simpler name:
</p>
<pre>    &gt; <code title = "iris-tutorial">kbws-move</code> 'Unspecified' '3CB7B210-7554-11E2-8D7C-2A08F545F304' 'expression_data_collection'<br>    Object moved to:<br>    prom_ws/Unspecified/expression_data_collection<br><br></pre>
<p> Now if you look at your workspace meta data, you will see that your
workspace is populated with more than
900 new objects. These new objects each correspond to a condition in
which gene expression data was
measured and is available.
</p>
<pre>    &gt; <code title = "iris-tutorial">kbws-meta</code><br>    Workspace ID: prom_published_fba_2<br>    Owner: msneddon<br>    Moddate: 2013-02-12T19:45:44<br>    Objects: 913<br>    User permission: a<br>    Global permission:r<br>    Workspace meta:<br>    1<br></pre>
<p>
Great, you now have gene expression data for <i>E. coli</i> in your
workspace. Unfortunately, that data refers to
genes by KBase feature IDs which are in the KBase genome kb|g.0. You
need gene expression data mapped to kb|g.1870,
which is the genome that is referenced by the model. These genomes are
very similar, so you can establish a mapping between
the IDs of the two genomes. There are a variety of ways to do this, but
a simple option, if both genomes are in the CDM, is
to use the corresponds method of the CDM API.
</p>
<pre>    &gt; <code title = "iris-tutorial">corresponds</code> 'kb|g.0' 'kb|g.1870' &gt; raw_map.txt<br></pre>
<p>
This command will attempt to establish a mapping between features from
kb|g.0 to features of kb|g.1870 and save the output
to a file named raw_map.txt on your local file system. You can inspect
the contents of this file like so:
</p>
<pre>    &gt; cat raw_map.txt<br>    kb|g.0.peg.2173	100.00	5	4	820	820	2	818	818	0.813	kb|g.1870.peg.1148<br>    kb|g.0.peg.3016	100.00	6	1	310	310	1	310	310	0.860	kb|g.1870.peg.2046<br>    kb|g.0.peg.4288	100.00	7	1	428	428	1	428	428	0.901	kb|g.1870.peg.3069<br>    ...<br></pre>
<p>
The first column contains a feature from kb|g.0, the last column
contains a mapping in kb|g.1870, and the inner columns
display some statistics about the mapping, which you can learn more
about by looking at documentation for the
corresponds method. Here you just need a two column file that contains
the kb|g.0 features in one column, and the
kb|g.1870 features in the second column. Do this by using the standard
linux 'cut' method to extract the 1st and the
11th column, and save the result to a file.
</p>
<pre>    &gt; cat raw_map.txt | cut -f 1,11 &gt; 0_to_1870_map.txt<br>    &gt; cat 0_to_1870_map.txt<br>    kb|g.0.peg.2173	kb|g.1870.peg.1148<br>    kb|g.0.peg.3016	kb|g.1870.peg.2046<br>    kb|g.0.peg.4288	kb|g.1870.peg.3069<br>    ...<br></pre>
<p>
Armed with a reasonable mapping between the genomes, you can now
relabel the expression data set in terms of kb|g.1870 using
a PROM service method.
</p>
<pre>    &gt; <code title = "iris-tutorial">prom-change-expression-data-namespace</code> -e 'expression_data_collection' -m '0_to_1870_map.txt' -v<br></pre>
<p>
And that's it. Running with the -v option provides some extra
information about the relabeling process. The
'expression_data_collection' is now defined in terms of kb|g.1870 and
can now be used in conjunction with the
FBA model iJR904. Note that again this step may take a several minutes
to complete due to the amount of gene
expression data that must be processed.
</p>
</div><div><h2>Load Regulatory Network</h2>
<p>The next step towards building a PROM constraints object is to load
a regulatory network. As with expression data,
regulatory network data is not always available for the genome in which
we are
interested. For this particular genome, however,
we are in luck. Run the following command to load a regulatory network
model for genome 'kb|g.1870' </p>
<pre>    <code title = "iris-tutorial">prom-load-regulatory-network</code> -g 'kb|g.1870' -v<br>    SUCCESS.<br>      -&gt; found 1 regulome model(s) for genome kb|g.1870<br>      -&gt; possible models are :<br>           -kb|g.1870.regulome.301 (regulomeSource:REGPRECISE_CURATED, tfRegulonCount:75)<br>      -&gt; selected regulome model with the most regulons (kb|g.1870.regulome.301)<br>      -&gt; compiled regulatory network with 1546 regulatory interactions<br>      -&gt; saving the regulatory network to your workspace with ID:86559BE0-7562-11E2-9E83-E214F545F304<br><br>    86559BE0-7562-11E2-9E83-E214F545F304<br></pre>
<p>Just as we did with the expression data collection, rename the
regulatory network so that it is easier to manipulate:
</p>
<pre>    &gt; <code title = "iris-tutorial">kbws-move</code> 'Unspecified' '86559BE0-7562-11E2-9E83-E214F545F304' 'regulatory_network'<br>    Object moved to:<br>    prom_ws/Unspecified/regulatory_network<br></pre>
<p>If a regulatory network is not available for your genome, you would
have received an error message stating this. If this
was the case, we again could have pulled a regulatory network for a
related <i>E. coli</i> genome, mapped the gene features
from one variant to the other, and used the method named
'prom-change-regulatory-network-namespace'. It works the same way
as the method to change the expression data namespace, except that it
creates a new regulatory network object in your workspace.
</p>
<p>As another note, you can inspect your regulatory network at any time
by fetching the regulatory network object from your
workspace using the kbws-get method. This is a convenient way to
inspect any object that is stored in your workspace during
this tutorial.
</p>
<pre>    &gt; <code title = "iris-tutorial">kbws-get</code> 'Unspecified' 'regulatory_network'<br>    kb|g.1870.peg.4264	kb|g.1870.peg.553<br>    kb|g.1870.peg.4264	kb|g.1870.peg.521<br>    kb|g.1870.peg.4264	kb|g.1870.peg.778<br>    kb|g.1870.peg.4264	kb|g.1870.peg.957<br>    kb|g.1870.peg.4264	kb|g.1870.peg.1032<br>    kb|g.1870.peg.4264	kb|g.1870.peg.263<br>    ....<br></pre>
<p>In this format, transcription factors are listed in the first column
and transcriptional targets are listed in the second
column. No other information (e.g., activator vs. repressor) is stored
because it is not required to create a set of PROM
constraints. You can use the KBase Regulation services to pull
additional information for the regulatory network models
of your genome of interest.
</p>
</div><div><h2>Create the PROM Constraints Object</h2>
<p>You can now create a PROM constraints object from the expression
data collection, the regulatory network, and the
genome object. To do so, call the prom-create-constraints method with
flags to indicate the data objects that
you created earlier.</p>
<pre>    &gt; <code title = "iris-tutorial">prom-create-constraints</code> -g 'kb|g.1870' -e 'expression_data_collection' -r 'regulatory_network' -v<br></pre>
<p>This method may also take a few minutes to run because it must pull
all the expression data in the collection
to calculate the transcriptional interaction probabilities. If you ran
the method with the '-v' option as indicated,
and the method was successful, a status message will be displayed.
</p>
<pre>    SUCCESS.<br>      -&gt; fetched genome object.<br>      -&gt; fetched hidden annotation object by reference with uuid: 'F874E632-6E81-11E2-9490-10D66F82D269'.<br>      -&gt; genome annotation has 4655 features.<br>          5 wallclock secs ( 0.09 usr +  0.37 sys =  0.46 CPU)<br>      -&gt; retrieved regulatory network with 3 regulatory interactions.<br>          5 wallclock secs ( 0.09 usr +  0.38 sys =  0.47 CPU)<br>      -&gt; retrieved expression data collection with 907 experiments.<br>          5 wallclock secs ( 0.09 usr +  0.38 sys =  0.47 CPU)<br>      -&gt; retrieved all expression data for each experiment in the collection<br>         583 wallclock secs ( 3.59 usr +  9.16 sys = 12.75 CPU)<br>      -&gt; computed regulation interaction probabilities<br>         583 wallclock secs ( 3.59 usr +  9.16 sys = 12.75 CPU)<br>      -&gt; saving the new PromModelConstraints object ID:kb|g.1870.promconstraint.3<br>         584 wallclock secs ( 3.60 usr +  9.17 sys = 12.77 CPU)<br></pre>
</div><div><h2>Run an FBA Model with a PROM Constraint</h2>
<p>You can finally run the FBA model with the new PROM constraint. To
associate the PROM constraint object
with the FBA, call kbfba-runfba:</p>
<pre>    &gt; <code title = "iris-tutorial">kbfba-runfba</code> iJR904 -p 'kb|g.1870.promconstraints.3'<br>    Flux balance analysis successful:<br>    Object ID: iJR904.fba.4<br>    Type: FBA<br>    Workspace: prom_ws<br>    Owner: msneddon<br>    Instance: 0<br>    Moddate: 2013-02-12T23:00:54<br>    Last cmd: runfba<br>    Modified by: msneddon<br>    Perm ref: 0DB89574-7568-11E2-AEF2-F6187082D269<br>    Checksum: 5363d7e64e3620d52f4efd91f8f00bd4<br>    object_value: 2.18666<br>    notes: <br>    media_uuid: 36A4741E-3A6A-11E2-837B-C916C85230E2<br></pre>
<p>You were able to successfully run the FBA model with the PROM
constraint object. As with any FBA run, you can view
the results by exporting the results object to an HTML file that you
can open in your browser using the kbfba-exportfba
method. See the FBA Modeling documenation for for details on
visualizing and analyzing the results of a model.
</p>
<p>If you closely inspected the results, you will discover that the
results are identical to running an FBA model without
a PROM constraint. The reason is that the PROM constraint object only
influences a model if you simulate a transcriptional regulator
knockout strain. Try running the following simulation with a TF
knockout: </p>
<pre>    &gt; <code title = "iris-tutorial">kbfba-runfba</code> iJR904 -p 'kb|g.1870.promconstraints.3' --geneko "kb|g.20848.cds.4235"<br>    Flux balance analysis successful:<br>    Object ID: iJR904.fba.5<br>    Type: FBA<br>    Workspace: prom_ws<br>    Owner: msneddon<br>    Instance: 0<br>    Moddate: 2013-02-12T19:04:09<br>    Last cmd: runfba<br>    Modified by: msneddon<br>    Perm ref: FAEA3B94-7546-11E2-B4AE-36E76F82D269<br>    Checksum: e42fd1db0389510403dab75bdd7b5b1b<br>    object_value: 0<br>    notes: <br>    media_uuid: 36A4741E-3A6A-11E2-837B-C916C85230E2<br></pre>
<p>By leveraging gene expression data together with a regulatory
network, you can now use PROM to increase the predictive
power of your model for transcription factor knockout strains.
</p>
</div>
