<?xml version="1.0" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
 <head>
  <title>
   Genome Annotation Service API
  </title>
  <meta content="text/html; charset=utf-8" http-equiv="content-type"/>
  <link href="mailto:root@ubuntu.chicago.kbase.us" rev="made"/>
 </head>
 <body style="background-color: white">
  <h1 id="NAME">
   NAME
  </h1>
  <p>
   GenomeAnnotation
  </p>
  <h1 id="DESCRIPTION">
   DESCRIPTION
  </h1>
  <p>
   API Access to the Genome Annotation Service.
  </p>
  <p>
   Provides support for gene calling, functional annotation, re-annotation. Use to extract annotation in formation about an existing genome, or to create new annotations.
  </p>
  <h1 id="METHODS">
   METHODS
  </h1>
  <h2 id="genomeTO_to_reconstructionTO">
   genomeTO_to_reconstructionTO
  </h2>
  <pre><code>  $return = $obj-&gt;genomeTO_to_reconstructionTO($genomeTO)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$genomeTO is a genomeTO
$return is a reconstructionTO
genomeTO is a reference to a hash where the following keys are defined:
	id has a value which is a genome_id
	scientific_name has a value which is a string
	domain has a value which is a string
	genetic_code has a value which is an int
	source has a value which is a string
	source_id has a value which is a string
	contigs has a value which is a reference to a list where each element is a contig
	features has a value which is a reference to a list where each element is a feature
genome_id is a string
contig is a reference to a hash where the following keys are defined:
	id has a value which is a contig_id
	dna has a value which is a string
contig_id is a string
feature is a reference to a hash where the following keys are defined:
	id has a value which is a feature_id
	location has a value which is a location
	type has a value which is a feature_type
	function has a value which is a string
	protein_translation has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	annotations has a value which is a reference to a list where each element is an annotation
feature_id is a string
location is a reference to a list where each element is a region_of_dna
region_of_dna is a reference to a list containing 4 items:
	0: a contig_id
	1: (begin) an int
	2: (strand) a string
	3: (length) an int
feature_type is a string
annotation is a reference to a list containing 3 items:
	0: (comment) a string
	1: (annotator) a string
	2: (annotation_time) an int
reconstructionTO is a reference to a hash where the following keys are defined:
	subsystems has a value which is a variant_subsystem_pairs
	bindings has a value which is a fid_role_pairs
	assignments has a value which is a fid_function_pairs
variant_subsystem_pairs is a reference to a list where each element is a variant_of_subsystem
variant_of_subsystem is a reference to a list containing 2 items:
	0: a subsystem
	1: a variant
subsystem is a string
variant is a string
fid_role_pairs is a reference to a list where each element is a fid_role_pair
fid_role_pair is a reference to a list containing 2 items:
	0: a fid
	1: a role
fid is a string
role is a string
fid_function_pairs is a reference to a list where each element is a fid_function_pair
fid_function_pair is a reference to a list containing 2 items:
	0: a fid
	1: a function
function is a string</pre>
   </dd>
   <dt id="Description">
    Description
   </dt>
   <dd>
   </dd>
  </dl>
  <h2 id="genomeTO_to_feature_data">
   genomeTO_to_feature_data
  </h2>
  <pre><code>  $return = $obj-&gt;genomeTO_to_feature_data($genomeTO)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types1">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$genomeTO is a genomeTO
$return is a fid_data_tuples
genomeTO is a reference to a hash where the following keys are defined:
	id has a value which is a genome_id
	scientific_name has a value which is a string
	domain has a value which is a string
	genetic_code has a value which is an int
	source has a value which is a string
	source_id has a value which is a string
	contigs has a value which is a reference to a list where each element is a contig
	features has a value which is a reference to a list where each element is a feature
genome_id is a string
contig is a reference to a hash where the following keys are defined:
	id has a value which is a contig_id
	dna has a value which is a string
contig_id is a string
feature is a reference to a hash where the following keys are defined:
	id has a value which is a feature_id
	location has a value which is a location
	type has a value which is a feature_type
	function has a value which is a string
	protein_translation has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	annotations has a value which is a reference to a list where each element is an annotation
feature_id is a string
location is a reference to a list where each element is a region_of_dna
region_of_dna is a reference to a list containing 4 items:
	0: a contig_id
	1: (begin) an int
	2: (strand) a string
	3: (length) an int
feature_type is a string
annotation is a reference to a list containing 3 items:
	0: (comment) a string
	1: (annotator) a string
	2: (annotation_time) an int
fid_data_tuples is a reference to a list where each element is a fid_data_tuple
fid_data_tuple is a reference to a list containing 4 items:
	0: a fid
	1: a md5
	2: a location
	3: a function
fid is a string
md5 is a string
function is a string</pre>
   </dd>
   <dt id="Description1">
    Description
   </dt>
   <dd>
   </dd>
  </dl>
  <h2 id="reconstructionTO_to_roles">
   reconstructionTO_to_roles
  </h2>
  <pre><code>  $return = $obj-&gt;reconstructionTO_to_roles($reconstructionTO)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types2">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$reconstructionTO is a reconstructionTO
$return is a reference to a list where each element is a role
reconstructionTO is a reference to a hash where the following keys are defined:
	subsystems has a value which is a variant_subsystem_pairs
	bindings has a value which is a fid_role_pairs
	assignments has a value which is a fid_function_pairs
variant_subsystem_pairs is a reference to a list where each element is a variant_of_subsystem
variant_of_subsystem is a reference to a list containing 2 items:
	0: a subsystem
	1: a variant
subsystem is a string
variant is a string
fid_role_pairs is a reference to a list where each element is a fid_role_pair
fid_role_pair is a reference to a list containing 2 items:
	0: a fid
	1: a role
fid is a string
role is a string
fid_function_pairs is a reference to a list where each element is a fid_function_pair
fid_function_pair is a reference to a list containing 2 items:
	0: a fid
	1: a function
function is a string</pre>
   </dd>
   <dt id="Description2">
    Description
   </dt>
   <dd>
   </dd>
  </dl>
  <h2 id="reconstructionTO_to_subsystems">
   reconstructionTO_to_subsystems
  </h2>
  <pre><code>  $return = $obj-&gt;reconstructionTO_to_subsystems($reconstructionTO)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types3">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$reconstructionTO is a reconstructionTO
$return is a variant_subsystem_pairs
reconstructionTO is a reference to a hash where the following keys are defined:
	subsystems has a value which is a variant_subsystem_pairs
	bindings has a value which is a fid_role_pairs
	assignments has a value which is a fid_function_pairs
variant_subsystem_pairs is a reference to a list where each element is a variant_of_subsystem
variant_of_subsystem is a reference to a list containing 2 items:
	0: a subsystem
	1: a variant
subsystem is a string
variant is a string
fid_role_pairs is a reference to a list where each element is a fid_role_pair
fid_role_pair is a reference to a list containing 2 items:
	0: a fid
	1: a role
fid is a string
role is a string
fid_function_pairs is a reference to a list where each element is a fid_function_pair
fid_function_pair is a reference to a list containing 2 items:
	0: a fid
	1: a function
function is a string</pre>
   </dd>
   <dt id="Description3">
    Description
   </dt>
   <dd>
   </dd>
  </dl>
  <h2 id="annotate_genome">
   annotate_genome
  </h2>
  <pre><code>  $return = $obj-&gt;annotate_genome($genomeTO)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types4">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$genomeTO is a genomeTO
$return is a genomeTO
genomeTO is a reference to a hash where the following keys are defined:
	id has a value which is a genome_id
	scientific_name has a value which is a string
	domain has a value which is a string
	genetic_code has a value which is an int
	source has a value which is a string
	source_id has a value which is a string
	contigs has a value which is a reference to a list where each element is a contig
	features has a value which is a reference to a list where each element is a feature
genome_id is a string
contig is a reference to a hash where the following keys are defined:
	id has a value which is a contig_id
	dna has a value which is a string
contig_id is a string
feature is a reference to a hash where the following keys are defined:
	id has a value which is a feature_id
	location has a value which is a location
	type has a value which is a feature_type
	function has a value which is a string
	protein_translation has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	annotations has a value which is a reference to a list where each element is an annotation
feature_id is a string
location is a reference to a list where each element is a region_of_dna
region_of_dna is a reference to a list containing 4 items:
	0: a contig_id
	1: (begin) an int
	2: (strand) a string
	3: (length) an int
feature_type is a string
annotation is a reference to a list containing 3 items:
	0: (comment) a string
	1: (annotator) a string
	2: (annotation_time) an int</pre>
   </dd>
   <dt id="Description4">
    Description
   </dt>
   <dd>
    <p>
     Given a genome object populated with contig data, perform gene calling and functional annotation and return the annotated genome. NOTE: Many of these "transformations" modify the input hash and copy the pointer. Be warned.
    </p>
   </dd>
  </dl>
  <h2 id="call_selenoproteins">
   call_selenoproteins
  </h2>
  <pre><code>  $return = $obj-&gt;call_selenoproteins($genomeTO)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types5">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$genomeTO is a genomeTO
$return is a genomeTO
genomeTO is a reference to a hash where the following keys are defined:
	id has a value which is a genome_id
	scientific_name has a value which is a string
	domain has a value which is a string
	genetic_code has a value which is an int
	source has a value which is a string
	source_id has a value which is a string
	contigs has a value which is a reference to a list where each element is a contig
	features has a value which is a reference to a list where each element is a feature
genome_id is a string
contig is a reference to a hash where the following keys are defined:
	id has a value which is a contig_id
	dna has a value which is a string
contig_id is a string
feature is a reference to a hash where the following keys are defined:
	id has a value which is a feature_id
	location has a value which is a location
	type has a value which is a feature_type
	function has a value which is a string
	protein_translation has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	annotations has a value which is a reference to a list where each element is an annotation
feature_id is a string
location is a reference to a list where each element is a region_of_dna
region_of_dna is a reference to a list containing 4 items:
	0: a contig_id
	1: (begin) an int
	2: (strand) a string
	3: (length) an int
feature_type is a string
annotation is a reference to a list containing 3 items:
	0: (comment) a string
	1: (annotator) a string
	2: (annotation_time) an int</pre>
   </dd>
   <dt id="Description5">
    Description
   </dt>
   <dd>
   </dd>
  </dl>
  <h2 id="call_pyrrolysoproteins">
   call_pyrrolysoproteins
  </h2>
  <pre><code>  $return = $obj-&gt;call_pyrrolysoproteins($genomeTO)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types6">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$genomeTO is a genomeTO
$return is a genomeTO
genomeTO is a reference to a hash where the following keys are defined:
	id has a value which is a genome_id
	scientific_name has a value which is a string
	domain has a value which is a string
	genetic_code has a value which is an int
	source has a value which is a string
	source_id has a value which is a string
	contigs has a value which is a reference to a list where each element is a contig
	features has a value which is a reference to a list where each element is a feature
genome_id is a string
contig is a reference to a hash where the following keys are defined:
	id has a value which is a contig_id
	dna has a value which is a string
contig_id is a string
feature is a reference to a hash where the following keys are defined:
	id has a value which is a feature_id
	location has a value which is a location
	type has a value which is a feature_type
	function has a value which is a string
	protein_translation has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	annotations has a value which is a reference to a list where each element is an annotation
feature_id is a string
location is a reference to a list where each element is a region_of_dna
region_of_dna is a reference to a list containing 4 items:
	0: a contig_id
	1: (begin) an int
	2: (strand) a string
	3: (length) an int
feature_type is a string
annotation is a reference to a list containing 3 items:
	0: (comment) a string
	1: (annotator) a string
	2: (annotation_time) an int</pre>
   </dd>
   <dt id="Description6">
    Description
   </dt>
   <dd>
   </dd>
  </dl>
  <h2 id="call_RNAs">
   call_RNAs
  </h2>
  <pre><code>  $return = $obj-&gt;call_RNAs($genomeTO)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types7">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$genomeTO is a genomeTO
$return is a genomeTO
genomeTO is a reference to a hash where the following keys are defined:
	id has a value which is a genome_id
	scientific_name has a value which is a string
	domain has a value which is a string
	genetic_code has a value which is an int
	source has a value which is a string
	source_id has a value which is a string
	contigs has a value which is a reference to a list where each element is a contig
	features has a value which is a reference to a list where each element is a feature
genome_id is a string
contig is a reference to a hash where the following keys are defined:
	id has a value which is a contig_id
	dna has a value which is a string
contig_id is a string
feature is a reference to a hash where the following keys are defined:
	id has a value which is a feature_id
	location has a value which is a location
	type has a value which is a feature_type
	function has a value which is a string
	protein_translation has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	annotations has a value which is a reference to a list where each element is an annotation
feature_id is a string
location is a reference to a list where each element is a region_of_dna
region_of_dna is a reference to a list containing 4 items:
	0: a contig_id
	1: (begin) an int
	2: (strand) a string
	3: (length) an int
feature_type is a string
annotation is a reference to a list containing 3 items:
	0: (comment) a string
	1: (annotator) a string
	2: (annotation_time) an int</pre>
   </dd>
   <dt id="Description7">
    Description
   </dt>
   <dd>
   </dd>
  </dl>
  <h2 id="call_CDSs">
   call_CDSs
  </h2>
  <pre><code>  $return = $obj-&gt;call_CDSs($genomeTO)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types8">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$genomeTO is a genomeTO
$return is a genomeTO
genomeTO is a reference to a hash where the following keys are defined:
	id has a value which is a genome_id
	scientific_name has a value which is a string
	domain has a value which is a string
	genetic_code has a value which is an int
	source has a value which is a string
	source_id has a value which is a string
	contigs has a value which is a reference to a list where each element is a contig
	features has a value which is a reference to a list where each element is a feature
genome_id is a string
contig is a reference to a hash where the following keys are defined:
	id has a value which is a contig_id
	dna has a value which is a string
contig_id is a string
feature is a reference to a hash where the following keys are defined:
	id has a value which is a feature_id
	location has a value which is a location
	type has a value which is a feature_type
	function has a value which is a string
	protein_translation has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	annotations has a value which is a reference to a list where each element is an annotation
feature_id is a string
location is a reference to a list where each element is a region_of_dna
region_of_dna is a reference to a list containing 4 items:
	0: a contig_id
	1: (begin) an int
	2: (strand) a string
	3: (length) an int
feature_type is a string
annotation is a reference to a list containing 3 items:
	0: (comment) a string
	1: (annotator) a string
	2: (annotation_time) an int</pre>
   </dd>
   <dt id="Description8">
    Description
   </dt>
   <dd>
   </dd>
  </dl>
  <h2 id="find_close_neighbors">
   find_close_neighbors
  </h2>
  <pre><code>  $return = $obj-&gt;find_close_neighbors($genomeTO)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types9">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$genomeTO is a genomeTO
$return is a genomeTO
genomeTO is a reference to a hash where the following keys are defined:
	id has a value which is a genome_id
	scientific_name has a value which is a string
	domain has a value which is a string
	genetic_code has a value which is an int
	source has a value which is a string
	source_id has a value which is a string
	contigs has a value which is a reference to a list where each element is a contig
	features has a value which is a reference to a list where each element is a feature
genome_id is a string
contig is a reference to a hash where the following keys are defined:
	id has a value which is a contig_id
	dna has a value which is a string
contig_id is a string
feature is a reference to a hash where the following keys are defined:
	id has a value which is a feature_id
	location has a value which is a location
	type has a value which is a feature_type
	function has a value which is a string
	protein_translation has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	annotations has a value which is a reference to a list where each element is an annotation
feature_id is a string
location is a reference to a list where each element is a region_of_dna
region_of_dna is a reference to a list containing 4 items:
	0: a contig_id
	1: (begin) an int
	2: (strand) a string
	3: (length) an int
feature_type is a string
annotation is a reference to a list containing 3 items:
	0: (comment) a string
	1: (annotator) a string
	2: (annotation_time) an int</pre>
   </dd>
   <dt id="Description9">
    Description
   </dt>
   <dd>
   </dd>
  </dl>
  <h2 id="assign_functions_to_CDSs">
   assign_functions_to_CDSs
  </h2>
  <pre><code>  $return = $obj-&gt;assign_functions_to_CDSs($genomeTO)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types10">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$genomeTO is a genomeTO
$return is a genomeTO
genomeTO is a reference to a hash where the following keys are defined:
	id has a value which is a genome_id
	scientific_name has a value which is a string
	domain has a value which is a string
	genetic_code has a value which is an int
	source has a value which is a string
	source_id has a value which is a string
	contigs has a value which is a reference to a list where each element is a contig
	features has a value which is a reference to a list where each element is a feature
genome_id is a string
contig is a reference to a hash where the following keys are defined:
	id has a value which is a contig_id
	dna has a value which is a string
contig_id is a string
feature is a reference to a hash where the following keys are defined:
	id has a value which is a feature_id
	location has a value which is a location
	type has a value which is a feature_type
	function has a value which is a string
	protein_translation has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	annotations has a value which is a reference to a list where each element is an annotation
feature_id is a string
location is a reference to a list where each element is a region_of_dna
region_of_dna is a reference to a list containing 4 items:
	0: a contig_id
	1: (begin) an int
	2: (strand) a string
	3: (length) an int
feature_type is a string
annotation is a reference to a list containing 3 items:
	0: (comment) a string
	1: (annotator) a string
	2: (annotation_time) an int</pre>
   </dd>
   <dt id="Description10">
    Description
   </dt>
   <dd>
   </dd>
  </dl>
  <h2 id="annotate_proteins">
   annotate_proteins
  </h2>
  <pre><code>  $return = $obj-&gt;annotate_proteins($genomeTO)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types11">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$genomeTO is a genomeTO
$return is a genomeTO
genomeTO is a reference to a hash where the following keys are defined:
	id has a value which is a genome_id
	scientific_name has a value which is a string
	domain has a value which is a string
	genetic_code has a value which is an int
	source has a value which is a string
	source_id has a value which is a string
	contigs has a value which is a reference to a list where each element is a contig
	features has a value which is a reference to a list where each element is a feature
genome_id is a string
contig is a reference to a hash where the following keys are defined:
	id has a value which is a contig_id
	dna has a value which is a string
contig_id is a string
feature is a reference to a hash where the following keys are defined:
	id has a value which is a feature_id
	location has a value which is a location
	type has a value which is a feature_type
	function has a value which is a string
	protein_translation has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	annotations has a value which is a reference to a list where each element is an annotation
feature_id is a string
location is a reference to a list where each element is a region_of_dna
region_of_dna is a reference to a list containing 4 items:
	0: a contig_id
	1: (begin) an int
	2: (strand) a string
	3: (length) an int
feature_type is a string
annotation is a reference to a list containing 3 items:
	0: (comment) a string
	1: (annotator) a string
	2: (annotation_time) an int</pre>
   </dd>
   <dt id="Description11">
    Description
   </dt>
   <dd>
    <p>
     Given a genome object populated with feature data, reannotate the features that have protein translations. Return the updated genome object.
    </p>
   </dd>
  </dl>
  <h2 id="call_CDSs_by_projection">
   call_CDSs_by_projection
  </h2>
  <pre><code>  $return = $obj-&gt;call_CDSs_by_projection($genomeTO)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types12">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$genomeTO is a genomeTO
$return is a genomeTO
genomeTO is a reference to a hash where the following keys are defined:
	id has a value which is a genome_id
	scientific_name has a value which is a string
	domain has a value which is a string
	genetic_code has a value which is an int
	source has a value which is a string
	source_id has a value which is a string
	contigs has a value which is a reference to a list where each element is a contig
	features has a value which is a reference to a list where each element is a feature
genome_id is a string
contig is a reference to a hash where the following keys are defined:
	id has a value which is a contig_id
	dna has a value which is a string
contig_id is a string
feature is a reference to a hash where the following keys are defined:
	id has a value which is a feature_id
	location has a value which is a location
	type has a value which is a feature_type
	function has a value which is a string
	protein_translation has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	annotations has a value which is a reference to a list where each element is an annotation
feature_id is a string
location is a reference to a list where each element is a region_of_dna
region_of_dna is a reference to a list containing 4 items:
	0: a contig_id
	1: (begin) an int
	2: (strand) a string
	3: (length) an int
feature_type is a string
annotation is a reference to a list containing 3 items:
	0: (comment) a string
	1: (annotator) a string
	2: (annotation_time) an int</pre>
   </dd>
   <dt id="Description12">
    Description
   </dt>
   <dd>
   </dd>
  </dl>
  <h2 id="get_strep_suis_repeats">
   get_strep_suis_repeats
  </h2>
  <pre><code>  $return = $obj-&gt;get_strep_suis_repeats($genomeTO)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types13">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$genomeTO is a genomeTO
$return is a genomeTO
genomeTO is a reference to a hash where the following keys are defined:
	id has a value which is a genome_id
	scientific_name has a value which is a string
	domain has a value which is a string
	genetic_code has a value which is an int
	source has a value which is a string
	source_id has a value which is a string
	contigs has a value which is a reference to a list where each element is a contig
	features has a value which is a reference to a list where each element is a feature
genome_id is a string
contig is a reference to a hash where the following keys are defined:
	id has a value which is a contig_id
	dna has a value which is a string
contig_id is a string
feature is a reference to a hash where the following keys are defined:
	id has a value which is a feature_id
	location has a value which is a location
	type has a value which is a feature_type
	function has a value which is a string
	protein_translation has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	annotations has a value which is a reference to a list where each element is an annotation
feature_id is a string
location is a reference to a list where each element is a region_of_dna
region_of_dna is a reference to a list containing 4 items:
	0: a contig_id
	1: (begin) an int
	2: (strand) a string
	3: (length) an int
feature_type is a string
annotation is a reference to a list containing 3 items:
	0: (comment) a string
	1: (annotator) a string
	2: (annotation_time) an int</pre>
   </dd>
   <dt id="Description13">
    Description
   </dt>
   <dd>
    <p>
     Interface to Strep repeats and "boxes" tools
    </p>
   </dd>
  </dl>
  <h2 id="get_strep_pneumo_repeats">
   get_strep_pneumo_repeats
  </h2>
  <pre><code>  $return = $obj-&gt;get_strep_pneumo_repeats($genomeTO)</code></pre>
  <dl>
   <dt id="Parameter-and-return-types14">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$genomeTO is a genomeTO
$return is a genomeTO
genomeTO is a reference to a hash where the following keys are defined:
	id has a value which is a genome_id
	scientific_name has a value which is a string
	domain has a value which is a string
	genetic_code has a value which is an int
	source has a value which is a string
	source_id has a value which is a string
	contigs has a value which is a reference to a list where each element is a contig
	features has a value which is a reference to a list where each element is a feature
genome_id is a string
contig is a reference to a hash where the following keys are defined:
	id has a value which is a contig_id
	dna has a value which is a string
contig_id is a string
feature is a reference to a hash where the following keys are defined:
	id has a value which is a feature_id
	location has a value which is a location
	type has a value which is a feature_type
	function has a value which is a string
	protein_translation has a value which is a string
	aliases has a value which is a reference to a list where each element is a string
	annotations has a value which is a reference to a list where each element is an annotation
feature_id is a string
location is a reference to a list where each element is a region_of_dna
region_of_dna is a reference to a list containing 4 items:
	0: a contig_id
	1: (begin) an int
	2: (strand) a string
	3: (length) an int
feature_type is a string
annotation is a reference to a list containing 3 items:
	0: (comment) a string
	1: (annotator) a string
	2: (annotation_time) an int</pre>
   </dd>
   <dt id="Description14">
    Description
   </dt>
   <dd>
   </dd>
  </dl>
  <h2 id="version">
   version
  </h2>
  <pre><code>  $return = $obj-&gt;version()</code></pre>
  <dl>
   <dt id="Parameter-and-return-types15">
    Parameter and return types
   </dt>
   <dd>
    <pre>
$return is a string
</pre>
   </dd>
   <dt id="Description15">
    Description
   </dt>
   <dd>
    <p>
     Return the module version. This is a Semantic Versioning number.
    </p>
   </dd>
  </dl>
  <h1 id="TYPES">
   TYPES
  </h1>
  <h2 id="md5">
   md5
  </h2>
  <dl>
   <dt id="Definition">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="md5s">
   md5s
  </h2>
  <dl>
   <dt id="Definition1">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list where each element is a md5
</pre>
   </dd>
  </dl>
  <h2 id="genome_id">
   genome_id
  </h2>
  <dl>
   <dt id="Definition2">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="feature_id">
   feature_id
  </h2>
  <dl>
   <dt id="Definition3">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="contig_id">
   contig_id
  </h2>
  <dl>
   <dt id="Definition4">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="feature_type">
   feature_type
  </h2>
  <dl>
   <dt id="Definition5">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="region_of_dna">
   region_of_dna
  </h2>
  <dl>
   <dt id="Description16">
    Description
   </dt>
   <dd>
    <p>
     A region of DNA is maintained as a tuple of four components:
    </p>
    <pre><code>                the contig
                the beginning position (from 1)
                the strand
                the length

           We often speak of "a region".  By "location", we mean a sequence
           of regions from the same genome (perhaps from distinct contigs).</code></pre>
   </dd>
   <dt id="Definition6">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 4 items:
0: a contig_id
1: (begin) an int
2: (strand) a string
3: (length) an int</pre>
   </dd>
  </dl>
  <h2 id="location">
   location
  </h2>
  <dl>
   <dt id="Description17">
    Description
   </dt>
   <dd>
    <p>
     a "location" refers to a sequence of regions
    </p>
   </dd>
   <dt id="Definition7">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list where each element is a region_of_dna
</pre>
   </dd>
  </dl>
  <h2 id="annotation">
   annotation
  </h2>
  <dl>
   <dt id="Definition8">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 3 items:
0: (comment) a string
1: (annotator) a string
2: (annotation_time) an int</pre>
   </dd>
  </dl>
  <h2 id="feature">
   feature
  </h2>
  <dl>
   <dt id="Description18">
    Description
   </dt>
   <dd>
    <p>
     represents a feature on the genome location on the contig with a type, and if a protein has translation, any aliases associated current history of annoation in style of SEED
    </p>
   </dd>
   <dt id="Definition9">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a feature_id
location has a value which is a location
type has a value which is a feature_type
function has a value which is a string
protein_translation has a value which is a string
aliases has a value which is a reference to a list where each element is a string
annotations has a value which is a reference to a list where each element is an annotation</pre>
   </dd>
  </dl>
  <h2 id="contig">
   contig
  </h2>
  <dl>
   <dt id="Description19">
    Description
   </dt>
   <dd>
    <p>
     Data for DNA contig
    </p>
   </dd>
   <dt id="Definition10">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a contig_id
dna has a value which is a string</pre>
   </dd>
  </dl>
  <h2 id="genomeTO">
   genomeTO
  </h2>
  <dl>
   <dt id="Description20">
    Description
   </dt>
   <dd>
    <p>
     All of the information about particular genome
    </p>
   </dd>
   <dt id="Definition11">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
id has a value which is a genome_id
scientific_name has a value which is a string
domain has a value which is a string
genetic_code has a value which is an int
source has a value which is a string
source_id has a value which is a string
contigs has a value which is a reference to a list where each element is a contig
features has a value which is a reference to a list where each element is a feature</pre>
   </dd>
  </dl>
  <h2 id="subsystem">
   subsystem
  </h2>
  <dl>
   <dt id="Definition12">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="variant">
   variant
  </h2>
  <dl>
   <dt id="Definition13">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="variant_of_subsystem">
   variant_of_subsystem
  </h2>
  <dl>
   <dt id="Definition14">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 2 items:
0: a subsystem
1: a variant</pre>
   </dd>
  </dl>
  <h2 id="variant_subsystem_pairs">
   variant_subsystem_pairs
  </h2>
  <dl>
   <dt id="Definition15">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list where each element is a variant_of_subsystem
</pre>
   </dd>
  </dl>
  <h2 id="fid">
   fid
  </h2>
  <dl>
   <dt id="Definition16">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="role">
   role
  </h2>
  <dl>
   <dt id="Definition17">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="function">
   function
  </h2>
  <dl>
   <dt id="Definition18">
    Definition
   </dt>
   <dd>
    <pre>
a string
</pre>
   </dd>
  </dl>
  <h2 id="fid_role_pair">
   fid_role_pair
  </h2>
  <dl>
   <dt id="Definition19">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 2 items:
0: a fid
1: a role</pre>
   </dd>
  </dl>
  <h2 id="fid_role_pairs">
   fid_role_pairs
  </h2>
  <dl>
   <dt id="Definition20">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list where each element is a fid_role_pair
</pre>
   </dd>
  </dl>
  <h2 id="fid_function_pair">
   fid_function_pair
  </h2>
  <dl>
   <dt id="Definition21">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 2 items:
0: a fid
1: a function</pre>
   </dd>
  </dl>
  <h2 id="fid_function_pairs">
   fid_function_pairs
  </h2>
  <dl>
   <dt id="Definition22">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list where each element is a fid_function_pair
</pre>
   </dd>
  </dl>
  <h2 id="reconstructionTO">
   reconstructionTO
  </h2>
  <dl>
   <dt id="Description21">
    Description
   </dt>
   <dd>
    <p>
     Metabolic reconstruction represents the set of subsystems that we infer are present in this genome
    </p>
   </dd>
   <dt id="Definition23">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a hash where the following keys are defined:
subsystems has a value which is a variant_subsystem_pairs
bindings has a value which is a fid_role_pairs
assignments has a value which is a fid_function_pairs</pre>
   </dd>
  </dl>
  <h2 id="fid_data_tuple">
   fid_data_tuple
  </h2>
  <dl>
   <dt id="Definition24">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list containing 4 items:
0: a fid
1: a md5
2: a location
3: a function</pre>
   </dd>
  </dl>
  <h2 id="fid_data_tuples">
   fid_data_tuples
  </h2>
  <dl>
   <dt id="Definition25">
    Definition
   </dt>
   <dd>
    <pre>
a reference to a list where each element is a fid_data_tuple
</pre>
   </dd>
  </dl>
 </body>
</html>
