

<h1 id="NAME">NAME</h1>

<p>Bio::KBase::Genotype_PhenotypeAPI::Client</p>

<h1 id="DESCRIPTION">DESCRIPTION</h1>

<h2 id="genomes_with_trait">genomes_with_trait</h2>

<pre><code>  $return = $obj-&gt;genomes_with_trait()</code></pre>

<dl>

<dt id="Parameter-and-return-types">Parameter and return types</dt>
<dd>



<pre>
$return is a kb_genome_list
kb_genome_list is a reference to a list where each element is a kb_genome
kb_genome is a string</pre>

</dd>
<dt id="Description">Description</dt>
<dd>

<p>get genomes with trait/phenotype data</p>

</dd>
</dl>

<h2 id="get_experiments">get_experiments</h2>

<pre><code>  $return = $obj-&gt;get_experiments($kb_genome)</code></pre>

<dl>

<dt id="Parameter-and-return-types1">Parameter and return types</dt>
<dd>



<pre>
$kb_genome is a kb_genome
$return is an experiment_details
kb_genome is a string
experiment_details is a reference to a list where each element is a reference to a list containing 3 items:
	0: a kb_study_experiment
	1: a design_experiment
	2: an originator
kb_study_experiment is a string
design_experiment is a string
originator is a string</pre>

</dd>
<dt id="Description1">Description</dt>
<dd>

<p>get all experiments corresponding to a genome.</p>

</dd>
</dl>

<h2 id="get_traits">get_traits</h2>

<pre><code>  $return = $obj-&gt;get_traits($kb_study_experiment)</code></pre>

<dl>

<dt id="Parameter-and-return-types2">Parameter and return types</dt>
<dd>



<pre>
$kb_study_experiment is a kb_study_experiment
$return is a trait_metadata
kb_study_experiment is a string
trait_metadata is a reference to a list where each element is a reference to a list containing 4 items:
	0: a trait
	1: a protocol
	2: a to_id
	3: a unit_of_measure
trait is a string
protocol is a string
to_id is a string
unit_of_measure is a string</pre>

</dd>
<dt id="Description2">Description</dt>
<dd>

<p>get list of traits in an experiment</p>

</dd>
</dl>

<h2 id="traits_to_variations">traits_to_variations</h2>

<pre><code>  $return = $obj-&gt;traits_to_variations($trait, $pvaluecutoff)</code></pre>

<dl>

<dt id="Parameter-and-return-types3">Parameter and return types</dt>
<dd>



<pre>
$trait is a trait
$pvaluecutoff is a pvaluecutoff
$return is a variation_details_with_trait_metadata
trait is a string
pvaluecutoff is a float
variation_details_with_trait_metadata is a reference to a hash where the key is a trait_metadata and the value is a variation_details
trait_metadata is a reference to a list where each element is a reference to a list containing 4 items:
	0: a trait
	1: a protocol
	2: a to_id
	3: a unit_of_measure
protocol is a string
to_id is a string
unit_of_measure is a string
variation_details is a reference to a list where each element is a reference to a list containing 4 items:
	0: a chromosome
	1: a position
	2: a pvalue
	3: a kb_chromosome
chromosome is a string
position is an int
pvalue is a float
kb_chromosome is a string</pre>

</dd>
<dt id="Description3">Description</dt>
<dd>

<p>Given a trait id and pvalue cutoff, the trait metadata and significant variations above the pvalue cutoff</p>

</dd>
</dl>

<h2 id="chromosome_position_from_variation_details">chromosome_position_from_variation_details</h2>

<pre><code>  $return = $obj-&gt;chromosome_position_from_variation_details($variation_details)</code></pre>

<dl>

<dt id="Parameter-and-return-types4">Parameter and return types</dt>
<dd>



<pre>
$variation_details is a variation_details
$return is a chromosomal_positions
variation_details is a reference to a list where each element is a reference to a list containing 4 items:
	0: a chromosome
	1: a position
	2: a pvalue
	3: a kb_chromosome
chromosome is a string
position is an int
pvalue is a float
kb_chromosome is a string
chromosomal_positions is a reference to a list where each element is a reference to a list containing 2 items:
	0: a chromosome
	1: a position</pre>

</dd>
<dt id="Description4">Description</dt>
<dd>

<p>variation details is a data structure with 4 columns, this function converts it into 2 columns for use in other functions</p>

</dd>
</dl>

<h2 id="traits_to_genes">traits_to_genes</h2>

<pre><code>  $return = $obj-&gt;traits_to_genes($trait, $pvaluecutoff, $distance)</code></pre>

<dl>

<dt id="Parameter-and-return-types5">Parameter and return types</dt>
<dd>



<pre>
$trait is a trait
$pvaluecutoff is a pvaluecutoff
$distance is a distance
$return is a gene_list_with_trait_metadata
trait is a string
pvaluecutoff is a float
distance is an int
gene_list_with_trait_metadata is a reference to a list where each element is a reference to a list containing 2 items:
	0: a trait_metadata
	1: a gene_list
trait_metadata is a reference to a list where each element is a reference to a list containing 4 items:
	0: a trait
	1: a protocol
	2: a to_id
	3: a unit_of_measure
protocol is a string
to_id is a string
unit_of_measure is a string
gene_list is a reference to a list where each element is a reference to a list containing 2 items:
	0: a kbase_gene
	1: a gene
kbase_gene is a string
gene is a string</pre>

</dd>
<dt id="Description5">Description</dt>
<dd>

<p>Given a trait, directly get the trait metadata and gene list.</p>

</dd>
</dl>

<h2 id="variations_to_genes">variations_to_genes</h2>

<pre><code>  $return = $obj-&gt;variations_to_genes($chromosomal_positions, $distance)</code></pre>

<dl>

<dt id="Parameter-and-return-types6">Parameter and return types</dt>
<dd>



<pre>
$chromosomal_positions is a chromosomal_positions
$distance is a distance
$return is a gene_list
chromosomal_positions is a reference to a list where each element is a reference to a list containing 2 items:
	0: a chromosome
	1: a position
chromosome is a string
position is an int
distance is an int
gene_list is a reference to a list where each element is a reference to a list containing 2 items:
	0: a kbase_gene
	1: a gene
kbase_gene is a string
gene is a string</pre>

</dd>
<dt id="Description6">Description</dt>
<dd>

<p>Given a set of variations (genomic locations), identify the genes near the variations</p>

</dd>
</dl>

<h2 id="find_common_snps">find_common_snps</h2>

<pre><code>  $return = $obj-&gt;find_common_snps($trait_list_pvalue)</code></pre>

<dl>

<dt id="Parameter-and-return-types7">Parameter and return types</dt>
<dd>



<pre>
$trait_list_pvalue is a trait_list_pvalue
$return is a chromosomal_positions
trait_list_pvalue is a reference to a list where each element is a reference to a list containing 2 items:
	0: a trait
	1: a pvaluecutoff
trait is a string
pvaluecutoff is a float
chromosomal_positions is a reference to a list where each element is a reference to a list containing 2 items:
	0: a chromosome
	1: a position
chromosome is a string
position is an int</pre>

</dd>
<dt id="Description7">Description</dt>
<dd>

<p>Given a set of traits, identify variants that are common to those traits above a certain pvalue cutoff</p>

</dd>
</dl>

<h2 id="selected_locations_to_genes">selected_locations_to_genes</h2>

<pre><code>  $return = $obj-&gt;selected_locations_to_genes($trait, $pmin, $pmax, $chromosomal_locations, $distance)</code></pre>

<dl>

<dt id="Parameter-and-return-types8">Parameter and return types</dt>
<dd>



<pre>
$trait is a trait
$pmin is a pmin
$pmax is a pmax
$chromosomal_locations is a chromosomal_locations
$distance is a distance
$return is a gene_list
trait is a string
pmin is a float
pmax is a float
chromosomal_locations is a reference to a list where each element is a reference to a list containing 3 items:
	0: a chromosome
	1: a start
	2: an end
chromosome is a string
start is an int
end is an int
distance is an int
gene_list is a reference to a list where each element is a reference to a list containing 2 items:
	0: a kbase_gene
	1: a gene
kbase_gene is a string
gene is a string</pre>

</dd>
<dt id="Description8">Description</dt>
<dd>

<p>Given pvalue range, list of chromosomal positions, and distance, this function find snp&acirc;&#128;&#153;s with pvalue within the given pvalue range. It then identifies genes within given distance and return list of genes. This function is written to take selection from manhattan plot as input and feed results to other widget</p>

</dd>
</dl>

<h1 id="TYPES">TYPES</h1>

<h2 id="kb_genome">kb_genome</h2>

<dl>

<dt id="Description9">Description</dt>
<dd>

<p>kbase id of genome eg. kb|g.3899</p>

</dd>
<dt id="Definition">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="genome_name">genome_name</h2>

<dl>

<dt id="Description10">Description</dt>
<dd>

<p>genome name eg. Arabidopsis thaliana</p>

</dd>
<dt id="Definition1">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="kb_study_experiment">kb_study_experiment</h2>

<dl>

<dt id="Description11">Description</dt>
<dd>

<p>kbase id of the experiment eg. kb|g.3899.exp.1</p>

</dd>
<dt id="Definition2">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="design_experiment">design_experiment</h2>

<dl>

<dt id="Description12">Description</dt>
<dd>

<p>Design information about the experiment eg. SNP Array chip or resequencing experiment eg. ~250K SNPs x 107 phenotpes</p>

</dd>
<dt id="Definition3">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="originator">originator</h2>

<dl>

<dt id="Description13">Description</dt>
<dd>

<p>Name of the individual or program that are the originators of the experiment eg. Atwell S, et. al.</p>

</dd>
<dt id="Definition4">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="trait">trait</h2>

<dl>

<dt id="Description14">Description</dt>
<dd>

<p>trait is the kbase id of the trait. kb|g.3899.trait.191</p>

</dd>
<dt id="Definition5">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="chromosome">chromosome</h2>

<dl>

<dt id="Description15">Description</dt>
<dd>

<p>chromosome is the source id (eg. TAIR id) of the chromosome eg. 1</p>

</dd>
<dt id="Definition6">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="kb_chromosome">kb_chromosome</h2>

<dl>

<dt id="Description16">Description</dt>
<dd>

<p>kb_chromosome is the kbase id of the chromosome eg. kb|g.22476.c.2496</p>

</dd>
<dt id="Definition7">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="position">position</h2>

<dl>

<dt id="Description17">Description</dt>
<dd>

<p>position is the location of variant on the chromosome</p>

</dd>
<dt id="Definition8">Definition</dt>
<dd>



<pre>
an int
</pre>

</dd>
</dl>

<h2 id="pvalue">pvalue</h2>

<dl>

<dt id="Definition9">Definition</dt>
<dd>



<pre>
a float
</pre>

</dd>
</dl>

<h2 id="gene">gene</h2>

<dl>

<dt id="Description18">Description</dt>
<dd>

<p>source id of the gene</p>

</dd>
<dt id="Definition10">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="kbase_gene">kbase_gene</h2>

<dl>

<dt id="Description19">Description</dt>
<dd>

<p>kbase id of the gene</p>

</dd>
<dt id="Definition11">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="pvaluecutoff">pvaluecutoff</h2>

<dl>

<dt id="Definition12">Definition</dt>
<dd>



<pre>
a float
</pre>

</dd>
</dl>

<h2 id="distance">distance</h2>

<dl>

<dt id="Definition13">Definition</dt>
<dd>



<pre>
an int
</pre>

</dd>
</dl>

<h2 id="rankcutoff">rankcutoff</h2>

<dl>

<dt id="Definition14">Definition</dt>
<dd>



<pre>
an int
</pre>

</dd>
</dl>

<h2 id="region">region</h2>

<dl>

<dt id="Definition15">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="pmin">pmin</h2>

<dl>

<dt id="Description20">Description</dt>
<dd>

<p>pmin is minimum selected pvalue</p>

</dd>
<dt id="Definition16">Definition</dt>
<dd>



<pre>
a float
</pre>

</dd>
</dl>

<h2 id="pmax">pmax</h2>

<dl>

<dt id="Description21">Description</dt>
<dd>

<p>pmax is maximum selected pvalue</p>

</dd>
<dt id="Definition17">Definition</dt>
<dd>



<pre>
a float
</pre>

</dd>
</dl>

<h2 id="start">start</h2>

<dl>

<dt id="Description22">Description</dt>
<dd>

<p>start is the start position of chromosomal location</p>

</dd>
<dt id="Definition18">Definition</dt>
<dd>



<pre>
an int
</pre>

</dd>
</dl>

<h2 id="end">end</h2>

<dl>

<dt id="Description23">Description</dt>
<dd>

<p>end is the end position of chromosomal location</p>

</dd>
<dt id="Definition19">Definition</dt>
<dd>



<pre>
an int
</pre>

</dd>
</dl>

<h2 id="protocol">protocol</h2>

<dl>

<dt id="Description24">Description</dt>
<dd>

<p>phenotype/trait measurement protocol</p>

</dd>
<dt id="Definition20">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="to_id">to_id</h2>

<dl>

<dt id="Description25">Description</dt>
<dd>

<p>trait ontology</p>

</dd>
<dt id="Definition21">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="unit_of_measure">unit_of_measure</h2>

<dl>

<dt id="Description26">Description</dt>
<dd>

<p>unit of measurement of a phenotype/trait</p>

</dd>
<dt id="Definition22">Definition</dt>
<dd>



<pre>
a string
</pre>

</dd>
</dl>

<h2 id="variation_details">variation_details</h2>

<dl>

<dt id="Description27">Description</dt>
<dd>

<p>variation_details has the output for manhattan plot</p>

</dd>
<dt id="Definition23">Definition</dt>
<dd>



<pre>
a reference to a list where each element is a reference to a list containing 4 items:
0: a chromosome
1: a position
2: a pvalue
3: a kb_chromosome</pre>

</dd>
</dl>

<h2 id="experiment_details">experiment_details</h2>

<dl>

<dt id="Description28">Description</dt>
<dd>

<p>experiment_details: kb_study_experiment, experiment design, orginator</p>

</dd>
<dt id="Definition24">Definition</dt>
<dd>



<pre>
a reference to a list where each element is a reference to a list containing 3 items:
0: a kb_study_experiment
1: a design_experiment
2: an originator</pre>

</dd>
</dl>

<h2 id="trait_metadata">trait_metadata</h2>

<dl>

<dt id="Description29">Description</dt>
<dd>

<p>experiment_details: kb_study_experiment, experiment design, orginator</p>

</dd>
<dt id="Definition25">Definition</dt>
<dd>



<pre>
a reference to a list where each element is a reference to a list containing 4 items:
0: a trait
1: a protocol
2: a to_id
3: a unit_of_measure</pre>

</dd>
</dl>

<h2 id="variation_details_with_trait_metadata">variation_details_with_trait_metadata</h2>

<dl>

<dt id="Description30">Description</dt>
<dd>

<p>variation_details with trait metadata</p>

</dd>
<dt id="Definition26">Definition</dt>
<dd>



<pre>
a reference to a hash where the key is a trait_metadata and the value is a variation_details
</pre>

</dd>
</dl>

<h2 id="chromosomal_positions">chromosomal_positions</h2>

<dl>

<dt id="Description31">Description</dt>
<dd>

<p>chromosomal positions: list of positions define by tuple chromosome, position; position can be from several different chromosomes</p>

</dd>
<dt id="Definition27">Definition</dt>
<dd>



<pre>
a reference to a list where each element is a reference to a list containing 2 items:
0: a chromosome
1: a position</pre>

</dd>
</dl>

<h2 id="chromosomal_locations">chromosomal_locations</h2>

<dl>

<dt id="Description32">Description</dt>
<dd>

<p>chromosomal locations: chromosome start end</p>

</dd>
<dt id="Definition28">Definition</dt>
<dd>



<pre>
a reference to a list where each element is a reference to a list containing 3 items:
0: a chromosome
1: a start
2: an end</pre>

</dd>
</dl>

<h2 id="trait_list_pvalue">trait_list_pvalue</h2>

<dl>

<dt id="Description33">Description</dt>
<dd>

<p>trait_list_pvalue:List of traits and pvalue</p>

</dd>
<dt id="Definition29">Definition</dt>
<dd>



<pre>
a reference to a list where each element is a reference to a list containing 2 items:
0: a trait
1: a pvaluecutoff</pre>

</dd>
</dl>

<h2 id="gene_list">gene_list</h2>

<dl>

<dt id="Description34">Description</dt>
<dd>

<p>list of genes</p>

</dd>
<dt id="Definition30">Definition</dt>
<dd>



<pre>
a reference to a list where each element is a reference to a list containing 2 items:
0: a kbase_gene
1: a gene</pre>

</dd>
</dl>

<h2 id="gene_list_with_trait_metadata">gene_list_with_trait_metadata</h2>

<dl>

<dt id="Description35">Description</dt>
<dd>

<p>Trait details and list of genes</p>

</dd>
<dt id="Definition31">Definition</dt>
<dd>



<pre>
a reference to a list where each element is a reference to a list containing 2 items:
0: a trait_metadata
1: a gene_list</pre>

</dd>
</dl>

<h2 id="kb_genome_list">kb_genome_list</h2>

<dl>

<dt id="Description36">Description</dt>
<dd>

<p>list of genomes</p>

</dd>
<dt id="Definition32">Definition</dt>
<dd>



<pre>
a reference to a list where each element is a kb_genome
</pre>

</dd>
</dl>


