<html>
<head>

<body bgcolor=white lang=EN-US style='tab-interval:.5in'>

<br />
<h1 align=center>Walk-thru of the KBase Notebook</h1>
<hr />

<br />
<br />
<span style='font-size:16.0pt'>1. Login with your KBase credentials in order to create, view and save your private notebooks.</span>
<br />
<img width=1272 height=1061 src="KBNB_walkthru_files/image001.png">

<br />
<br />
<span style='font-size:16.0pt'>2. Create a new notebook if you like.</span>
<br />
<img width=1272 height=1061 src="KBNB_walkthru_files/image002.png">

<br />
<br />
<span style='font-size:16.0pt'>3. To get started you might want open an existing KBase notebook.</span>
<br />
<img width=1272 height=1061 src="KBNB_walkthru_files/image003.png">

<br />
<br />
<span style='font-size:16.0pt'>4. You can also create a copy of an existing notebook.</span>
<br />
<img width=1272 height=1061 src="KBNB_walkthru_files/image004.png">

<br />
<br />
<span style='font-size:16.0pt'>5. This notebook shows you how to examine sequence quality information in KBase. Reading thru this notebook can give you an idea of how analyses can be performed and displayed in KBase.</span>
<br />
<img width=1272 height=1061 src="KBNB_walkthru_files/image005.png">

<br />
<br />
<span style='font-size:16.0pt'>6. In the top right-hand corner you can click on the "Analysis Builder" button to open the visual analysis builder. This is a convenient way for any user to get started, but especially those without prior programming experience.</span>
<br />
<img width=1272 height=1061 src="KBNB_walkthru_files/image006.png">

<br />
<br />
<span style='font-size:16.0pt'>7. Before we go any further, try creating a new notebook by clicking the "+" symbol next to the tab of the notebook that you already opened.</span>
<br />
<img width=1272 height=1061 src="KBNB_walkthru_files/image007.png">

<br />
<br />
<span style='font-size:16.0pt'>8. Then, give your new notebook a name and click "Launch".</span>
<br />
<img width=1272 height=1061 src="KBNB_walkthru_files/image008.png">

<br />
<br />
<span style='font-size:16.0pt'>9. The "import data" tab in the Analysis Builder is where you import the metagenomes that you're interested in analyzing. Below you can see we are viewing all available metagenomes by their "sequencing type" but have restricted our search set to include only WGS (Whole Genome Shotgun) metagenomes. By selecting the search field from the drop-down menu and then typing in your search criteria and hitting enter, you can enter multiple search criteria for different search fields.</span>
<br />
<img width=1272 height=1061 src="KBNB_walkthru_files/image009.png">

<br />
<br />
<span style='font-size:16.0pt'>10. Notice in the following picture, that while we are still viewing only WGS metagenomes, we have changed the field to "id" and entered the beginning of our metagenome sample ID to narrow our list of metagenomes. Then, we selected the two metagenomes we are interested in (mgm4440275.3 and mgm4440279.3) and clicked on the right arrow button to add them to our selected metagenomes list.</span>
<br />
<img width=1272 height=1061 src="KBNB_walkthru_files/image010.png">

<br />
<br />
<span style='font-size:16.0pt'>11. Next, clicking on the green checkmark loads the data for these metagenomes. You will see that the python code for loading this data is automatically generated in the notebook below. The line numbers to the left of each cell in the notebook keep track of the order in which the cells were run. Sometimes loading the initial data can take a few minutes. During this time, instead of a line number you will see [*] that will indicate the command is still running. Once the code in that line has finished running, the line number will appear in place of the asterisk.</span>
<br />
<img width=1272 height=1061 src="KBNB_walkthru_files/image011.png">

<br />
<br />
<span style='font-size:16.0pt'>12. Now we will click on the "select data" tab and select what we are interested in analyzing. Perhaps we only want to look at a subset of the samples we loaded. We also need to select the type of data we are going to visually analyze. Once we are ready, we again click on the green checkmark to generate the python code necessary for performing this subselection. Note that our subselection will be loaded into the variable listed under "select target -> variable name".</span>
<br />
<img width=1272 height=1061 src="KBNB_walkthru_files/image012.png">

<br />
<br />
<span style='font-size:16.0pt'>13. Finally we want to go to the "visualize data" tab to generate our visual analysis. There are various analysis types to choose from at the top of this tab. Below, we have selected a heatmap with the default settings. Note that you must also select which variable from the "select data" step that you would like to analyze under "available variables". In our case there is only one to choose from, but you can have multiple. When we're ready, we again click the green checkmark.</span>
<br />
<img width=1272 height=1061 src="KBNB_walkthru_files/image013.png">

<br />
<br />
<span style='font-size:16.0pt'>14. The next slide shows the graph that was generated inline which we can view by scrolling down in our notebook.</span>
<br />
<img width=1272 height=1061 src="KBNB_walkthru_files/image014.png">

<br />
<br />
<span style='font-size:16.0pt'>15. Scrolling to the top of the page, you will find a green button (with an eye on it) next to the "Analysis Builder" button, which will open a new browser window containing only the visual analyses generated in the notebook.</span>
<br />
<img width=1272 height=1061 src="KBNB_walkthru_files/image015.png">

<br />
<br />
<span style='font-size:16.0pt'>16. This last slide shows what this new window would look like in our example.</span>
<br />
<img width=1272 height=1061 src="KBNB_walkthru_files/image016.png">

</body>
</html>
