<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_AlignmentTree"
>all_entities_AlignmentTree</a></h2>

<p>Return all instances of the AlignmentTree entity.</p>

<p>An alignment arranges a group of protein sequences so that they match.
Each alignment is associated with a phylogenetic tree that describes how the sequences developed and their evolutionary distance.
The actual tree and alignment FASTA are stored in separate flat files.
The Kbase will maintain a set of alignments and associated trees.
The majority of these will be based on protein sequences.
We will not have a comprehensive set but we will have tens of thousands of such alignments,
and we view them as an imporant resource to support annotation.
The alignments/trees will include the tools and parameters used to construct them.
Access to the underlying sequences and trees in a form convenient to existing tools will be supported.</p>

<p>Example:</p>

<pre>    all_entities_AlignmentTree -a </pre>

<p>would retrieve all entities of type AlignmentTree and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AlignmentTree entity has the following relationship links:</p>

<dl>
<dt><a name="Aligns_ProteinSequence"
>Aligns ProteinSequence</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="alignment_method"
>alignment_method</a></dt>

<dd>
<dt><a name="alignment_parameters"
>alignment_parameters</a></dt>

<dd>
<dt><a name="alignment_properties"
>alignment_properties</a></dt>

<dd>
<dt><a name="tree_method"
>tree_method</a></dt>

<dd>
<dt><a name="tree_parameters"
>tree_parameters</a></dt>

<dd>
<dt><a name="tree_properties"
>tree_properties</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Annotation"
>all_entities_Annotation</a></h2>

<p>Return all instances of the Annotation entity.</p>

<p>An annotation is a comment attached to a feature.
Annotations are used to track the history of a feature&#39;s functional assignments and any related issues.
The key is the feature ID followed by a colon and a complemented ten-digit sequence number.</p>

<p>Example:</p>

<pre>    all_entities_Annotation -a </pre>

<p>would retrieve all entities of type Annotation and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Annotation entity has the following relationship links:</p>

<dl>
<dt><a name="Annotates_Feature"
>Annotates Feature</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="annotator"
>annotator</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>

<dd>
<dt><a name="annotation_time"
>annotation_time</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_AtomicRegulon"
>all_entities_AtomicRegulon</a></h2>

<p>Return all instances of the AtomicRegulon entity.</p>

<p>An atomic regulon is an indivisible group of coregulated features on a single genome.
Atomic regulons are constructed so that a given feature can only belong to one.
Because of this,
the expression levels for atomic regulons represent in some sense the state of a cell.
An atomicRegulon is a set of protein-encoding genes that are believed to have identical expression profiles (i.e.,
they will all be expressed or none will be expressed in the vast majority of conditions).
These are sometimes referred to as &#34;atomic regulons&#34;.
Note that there are more common notions of &#34;coregulated set of genes&#34; based on the notion that a single regulatory mechanism impacts an entire set of genes.
Since multiple other mechanisms may impact overlapping sets,
the genes impacted by a regulatory mechanism need not all share the same expression profile.
We use a distinct notion (CoregulatedSet) to reference sets of genes impacted by a single regulatory mechanism (i.e.,
by a single transcription regulator).</p>

<p>Example:</p>

<pre>    all_entities_AtomicRegulon -a </pre>

<p>would retrieve all entities of type AtomicRegulon and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AtomicRegulon entity has the following relationship links:</p>

<dl>
<dt><a name="IsAffectedIn_Experiment"
>IsAffectedIn Experiment</a></dt>

<dd>
<dt><a name="IsFormedOf_Feature"
>IsFormedOf Feature</a></dt>

<dd>
<dt><a name="ReflectsStateOf_Genome"
>ReflectsStateOf Genome</a></dt>

<dd>
<dt><a name="WasGeneratedFrom_ProbeSet"
>WasGeneratedFrom ProbeSet</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Attribute"
>all_entities_Attribute</a></h2>

<p>Return all instances of the Attribute entity.</p>

<p>An attribute describes a category of condition or characteristic for an experiment.
The goals of the experiment can be inferred from its values for all the attributes of interest.</p>

<p>Example:</p>

<pre>    all_entities_Attribute -a </pre>

<p>would retrieve all entities of type Attribute and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Attribute entity has the following relationship links:</p>

<dl>
<dt><a name="HasValueIn_Experiment"
>HasValueIn Experiment</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="description"
>description</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Biomass"
>all_entities_Biomass</a></h2>

<p>Return all instances of the Biomass entity.</p>

<p>A biomass is a collection of compounds in a specific ratio and in specific compartments that are necessary for a cell to function properly.
The prediction of biomasses is key to the functioning of the model.</p>

<p>Example:</p>

<pre>    all_entities_Biomass -a </pre>

<p>would retrieve all entities of type Biomass and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Biomass entity has the following relationship links:</p>

<dl>
<dt><a name="IsComprisedOf_BiomassCompound"
>IsComprisedOf BiomassCompound</a></dt>

<dd>
<dt><a name="IsManagedBy_Model"
>IsManagedBy Model</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_BiomassCompound"
>all_entities_BiomassCompound</a></h2>

<p>Return all instances of the BiomassCompound entity.</p>

<p>A Biomass Compound represents the occurrence of a particular compound in a biomass.</p>

<p>Example:</p>

<pre>    all_entities_BiomassCompound -a </pre>

<p>would retrieve all entities of type BiomassCompound and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The BiomassCompound entity has the following relationship links:</p>

<dl>
<dt><a name="Comprises_Biomass"
>Comprises Biomass</a></dt>

<dd>
<dt><a name="IsUsageOf_Compound"
>IsUsageOf Compound</a></dt>

<dd>
<dt><a name="Targets_ModelCompartment"
>Targets ModelCompartment</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="coefficient"
>coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Compartment"
>all_entities_Compartment</a></h2>

<p>Return all instances of the Compartment entity.</p>

<p>A compartment is a section of a single model that represents the environment in which a reaction takes place (e.g.
cell wall).</p>

<p>Example:</p>

<pre>    all_entities_Compartment -a </pre>

<p>would retrieve all entities of type Compartment and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Compartment entity has the following relationship links:</p>

<dl>
<dt><a name="IsDefaultFor_Reaction"
>IsDefaultFor Reaction</a></dt>

<dd>
<dt><a name="IsDefaultLocationOf_Reagent"
>IsDefaultLocationOf Reagent</a></dt>

<dd>
<dt><a name="IsInstantiatedBy_ModelCompartment"
>IsInstantiatedBy ModelCompartment</a></dt>

<dd>
<dt><a name="IsProposedLocationOf_ReactionRule"
>IsProposedLocationOf ReactionRule</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Complex"
>all_entities_Complex</a></h2>

<p>Return all instances of the Complex entity.</p>

<p>A complex is a set of chemical reactions that act in concert to effect a role.</p>

<p>Example:</p>

<pre>    all_entities_Complex -a </pre>

<p>would retrieve all entities of type Complex and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Complex entity has the following relationship links:</p>

<dl>
<dt><a name="HasStep_ReactionRule"
>HasStep ReactionRule</a></dt>

<dd>
<dt><a name="IsTriggeredBy_Role"
>IsTriggeredBy Role</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Compound"
>all_entities_Compound</a></h2>

<p>Return all instances of the Compound entity.</p>

<p>A compound is a chemical that participates in a reaction.
Both ligands and reaction components are treated as compounds.</p>

<p>Example:</p>

<pre>    all_entities_Compound -a </pre>

<p>would retrieve all entities of type Compound and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Compound entity has the following relationship links:</p>

<dl>
<dt><a name="HasUsage_BiomassCompound"
>HasUsage BiomassCompound</a></dt>

<dd>
<dt><a name="IsPresentIn_Media"
>IsPresentIn Media</a></dt>

<dd>
<dt><a name="IsShownOn_Diagram"
>IsShownOn Diagram</a></dt>

<dd>
<dt><a name="IsTerminusFor_Scenario"
>IsTerminusFor Scenario</a></dt>

<dd>
<dt><a name="ParticipatesAs_Reagent"
>ParticipatesAs Reagent</a></dt>

<dd>
<dt><a name="UsesAliasForCompound_Source"
>UsesAliasForCompound Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="uncharged_formula"
>uncharged_formula</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Contig"
>all_entities_Contig</a></h2>

<p>Return all instances of the Contig entity.</p>

<p>A contig is thought of as composing a part of the DNA associated with a specific genome.
It is represented as an ID (including the genome ID) and a ContigSequence.
We do not think of strings of DNA from,
say,
a metgenomic sample as &#34;contigs&#34;,
since there is no associated genome (these would be considered ContigSequences).
This use of the term &#34;ContigSequence&#34;,
rather than just &#34;DNA sequence&#34;,
may turn out to be a bad idea.
For now,
you should just realize that a Contig has an associated genome,
but a ContigSequence does not.</p>

<p>Example:</p>

<pre>    all_entities_Contig -a </pre>

<p>would retrieve all entities of type Contig and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Contig entity has the following relationship links:</p>

<dl>
<dt><a name="HasAsSequence_ContigSequence"
>HasAsSequence ContigSequence</a></dt>

<dd>
<dt><a name="IsAlignedIn_Variation"
>IsAlignedIn Variation</a></dt>

<dd>
<dt><a name="IsComponentOf_Genome"
>IsComponentOf Genome</a></dt>

<dd>
<dt><a name="IsLocusFor_Feature"
>IsLocusFor Feature</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_ContigChunk"
>all_entities_ContigChunk</a></h2>

<p>Return all instances of the ContigChunk entity.</p>

<p>ContigChunks are strings of DNA thought of as being a string in a 4-character alphabet with an associated ID.
We allow a broader alphabet that includes U (for RNA) and the standard ambiguity characters.
The notion of ContigChunk was introduced to avoid transferring/manipulating huge contigs to access small substrings.
A ContigSequence is formed by concatenating a set of one or more ContigChunks.
Thus,
ContigChunks are the basic units moved from the database to memory.
Their existence should be hidden from users in most circumstances (users are expected to request substrings of ContigSequences,
and the Kbase software locates the appropriate ContigChunks).</p>

<p>Example:</p>

<pre>    all_entities_ContigChunk -a </pre>

<p>would retrieve all entities of type ContigChunk and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ContigChunk entity has the following relationship links:</p>

<dl>
<dt><a name="IsSectionOf_ContigSequence"
>IsSectionOf ContigSequence</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_ContigSequence"
>all_entities_ContigSequence</a></h2>

<p>Return all instances of the ContigSequence entity.</p>

<p>ContigSequences are strings of DNA.
Contigs have an associated genome,
but ContigSequences do not..
We can think of random samples of DNA as a set of ContigSequences.
There are no length constraints imposed on ContigSequences -- they can be either very short or very long.
The basic unit of data that is moved to/from the database is the ContigChunk,
from which ContigSequences are formed.
The key of a ContigSequence is the sequence&#39;s MD5 identifier.</p>

<p>Example:</p>

<pre>    all_entities_ContigSequence -a </pre>

<p>would retrieve all entities of type ContigSequence and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ContigSequence entity has the following relationship links:</p>

<dl>
<dt><a name="HasSection_ContigChunk"
>HasSection ContigChunk</a></dt>

<dd>
<dt><a name="IsSequenceOf_Contig"
>IsSequenceOf Contig</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="length"
>length</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_CoregulatedSet"
>all_entities_CoregulatedSet</a></h2>

<p>Return all instances of the CoregulatedSet entity.</p>

<p>We need to represent sets of genes that are coregulated via some regulatory mechanism.
In particular,
we wish to represent genes that are coregulated using transcription binding sites and corresponding transcription regulatory proteins.
We represent a coregulated set (which may,
or may not,
be considered an atomic regulon) using CoregulatedSet.</p>

<p>Example:</p>

<pre>    all_entities_CoregulatedSet -a </pre>

<p>would retrieve all entities of type CoregulatedSet and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The CoregulatedSet entity has the following relationship links:</p>

<dl>
<dt><a name="IsBoundBy_Feature"
>IsBoundBy Feature</a></dt>

<dd>
<dt><a name="IsControlledUsing_Feature"
>IsControlledUsing Feature</a></dt>

<dd>
<dt><a name="IsRegulatedSetOf_Feature"
>IsRegulatedSetOf Feature</a></dt>

<dd>
<dt><a name="WasFormulatedBy_Source"
>WasFormulatedBy Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Diagram"
>all_entities_Diagram</a></h2>

<p>Return all instances of the Diagram entity.</p>

<p>A functional diagram describes a network of chemical reactions,
often comprising a single subsystem.</p>

<p>Example:</p>

<pre>    all_entities_Diagram -a </pre>

<p>would retrieve all entities of type Diagram and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Diagram entity has the following relationship links:</p>

<dl>
<dt><a name="Displays_Reaction"
>Displays Reaction</a></dt>

<dd>
<dt><a name="IncludesPartOf_Scenario"
>IncludesPartOf Scenario</a></dt>

<dd>
<dt><a name="IsRelevantFor_Subsystem"
>IsRelevantFor Subsystem</a></dt>

<dd>
<dt><a name="Shows_Compound"
>Shows Compound</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_EcNumber"
>all_entities_EcNumber</a></h2>

<p>Return all instances of the EcNumber entity.</p>

<p>EC numbers are assigned by the Enzyme Commission,
and consist of four numbers separated by periods,
each indicating a successively smaller cateogry of enzymes.</p>

<p>Example:</p>

<pre>    all_entities_EcNumber -a </pre>

<p>would retrieve all entities of type EcNumber and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The EcNumber entity has the following relationship links:</p>

<dl>
<dt><a name="IsConsistentWith_Role"
>IsConsistentWith Role</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="obsolete"
>obsolete</a></dt>

<dd>
<dt><a name="replacedby"
>replacedby</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Experiment"
>all_entities_Experiment</a></h2>

<p>Return all instances of the Experiment entity.</p>

<p>An experiment is a combination of conditions for which gene expression information is desired.
The result of the experiment is a set of expression levels for features under the given conditions.</p>

<p>Example:</p>

<pre>    all_entities_Experiment -a </pre>

<p>would retrieve all entities of type Experiment and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Experiment entity has the following relationship links:</p>

<dl>
<dt><a name="AffectsLevelOf_AtomicRegulon"
>AffectsLevelOf AtomicRegulon</a></dt>

<dd>
<dt><a name="HasResultsFor_ProbeSet"
>HasResultsFor ProbeSet</a></dt>

<dd>
<dt><a name="HasValueFor_Attribute"
>HasValueFor Attribute</a></dt>

<dd>
<dt><a name="IndicatesSignalFor_Feature"
>IndicatesSignalFor Feature</a></dt>

<dd>
<dt><a name="OperatesIn_Media"
>OperatesIn Media</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source"
>source</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Family"
>all_entities_Family</a></h2>

<p>Return all instances of the Family entity.</p>

<p>The Kbase will support the maintenance of protein families (as sets of Features with associated translations).
We are initially only supporting the notion of a family as composed of a set of isofunctional homologs.
That is,
the families we initially support should be thought of as containing protein-encoding genes whose associated sequences all implement the same function (we do understand that the notion of &#34;function&#34; is somewhat ambiguous,
so let us sweep this under the rug by calling a functional role a &#34;primitive concept&#34;).
We currently support families in which the members are translations of features,
and we think of Features as having an associated function.
Identical protein sequences as products of translating distinct genes may or may not have identical functions,
and we allow multiple members of the same Family to share identical protein sequences.
This may be justified,
since in a very,
very,
very few cases identical proteins do,
in fact,
have distinct functions.
We would prefer to reach the point where our Families are sets of protein sequence,
rather than sets of protein-encoding Features.</p>

<p>Example:</p>

<pre>    all_entities_Family -a </pre>

<p>would retrieve all entities of type Family and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Family entity has the following relationship links:</p>

<dl>
<dt><a name="HasMember_Feature"
>HasMember Feature</a></dt>

<dd>
<dt><a name="IsCoupledTo_Family"
>IsCoupledTo Family</a></dt>

<dd>
<dt><a name="IsCoupledWith_Family"
>IsCoupledWith Family</a></dt>

<dd>
<dt><a name="IsFamilyFor_Role"
>IsFamilyFor Role</a></dt>

<dd>
<dt><a name="IsRepresentedIn_Genome"
>IsRepresentedIn Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Feature"
>all_entities_Feature</a></h2>

<p>Return all instances of the Feature entity.</p>

<p>A feature (sometimes also called a gene) is a part of a genome that is of special interest.
Features may be spread across multiple DNA sequences (contigs) of a genome,
but never across more than one genome.
Each feature in the database has a unique ID that functions as its ID in this table.
Normally a Feature is just a single contigous region on a contig.
Features have types,
and an appropriate choice of available types allows the support of protein-encoding genes,
exons,
RNA genes,
binding sites,
pathogenicity islands,
or whatever.</p>

<p>Example:</p>

<pre>    all_entities_Feature -a </pre>

<p>would retrieve all entities of type Feature and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Feature entity has the following relationship links:</p>

<dl>
<dt><a name="Controls_CoregulatedSet"
>Controls CoregulatedSet</a></dt>

<dd>
<dt><a name="Encompasses_Feature"
>Encompasses Feature</a></dt>

<dd>
<dt><a name="HasCoregulationWith_Feature"
>HasCoregulationWith Feature</a></dt>

<dd>
<dt><a name="HasFunctional_Role"
>HasFunctional Role</a></dt>

<dd>
<dt><a name="HasIndicatedSignalFrom_Experiment"
>HasIndicatedSignalFrom Experiment</a></dt>

<dd>
<dt><a name="HasLevelsFrom_ProbeSet"
>HasLevelsFrom ProbeSet</a></dt>

<dd>
<dt><a name="IsAnnotatedBy_Annotation"
>IsAnnotatedBy Annotation</a></dt>

<dd>
<dt><a name="IsBindingSiteFor_CoregulatedSet"
>IsBindingSiteFor CoregulatedSet</a></dt>

<dd>
<dt><a name="IsContainedIn_SSCell"
>IsContainedIn SSCell</a></dt>

<dd>
<dt><a name="IsCoregulatedWith_Feature"
>IsCoregulatedWith Feature</a></dt>

<dd>
<dt><a name="IsEncompassedIn_Feature"
>IsEncompassedIn Feature</a></dt>

<dd>
<dt><a name="IsExemplarOf_Role"
>IsExemplarOf Role</a></dt>

<dd>
<dt><a name="IsFormedInto_AtomicRegulon"
>IsFormedInto AtomicRegulon</a></dt>

<dd>
<dt><a name="IsInPair_Pairing"
>IsInPair Pairing</a></dt>

<dd>
<dt><a name="IsLocatedIn_Contig"
>IsLocatedIn Contig</a></dt>

<dd>
<dt><a name="IsMemberOf_Family"
>IsMemberOf Family</a></dt>

<dd>
<dt><a name="IsOwnedBy_Genome"
>IsOwnedBy Genome</a></dt>

<dd>
<dt><a name="IsRegulatedIn_CoregulatedSet"
>IsRegulatedIn CoregulatedSet</a></dt>

<dd>
<dt><a name="Produces_ProteinSequence"
>Produces ProteinSequence</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Genome"
>all_entities_Genome</a></h2>

<p>Return all instances of the Genome entity.</p>

<p>The Kbase houses a large and growing set of genomes.
We often have multiple genomes that have identical DNA.
These usually have distinct gene calls and annotations,
but not always.
We consider the Kbase to be a framework for managing hundreds of thousands of genomes and offering the tools needed to support compartive analysis on large sets of genomes,
some of which are virtually identical.
Each genome has an MD5 value computed from the DNA that is associated with the genome.
Hence,
it is easy to recognize when you have identical genomes,
perhaps annotated by distinct groups.</p>

<p>Example:</p>

<pre>    all_entities_Genome -a </pre>

<p>would retrieve all entities of type Genome and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Genome entity has the following relationship links:</p>

<dl>
<dt><a name="HadResultsProducedBy_ProbeSet"
>HadResultsProducedBy ProbeSet</a></dt>

<dd>
<dt><a name="HasRepresentativeOf_Family"
>HasRepresentativeOf Family</a></dt>

<dd>
<dt><a name="IsCollectedInto_OTU"
>IsCollectedInto OTU</a></dt>

<dd>
<dt><a name="IsComposedOf_Contig"
>IsComposedOf Contig</a></dt>

<dd>
<dt><a name="IsConfiguredBy_AtomicRegulon"
>IsConfiguredBy AtomicRegulon</a></dt>

<dd>
<dt><a name="IsInTaxa_TaxonomicGrouping"
>IsInTaxa TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsModeledBy_Model"
>IsModeledBy Model</a></dt>

<dd>
<dt><a name="IsOwnerOf_Feature"
>IsOwnerOf Feature</a></dt>

<dd>
<dt><a name="Uses_SSRow"
>Uses SSRow</a></dt>

<dd>
<dt><a name="WasSubmittedBy_Source"
>WasSubmittedBy Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Identifier"
>all_entities_Identifier</a></h2>

<p>Return all instances of the Identifier entity.</p>

<p>An identifier is an alternate name for a protein sequence.
The identifier is typically stored in a prefixed form that indicates the database it came from.</p>

<p>Example:</p>

<pre>    all_entities_Identifier -a </pre>

<p>would retrieve all entities of type Identifier and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Identifier entity has the following relationship links:</p>

<dl>
<dt><a name="HasAssertionFrom_Source"
>HasAssertionFrom Source</a></dt>

<dd>
<dt><a name="Names_ProteinSequence"
>Names ProteinSequence</a></dt>

<dd>
<dt><a name="WasImportedFrom_Source"
>WasImportedFrom Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source"
>source</a></dt>

<dd>
<dt><a name="natural_form"
>natural_form</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Media"
>all_entities_Media</a></h2>

<p>Return all instances of the Media entity.</p>

<p>A media describes the chemical content of the solution in which cells are grown in an experiment or for the purposes of a model.
The key is the common media name.
The nature of the media is described by its relationship to its constituent compounds.</p>

<p>Example:</p>

<pre>    all_entities_Media -a </pre>

<p>would retrieve all entities of type Media and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Media entity has the following relationship links:</p>

<dl>
<dt><a name="HasPresenceOf_Compound"
>HasPresenceOf Compound</a></dt>

<dd>
<dt><a name="IsUtilizedIn_Experiment"
>IsUtilizedIn Experiment</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Model"
>all_entities_Model</a></h2>

<p>Return all instances of the Model entity.</p>

<p>A model specifies a relationship between sets of features and reactions in a cell.
It is used to simulate cell growth and gene knockouts to validate annotations.</p>

<p>Example:</p>

<pre>    all_entities_Model -a </pre>

<p>would retrieve all entities of type Model and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Model entity has the following relationship links:</p>

<dl>
<dt><a name="IsARequirementIn_Requirement"
>IsARequirementIn Requirement</a></dt>

<dd>
<dt><a name="IsDividedInto_ModelCompartment"
>IsDividedInto ModelCompartment</a></dt>

<dd>
<dt><a name="Manages_Biomass"
>Manages Biomass</a></dt>

<dd>
<dt><a name="Models_Genome"
>Models Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_ModelCompartment"
>all_entities_ModelCompartment</a></h2>

<p>Return all instances of the ModelCompartment entity.</p>

<p>The Model Compartment represents a section of a cell (e.g.
cell wall,
cytoplasm) as it appears in a specific model.</p>

<p>Example:</p>

<pre>    all_entities_ModelCompartment -a </pre>

<p>would retrieve all entities of type ModelCompartment and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ModelCompartment entity has the following relationship links:</p>

<dl>
<dt><a name="IsDivisionOf_Model"
>IsDivisionOf Model</a></dt>

<dd>
<dt><a name="IsInstanceOf_Compartment"
>IsInstanceOf Compartment</a></dt>

<dd>
<dt><a name="IsRealLocationOf_Requirement"
>IsRealLocationOf Requirement</a></dt>

<dd>
<dt><a name="IsTargetOf_BiomassCompound"
>IsTargetOf BiomassCompound</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_OTU"
>all_entities_OTU</a></h2>

<p>Return all instances of the OTU entity.</p>

<p>An OTU (Organism Taxonomic Unit) is a named group of related genomes.</p>

<p>Example:</p>

<pre>    all_entities_OTU -a </pre>

<p>would retrieve all entities of type OTU and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The OTU entity has the following relationship links:</p>

<dl>
<dt><a name="IsCollectionOf_Genome"
>IsCollectionOf Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Pairing"
>all_entities_Pairing</a></h2>

<p>Return all instances of the Pairing entity.</p>

<p>A pairing indicates that two features are found close together in a genome.
Not all possible pairings are stored in the database; only those that are considered for some reason to be significant for annotation purposes.The key of the pairing is the concatenation of the feature IDs in alphabetical order with an intervening colon.</p>

<p>Example:</p>

<pre>    all_entities_Pairing -a </pre>

<p>would retrieve all entities of type Pairing and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Pairing entity has the following relationship links:</p>

<dl>
<dt><a name="Determines_PairSet"
>Determines PairSet</a></dt>

<dd>
<dt><a name="IsPairOf_Feature"
>IsPairOf Feature</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_PairSet"
>all_entities_PairSet</a></h2>

<p>Return all instances of the PairSet entity.</p>

<p>A PairSet is a precompute set of pairs or genes.
Each pair occurs close to one another of the chromosome.
We believe that all of the first members of the pairs correspond to one another (are quite similar),
as do all of the second members of the pairs.
These pairs (from prokaryotic genomes) offer on of the most powerful clues relating to uncharacterized genes/peroteins.</p>

<p>Example:</p>

<pre>    all_entities_PairSet -a </pre>

<p>would retrieve all entities of type PairSet and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The PairSet entity has the following relationship links:</p>

<dl>
<dt><a name="IsDeterminedBy_Pairing"
>IsDeterminedBy Pairing</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="score"
>score</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_ProbeSet"
>all_entities_ProbeSet</a></h2>

<p>Return all instances of the ProbeSet entity.</p>

<p>A probe set is a device containing multiple probe sequences for use in gene expression experiments.</p>

<p>Example:</p>

<pre>    all_entities_ProbeSet -a </pre>

<p>would retrieve all entities of type ProbeSet and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ProbeSet entity has the following relationship links:</p>

<dl>
<dt><a name="GeneratedLevelsFor_AtomicRegulon"
>GeneratedLevelsFor AtomicRegulon</a></dt>

<dd>
<dt><a name="HasResultsIn_Experiment"
>HasResultsIn Experiment</a></dt>

<dd>
<dt><a name="IndicatedLevelsFor_Feature"
>IndicatedLevelsFor Feature</a></dt>

<dd>
<dt><a name="ProducedResultsFor_Genome"
>ProducedResultsFor Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_ProteinSequence"
>all_entities_ProteinSequence</a></h2>

<p>Return all instances of the ProteinSequence entity.</p>

<p>We use the concept of ProteinSequence as an amino acid string with an associated MD5 value.
It is easy to access the set of Features that relate to a ProteinSequence.
While function is still associated with Features (and may be for some time),
publications are associated with ProteinSequences (and the inferred impact on Features is through the relationship connecting ProteinSequences to Features).</p>

<p>Example:</p>

<pre>    all_entities_ProteinSequence -a </pre>

<p>would retrieve all entities of type ProteinSequence and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ProteinSequence entity has the following relationship links:</p>

<dl>
<dt><a name="IsATopicOf_Publication"
>IsATopicOf Publication</a></dt>

<dd>
<dt><a name="IsAlignedBy_AlignmentTree"
>IsAlignedBy AlignmentTree</a></dt>

<dd>
<dt><a name="IsNamedBy_Identifier"
>IsNamedBy Identifier</a></dt>

<dd>
<dt><a name="IsProjectedOnto_ProteinSequence"
>IsProjectedOnto ProteinSequence</a></dt>

<dd>
<dt><a name="IsProteinFor_Feature"
>IsProteinFor Feature</a></dt>

<dd>
<dt><a name="ProjectsOnto_ProteinSequence"
>ProjectsOnto ProteinSequence</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Publication"
>all_entities_Publication</a></h2>

<p>Return all instances of the Publication entity.</p>

<p>Annotators attach publications to ProteinSequences.
The criteria we have used to gather such connections is a bit nonstandard.
We have sought to attach publications to ProteinSequences when the publication includes an expert asserting a belief or estimate of function.
The paper may not be the original characterization.
Further,
it may not even discuss a sequence protein (much of the literature is very valuable,
but reports work on proteins in strains that have not yet been sequenced).
On the other hand,
reports of sequencing regions of a chromosome (with no specific assertion of a clear function) should not be attached.
The attached publications give an ID (usually a Pubmed ID),
a URL to the paper (when we have it),
and a title (when we have it).</p>

<p>Example:</p>

<pre>    all_entities_Publication -a </pre>

<p>would retrieve all entities of type Publication and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Publication entity has the following relationship links:</p>

<dl>
<dt><a name="Concerns_ProteinSequence"
>Concerns ProteinSequence</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="link"
>link</a></dt>

<dd>
<dt><a name="pubdate"
>pubdate</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Reaction"
>all_entities_Reaction</a></h2>

<p>Return all instances of the Reaction entity.</p>

<p>A reaction is a chemical process that converts one set of compounds (substrate) to another set (products).</p>

<p>Example:</p>

<pre>    all_entities_Reaction -a </pre>

<p>would retrieve all entities of type Reaction and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Reaction entity has the following relationship links:</p>

<dl>
<dt><a name="Involves_Reagent"
>Involves Reagent</a></dt>

<dd>
<dt><a name="IsDisplayedOn_Diagram"
>IsDisplayedOn Diagram</a></dt>

<dd>
<dt><a name="IsRequiredBy_Requirement"
>IsRequiredBy Requirement</a></dt>

<dd>
<dt><a name="IsUsedAs_ReactionRule"
>IsUsedAs ReactionRule</a></dt>

<dd>
<dt><a name="ParticipatesIn_Scenario"
>ParticipatesIn Scenario</a></dt>

<dd>
<dt><a name="RunsByDefaultIn_Compartment"
>RunsByDefaultIn Compartment</a></dt>

<dd>
<dt><a name="UsesAliasForReaction_Source"
>UsesAliasForReaction Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="equation"
>equation</a></dt>

<dd>
<dt><a name="reversibility"
>reversibility</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_ReactionRule"
>all_entities_ReactionRule</a></h2>

<p>Return all instances of the ReactionRule entity.</p>

<p>A reaction rule represents the way a reaction takes place within the context of a model.</p>

<p>Example:</p>

<pre>    all_entities_ReactionRule -a </pre>

<p>would retrieve all entities of type ReactionRule and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ReactionRule entity has the following relationship links:</p>

<dl>
<dt><a name="HasProposedLocationIn_Compartment"
>HasProposedLocationIn Compartment</a></dt>

<dd>
<dt><a name="IsStepOf_Complex"
>IsStepOf Complex</a></dt>

<dd>
<dt><a name="IsUseOf_Reaction"
>IsUseOf Reaction</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Reagent"
>all_entities_Reagent</a></h2>

<p>Return all instances of the Reagent entity.</p>

<p>This entity represents a compound as it is used by a specific reaction.
A reaction involves many compounds,
and a compound can be involved in many reactions.
The reagent describes the use of the compound by a specific reaction.</p>

<p>Example:</p>

<pre>    all_entities_Reagent -a </pre>

<p>would retrieve all entities of type Reagent and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Reagent entity has the following relationship links:</p>

<dl>
<dt><a name="HasDefaultLocation_Compartment"
>HasDefaultLocation Compartment</a></dt>

<dd>
<dt><a name="IsInvolvedIn_Reaction"
>IsInvolvedIn Reaction</a></dt>

<dd>
<dt><a name="IsParticipationOf_Compound"
>IsParticipationOf Compound</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="stoichiometry"
>stoichiometry</a></dt>

<dd>
<dt><a name="cofactor"
>cofactor</a></dt>

<dd>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="transport_coefficient"
>transport_coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Requirement"
>all_entities_Requirement</a></h2>

<p>Return all instances of the Requirement entity.</p>

<p>A requirement describes the way a reaction fits into a model.</p>

<p>Example:</p>

<pre>    all_entities_Requirement -a </pre>

<p>would retrieve all entities of type Requirement and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Requirement entity has the following relationship links:</p>

<dl>
<dt><a name="HasRealLocationIn_ModelCompartment"
>HasRealLocationIn ModelCompartment</a></dt>

<dd>
<dt><a name="IsARequirementOf_Model"
>IsARequirementOf Model</a></dt>

<dd>
<dt><a name="Requires_Reaction"
>Requires Reaction</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>

<dd>
<dt><a name="proton"
>proton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Role"
>all_entities_Role</a></h2>

<p>Return all instances of the Role entity.</p>

<p>A role describes a biological function that may be fulfilled by a feature.
One of the main goals of the database is to assign features to roles.
Most roles are effected by the construction of proteins.
Some,
however,
deal with functional regulation and message transmission.</p>

<p>Example:</p>

<pre>    all_entities_Role -a </pre>

<p>would retrieve all entities of type Role and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Role entity has the following relationship links:</p>

<dl>
<dt><a name="DeterminesFunctionOf_Family"
>DeterminesFunctionOf Family</a></dt>

<dd>
<dt><a name="HasAsExemplar_Feature"
>HasAsExemplar Feature</a></dt>

<dd>
<dt><a name="IsConsistentTo_EcNumber"
>IsConsistentTo EcNumber</a></dt>

<dd>
<dt><a name="IsFunctionalIn_Feature"
>IsFunctionalIn Feature</a></dt>

<dd>
<dt><a name="IsIncludedIn_Subsystem"
>IsIncludedIn Subsystem</a></dt>

<dd>
<dt><a name="IsRoleOf_SSCell"
>IsRoleOf SSCell</a></dt>

<dd>
<dt><a name="Triggers_Complex"
>Triggers Complex</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Scenario"
>all_entities_Scenario</a></h2>

<p>Return all instances of the Scenario entity.</p>

<p>A scenario is a partial instance of a subsystem with a defined set of reactions.
Each scenario converts input compounds to output compounds using reactions.
The scenario may use all of the reactions controlled by a subsystem or only some,
and may also incorporate additional reactions.
Because scenario names are not unique,
the actual scenario ID is a number.</p>

<p>Example:</p>

<pre>    all_entities_Scenario -a </pre>

<p>would retrieve all entities of type Scenario and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Scenario entity has the following relationship links:</p>

<dl>
<dt><a name="HasAsTerminus_Compound"
>HasAsTerminus Compound</a></dt>

<dd>
<dt><a name="HasParticipant_Reaction"
>HasParticipant Reaction</a></dt>

<dd>
<dt><a name="Overlaps_Diagram"
>Overlaps Diagram</a></dt>

<dd>
<dt><a name="Validates_Subsystem"
>Validates Subsystem</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="common_name"
>common_name</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Source"
>all_entities_Source</a></h2>

<p>Return all instances of the Source entity.</p>

<p>A source is a user or organization that is permitted to assign its own identifiers or to submit bioinformatic objects to the database.</p>

<p>Example:</p>

<pre>    all_entities_Source -a </pre>

<p>would retrieve all entities of type Source and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Source entity has the following relationship links:</p>

<dl>
<dt><a name="Asserts_Identifier"
>Asserts Identifier</a></dt>

<dd>
<dt><a name="Formulated_CoregulatedSet"
>Formulated CoregulatedSet</a></dt>

<dd>
<dt><a name="HasCompoundAliasFrom_Compound"
>HasCompoundAliasFrom Compound</a></dt>

<dd>
<dt><a name="HasReactionAliasFrom_Reaction"
>HasReactionAliasFrom Reaction</a></dt>

<dd>
<dt><a name="Imported_Identifier"
>Imported Identifier</a></dt>

<dd>
<dt><a name="Provided_Subsystem"
>Provided Subsystem</a></dt>

<dd>
<dt><a name="Submitted_Genome"
>Submitted Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_SSCell"
>all_entities_SSCell</a></h2>

<p>Return all instances of the SSCell entity.</p>

<p>An SSCell (SpreadSheet Cell) represents a role as it occurs in a subsystem spreadsheet row.
The key is a colon-delimited triple containing an MD5 hash of the subsystem ID followed by a genome ID (with optional region string) and a role abbreviation.</p>

<p>Example:</p>

<pre>    all_entities_SSCell -a </pre>

<p>would retrieve all entities of type SSCell and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SSCell entity has the following relationship links:</p>

<dl>
<dt><a name="Contains_Feature"
>Contains Feature</a></dt>

<dd>
<dt><a name="HasRole_Role"
>HasRole Role</a></dt>

<dd>
<dt><a name="IsRoleFor_SSRow"
>IsRoleFor SSRow</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_SSRow"
>all_entities_SSRow</a></h2>

<p>Return all instances of the SSRow entity.</p>

<p>An SSRow (that is,
a row in a subsystem spreadsheet) represents a collection of functional roles present in the Features of a single Genome.
The roles are part of a designated subsystem,
and the fids associated with each role are included in the row,
That is,
a row amounts to an instance of a subsystem as it exists in a specific,
designated genome.</p>

<p>Example:</p>

<pre>    all_entities_SSRow -a </pre>

<p>would retrieve all entities of type SSRow and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SSRow entity has the following relationship links:</p>

<dl>
<dt><a name="Implements_Variant"
>Implements Variant</a></dt>

<dd>
<dt><a name="IsRowOf_SSCell"
>IsRowOf SSCell</a></dt>

<dd>
<dt><a name="IsUsedBy_Genome"
>IsUsedBy Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="curated"
>curated</a></dt>

<dd>
<dt><a name="region"
>region</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Subsystem"
>all_entities_Subsystem</a></h2>

<p>Return all instances of the Subsystem entity.</p>

<p>A subsystem is a set of functional roles that have been annotated simultaneously (e.g.,
the roles present in a specific pathway),
with an associated subsystem spreadsheet which encodes the fids in each genome that implement the functional roles in the subsystem.</p>

<p>Example:</p>

<pre>    all_entities_Subsystem -a </pre>

<p>would retrieve all entities of type Subsystem and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Subsystem entity has the following relationship links:</p>

<dl>
<dt><a name="Describes_Variant"
>Describes Variant</a></dt>

<dd>
<dt><a name="Includes_Role"
>Includes Role</a></dt>

<dd>
<dt><a name="IsInClass_SubsystemClass"
>IsInClass SubsystemClass</a></dt>

<dd>
<dt><a name="IsRelevantTo_Diagram"
>IsRelevantTo Diagram</a></dt>

<dd>
<dt><a name="IsSubInstanceOf_Scenario"
>IsSubInstanceOf Scenario</a></dt>

<dd>
<dt><a name="WasProvidedBy_Source"
>WasProvidedBy Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_SubsystemClass"
>all_entities_SubsystemClass</a></h2>

<p>Return all instances of the SubsystemClass entity.</p>

<p>Subsystem classes impose a hierarchical organization on the subsystems.</p>

<p>Example:</p>

<pre>    all_entities_SubsystemClass -a </pre>

<p>would retrieve all entities of type SubsystemClass and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SubsystemClass entity has the following relationship links:</p>

<dl>
<dt><a name="IsClassFor_Subsystem"
>IsClassFor Subsystem</a></dt>

<dd>
<dt><a name="IsSubclassOf_SubsystemClass"
>IsSubclassOf SubsystemClass</a></dt>

<dd>
<dt><a name="IsSuperclassOf_SubsystemClass"
>IsSuperclassOf SubsystemClass</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_TaxonomicGrouping"
>all_entities_TaxonomicGrouping</a></h2>

<p>Return all instances of the TaxonomicGrouping entity.</p>

<p>We associate with most genomes a &#34;taxonomy&#34; based on the NCBI taxonomy.
This includes,
for each genome,
a list of ever larger taxonomic groups.</p>

<p>Example:</p>

<pre>    all_entities_TaxonomicGrouping -a </pre>

<p>would retrieve all entities of type TaxonomicGrouping and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The TaxonomicGrouping entity has the following relationship links:</p>

<dl>
<dt><a name="IsGroupFor_TaxonomicGrouping"
>IsGroupFor TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsInGroup_TaxonomicGrouping"
>IsInGroup TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsTaxonomyOf_Genome"
>IsTaxonomyOf Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Variant"
>all_entities_Variant</a></h2>

<p>Return all instances of the Variant entity.</p>

<p>Each subsystem may include the designation of distinct variants.
Thus,
there may be three closely-related,
but distinguishable forms of histidine degradation.
Each form would be called a &#34;variant&#34;,
with an associated code,
and all genomes implementing a specific variant can easily be accessed.</p>

<p>Example:</p>

<pre>    all_entities_Variant -a </pre>

<p>would retrieve all entities of type Variant and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Variant entity has the following relationship links:</p>

<dl>
<dt><a name="IsDescribedBy_Subsystem"
>IsDescribedBy Subsystem</a></dt>

<dd>
<dt><a name="IsImplementedBy_SSRow"
>IsImplementedBy SSRow</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="role_rule"
>role_rule</a></dt>

<dd>
<dt><a name="code"
>code</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_entities_Variation"
>all_entities_Variation</a></h2>

<p>Return all instances of the Variation entity.</p>

<p>A variation describes a set of aligned regions in two or more contigs.</p>

<p>Example:</p>

<pre>    all_entities_Variation -a </pre>

<p>would retrieve all entities of type Variation and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Variation entity has the following relationship links:</p>

<dl>
<dt><a name="IsAlignmentFor_Contig"
>IsAlignmentFor Contig</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="notes"
>notes</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_AlignmentTree"
>get_entity_AlignmentTree</a></h2>

<p>An alignment arranges a group of protein sequences so that they match.
Each alignment is associated with a phylogenetic tree that describes how the sequences developed and their evolutionary distance.
The actual tree and alignment FASTA are stored in separate flat files.
The Kbase will maintain a set of alignments and associated trees.
The majority of these will be based on protein sequences.
We will not have a comprehensive set but we will have tens of thousands of such alignments,
and we view them as an imporant resource to support annotation.
The alignments/trees will include the tools and parameters used to construct them.
Access to the underlying sequences and trees in a form convenient to existing tools will be supported.</p>

<p>Example:</p>

<pre>    get_entity_AlignmentTree -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AlignmentTree entity has the following relationship links:</p>

<dl>
<dt><a name="Aligns_ProteinSequence"
>Aligns ProteinSequence</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="alignment_method"
>alignment_method</a></dt>

<dd>
<dt><a name="alignment_parameters"
>alignment_parameters</a></dt>

<dd>
<dt><a name="alignment_properties"
>alignment_properties</a></dt>

<dd>
<dt><a name="tree_method"
>tree_method</a></dt>

<dd>
<dt><a name="tree_parameters"
>tree_parameters</a></dt>

<dd>
<dt><a name="tree_properties"
>tree_properties</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Annotation"
>get_entity_Annotation</a></h2>

<p>An annotation is a comment attached to a feature.
Annotations are used to track the history of a feature&#39;s functional assignments and any related issues.
The key is the feature ID followed by a colon and a complemented ten-digit sequence number.</p>

<p>Example:</p>

<pre>    get_entity_Annotation -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Annotation entity has the following relationship links:</p>

<dl>
<dt><a name="Annotates_Feature"
>Annotates Feature</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="annotator"
>annotator</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>

<dd>
<dt><a name="annotation_time"
>annotation_time</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_AtomicRegulon"
>get_entity_AtomicRegulon</a></h2>

<p>An atomic regulon is an indivisible group of coregulated features on a single genome.
Atomic regulons are constructed so that a given feature can only belong to one.
Because of this,
the expression levels for atomic regulons represent in some sense the state of a cell.
An atomicRegulon is a set of protein-encoding genes that are believed to have identical expression profiles (i.e.,
they will all be expressed or none will be expressed in the vast majority of conditions).
These are sometimes referred to as &#34;atomic regulons&#34;.
Note that there are more common notions of &#34;coregulated set of genes&#34; based on the notion that a single regulatory mechanism impacts an entire set of genes.
Since multiple other mechanisms may impact overlapping sets,
the genes impacted by a regulatory mechanism need not all share the same expression profile.
We use a distinct notion (CoregulatedSet) to reference sets of genes impacted by a single regulatory mechanism (i.e.,
by a single transcription regulator).</p>

<p>Example:</p>

<pre>    get_entity_AtomicRegulon -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AtomicRegulon entity has the following relationship links:</p>

<dl>
<dt><a name="IsAffectedIn_Experiment"
>IsAffectedIn Experiment</a></dt>

<dd>
<dt><a name="IsFormedOf_Feature"
>IsFormedOf Feature</a></dt>

<dd>
<dt><a name="ReflectsStateOf_Genome"
>ReflectsStateOf Genome</a></dt>

<dd>
<dt><a name="WasGeneratedFrom_ProbeSet"
>WasGeneratedFrom ProbeSet</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Attribute"
>get_entity_Attribute</a></h2>

<p>An attribute describes a category of condition or characteristic for an experiment.
The goals of the experiment can be inferred from its values for all the attributes of interest.</p>

<p>Example:</p>

<pre>    get_entity_Attribute -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Attribute entity has the following relationship links:</p>

<dl>
<dt><a name="HasValueIn_Experiment"
>HasValueIn Experiment</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="description"
>description</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Biomass"
>get_entity_Biomass</a></h2>

<p>A biomass is a collection of compounds in a specific ratio and in specific compartments that are necessary for a cell to function properly.
The prediction of biomasses is key to the functioning of the model.</p>

<p>Example:</p>

<pre>    get_entity_Biomass -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Biomass entity has the following relationship links:</p>

<dl>
<dt><a name="IsComprisedOf_BiomassCompound"
>IsComprisedOf BiomassCompound</a></dt>

<dd>
<dt><a name="IsManagedBy_Model"
>IsManagedBy Model</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_BiomassCompound"
>get_entity_BiomassCompound</a></h2>

<p>A Biomass Compound represents the occurrence of a particular compound in a biomass.</p>

<p>Example:</p>

<pre>    get_entity_BiomassCompound -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The BiomassCompound entity has the following relationship links:</p>

<dl>
<dt><a name="Comprises_Biomass"
>Comprises Biomass</a></dt>

<dd>
<dt><a name="IsUsageOf_Compound"
>IsUsageOf Compound</a></dt>

<dd>
<dt><a name="Targets_ModelCompartment"
>Targets ModelCompartment</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="coefficient"
>coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Compartment"
>get_entity_Compartment</a></h2>

<p>A compartment is a section of a single model that represents the environment in which a reaction takes place (e.g.
cell wall).</p>

<p>Example:</p>

<pre>    get_entity_Compartment -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Compartment entity has the following relationship links:</p>

<dl>
<dt><a name="IsDefaultFor_Reaction"
>IsDefaultFor Reaction</a></dt>

<dd>
<dt><a name="IsDefaultLocationOf_Reagent"
>IsDefaultLocationOf Reagent</a></dt>

<dd>
<dt><a name="IsInstantiatedBy_ModelCompartment"
>IsInstantiatedBy ModelCompartment</a></dt>

<dd>
<dt><a name="IsProposedLocationOf_ReactionRule"
>IsProposedLocationOf ReactionRule</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Complex"
>get_entity_Complex</a></h2>

<p>A complex is a set of chemical reactions that act in concert to effect a role.</p>

<p>Example:</p>

<pre>    get_entity_Complex -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Complex entity has the following relationship links:</p>

<dl>
<dt><a name="HasStep_ReactionRule"
>HasStep ReactionRule</a></dt>

<dd>
<dt><a name="IsTriggeredBy_Role"
>IsTriggeredBy Role</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Compound"
>get_entity_Compound</a></h2>

<p>A compound is a chemical that participates in a reaction.
Both ligands and reaction components are treated as compounds.</p>

<p>Example:</p>

<pre>    get_entity_Compound -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Compound entity has the following relationship links:</p>

<dl>
<dt><a name="HasUsage_BiomassCompound"
>HasUsage BiomassCompound</a></dt>

<dd>
<dt><a name="IsPresentIn_Media"
>IsPresentIn Media</a></dt>

<dd>
<dt><a name="IsShownOn_Diagram"
>IsShownOn Diagram</a></dt>

<dd>
<dt><a name="IsTerminusFor_Scenario"
>IsTerminusFor Scenario</a></dt>

<dd>
<dt><a name="ParticipatesAs_Reagent"
>ParticipatesAs Reagent</a></dt>

<dd>
<dt><a name="UsesAliasForCompound_Source"
>UsesAliasForCompound Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="uncharged_formula"
>uncharged_formula</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Contig"
>get_entity_Contig</a></h2>

<p>A contig is thought of as composing a part of the DNA associated with a specific genome.
It is represented as an ID (including the genome ID) and a ContigSequence.
We do not think of strings of DNA from,
say,
a metgenomic sample as &#34;contigs&#34;,
since there is no associated genome (these would be considered ContigSequences).
This use of the term &#34;ContigSequence&#34;,
rather than just &#34;DNA sequence&#34;,
may turn out to be a bad idea.
For now,
you should just realize that a Contig has an associated genome,
but a ContigSequence does not.</p>

<p>Example:</p>

<pre>    get_entity_Contig -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Contig entity has the following relationship links:</p>

<dl>
<dt><a name="HasAsSequence_ContigSequence"
>HasAsSequence ContigSequence</a></dt>

<dd>
<dt><a name="IsAlignedIn_Variation"
>IsAlignedIn Variation</a></dt>

<dd>
<dt><a name="IsComponentOf_Genome"
>IsComponentOf Genome</a></dt>

<dd>
<dt><a name="IsLocusFor_Feature"
>IsLocusFor Feature</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_ContigChunk"
>get_entity_ContigChunk</a></h2>

<p>ContigChunks are strings of DNA thought of as being a string in a 4-character alphabet with an associated ID.
We allow a broader alphabet that includes U (for RNA) and the standard ambiguity characters.
The notion of ContigChunk was introduced to avoid transferring/manipulating huge contigs to access small substrings.
A ContigSequence is formed by concatenating a set of one or more ContigChunks.
Thus,
ContigChunks are the basic units moved from the database to memory.
Their existence should be hidden from users in most circumstances (users are expected to request substrings of ContigSequences,
and the Kbase software locates the appropriate ContigChunks).</p>

<p>Example:</p>

<pre>    get_entity_ContigChunk -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ContigChunk entity has the following relationship links:</p>

<dl>
<dt><a name="IsSectionOf_ContigSequence"
>IsSectionOf ContigSequence</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_ContigSequence"
>get_entity_ContigSequence</a></h2>

<p>ContigSequences are strings of DNA.
Contigs have an associated genome,
but ContigSequences do not..
We can think of random samples of DNA as a set of ContigSequences.
There are no length constraints imposed on ContigSequences -- they can be either very short or very long.
The basic unit of data that is moved to/from the database is the ContigChunk,
from which ContigSequences are formed.
The key of a ContigSequence is the sequence&#39;s MD5 identifier.</p>

<p>Example:</p>

<pre>    get_entity_ContigSequence -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ContigSequence entity has the following relationship links:</p>

<dl>
<dt><a name="HasSection_ContigChunk"
>HasSection ContigChunk</a></dt>

<dd>
<dt><a name="IsSequenceOf_Contig"
>IsSequenceOf Contig</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="length"
>length</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_CoregulatedSet"
>get_entity_CoregulatedSet</a></h2>

<p>We need to represent sets of genes that are coregulated via some regulatory mechanism.
In particular,
we wish to represent genes that are coregulated using transcription binding sites and corresponding transcription regulatory proteins.
We represent a coregulated set (which may,
or may not,
be considered an atomic regulon) using CoregulatedSet.</p>

<p>Example:</p>

<pre>    get_entity_CoregulatedSet -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The CoregulatedSet entity has the following relationship links:</p>

<dl>
<dt><a name="IsBoundBy_Feature"
>IsBoundBy Feature</a></dt>

<dd>
<dt><a name="IsControlledUsing_Feature"
>IsControlledUsing Feature</a></dt>

<dd>
<dt><a name="IsRegulatedSetOf_Feature"
>IsRegulatedSetOf Feature</a></dt>

<dd>
<dt><a name="WasFormulatedBy_Source"
>WasFormulatedBy Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Diagram"
>get_entity_Diagram</a></h2>

<p>A functional diagram describes a network of chemical reactions,
often comprising a single subsystem.</p>

<p>Example:</p>

<pre>    get_entity_Diagram -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Diagram entity has the following relationship links:</p>

<dl>
<dt><a name="Displays_Reaction"
>Displays Reaction</a></dt>

<dd>
<dt><a name="IncludesPartOf_Scenario"
>IncludesPartOf Scenario</a></dt>

<dd>
<dt><a name="IsRelevantFor_Subsystem"
>IsRelevantFor Subsystem</a></dt>

<dd>
<dt><a name="Shows_Compound"
>Shows Compound</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_EcNumber"
>get_entity_EcNumber</a></h2>

<p>EC numbers are assigned by the Enzyme Commission,
and consist of four numbers separated by periods,
each indicating a successively smaller cateogry of enzymes.</p>

<p>Example:</p>

<pre>    get_entity_EcNumber -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The EcNumber entity has the following relationship links:</p>

<dl>
<dt><a name="IsConsistentWith_Role"
>IsConsistentWith Role</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="obsolete"
>obsolete</a></dt>

<dd>
<dt><a name="replacedby"
>replacedby</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Experiment"
>get_entity_Experiment</a></h2>

<p>An experiment is a combination of conditions for which gene expression information is desired.
The result of the experiment is a set of expression levels for features under the given conditions.</p>

<p>Example:</p>

<pre>    get_entity_Experiment -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Experiment entity has the following relationship links:</p>

<dl>
<dt><a name="AffectsLevelOf_AtomicRegulon"
>AffectsLevelOf AtomicRegulon</a></dt>

<dd>
<dt><a name="HasResultsFor_ProbeSet"
>HasResultsFor ProbeSet</a></dt>

<dd>
<dt><a name="HasValueFor_Attribute"
>HasValueFor Attribute</a></dt>

<dd>
<dt><a name="IndicatesSignalFor_Feature"
>IndicatesSignalFor Feature</a></dt>

<dd>
<dt><a name="OperatesIn_Media"
>OperatesIn Media</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source"
>source</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Family"
>get_entity_Family</a></h2>

<p>The Kbase will support the maintenance of protein families (as sets of Features with associated translations).
We are initially only supporting the notion of a family as composed of a set of isofunctional homologs.
That is,
the families we initially support should be thought of as containing protein-encoding genes whose associated sequences all implement the same function (we do understand that the notion of &#34;function&#34; is somewhat ambiguous,
so let us sweep this under the rug by calling a functional role a &#34;primitive concept&#34;).
We currently support families in which the members are translations of features,
and we think of Features as having an associated function.
Identical protein sequences as products of translating distinct genes may or may not have identical functions,
and we allow multiple members of the same Family to share identical protein sequences.
This may be justified,
since in a very,
very,
very few cases identical proteins do,
in fact,
have distinct functions.
We would prefer to reach the point where our Families are sets of protein sequence,
rather than sets of protein-encoding Features.</p>

<p>Example:</p>

<pre>    get_entity_Family -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Family entity has the following relationship links:</p>

<dl>
<dt><a name="HasMember_Feature"
>HasMember Feature</a></dt>

<dd>
<dt><a name="IsCoupledTo_Family"
>IsCoupledTo Family</a></dt>

<dd>
<dt><a name="IsCoupledWith_Family"
>IsCoupledWith Family</a></dt>

<dd>
<dt><a name="IsFamilyFor_Role"
>IsFamilyFor Role</a></dt>

<dd>
<dt><a name="IsRepresentedIn_Genome"
>IsRepresentedIn Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Feature"
>get_entity_Feature</a></h2>

<p>A feature (sometimes also called a gene) is a part of a genome that is of special interest.
Features may be spread across multiple DNA sequences (contigs) of a genome,
but never across more than one genome.
Each feature in the database has a unique ID that functions as its ID in this table.
Normally a Feature is just a single contigous region on a contig.
Features have types,
and an appropriate choice of available types allows the support of protein-encoding genes,
exons,
RNA genes,
binding sites,
pathogenicity islands,
or whatever.</p>

<p>Example:</p>

<pre>    get_entity_Feature -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Feature entity has the following relationship links:</p>

<dl>
<dt><a name="Controls_CoregulatedSet"
>Controls CoregulatedSet</a></dt>

<dd>
<dt><a name="Encompasses_Feature"
>Encompasses Feature</a></dt>

<dd>
<dt><a name="HasCoregulationWith_Feature"
>HasCoregulationWith Feature</a></dt>

<dd>
<dt><a name="HasFunctional_Role"
>HasFunctional Role</a></dt>

<dd>
<dt><a name="HasIndicatedSignalFrom_Experiment"
>HasIndicatedSignalFrom Experiment</a></dt>

<dd>
<dt><a name="HasLevelsFrom_ProbeSet"
>HasLevelsFrom ProbeSet</a></dt>

<dd>
<dt><a name="IsAnnotatedBy_Annotation"
>IsAnnotatedBy Annotation</a></dt>

<dd>
<dt><a name="IsBindingSiteFor_CoregulatedSet"
>IsBindingSiteFor CoregulatedSet</a></dt>

<dd>
<dt><a name="IsContainedIn_SSCell"
>IsContainedIn SSCell</a></dt>

<dd>
<dt><a name="IsCoregulatedWith_Feature"
>IsCoregulatedWith Feature</a></dt>

<dd>
<dt><a name="IsEncompassedIn_Feature"
>IsEncompassedIn Feature</a></dt>

<dd>
<dt><a name="IsExemplarOf_Role"
>IsExemplarOf Role</a></dt>

<dd>
<dt><a name="IsFormedInto_AtomicRegulon"
>IsFormedInto AtomicRegulon</a></dt>

<dd>
<dt><a name="IsInPair_Pairing"
>IsInPair Pairing</a></dt>

<dd>
<dt><a name="IsLocatedIn_Contig"
>IsLocatedIn Contig</a></dt>

<dd>
<dt><a name="IsMemberOf_Family"
>IsMemberOf Family</a></dt>

<dd>
<dt><a name="IsOwnedBy_Genome"
>IsOwnedBy Genome</a></dt>

<dd>
<dt><a name="IsRegulatedIn_CoregulatedSet"
>IsRegulatedIn CoregulatedSet</a></dt>

<dd>
<dt><a name="Produces_ProteinSequence"
>Produces ProteinSequence</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Genome"
>get_entity_Genome</a></h2>

<p>The Kbase houses a large and growing set of genomes.
We often have multiple genomes that have identical DNA.
These usually have distinct gene calls and annotations,
but not always.
We consider the Kbase to be a framework for managing hundreds of thousands of genomes and offering the tools needed to support compartive analysis on large sets of genomes,
some of which are virtually identical.
Each genome has an MD5 value computed from the DNA that is associated with the genome.
Hence,
it is easy to recognize when you have identical genomes,
perhaps annotated by distinct groups.</p>

<p>Example:</p>

<pre>    get_entity_Genome -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Genome entity has the following relationship links:</p>

<dl>
<dt><a name="HadResultsProducedBy_ProbeSet"
>HadResultsProducedBy ProbeSet</a></dt>

<dd>
<dt><a name="HasRepresentativeOf_Family"
>HasRepresentativeOf Family</a></dt>

<dd>
<dt><a name="IsCollectedInto_OTU"
>IsCollectedInto OTU</a></dt>

<dd>
<dt><a name="IsComposedOf_Contig"
>IsComposedOf Contig</a></dt>

<dd>
<dt><a name="IsConfiguredBy_AtomicRegulon"
>IsConfiguredBy AtomicRegulon</a></dt>

<dd>
<dt><a name="IsInTaxa_TaxonomicGrouping"
>IsInTaxa TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsModeledBy_Model"
>IsModeledBy Model</a></dt>

<dd>
<dt><a name="IsOwnerOf_Feature"
>IsOwnerOf Feature</a></dt>

<dd>
<dt><a name="Uses_SSRow"
>Uses SSRow</a></dt>

<dd>
<dt><a name="WasSubmittedBy_Source"
>WasSubmittedBy Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Identifier"
>get_entity_Identifier</a></h2>

<p>An identifier is an alternate name for a protein sequence.
The identifier is typically stored in a prefixed form that indicates the database it came from.</p>

<p>Example:</p>

<pre>    get_entity_Identifier -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Identifier entity has the following relationship links:</p>

<dl>
<dt><a name="HasAssertionFrom_Source"
>HasAssertionFrom Source</a></dt>

<dd>
<dt><a name="Names_ProteinSequence"
>Names ProteinSequence</a></dt>

<dd>
<dt><a name="WasImportedFrom_Source"
>WasImportedFrom Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source"
>source</a></dt>

<dd>
<dt><a name="natural_form"
>natural_form</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Media"
>get_entity_Media</a></h2>

<p>A media describes the chemical content of the solution in which cells are grown in an experiment or for the purposes of a model.
The key is the common media name.
The nature of the media is described by its relationship to its constituent compounds.</p>

<p>Example:</p>

<pre>    get_entity_Media -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Media entity has the following relationship links:</p>

<dl>
<dt><a name="HasPresenceOf_Compound"
>HasPresenceOf Compound</a></dt>

<dd>
<dt><a name="IsUtilizedIn_Experiment"
>IsUtilizedIn Experiment</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Model"
>get_entity_Model</a></h2>

<p>A model specifies a relationship between sets of features and reactions in a cell.
It is used to simulate cell growth and gene knockouts to validate annotations.</p>

<p>Example:</p>

<pre>    get_entity_Model -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Model entity has the following relationship links:</p>

<dl>
<dt><a name="IsARequirementIn_Requirement"
>IsARequirementIn Requirement</a></dt>

<dd>
<dt><a name="IsDividedInto_ModelCompartment"
>IsDividedInto ModelCompartment</a></dt>

<dd>
<dt><a name="Manages_Biomass"
>Manages Biomass</a></dt>

<dd>
<dt><a name="Models_Genome"
>Models Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_ModelCompartment"
>get_entity_ModelCompartment</a></h2>

<p>The Model Compartment represents a section of a cell (e.g.
cell wall,
cytoplasm) as it appears in a specific model.</p>

<p>Example:</p>

<pre>    get_entity_ModelCompartment -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ModelCompartment entity has the following relationship links:</p>

<dl>
<dt><a name="IsDivisionOf_Model"
>IsDivisionOf Model</a></dt>

<dd>
<dt><a name="IsInstanceOf_Compartment"
>IsInstanceOf Compartment</a></dt>

<dd>
<dt><a name="IsRealLocationOf_Requirement"
>IsRealLocationOf Requirement</a></dt>

<dd>
<dt><a name="IsTargetOf_BiomassCompound"
>IsTargetOf BiomassCompound</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_OTU"
>get_entity_OTU</a></h2>

<p>An OTU (Organism Taxonomic Unit) is a named group of related genomes.</p>

<p>Example:</p>

<pre>    get_entity_OTU -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The OTU entity has the following relationship links:</p>

<dl>
<dt><a name="IsCollectionOf_Genome"
>IsCollectionOf Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Pairing"
>get_entity_Pairing</a></h2>

<p>A pairing indicates that two features are found close together in a genome.
Not all possible pairings are stored in the database; only those that are considered for some reason to be significant for annotation purposes.The key of the pairing is the concatenation of the feature IDs in alphabetical order with an intervening colon.</p>

<p>Example:</p>

<pre>    get_entity_Pairing -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Pairing entity has the following relationship links:</p>

<dl>
<dt><a name="Determines_PairSet"
>Determines PairSet</a></dt>

<dd>
<dt><a name="IsPairOf_Feature"
>IsPairOf Feature</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_PairSet"
>get_entity_PairSet</a></h2>

<p>A PairSet is a precompute set of pairs or genes.
Each pair occurs close to one another of the chromosome.
We believe that all of the first members of the pairs correspond to one another (are quite similar),
as do all of the second members of the pairs.
These pairs (from prokaryotic genomes) offer on of the most powerful clues relating to uncharacterized genes/peroteins.</p>

<p>Example:</p>

<pre>    get_entity_PairSet -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The PairSet entity has the following relationship links:</p>

<dl>
<dt><a name="IsDeterminedBy_Pairing"
>IsDeterminedBy Pairing</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="score"
>score</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_ProbeSet"
>get_entity_ProbeSet</a></h2>

<p>A probe set is a device containing multiple probe sequences for use in gene expression experiments.</p>

<p>Example:</p>

<pre>    get_entity_ProbeSet -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ProbeSet entity has the following relationship links:</p>

<dl>
<dt><a name="GeneratedLevelsFor_AtomicRegulon"
>GeneratedLevelsFor AtomicRegulon</a></dt>

<dd>
<dt><a name="HasResultsIn_Experiment"
>HasResultsIn Experiment</a></dt>

<dd>
<dt><a name="IndicatedLevelsFor_Feature"
>IndicatedLevelsFor Feature</a></dt>

<dd>
<dt><a name="ProducedResultsFor_Genome"
>ProducedResultsFor Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_ProteinSequence"
>get_entity_ProteinSequence</a></h2>

<p>We use the concept of ProteinSequence as an amino acid string with an associated MD5 value.
It is easy to access the set of Features that relate to a ProteinSequence.
While function is still associated with Features (and may be for some time),
publications are associated with ProteinSequences (and the inferred impact on Features is through the relationship connecting ProteinSequences to Features).</p>

<p>Example:</p>

<pre>    get_entity_ProteinSequence -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ProteinSequence entity has the following relationship links:</p>

<dl>
<dt><a name="IsATopicOf_Publication"
>IsATopicOf Publication</a></dt>

<dd>
<dt><a name="IsAlignedBy_AlignmentTree"
>IsAlignedBy AlignmentTree</a></dt>

<dd>
<dt><a name="IsNamedBy_Identifier"
>IsNamedBy Identifier</a></dt>

<dd>
<dt><a name="IsProjectedOnto_ProteinSequence"
>IsProjectedOnto ProteinSequence</a></dt>

<dd>
<dt><a name="IsProteinFor_Feature"
>IsProteinFor Feature</a></dt>

<dd>
<dt><a name="ProjectsOnto_ProteinSequence"
>ProjectsOnto ProteinSequence</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Publication"
>get_entity_Publication</a></h2>

<p>Annotators attach publications to ProteinSequences.
The criteria we have used to gather such connections is a bit nonstandard.
We have sought to attach publications to ProteinSequences when the publication includes an expert asserting a belief or estimate of function.
The paper may not be the original characterization.
Further,
it may not even discuss a sequence protein (much of the literature is very valuable,
but reports work on proteins in strains that have not yet been sequenced).
On the other hand,
reports of sequencing regions of a chromosome (with no specific assertion of a clear function) should not be attached.
The attached publications give an ID (usually a Pubmed ID),
a URL to the paper (when we have it),
and a title (when we have it).</p>

<p>Example:</p>

<pre>    get_entity_Publication -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Publication entity has the following relationship links:</p>

<dl>
<dt><a name="Concerns_ProteinSequence"
>Concerns ProteinSequence</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="link"
>link</a></dt>

<dd>
<dt><a name="pubdate"
>pubdate</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Reaction"
>get_entity_Reaction</a></h2>

<p>A reaction is a chemical process that converts one set of compounds (substrate) to another set (products).</p>

<p>Example:</p>

<pre>    get_entity_Reaction -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Reaction entity has the following relationship links:</p>

<dl>
<dt><a name="Involves_Reagent"
>Involves Reagent</a></dt>

<dd>
<dt><a name="IsDisplayedOn_Diagram"
>IsDisplayedOn Diagram</a></dt>

<dd>
<dt><a name="IsRequiredBy_Requirement"
>IsRequiredBy Requirement</a></dt>

<dd>
<dt><a name="IsUsedAs_ReactionRule"
>IsUsedAs ReactionRule</a></dt>

<dd>
<dt><a name="ParticipatesIn_Scenario"
>ParticipatesIn Scenario</a></dt>

<dd>
<dt><a name="RunsByDefaultIn_Compartment"
>RunsByDefaultIn Compartment</a></dt>

<dd>
<dt><a name="UsesAliasForReaction_Source"
>UsesAliasForReaction Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="equation"
>equation</a></dt>

<dd>
<dt><a name="reversibility"
>reversibility</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_ReactionRule"
>get_entity_ReactionRule</a></h2>

<p>A reaction rule represents the way a reaction takes place within the context of a model.</p>

<p>Example:</p>

<pre>    get_entity_ReactionRule -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ReactionRule entity has the following relationship links:</p>

<dl>
<dt><a name="HasProposedLocationIn_Compartment"
>HasProposedLocationIn Compartment</a></dt>

<dd>
<dt><a name="IsStepOf_Complex"
>IsStepOf Complex</a></dt>

<dd>
<dt><a name="IsUseOf_Reaction"
>IsUseOf Reaction</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Reagent"
>get_entity_Reagent</a></h2>

<p>This entity represents a compound as it is used by a specific reaction.
A reaction involves many compounds,
and a compound can be involved in many reactions.
The reagent describes the use of the compound by a specific reaction.</p>

<p>Example:</p>

<pre>    get_entity_Reagent -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Reagent entity has the following relationship links:</p>

<dl>
<dt><a name="HasDefaultLocation_Compartment"
>HasDefaultLocation Compartment</a></dt>

<dd>
<dt><a name="IsInvolvedIn_Reaction"
>IsInvolvedIn Reaction</a></dt>

<dd>
<dt><a name="IsParticipationOf_Compound"
>IsParticipationOf Compound</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="stoichiometry"
>stoichiometry</a></dt>

<dd>
<dt><a name="cofactor"
>cofactor</a></dt>

<dd>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="transport_coefficient"
>transport_coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Requirement"
>get_entity_Requirement</a></h2>

<p>A requirement describes the way a reaction fits into a model.</p>

<p>Example:</p>

<pre>    get_entity_Requirement -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Requirement entity has the following relationship links:</p>

<dl>
<dt><a name="HasRealLocationIn_ModelCompartment"
>HasRealLocationIn ModelCompartment</a></dt>

<dd>
<dt><a name="IsARequirementOf_Model"
>IsARequirementOf Model</a></dt>

<dd>
<dt><a name="Requires_Reaction"
>Requires Reaction</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>

<dd>
<dt><a name="proton"
>proton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Role"
>get_entity_Role</a></h2>

<p>A role describes a biological function that may be fulfilled by a feature.
One of the main goals of the database is to assign features to roles.
Most roles are effected by the construction of proteins.
Some,
however,
deal with functional regulation and message transmission.</p>

<p>Example:</p>

<pre>    get_entity_Role -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Role entity has the following relationship links:</p>

<dl>
<dt><a name="DeterminesFunctionOf_Family"
>DeterminesFunctionOf Family</a></dt>

<dd>
<dt><a name="HasAsExemplar_Feature"
>HasAsExemplar Feature</a></dt>

<dd>
<dt><a name="IsConsistentTo_EcNumber"
>IsConsistentTo EcNumber</a></dt>

<dd>
<dt><a name="IsFunctionalIn_Feature"
>IsFunctionalIn Feature</a></dt>

<dd>
<dt><a name="IsIncludedIn_Subsystem"
>IsIncludedIn Subsystem</a></dt>

<dd>
<dt><a name="IsRoleOf_SSCell"
>IsRoleOf SSCell</a></dt>

<dd>
<dt><a name="Triggers_Complex"
>Triggers Complex</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Scenario"
>get_entity_Scenario</a></h2>

<p>A scenario is a partial instance of a subsystem with a defined set of reactions.
Each scenario converts input compounds to output compounds using reactions.
The scenario may use all of the reactions controlled by a subsystem or only some,
and may also incorporate additional reactions.
Because scenario names are not unique,
the actual scenario ID is a number.</p>

<p>Example:</p>

<pre>    get_entity_Scenario -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Scenario entity has the following relationship links:</p>

<dl>
<dt><a name="HasAsTerminus_Compound"
>HasAsTerminus Compound</a></dt>

<dd>
<dt><a name="HasParticipant_Reaction"
>HasParticipant Reaction</a></dt>

<dd>
<dt><a name="Overlaps_Diagram"
>Overlaps Diagram</a></dt>

<dd>
<dt><a name="Validates_Subsystem"
>Validates Subsystem</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="common_name"
>common_name</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Source"
>get_entity_Source</a></h2>

<p>A source is a user or organization that is permitted to assign its own identifiers or to submit bioinformatic objects to the database.</p>

<p>Example:</p>

<pre>    get_entity_Source -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Source entity has the following relationship links:</p>

<dl>
<dt><a name="Asserts_Identifier"
>Asserts Identifier</a></dt>

<dd>
<dt><a name="Formulated_CoregulatedSet"
>Formulated CoregulatedSet</a></dt>

<dd>
<dt><a name="HasCompoundAliasFrom_Compound"
>HasCompoundAliasFrom Compound</a></dt>

<dd>
<dt><a name="HasReactionAliasFrom_Reaction"
>HasReactionAliasFrom Reaction</a></dt>

<dd>
<dt><a name="Imported_Identifier"
>Imported Identifier</a></dt>

<dd>
<dt><a name="Provided_Subsystem"
>Provided Subsystem</a></dt>

<dd>
<dt><a name="Submitted_Genome"
>Submitted Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_SSCell"
>get_entity_SSCell</a></h2>

<p>An SSCell (SpreadSheet Cell) represents a role as it occurs in a subsystem spreadsheet row.
The key is a colon-delimited triple containing an MD5 hash of the subsystem ID followed by a genome ID (with optional region string) and a role abbreviation.</p>

<p>Example:</p>

<pre>    get_entity_SSCell -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SSCell entity has the following relationship links:</p>

<dl>
<dt><a name="Contains_Feature"
>Contains Feature</a></dt>

<dd>
<dt><a name="HasRole_Role"
>HasRole Role</a></dt>

<dd>
<dt><a name="IsRoleFor_SSRow"
>IsRoleFor SSRow</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_SSRow"
>get_entity_SSRow</a></h2>

<p>An SSRow (that is,
a row in a subsystem spreadsheet) represents a collection of functional roles present in the Features of a single Genome.
The roles are part of a designated subsystem,
and the fids associated with each role are included in the row,
That is,
a row amounts to an instance of a subsystem as it exists in a specific,
designated genome.</p>

<p>Example:</p>

<pre>    get_entity_SSRow -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SSRow entity has the following relationship links:</p>

<dl>
<dt><a name="Implements_Variant"
>Implements Variant</a></dt>

<dd>
<dt><a name="IsRowOf_SSCell"
>IsRowOf SSCell</a></dt>

<dd>
<dt><a name="IsUsedBy_Genome"
>IsUsedBy Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="curated"
>curated</a></dt>

<dd>
<dt><a name="region"
>region</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Subsystem"
>get_entity_Subsystem</a></h2>

<p>A subsystem is a set of functional roles that have been annotated simultaneously (e.g.,
the roles present in a specific pathway),
with an associated subsystem spreadsheet which encodes the fids in each genome that implement the functional roles in the subsystem.</p>

<p>Example:</p>

<pre>    get_entity_Subsystem -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Subsystem entity has the following relationship links:</p>

<dl>
<dt><a name="Describes_Variant"
>Describes Variant</a></dt>

<dd>
<dt><a name="Includes_Role"
>Includes Role</a></dt>

<dd>
<dt><a name="IsInClass_SubsystemClass"
>IsInClass SubsystemClass</a></dt>

<dd>
<dt><a name="IsRelevantTo_Diagram"
>IsRelevantTo Diagram</a></dt>

<dd>
<dt><a name="IsSubInstanceOf_Scenario"
>IsSubInstanceOf Scenario</a></dt>

<dd>
<dt><a name="WasProvidedBy_Source"
>WasProvidedBy Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_SubsystemClass"
>get_entity_SubsystemClass</a></h2>

<p>Subsystem classes impose a hierarchical organization on the subsystems.</p>

<p>Example:</p>

<pre>    get_entity_SubsystemClass -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SubsystemClass entity has the following relationship links:</p>

<dl>
<dt><a name="IsClassFor_Subsystem"
>IsClassFor Subsystem</a></dt>

<dd>
<dt><a name="IsSubclassOf_SubsystemClass"
>IsSubclassOf SubsystemClass</a></dt>

<dd>
<dt><a name="IsSuperclassOf_SubsystemClass"
>IsSuperclassOf SubsystemClass</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_TaxonomicGrouping"
>get_entity_TaxonomicGrouping</a></h2>

<p>We associate with most genomes a &#34;taxonomy&#34; based on the NCBI taxonomy.
This includes,
for each genome,
a list of ever larger taxonomic groups.</p>

<p>Example:</p>

<pre>    get_entity_TaxonomicGrouping -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The TaxonomicGrouping entity has the following relationship links:</p>

<dl>
<dt><a name="IsGroupFor_TaxonomicGrouping"
>IsGroupFor TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsInGroup_TaxonomicGrouping"
>IsInGroup TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsTaxonomyOf_Genome"
>IsTaxonomyOf Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Variant"
>get_entity_Variant</a></h2>

<p>Each subsystem may include the designation of distinct variants.
Thus,
there may be three closely-related,
but distinguishable forms of histidine degradation.
Each form would be called a &#34;variant&#34;,
with an associated code,
and all genomes implementing a specific variant can easily be accessed.</p>

<p>Example:</p>

<pre>    get_entity_Variant -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Variant entity has the following relationship links:</p>

<dl>
<dt><a name="IsDescribedBy_Subsystem"
>IsDescribedBy Subsystem</a></dt>

<dd>
<dt><a name="IsImplementedBy_SSRow"
>IsImplementedBy SSRow</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="role_rule"
>role_rule</a></dt>

<dd>
<dt><a name="code"
>code</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_entity_Variation"
>get_entity_Variation</a></h2>

<p>A variation describes a set of aligned regions in two or more contigs.</p>

<p>Example:</p>

<pre>    get_entity_Variation -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each filed in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Variation entity has the following relationship links:</p>

<dl>
<dt><a name="IsAlignmentFor_Contig"
>IsAlignmentFor Contig</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>Use the specified column to define the id of the entity to retrieve.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="notes"
>notes</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_AffectsLevelOf"
>get_relationship_AffectsLevelOf</a></h2>

<p>This relationship indicates the expression level of an atomic regulon for a given experiment.</p>

<p>Example:</p>

<pre>    get_relationship_AffectsLevelOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="level"
>level</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the AtomicRegulon entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Aligns"
>get_relationship_Aligns</a></h2>

<p>This relationship connects each alignment to its constituent protein sequences.
Each alignment contains many protein sequences,
and a single sequence can be in many alignments.
Parts of a single protein can occur in multiple places in an alignment.
The sequence-id field is used to keep these unique,
and is the string that represents the sequence in the alignment and tree text.</p>

<p>Example:</p>

<pre>    get_relationship_Aligns -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the AlignmentTree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="alignment_method"
>alignment_method</a></dt>

<dd>
<dt><a name="alignment_parameters"
>alignment_parameters</a></dt>

<dd>
<dt><a name="alignment_properties"
>alignment_properties</a></dt>

<dd>
<dt><a name="tree_method"
>tree_method</a></dt>

<dd>
<dt><a name="tree_parameters"
>tree_parameters</a></dt>

<dd>
<dt><a name="tree_properties"
>tree_properties</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="begin"
>begin</a></dt>

<dd>
<dt><a name="end"
>end</a></dt>

<dd>
<dt><a name="len"
>len</a></dt>

<dd>
<dt><a name="sequence_id"
>sequence_id</a></dt>

<dd>
<dt><a name="properties"
>properties</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Annotates"
>get_relationship_Annotates</a></h2>

<p>This relationship connects a feature to its annotations.
A feature may have multiple annotations,
but an annotation belongs to only one feature.</p>

<p>Example:</p>

<pre>    get_relationship_Annotates -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Annotation entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="annotator"
>annotator</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>

<dd>
<dt><a name="annotation_time"
>annotation_time</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Asserts"
>get_relationship_Asserts</a></h2>

<p>Sources (users) can make assertions about identifiers using the annotation clearinghouse.
When a user makes a new assertion about an identifier,
it erases the old one.</p>

<p>Example:</p>

<pre>    get_relationship_Asserts -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Identifier entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>

<dd>
<dt><a name="natural_form"
>natural_form</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="expert"
>expert</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Comprises"
>get_relationship_Comprises</a></h2>

<p>This relationship connects a biomass to the compound specifications that define it.</p>

<p>Example:</p>

<pre>    get_relationship_Comprises -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Biomass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the BiomassCompound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="coefficient"
>coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Concerns"
>get_relationship_Concerns</a></h2>

<p>This relationship connects a publication to the protein sequences it describes.</p>

<p>Example:</p>

<pre>    get_relationship_Concerns -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Publication entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="link"
>link</a></dt>

<dd>
<dt><a name="pubdate"
>pubdate</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Contains"
>get_relationship_Contains</a></h2>

<p>This relationship connects a subsystem spreadsheet cell to the features that occur in it.
A feature may occur in many machine roles and a machine role may contain many features.
The subsystem annotation process is essentially the maintenance of this relationship.</p>

<p>Example:</p>

<pre>    get_relationship_Contains -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSCell entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Controls"
>get_relationship_Controls</a></h2>

<p>This relationship connects a coregulated set to the features that are used as its transcription factors.</p>

<p>Example:</p>

<pre>    get_relationship_Controls -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the CoregulatedSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Describes"
>get_relationship_Describes</a></h2>

<p>This relationship connects a subsystem to the individual variants used to implement it.
Each variant contains a slightly different subset of the roles in the parent subsystem.</p>

<p>Example:</p>

<pre>    get_relationship_Describes -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Variant entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="role_rule"
>role_rule</a></dt>

<dd>
<dt><a name="code"
>code</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Determines"
>get_relationship_Determines</a></h2>

<p>A functional coupling evidence set exists because it has pairings in it,
and this relationship connects the evidence set to its constituent pairings.
A pairing cam belong to multiple evidence sets.</p>

<p>Example:</p>

<pre>    get_relationship_Determines -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the PairSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="score"
>score</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="inverted"
>inverted</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Pairing entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_DeterminesFunctionOf"
>get_relationship_DeterminesFunctionOf</a></h2>

<p>This relationship connects an isofunctional family to the roles that make up its assigned function.</p>

<p>Example:</p>

<pre>    get_relationship_DeterminesFunctionOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Displays"
>get_relationship_Displays</a></h2>

<p>This relationship connects a diagram to its reactions.
A diagram shows multiple reactions,
and a reaction can be on many diagrams.</p>

<p>Example:</p>

<pre>    get_relationship_Displays -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Diagram entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="location"
>location</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="equation"
>equation</a></dt>

<dd>
<dt><a name="reversibility"
>reversibility</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Encompasses"
>get_relationship_Encompasses</a></h2>

<p>This relationship connects a feature to a related feature; for example,
it would connect a gene to its constituent splice variants,
and the splice variants to their exons.</p>

<p>Example:</p>

<pre>    get_relationship_Encompasses -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Formulated"
>get_relationship_Formulated</a></h2>

<p>This relationship connects a coregulated set to the source organization that originally computed it.</p>

<p>Example:</p>

<pre>    get_relationship_Formulated -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the CoregulatedSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_GeneratedLevelsFor"
>get_relationship_GeneratedLevelsFor</a></h2>

<p>This relationship connects an atomic regulon to a probe set from which experimental data was produced for its features.
It contains a vector of the expression levels.</p>

<p>Example:</p>

<pre>    get_relationship_GeneratedLevelsFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProbeSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="level_vector"
>level_vector</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the AtomicRegulon entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HadResultsProducedBy"
>get_relationship_HadResultsProducedBy</a></h2>

<p>This relationship connects a probe set to a genome for which it was used to produce experimental results.
In general,
a probe set is used for only one genome and vice versa,
but this is not a requirement.</p>

<p>Example:</p>

<pre>    get_relationship_HadResultsProducedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProbeSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasAsExemplar"
>get_relationship_HasAsExemplar</a></h2>

<p>This relationship links a role to a feature that provides a typical example of how the role is implemented.</p>

<p>Example:</p>

<pre>    get_relationship_HasAsExemplar -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasAsSequence"
>get_relationship_HasAsSequence</a></h2>

<p>This relationship connects a Contig as it occurs in a genome to the Contig Sequence that represents the physical DNA base pairs.
A contig sequence may represent many contigs,
but each contig has only one sequence.</p>

<p>Example:</p>

<pre>    get_relationship_HasAsSequence -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ContigSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="length"
>length</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasAssertionFrom"
>get_relationship_HasAssertionFrom</a></h2>

<p>Sources (users) can make assertions about identifiers using the annotation clearinghouse.
When a user makes a new assertion about an identifier,
it erases the old one.</p>

<p>Example:</p>

<pre>    get_relationship_HasAssertionFrom -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Identifier entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>

<dd>
<dt><a name="natural_form"
>natural_form</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="expert"
>expert</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasAsTerminus"
>get_relationship_HasAsTerminus</a></h2>

<p>A terminus for a scenario is a compound that acts as its input or output.
A compound can be the terminus for many scenarios,
and a scenario will have many termini.
The relationship attributes indicate whether the compound is an input to the scenario or an output.
In some cases,
there may be multiple alternative output groups.
This is also indicated by the attributes.</p>

<p>Example:</p>

<pre>    get_relationship_HasAsTerminus -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="uncharged_formula"
>uncharged_formula</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="group_number"
>group_number</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Scenario entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="common_name"
>common_name</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasCompoundAliasFrom"
>get_relationship_HasCompoundAliasFrom</a></h2>

<p>This relationship connects a source (database or organization) with the compounds for which it has assigned names (aliases).
The alias itself is stored as intersection data.</p>

<p>Example:</p>

<pre>    get_relationship_HasCompoundAliasFrom -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="uncharged_formula"
>uncharged_formula</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasCoregulationWith"
>get_relationship_HasCoregulationWith</a></h2>

<p>This relationship connects a feature with another feature in the same genome with which it appears to be coregulated as a result of expression data analysis.</p>

<p>Example:</p>

<pre>    get_relationship_HasCoregulationWith -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="coefficient"
>coefficient</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasDefaultLocation"
>get_relationship_HasDefaultLocation</a></h2>

<p>This relationship connects a reagent to the compartment which is its default location during the reaction.</p>

<p>Example:</p>

<pre>    get_relationship_HasDefaultLocation -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compartment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reagent entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="stoichiometry"
>stoichiometry</a></dt>

<dd>
<dt><a name="cofactor"
>cofactor</a></dt>

<dd>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="transport_coefficient"
>transport_coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasFunctional"
>get_relationship_HasFunctional</a></h2>

<p>This relationship connects a role with the features in which it plays a functional part.</p>

<p>Example:</p>

<pre>    get_relationship_HasFunctional -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasIndicatedSignalFrom"
>get_relationship_HasIndicatedSignalFrom</a></h2>

<p>This relationship connects an experiment to a feature.
The feature expression levels inferred from the experimental results are stored here.</p>

<p>Example:</p>

<pre>    get_relationship_HasIndicatedSignalFrom -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="rma_value"
>rma_value</a></dt>

<dd>
<dt><a name="level"
>level</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasLevelsFrom"
>get_relationship_HasLevelsFrom</a></h2>

<p>This relationship connects a feature to a probe set from which experimental data was produced for the feature.
It contains a vector of the expression levels.</p>

<p>Example:</p>

<pre>    get_relationship_HasLevelsFrom -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProbeSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="level_vector"
>level_vector</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasMember"
>get_relationship_HasMember</a></h2>

<p>This relationship connects each feature family to its constituent features.
A family always has many features,
and a single feature can be found in many families.</p>

<p>Example:</p>

<pre>    get_relationship_HasMember -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasParticipant"
>get_relationship_HasParticipant</a></h2>

<p>A scenario consists of many participant reactions that convert the input compounds to output compounds.
A single reaction may participate in many scenarios.</p>

<p>Example:</p>

<pre>    get_relationship_HasParticipant -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Scenario entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="common_name"
>common_name</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="equation"
>equation</a></dt>

<dd>
<dt><a name="reversibility"
>reversibility</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasPresenceOf"
>get_relationship_HasPresenceOf</a></h2>

<p>This relationship connects a media to the compounds that occur in it.
The intersection data describes how much of each compound can be found.</p>

<p>Example:</p>

<pre>    get_relationship_HasPresenceOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Media entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="concentration"
>concentration</a></dt>

<dd>
<dt><a name="minimum_flux"
>minimum_flux</a></dt>

<dd>
<dt><a name="maximum_flux"
>maximum_flux</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="uncharged_formula"
>uncharged_formula</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasProposedLocationIn"
>get_relationship_HasProposedLocationIn</a></h2>

<p>This relationship connects a reaction as it is used in a complex to the compartments in which it usually takes place.
Most reactions take place in a single compartment.
Transporters take place in two compartments.</p>

<p>Example:</p>

<pre>    get_relationship_HasProposedLocationIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compartment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ReactionRule entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasReactionAliasFrom"
>get_relationship_HasReactionAliasFrom</a></h2>

<p>This relationship connects a source (database or organization) with the reactions for which it has assigned names (aliases).
The alias itself is stored as intersection data.</p>

<p>Example:</p>

<pre>    get_relationship_HasReactionAliasFrom -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="equation"
>equation</a></dt>

<dd>
<dt><a name="reversibility"
>reversibility</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasRealLocationIn"
>get_relationship_HasRealLocationIn</a></h2>

<p>This relationship connects a model&#39;s instance of a reaction to the compartments in which it takes place.
Most instances take place in a single compartment.
Transporters use two compartments.</p>

<p>Example:</p>

<pre>    get_relationship_HasRealLocationIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ModelCompartment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Requirement entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>

<dd>
<dt><a name="proton"
>proton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasRepresentativeOf"
>get_relationship_HasRepresentativeOf</a></h2>

<p>This relationship connects a genome to the FIGfam protein families for which it has representative proteins.
This information can be computed from other relationships,
but it is provided explicitly to allow fast access to a genome&#39;s FIGfam profile.</p>

<p>Example:</p>

<pre>    get_relationship_HasRepresentativeOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasResultsFor"
>get_relationship_HasResultsFor</a></h2>

<p>This relationship connects a probe set to the experiments that were applied to it.</p>

<p>Example:</p>

<pre>    get_relationship_HasResultsFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProbeSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasResultsIn"
>get_relationship_HasResultsIn</a></h2>

<p>This relationship connects a probe set to the experiments that were applied to it.</p>

<p>Example:</p>

<pre>    get_relationship_HasResultsIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProbeSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasRole"
>get_relationship_HasRole</a></h2>

<p>This relationship connects a role to the machine roles that represent its appearance in a molecular machine.
A machine role has exactly one associated role,
but a role may be represented by many machine roles.</p>

<p>Example:</p>

<pre>    get_relationship_HasRole -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSCell entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasSection"
>get_relationship_HasSection</a></h2>

<p>This relationship connects a contig&#39;s sequence to its DNA sequences.</p>

<p>Example:</p>

<pre>    get_relationship_HasSection -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ContigSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="length"
>length</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ContigChunk entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasStep"
>get_relationship_HasStep</a></h2>

<p>This relationship connects a complex to the reaction rules for the reactions that work together to make the complex happen.</p>

<p>Example:</p>

<pre>    get_relationship_HasStep -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Complex entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ReactionRule entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasUsage"
>get_relationship_HasUsage</a></h2>

<p>This relationship connects a biomass compound specification to the compounds for which it is relevant.</p>

<p>Example:</p>

<pre>    get_relationship_HasUsage -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="uncharged_formula"
>uncharged_formula</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the BiomassCompound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="coefficient"
>coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasValueFor"
>get_relationship_HasValueFor</a></h2>

<p>This relationship connects an experiment to its attributes.
The attribute values are stored here.</p>

<p>Example:</p>

<pre>    get_relationship_HasValueFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Attribute entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_HasValueIn"
>get_relationship_HasValueIn</a></h2>

<p>This relationship connects an experiment to its attributes.
The attribute values are stored here.</p>

<p>Example:</p>

<pre>    get_relationship_HasValueIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Attribute entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Implements"
>get_relationship_Implements</a></h2>

<p>This relationship connects a variant to the physical machines that implement it in the genomes.
A variant is implemented by many machines,
but a machine belongs to only one variant.</p>

<p>Example:</p>

<pre>    get_relationship_Implements -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Variant entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="role_rule"
>role_rule</a></dt>

<dd>
<dt><a name="code"
>code</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="curated"
>curated</a></dt>

<dd>
<dt><a name="region"
>region</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Imported"
>get_relationship_Imported</a></h2>

<p>This relationship specifies the import source for identifiers.
It is used when reloading identifiers to provide for rapid deletion of the previous load&#39;s results.</p>

<p>Example:</p>

<pre>    get_relationship_Imported -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Identifier entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>

<dd>
<dt><a name="natural_form"
>natural_form</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Includes"
>get_relationship_Includes</a></h2>

<p>A subsystem is defined by its roles.
The subsystem&#39;s variants contain slightly different sets of roles,
but all of the roles in a variant must be connected to the parent subsystem by this relationship.
A subsystem always has at least one role,
and a role always belongs to at least one subsystem.</p>

<p>Example:</p>

<pre>    get_relationship_Includes -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>

<dd>
<dt><a name="abbreviation"
>abbreviation</a></dt>

<dd>
<dt><a name="auxiliary"
>auxiliary</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IncludesPartOf"
>get_relationship_IncludesPartOf</a></h2>

<p>A Scenario overlaps a diagram when the diagram displays a portion of the reactions that make up the scenario.
A scenario may overlap many diagrams,
and a diagram may be include portions of many scenarios.</p>

<p>Example:</p>

<pre>    get_relationship_IncludesPartOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Scenario entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="common_name"
>common_name</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Diagram entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IndicatedLevelsFor"
>get_relationship_IndicatedLevelsFor</a></h2>

<p>This relationship connects a feature to a probe set from which experimental data was produced for the feature.
It contains a vector of the expression levels.</p>

<p>Example:</p>

<pre>    get_relationship_IndicatedLevelsFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProbeSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="level_vector"
>level_vector</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IndicatesSignalFor"
>get_relationship_IndicatesSignalFor</a></h2>

<p>This relationship connects an experiment to a feature.
The feature expression levels inferred from the experimental results are stored here.</p>

<p>Example:</p>

<pre>    get_relationship_IndicatesSignalFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="rma_value"
>rma_value</a></dt>

<dd>
<dt><a name="level"
>level</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Involves"
>get_relationship_Involves</a></h2>

<p>This relationship connects a reaction to the reagents representing the compounds that participate in it.</p>

<p>Example:</p>

<pre>    get_relationship_Involves -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="equation"
>equation</a></dt>

<dd>
<dt><a name="reversibility"
>reversibility</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reagent entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="stoichiometry"
>stoichiometry</a></dt>

<dd>
<dt><a name="cofactor"
>cofactor</a></dt>

<dd>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="transport_coefficient"
>transport_coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsAffectedIn"
>get_relationship_IsAffectedIn</a></h2>

<p>This relationship indicates the expression level of an atomic regulon for a given experiment.</p>

<p>Example:</p>

<pre>    get_relationship_IsAffectedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="level"
>level</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the AtomicRegulon entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsAlignedBy"
>get_relationship_IsAlignedBy</a></h2>

<p>This relationship connects each alignment to its constituent protein sequences.
Each alignment contains many protein sequences,
and a single sequence can be in many alignments.
Parts of a single protein can occur in multiple places in an alignment.
The sequence-id field is used to keep these unique,
and is the string that represents the sequence in the alignment and tree text.</p>

<p>Example:</p>

<pre>    get_relationship_IsAlignedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the AlignmentTree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="alignment_method"
>alignment_method</a></dt>

<dd>
<dt><a name="alignment_parameters"
>alignment_parameters</a></dt>

<dd>
<dt><a name="alignment_properties"
>alignment_properties</a></dt>

<dd>
<dt><a name="tree_method"
>tree_method</a></dt>

<dd>
<dt><a name="tree_parameters"
>tree_parameters</a></dt>

<dd>
<dt><a name="tree_properties"
>tree_properties</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="begin"
>begin</a></dt>

<dd>
<dt><a name="end"
>end</a></dt>

<dd>
<dt><a name="len"
>len</a></dt>

<dd>
<dt><a name="sequence_id"
>sequence_id</a></dt>

<dd>
<dt><a name="properties"
>properties</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsAlignedIn"
>get_relationship_IsAlignedIn</a></h2>

<p>This relationship connects each variation to the contig regions that it aligns.</p>

<p>Example:</p>

<pre>    get_relationship_IsAlignedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="start"
>start</a></dt>

<dd>
<dt><a name="len"
>len</a></dt>

<dd>
<dt><a name="dir"
>dir</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Variation entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsAlignmentFor"
>get_relationship_IsAlignmentFor</a></h2>

<p>This relationship connects each variation to the contig regions that it aligns.</p>

<p>Example:</p>

<pre>    get_relationship_IsAlignmentFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="start"
>start</a></dt>

<dd>
<dt><a name="len"
>len</a></dt>

<dd>
<dt><a name="dir"
>dir</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Variation entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsAnnotatedBy"
>get_relationship_IsAnnotatedBy</a></h2>

<p>This relationship connects a feature to its annotations.
A feature may have multiple annotations,
but an annotation belongs to only one feature.</p>

<p>Example:</p>

<pre>    get_relationship_IsAnnotatedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Annotation entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="annotator"
>annotator</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>

<dd>
<dt><a name="annotation_time"
>annotation_time</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsARequirementIn"
>get_relationship_IsARequirementIn</a></h2>

<p>This relationship connects a model to its requirements.
A requirement represents the use of a reaction in a single model.</p>

<p>Example:</p>

<pre>    get_relationship_IsARequirementIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Model entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Requirement entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>

<dd>
<dt><a name="proton"
>proton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsARequirementOf"
>get_relationship_IsARequirementOf</a></h2>

<p>This relationship connects a model to its requirements.
A requirement represents the use of a reaction in a single model.</p>

<p>Example:</p>

<pre>    get_relationship_IsARequirementOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Model entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Requirement entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>

<dd>
<dt><a name="proton"
>proton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsATopicOf"
>get_relationship_IsATopicOf</a></h2>

<p>This relationship connects a publication to the protein sequences it describes.</p>

<p>Example:</p>

<pre>    get_relationship_IsATopicOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Publication entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="link"
>link</a></dt>

<dd>
<dt><a name="pubdate"
>pubdate</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsBindingSiteFor"
>get_relationship_IsBindingSiteFor</a></h2>

<p>This relationship connects a coregulated set to the binding site to which its feature attaches.</p>

<p>Example:</p>

<pre>    get_relationship_IsBindingSiteFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the CoregulatedSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsBoundBy"
>get_relationship_IsBoundBy</a></h2>

<p>This relationship connects a coregulated set to the binding site to which its feature attaches.</p>

<p>Example:</p>

<pre>    get_relationship_IsBoundBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the CoregulatedSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsClassFor"
>get_relationship_IsClassFor</a></h2>

<p>This relationship connects each subsystem class with the subsystems that belong to it.
A class can contain many subsystems,
but a subsystem is only in one class.
Some subsystems are not in any class,
but this is usually a temporary condition.</p>

<p>Example:</p>

<pre>    get_relationship_IsClassFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the SubsystemClass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsCollectedInto"
>get_relationship_IsCollectedInto</a></h2>

<p>A genome belongs to only one genome set.
For each set,
this relationship marks the genome to be used as its representative.</p>

<p>Example:</p>

<pre>    get_relationship_IsCollectedInto -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the OTU entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="representative"
>representative</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsCollectionOf"
>get_relationship_IsCollectionOf</a></h2>

<p>A genome belongs to only one genome set.
For each set,
this relationship marks the genome to be used as its representative.</p>

<p>Example:</p>

<pre>    get_relationship_IsCollectionOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the OTU entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="representative"
>representative</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsComponentOf"
>get_relationship_IsComponentOf</a></h2>

<p>This relationship connects a genome to its constituent contigs.
Unlike contig sequences,
a contig belongs to only one genome.</p>

<p>Example:</p>

<pre>    get_relationship_IsComponentOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsComposedOf"
>get_relationship_IsComposedOf</a></h2>

<p>This relationship connects a genome to its constituent contigs.
Unlike contig sequences,
a contig belongs to only one genome.</p>

<p>Example:</p>

<pre>    get_relationship_IsComposedOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsComprisedOf"
>get_relationship_IsComprisedOf</a></h2>

<p>This relationship connects a biomass to the compound specifications that define it.</p>

<p>Example:</p>

<pre>    get_relationship_IsComprisedOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Biomass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the BiomassCompound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="coefficient"
>coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsConfiguredBy"
>get_relationship_IsConfiguredBy</a></h2>

<p>This relationship connects a genome to the atomic regulons that describe its state.</p>

<p>Example:</p>

<pre>    get_relationship_IsConfiguredBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the AtomicRegulon entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsConsistentTo"
>get_relationship_IsConsistentTo</a></h2>

<p>This relationship connects a functional role to the EC numbers consistent with the chemistry described in the role.</p>

<p>Example:</p>

<pre>    get_relationship_IsConsistentTo -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the EcNumber entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="obsolete"
>obsolete</a></dt>

<dd>
<dt><a name="replacedby"
>replacedby</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsConsistentWith"
>get_relationship_IsConsistentWith</a></h2>

<p>This relationship connects a functional role to the EC numbers consistent with the chemistry described in the role.</p>

<p>Example:</p>

<pre>    get_relationship_IsConsistentWith -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the EcNumber entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="obsolete"
>obsolete</a></dt>

<dd>
<dt><a name="replacedby"
>replacedby</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsContainedIn"
>get_relationship_IsContainedIn</a></h2>

<p>This relationship connects a subsystem spreadsheet cell to the features that occur in it.
A feature may occur in many machine roles and a machine role may contain many features.
The subsystem annotation process is essentially the maintenance of this relationship.</p>

<p>Example:</p>

<pre>    get_relationship_IsContainedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSCell entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsControlledUsing"
>get_relationship_IsControlledUsing</a></h2>

<p>This relationship connects a coregulated set to the features that are used as its transcription factors.</p>

<p>Example:</p>

<pre>    get_relationship_IsControlledUsing -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the CoregulatedSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsCoregulatedWith"
>get_relationship_IsCoregulatedWith</a></h2>

<p>This relationship connects a feature with another feature in the same genome with which it appears to be coregulated as a result of expression data analysis.</p>

<p>Example:</p>

<pre>    get_relationship_IsCoregulatedWith -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="coefficient"
>coefficient</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsCoupledTo"
>get_relationship_IsCoupledTo</a></h2>

<p>This relationship connects two FIGfams that we believe to be related either because their members occur in proximity on chromosomes or because the members are expressed together.
Such a relationship is evidence the functions of the FIGfams are themselves related.
This relationship is commutative; only the instance in which the first FIGfam has a lower ID than the second is stored.</p>

<p>Example:</p>

<pre>    get_relationship_IsCoupledTo -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="co_occurrence_evidence"
>co_occurrence_evidence</a></dt>

<dd>
<dt><a name="co_expression_evidence"
>co_expression_evidence</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsCoupledWith"
>get_relationship_IsCoupledWith</a></h2>

<p>This relationship connects two FIGfams that we believe to be related either because their members occur in proximity on chromosomes or because the members are expressed together.
Such a relationship is evidence the functions of the FIGfams are themselves related.
This relationship is commutative; only the instance in which the first FIGfam has a lower ID than the second is stored.</p>

<p>Example:</p>

<pre>    get_relationship_IsCoupledWith -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="co_occurrence_evidence"
>co_occurrence_evidence</a></dt>

<dd>
<dt><a name="co_expression_evidence"
>co_expression_evidence</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsDefaultFor"
>get_relationship_IsDefaultFor</a></h2>

<p>This relationship connects a reaction to the compartment in which it runs by default.</p>

<p>Example:</p>

<pre>    get_relationship_IsDefaultFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compartment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="equation"
>equation</a></dt>

<dd>
<dt><a name="reversibility"
>reversibility</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsDefaultLocationOf"
>get_relationship_IsDefaultLocationOf</a></h2>

<p>This relationship connects a reagent to the compartment which is its default location during the reaction.</p>

<p>Example:</p>

<pre>    get_relationship_IsDefaultLocationOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compartment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reagent entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="stoichiometry"
>stoichiometry</a></dt>

<dd>
<dt><a name="cofactor"
>cofactor</a></dt>

<dd>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="transport_coefficient"
>transport_coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsDescribedBy"
>get_relationship_IsDescribedBy</a></h2>

<p>This relationship connects a subsystem to the individual variants used to implement it.
Each variant contains a slightly different subset of the roles in the parent subsystem.</p>

<p>Example:</p>

<pre>    get_relationship_IsDescribedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Variant entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="role_rule"
>role_rule</a></dt>

<dd>
<dt><a name="code"
>code</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsDeterminedBy"
>get_relationship_IsDeterminedBy</a></h2>

<p>A functional coupling evidence set exists because it has pairings in it,
and this relationship connects the evidence set to its constituent pairings.
A pairing cam belong to multiple evidence sets.</p>

<p>Example:</p>

<pre>    get_relationship_IsDeterminedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the PairSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="score"
>score</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="inverted"
>inverted</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Pairing entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsDisplayedOn"
>get_relationship_IsDisplayedOn</a></h2>

<p>This relationship connects a diagram to its reactions.
A diagram shows multiple reactions,
and a reaction can be on many diagrams.</p>

<p>Example:</p>

<pre>    get_relationship_IsDisplayedOn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Diagram entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="location"
>location</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="equation"
>equation</a></dt>

<dd>
<dt><a name="reversibility"
>reversibility</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsDividedInto"
>get_relationship_IsDividedInto</a></h2>

<p>This relationship connects a model to the cell compartments that participate in the model.</p>

<p>Example:</p>

<pre>    get_relationship_IsDividedInto -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Model entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ModelCompartment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsDivisionOf"
>get_relationship_IsDivisionOf</a></h2>

<p>This relationship connects a model to the cell compartments that participate in the model.</p>

<p>Example:</p>

<pre>    get_relationship_IsDivisionOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Model entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ModelCompartment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsEncompassedIn"
>get_relationship_IsEncompassedIn</a></h2>

<p>This relationship connects a feature to a related feature; for example,
it would connect a gene to its constituent splice variants,
and the splice variants to their exons.</p>

<p>Example:</p>

<pre>    get_relationship_IsEncompassedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsExemplarOf"
>get_relationship_IsExemplarOf</a></h2>

<p>This relationship links a role to a feature that provides a typical example of how the role is implemented.</p>

<p>Example:</p>

<pre>    get_relationship_IsExemplarOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsFamilyFor"
>get_relationship_IsFamilyFor</a></h2>

<p>This relationship connects an isofunctional family to the roles that make up its assigned function.</p>

<p>Example:</p>

<pre>    get_relationship_IsFamilyFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsFormedInto"
>get_relationship_IsFormedInto</a></h2>

<p>This relationship connects each feature to the atomic regulon to which it belongs.</p>

<p>Example:</p>

<pre>    get_relationship_IsFormedInto -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the AtomicRegulon entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsFormedOf"
>get_relationship_IsFormedOf</a></h2>

<p>This relationship connects each feature to the atomic regulon to which it belongs.</p>

<p>Example:</p>

<pre>    get_relationship_IsFormedOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the AtomicRegulon entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsFunctionalIn"
>get_relationship_IsFunctionalIn</a></h2>

<p>This relationship connects a role with the features in which it plays a functional part.</p>

<p>Example:</p>

<pre>    get_relationship_IsFunctionalIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsGroupFor"
>get_relationship_IsGroupFor</a></h2>

<p>The recursive IsGroupFor relationship organizes taxonomic groupings into a hierarchy based on the standard organism taxonomy.</p>

<p>Example:</p>

<pre>    get_relationship_IsGroupFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the TaxonomicGrouping entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the TaxonomicGrouping entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsImplementedBy"
>get_relationship_IsImplementedBy</a></h2>

<p>This relationship connects a variant to the physical machines that implement it in the genomes.
A variant is implemented by many machines,
but a machine belongs to only one variant.</p>

<p>Example:</p>

<pre>    get_relationship_IsImplementedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Variant entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="role_rule"
>role_rule</a></dt>

<dd>
<dt><a name="code"
>code</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="curated"
>curated</a></dt>

<dd>
<dt><a name="region"
>region</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsInClass"
>get_relationship_IsInClass</a></h2>

<p>This relationship connects each subsystem class with the subsystems that belong to it.
A class can contain many subsystems,
but a subsystem is only in one class.
Some subsystems are not in any class,
but this is usually a temporary condition.</p>

<p>Example:</p>

<pre>    get_relationship_IsInClass -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the SubsystemClass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsIncludedIn"
>get_relationship_IsIncludedIn</a></h2>

<p>A subsystem is defined by its roles.
The subsystem&#39;s variants contain slightly different sets of roles,
but all of the roles in a variant must be connected to the parent subsystem by this relationship.
A subsystem always has at least one role,
and a role always belongs to at least one subsystem.</p>

<p>Example:</p>

<pre>    get_relationship_IsIncludedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>

<dd>
<dt><a name="abbreviation"
>abbreviation</a></dt>

<dd>
<dt><a name="auxiliary"
>auxiliary</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsInGroup"
>get_relationship_IsInGroup</a></h2>

<p>The recursive IsGroupFor relationship organizes taxonomic groupings into a hierarchy based on the standard organism taxonomy.</p>

<p>Example:</p>

<pre>    get_relationship_IsInGroup -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the TaxonomicGrouping entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the TaxonomicGrouping entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsInPair"
>get_relationship_IsInPair</a></h2>

<p>A pairing contains exactly two protein sequences.
A protein sequence can belong to multiple pairings.
When going from a protein sequence to its pairings,
they are presented in alphabetical order by sequence key.</p>

<p>Example:</p>

<pre>    get_relationship_IsInPair -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Pairing entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsInstanceOf"
>get_relationship_IsInstanceOf</a></h2>

<p>This relationship connects a compartment to the instances of that compartment that occur in models.</p>

<p>Example:</p>

<pre>    get_relationship_IsInstanceOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compartment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ModelCompartment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsInstantiatedBy"
>get_relationship_IsInstantiatedBy</a></h2>

<p>This relationship connects a compartment to the instances of that compartment that occur in models.</p>

<p>Example:</p>

<pre>    get_relationship_IsInstantiatedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compartment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ModelCompartment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsInTaxa"
>get_relationship_IsInTaxa</a></h2>

<p>A genome is assigned to a particular point in the taxonomy tree,
but not necessarily to a leaf node.
In some cases,
the exact species and strain is not available when inserting the genome,
so it is placed at the lowest node that probably contains the actual genome.</p>

<p>Example:</p>

<pre>    get_relationship_IsInTaxa -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the TaxonomicGrouping entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsInvolvedIn"
>get_relationship_IsInvolvedIn</a></h2>

<p>This relationship connects a reaction to the reagents representing the compounds that participate in it.</p>

<p>Example:</p>

<pre>    get_relationship_IsInvolvedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="equation"
>equation</a></dt>

<dd>
<dt><a name="reversibility"
>reversibility</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reagent entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="stoichiometry"
>stoichiometry</a></dt>

<dd>
<dt><a name="cofactor"
>cofactor</a></dt>

<dd>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="transport_coefficient"
>transport_coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsLocatedIn"
>get_relationship_IsLocatedIn</a></h2>

<p>A feature is a set of DNA sequence fragments.
Most features are a single contiquous fragment,
so they are located in only one DNA sequence; however,
fragments have a maximum length,
so even a single contiguous feature may participate in this relationship multiple times.
A few features belong to multiple DNA sequences.
In that case,
however,
all the DNA sequences belong to the same genome.
A DNA sequence itself will frequently have thousands of features connected to it.</p>

<p>Example:</p>

<pre>    get_relationship_IsLocatedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="ordinal"
>ordinal</a></dt>

<dd>
<dt><a name="begin"
>begin</a></dt>

<dd>
<dt><a name="len"
>len</a></dt>

<dd>
<dt><a name="dir"
>dir</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsLocusFor"
>get_relationship_IsLocusFor</a></h2>

<p>A feature is a set of DNA sequence fragments.
Most features are a single contiquous fragment,
so they are located in only one DNA sequence; however,
fragments have a maximum length,
so even a single contiguous feature may participate in this relationship multiple times.
A few features belong to multiple DNA sequences.
In that case,
however,
all the DNA sequences belong to the same genome.
A DNA sequence itself will frequently have thousands of features connected to it.</p>

<p>Example:</p>

<pre>    get_relationship_IsLocusFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="ordinal"
>ordinal</a></dt>

<dd>
<dt><a name="begin"
>begin</a></dt>

<dd>
<dt><a name="len"
>len</a></dt>

<dd>
<dt><a name="dir"
>dir</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsManagedBy"
>get_relationship_IsManagedBy</a></h2>

<p>This relationship connects a model to the biomasses that are monitored to determine whether or not the model is effective.</p>

<p>Example:</p>

<pre>    get_relationship_IsManagedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Model entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Biomass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsMemberOf"
>get_relationship_IsMemberOf</a></h2>

<p>This relationship connects each feature family to its constituent features.
A family always has many features,
and a single feature can be found in many families.</p>

<p>Example:</p>

<pre>    get_relationship_IsMemberOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsModeledBy"
>get_relationship_IsModeledBy</a></h2>

<p>A genome can be modeled by many different models,
but a model belongs to only one genome.</p>

<p>Example:</p>

<pre>    get_relationship_IsModeledBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Model entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsNamedBy"
>get_relationship_IsNamedBy</a></h2>

<p>The normal case is that an identifier names a single protein sequence,
while a protein sequence can have many identifiers,
but some identifiers name multiple sequences.</p>

<p>Example:</p>

<pre>    get_relationship_IsNamedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Identifier entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>

<dd>
<dt><a name="natural_form"
>natural_form</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsOwnedBy"
>get_relationship_IsOwnedBy</a></h2>

<p>This relationship connects a genome to the features it contains.
Though technically redundant (the information is available from the feature&#39;s contigs),
it simplifies the extremely common process of finding all features for a genome.</p>

<p>Example:</p>

<pre>    get_relationship_IsOwnedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsOwnerOf"
>get_relationship_IsOwnerOf</a></h2>

<p>This relationship connects a genome to the features it contains.
Though technically redundant (the information is available from the feature&#39;s contigs),
it simplifies the extremely common process of finding all features for a genome.</p>

<p>Example:</p>

<pre>    get_relationship_IsOwnerOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsPairOf"
>get_relationship_IsPairOf</a></h2>

<p>A pairing contains exactly two protein sequences.
A protein sequence can belong to multiple pairings.
When going from a protein sequence to its pairings,
they are presented in alphabetical order by sequence key.</p>

<p>Example:</p>

<pre>    get_relationship_IsPairOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Pairing entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsParticipationOf"
>get_relationship_IsParticipationOf</a></h2>

<p>This relationship connects a compound to the reagents that represent its participation in reactions.</p>

<p>Example:</p>

<pre>    get_relationship_IsParticipationOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="uncharged_formula"
>uncharged_formula</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reagent entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="stoichiometry"
>stoichiometry</a></dt>

<dd>
<dt><a name="cofactor"
>cofactor</a></dt>

<dd>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="transport_coefficient"
>transport_coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsPresentIn"
>get_relationship_IsPresentIn</a></h2>

<p>This relationship connects a media to the compounds that occur in it.
The intersection data describes how much of each compound can be found.</p>

<p>Example:</p>

<pre>    get_relationship_IsPresentIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Media entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="concentration"
>concentration</a></dt>

<dd>
<dt><a name="minimum_flux"
>minimum_flux</a></dt>

<dd>
<dt><a name="maximum_flux"
>maximum_flux</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="uncharged_formula"
>uncharged_formula</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsProjectedOnto"
>get_relationship_IsProjectedOnto</a></h2>

<p>This relationship connects two protein sequences for which a clear bidirectional best hit exists in known genomes.
The attributes of the relationship describe how good the relationship is between the proteins.
The relationship is bidirectional and symmetric,
but is only stored in one direction (lower ID to higher ID).</p>

<p>Example:</p>

<pre>    get_relationship_IsProjectedOnto -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="gene_context"
>gene_context</a></dt>

<dd>
<dt><a name="percent_identity"
>percent_identity</a></dt>

<dd>
<dt><a name="score"
>score</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsProposedLocationOf"
>get_relationship_IsProposedLocationOf</a></h2>

<p>This relationship connects a reaction as it is used in a complex to the compartments in which it usually takes place.
Most reactions take place in a single compartment.
Transporters take place in two compartments.</p>

<p>Example:</p>

<pre>    get_relationship_IsProposedLocationOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compartment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ReactionRule entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsProteinFor"
>get_relationship_IsProteinFor</a></h2>

<p>This relationship connects a peg feature to the protein sequence it produces (if any).
Only peg features participate in this relationship.
A single protein sequence will frequently be produced by many features.</p>

<p>Example:</p>

<pre>    get_relationship_IsProteinFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsRealLocationOf"
>get_relationship_IsRealLocationOf</a></h2>

<p>This relationship connects a model&#39;s instance of a reaction to the compartments in which it takes place.
Most instances take place in a single compartment.
Transporters use two compartments.</p>

<p>Example:</p>

<pre>    get_relationship_IsRealLocationOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ModelCompartment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Requirement entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>

<dd>
<dt><a name="proton"
>proton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsRegulatedIn"
>get_relationship_IsRegulatedIn</a></h2>

<p>This relationship connects a feature to the set of coregulated features.</p>

<p>Example:</p>

<pre>    get_relationship_IsRegulatedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the CoregulatedSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsRegulatedSetOf"
>get_relationship_IsRegulatedSetOf</a></h2>

<p>This relationship connects a feature to the set of coregulated features.</p>

<p>Example:</p>

<pre>    get_relationship_IsRegulatedSetOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the CoregulatedSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsRelevantFor"
>get_relationship_IsRelevantFor</a></h2>

<p>This relationship connects a diagram to the subsystems that are depicted on it.
Only diagrams which are useful in curating or annotation the subsystem are specified in this relationship.</p>

<p>Example:</p>

<pre>    get_relationship_IsRelevantFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Diagram entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsRelevantTo"
>get_relationship_IsRelevantTo</a></h2>

<p>This relationship connects a diagram to the subsystems that are depicted on it.
Only diagrams which are useful in curating or annotation the subsystem are specified in this relationship.</p>

<p>Example:</p>

<pre>    get_relationship_IsRelevantTo -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Diagram entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsRepresentedIn"
>get_relationship_IsRepresentedIn</a></h2>

<p>This relationship connects a genome to the FIGfam protein families for which it has representative proteins.
This information can be computed from other relationships,
but it is provided explicitly to allow fast access to a genome&#39;s FIGfam profile.</p>

<p>Example:</p>

<pre>    get_relationship_IsRepresentedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsRequiredBy"
>get_relationship_IsRequiredBy</a></h2>

<p>This relationship links a reaction to the way it is used in a model.</p>

<p>Example:</p>

<pre>    get_relationship_IsRequiredBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="equation"
>equation</a></dt>

<dd>
<dt><a name="reversibility"
>reversibility</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Requirement entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>

<dd>
<dt><a name="proton"
>proton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsRoleFor"
>get_relationship_IsRoleFor</a></h2>

<p>This relationship connects a subsystem spreadsheet row to its constituent spreadsheet cells.</p>

<p>Example:</p>

<pre>    get_relationship_IsRoleFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="curated"
>curated</a></dt>

<dd>
<dt><a name="region"
>region</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSCell entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsRoleOf"
>get_relationship_IsRoleOf</a></h2>

<p>This relationship connects a role to the machine roles that represent its appearance in a molecular machine.
A machine role has exactly one associated role,
but a role may be represented by many machine roles.</p>

<p>Example:</p>

<pre>    get_relationship_IsRoleOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSCell entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsRowOf"
>get_relationship_IsRowOf</a></h2>

<p>This relationship connects a subsystem spreadsheet row to its constituent spreadsheet cells.</p>

<p>Example:</p>

<pre>    get_relationship_IsRowOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="curated"
>curated</a></dt>

<dd>
<dt><a name="region"
>region</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSCell entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsSectionOf"
>get_relationship_IsSectionOf</a></h2>

<p>This relationship connects a contig&#39;s sequence to its DNA sequences.</p>

<p>Example:</p>

<pre>    get_relationship_IsSectionOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ContigSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="length"
>length</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ContigChunk entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsSequenceOf"
>get_relationship_IsSequenceOf</a></h2>

<p>This relationship connects a Contig as it occurs in a genome to the Contig Sequence that represents the physical DNA base pairs.
A contig sequence may represent many contigs,
but each contig has only one sequence.</p>

<p>Example:</p>

<pre>    get_relationship_IsSequenceOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ContigSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="length"
>length</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsShownOn"
>get_relationship_IsShownOn</a></h2>

<p>This relationship indicates that a compound appears on a particular diagram.
The same compound can appear on many diagrams,
and a diagram always contains many compounds.</p>

<p>Example:</p>

<pre>    get_relationship_IsShownOn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Diagram entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="location"
>location</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="uncharged_formula"
>uncharged_formula</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsStepOf"
>get_relationship_IsStepOf</a></h2>

<p>This relationship connects a complex to the reaction rules for the reactions that work together to make the complex happen.</p>

<p>Example:</p>

<pre>    get_relationship_IsStepOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Complex entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ReactionRule entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsSubclassOf"
>get_relationship_IsSubclassOf</a></h2>

<p>This is a recursive relationship that imposes a hierarchy on the subsystem classes.</p>

<p>Example:</p>

<pre>    get_relationship_IsSubclassOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the SubsystemClass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SubsystemClass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsSubInstanceOf"
>get_relationship_IsSubInstanceOf</a></h2>

<p>This relationship connects a scenario to its subsystem it validates.
A scenario belongs to exactly one subsystem,
but a subsystem may have multiple scenarios.</p>

<p>Example:</p>

<pre>    get_relationship_IsSubInstanceOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Scenario entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="common_name"
>common_name</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsSuperclassOf"
>get_relationship_IsSuperclassOf</a></h2>

<p>This is a recursive relationship that imposes a hierarchy on the subsystem classes.</p>

<p>Example:</p>

<pre>    get_relationship_IsSuperclassOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the SubsystemClass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SubsystemClass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsTargetOf"
>get_relationship_IsTargetOf</a></h2>

<p>This relationship connects a compound in a biomass to the compartment in which it is supposed to appear.</p>

<p>Example:</p>

<pre>    get_relationship_IsTargetOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ModelCompartment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the BiomassCompound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="coefficient"
>coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsTaxonomyOf"
>get_relationship_IsTaxonomyOf</a></h2>

<p>A genome is assigned to a particular point in the taxonomy tree,
but not necessarily to a leaf node.
In some cases,
the exact species and strain is not available when inserting the genome,
so it is placed at the lowest node that probably contains the actual genome.</p>

<p>Example:</p>

<pre>    get_relationship_IsTaxonomyOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the TaxonomicGrouping entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsTerminusFor"
>get_relationship_IsTerminusFor</a></h2>

<p>A terminus for a scenario is a compound that acts as its input or output.
A compound can be the terminus for many scenarios,
and a scenario will have many termini.
The relationship attributes indicate whether the compound is an input to the scenario or an output.
In some cases,
there may be multiple alternative output groups.
This is also indicated by the attributes.</p>

<p>Example:</p>

<pre>    get_relationship_IsTerminusFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="uncharged_formula"
>uncharged_formula</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="group_number"
>group_number</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Scenario entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="common_name"
>common_name</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsTriggeredBy"
>get_relationship_IsTriggeredBy</a></h2>

<p>A complex can be triggered by many roles.
A role can trigger many complexes.</p>

<p>Example:</p>

<pre>    get_relationship_IsTriggeredBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Complex entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="optional"
>optional</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsUsageOf"
>get_relationship_IsUsageOf</a></h2>

<p>This relationship connects a biomass compound specification to the compounds for which it is relevant.</p>

<p>Example:</p>

<pre>    get_relationship_IsUsageOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="uncharged_formula"
>uncharged_formula</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the BiomassCompound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="coefficient"
>coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsUsedAs"
>get_relationship_IsUsedAs</a></h2>

<p>This relationship connects a reaction to its usage in specific complexes.</p>

<p>Example:</p>

<pre>    get_relationship_IsUsedAs -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="equation"
>equation</a></dt>

<dd>
<dt><a name="reversibility"
>reversibility</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ReactionRule entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsUsedBy"
>get_relationship_IsUsedBy</a></h2>

<p>This relationship connects a genome to the machines that form its metabolic pathways.
A genome can use many machines,
but a machine is used by exactly one genome.</p>

<p>Example:</p>

<pre>    get_relationship_IsUsedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="curated"
>curated</a></dt>

<dd>
<dt><a name="region"
>region</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsUseOf"
>get_relationship_IsUseOf</a></h2>

<p>This relationship connects a reaction to its usage in specific complexes.</p>

<p>Example:</p>

<pre>    get_relationship_IsUseOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="equation"
>equation</a></dt>

<dd>
<dt><a name="reversibility"
>reversibility</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ReactionRule entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_IsUtilizedIn"
>get_relationship_IsUtilizedIn</a></h2>

<p>This relationship connects an experiment to the media in which the experiment took place.</p>

<p>Example:</p>

<pre>    get_relationship_IsUtilizedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Media entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Manages"
>get_relationship_Manages</a></h2>

<p>This relationship connects a model to the biomasses that are monitored to determine whether or not the model is effective.</p>

<p>Example:</p>

<pre>    get_relationship_Manages -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Model entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Biomass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Models"
>get_relationship_Models</a></h2>

<p>A genome can be modeled by many different models,
but a model belongs to only one genome.</p>

<p>Example:</p>

<pre>    get_relationship_Models -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Model entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Names"
>get_relationship_Names</a></h2>

<p>The normal case is that an identifier names a single protein sequence,
while a protein sequence can have many identifiers,
but some identifiers name multiple sequences.</p>

<p>Example:</p>

<pre>    get_relationship_Names -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Identifier entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>

<dd>
<dt><a name="natural_form"
>natural_form</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_OperatesIn"
>get_relationship_OperatesIn</a></h2>

<p>This relationship connects an experiment to the media in which the experiment took place.</p>

<p>Example:</p>

<pre>    get_relationship_OperatesIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Media entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Overlaps"
>get_relationship_Overlaps</a></h2>

<p>A Scenario overlaps a diagram when the diagram displays a portion of the reactions that make up the scenario.
A scenario may overlap many diagrams,
and a diagram may be include portions of many scenarios.</p>

<p>Example:</p>

<pre>    get_relationship_Overlaps -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Scenario entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="common_name"
>common_name</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Diagram entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_ParticipatesAs"
>get_relationship_ParticipatesAs</a></h2>

<p>This relationship connects a compound to the reagents that represent its participation in reactions.</p>

<p>Example:</p>

<pre>    get_relationship_ParticipatesAs -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="uncharged_formula"
>uncharged_formula</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reagent entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="stoichiometry"
>stoichiometry</a></dt>

<dd>
<dt><a name="cofactor"
>cofactor</a></dt>

<dd>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="transport_coefficient"
>transport_coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_ParticipatesIn"
>get_relationship_ParticipatesIn</a></h2>

<p>A scenario consists of many participant reactions that convert the input compounds to output compounds.
A single reaction may participate in many scenarios.</p>

<p>Example:</p>

<pre>    get_relationship_ParticipatesIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Scenario entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="common_name"
>common_name</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="equation"
>equation</a></dt>

<dd>
<dt><a name="reversibility"
>reversibility</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_ProducedResultsFor"
>get_relationship_ProducedResultsFor</a></h2>

<p>This relationship connects a probe set to a genome for which it was used to produce experimental results.
In general,
a probe set is used for only one genome and vice versa,
but this is not a requirement.</p>

<p>Example:</p>

<pre>    get_relationship_ProducedResultsFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProbeSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Produces"
>get_relationship_Produces</a></h2>

<p>This relationship connects a peg feature to the protein sequence it produces (if any).
Only peg features participate in this relationship.
A single protein sequence will frequently be produced by many features.</p>

<p>Example:</p>

<pre>    get_relationship_Produces -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_ProjectsOnto"
>get_relationship_ProjectsOnto</a></h2>

<p>This relationship connects two protein sequences for which a clear bidirectional best hit exists in known genomes.
The attributes of the relationship describe how good the relationship is between the proteins.
The relationship is bidirectional and symmetric,
but is only stored in one direction (lower ID to higher ID).</p>

<p>Example:</p>

<pre>    get_relationship_ProjectsOnto -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="gene_context"
>gene_context</a></dt>

<dd>
<dt><a name="percent_identity"
>percent_identity</a></dt>

<dd>
<dt><a name="score"
>score</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Provided"
>get_relationship_Provided</a></h2>

<p>This relationship connects a source (core) database to the subsystems it submitted to the knowledge base.</p>

<p>Example:</p>

<pre>    get_relationship_Provided -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_ReflectsStateOf"
>get_relationship_ReflectsStateOf</a></h2>

<p>This relationship connects a genome to the atomic regulons that describe its state.</p>

<p>Example:</p>

<pre>    get_relationship_ReflectsStateOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the AtomicRegulon entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Requires"
>get_relationship_Requires</a></h2>

<p>This relationship links a reaction to the way it is used in a model.</p>

<p>Example:</p>

<pre>    get_relationship_Requires -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="equation"
>equation</a></dt>

<dd>
<dt><a name="reversibility"
>reversibility</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Requirement entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>

<dd>
<dt><a name="proton"
>proton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_RunsByDefaultIn"
>get_relationship_RunsByDefaultIn</a></h2>

<p>This relationship connects a reaction to the compartment in which it runs by default.</p>

<p>Example:</p>

<pre>    get_relationship_RunsByDefaultIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compartment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="equation"
>equation</a></dt>

<dd>
<dt><a name="reversibility"
>reversibility</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Shows"
>get_relationship_Shows</a></h2>

<p>This relationship indicates that a compound appears on a particular diagram.
The same compound can appear on many diagrams,
and a diagram always contains many compounds.</p>

<p>Example:</p>

<pre>    get_relationship_Shows -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Diagram entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="location"
>location</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="uncharged_formula"
>uncharged_formula</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Submitted"
>get_relationship_Submitted</a></h2>

<p>This relationship connects a genome to the core database from which it was loaded.</p>

<p>Example:</p>

<pre>    get_relationship_Submitted -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Targets"
>get_relationship_Targets</a></h2>

<p>This relationship connects a compound in a biomass to the compartment in which it is supposed to appear.</p>

<p>Example:</p>

<pre>    get_relationship_Targets -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ModelCompartment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the BiomassCompound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="coefficient"
>coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Triggers"
>get_relationship_Triggers</a></h2>

<p>A complex can be triggered by many roles.
A role can trigger many complexes.</p>

<p>Example:</p>

<pre>    get_relationship_Triggers -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Complex entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="optional"
>optional</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Uses"
>get_relationship_Uses</a></h2>

<p>This relationship connects a genome to the machines that form its metabolic pathways.
A genome can use many machines,
but a machine is used by exactly one genome.</p>

<p>Example:</p>

<pre>    get_relationship_Uses -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="curated"
>curated</a></dt>

<dd>
<dt><a name="region"
>region</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_UsesAliasForCompound"
>get_relationship_UsesAliasForCompound</a></h2>

<p>This relationship connects a source (database or organization) with the compounds for which it has assigned names (aliases).
The alias itself is stored as intersection data.</p>

<p>Example:</p>

<pre>    get_relationship_UsesAliasForCompound -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="uncharged_formula"
>uncharged_formula</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_UsesAliasForReaction"
>get_relationship_UsesAliasForReaction</a></h2>

<p>This relationship connects a source (database or organization) with the reactions for which it has assigned names (aliases).
The alias itself is stored as intersection data.</p>

<p>Example:</p>

<pre>    get_relationship_UsesAliasForReaction -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="equation"
>equation</a></dt>

<dd>
<dt><a name="reversibility"
>reversibility</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_Validates"
>get_relationship_Validates</a></h2>

<p>This relationship connects a scenario to its subsystem it validates.
A scenario belongs to exactly one subsystem,
but a subsystem may have multiple scenarios.</p>

<p>Example:</p>

<pre>    get_relationship_Validates -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Scenario entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="common_name"
>common_name</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_WasFormulatedBy"
>get_relationship_WasFormulatedBy</a></h2>

<p>This relationship connects a coregulated set to the source organization that originally computed it.</p>

<p>Example:</p>

<pre>    get_relationship_WasFormulatedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the CoregulatedSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_WasGeneratedFrom"
>get_relationship_WasGeneratedFrom</a></h2>

<p>This relationship connects an atomic regulon to a probe set from which experimental data was produced for its features.
It contains a vector of the expression levels.</p>

<p>Example:</p>

<pre>    get_relationship_WasGeneratedFrom -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProbeSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="level_vector"
>level_vector</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the AtomicRegulon entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_WasImportedFrom"
>get_relationship_WasImportedFrom</a></h2>

<p>This relationship specifies the import source for identifiers.
It is used when reloading identifiers to provide for rapid deletion of the previous load&#39;s results.</p>

<p>Example:</p>

<pre>    get_relationship_WasImportedFrom -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Identifier entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>

<dd>
<dt><a name="natural_form"
>natural_form</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_WasProvidedBy"
>get_relationship_WasProvidedBy</a></h2>

<p>This relationship connects a source (core) database to the subsystems it submitted to the knowledge base.</p>

<p>Example:</p>

<pre>    get_relationship_WasProvidedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="get_relationship_WasSubmittedBy"
>get_relationship_WasSubmittedBy</a></h2>

<p>This relationship connects a genome to the core database from which it was loaded.</p>

<p>Example:</p>

<pre>    get_relationship_WasSubmittedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing id is not the last.</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added for each requested field. Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_AlignmentTree"
>query_entity_AlignmentTree</a></h2>

<p>Query the entity AlignmentTree.</p>

<p>An alignment arranges a group of protein sequences so that they match.
Each alignment is associated with a phylogenetic tree that describes how the sequences developed and their evolutionary distance.
The actual tree and alignment FASTA are stored in separate flat files.
The Kbase will maintain a set of alignments and associated trees.
The majority of these will be based on protein sequences.
We will not have a comprehensive set but we will have tens of thousands of such alignments,
and we view them as an imporant resource to support annotation.
The alignments/trees will include the tools and parameters used to construct them.
Access to the underlying sequences and trees in a form convenient to existing tools will be supported.</p>

<p>Example:</p>

<pre>    query_entity_AlignmentTree -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AlignmentTree entity has the following relationship links:</p>

<dl>
<dt><a name="Aligns_ProteinSequence"
>Aligns ProteinSequence</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="alignment_method"
>alignment_method</a></dt>

<dd>
<dt><a name="alignment_parameters"
>alignment_parameters</a></dt>

<dd>
<dt><a name="alignment_properties"
>alignment_properties</a></dt>

<dd>
<dt><a name="tree_method"
>tree_method</a></dt>

<dd>
<dt><a name="tree_parameters"
>tree_parameters</a></dt>

<dd>
<dt><a name="tree_properties"
>tree_properties</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Annotation"
>query_entity_Annotation</a></h2>

<p>Query the entity Annotation.</p>

<p>An annotation is a comment attached to a feature.
Annotations are used to track the history of a feature&#39;s functional assignments and any related issues.
The key is the feature ID followed by a colon and a complemented ten-digit sequence number.</p>

<p>Example:</p>

<pre>    query_entity_Annotation -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Annotation entity has the following relationship links:</p>

<dl>
<dt><a name="Annotates_Feature"
>Annotates Feature</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="annotator"
>annotator</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>

<dd>
<dt><a name="annotation_time"
>annotation_time</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_AtomicRegulon"
>query_entity_AtomicRegulon</a></h2>

<p>Query the entity AtomicRegulon.</p>

<p>An atomic regulon is an indivisible group of coregulated features on a single genome.
Atomic regulons are constructed so that a given feature can only belong to one.
Because of this,
the expression levels for atomic regulons represent in some sense the state of a cell.
An atomicRegulon is a set of protein-encoding genes that are believed to have identical expression profiles (i.e.,
they will all be expressed or none will be expressed in the vast majority of conditions).
These are sometimes referred to as &#34;atomic regulons&#34;.
Note that there are more common notions of &#34;coregulated set of genes&#34; based on the notion that a single regulatory mechanism impacts an entire set of genes.
Since multiple other mechanisms may impact overlapping sets,
the genes impacted by a regulatory mechanism need not all share the same expression profile.
We use a distinct notion (CoregulatedSet) to reference sets of genes impacted by a single regulatory mechanism (i.e.,
by a single transcription regulator).</p>

<p>Example:</p>

<pre>    query_entity_AtomicRegulon -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AtomicRegulon entity has the following relationship links:</p>

<dl>
<dt><a name="IsAffectedIn_Experiment"
>IsAffectedIn Experiment</a></dt>

<dd>
<dt><a name="IsFormedOf_Feature"
>IsFormedOf Feature</a></dt>

<dd>
<dt><a name="ReflectsStateOf_Genome"
>ReflectsStateOf Genome</a></dt>

<dd>
<dt><a name="WasGeneratedFrom_ProbeSet"
>WasGeneratedFrom ProbeSet</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Attribute"
>query_entity_Attribute</a></h2>

<p>Query the entity Attribute.</p>

<p>An attribute describes a category of condition or characteristic for an experiment.
The goals of the experiment can be inferred from its values for all the attributes of interest.</p>

<p>Example:</p>

<pre>    query_entity_Attribute -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Attribute entity has the following relationship links:</p>

<dl>
<dt><a name="HasValueIn_Experiment"
>HasValueIn Experiment</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="description"
>description</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Biomass"
>query_entity_Biomass</a></h2>

<p>Query the entity Biomass.</p>

<p>A biomass is a collection of compounds in a specific ratio and in specific compartments that are necessary for a cell to function properly.
The prediction of biomasses is key to the functioning of the model.</p>

<p>Example:</p>

<pre>    query_entity_Biomass -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Biomass entity has the following relationship links:</p>

<dl>
<dt><a name="IsComprisedOf_BiomassCompound"
>IsComprisedOf BiomassCompound</a></dt>

<dd>
<dt><a name="IsManagedBy_Model"
>IsManagedBy Model</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_BiomassCompound"
>query_entity_BiomassCompound</a></h2>

<p>Query the entity BiomassCompound.</p>

<p>A Biomass Compound represents the occurrence of a particular compound in a biomass.</p>

<p>Example:</p>

<pre>    query_entity_BiomassCompound -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The BiomassCompound entity has the following relationship links:</p>

<dl>
<dt><a name="Comprises_Biomass"
>Comprises Biomass</a></dt>

<dd>
<dt><a name="IsUsageOf_Compound"
>IsUsageOf Compound</a></dt>

<dd>
<dt><a name="Targets_ModelCompartment"
>Targets ModelCompartment</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="coefficient"
>coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Compartment"
>query_entity_Compartment</a></h2>

<p>Query the entity Compartment.</p>

<p>A compartment is a section of a single model that represents the environment in which a reaction takes place (e.g.
cell wall).</p>

<p>Example:</p>

<pre>    query_entity_Compartment -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Compartment entity has the following relationship links:</p>

<dl>
<dt><a name="IsDefaultFor_Reaction"
>IsDefaultFor Reaction</a></dt>

<dd>
<dt><a name="IsDefaultLocationOf_Reagent"
>IsDefaultLocationOf Reagent</a></dt>

<dd>
<dt><a name="IsInstantiatedBy_ModelCompartment"
>IsInstantiatedBy ModelCompartment</a></dt>

<dd>
<dt><a name="IsProposedLocationOf_ReactionRule"
>IsProposedLocationOf ReactionRule</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Complex"
>query_entity_Complex</a></h2>

<p>Query the entity Complex.</p>

<p>A complex is a set of chemical reactions that act in concert to effect a role.</p>

<p>Example:</p>

<pre>    query_entity_Complex -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Complex entity has the following relationship links:</p>

<dl>
<dt><a name="HasStep_ReactionRule"
>HasStep ReactionRule</a></dt>

<dd>
<dt><a name="IsTriggeredBy_Role"
>IsTriggeredBy Role</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Compound"
>query_entity_Compound</a></h2>

<p>Query the entity Compound.</p>

<p>A compound is a chemical that participates in a reaction.
Both ligands and reaction components are treated as compounds.</p>

<p>Example:</p>

<pre>    query_entity_Compound -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Compound entity has the following relationship links:</p>

<dl>
<dt><a name="HasUsage_BiomassCompound"
>HasUsage BiomassCompound</a></dt>

<dd>
<dt><a name="IsPresentIn_Media"
>IsPresentIn Media</a></dt>

<dd>
<dt><a name="IsShownOn_Diagram"
>IsShownOn Diagram</a></dt>

<dd>
<dt><a name="IsTerminusFor_Scenario"
>IsTerminusFor Scenario</a></dt>

<dd>
<dt><a name="ParticipatesAs_Reagent"
>ParticipatesAs Reagent</a></dt>

<dd>
<dt><a name="UsesAliasForCompound_Source"
>UsesAliasForCompound Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="uncharged_formula"
>uncharged_formula</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Contig"
>query_entity_Contig</a></h2>

<p>Query the entity Contig.</p>

<p>A contig is thought of as composing a part of the DNA associated with a specific genome.
It is represented as an ID (including the genome ID) and a ContigSequence.
We do not think of strings of DNA from,
say,
a metgenomic sample as &#34;contigs&#34;,
since there is no associated genome (these would be considered ContigSequences).
This use of the term &#34;ContigSequence&#34;,
rather than just &#34;DNA sequence&#34;,
may turn out to be a bad idea.
For now,
you should just realize that a Contig has an associated genome,
but a ContigSequence does not.</p>

<p>Example:</p>

<pre>    query_entity_Contig -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Contig entity has the following relationship links:</p>

<dl>
<dt><a name="HasAsSequence_ContigSequence"
>HasAsSequence ContigSequence</a></dt>

<dd>
<dt><a name="IsAlignedIn_Variation"
>IsAlignedIn Variation</a></dt>

<dd>
<dt><a name="IsComponentOf_Genome"
>IsComponentOf Genome</a></dt>

<dd>
<dt><a name="IsLocusFor_Feature"
>IsLocusFor Feature</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_ContigChunk"
>query_entity_ContigChunk</a></h2>

<p>Query the entity ContigChunk.</p>

<p>ContigChunks are strings of DNA thought of as being a string in a 4-character alphabet with an associated ID.
We allow a broader alphabet that includes U (for RNA) and the standard ambiguity characters.
The notion of ContigChunk was introduced to avoid transferring/manipulating huge contigs to access small substrings.
A ContigSequence is formed by concatenating a set of one or more ContigChunks.
Thus,
ContigChunks are the basic units moved from the database to memory.
Their existence should be hidden from users in most circumstances (users are expected to request substrings of ContigSequences,
and the Kbase software locates the appropriate ContigChunks).</p>

<p>Example:</p>

<pre>    query_entity_ContigChunk -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ContigChunk entity has the following relationship links:</p>

<dl>
<dt><a name="IsSectionOf_ContigSequence"
>IsSectionOf ContigSequence</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_ContigSequence"
>query_entity_ContigSequence</a></h2>

<p>Query the entity ContigSequence.</p>

<p>ContigSequences are strings of DNA.
Contigs have an associated genome,
but ContigSequences do not..
We can think of random samples of DNA as a set of ContigSequences.
There are no length constraints imposed on ContigSequences -- they can be either very short or very long.
The basic unit of data that is moved to/from the database is the ContigChunk,
from which ContigSequences are formed.
The key of a ContigSequence is the sequence&#39;s MD5 identifier.</p>

<p>Example:</p>

<pre>    query_entity_ContigSequence -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ContigSequence entity has the following relationship links:</p>

<dl>
<dt><a name="HasSection_ContigChunk"
>HasSection ContigChunk</a></dt>

<dd>
<dt><a name="IsSequenceOf_Contig"
>IsSequenceOf Contig</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="length"
>length</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_CoregulatedSet"
>query_entity_CoregulatedSet</a></h2>

<p>Query the entity CoregulatedSet.</p>

<p>We need to represent sets of genes that are coregulated via some regulatory mechanism.
In particular,
we wish to represent genes that are coregulated using transcription binding sites and corresponding transcription regulatory proteins.
We represent a coregulated set (which may,
or may not,
be considered an atomic regulon) using CoregulatedSet.</p>

<p>Example:</p>

<pre>    query_entity_CoregulatedSet -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The CoregulatedSet entity has the following relationship links:</p>

<dl>
<dt><a name="IsBoundBy_Feature"
>IsBoundBy Feature</a></dt>

<dd>
<dt><a name="IsControlledUsing_Feature"
>IsControlledUsing Feature</a></dt>

<dd>
<dt><a name="IsRegulatedSetOf_Feature"
>IsRegulatedSetOf Feature</a></dt>

<dd>
<dt><a name="WasFormulatedBy_Source"
>WasFormulatedBy Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Diagram"
>query_entity_Diagram</a></h2>

<p>Query the entity Diagram.</p>

<p>A functional diagram describes a network of chemical reactions,
often comprising a single subsystem.</p>

<p>Example:</p>

<pre>    query_entity_Diagram -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Diagram entity has the following relationship links:</p>

<dl>
<dt><a name="Displays_Reaction"
>Displays Reaction</a></dt>

<dd>
<dt><a name="IncludesPartOf_Scenario"
>IncludesPartOf Scenario</a></dt>

<dd>
<dt><a name="IsRelevantFor_Subsystem"
>IsRelevantFor Subsystem</a></dt>

<dd>
<dt><a name="Shows_Compound"
>Shows Compound</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_EcNumber"
>query_entity_EcNumber</a></h2>

<p>Query the entity EcNumber.</p>

<p>EC numbers are assigned by the Enzyme Commission,
and consist of four numbers separated by periods,
each indicating a successively smaller cateogry of enzymes.</p>

<p>Example:</p>

<pre>    query_entity_EcNumber -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The EcNumber entity has the following relationship links:</p>

<dl>
<dt><a name="IsConsistentWith_Role"
>IsConsistentWith Role</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="obsolete"
>obsolete</a></dt>

<dd>
<dt><a name="replacedby"
>replacedby</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Experiment"
>query_entity_Experiment</a></h2>

<p>Query the entity Experiment.</p>

<p>An experiment is a combination of conditions for which gene expression information is desired.
The result of the experiment is a set of expression levels for features under the given conditions.</p>

<p>Example:</p>

<pre>    query_entity_Experiment -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Experiment entity has the following relationship links:</p>

<dl>
<dt><a name="AffectsLevelOf_AtomicRegulon"
>AffectsLevelOf AtomicRegulon</a></dt>

<dd>
<dt><a name="HasResultsFor_ProbeSet"
>HasResultsFor ProbeSet</a></dt>

<dd>
<dt><a name="HasValueFor_Attribute"
>HasValueFor Attribute</a></dt>

<dd>
<dt><a name="IndicatesSignalFor_Feature"
>IndicatesSignalFor Feature</a></dt>

<dd>
<dt><a name="OperatesIn_Media"
>OperatesIn Media</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source"
>source</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Family"
>query_entity_Family</a></h2>

<p>Query the entity Family.</p>

<p>The Kbase will support the maintenance of protein families (as sets of Features with associated translations).
We are initially only supporting the notion of a family as composed of a set of isofunctional homologs.
That is,
the families we initially support should be thought of as containing protein-encoding genes whose associated sequences all implement the same function (we do understand that the notion of &#34;function&#34; is somewhat ambiguous,
so let us sweep this under the rug by calling a functional role a &#34;primitive concept&#34;).
We currently support families in which the members are translations of features,
and we think of Features as having an associated function.
Identical protein sequences as products of translating distinct genes may or may not have identical functions,
and we allow multiple members of the same Family to share identical protein sequences.
This may be justified,
since in a very,
very,
very few cases identical proteins do,
in fact,
have distinct functions.
We would prefer to reach the point where our Families are sets of protein sequence,
rather than sets of protein-encoding Features.</p>

<p>Example:</p>

<pre>    query_entity_Family -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Family entity has the following relationship links:</p>

<dl>
<dt><a name="HasMember_Feature"
>HasMember Feature</a></dt>

<dd>
<dt><a name="IsCoupledTo_Family"
>IsCoupledTo Family</a></dt>

<dd>
<dt><a name="IsCoupledWith_Family"
>IsCoupledWith Family</a></dt>

<dd>
<dt><a name="IsFamilyFor_Role"
>IsFamilyFor Role</a></dt>

<dd>
<dt><a name="IsRepresentedIn_Genome"
>IsRepresentedIn Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Feature"
>query_entity_Feature</a></h2>

<p>Query the entity Feature.</p>

<p>A feature (sometimes also called a gene) is a part of a genome that is of special interest.
Features may be spread across multiple DNA sequences (contigs) of a genome,
but never across more than one genome.
Each feature in the database has a unique ID that functions as its ID in this table.
Normally a Feature is just a single contigous region on a contig.
Features have types,
and an appropriate choice of available types allows the support of protein-encoding genes,
exons,
RNA genes,
binding sites,
pathogenicity islands,
or whatever.</p>

<p>Example:</p>

<pre>    query_entity_Feature -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Feature entity has the following relationship links:</p>

<dl>
<dt><a name="Controls_CoregulatedSet"
>Controls CoregulatedSet</a></dt>

<dd>
<dt><a name="Encompasses_Feature"
>Encompasses Feature</a></dt>

<dd>
<dt><a name="HasCoregulationWith_Feature"
>HasCoregulationWith Feature</a></dt>

<dd>
<dt><a name="HasFunctional_Role"
>HasFunctional Role</a></dt>

<dd>
<dt><a name="HasIndicatedSignalFrom_Experiment"
>HasIndicatedSignalFrom Experiment</a></dt>

<dd>
<dt><a name="HasLevelsFrom_ProbeSet"
>HasLevelsFrom ProbeSet</a></dt>

<dd>
<dt><a name="IsAnnotatedBy_Annotation"
>IsAnnotatedBy Annotation</a></dt>

<dd>
<dt><a name="IsBindingSiteFor_CoregulatedSet"
>IsBindingSiteFor CoregulatedSet</a></dt>

<dd>
<dt><a name="IsContainedIn_SSCell"
>IsContainedIn SSCell</a></dt>

<dd>
<dt><a name="IsCoregulatedWith_Feature"
>IsCoregulatedWith Feature</a></dt>

<dd>
<dt><a name="IsEncompassedIn_Feature"
>IsEncompassedIn Feature</a></dt>

<dd>
<dt><a name="IsExemplarOf_Role"
>IsExemplarOf Role</a></dt>

<dd>
<dt><a name="IsFormedInto_AtomicRegulon"
>IsFormedInto AtomicRegulon</a></dt>

<dd>
<dt><a name="IsInPair_Pairing"
>IsInPair Pairing</a></dt>

<dd>
<dt><a name="IsLocatedIn_Contig"
>IsLocatedIn Contig</a></dt>

<dd>
<dt><a name="IsMemberOf_Family"
>IsMemberOf Family</a></dt>

<dd>
<dt><a name="IsOwnedBy_Genome"
>IsOwnedBy Genome</a></dt>

<dd>
<dt><a name="IsRegulatedIn_CoregulatedSet"
>IsRegulatedIn CoregulatedSet</a></dt>

<dd>
<dt><a name="Produces_ProteinSequence"
>Produces ProteinSequence</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Genome"
>query_entity_Genome</a></h2>

<p>Query the entity Genome.</p>

<p>The Kbase houses a large and growing set of genomes.
We often have multiple genomes that have identical DNA.
These usually have distinct gene calls and annotations,
but not always.
We consider the Kbase to be a framework for managing hundreds of thousands of genomes and offering the tools needed to support compartive analysis on large sets of genomes,
some of which are virtually identical.
Each genome has an MD5 value computed from the DNA that is associated with the genome.
Hence,
it is easy to recognize when you have identical genomes,
perhaps annotated by distinct groups.</p>

<p>Example:</p>

<pre>    query_entity_Genome -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Genome entity has the following relationship links:</p>

<dl>
<dt><a name="HadResultsProducedBy_ProbeSet"
>HadResultsProducedBy ProbeSet</a></dt>

<dd>
<dt><a name="HasRepresentativeOf_Family"
>HasRepresentativeOf Family</a></dt>

<dd>
<dt><a name="IsCollectedInto_OTU"
>IsCollectedInto OTU</a></dt>

<dd>
<dt><a name="IsComposedOf_Contig"
>IsComposedOf Contig</a></dt>

<dd>
<dt><a name="IsConfiguredBy_AtomicRegulon"
>IsConfiguredBy AtomicRegulon</a></dt>

<dd>
<dt><a name="IsInTaxa_TaxonomicGrouping"
>IsInTaxa TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsModeledBy_Model"
>IsModeledBy Model</a></dt>

<dd>
<dt><a name="IsOwnerOf_Feature"
>IsOwnerOf Feature</a></dt>

<dd>
<dt><a name="Uses_SSRow"
>Uses SSRow</a></dt>

<dd>
<dt><a name="WasSubmittedBy_Source"
>WasSubmittedBy Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Identifier"
>query_entity_Identifier</a></h2>

<p>Query the entity Identifier.</p>

<p>An identifier is an alternate name for a protein sequence.
The identifier is typically stored in a prefixed form that indicates the database it came from.</p>

<p>Example:</p>

<pre>    query_entity_Identifier -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Identifier entity has the following relationship links:</p>

<dl>
<dt><a name="HasAssertionFrom_Source"
>HasAssertionFrom Source</a></dt>

<dd>
<dt><a name="Names_ProteinSequence"
>Names ProteinSequence</a></dt>

<dd>
<dt><a name="WasImportedFrom_Source"
>WasImportedFrom Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source"
>source</a></dt>

<dd>
<dt><a name="natural_form"
>natural_form</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Media"
>query_entity_Media</a></h2>

<p>Query the entity Media.</p>

<p>A media describes the chemical content of the solution in which cells are grown in an experiment or for the purposes of a model.
The key is the common media name.
The nature of the media is described by its relationship to its constituent compounds.</p>

<p>Example:</p>

<pre>    query_entity_Media -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Media entity has the following relationship links:</p>

<dl>
<dt><a name="HasPresenceOf_Compound"
>HasPresenceOf Compound</a></dt>

<dd>
<dt><a name="IsUtilizedIn_Experiment"
>IsUtilizedIn Experiment</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Model"
>query_entity_Model</a></h2>

<p>Query the entity Model.</p>

<p>A model specifies a relationship between sets of features and reactions in a cell.
It is used to simulate cell growth and gene knockouts to validate annotations.</p>

<p>Example:</p>

<pre>    query_entity_Model -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Model entity has the following relationship links:</p>

<dl>
<dt><a name="IsARequirementIn_Requirement"
>IsARequirementIn Requirement</a></dt>

<dd>
<dt><a name="IsDividedInto_ModelCompartment"
>IsDividedInto ModelCompartment</a></dt>

<dd>
<dt><a name="Manages_Biomass"
>Manages Biomass</a></dt>

<dd>
<dt><a name="Models_Genome"
>Models Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_ModelCompartment"
>query_entity_ModelCompartment</a></h2>

<p>Query the entity ModelCompartment.</p>

<p>The Model Compartment represents a section of a cell (e.g.
cell wall,
cytoplasm) as it appears in a specific model.</p>

<p>Example:</p>

<pre>    query_entity_ModelCompartment -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ModelCompartment entity has the following relationship links:</p>

<dl>
<dt><a name="IsDivisionOf_Model"
>IsDivisionOf Model</a></dt>

<dd>
<dt><a name="IsInstanceOf_Compartment"
>IsInstanceOf Compartment</a></dt>

<dd>
<dt><a name="IsRealLocationOf_Requirement"
>IsRealLocationOf Requirement</a></dt>

<dd>
<dt><a name="IsTargetOf_BiomassCompound"
>IsTargetOf BiomassCompound</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_OTU"
>query_entity_OTU</a></h2>

<p>Query the entity OTU.</p>

<p>An OTU (Organism Taxonomic Unit) is a named group of related genomes.</p>

<p>Example:</p>

<pre>    query_entity_OTU -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The OTU entity has the following relationship links:</p>

<dl>
<dt><a name="IsCollectionOf_Genome"
>IsCollectionOf Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Pairing"
>query_entity_Pairing</a></h2>

<p>Query the entity Pairing.</p>

<p>A pairing indicates that two features are found close together in a genome.
Not all possible pairings are stored in the database; only those that are considered for some reason to be significant for annotation purposes.The key of the pairing is the concatenation of the feature IDs in alphabetical order with an intervening colon.</p>

<p>Example:</p>

<pre>    query_entity_Pairing -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Pairing entity has the following relationship links:</p>

<dl>
<dt><a name="Determines_PairSet"
>Determines PairSet</a></dt>

<dd>
<dt><a name="IsPairOf_Feature"
>IsPairOf Feature</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_PairSet"
>query_entity_PairSet</a></h2>

<p>Query the entity PairSet.</p>

<p>A PairSet is a precompute set of pairs or genes.
Each pair occurs close to one another of the chromosome.
We believe that all of the first members of the pairs correspond to one another (are quite similar),
as do all of the second members of the pairs.
These pairs (from prokaryotic genomes) offer on of the most powerful clues relating to uncharacterized genes/peroteins.</p>

<p>Example:</p>

<pre>    query_entity_PairSet -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The PairSet entity has the following relationship links:</p>

<dl>
<dt><a name="IsDeterminedBy_Pairing"
>IsDeterminedBy Pairing</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="score"
>score</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_ProbeSet"
>query_entity_ProbeSet</a></h2>

<p>Query the entity ProbeSet.</p>

<p>A probe set is a device containing multiple probe sequences for use in gene expression experiments.</p>

<p>Example:</p>

<pre>    query_entity_ProbeSet -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ProbeSet entity has the following relationship links:</p>

<dl>
<dt><a name="GeneratedLevelsFor_AtomicRegulon"
>GeneratedLevelsFor AtomicRegulon</a></dt>

<dd>
<dt><a name="HasResultsIn_Experiment"
>HasResultsIn Experiment</a></dt>

<dd>
<dt><a name="IndicatedLevelsFor_Feature"
>IndicatedLevelsFor Feature</a></dt>

<dd>
<dt><a name="ProducedResultsFor_Genome"
>ProducedResultsFor Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_ProteinSequence"
>query_entity_ProteinSequence</a></h2>

<p>Query the entity ProteinSequence.</p>

<p>We use the concept of ProteinSequence as an amino acid string with an associated MD5 value.
It is easy to access the set of Features that relate to a ProteinSequence.
While function is still associated with Features (and may be for some time),
publications are associated with ProteinSequences (and the inferred impact on Features is through the relationship connecting ProteinSequences to Features).</p>

<p>Example:</p>

<pre>    query_entity_ProteinSequence -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ProteinSequence entity has the following relationship links:</p>

<dl>
<dt><a name="IsATopicOf_Publication"
>IsATopicOf Publication</a></dt>

<dd>
<dt><a name="IsAlignedBy_AlignmentTree"
>IsAlignedBy AlignmentTree</a></dt>

<dd>
<dt><a name="IsNamedBy_Identifier"
>IsNamedBy Identifier</a></dt>

<dd>
<dt><a name="IsProjectedOnto_ProteinSequence"
>IsProjectedOnto ProteinSequence</a></dt>

<dd>
<dt><a name="IsProteinFor_Feature"
>IsProteinFor Feature</a></dt>

<dd>
<dt><a name="ProjectsOnto_ProteinSequence"
>ProjectsOnto ProteinSequence</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Publication"
>query_entity_Publication</a></h2>

<p>Query the entity Publication.</p>

<p>Annotators attach publications to ProteinSequences.
The criteria we have used to gather such connections is a bit nonstandard.
We have sought to attach publications to ProteinSequences when the publication includes an expert asserting a belief or estimate of function.
The paper may not be the original characterization.
Further,
it may not even discuss a sequence protein (much of the literature is very valuable,
but reports work on proteins in strains that have not yet been sequenced).
On the other hand,
reports of sequencing regions of a chromosome (with no specific assertion of a clear function) should not be attached.
The attached publications give an ID (usually a Pubmed ID),
a URL to the paper (when we have it),
and a title (when we have it).</p>

<p>Example:</p>

<pre>    query_entity_Publication -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Publication entity has the following relationship links:</p>

<dl>
<dt><a name="Concerns_ProteinSequence"
>Concerns ProteinSequence</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="link"
>link</a></dt>

<dd>
<dt><a name="pubdate"
>pubdate</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Reaction"
>query_entity_Reaction</a></h2>

<p>Query the entity Reaction.</p>

<p>A reaction is a chemical process that converts one set of compounds (substrate) to another set (products).</p>

<p>Example:</p>

<pre>    query_entity_Reaction -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Reaction entity has the following relationship links:</p>

<dl>
<dt><a name="Involves_Reagent"
>Involves Reagent</a></dt>

<dd>
<dt><a name="IsDisplayedOn_Diagram"
>IsDisplayedOn Diagram</a></dt>

<dd>
<dt><a name="IsRequiredBy_Requirement"
>IsRequiredBy Requirement</a></dt>

<dd>
<dt><a name="IsUsedAs_ReactionRule"
>IsUsedAs ReactionRule</a></dt>

<dd>
<dt><a name="ParticipatesIn_Scenario"
>ParticipatesIn Scenario</a></dt>

<dd>
<dt><a name="RunsByDefaultIn_Compartment"
>RunsByDefaultIn Compartment</a></dt>

<dd>
<dt><a name="UsesAliasForReaction_Source"
>UsesAliasForReaction Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="msid"
>msid</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="equation"
>equation</a></dt>

<dd>
<dt><a name="reversibility"
>reversibility</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_ReactionRule"
>query_entity_ReactionRule</a></h2>

<p>Query the entity ReactionRule.</p>

<p>A reaction rule represents the way a reaction takes place within the context of a model.</p>

<p>Example:</p>

<pre>    query_entity_ReactionRule -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ReactionRule entity has the following relationship links:</p>

<dl>
<dt><a name="HasProposedLocationIn_Compartment"
>HasProposedLocationIn Compartment</a></dt>

<dd>
<dt><a name="IsStepOf_Complex"
>IsStepOf Complex</a></dt>

<dd>
<dt><a name="IsUseOf_Reaction"
>IsUseOf Reaction</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Reagent"
>query_entity_Reagent</a></h2>

<p>Query the entity Reagent.</p>

<p>This entity represents a compound as it is used by a specific reaction.
A reaction involves many compounds,
and a compound can be involved in many reactions.
The reagent describes the use of the compound by a specific reaction.</p>

<p>Example:</p>

<pre>    query_entity_Reagent -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Reagent entity has the following relationship links:</p>

<dl>
<dt><a name="HasDefaultLocation_Compartment"
>HasDefaultLocation Compartment</a></dt>

<dd>
<dt><a name="IsInvolvedIn_Reaction"
>IsInvolvedIn Reaction</a></dt>

<dd>
<dt><a name="IsParticipationOf_Compound"
>IsParticipationOf Compound</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="stoichiometry"
>stoichiometry</a></dt>

<dd>
<dt><a name="cofactor"
>cofactor</a></dt>

<dd>
<dt><a name="compartment_index"
>compartment_index</a></dt>

<dd>
<dt><a name="transport_coefficient"
>transport_coefficient</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Requirement"
>query_entity_Requirement</a></h2>

<p>Query the entity Requirement.</p>

<p>A requirement describes the way a reaction fits into a model.</p>

<p>Example:</p>

<pre>    query_entity_Requirement -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Requirement entity has the following relationship links:</p>

<dl>
<dt><a name="HasRealLocationIn_ModelCompartment"
>HasRealLocationIn ModelCompartment</a></dt>

<dd>
<dt><a name="IsARequirementOf_Model"
>IsARequirementOf Model</a></dt>

<dd>
<dt><a name="Requires_Reaction"
>Requires Reaction</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="transproton"
>transproton</a></dt>

<dd>
<dt><a name="proton"
>proton</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Role"
>query_entity_Role</a></h2>

<p>Query the entity Role.</p>

<p>A role describes a biological function that may be fulfilled by a feature.
One of the main goals of the database is to assign features to roles.
Most roles are effected by the construction of proteins.
Some,
however,
deal with functional regulation and message transmission.</p>

<p>Example:</p>

<pre>    query_entity_Role -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Role entity has the following relationship links:</p>

<dl>
<dt><a name="DeterminesFunctionOf_Family"
>DeterminesFunctionOf Family</a></dt>

<dd>
<dt><a name="HasAsExemplar_Feature"
>HasAsExemplar Feature</a></dt>

<dd>
<dt><a name="IsConsistentTo_EcNumber"
>IsConsistentTo EcNumber</a></dt>

<dd>
<dt><a name="IsFunctionalIn_Feature"
>IsFunctionalIn Feature</a></dt>

<dd>
<dt><a name="IsIncludedIn_Subsystem"
>IsIncludedIn Subsystem</a></dt>

<dd>
<dt><a name="IsRoleOf_SSCell"
>IsRoleOf SSCell</a></dt>

<dd>
<dt><a name="Triggers_Complex"
>Triggers Complex</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Scenario"
>query_entity_Scenario</a></h2>

<p>Query the entity Scenario.</p>

<p>A scenario is a partial instance of a subsystem with a defined set of reactions.
Each scenario converts input compounds to output compounds using reactions.
The scenario may use all of the reactions controlled by a subsystem or only some,
and may also incorporate additional reactions.
Because scenario names are not unique,
the actual scenario ID is a number.</p>

<p>Example:</p>

<pre>    query_entity_Scenario -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Scenario entity has the following relationship links:</p>

<dl>
<dt><a name="HasAsTerminus_Compound"
>HasAsTerminus Compound</a></dt>

<dd>
<dt><a name="HasParticipant_Reaction"
>HasParticipant Reaction</a></dt>

<dd>
<dt><a name="Overlaps_Diagram"
>Overlaps Diagram</a></dt>

<dd>
<dt><a name="Validates_Subsystem"
>Validates Subsystem</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="common_name"
>common_name</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Source"
>query_entity_Source</a></h2>

<p>Query the entity Source.</p>

<p>A source is a user or organization that is permitted to assign its own identifiers or to submit bioinformatic objects to the database.</p>

<p>Example:</p>

<pre>    query_entity_Source -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Source entity has the following relationship links:</p>

<dl>
<dt><a name="Asserts_Identifier"
>Asserts Identifier</a></dt>

<dd>
<dt><a name="Formulated_CoregulatedSet"
>Formulated CoregulatedSet</a></dt>

<dd>
<dt><a name="HasCompoundAliasFrom_Compound"
>HasCompoundAliasFrom Compound</a></dt>

<dd>
<dt><a name="HasReactionAliasFrom_Reaction"
>HasReactionAliasFrom Reaction</a></dt>

<dd>
<dt><a name="Imported_Identifier"
>Imported Identifier</a></dt>

<dd>
<dt><a name="Provided_Subsystem"
>Provided Subsystem</a></dt>

<dd>
<dt><a name="Submitted_Genome"
>Submitted Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_SSCell"
>query_entity_SSCell</a></h2>

<p>Query the entity SSCell.</p>

<p>An SSCell (SpreadSheet Cell) represents a role as it occurs in a subsystem spreadsheet row.
The key is a colon-delimited triple containing an MD5 hash of the subsystem ID followed by a genome ID (with optional region string) and a role abbreviation.</p>

<p>Example:</p>

<pre>    query_entity_SSCell -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SSCell entity has the following relationship links:</p>

<dl>
<dt><a name="Contains_Feature"
>Contains Feature</a></dt>

<dd>
<dt><a name="HasRole_Role"
>HasRole Role</a></dt>

<dd>
<dt><a name="IsRoleFor_SSRow"
>IsRoleFor SSRow</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_SSRow"
>query_entity_SSRow</a></h2>

<p>Query the entity SSRow.</p>

<p>An SSRow (that is,
a row in a subsystem spreadsheet) represents a collection of functional roles present in the Features of a single Genome.
The roles are part of a designated subsystem,
and the fids associated with each role are included in the row,
That is,
a row amounts to an instance of a subsystem as it exists in a specific,
designated genome.</p>

<p>Example:</p>

<pre>    query_entity_SSRow -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SSRow entity has the following relationship links:</p>

<dl>
<dt><a name="Implements_Variant"
>Implements Variant</a></dt>

<dd>
<dt><a name="IsRowOf_SSCell"
>IsRowOf SSCell</a></dt>

<dd>
<dt><a name="IsUsedBy_Genome"
>IsUsedBy Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="curated"
>curated</a></dt>

<dd>
<dt><a name="region"
>region</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Subsystem"
>query_entity_Subsystem</a></h2>

<p>Query the entity Subsystem.</p>

<p>A subsystem is a set of functional roles that have been annotated simultaneously (e.g.,
the roles present in a specific pathway),
with an associated subsystem spreadsheet which encodes the fids in each genome that implement the functional roles in the subsystem.</p>

<p>Example:</p>

<pre>    query_entity_Subsystem -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Subsystem entity has the following relationship links:</p>

<dl>
<dt><a name="Describes_Variant"
>Describes Variant</a></dt>

<dd>
<dt><a name="Includes_Role"
>Includes Role</a></dt>

<dd>
<dt><a name="IsInClass_SubsystemClass"
>IsInClass SubsystemClass</a></dt>

<dd>
<dt><a name="IsRelevantTo_Diagram"
>IsRelevantTo Diagram</a></dt>

<dd>
<dt><a name="IsSubInstanceOf_Scenario"
>IsSubInstanceOf Scenario</a></dt>

<dd>
<dt><a name="WasProvidedBy_Source"
>WasProvidedBy Source</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_SubsystemClass"
>query_entity_SubsystemClass</a></h2>

<p>Query the entity SubsystemClass.</p>

<p>Subsystem classes impose a hierarchical organization on the subsystems.</p>

<p>Example:</p>

<pre>    query_entity_SubsystemClass -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SubsystemClass entity has the following relationship links:</p>

<dl>
<dt><a name="IsClassFor_Subsystem"
>IsClassFor Subsystem</a></dt>

<dd>
<dt><a name="IsSubclassOf_SubsystemClass"
>IsSubclassOf SubsystemClass</a></dt>

<dd>
<dt><a name="IsSuperclassOf_SubsystemClass"
>IsSuperclassOf SubsystemClass</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_TaxonomicGrouping"
>query_entity_TaxonomicGrouping</a></h2>

<p>Query the entity TaxonomicGrouping.</p>

<p>We associate with most genomes a &#34;taxonomy&#34; based on the NCBI taxonomy.
This includes,
for each genome,
a list of ever larger taxonomic groups.</p>

<p>Example:</p>

<pre>    query_entity_TaxonomicGrouping -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The TaxonomicGrouping entity has the following relationship links:</p>

<dl>
<dt><a name="IsGroupFor_TaxonomicGrouping"
>IsGroupFor TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsInGroup_TaxonomicGrouping"
>IsInGroup TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsTaxonomyOf_Genome"
>IsTaxonomyOf Genome</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Variant"
>query_entity_Variant</a></h2>

<p>Query the entity Variant.</p>

<p>Each subsystem may include the designation of distinct variants.
Thus,
there may be three closely-related,
but distinguishable forms of histidine degradation.
Each form would be called a &#34;variant&#34;,
with an associated code,
and all genomes implementing a specific variant can easily be accessed.</p>

<p>Example:</p>

<pre>    query_entity_Variant -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Variant entity has the following relationship links:</p>

<dl>
<dt><a name="IsDescribedBy_Subsystem"
>IsDescribedBy Subsystem</a></dt>

<dd>
<dt><a name="IsImplementedBy_SSRow"
>IsImplementedBy SSRow</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="role_rule"
>role_rule</a></dt>

<dd>
<dt><a name="code"
>code</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="query_entity_Variation"
>query_entity_Variation</a></h2>

<p>Query the entity Variation.</p>

<p>A variation describes a set of aligned regions in two or more contigs.</p>

<p>Example:</p>

<pre>    query_entity_Variation -a </pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Variation entity has the following relationship links:</p>

<dl>
<dt><a name="IsAlignmentFor_Contig"
>IsAlignmentFor Contig</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-is_field,value"
>-is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="-like_field,value"
>-like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="-op_operator,field,value"
>-op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text base alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="-a"
>-a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="notes"
>notes</a></dt>
</dl>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file containing a column for each requested field.</p>
<hr>
