<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="all_roles_used_in_models"
>all_roles_used_in_models</a></h2>

<p>The all_roles_used_in_models allows a user to access the set of roles that are included in current models.
This is important.
There are far fewer roles used in models than overall.
Hence,
the returned set represents the minimal set we need to clean up in order to properly support modeling.</p>

<p>Example:</p>

<pre>    all_roles_used_in_models &#62; output</pre>

<p>This is a pipe command. The is no input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call all_roles_used_in_models. It is documented as follows:</p>

<pre>  $return = $obj-&#62;all_roles_used_in_models()</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$return is a roles
roles is a reference to a list where each element is a role
role is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The output is a file in which each line contains just a role used in the construction of models.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="atomic_regulons_to_fids"
>atomic_regulons_to_fids</a></h2>

<p>Example:</p>

<pre>    atomic_regulons_to_fids [arguments] &#60; input &#62; output</pre>

<p>Atomic_regulons are sets of fids that are believed to have identical expression profiles. The estmates were made using expression data, operon estimates, and subsystems.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call atomic_regulons_to_fids. It is documented as follows:</p>

<pre>  $return = $obj-&#62;atomic_regulons_to_fids($regulons)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$regulons is a regulons
$return is a reference to a hash where the key is a regulon and the value is a fids
regulons is a reference to a list where each element is a regulon
regulon is a string
fids is a reference to a list where each element is a fid
fid is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="close_genomes"
>close_genomes</a></h2>

<p>Example:</p>

<pre>    close_genomes [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the identifier.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call close_genomes. It is documented as follows:</p>

<pre>  $return = $obj-&#62;close_genomes($genomes, $how, $n)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$genomes is a genomes
$how is a how
$n is an int
$return is a reference to a hash where the key is a genome and the value is a genomes
genomes is a reference to a list where each element is a genome
genome is a string
how is an int


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the identifier is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>

<dd>
<dt><a name="-n_N_[_N_is_the_number_of_close_genomes_desired_]"
>-n N [ N is the number of close genomes desired ]</a></dt>

<dd>
<dt><a 
>-how Algorithm [ optional: 0 -&#62; default, 1 -&#62; just match scientific names of genomes 2 -&#62; use SSU rRNA (not yet implemented) 3 -&#62; use &#34;universal proteins&#34; (not yet implemented) ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added (a close genome)</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="co_occurrence_evidence"
>co_occurrence_evidence</a></h2>

<p>co-occurence_evidence is used to retrieve the detailed pairs of genes that go into the computation of co-occurence scores.
The scores reflect an estimate of the number of distinct OTUs that contain an instance of a co-occuring pair.
This routine returns as evidence a list of all the pairs that went into the computation.</p>

<p>The input to the computation is table in which one of the columns contains co-occurring pairs.
For example,</p>

<pre>     echo &#39;kb|g.0.peg.101:kb|g.0.peg.263&#39; | co_occurrence_evidence </pre>

<p>takes in a table with 1 column and one row. The input row is a single pair of related fids joined using a &#39;:&#39;.</p>

<p>The output is a table with an added (usually huge) column containing the evidence that the pair of fids are,in fact, related. The evidence is a comma-separated list of pairs (each pair being two fids joined with &#39;:&#39;). Thus the command above produces</p>

<pre>           kb|g.0.peg.101:kb|g.0.peg.263        kb|g.0.peg.101:kb|g.0.peg.263,kb|g.10.peg.3334:kb|g.10.peg.3978,...</pre>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call co_occurrence_evidence. It is documented as follows:</p>

<pre>  $return = $obj-&#62;co_occurrence_evidence($pairs_of_fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$pairs_of_fids is a pairs_of_fids
$return is a reference to a list where each element is a reference to a list containing 2 items:
	0: a pair_of_fids
	1: an evidence
pairs_of_fids is a reference to a list where each element is a pair_of_fids
pair_of_fids is a reference to a list containing 2 items:
	0: a fid
	1: a fid
fid is a string
evidence is a reference to a list where each element is a pair_of_fids


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column (containing large evidence strings) added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="complexes_to_complex_data"
>complexes_to_complex_data</a></h2>

<p>Example:</p>

<pre>    complexes_to_complex_data [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call complexes_to_complex_data. It is documented as follows:</p>

<pre>  $return = $obj-&#62;complexes_to_complex_data($complexes)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$complexes is a complexes
$return is a reference to a hash where the key is a complex and the value is a complex_data
complexes is a reference to a list where each element is a complex
complex is a string
complex_data is a reference to a hash where the following keys are defined:
	complex_name has a value which is a name
	complex_roles has a value which is a roles_with_flags
	complex_reactions has a value which is a reactions
name is a string
roles_with_flags is a reference to a list where each element is a role_with_flag
role_with_flag is a reference to a list containing 2 items:
	0: a role
	1: an optional
role is a string
optional is a string
reactions is a reference to a list where each element is a reaction
reaction is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added in this order: complex_name complex_reactions complex_roles (flag role).</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="complexes_to_roles"
>complexes_to_roles</a></h2>

<p>Complexes catalyze reactions and are made up of one or more roles.
The roles that make up a complex can be &#34;optional&#34;,
or not.</p>

<p>This command takes as input a table with a column composed of complex ids and adds one column (containing a role that is part of the complex).
The &#34;optional&#34; field is not returned,
so you do not get back a field indicating whether or not the role is optional.
If you need that information,
we recommend that you use</p>

<pre>    get_relationship_IsTriggeredBy -rel optional -to id</pre>

<p>This would tack on both the optional flag and the role.</p>

<p>Example:</p>

<pre>    complexes_to_roles [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call complexes_to_roles. It is documented as follows:</p>

<pre>  $return = $obj-&#62;complexes_to_roles($complexes)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$complexes is a complexes
$return is a reference to a hash where the key is a complexes and the value is a roles
complexes is a reference to a list where each element is a complex
complex is a string
roles is a reference to a list where each element is a role
role is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="contigs_to_lengths"
>contigs_to_lengths</a></h2>

<p>In some cases,
one wishes to know just the lengths of the contigs,
rather than their actual DNA sequence (e.g.,
suppose that you wished to know if a gene boundary occured within 100 bp of the end of the contig).
To avoid requiring a user to access the entire DNA sequence,
we offer the ability to retrieve just the contig lengths.
Input to the routine is a list of contig IDs.
The routine returns a mapping from contig IDs to lengths</p>

<p>Example:</p>

<pre>    contigs_to_lengths [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call contigs_to_lengths. It is documented as follows:</p>

<pre>  $return = $obj-&#62;contigs_to_lengths($contigs)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$contigs is a contigs
$return is a reference to a hash where the key is a contig and the value is a length
contigs is a reference to a list where each element is a contig
contig is a string
length is an int


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column (the contig length) added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="contigs_to_md5s"
>contigs_to_md5s</a></h2>

<p>contigs_to_md5s can be used to acquire MD5 values for each of a list of contigs.
The quickest way to determine whether two contigs are identical is to compare their associated MD5 values,
eliminating the need to retrieve the sequence of each and compare them.</p>

<p>The routine takes as input a list of contig IDs.
The output is a mapping from contig ID to MD5 value.</p>

<p>Example:</p>

<pre>    contigs_to_md5s [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call contigs_to_md5s. It is documented as follows:</p>

<pre>  $return = $obj-&#62;contigs_to_md5s($contigs)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$contigs is a contigs
$return is a reference to a hash where the key is a contig and the value is a md5
contigs is a reference to a list where each element is a contig
contig is a string
md5 is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column (the contig md5) added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="contigs_to_sequences"
>contigs_to_sequences</a></h2>

<p>contigs_to_sequences is used to access the DNA sequence associated with each of a set of input contigs.
It takes as input a set of contig IDs (from which the genome can be determined) and produces a mapping from the input IDs to the returned DNA sequence in each case.</p>

<p>Normally,
people want to get back a fasta file,
so that is the default (the input table is essentially discarded,
keeping only the fasta file).
The alternative to to add a column to the table (containing the contig sequence).
The &#34;-fasta=0&#34; argument should be used,
if you do not want fasta output.</p>

<p>Example:</p>

<pre>    contigs_to_sequences [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call contigs_to_sequences. It is documented as follows:</p>

<pre>  $return = $obj-&#62;contigs_to_sequences($contigs)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$contigs is a contigs
$return is a reference to a hash where the key is a contig and the value is a dna
contigs is a reference to a list where each element is a contig
contig is a string
dna is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the contig IDs is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>

<dd>
<dt><a name="-fasta"
>-fasta</a></dt>

<dd>
<p>This is used to request a fasta output file (dropping all of the other columns in the input lines). It defaults to outputing just a fasta entry.</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="corresponds"
>corresponds</a></h2>

<p>Example:</p>

<pre>    corresponds [arguments] &#60; input &#62; output</pre>

<p>The input file should be a table with one column containing fids to be projected to a designated genome.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the identifier.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call corresponds. It is documented as follows:</p>

<pre>  $return = $obj-&#62;corresponds($fids, $genome)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$genome is a genome
$return is a reference to a hash where the key is a fid and the value is a correspondence
fids is a reference to a list where each element is a fid
fid is a string
genome is a string
correspondence is a reference to a hash where the following keys are defined:
	to has a value which is a fid
	ncontext has a value which is an int
	b1 has a value which is an int
	e1 has a value which is an int
	ln1 has a value which is an int
	b2 has a value which is an int
	e2 has a value which is an int
	ln2 has a value which is an int
	score has a value which is an int


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-g_Genome"
>-g Genome</a></dt>

<dd>
<p>This is used to designate a genome in the CS. The command will try to project fids from the input to fids in this designated genome.</p>

<dt><a name="-tsg_Genome"
>-tsg Genome</a></dt>

<dd>
<p>This is used to designate a genome in the default TS. The command will try to project fids from the input to fids in this designated genome. This argument and the &#60;b&#62;-g&#60;/b&#62; argument ar mutually exclusive. If they both are specified then the genome in the CS will be used.</p>

<p>= item -a Cutoff</p>

<p>This requests an abbreviated format in which only two columns are added: the projection score and the fid in the target genome. If this is not used 10 column are added. The Cutoff specifies a minimum projection score.</p>

<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the identifier is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added. If the abbreviated format is requested, two columns get added (sc and the fid projected to). If the abbreviated format is not requested, ten columns will be added</p>

<p>[percent-identity,matching-context,b1,e1,ln1,b2,e2,ln2,sc,to-fid]</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="equiv_sequence_assertions"
>equiv_sequence_assertions</a></h2>

<p>Different groups have made assertions of function for numerous protein sequences.
The equiv_sequence_assertions allows the user to gather function assertions from all of the sources.
Each assertion includes a field indicating whether the person making the assertion viewed themself as an &#34;expert&#34;.
The routine gathers assertions for all proteins having identical protein sequence.</p>

<p>Example:</p>

<pre>    equiv_sequence_assertions [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain a protein identifier (i.e., an md5 value). If another column contains the md5s use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the protein identifiers.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call equiv_sequence_assertions. It is documented as follows:</p>

<pre>  $return = $obj-&#62;equiv_sequence_assertions($proteins)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$proteins is a proteins
$return is a reference to a hash where the key is a protein and the value is a function_assertions
proteins is a reference to a list where each element is a protein
protein is a string
function_assertions is a reference to a list where each element is a function_assertion
function_assertion is a reference to a list containing 4 items:
	0: an id
	1: a function
	2: a source
	3: an expert
id is a string
function is a string
source is a string
expert is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the md5s is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added (identifier,source,function,expert).</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="fids_to_annotations"
>fids_to_annotations</a></h2>

<p>This routine takes as input a list of fids.
It retrieves the existing annotations for each fid,
including the text of the annotation,
who made the annotation and when (as seconds from the epoch).</p>

<p>Example:</p>

<pre>    fids_to_annotations [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call fids_to_annotations. It is documented as follows:</p>

<pre>  $return = $obj-&#62;fids_to_annotations($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is an annotations
fids is a reference to a list where each element is a fid
fid is a string
annotations is a reference to a list where each element is an annotation
annotation is a reference to a list containing 3 items:
	0: a comment
	1: an annotator
	2: an annotation_time
comment is a string
annotator is a string
annotation_time is an int


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="fids_to_atomic_regulons"
>fids_to_atomic_regulons</a></h2>

<p>The fids_to_atomic_regulons allows one to map fids into regulons that contain the fids.
Normally a fid will be in at most one regulon,
but we support multiple regulons.</p>

<p>Example:</p>

<pre>    fids_to_atomic_regulons [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call fids_to_regulons. It is documented as follows:</p>

<pre>  $return = $obj-&#62;fids_to_regulons($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is a regulon_size_pairs
fids is a reference to a list where each element is a fid
fid is a string
regulon_size_pairs is a reference to a list where each element is a regulon_size_pair
regulon_size_pair is a reference to a list containing 2 items:
	0: a regulon
	1: a regulon_size
regulon is a string
regulon_size is an int


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="fids_to_co_occurring_fids"
>fids_to_co_occurring_fids</a></h2>

<p>One of the most powerful clues to function relates to conserved clusters of genes on the chromosome (in prokaryotic genomes).
We have attempted to record pairs of genes that tend to occur close to one another on the chromosome.
To meaningfully do this,
we need to construct similarity-based mappings between genes in distinct genomes.
We have constructed such mappings for many (but not all) genomes maintained in the Kbase CS.
The prokaryotic geneomes in the CS are grouped into OTUs by ribosomal RNA (genomes within a single OTU have SSU rRNA that is greater than 97% identical).
If two genes occur close to one another (i.e.,
corresponding genes occur close to one another),
then we assign a score,
which is the number of distinct OTUs in which such clustering is detected.
This allows one to normalize for situations in which hundreds of corresponding genes are detected,
but they all come from very closely related genomes.</p>

<p>The significance of the score relates to the number of genomes in the database.
We recommend that you take the time to look at a set of scored pairs and determine approximately what percentage appear to be actually related for a few cutoff values.</p>

<p>Example:</p>

<pre>    fids_to_co_occurring_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call fids_to_co_occurring_fids. It is documented as follows:</p>

<pre>  $return = $obj-&#62;fids_to_co_occurring_fids($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is a scored_fids
fids is a reference to a list where each element is a fid
fid is a string
scored_fids is a reference to a list where each element is a scored_fid
scored_fid is a reference to a list containing 2 items:
	0: a fid
	1: a float


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added. For each line of input there may be multiple lines of output, one for each co_occuring fid. Two columns are added, the score and the co_occurring fid.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="fids_to_coexpressed_fids"
>fids_to_coexpressed_fids</a></h2>

<p>The routine fids_to_coexpressed_fids returns (for each input fid) a list of features that appear to be coexpressed.
That is,
for an input fid,
we determine the set of fids from the same genome that have Pearson Correlation Coefficients (based on normalized expression data) greater than 0.5 or less than -0.5.</p>

<p>Example:</p>

<pre>    fids_to_coexpressed_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call fids_to_coexpressed_fids. It is documented as follows:</p>

<pre>  $return = $obj-&#62;fids_to_coexpressed_fids($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is a scored_fids
fids is a reference to a list where each element is a fid
fid is a string
scored_fids is a reference to a list where each element is a scored_fid
scored_fid is a reference to a list containing 2 items:
	0: a fid
	1: a float


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="fids_to_dna_sequences"
>fids_to_dna_sequences</a></h2>

<p>fids_to_dna_sequences allows the user to look up the DNA sequences corresponding to each of a set of fids.</p>

<p>Example:</p>

<pre>    fids_to_dna_sequences [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call fids_to_dna_sequences. It is documented as follows:</p>

<pre>  $return = $obj-&#62;fids_to_dna_sequences($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is a dna_sequence
fids is a reference to a list where each element is a fid
fid is a string
dna_sequence is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>

<dd>
<dt><a name="-fasta"
>-fasta</a></dt>

<dd>
<p>This is used to request a fasta output file (dropping all of the other columns in the input lines). It defaults to outputing just a fasta entry.</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is jsut a fasta file with the sequence. You can also get a tab-delimited file by using -fasta=0. The tab-delimited format consists of the input file with an extra column of sequence added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="fids_to_feature_data"
>fids_to_feature_data</a></h2>

<p>Example:</p>

<pre>    fids_to_feature_data [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call fids_to_feature_data. It is documented as follows:</p>

<pre>  $return = $obj-&#62;fids_to_feature_data($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is a feature_data
fids is a reference to a list where each element is a fid
fid is a string
feature_data is a reference to a hash where the following keys are defined:
	feature_id has a value which is a fid
	genome_name has a value which is a string
	feature_function has a value which is a string
	feature_length has a value which is an int
	feature_publications has a value which is a pubrefs
	feature_location has a value which is a location
pubrefs is a reference to a list where each element is a pubref
pubref is a reference to a list containing 3 items:
	0: a string
	1: a string
	2: a string
location is a reference to a list where each element is a region_of_dna
region_of_dna is a reference to a list containing 4 items:
	0: a contig
	1: a begin
	2: a strand
	3: a length
contig is a string
begin is an int
strand is a string
length is an int


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added in this order:</p>

<dl>
<dt><a name="feature_id"
>feature_id</a></dt>

<dd>
<p>Feature identifier</p>

<dt><a name="feature_publications"
>feature_publications</a></dt>

<dd>
<p>List of Pubmed identifiers for publications associated with this feature.</p>

<dt><a name="genome_name"
>genome_name</a></dt>

<dd>
<p>Name of the genome this feature is present in.</p>

<dt><a name="feature_location"
>feature_location</a></dt>

<dd>
<p>Location string describing the location of this feature in its genome.</p>

<dt><a name="feature_length"
>feature_length</a></dt>

<dd>
<p>Length of the DNA sequence of this feature.</p>

<dt><a name="feature_function"
>feature_function</a></dt>

<dd>
<p>The function currently assigned to this feature.</p>
</dd>
</dl>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="fids_to_functions"
>fids_to_functions</a></h2>

<p>This routine takes as input a list of fids and returns a mapping from the fids to their assigned functions.</p>

<p>Example:</p>

<pre>    fids_to_functions [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call fids_to_functions. It is documented as follows:</p>

<pre>  $return = $obj-&#62;fids_to_functions($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is a function
fids is a reference to a list where each element is a fid
fid is a string
function is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="fids_to_genomes"
>fids_to_genomes</a></h2>

<p>Example:</p>

<pre>    fids_to_genomes [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call fids_to_genomes. It is documented as follows:</p>

<pre>  $return = $obj-&#62;fids_to_genomes($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is a genome
fids is a reference to a list where each element is a fid
fid is a string
genome is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="fids_to_literature"
>fids_to_literature</a></h2>

<p>We try to associate features and publications,
when the publications constitute supporting evidence of the function.
We connect a paper to a feature when we believe that an &#34;expert&#34; has asserted that the function of the feature is basically what we have associated with the feature.
Thus,
we might attach a paper reporting the crystal structure of a protein,
even though the paper is clearly not the paper responsible for the original characterization.
Our position in this matter is somewhat controversial,
but we are seeking to characterize some assertions as relatively solid,
and this strategy seems to support that goal.
Please note that we certainly wish we could also capture original publications,
and when experts can provide those connections,
we hope that they will help record the associations.</p>

<p>Example:</p>

<pre>    fids_to_literature [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call fids_to_literature. It is documented as follows:</p>

<pre>  $return = $obj-&#62;fids_to_literature($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is a pubrefs
fids is a reference to a list where each element is a fid
fid is a string
pubrefs is a reference to a list where each element is a pubref
pubref is a reference to a list containing 3 items:
	0: a string
	1: a string
	2: a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="fids_to_locations"
>fids_to_locations</a></h2>

<p>A &#34;location&#34; is a sequence of &#34;regions&#34;.
A region is a contiguous set of bases in a contig.
We work with locations in both the string form and as structures.
fids_to_locations takes as input a list of fids.
For each fid,
a structured location is returned.
The location is a list of regions; a region is given as a pointer to a list containing</p>

<pre>             the contig,
             the beginning base in the contig (from 1).
             the strand (+ or -), and
             the length</pre>

<p>Note that specifying a region using these 4 values allows you to represent a single base-pair region on either strand unambiguously (which giving begin/end pairs does not achieve).</p>

<p>Example:</p>

<pre>    fids_to_locations [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call fids_to_locations. It is documented as follows:</p>

<pre>  $return = $obj-&#62;fids_to_locations($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is a location
fids is a reference to a list where each element is a fid
fid is a string
location is a reference to a list where each element is a region_of_dna
region_of_dna is a reference to a list containing 4 items:
	0: a contig
	1: a begin
	2: a strand
	3: a length
contig is a string
begin is an int
strand is a string
length is an int


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="fids_to_protein_families"
>fids_to_protein_families</a></h2>

<p>Kbase supports the creation and maintence of protein families.
Each family is intended to contain a set of isofunctional homologs.
Currently,
the families are collections of translations of features,
rather than of just protein sequences (represented by md5s,
for example).
fids_to_protein_families supports access to the features that have been grouped into a family.
Ideally,
each feature in a family would have the same assigned function.
This is not always true,
but probably should be.</p>

<p>Example:</p>

<pre>    fids_to_protein_families [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call fids_to_protein_families. It is documented as follows:</p>

<pre>  $return = $obj-&#62;fids_to_protein_families($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is a protein_families
fids is a reference to a list where each element is a fid
fid is a string
protein_families is a reference to a list where each element is a protein_family
protein_family is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="fids_to_protein_sequences"
>fids_to_protein_sequences</a></h2>

<p>fids_to_protein_sequences allows the user to look up the amino acid sequences corresponding to each of a set of fids.
You can also get the sequence from proteins (i.e.,
md5 values).
This routine saves you having to look up the md5 sequence and then accessing the protein string in a separate call.</p>

<p>Example:</p>

<pre>    fids_to_protein_sequences [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call fids_to_protein_sequences. It is documented as follows:</p>

<pre>  $return = $obj-&#62;fids_to_protein_sequences($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is a protein_sequence
fids is a reference to a list where each element is a fid
fid is a string
protein_sequence is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>

<dd>
<dt><a name="-fasta"
>-fasta</a></dt>

<dd>
<p>This is used to request a fasta output file (dropping all of the other columns in the input lines). It defaults to outputing just a fasta entry.</p>

<dt><a name="-fc_Columns_[_construct_comment_for_fasta_from_these_columns_]"
>-fc Columns [ construct comment for fasta from these columns ]</a></dt>

<dd>
<p>This is used to ask for &#34;fasta comments&#34; formed from one or more columns (comma-separated)</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is jsut a fasta file with the sequence. You can also get a tab-delimited file by using -fasta=0. The tab-delimited format consists of the input file with an extra column of sequence added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="fids_to_proteins"
>fids_to_proteins</a></h2>

<p>Example:</p>

<pre>    fids_to_proteins [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call fids_to_proteins. It is documented as follows:</p>

<pre>  $return = $obj-&#62;fids_to_proteins($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is a md5
fids is a reference to a list where each element is a fid
fid is a string
md5 is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="fids_to_regulon_data"
>fids_to_regulon_data</a></h2>

<p>Example:</p>

<pre>    fids_to_regulon_data [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call fids_to_regulon_data. It is documented as follows:</p>

<pre>  $return = $obj-&#62;fids_to_regulon_data($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is a regulons_data
fids is a reference to a list where each element is a fid
fid is a string
regulons_data is a reference to a list where each element is a regulon_data
regulon_data is a reference to a hash where the following keys are defined:
	regulon_id has a value which is a regulon
	regulon_set has a value which is a fids
	tfs has a value which is a fids
regulon is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>

<dd>
<dt><a name="-h"
>-h</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="-fields_field-list"
>-fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available: regulon_id, regulon_set, tfs.</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="fids_to_roles"
>fids_to_roles</a></h2>

<p>Given a feature,
one can get the set of roles it implements using fid_to_roles.
Remember,
a protein can be multifunctional -- implementing several roles.
This can occur due to fusions or to broad specificity of substrate.</p>

<p>Example:</p>

<pre>    fids_to_roles [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call fids_to_roles. It is documented as follows:</p>

<pre>  $return = $obj-&#62;fids_to_roles($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is a roles
fids is a reference to a list where each element is a fid
fid is a string
roles is a reference to a list where each element is a role
role is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="fids_to_subsystem_data"
>fids_to_subsystem_data</a></h2>

<p>Example:</p>

<pre>    fids_to_subsystem_data [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call fids_to_subsystem_data. It is documented as follows:</p>

<pre>  $return = $obj-&#62;fids_to_subsystem_data($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is a ss_var_role_tuples
fids is a reference to a list where each element is a fid
fid is a string
ss_var_role_tuples is a reference to a list where each element is a ss_var_role_tuple
ss_var_role_tuple is a reference to a list containing 3 items:
	0: a subsystem
	1: a variant
	2: a role
subsystem is a string
variant is a string
role is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="fids_to_subsystems"
>fids_to_subsystems</a></h2>

<p>fids in subsystems normally have somewhat more reliable assigned functions than those not in subsystems.
Hence,
it is common to ask &#34;Is this protein-encoding gene included in any subsystems?&#34; fids_to_subsystems can be used to see which subsystems contain a fid (or,
you can submit as input a set of fids and get the subsystems for each).</p>

<p>Example:</p>

<pre>    fids_to_subsystems [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call fids_to_subsystems. It is documented as follows:</p>

<pre>  $return = $obj-&#62;fids_to_subsystems($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is a subsystems
fids is a reference to a list where each element is a fid
fid is a string
subsystems is a reference to a list where each element is a subsystem
subsystem is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="genomes_to_contigs"
>genomes_to_contigs</a></h2>

<p>The routine genomes_to_contigs can be used to retrieve the IDs of the contigs associated with each of a list of input genomes.
The routine constructs a mapping from genome ID to the list of contigs included in the genome.</p>

<p>Example:</p>

<pre>    genomes_to_contigs [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each line of input, there can be multiple output lines, one per contig. The contig id is added to the end of each line.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call genomes_to_contigs. It is documented as follows:</p>

<pre>  $return = $obj-&#62;genomes_to_contigs($genomes)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$genomes is a genomes
$return is a reference to a hash where the key is a genome and the value is a contigs
genomes is a reference to a list where each element is a genome
genome is a string
contigs is a reference to a list where each element is a contig
contig is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="genomes_to_fids"
>genomes_to_fids</a></h2>

<p>genomes_to_fids is used to get the fids included in specific genomes.
It is often the case that you want just one or two types of fids -- hence,
the types_of_fids argument.</p>

<p>Example:</p>

<pre>    genomes_to_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each input line, ther can be many output lines, one per feature. The feature id is added to the end of the line.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call genomes_to_fids. It is documented as follows:</p>

<pre>  $return = $obj-&#62;genomes_to_fids($genomes, $types_of_fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$genomes is a genomes
$types_of_fids is a types_of_fids
$return is a reference to a hash where the key is a genome and the value is a fids
genomes is a reference to a list where each element is a genome
genome is a string
types_of_fids is a reference to a list where each element is a type_of_fid
type_of_fid is a string
fids is a reference to a list where each element is a fid
fid is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<p>This is the type of the fid requested, i.e. peg, rna, etc.</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="genomes_to_genome_data"
>genomes_to_genome_data</a></h2>

<p>Example:</p>

<pre>    genomes_to_genome_data [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call genomes_to_genome_data. It is documented as follows:</p>

<pre>  $return = $obj-&#62;genomes_to_genome_data($genomes)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$genomes is a genomes
$return is a reference to a hash where the key is a genome and the value is a genome_data
genomes is a reference to a list where each element is a genome
genome is a string
genome_data is a reference to a hash where the following keys are defined:
	complete has a value which is an int
	contigs has a value which is an int
	dna_size has a value which is an int
	gc_content has a value which is a float
	genetic_code has a value which is an int
	pegs has a value which is an int
	rnas has a value which is an int
	scientific_name has a value which is a string
	taxonomy has a value which is a string
	genome_md5 has a value which is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added in this order: rnas gc_content dna_size taxonomy scientific_name contigs genome_md5 pegs genetic_code complete.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="genomes_to_md5s"
>genomes_to_md5s</a></h2>

<p>The routine genomes_to_md5s can be used to look up the MD5 value associated with each of a set of genomes.
The MD5 values are computed when the genome is loaded,
so this routine just retrieves the precomputed values.</p>

<p>Note that the MD5 value of a genome is independent of the contig names and case of the DNA sequences that make up the genome.</p>

<p>Example:</p>

<pre>    genomes_to_md5s [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call genomes_to_md5s. It is documented as follows:</p>

<pre>  $return = $obj-&#62;genomes_to_md5s($genomes)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$genomes is a genomes
$return is a reference to a hash where the key is a genome and the value is a md5
genomes is a reference to a list where each element is a genome
genome is a string
md5 is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="genomes_to_subsystems"
>genomes_to_subsystems</a></h2>

<p>A user can invoke genomes_to_subsystems to rerieve the names of the subsystems relevant to each genome.
The input is a list of genomes.
The output is a mapping from genome to a list of 2-tuples,
where each 2-tuple give a variant code and a subsystem name.
Variant codes of -1 (or *-1) amount to assertions that the genome contains no active variant.
A variant code of 0 means &#34;work in progress&#34;,
and presence or absence of the subsystem in the genome should be undetermined.</p>

<p>Example:</p>

<pre>    genomes_to_subsystems [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call genomes_to_subsystems. It is documented as follows:</p>

<pre>  $return = $obj-&#62;genomes_to_subsystems($genomes)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$genomes is a genomes
$return is a reference to a hash where the key is a genome and the value is a variant_subsystem_pairs
genomes is a reference to a list where each element is a genome
genome is a string
variant_subsystem_pairs is a reference to a list where each element is a variant_of_subsystem
variant_of_subsystem is a reference to a list containing 2 items:
	0: a subsystem
	1: a variant
subsystem is a string
variant is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added. For each line of the input file there can be many output files, one per subsystem. A single field of variant,subsystem is added to the end of each line.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="genomes_to_taxonomies"
>genomes_to_taxonomies</a></h2>

<p>The routine genomes_to_taxonomies can be used to retrieve taxonomic information for each of a list of input genomes.
For each genome in the input list of genomes,
a list of taxonomic groups is returned.
Kbase will use the groups maintained by NCBI.
For an NCBI taxonomic string like</p>

<pre>     cellular organisms;
     Bacteria;
     Proteobacteria;
     Gammaproteobacteria;
     Enterobacteriales;
     Enterobacteriaceae;
     Escherichia;
     Escherichia coli</pre>

<p>associated with the strain &#39;Escherichia coli 1412&#39;, this routine would return a list of these taxonomic groups:</p>

<pre>     [&#39;Bacteria&#39;,
      &#39;Proteobacteria&#39;,
      &#39;Gammaproteobacteria&#39;,
      &#39;Enterobacteriales&#39;,
      &#39;Enterobacteriaceae&#39;,
      &#39;Escherichia&#39;,
      &#39;Escherichia coli&#39;,
      &#39;Escherichia coli 1412&#39;
     ]</pre>

<p>That is, the initial &#34;cellular organisms&#34; has been deleted, and the strain ID has been added as the last &#34;grouping&#34;.</p>

<p>The output is a mapping from genome IDs to lists of the form shown above.</p>

<p>Example:</p>

<pre>    genomes_to_taxonomies [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call genomes_to_taxonomies. It is documented as follows:</p>

<pre>  $return = $obj-&#62;genomes_to_taxonomies($genomes)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$genomes is a genomes
$return is a reference to a hash where the key is a genome and the value is a taxonomic_groups
genomes is a reference to a list where each element is a genome
genome is a string
taxonomic_groups is a reference to a list where each element is a taxonomic_group
taxonomic_group is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added. One extra column is added, consisting of the taxonomy list presented as a single string with items separated by a &#34;:&#34;.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="locations_to_dna_sequences"
>locations_to_dna_sequences</a></h2>

<p>locations_to_dna_sequences takes as input a list of locations (each in the form of a list of regions).
The routine constructs 2-tuples composed of</p>

<pre>     [the input location,the dna string]</pre>

<p>The returned DNA string is formed by concatenating the DNA for each of the regions that make up the location.</p>

<p>Example:</p>

<pre>    locations_to_dna_sequences [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call locations_to_dna_sequences. It is documented as follows:</p>

<pre>  $dna_seqs = $obj-&#62;locations_to_dna_sequences($locations)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$locations is a locations
$dna_seqs is a reference to a list where each element is a reference to a list containing 2 items:
	0: a location
	1: a dna
locations is a reference to a list where each element is a location
location is a reference to a list where each element is a region_of_dna
region_of_dna is a reference to a list containing 4 items:
	0: a contig
	1: a begin
	2: a strand
	3: a length
contig is a string
begin is an int
strand is a string
length is an int
dna is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="locations_to_fids"
>locations_to_fids</a></h2>

<p>It is frequently the case that one wishes to look up the genes that occur in a given region of a contig.
Location_to_fids can be used to extract such sets of genes for each region in the input set of regions.
We define a gene as &#34;occuring&#34; in a region if the location of the gene overlaps the designated region.</p>

<p>This command takes as input a table in which one column is a location.
It adds an extra column which will contain a feature that overlaps the location.
if n features overlap a location,
then that input line will generate n output lines (each with an extra column giving one of the overlapping fids).</p>

<p>Example:</p>

<pre>    locations_to_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the location. If another column contains the location use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the location.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call locations_to_fids. It is documented as follows:</p>

<pre>  $return = $obj-&#62;locations_to_fids($region_of_dna_strings)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$region_of_dna_strings is a region_of_dna_strings
$return is a reference to a hash where the key is a region_of_dna_string and the value is a fids
region_of_dna_strings is a reference to a list where each element is a region_of_dna_string
region_of_dna_string is a string
fids is a reference to a list where each element is a fid
fid is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column (an overlapping fid) added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="md5s_to_genomes"
>md5s_to_genomes</a></h2>

<p>md5s to genomes is used to get the genomes associated with each of a list of input md5 values.</p>

<pre>           The routine takes as input a list of MD5 values.  It constructs a mapping from each input
           MD5 value to a list of genomes that share the same MD5 value.

           The MD5 value for a genome is independent of the names of contigs and the case of the DNA sequence
           data.</pre>

<p>Example:</p>

<pre>    md5s_to_genomes [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call md5s_to_genomes. It is documented as follows:</p>

<pre>  $return = $obj-&#62;md5s_to_genomes($md5s)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$md5s is a md5s
$return is a reference to a hash where the key is a md5 and the value is a genomes
md5s is a reference to a list where each element is a md5
md5 is a string
genomes is a reference to a list where each element is a genome
genome is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="otu_members"
>otu_members</a></h2>

<p>We deine the notion of OTU loosely.
We try to keep genomes with 97% identical SSUrRNAs within the same OTU.
We pick a representative genome for each OTU,
and we try to keep all genomes that have 97% identical SSU rRNAs in the same OTU.</p>

<p>This command is intended to add a column containing the genomes from the same OTU.</p>

<p>Example:</p>

<pre>    otu_members [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain a genome identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call otu_members. It is documented as follows:</p>

<pre>  $return = $obj-&#62;otu_members($genomes)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$genomes is a genomes
$return is a reference to a hash where the key is a genome and the value is a reference to a hash where the key is a genome and the value is a genome_name
genomes is a reference to a list where each element is a genome
genome is a string
genome_name is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing genome identifiers is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column (a genome from the same OTU) added. If there are N genomes in the OTU for the genome in an input line, then N output lines will be generated, each containing a single genome ID from the same OTU.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="protein_families_to_co_occurring_families"
>protein_families_to_co_occurring_families</a></h2>

<p>Since we accumulate data relating to the co-occurrence (i.e.,
chromosomal clustering) of genes in prokaryotic genomes,
we can note which pairs of genes tend to co-occur.
From this data,
one can compute the protein families that tend to co-occur (i.e.,
tend to cluster on the chromosome).
This allows one to formulate conjectures for unclustered pairs,
based on clustered pairs from the same protein_families.</p>

<p>Example:</p>

<pre>    protein_families_to_co_occurring_families [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call protein_families_to_co_occurring_families. It is documented as follows:</p>

<pre>  $return = $obj-&#62;protein_families_to_co_occurring_families($protein_families)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$protein_families is a protein_families
$return is a reference to a hash where the key is a protein_family and the value is a fc_protein_families
protein_families is a reference to a list where each element is a protein_family
protein_family is a string
fc_protein_families is a reference to a list where each element is a fc_protein_family
fc_protein_family is a reference to a list containing 3 items:
	0: a protein_family
	1: a score
	2: a function
score is a float
function is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="protein_families_to_fids"
>protein_families_to_fids</a></h2>

<p>protein_families_to_fids can be used to access the set of fids represented by each of a set of protein_families.
We define protein_families as sets of fids (rather than sets of MD5s.
This may,
or may not,
be a mistake.</p>

<p>Example:</p>

<pre>    protein_families_to_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each input line, there are multiple output lines, one for each fid in the family, the fid is added to the end of each output line.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call protein_families_to_fids. It is documented as follows:</p>

<pre>  $return = $obj-&#62;protein_families_to_fids($protein_families)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$protein_families is a protein_families
$return is a reference to a hash where the key is a protein_family and the value is a fids
protein_families is a reference to a list where each element is a protein_family
protein_family is a string
fids is a reference to a list where each element is a fid
fid is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="protein_families_to_functions"
>protein_families_to_functions</a></h2>

<p>protein_families_to_functions can be used to extract the set of functions assigned to the fids that make up the family.
Each input protein_family is mapped to a family function.</p>

<p>Example:</p>

<pre>    protein_families_to_functions [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call protein_families_to_functions. It is documented as follows:</p>

<pre>  $return = $obj-&#62;protein_families_to_functions($protein_families)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$protein_families is a protein_families
$return is a reference to a hash where the key is a protein_family and the value is a fid_function_pairs
protein_families is a reference to a list where each element is a protein_family
protein_family is a string
fid_function_pairs is a reference to a list where each element is a fid_function_pair
fid_function_pair is a reference to a list containing 2 items:
	0: a fid
	1: a function
fid is a string
function is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="protein_families_to_proteins"
>protein_families_to_proteins</a></h2>

<p>protein_families_to_proteins can be used to access the set of proteins (i.e.,
the set of MD5 values) represented by each of a set of protein_families.
We define protein_families as sets of fids (rather than sets of MD5s.
This may,
or may not,
be a mistake.</p>

<p>Example:</p>

<pre>    protein_families_to_proteins [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call protein_families_to_proteins. It is documented as follows:</p>

<pre>  $return = $obj-&#62;protein_families_to_proteins($protein_families)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$protein_families is a protein_families
$return is a reference to a hash where the key is a protein_family and the value is a proteins
protein_families is a reference to a list where each element is a protein_family
protein_family is a string
proteins is a reference to a list where each element is a protein
protein is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. For each input line, there are multiple output lines, one for each protein in the family. The protein is added to the end of each line.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="proteins_to_fids"
>proteins_to_fids</a></h2>

<p>proteins_to_fids takes as input a list of proteins (i.e.,
a list of md5s) and returns for each a set of protein-encoding fids that have the designated sequence as their translation.
That is,
for each sequence,
the returned fids will be the entire set (within Kbase) that have the sequence as a translation.</p>

<p>Example:</p>

<pre>    proteins_to_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each input line, there are multiple output lines, one for each fid the protein maps to. The fid is added to the end of each line.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call proteins_to_fids. It is documented as follows:</p>

<pre>  $return = $obj-&#62;proteins_to_fids($proteins)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$proteins is a proteins
$return is a reference to a hash where the key is a protein and the value is a fids
proteins is a reference to a list where each element is a protein
protein is a string
fids is a reference to a list where each element is a fid
fid is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="proteins_to_functions"
>proteins_to_functions</a></h2>

<p>The routine proteins_to_functions allows users to access functions associated with specific protein sequences.
The input proteins are given as a list of MD5 values (these MD5 values each correspond to a specific protein sequence).
For each input MD5 value,
a list of [feature-id,function] pairs is constructed and returned.
Note that there are many cases in which a single protein sequence corresponds to the translation associated with multiple protein-encoding genes,
and each may have distinct functions (an undesirable situation,
we grant).</p>

<p>This function allows you to access all of the functions assigned (by all annotation groups represented in Kbase) to each of a set of sequences.</p>

<p>Example:</p>

<pre>    proteins_to_functions [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each input line there are multiple output lines, one for each fid the protein maps to. Two columns are added to each output line, a fid and a function.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call proteins_to_functions. It is documented as follows:</p>

<pre>  $return = $obj-&#62;proteins_to_functions($proteins)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$proteins is a proteins
$return is a reference to a hash where the key is a protein and the value is a fid_function_pairs
proteins is a reference to a list where each element is a protein
protein is a string
fid_function_pairs is a reference to a list where each element is a fid_function_pair
fid_function_pair is a reference to a list containing 2 items:
	0: a fid
	1: a function
fid is a string
function is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="proteins_to_literature"
>proteins_to_literature</a></h2>

<p>The routine proteins_to_literature can be used to extract the list of papers we have associated with specific protein sequences.
The user should note that in many cases the association of a paper with a protein sequence is not precise.
That is,
the paper may actually describe a closely-related protein (that may not yet even be in a sequenced genome).
Annotators attempt to use best judgement when associating literature and proteins.
Publication references include [pubmed ID,URL for the paper,
title of the paper].
In some cases,
the URL and title are omitted.
In theory,
we can extract them from PubMed and we will attempt to do so.</p>

<p>Example:</p>

<pre>    proteins_to_literature [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call proteins_to_literature. It is documented as follows:</p>

<pre>  $return = $obj-&#62;proteins_to_literature($proteins)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$proteins is a proteins
$return is a reference to a hash where the key is a protein and the value is a pubrefs
proteins is a reference to a list where each element is a protein
protein is a string
pubrefs is a reference to a list where each element is a pubref
pubref is a reference to a list containing 3 items:
	0: a string
	1: a string
	2: a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="proteins_to_protein_families"
>proteins_to_protein_families</a></h2>

<p>Protein families contain a set of isofunctional homologs.
proteins_to_protein_families can be used to look up is used to get the set of protein_families containing a specified protein.
For performance reasons,
you can submit a batch of proteins (i.e.,
a list of proteins),
and for each input protein,
you get back a set (possibly empty) of protein_families.
Specific collections of families (e.g.,
FIGfams) usually require that a protein be in at most one family.
However,
we will be integrating protein families from a number of sources,
and so a protein can be in multiple families.</p>

<p>Example:</p>

<pre>    proteins_to_protein_families [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the protein.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each protein, the family it belongs to is added at the end of the input line.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call proteins_to_protein_families. It is documented as follows:</p>

<pre>  $return = $obj-&#62;proteins_to_protein_families($proteins)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$proteins is a proteins
$return is a reference to a hash where the key is a protein and the value is a protein_families
proteins is a reference to a list where each element is a protein
protein is a string
protein_families is a reference to a list where each element is a protein_family
protein_family is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="proteins_to_roles"
>proteins_to_roles</a></h2>

<p>The routine proteins_to_roles allows a user to gather the set of functional roles that are associated with specifc protein sequences.
A single protein sequence (designated by an MD5 value) may have numerous associated functions,
since functions are treated as an attribute of the feature,
and multiple features may have precisely the same translation.
In our experience,
it is not uncommon,
even for the best annotation teams,
to assign distinct functions (and,
hence,
functional roles) to identical protein sequences.</p>

<p>For each input MD5 value,
this routine gathers the set of features (fids) that share the same sequence,
collects the associated functions,
expands these into functional roles (for multi-functional proteins),
and returns the set of roles that results.</p>

<p>Note that,
if the user wishes to see the specific features that have the assigned functional roles,
they should use proteins_to_functions instead (it returns the fids associated with each assigned function).</p>

<p>Example:</p>

<pre>    proteins_to_roles [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each input line, there can be multiple output lines, one for each role the protein can map to. The role is added to the end of each line.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call proteins_to_roles. It is documented as follows:</p>

<pre>  $return = $obj-&#62;proteins_to_roles($proteins)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$proteins is a proteins
$return is a reference to a hash where the key is a protein and the value is a roles
proteins is a reference to a list where each element is a protein
protein is a string
roles is a reference to a list where each element is a role
role is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="reaction_strings"
>reaction_strings</a></h2>

<p>Reaction_strings are text strings that represent (albeit crudely) the details of Reactions.</p>

<p>Example:</p>

<pre>    reaction_strings [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain a reaction ID. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call reaction_strings. It is documented as follows:</p>

<pre>  $return = $obj-&#62;reaction_strings($reactions, $name_parameter)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$reactions is a reactions
$name_parameter is a name_parameter
$return is a reference to a hash where the key is a reaction and the value is a string
reactions is a reference to a list where each element is a reaction
reaction is a string
name_parameter is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column (the readable reaction string) added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="reactions_to_complexes"
>reactions_to_complexes</a></h2>

<p>Reactions are thought of as being either spontaneous or implemented by one or more Complexes.
Complexes connect to Roles.
Hence,
the connection of fids or roles to reactions goes through Complexes.</p>

<p>Complexes are best thought of as things like the ribosome or ATP synthase,
but when a single protein is not grouped with others,
we still think of it as a complex made up of a single member.</p>

<p>Example:</p>

<pre>    reactions_to_complexes [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the reaction identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call reactions_to_complexes. It is documented as follows:</p>

<pre>  $return = $obj-&#62;reactions_to_complexes($reactions)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$reactions is a reactions
$return is a reference to a hash where the key is a reaction and the value is a complexes
reactions is a reference to a list where each element is a reaction
reaction is a string
complexes is a reference to a list where each element is a complex
complex is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column (the complex ID) added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="regulons_to_fids"
>regulons_to_fids</a></h2>

<p>The CS contains two similar notions:</p>

<pre>    1. atomic regulons, which are sets of fids thought to have exactly the
       same expression profiles (derived from operons, subsystems, and expression data)

    2. regulons which are thought of as a set of genes impacted by a transcription factor.
       This second notion is built up looking for binding-site/transcription-factor pairs 
       and reconciling the insights with operons and subsystems.</pre>

<p>Clearly the two notions are closely related.</p>

<p>This command takes as input a table in which one column is composed of regulon ids, and it adds a column corresponding to the fids that make up the regulon.</p>

<p>Example:</p>

<pre>    regulons_to_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the regulon identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the regulon ID.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call regulons_to_fids. It is documented as follows:</p>

<pre>  $return = $obj-&#62;regulons_to_fids($regulons)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$regulons is a regulons
$return is a reference to a hash where the key is a regulon and the value is a fids
regulons is a reference to a list where each element is a regulon
regulon is a string
fids is a reference to a list where each element is a fid
fid is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column (fids in the regulon) added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="representative"
>representative</a></h2>

<p>Genomes within KBase have been grouped into OTUs.
Usually,
members of the same OTU have SSU RNAs that are 97% or more identical,
and usually two genomes with 97% identical SSU RNAs are part of the same OTU.
However,
we do not guarantee either assertion,
and there are some genomes that have not been placed into OTUs (usually due to a missing or truncated rRNA).</p>

<p>Example:</p>

<pre>    representative [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the genome identifer. If another column contains the genome identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the genome ID..</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call representative. It is documented as follows:</p>

<pre>  $return = $obj-&#62;representative($genomes)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$genomes is a genomes
$return is a reference to a hash where the key is a genome and the value is a genome
genomes is a reference to a list where each element is a genome
genome is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the genome IDs is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column added (the ID of the representative genome).</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="representative_sequences"
>representative_sequences</a></h2>

<p>Example:</p>

<pre>    representative_sequences [arguments] &#60; input &#62; output</pre>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call representative_sequences. It is documented as follows:</p>

<p>we return two arguments. The first is the list of representative triples, and the second is the list of sets (the first entry always being the representative sequence)</p>

<dl>
<dt><a name="-e_Existing_Representatives"
>-e Existing Representatives</a></dt>

<dd>
<p>These are sequences that are currently represenatives and the command extends this set.</p>

<dt><a name="-o_order-option"
>-o order-option</a></dt>

<dd>
<p>order sequences using the designated option (note that -b is another way to get a long-to-short ordering). Supported options are</p>

<pre>    long-to-short
    default (as is)</pre>

<dt><a name="-b"
>-b</a></dt>

<dd>
<p>order input sequences by size (long to short)</p>

<dt><a name="-c_cluster_type"
>-c cluster_type</a></dt>

<dd>
<p>behavior of clustering algorithm (0 or 1, D=1)</p>

<p>cluster_type 0 is the original method, which has only the representative for each group in the blast database. This can randomly segregate distant members of groups, regardless of the placement of other very similar sequences.</p>

<p>cluster_type 1 adds more diverse representatives of a group in the blast database. This is slightly more expensive, but is much less likely to split close relatives into different groups.</p>

<dt><a name="-d_seq_clust_dir_-_directory_for_files_of_clustered_sequencees"
>-d seq_clust_dir - directory for files of clustered sequencees</a></dt>

<dd>
<p>With the -d option, each cluster of sequences is written to a distinct file in the specified directory.</p>

<dt><a name="-f_id_clust_file_-_file_with_one_line_per_cluster,_listing_its_ids"
>-f id_clust_file - file with one line per cluster, listing its ids</a></dt>

<dd>
<p>With the -f option, for each cluster, a tab-separated list of ids is written to the specified file.</p>

<dt><a name="-m_measure_of_sim_-_measure_of_similarity_to_use:"
>-m measure_of_sim - measure of similarity to use:</a></dt>

<dd>
<p>Sequences are removed if there similarity to a &#34;kept&#34; sequence exceeds a specified threshold (see -similarity below)</p>

<p>The possible measures of similarity that you can specify are as follows:</p>

<p>identity_fraction (default), positive_fraction (proteins only), or score_per_position (0-2 bits)</p>

<dt><a name="-s_similarity_-_similarity_required_to_be_clustered_(D_=_0.8)"
>-s similarity - similarity required to be clustered (D = 0.8)</a></dt>

<dd>
<p>The similarity threshhold used to determine when sequences are deleted (but represented by a kept sequence).</p>

<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$seq_set is a seq_set
$rep_seq_parms is a rep_seq_parms
$return_1 is an id_set
$return_2 is a reference to a list where each element is an id_set
seq_set is a reference to a list where each element is a seq_triple
seq_triple is a reference to a list containing 3 items:
	0: an id
	1: a comment
	2: a sequence
id is a string
comment is a string
sequence is a string
rep_seq_parms is a reference to a hash where the following keys are defined:
	existing_reps has a value which is a seq_set
	order has a value which is an int
	alg has a value which is an int
	type_sim has a value which is an int
	cutoff has a value which is a float
id_set is a reference to a list where each element is an id


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Input_Format"
>Input Format</a></h3>

<pre>    The input is a fasta-formatted set of sequences.  These sequences
    should not contain indels.</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<pre>    FASTA output of the representatives is always written to STDOUT.

    The -d option will cause a directory to be built containing the clusters.

    The -f option will cause an abbreviated format of the clusters (just IDs) to
    be written</pre>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="roles_to_complexes"
>roles_to_complexes</a></h2>

<p>roles_to_complexes allows a user to connect Roles to Complexes,
from there,
the connection exists to Reactions (although in the actual ER-model model,
the connection from Complex to Reaction goes through ReactionComplex).
Since Roles also connect to fids,
the connection between fids and Reactions is induced.</p>

<p>Example:</p>

<pre>    roles_to_complexes [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each line of input there may be multiple lines of output, one per complex associated with the role. Two columns are appended to the input line, the optional flag, and the complex id.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call roles_to_complexes. It is documented as follows:</p>

<pre>  $return = $obj-&#62;roles_to_complexes($roles)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$roles is a roles
$return is a reference to a hash where the key is a role and the value is a complexes
roles is a reference to a list where each element is a role
role is a string
complexes is a reference to a list where each element is a complex
complex is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="roles_to_fids"
>roles_to_fids</a></h2>

<pre>            You often wish to find the fids in one or more genomes that
            implement specific functional roles.  To do this, you can use
            roles_to_fids.</pre>

<p>Example:</p>

<pre>    roles_to_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call roles_to_fids. It is documented as follows:</p>

<pre>  $return = $obj-&#62;roles_to_fids($roles, $genomes)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$roles is a roles
$genomes is a genomes
$return is a reference to a hash where the key is a role and the value is a fid
roles is a reference to a list where each element is a role
role is a string
genomes is a reference to a list where each element is a genome
genome is a string
fid is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>

<dd>
<dt><a name="genomes_-_A_list_of_genomes_to_restricty_the_output"
>genomes - A list of genomes to restricty the output</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="roles_to_protein_families"
>roles_to_protein_families</a></h2>

<p>roles_to_protein_families can be used to locate the protein families containing features that have assigned functions implying that they implement designated roles.
Note that for any input role (given as a role description),
you may have a set of distinct protein_families returned.</p>

<p>Example:</p>

<pre>    roles_to_protein_families [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call roles_to_protein_families. It is documented as follows:</p>

<pre>  $return = $obj-&#62;roles_to_protein_families($roles)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$roles is a roles
$return is a reference to a hash where the key is a role and the value is a protein_families
roles is a reference to a list where each element is a role
role is a string
protein_families is a reference to a list where each element is a protein_family
protein_family is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="roles_to_proteins"
>roles_to_proteins</a></h2>

<p>roles_to_proteins can be used to extract the set of proteins (designated by MD5 values) that currently are believed to implement a given role.
Note that the proteins may be multifunctional,
meaning that they may be implementing other roles,
as well.</p>

<p>Example:</p>

<pre>    roles_to_proteins [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call roles_to_proteins. It is documented as follows:</p>

<pre>  $return = $obj-&#62;roles_to_proteins($roles)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$roles is a roles
$return is a reference to a hash where the key is a role and the value is a proteins
roles is a reference to a list where each element is a role
role is a string
proteins is a reference to a list where each element is a protein
protein is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added. For each input line there can be multiple output lines, one per protein associated with the role. Two extra columns are added to the output, function and protein.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="roles_to_subsystems"
>roles_to_subsystems</a></h2>

<p>roles_to_subsystems can be used to access the set of subsystems that include specific roles.
The input is a list of roles (i.e.,
role descriptions),
and a mapping is returned as a hash with key role description and values composed of sets of susbsystem names.</p>

<p>Example:</p>

<pre>    roles_to_subsystems [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call roles_to_subsystems. It is documented as follows:</p>

<pre>  $return = $obj-&#62;roles_to_subsystems($roles)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$roles is a roles
$return is a reference to a hash where the key is a role and the value is a subsystems
roles is a reference to a list where each element is a role
role is a string
subsystems is a reference to a list where each element is a subsystem
subsystem is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added. For each line of input there can be multiple lines of output, one per subsystem containing the role. The subsystem name is added to the output lines.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="subsystems_to_fids"
>subsystems_to_fids</a></h2>

<p>This command gives to access to all of the rows in a subsystem.
It does not support selecting just rows for specific genomes (although the underlying API routine subsystems_to_fids does).
This command gives you all of the</p>

<pre>    [genome,variant-code,fids]</pre>

<p>for an input subsystem.</p>

<p>Example:</p>

<pre>    subsystems_to_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain a subssytem name. If another column contains the subsystem use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call subsystems_to_fids. It is documented as follows:</p>

<pre>  $return = $obj-&#62;subsystems_to_fids($subsystems, $genomes)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$subsystems is a subsystems
$genomes is a genomes
$return is a reference to a hash where the key is a subsystem and the value is a reference to a hash where the key is a genome and the value is a reference to a list containing 2 items:
	0: a variant
	1: a fids
subsystems is a reference to a list where each element is a subsystem
subsystem is a string
genomes is a reference to a list where each element is a genome
genome is a string
variant is a string
fids is a reference to a list where each element is a fid
fid is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added. These will be [Genome,Variant,FIF] (i.e., three columns are appended).</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="subsystems_to_genomes"
>subsystems_to_genomes</a></h2>

<p>This command takes as input a table with a column containing subsystem names.
An extra column is appended to the table containing genome IDs (for those genomes included in the subsystem with an active variant code (not matching /\*?(o|-1)/).</p>

<p>Example:</p>

<pre>    subsystems_to_genomes [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the subsystem name. If another column contains the subsystem name use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call subsystems_to_genomes. It is documented as follows:</p>

<pre>  $return = $obj-&#62;subsystems_to_genomes($subsystems)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$subsystems is a subsystems
$return is a reference to a hash where the key is a subsystem and the value is a reference to a list where each element is a reference to a list containing 2 items:
	0: a variant
	1: a genome
subsystems is a reference to a list where each element is a subsystem
subsystem is a string
variant is a string
genome is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with two extra columns added (the variant code and the genome).</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="subsystems_to_roles"
>subsystems_to_roles</a></h2>

<p>The command subsystem_to_roles takes as input a table with a column containing subsystem IDs.
The table is extended with an extra column set to roles that make up the subsystems.
For each input line (designating a subsystem),
a line will be added to the output for each role that occurs in the subsystem.</p>

<p>You can ask for auxiliary roles,
but the default is to leave them out.
If you want them,
use &#34;-aux=1&#34;.
Example:</p>

<pre>    subsystems_to_roles [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the subsystem name. If another column contains the subsystem, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call subsystems_to_roles. It is documented as follows:</p>

<pre>  $return = $obj-&#62;subsystems_to_roles($subsystems, $aux)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$subsystems is a subsystems
$aux is an aux
$return is a reference to a hash where the key is a subsystem and the value is a roles
subsystems is a reference to a list where each element is a subsystem
subsystem is a string
aux is an int
roles is a reference to a list where each element is a role
role is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column (Role) added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="subsystems_to_spreadsheets"
>subsystems_to_spreadsheets</a></h2>

<p>The subsystem_to_spreadsheets command allows the user to output the entire spreadsheet for a set of input subsystems.
The input is a table with a column containing subsystem names.
The output is a table with 4 appended columns</p>

<pre>     [Genome,variant-code,role,fid]</pre>

<p>Example:</p>

<pre>    subsystems_to_spreadsheets [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the subsystem name. If another column contains the subsystem use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call subsystems_to_spreadsheets. It is documented as follows:</p>

<pre>  $return = $obj-&#62;subsystems_to_spreadsheets($subsystems, $genomes)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$subsystems is a subsystems
$genomes is a genomes
$return is a reference to a hash where the key is a subsystem and the value is a reference to a hash where the key is a genome and the value is a row
subsystems is a reference to a list where each element is a subsystem
subsystem is a string
genomes is a reference to a list where each element is a genome
genome is a string
row is a reference to a list containing 2 items:
	0: a variant
	1: a reference to a hash where the key is a role and the value is a fids
variant is a string
role is a string
fids is a reference to a list where each element is a fid
fid is a string


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with 4 extra columns added (Genome,Variant,Role,Fid).</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="text_search"
>text_search</a></h2>

<p>text_search performs a search against a full-text index maintained for the CDMI.
The parameter &#34;input&#34; is the text string to be searched for.
The parameter &#34;entities&#34; defines the entities to be searched.
If the list is empty,
all indexed entities will be searched.
The &#34;start&#34; and &#34;count&#34; parameters limit the results to &#34;count&#34; hits starting at &#34;start&#34;.</p>

<p>Example:</p>

<pre>    text_search [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Documentation_for_underlying_call"
>Documentation for underlying call</a></h3>

<p>This script is a wrapper for the CDMI-API call text_search. It is documented as follows:</p>

<pre>  $return = $obj-&#62;text_search($fids)</pre>

<dl>
<dt><a name="Parameter_and_return_types"
>Parameter and return types</a></dt>

<dd>
<pre>
$fids is a fids
$return is a reference to a hash where the key is a fid and the value is an annotations
fids is a reference to a list where each element is a fid
fid is a string
annotations is a reference to a list where each element is an annotation
annotation is a reference to a list containing 3 items:
	0: a comment
	1: an annotator
	2: an annotation_time
comment is a string
annotator is a string
annotation_time is an int


</pre>

</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
