<h1>Extracting Data from the CS Using the CS-API: Some Typical Examples</h1>
<h2>Introduction</h2>
 We will cover the basic tools for accessing data in the CS via the <a name="CS-API256">CS-API</a>
 (as opposed to the more common use of the command-line tools. We will give small test programs along with displayed output in hopes that users can easily generate what they need by making minor modifications.<br />
<br />
<br />
<br />
<div style="text-align: center;"><big style="font-weight: bold;"><big>Table of Contents</big></big><br /></div>
<ul><li><a href="#mozTocId403269" name="Genome257"> Extracting Data Corresponding to the Genome Entity</a><li><a href="#mozTocId789578" name="Feature258"> Extracting Data Corresponding to the Feature Entity</a><li><a href="#mozTocId477963">Looking Up the Functions Assigned to Fids</a><li><a href="#mozTocId279369" name="Feature260"> Getting the DNA Sequence for Feature Entities</a><li><a href="#mozTocId231552">Extracting the Protein Families that Contain One or More Features</a><li><a href="#mozTocId439283" name="Feature261">Getting Literature Related to a Given Feature</a><li><a href="#mozTocId787030">Accessing the Annotations Associated with One or More Features</a><li><a href="#mozTocId948659" name="Feature262">Locating Features that Tend to Co-occur with a Given Feature</a><li><a href="#mozTocId611557" name="Feature263">Locating Subsystems that Contain a Given Feature</a><li><a href="#mozTocId182481" name="Atomic Regulons264">What Atomic Regulons Contain a fid (if any)?</a><li><a href="#mozTocId300685" name="Coexpressed266">Getting the Pearson Correlation Coefficient for Apparently Coexpressed Features</a><li><a href="#mozTocId57110" name="Protein Sequence267">Finding All Features with the Same Protein Sequence</a><li><a href="#mozTocId586120" name="Family269">Accessing Members of a Protein Family and the Family Functions</a><li><a href="#mozTocId458855" name="Co-occurring270">Getting a Set of Co-occurring Protein Families</a><li><a href="#mozTocId193709" name="Role271">Given a Role, which Subsystems Include it?</a><li><a href="#mozTocId869371">Going from Roles to Complexes to Reactions to Printable Versions of Reactions</a></ul>
<br />
<h3><a class="mozTocH2" name="mozTocId403269"></a> Extracting Data Corresponding to the <b><a name="Genome272">Genome</a></b> Entity</h3>
 You can access data relating to several genomes with a single invocation of <i><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_genome_data" target='_blank'>genomes_to_genome_data</a></i>
, but most of the time you will be after data relating to a single genome (in the example, <i>kb|g.0</i>
). <br />
<pre>use strict;<br />use Data::Dumper;<br />use Bio::KBase::CDMI::CDMIClient;<br />use Bio::KBase::Utilities::ScriptThing;<br />my $csO = Bio::KBase::CDMI::CDMIClient-&gt;new_for_script();<br /><br />my $gH = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_genome_data" target='_blank'>genomes_to_genome_data</a>([&#39;kb|g.0&#39;]);<br />print &amp;Dumper($gH)<br /><br /></pre>
 produces the following output: <br />
<pre>$VAR1 = {<br />          &#39;kb|g.0&#39; =&gt; {<br />                        &#39;rnas&#39; =&gt; &#39;170&#39;,<br />                        &#39;gc_content&#39; =&gt; &#39;50.7888716661698&#39;,<br />                        &#39;dna_size&#39; =&gt; &#39;4639221&#39;,<br />                        &#39;taxonomy&#39; =&gt; &#39;Bacteria; Proteobacteria; Gammaproteobacteria; Enterobacteriales; Enterobacteriaceae; Escherichia; Escherichia coli K-12&#39;,<br />                        &#39;scientific_name&#39; =&gt; &#39;Escherichia coli K12&#39;,<br />                        &#39;contigs&#39; =&gt; &#39;1&#39;,<br />                        &#39;genome_md5&#39; =&gt; &#39;b4ff0a0fea9686b26b4e2a3cf7b6adbf&#39;,<br />                        &#39;pegs&#39; =&gt; &#39;4308&#39;,<br />                        &#39;genetic_code&#39; =&gt; &#39;11&#39;,<br />                        &#39;complete&#39; =&gt; &#39;1&#39;<br />                      }<br />        };<br /><br /></pre>
 You might wish to determine which <a name="subsystems273">subsystems</a>
 include a given genome. To get that, you should use something like <br />
<pre>use strict;<br />use Data::Dumper;<br />use Bio::KBase::CDMI::CDMIClient;<br />use Bio::KBase::Utilities::ScriptThing;<br />my $csO = Bio::KBase::CDMI::CDMIClient-&gt;new_for_script();<br /><br />my $genomes = [&#39;kb|g.0&#39;];<br />my $gH     = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#genomes_to_subsystems" target='_blank'>genomes_to_subsystems</a>($genomes);<br />my $dataH  = {};<br />foreach my $g (@$genomes)<br />{<br />    my $x  = $gH-&gt;{$g};<br />    if (! $x)<br />    {<br />	$dataH-&gt;{$g} = [];<br />    }<br />    else<br />    {<br />	# we reorder the ouput and throw out entries with variants that are not active<br />	my @pairs = map { [$_-&gt;[1],$_-&gt;[0]] } grep { $_-&gt;[0] !~ /^\*?(0|-1)$/ } @$x;<br />	$dataH-&gt;{$g} = \@pairs;<br />    }<br />}<br /><br />print &amp;Dumper($dataH);<br /><br /></pre>
 The data passed back includes <a name="subsystems274">subsystems</a>
 in which the genome has been placed with a variant code indicating that it is not active. You probably will wish to exclude these, and the snippet of code shows you how to do it. Running the code should produce something like <br />
<pre>$VAR1 = {<br />          &#39;kb|g.0&#39; =&gt; [<br />                        [<br />                          &#39;Restriction-Modification System&#39;,<br />                          &#39;1.0&#39;<br />                        ],<br />                        [<br />                          &#39;Citrate Metabolism, Transport, and Regulation&#39;,<br />                          &#39;1&#39;<br />                        ],<br />                        [<br />                          &#39;The usher protein HtrE fimbrial cluster&#39;,<br />                          &#39;1&#39;<br />                        ],<br />                        [<br />                          &#39;Methylglyoxal Metabolism&#39;,<br />                          &#39;3.0&#39;<br />                        ],<br />                        [<br />                          &#39;Synthesis of osmoregulated periplasmic glucans&#39;,<br />                          &#39;1.3&#39;<br />                        ],<br />                        [<br />                          &#39;At5g37530 (CsdL protein family)&#39;,<br />                          &#39;1&#39;<br />                        ],<br />                        [<br />                          &#39;Housecleaning nucleoside triphosphate pyrophosphatases&#39;,<br />                          &#39;1&#39;<br />                        ],<br />                        [<br />                          &#39;A Hypothetical Protein Related to Proline Metabolism&#39;,<br />                          &#39;1&#39;<br />                        ],<br />                        [<br />                          &#39;Rcs phosphorelay signal transduction pathway&#39;,<br />                          &#39;1&#39;<br />                        ],<br />.<br />.<br />.<br />                      ]<br />        };<br /><br /><br /></pre>
 The 2-tuples are [<a name="subsystem275">subsystem</a>
-name,variant-code]. <p>
<h3><a class="mozTocH2" name="mozTocId789578"></a> Extracting Data Corresponding to the <b><a name="Feature276">Feature</a></b> Entity</h3>
 You can get data for one or more <i>Features</i>
 using something like <br />
<pre>use strict;<br />use Data::Dumper;<br />use Bio::KBase::CDMI::CDMIClient;<br />use Bio::KBase::Utilities::ScriptThing;<br />my $csO = Bio::KBase::CDMI::CDMIClient-&gt;new_for_script();<br /><br />my $dataH = &amp;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_feature_data" target='_blank'>fids_to_feature_data</a>($csO,[&#39;kb|g.0.peg.2&#39;,&#39;kb|g.0.peg.3&#39;]);<br />print &amp;Dumper($dataH);<br /><br />sub <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_feature_data" target='_blank'>fids_to_feature_data</a> {<br />    my($csO,$fids) = @_;<br /><br />    my $dataH = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_feature_data" target='_blank'>fids_to_feature_data</a>($fids);<br />    foreach my $fid (keys(%$dataH))<br />    {<br />	$dataH-&gt;{$fid}-&gt;{feature_location} = &amp;loc_to_locstring($dataH-&gt;{$fid}-&gt;{feature_location});<br />    }<br />    return $dataH;<br />}<br /><br />sub loc_to_locstring {<br />    my($loc) = @_;<br /><br />    return join(&quot;,&quot;,map { &quot;$_-&gt;[0]\_$_-&gt;[1]$_-&gt;[2]$_-&gt;[3]&quot; } @$loc);<br />}<br /><br /><br /></pre>
 which produces the following output: <br />
<pre>$VAR1 = {<br />          &#39;kb|g.0.peg.3&#39; =&gt; {<br />                              &#39;feature_id&#39; =&gt; &#39;kb|g.0.peg.3&#39;,<br />                              &#39;feature_publications&#39; =&gt; [],<br />                              &#39;feature_length&#39; =&gt; &#39;1986&#39;,<br />                              &#39;feature_location&#39; =&gt; &#39;kb|g.0.c.1_1342781-1986&#39;,<br />                              &#39;genome_name&#39; =&gt; &#39;Escherichia coli K12&#39;,<br />                              &#39;feature_function&#39; =&gt; &#39;hypothetical protein&#39;<br />                            },<br />          &#39;kb|g.0.peg.2&#39; =&gt; {<br />                              &#39;feature_id&#39; =&gt; &#39;kb|g.0.peg.2&#39;,<br />                              &#39;feature_publications&#39; =&gt; [<br />                                                          [<br />                                                            &#39;11527384&#39;,<br />                                                            &#39;http://www.ncbi.nlm.nih.gov/pubmed/11527384&#39;,<br />                                                            &#39;CueO is a multi-copper oxidase that confers copper tolerance in Escherichia coli.&#39;<br />                                                          ],<br />                                                          [<br />                                                            &#39;11867755&#39;,<br />                                                            &#39;http://www.ncbi.nlm.nih.gov/pubmed/11867755&#39;,<br />                                                            &#39;Crystal structure and electron transfer kinetics of CueO, a multicopper oxidase required for copper homeostasis in Escherichia coli.&#39;<br />                                                          ]<br />                                                        ],<br />                              &#39;feature_length&#39; =&gt; &#39;1551&#39;,<br />                              &#39;feature_location&#39; =&gt; &#39;kb|g.0.c.1_137083+1551&#39;,<br />                              &#39;genome_name&#39; =&gt; &#39;Escherichia coli K12&#39;,<br />                              &#39;feature_function&#39; =&gt; &#39;Blue copper oxidase CueO precursor&#39;<br />                            }<br />        };<br /><br /></pre>
 Note that we converted the <i>location</i>
 from a list of regions to a printable string. You may well wish to retain it in its list format, if you intend to work with it. <h3><a class="mozTocH2" name="mozTocId477963"></a>Looking Up the Functions Assigned to Fids</h3>
 To find out the current functions assigned to one or more fids, use something like <br />
<pre>use strict;<br />use Data::Dumper;<br />use Bio::KBase::CDMI::CDMIClient;<br />use Bio::KBase::Utilities::ScriptThing;<br />my $csO = Bio::KBase::CDMI::CDMIClient-&gt;new_for_script();<br /><br />my $fids  = [&#39;kb|g.0.peg.2&#39;,&#39;kb|g.0.peg.4&#39;];<br /><br />my $fidH = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_functions" target='_blank'>fids_to_functions</a>($fids);<br />print &amp;Dumper($fidH);<br /><br /></pre>
 This prodcuces output similar to <br />
<pre>$VAR1 = {<br />          &#39;kb|g.0.peg.2&#39; =&gt; &#39;Blue copper oxidase CueO precursor&#39;,<br />          &#39;kb|g.0.peg.4&#39; =&gt; &#39;L,D-transpeptidase YcfS&#39;<br />        };<br /><br /></pre>
<h3><a class="mozTocH2" name="mozTocId279369"></a> Getting the DNA Sequence for <b><a name="Feature277">Feature</a></b> Entities</h3>
 The following short piece of code extracts the <a name="DNA278">DNA</a>
 sequences of two features and reformats them as fasta entries. <br />
<pre>use strict;<br />use Data::Dumper;<br />use Bio::KBase::CDMI::CDMIClient;<br />use Bio::KBase::Utilities::ScriptThing;<br />my $csO = Bio::KBase::CDMI::CDMIClient-&gt;new_for_script();<br /><br />my $dataH = &amp;ids_to_dna_sequences($csO,[&#39;kb|g.0.peg.2&#39;,&#39;kb|g.0.peg.3&#39;]);<br />print &amp;Dumper($dataH);<br /><br />sub ids_to_dna_sequences {<br />    my($csO,$fids) = @_;<br /><br />    my $to_seqH  = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_dna_sequences" target='_blank'>fids_to_dna_sequences</a>($fids);<br />    my $dataH  = {};<br />    foreach my $fid (@$fids)<br />    {<br />	my $seq = $to_seqH-&gt;{$fid};<br />	$dataH-&gt;{$fid} = &amp;to_fasta($fid,$seq);<br />    }<br />    return $dataH;<br />}<br /><br />sub to_fasta {<br />    my($id,$seq) = @_;<br /><br />    return &amp;SeedUtils::create_fasta_record($id,&#39;&#39;,$seq);<br />}<br /><br /><br /></pre>
 It produces the following output: <br />
<pre>$VAR1 = {<br />          &#39;kb|g.0.peg.3&#39; =&gt; &#39;&gt;kb|g.0.peg.3<br />atgaaaaccgttagggagtccacaacgttgtacaactttctcggatcgcacaatccatac<br />tggcggttgacggaaagcagcgatgttttgcgcttttctaccaccgaaaccacagaacct<br />gatcgtacattgcagttatctgccgaacaggctgctcgcatcagggaaatgacggtcatc<br />acctccagcctgatgatgagtctgaccgtcgatgaaagcgatctttctgtgcatctggta<br />ggacgaaaaatcaataaacgggaatgggctggcaacgcgtctgcatggcatgacacaccg<br />gcggttgctcgtgatttatcacacgggctttcctttgctgagcaggtagtttctgaagca<br />cattccgcaatagtgattctcgacagccgggggaatatccaacgcttcaatcggttatgt<br />gaagattacacagggttgaaagaacacgacgtcattgggcaaagcgtgtttaaactgttt<br />atgagccgtcgtgaagctgcggcatccaggcgcaataaccgtgtattttttcgaagcggc<br />aatgcatatgaagtcgaactgtggataccaacatgtaaaggccagcggctgtttctgttt<br />cgcaataaatttgtccacagcggcagtggcaaaaacgagatttttttaatctgttccggc<br />accgacattaccgaagagcgccgcgctcaggagcgactgcgtattctggcaaataccgac<br />agtatcaccggactgccgaatcgtaacgcaatgcaggatttaatcgatcacgctattaat<br />catgcagataacaataaagttggggttgtgtatcttgatttggataatttcaaaaaggtc<br />aacgacgcctatgggcatttgtttggtgaccagttattacgcgacgtgtcattggctatt<br />ttaagctgtctcgaacatgaccaggtgttggcgcgtccaggtggggatgagtttctggta<br />ctggcatccaacacctcacaaagcgcgctggaagcaatggcatcacgaattttgacccgc<br />ttacggctcccctttcgcattggtttaattgaagtttataccagctgttcagtaggtatt<br />gcactctctcccgaacatggttcagacagcacggctattattcgtcacgccgacacagca<br />atgtacacagcgaaggaaggcggacgaggacaattttgcgtttttaccccagaaatgaat<br />caacgggtatttgaatatctctggctggataccaacttgcgtaaagcactggaaaacgat<br />cagttggttattcactatcaaccgaaaatcacctggcgtggcgaagtgcgcagtctggaa<br />gcactagtacgttggcagtcacctgaacgtgggttgattccaccgttggacttcatttcc<br />tacgccgaagagtcagggctaattgtgcctttaggccgttgggtgattctcgatgtcgta<br />cgccaggtggcaaagtggcgggataaaggcataaacctgcgagtggcggtaaatatttct<br />gcacgtcagctcgccgatcaaaccattttcaccgccctgaaacaggttctccaggaactc<br />aattttgaatactgccctatagatgttgaactgacagagagttgtctgattgagaatgat<br />gaactggcactgtctgttattcaacaatttagccaactaggtgcgcaagtgcatctggat<br />gattttggtaccggctactcttcactttcgcaactggcgcgctttccgatcgatgccatc<br />aaacttgaccaggtttttgttcgagatattcacaaacaacctgtctcgcagtcactggtc<br />cgggcgatcgtcgctgtggcccaggcattgaatcttcaggtgatcgccgaaggtgtagag<br />agtgcaaaggaagatgcttttttaaccaagaacgggatcaatgagcggcaaggatttttg<br />tttgccaaaccgatgcccgccgtcgccttcgaacgctggtataaacgctatctgaagcgc<br />gcataa<br />&#39;,<br />          &#39;kb|g.0.peg.2&#39; =&gt; &#39;&gt;kb|g.0.peg.2<br />atgcaacgtcgtgatttcttaaaatattccgtcgcgctgggtgtggcttcggctttgccg<br />ctgtggagccgcgcagtatttgcggcagaacgcccaacgttaccgatccctgatttgctc<br />acgaccgatgcccgtaatcgcattcagttaactattggcgcaggccagtccacctttggc<br />gggaaaactgcaactacctggggctataacggcaatctgctggggccggcggtgaaatta<br />cagcgcggcaaagcggtaacggttgatatctacaaccaactgacggaagagacaacgttg<br />cactggcacgggctggaagtaccgggtgaagtcgacggcggcccgcagggaattattccg<br />ccaggtggcaagcgctcggtgacgttgaacgttgatcaacctgccgctacctgctggttc<br />catccgcatcagcacggcaaaaccgggcgacaggtggcgatggggctggctgggctggtg<br />gtgattgaagatgacgagatcctgaaattaatgctgccaaaacagtggggtatcgatgat<br />gttccggtgatcgttcaggataagaaatttagcgccgacgggcagattgattatcaactg<br />gatgtgatgaccgccgccgtgggctggtttggcgatacgttgctgaccaacggtgcaatc<br />tacccgcaacacgctgccccgcgtggttggctgcgcctgcgtttgctcaatggctgtaat<br />gcccgttcgctcaatttcgccaccagcgacaatcgcccgctgtatgtgattgccagcgac<br />ggtggtctgctacctgaaccagtgaaggtgagcgaactgccggtgctgatgggcgagcgt<br />tttgaagtgctggtggaggttaacgataacaaaccctttgacctggtgacgctgccggtc<br />agccagatggggatggcgattgcgccgtttgataagcctcatccggtaatgcggattcag<br />ccgattgctattagtgcctccggtgctttgccagacacattaagtagcctgcctgcgtta<br />ccttcgctggaagggctgacggtacgcaagctgcaactctctatggacccgatgctcgat<br />atgatggggatgcagatgctaatggagaaatatggcgatcaggcgatggccgggatggat<br />cacagccagatgatgggccatatggggcacggcaatatgaatcatatgaaccacggcggg<br />aagttcgatttccaccatgccaacaaaatcaacggtcaggcgtttgatatgaacaagccg<br />atgtttgcggcggcgaaagggcaatacgaacgttgggttatctctggcgtgggcgacatg<br />atgctgcatccgttccatatccacggcacgcagttccgtatcttgtcagaaaatggcaaa<br />ccgccagcggctcatcgcgcgggctggaaagataccgttaaggtagaaggtaatgtcagc<br />gaagtgctggtgaagtttaatcacgatgcaccgaaagaacatgcttatatggcgcactgc<br />catctgctggagcatgaagatacggggatgatgttagggtttacggtataa<br />&#39;<br />        };<br /><br /><br /></pre>
 Changing <i><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_dna_sequences" target='_blank'>fids_to_dna_sequences</a></i>
 to <i><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_protein_sequences" target='_blank'>fids_to_protein_sequences</a></i>
 would produce <br />
<pre>$VAR1 = {<br />          &#39;kb|g.0.peg.3&#39; =&gt; &#39;&gt;kb|g.0.peg.3<br />MKTVRESTTLYNFLGSHNPYWRLTESSDVLRFSTTETTEPDRTLQLSAEQAARIREMTVI<br />TSSLMMSLTVDESDLSVHLVGRKINKREWAGNASAWHDTPAVARDLSHGLSFAEQVVSEA<br />HSAIVILDSRGNIQRFNRLCEDYTGLKEHDVIGQSVFKLFMSRREAAASRRNNRVFFRSG<br />NAYEVELWIPTCKGQRLFLFRNKFVHSGSGKNEIFLICSGTDITEERRAQERLRILANTD<br />SITGLPNRNAMQDLIDHAINHADNNKVGVVYLDLDNFKKVNDAYGHLFGDQLLRDVSLAI<br />LSCLEHDQVLARPGGDEFLVLASNTSQSALEAMASRILTRLRLPFRIGLIEVYTSCSVGI<br />ALSPEHGSDSTAIIRHADTAMYTAKEGGRGQFCVFTPEMNQRVFEYLWLDTNLRKALEND<br />QLVIHYQPKITWRGEVRSLEALVRWQSPERGLIPPLDFISYAEESGLIVPLGRWVILDVV<br />RQVAKWRDKGINLRVAVNISARQLADQTIFTALKQVLQELNFEYCPIDVELTESCLIEND<br />ELALSVIQQFSQLGAQVHLDDFGTGYSSLSQLARFPIDAIKLDQVFVRDIHKQPVSQSLV<br />RAIVAVAQALNLQVIAEGVESAKEDAFLTKNGINERQGFLFAKPMPAVAFERWYKRYLKR<br />A<br />&#39;,<br />          &#39;kb|g.0.peg.2&#39; =&gt; &#39;&gt;kb|g.0.peg.2<br />MQRRDFLKYSVALGVASALPLWSRAVFAAERPTLPIPDLLTTDARNRIQLTIGAGQSTFG<br />GKTATTWGYNGNLLGPAVKLQRGKAVTVDIYNQLTEETTLHWHGLEVPGEVDGGPQGIIP<br />PGGKRSVTLNVDQPAATCWFHPHQHGKTGRQVAMGLAGLVVIEDDEILKLMLPKQWGIDD<br />VPVIVQDKKFSADGQIDYQLDVMTAAVGWFGDTLLTNGAIYPQHAAPRGWLRLRLLNGCN<br />ARSLNFATSDNRPLYVIASDGGLLPEPVKVSELPVLMGERFEVLVEVNDNKPFDLVTLPV<br />SQMGMAIAPFDKPHPVMRIQPIAISASGALPDTLSSLPALPSLEGLTVRKLQLSMDPMLD<br />MMGMQMLMEKYGDQAMAGMDHSQMMGHMGHGNMNHMNHGGKFDFHHANKINGQAFDMNKP<br />MFAAAKGQYERWVISGVGDMMLHPFHIHGTQFRILSENGKPPAAHRAGWKDTVKVEGNVS<br />EVLVKFNHDAPKEHAYMAHCHLLEHEDTGMMLGFTV<br />&#39;<br />        };<br /><br /></pre>
<h3><a class="mozTocH2" name="mozTocId231552"></a>Extracting the Protein Families that Contain One or More Features</h3>
 If you wish to find out what <a name="protein families279">protein families</a>
 include a specific feature, you should use <i><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_protein_families" target='_blank'>fids_to_protein_families</a></i>
. Here is a simple example that looks up the <a name="families280">families</a>
 for each of two fids: <br />
<pre>use strict;<br />use Data::Dumper;<br />use Bio::KBase::CDMI::CDMIClient;<br />use Bio::KBase::Utilities::ScriptThing;<br />my $csO = Bio::KBase::CDMI::CDMIClient-&gt;new_for_script();<br /><br />my $ffH = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_protein_families" target='_blank'>fids_to_protein_families</a>([&#39;kb|g.1841.peg.788&#39;,&#39;kb|g.1841.peg.368&#39;]);<br />print &amp;Dumper($ffH);<br /><br /><br /></pre>
 Running the program should produce somnething like <br />
<pre>$VAR1 = {<br />          &#39;kb|g.1841.peg.368&#39; =&gt; [<br />                                   &#39;FIG00001992&#39;<br />                                 ],<br />          &#39;kb|g.1841.peg.788&#39; =&gt; [<br />                                   &#39;FIG01303839&#39;<br />                                 ]<br />        };<br /><br /></pre>
<h3><a class="mozTocH2" name="mozTocId439283"></a>Getting Literature Related to a Given Feature</h3>
 If you wish to get the <a name="literature281">literature</a>
 connected to a set of fids, you can use <i><a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_literature" target='_blank'>fids_to_literature</a></i>
: <br />
<pre>use strict;<br />use Data::Dumper;<br />use Bio::KBase::CDMI::CDMIClient;<br />use Bio::KBase::Utilities::ScriptThing;<br />my $csO = Bio::KBase::CDMI::CDMIClient-&gt;new_for_script();<br /><br />my $litH = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_literature" target='_blank'>fids_to_literature</a>([&#39;kb|g.0.peg.2&#39;,&#39;kb|g.0.peg.3&#39;,&#39;kb|g.0.peg.4&#39;]);<br />print &amp;Dumper($litH);<br /><br /><br /></pre>
 This produces <br />
<pre>$VAR1 = {<br />          &#39;kb|g.0.peg.2&#39; =&gt; [<br />                              [<br />                                &#39;11527384&#39;,<br />                                &#39;http://www.ncbi.nlm.nih.gov/pubmed/11527384&#39;,<br />                                &#39;CueO is a multi-copper oxidase that confers copper tolerance in Escherichia coli.&#39;<br />                              ],<br />                              [<br />                                &#39;11867755&#39;,<br />                                &#39;http://www.ncbi.nlm.nih.gov/pubmed/11867755&#39;,<br />                                &#39;Crystal structure and electron transfer kinetics of CueO, a multicopper oxidase required for copper homeostasis in Escherichia coli.&#39;<br />                              ]<br />                            ]<br />        };<br /><br /><br /></pre>
 Note that two of the three features had no connected <a name="literature282">literature</a>
 and were left out of the returned hash. <h3><a class="mozTocH2" name="mozTocId787030"></a>Accessing the Annotations Associated with One or More Features</h3>
 Features may have attached <a name="annotations283">annotations</a>
. Annotations are thought of as 3-tuples: <ol><li>a text string (often describing a change in assigned function), <li>who made the annotation, and <li>a timestamp giving the precise time at which the annotation was made. </ol>
 In the little example program we show, the timestamp is expanded into a readable form for one of the features: <br />
<pre>use strict;<br />use Data::Dumper;<br />use Bio::KBase::CDMI::CDMIClient;<br />use Bio::KBase::Utilities::ScriptThing;<br />my $csO = Bio::KBase::CDMI::CDMIClient-&gt;new_for_script();<br /><br />my $annH = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_annotations" target='_blank'>fids_to_annotations</a>([&#39;kb|g.0.peg.2&#39;,&#39;kb|g.0.peg.3&#39;,&#39;kb|g.0.peg.4&#39;]);<br />my @tab  = map { [scalar localtime($_-&gt;[2]),$_-&gt;[1],$_-&gt;[0]] } <br />           sort { ($b-&gt;[2] &lt;=&gt; $a-&gt;[2]) or ($a-&gt;[0] cmp $b-&gt;[0]) } <br />           @{$annH-&gt;{&#39;kb|g.0.peg.2&#39;}};<br />print &amp;Dumper($annH,\@tab);<br /><br /><br /></pre>
 This produces as output <br />
<pre>$VAR1 = {<br />          &#39;kb|g.0.peg.3&#39; =&gt; [<br />                              [<br />                                &#39;Set function to<br />hypothetical protein&#39;,<br />                                &#39;EC&#39;,<br />                                &#39;1116992974&#39;<br />                              ],<br />                              [<br />                                &#39;Set function to<br />Sensory box/GGDEF family protein&#39;,<br />                                &#39;master&#39;,<br />                                &#39;1088696623&#39;<br />                              ]<br />                            ],<br />          &#39;kb|g.0.peg.2&#39; =&gt; [<br />                              [<br />                                &#39;Set function to<br />Blue copper oxidase CueO precursor&#39;,<br />                                &#39;OlgaV&#39;,<br />                                1233354582<br />                              ],<br />                              [<br />                                &#39;Role changed from \&#39;Blue copper oxidase cueO precursor\&#39; to \&#39;Blue copper oxidase CueO precursor\&#39;&#39;,<br />                                &#39;OlgaV&#39;,<br />                                1233354582<br />                              ],<br />                              [<br />                                &#39;Function set by OlgaV at 1233354582<br />Blue copper oxidase CueO precursor&#39;,<br />                                &#39;OlgaV&#39;,<br />                                1233354582<br />                              ]<br />                            ],<br />          &#39;kb|g.0.peg.4&#39; =&gt; [<br />                              [<br />                                &#39;Set function to<br />L,D-transpeptidase YcfS&#39;,<br />                                &#39;claudia&#39;,<br />                                &#39;1215629062&#39;<br />                              ],<br />                              [<br />                                &#39;Set function to<br />L,D-transpeptidase YcfS&#39;,<br />                                &#39;claudia&#39;,<br />                                &#39;1215628461&#39;<br />                              ],<br />                              [<br />                                &#39;L,D-transpeptidase YcfS&#39;,<br />                                &#39;claudia&#39;,<br />                                &#39;1215628461&#39;<br />                              ],<br />                              [<br />                                &#39;Set function to<br />Protein erfK/srfK precursor&#39;,<br />                                &#39;claudia&#39;,<br />                                &#39;1215627140&#39;<br />                              ],<br />                              [<br />                                &#39;Set function to<br />hypothetical protein&#39;,<br />                                &#39;EC&#39;,<br />                                &#39;1116992961&#39;<br />                              ],<br />                              [<br />                                &#39;Set function to<br />Protein erfK/srfK precursor&#39;,<br />                                &#39;master&#39;,<br />                                &#39;1088690624&#39;<br />                              ]<br />                            ]<br />        };<br />$VAR2 = [<br />          [<br />            &#39;Fri Jan 30 16:29:42 2009&#39;,<br />            &#39;OlgaV&#39;,<br />            &#39;Function set by OlgaV at 1233354582<br />Blue copper oxidase CueO precursor&#39;<br />          ],<br />          [<br />            &#39;Fri Jan 30 16:29:42 2009&#39;,<br />            &#39;OlgaV&#39;,<br />            &#39;Role changed from \&#39;Blue copper oxidase cueO precursor\&#39; to \&#39;Blue copper oxidase CueO precursor\&#39;&#39;<br />          ],<br />          [<br />            &#39;Fri Jan 30 16:29:42 2009&#39;,<br />            &#39;OlgaV&#39;,<br />            &#39;Set function to<br />Blue copper oxidase CueO precursor&#39;<br />          ]<br />        ];<br /><br /></pre>
 Note that the text of <a name="annotations284">annotations</a>
 often include newlines. <h3><a class="mozTocH2" name="mozTocId948659"></a>Locating Features that Tend to Co-occur with a Given Feature</h3>
 Preserved contiguity on the <a name="chromosome285">chromosome</a>
 is one of the more important clues relating to function. To find the fids that appear to be part of a <a name="conserved neighborhood286">conserved neighborhood</a>
, use something like <br />
<pre>use strict;<br />use Data::Dumper;<br />use Bio::KBase::CDMI::CDMIClient;<br />use Bio::KBase::Utilities::ScriptThing;<br />my $csO = Bio::KBase::CDMI::CDMIClient-&gt;new_for_script();<br /><br />my $dataH =  $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_co_occurring_fids" target='_blank'>fids_to_co_occurring_fids</a>([&#39;kb|g.1841.peg.788&#39;]);<br />print &amp;Dumper($dataH);<br /><br /><br /></pre>
 This would produce <br />
<pre>$VAR1 = {<br />          &#39;kb|g.1841.peg.788&#39; =&gt; [<br />                                   [<br />                                     &#39;kb|g.1841.peg.659&#39;,<br />                                     &#39;13&#39;<br />                                   ],<br />                                   [<br />                                     &#39;kb|g.1841.peg.896&#39;,<br />                                     &#39;43&#39;<br />                                   ],<br />                                   [<br />                                     &#39;kb|g.1841.peg.837&#39;,<br />                                     &#39;14&#39;<br />                                   ],<br />                                   [<br />                                     &#39;kb|g.1841.peg.368&#39;,<br />                                     &#39;23&#39;<br />                                   ]<br />                                 ]<br />        };<br /><br /></pre>
 The list returned for <i>kb|g.1841.<a name="peg287">peg</a>.788</i>
 is a list of 2-tuples: [fid,score]. The score is computed as the number of distinct OTUs in which genes that look quite similar occur close to one another on the <a name="chromosome288">chromosome</a>
. <h3><a class="mozTocH2" name="mozTocId611557"></a>Locating Subsystems that Contain a Given Feature</h3>
 Suppose that you have a <a name="Feature289">Feature</a>
 (i.e., a fid), and you would like to know what <a name="subsystems290">subsystems</a>
 contain it. You might use something like <br />
<pre>use strict;<br />use Data::Dumper;<br />use Bio::KBase::CDMI::CDMIClient;<br />use Bio::KBase::Utilities::ScriptThing;<br />my $csO = Bio::KBase::CDMI::CDMIClient-&gt;new_for_script();<br /><br />my $fidH = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_subsystem_data" target='_blank'>fids_to_subsystem_data</a>([&#39;kb|g.1841.peg.2636&#39;,&#39;kb|g.1841.peg.3010&#39;]);<br />print &amp;Dumper($fidH);<br /><br /><br /></pre>
 which would produce something like <br />
<pre>$VAR1 = {<br />          &#39;kb|g.1841.peg.2636&#39; =&gt; [<br />                                    [<br />                                      &#39;Arginine Deiminase Pathway&#39;,<br />                                      &#39;1.x&#39;,<br />                                      &#39;Carbamate kinase (EC 2.7.2.2)&#39;<br />                                    ],<br />                                    [<br />                                      &#39;Arginine and Ornithine Degradation&#39;,<br />                                      &#39;1.1234&#39;,<br />                                      &#39;Carbamate kinase (EC 2.7.2.2)&#39;<br />                                    ],<br />                                    [<br />                                      &#39;Polyamine Metabolism&#39;,<br />                                      &#39;1&#39;,<br />                                      &#39;Carbamate kinase (EC 2.7.2.2)&#39;<br />                                    ]<br />                                  ],<br />          &#39;kb|g.1841.peg.3010&#39; =&gt; [<br />                                    [<br />                                      &#39;Arginine Biosynthesis extended&#39;,<br />                                      &#39;1.0&#39;,<br />                                      &#39;Ornithine carbamoyltransferase (EC 2.1.3.3)&#39;<br />                                    ],<br />                                    [<br />                                      &#39;Arginine Deiminase Pathway&#39;,<br />                                      &#39;1.x&#39;,<br />                                      &#39;Ornithine carbamoyltransferase (EC 2.1.3.3)&#39;<br />                                    ],<br />                                    [<br />                                      &#39;Arginine and Ornithine Degradation&#39;,<br />                                      &#39;1.1234&#39;,<br />                                      &#39;Ornithine carbamoyltransferase (EC 2.1.3.3)&#39;<br />                                    ]<br />                                  ]<br />        };<br /><br /></pre>
 That is, for each fid in the list you give as input, you get back a pointer to a list of 3-tuples. Each 3-tuple contains <ol><li>a <a name="subsystem291">subsystem</a> name, <li>a variant code associated with the row in the <a name="subsystem292">subsystem</a> containing the fid, and <li>the role associated with the column containing the fid. </ol>
<h3><a class="mozTocH2" name="mozTocId182481"></a>What Atomic Regulons Contain a fid (if any)?</h3>
 For a very few genomes, we have substantial expression data. For these we generated expression profiles and tried to form sets of genes with identical expression profiles. We called these <i>atomic <a name="regulons293">regulons</a></i>
. You can see whether or not features are in atomic <a name="regulons294">regulons</a>
 using <br />
<pre>use strict;<br />use Data::Dumper;<br />use Bio::KBase::CDMI::CDMIClient;<br />use Bio::KBase::Utilities::ScriptThing;<br />my $csO = Bio::KBase::CDMI::CDMIClient-&gt;new_for_script();<br /><br />my $fidH = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_atomic_regulons" target='_blank'>fids_to_atomic_regulons</a>([&#39;kb|g.0.peg.2&#39;,&#39;kb|g.0.peg.20&#39;]);<br />print &amp;Dumper($fidH);<br /><br /><br /></pre>
 which produces something like <br />
<pre>$VAR1 = {<br />          &#39;kb|g.0.peg.20&#39; =&gt; [<br />                               [<br />                                 &#39;kb|g.0.ar.86&#39;,<br />                                 5<br />                               ]<br />                             ]<br />        };<br /><br /><br /></pre>
 Suppose that you wished to find out which other fids are in atomic <a name="regulon295">regulon</a>
 <i>kb|g.0.ar.86</i>
. You could use <br />
<pre>use strict;<br />use Data::Dumper;<br />use Bio::KBase::CDMI::CDMIClient;<br />use Bio::KBase::Utilities::ScriptThing;<br />my $csO = Bio::KBase::CDMI::CDMIClient-&gt;new_for_script();<br /><br />my $ar      = &#39;kb|g.0.ar.86&#39;;<br />my $arH     = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#atomic_regulons_to_fids" target='_blank'>atomic_regulons_to_fids</a>([$ar]);<br />my $fids    = $arH-&gt;{$ar};<br />my $funcH   = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_functions" target='_blank'>fids_to_functions</a>($fids);<br />print &amp;Dumper($fids,$funcH);<br /><br /></pre>
 which produces <br />
<pre>$VAR1 = [<br />          &#39;kb|g.0.peg.146&#39;,<br />          &#39;kb|g.0.peg.20&#39;,<br />          &#39;kb|g.0.peg.205&#39;,<br />          &#39;kb|g.0.peg.240&#39;,<br />          &#39;kb|g.0.peg.66&#39;<br />        ];<br />$VAR2 = {<br />          &#39;kb|g.0.peg.66&#39; =&gt; &#39;Oligopeptide transport system permease protein OppC (TC 3.A.1.5.1)&#39;,<br />          &#39;kb|g.0.peg.205&#39; =&gt; &#39;Oligopeptide ABC transporter, periplasmic oligopeptide-binding protein OppA (TC 3.A.1.5.1)&#39;,<br />          &#39;kb|g.0.peg.240&#39; =&gt; &#39;Oligopeptide transport system permease protein OppB (TC 3.A.1.5.1)&#39;,<br />          &#39;kb|g.0.peg.146&#39; =&gt; &#39;Oligopeptide transport ATP-binding protein OppD (TC 3.A.1.5.1)&#39;,<br />          &#39;kb|g.0.peg.20&#39; =&gt; &#39;Oligopeptide transport ATP-binding protein OppF (TC 3.A.1.5.1)&#39;<br />        };<br /><br /><br /></pre>
<h3><a class="mozTocH2" name="mozTocId300685"></a>Getting the Pearson Correlation Coefficient for Apparently Coexpressed Features</h3>
 Suppose that you had fids from one of the somewhat rare genomes for which we have substantial expression data. To get the PCC values for apparently co-expressed fids, use <br />
<pre>use strict;<br />use Data::Dumper;<br />use Bio::KBase::CDMI::CDMIClient;<br />use Bio::KBase::Utilities::ScriptThing;<br />my $csO = Bio::KBase::CDMI::CDMIClient-&gt;new_for_script();<br /><br />my $fids   = [&#39;kb|g.0.peg.2257&#39;,&#39;kb|g.0.peg.1094&#39;];<br />my $coexpH = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_coexpressed_fids" target='_blank'>fids_to_coexpressed_fids</a>($fids);<br />print &amp;Dumper($coexpH);<br /><br /></pre>
 You will get a list of the features with scores that have an absolute value greater than or equal to 0.5. <br />
<pre>$VAR1 = {<br />          &#39;kb|g.0.peg.2257&#39; =&gt; [<br />                                 [<br />                                   &#39;kb|g.0.peg.1094&#39;,<br />                                   &#39;0.501&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.865&#39;,<br />                                   &#39;0.515&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.2113&#39;,<br />                                   &#39;0.524&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.1497&#39;,<br />                                   &#39;0.531&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.1388&#39;,<br />                                   &#39;0.546&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.1691&#39;,<br />                                   &#39;0.883&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.1401&#39;,<br />                                   &#39;0.897&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.2004&#39;,<br />                                   &#39;0.767&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.2108&#39;,<br />                                   &#39;0.769&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.2681&#39;,<br />                                   &#39;0.512&#39;<br />                                 ]<br />                               ],<br />          &#39;kb|g.0.peg.1094&#39; =&gt; [<br />                                 [<br />                                   &#39;kb|g.0.peg.185&#39;,<br />                                   &#39;0.582&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.939&#39;,<br />                                   &#39;0.516&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.547&#39;,<br />                                   &#39;0.504&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.518&#39;,<br />                                   &#39;0.618&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.627&#39;,<br />                                   &#39;0.595&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.1338&#39;,<br />                                   &#39;0.502&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.510&#39;,<br />                                   &#39;0.522&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.966&#39;,<br />                                   &#39;0.607&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.974&#39;,<br />                                   &#39;0.561&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.793&#39;,<br />                                   &#39;0.529&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.813&#39;,<br />                                   &#39;0.536&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.600&#39;,<br />                                   &#39;0.557&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.488&#39;,<br />                                   &#39;0.573&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.671&#39;,<br />                                   &#39;0.678&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.786&#39;,<br />                                   &#39;0.612&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.2307&#39;,<br />                                   &#39;0.513&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.1781&#39;,<br />                                   &#39;0.505&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.2282&#39;,<br />                                   &#39;0.597&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.1578&#39;,<br />                                   &#39;0.549&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.1422&#39;,<br />                                   &#39;0.547&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.2071&#39;,<br />                                   &#39;0.561&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.2309&#39;,<br />                                   &#39;0.514&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.1583&#39;,<br />                                   &#39;0.522&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.2331&#39;,<br />                                   &#39;0.531&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.1371&#39;,<br />                                   &#39;0.530&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.1656&#39;,<br />                                   &#39;0.545&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.2257&#39;,<br />                                   &#39;0.501&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.2359&#39;,<br />                                   &#39;0.513&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.3081&#39;,<br />                                   &#39;0.541&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.2681&#39;,<br />                                   &#39;0.603&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.2473&#39;,<br />                                   &#39;0.582&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.3335&#39;,<br />                                   &#39;0.588&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.2615&#39;,<br />                                   &#39;-0.525&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.2474&#39;,<br />                                   &#39;0.522&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.2388&#39;,<br />                                   &#39;0.603&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.3278&#39;,<br />                                   &#39;0.675&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.3283&#39;,<br />                                   &#39;0.596&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.3795&#39;,<br />                                   &#39;0.505&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.3573&#39;,<br />                                   &#39;0.572&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.3809&#39;,<br />                                   &#39;0.525&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.3576&#39;,<br />                                   &#39;0.543&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.3387&#39;,<br />                                   &#39;0.516&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.3937&#39;,<br />                                   &#39;0.533&#39;<br />                                 ],<br />                                 [<br />                                   &#39;kb|g.0.peg.4032&#39;,<br />                                   &#39;0.636&#39;<br />                                 ]<br />                               ]<br />        };<br /><br /></pre>
<h3><a class="mozTocH2" name="mozTocId57110"></a>Finding All Features with the Same Protein Sequence</h3>
 Suppose that you have a fid, and you want to compute the set of equivalent fids (in the sense that they have exactly the same protein sequence). You might use something like <br />
<pre>use strict;<br />use Data::Dumper;<br />use Bio::KBase::CDMI::CDMIClient;<br />use Bio::KBase::Utilities::ScriptThing;<br />my $csO = Bio::KBase::CDMI::CDMIClient-&gt;new_for_script();<br /><br />my $fid  = &#39;kb|g.0.peg.2&#39;;<br /><br />my $fidH = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_proteins" target='_blank'>fids_to_proteins</a>([$fid]);<br />my $md5  = $fidH-&gt;{$fid};<br />my $md5H = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#proteins_to_fids" target='_blank'>proteins_to_fids</a>([$md5]);<br />my $fids = $md5H-&gt;{$md5};<br />print &amp;Dumper($fids);<br /><br /><br /></pre>
 That is, first you lookup the <a name="md5296">md5</a>
 value for the protein sequence of the given fid, and then you ask for the set of fids with that <a name="md5297">md5</a>
 value for their protein sequence. Running the little program produces something like <br />
<pre>$VAR1 = [<br />          &#39;kb|g.0.peg.2&#39;,<br />          &#39;kb|g.10012.peg.2654&#39;,<br />          &#39;kb|g.1609.peg.1954&#39;,<br />          &#39;kb|g.1610.peg.2264&#39;,<br />          &#39;kb|g.1748.peg.3345&#39;,<br />          &#39;kb|g.1870.peg.354&#39;,<br />          &#39;kb|g.2136.peg.2112&#39;,<br />          &#39;kb|g.2226.peg.2607&#39;,<br />          &#39;kb|g.2295.peg.2245&#39;,<br />          &#39;kb|g.2590.peg.869&#39;,<br />          &#39;kb|g.2806.peg.1989&#39;,<br />          &#39;kb|g.2810.peg.2351&#39;,<br />          &#39;kb|g.2891.peg.4128&#39;,<br />          &#39;kb|g.2893.peg.2189&#39;,<br />          &#39;kb|g.2942.peg.2402&#39;,<br />          &#39;kb|g.3205.peg.2854&#39;,<br />          &#39;kb|g.3206.peg.2686&#39;,<br />          &#39;kb|g.3207.peg.82&#39;,<br />          &#39;kb|g.3211.peg.3447&#39;,<br />          &#39;kb|g.3505.peg.2125&#39;,<br />          &#39;kb|g.3508.peg.3768&#39;,<br />          &#39;kb|g.3511.peg.4795&#39;,<br />          &#39;kb|g.3558.peg.499&#39;,<br />          &#39;kb|g.842.peg.645&#39;,<br />          &#39;kb|g.844.peg.423&#39;,<br />          &#39;kb|g.9268.peg.514&#39;,<br />          &#39;kb|g.9269.peg.538&#39;,<br />          &#39;kb|g.9450.peg.3258&#39;,<br />          &#39;kb|g.954.peg.149&#39;,<br />          &#39;kb|g.955.peg.594&#39;,<br />          &#39;kb|g.9604.peg.3253&#39;,<br />          &#39;kb|g.976.peg.3756&#39;,<br />          &#39;kb|g.977.peg.814&#39;<br />        ];<br /><br /></pre>
<h3><a class="mozTocH2" name="mozTocId586120"></a>Accessing Members of a Protein Family and the Family Functions</h3>
 Suppose that you have a set of <a name="protein families298">protein families</a>
 and you wish to lookup their functions and members. This can be done using code like <br />
<pre>new_for_script();<br /><br />my $protein_families = [&quot;FIG01303839&quot;];<br />my $famsH            = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_functions" target='_blank'>protein_families_to_functions</a>($protein_families);<br />my $pegsH            = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_fids" target='_blank'>protein_families_to_fids</a>($protein_families);<br />print &amp;Dumper($famsH,$pegsH);<br />br&gt;</pre>
 which priduces output like <br />
<pre>$VAR1 = {<br />          &#39;FIG01303839&#39; =&gt; &#39;Enoyl-CoA hydratase (EC 4.2.1.17) / 3-hydroxyacyl-CoA dehydrogenase (EC 1.1.1.35) / 3-hydroxybutyryl-CoA epimerase (EC 5.1.2.3)&#39;<br />        };<br />$VAR2 = {<br />          &#39;FIG01303839&#39; =&gt; [<br />                             &#39;kb|g.22.peg.5488&#39;,<br />                             &#39;kb|g.26.peg.1073&#39;,<br />                             &#39;kb|g.45.peg.6310&#39;,<br />                             &#39;kb|g.55.peg.4641&#39;,<br />                             &#39;kb|g.56.peg.177&#39;,<br />                             &#39;kb|g.59.peg.1996&#39;,<br />                             &#39;kb|g.64.peg.2224&#39;,<br />                             &#39;kb|g.67.peg.1310&#39;,<br />                             &#39;kb|g.75.peg.2967&#39;,<br />                             &#39;kb|g.76.peg.670&#39;,<br />                             &#39;kb|g.77.peg.1250&#39;,<br />			     .<br />			     .<br />			     .<br />                             &#39;kb|g.3883.peg.1974&#39;,<br />                             &#39;kb|g.3884.peg.4283&#39;<br />                           ]<br />        };<br /><br /></pre>
<h3><a class="mozTocH2" name="mozTocId458855"></a>Getting a Set of Co-occurring Protein Families</h3>
 Protein <a name="families299">families</a>
 are thought of as co-occurring if members of the <a name="families300">families</a>
 tend to occur close to one another on the <a name="chromosome301">chromosome</a>
. That is, it is basically conserved contiguity. <p> To find which <a name="protein families302">protein families</a> tend to co-occur, you can use code like <br />
<pre>use strict;<br />use Data::Dumper;<br />use Bio::KBase::CDMI::CDMIClient;<br />use Bio::KBase::Utilities::ScriptThing;<br />my $csO = Bio::KBase::CDMI::CDMIClient-&gt;new_for_script();<br /><br />my $fams    = [&#39;FIG01303839&#39;,&#39;FIG00001992&#39;];<br />my $funcsH  = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_functions" target='_blank'>protein_families_to_functions</a>($fams);<br />my $famsH   = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_co_occurring_families" target='_blank'>protein_families_to_co_occurring_families</a>($fams);<br />print &amp;Dumper($funcsH,$famsH);<br /><br /></pre>
 This simple program displays the functions of the input <a name="families303">families</a>
 (two of them in this simple case), and then displays co-occurring <a name="families304">families</a>
 for each of the input <a name="families305">families</a>
. It produces <br />
<pre>$VAR1 = {<br />          &#39;FIG01303839&#39; =&gt; &#39;Enoyl-CoA hydratase (EC 4.2.1.17) / 3-hydroxyacyl-CoA dehydrogenase (EC 1.1.1.35) / 3-hydroxybutyryl-CoA epimerase (EC 5.1.2.3)&#39;,<br />          &#39;FIG00001992&#39; =&gt; &#39;Phosphohistidine phosphatase SixA&#39;<br />        };<br />$VAR2 = {<br />          &#39;FIG01303839&#39; =&gt; [<br />                             [<br />                               &#39;FIG01304970&#39;,<br />                               29,<br />                               &#39;Modification methylase, HemK family (EC 2.1.1.72)&#39;<br />                             ]<br />                           ],<br />          &#39;FIG00001992&#39; =&gt; [<br />                             [<br />                               &#39;FIG00002110&#39;,<br />                               15,<br />                               &#39;2-keto-3-deoxy-D-arabino-heptulosonate-7-phosphate synthase I alpha (EC 2.5.1.54)&#39;<br />                             ],<br />                             [<br />                               &#39;FIG00002671&#39;,<br />                               84,<br />                               &#39;Protease III precursor (EC 3.4.24.55)&#39;<br />                             ],<br />                             [<br />                               &#39;FIG00003087&#39;,<br />                               103,<br />                               &#39;Murein endopeptidase&#39;<br />                             ],<br />                             [<br />                               &#39;FIG00007476&#39;,<br />                               184,<br />                               &#39;Periplasmic fimbrial chaperone StfD&#39;<br />                             ],<br />                             [<br />                               &#39;FIG00009029&#39;,<br />                               186,<br />                               &#39;Major fimbrial subunit StfA&#39;<br />                             ],<br />                             [<br />                               &#39;FIG00009068&#39;,<br />                               43,<br />                               &#39;RNA polymerase sigma factor for flagellar operon&#39;<br />                             ],<br />                             [<br />                               &#39;FIG00011116&#39;,<br />                               140,<br />                               &#39;Fimbriae usher protein StfC&#39;<br />                             ],<br />                             [<br />                               &#39;FIG00013366&#39;,<br />                               13,<br />                               &#39;ADP-heptose--lipooligosaccharide heptosyltransferase II (EC 2.4.1.-)&#39;<br />                             ],<br />                             [<br />                               &#39;FIG00029709&#39;,<br />                               45,<br />                               &#39;Flagellar biosynthesis protein FlhA&#39;<br />                             ],<br />                             [<br />                               &#39;FIG00030797&#39;,<br />                               68,<br />                               &#39;C4-type zinc finger protein, DksA/TraR family&#39;<br />                             ],<br />                             [<br />                               &#39;FIG00039085&#39;,<br />                               69,<br />                               &#39;Mlr0777 protein&#39;<br />                             ],<br />                             [<br />                               &#39;FIG00138924&#39;,<br />                               63,<br />                               &#39;Amino acid regulated cytosolic protein&#39;<br />                             ],<br />                             [<br />                               &#39;FIG00146544&#39;,<br />                               128,<br />                               &#39;Ribosomal protein L3 methyltransferase&#39;<br />                             ],<br />                             [<br />                               &#39;FIG00229397&#39;,<br />                               71,<br />                               &#39;GTP-binding protein&#39;<br />                             ],<br />                             [<br />                               &#39;FIG01220567&#39;,<br />                               42,<br />                               &#39;FIG01220568: hypothetical protein&#39;<br />                             ],<br />                             [<br />                               &#39;FIG01272345&#39;,<br />                               134,<br />                               &#39;Putative membrane protein YfcA&#39;<br />                             ],<br />                             [<br />                               &#39;FIG01303839&#39;,<br />                               328,<br />                               &#39;Enoyl-CoA hydratase (EC 4.2.1.17) / 3-hydroxyacyl-CoA dehydrogenase (EC 1.1.1.35) / 3-hydroxybutyryl-CoA epimerase (EC 5.1.2.3)&#39;<br />                             ],<br />                             [<br />                               &#39;FIG01304970&#39;,<br />                               52,<br />                               &#39;Modification methylase, HemK family (EC 2.1.1.72)&#39;<br />                             ]<br />                           ]<br />        };<br /><br /></pre>
 Did you know that a fid implementing <i>&#39;Enoyl-CoA hydratase (EC 4.2.1.17) / 3-hydroxyacyl-CoA dehydrogenase (EC 1.1.1.35) / 3-hydroxybutyryl-CoA epimerase (EC 5.1.2.3)</i>
 co-occurs with a <a name="peg306">peg</a>
 assigned the function <i>Phosphohistidine phosphatase SixA</i>
 in 328 distinct OTUs? <h3><a class="mozTocH2" name="mozTocId193709"></a>Given a Role, which Subsystems Include it?</h3>
 The notion of <i><a name="Role307">Role</a></i>
 is central to that of a controlled vocabulary. <i>Subsystems</i>
 are sets of such roles and compose the framework within which they are connected to specific genes. So, given a role like <i>Methionyl-tRNA synthetase (EC 6.1.1.10)</i>
, which <a name="subsystem308">subsystem</a>
 or <a name="subsystems309">subsystems</a>
 are used to propagate the <a name="annotations310">annotations</a>
? <br />
<pre>use strict;<br />use Data::Dumper;<br />use Bio::KBase::CDMI::CDMIClient;<br />use Bio::KBase::Utilities::ScriptThing;<br />my $csO = Bio::KBase::CDMI::CDMIClient-&gt;new_for_script();<br /><br />my $role = &quot;SSU ribosomal protein S12p (S23e)&quot;;<br /><br />my $roleH = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#roles_to_subsystems" target='_blank'>roles_to_subsystems</a>([$role]);<br />print &amp;Dumper($roleH);<br /><br /></pre>
 This produces the following output: <br />
<pre>$VAR1 = {<br />          &#39;SSU ribosomal protein S12p (S23e)&#39; =&gt; [<br />                                                   &#39;271-Bsub&#39;,<br />                                                   &#39;EGS prediction&#39;,<br />                                                   &#39;Mycobacterium virulence operon involved in protein synthesis (SSU ribosomal proteins)&#39;,<br />                                                   &#39;Ribosomal protein S12p Asp methylthiotransferase&#39;,<br />                                                   &#39;Ribosome SSU bacterial&#39;,<br />                                                   &#39;Virulence operon involved in protein synthesis (SSU ribosomal proteins)&#39;<br />                                                 ]<br />        };<br /><br /></pre>
 That is, the role is in six distinct <a name="subsystems311">subsystems</a>
. <h3><a class="mozTocH2" name="mozTocId869371"></a>Going from Roles to Complexes to Reactions to Printable Versions of Reactions</h3>
<i>Roles</i>
 are parts of <i>complexes</i>
. A complex is made up of one or more roles. Some complexes implement reactions. The IDs for complexes and reactions are unreadable strings, but they can be expanded into meaningful strings. This little example shows how the concepts fit together: <br />
<pre>use strict;<br />use Data::Dumper;<br />use Bio::KBase::CDMI::CDMIClient;<br />use Bio::KBase::Utilities::ScriptThing;<br />my $csO = Bio::KBase::CDMI::CDMIClient-&gt;new_for_script();<br /><br />my $role   = &#39;Pyruvate kinase (EC 2.7.1.40)&#39;;<br />my $roleH  = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#roles_to_complexes" target='_blank'>roles_to_complexes</a>([$role]);<br />my @complexes = map { $_-&gt;[0] } @{$roleH-&gt;{$role}};<br />my $complexH = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#complexes_to_complex_data" target='_blank'>complexes_to_complex_data</a>(\@complexes);<br />foreach my $complex (keys(%$complexH))<br />{<br />    my $complex_data = $complexH-&gt;{$complex};<br />    my $reactions    = $complex_data-&gt;{complex_reactions};<br />    if ($reactions &amp;&amp; (@$reactions &gt; 0))<br />    {<br />	my $reactionH = $csO-&gt;<a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#reaction_strings" target='_blank'>reaction_strings</a>($reactions);<br />	foreach my $reaction (keys(%$reactionH))<br />	{<br />	    my $readable = $reactionH-&gt;{$reaction};<br />	    print join(&quot;\t&quot;,($complex,$reaction,$readable)),&quot;\n&quot;;<br />	}<br />    }<br />}<br /><br /><br /></pre>
 The output of the program is a single line: <br />
<pre>2AD4B58C-66F7-11E1-B48F-62E56F82D269	4DC986F0-66F5-11E1-B48F-62E56F82D269	ATP + Pyruvate &lt;=&gt; ADP + Phosphoenolpyruvate<br /><br /></pre>
 So, the given role connects to one complex, which in turn implements a single reaction (which is one of the steps in glycolysis). 