<h1>Tutorials to Help you Get Started</h1>
 In order to enable users of the KBase to get started with the system we have developed a series of tutorials ranging from the basic software installation how-to to detailed analysis of exemplars for functional roles. You may browse the tutorials from the menu on the right-hand side of the page. They are arranged roughly in order of complexity. However, here we present a short outline of what is available, and you might do well to first scan it quickly. <br />
<hr />
<ol>
 <li><b>IRIS</b>: a Browser-based Framework for Interacting with KBASE <ul>
   <li> Introduction<ul style="font-style: italic;">
    <li> The KBase IRIS&nbsp; interface allows one to run all the of the<span style="font-weight: bold;"> </span>Kbase command line scripts and some useful Unix tools from a web based interface without downloading any software. The interface keeps track of who you are and stores your results and history on our KBase IRIS server. The beta version of the IRIS interface is at <a href="http://bioseed.mcs.anl.gov/%7Eolson/up.html" target="_blank">the Intera</a><a href="http://bioseed.mcs.anl.gov/%7Eolson/up.html" target="_blank">ctive KBase</a>.</ul>
   <li> Examples </ul>
 <li><b>Command Line Scripts</b><ul>
   <li> Accessing CS Data <ul>
     <li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/" target="_blank">Some Basic Exercises Using the DOE Kbase (ERDB work)</a><span style="font-family: monospace;"></span><ul>
      <li><span style="font-family: monospace;"></span><span style="font-style: italic;">The DOE Kbase includes an integrated database of genomic data called the <a name="Central Store1">Central Store</a> (CS). Extracting data from the CS can be achieved using a defined <a name="API2">API</a>. It can also be accessed via a set of command-line scripts that allow a fairly simple and consistent mechanism for access, once a few general principles are understood.</span><br />
     </ul>
     <li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/" target="_blank">Getting What You Need from the CS Using Command-Line Scripts</a><ul>
      <li><span style="font-style: italic;">The <a name="Central Store3">Central Store</a> (CS) is the Kbase integration of the data needed to support the creation and validation of metabolic and regulatory models. It will certainly be used for many other purposes, as well; but, its creation is being driven by the needs of the modelling community.</span><br />
     </ul>
     <li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/" target="_blank"> Extending the CS-Commands with Operators</a><ul>
      <li><span style="font-style: italic;">The <a name="CS-API4">CS-API</a> allows a fairly rich set of command to access data from the CS. We believe that, if we supplement this <a name="API5">API</a> with just a few routines and then construct the corresponding commands, we will produce an extremely effective technology for serious work.</span><br />
     </ul>
    </ul>
   <li> Applications <ul>
     <li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/" name="Identical Genomes6" target="_blank"> Processing Identical Genomes</a><ul>
      <li><span style="font-style: italic;">The KBase is intended as an environment in which identical genomes, annotated by distinct groups, can reside and be compared. We anticipate that there will soon be thousands of &quot;essentially identical&quot; genomes, due to the rapid increase in sequencing capacity. We will cover &quot;very similar&quot; genomesin other tutorials. This short document will just make comments relating to the use of identical genomes.</span></ul>
     <li>Translation to the Abstract Function Vocabulary<ul>
      <li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/" name="Exemplars7" target="_blank"> Towards a Controlled Vocabulary Part 1: Defining Exemplars</a><ul>
       <li><span style="font-style: italic;">Much has been said about the desirability of a controlled vocabulary for protein function and how it might be achieved. One of the central goals of the KBase effort will be to develop detailed, predictive models of microbes. A consistent, controlled vocabulary of protein function will be needed to support automated generation of these models.</span><br />
      </ul>
      <li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/" name="Exemplars8" target="_blank"> Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</a><ul>
       <li><span style="font-style: italic;">We now consider the issue of creating a translation table that maps fids from some sources of <a name="annotations9">annotations</a> into the exemplars.</span><br />
      </ul>
      <li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/" target="_blank"> Towards a Controlled Vocabulary Part 3: Making the MOL</a><ul>
       <li><span style="font-style: italic;">We complete this discussion of how to reach consistent <a name="annotations10">annotations</a> in an abstract vocabulary by discussing how one might map <a name="MicrobesOnLine11">MicrobesOnLine</a> (MOL) features to the abstract vocabulary.</span><br />
      </ul>
     </ul>
    </ul>
  </ul>
 <li><b><a name="API12">API</a>&#39;s</b><ul>
   <li> Getting Started <ul>
     <li><a href="http://kbase.us/developer-zone/tutorials/getting-started/getting-started-with-the-kbase/" name="CPAN13" target="_blank">Getting Started with the KBase (Installing from CPAN)</a><ul>
      <li><span style="font-style: italic;">In order to use the KBase command line scripts as we discuss in these tutorials, you will need to install the KBase client software on your computer.</span><br />
     </ul>
     <li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/" name="SEED14" target="_blank">KBase Development in the SEED Environment</a><ul>
      <li><span style="font-style: italic;">The KBase development environment is still under construction. To accelerate progress, we have created a framework within the <a name="SEED15">SEED</a> environment that allows us to develop and test software, as well as creating and maintaining tutorials and documentation. At this stage, we have employed this environment to support worrk on the ID server, the CS and the documentation. We intend to move to a native KBase development environment, once it is well-defined. Until then, this document is intended to help KBase developers with access to the <a name="SEED16">SEED</a> environment to get started.</span><br />
     </ul>
     <li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/" name="CS-API17" target="_blank">Extracting Data from the CS Using the CS-API: Some Typical Examples</a><ul>
      <li><span style="font-style: italic;">We will cover the basic tools for accessing data in the CS via the <a name="CS-API18">CS-API</a> (as opposed to the more common use of the command-line tools. We will give small test programs along with displayed output in hopes that users can easily generate what they need by making minor modifications.</span><br />
     </ul>
    </ul>
   <li> Applications <ul>
     <li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/looking-for-features-with-identical-sequence-that-have-different-annotations/" target="_blank">Looking for Features with Identical Sequence that have Different Annotations</a><ul>
      <li><span style="font-style: italic;">In this short example, we write code that reads in a file in which each line is supposed to contain the ID of a feature of type &#39;<a name="peg19">peg</a>&#39; (protein-encoding gene). For each of these input pegs we gather all features that have exactly the same translation (i.e., protein sequence), gathe their <a name="annotations20">annotations</a>, and check to see if they are all identical. If not, the set of features and inconsistent <a name="annotations21">annotations</a> gets displayed.</span><br />
     </ul>
     <li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/comparing-the-functional-roles-in-two-genomes/" target="_blank">Comparing the Functional Roles in Two Genomes</a><ul>
      <li><span style="font-style: italic;">As the KBase begins to be used to support development and maintenance of metabolic models, it becomes important that we be able to rapidly compare the functional roles that are implemented by protein-encoding genes in a pair of genomes. In the most common case, we will be looking for errors in <a name="annotations22">annotations</a> between very close genomes. In that case, most of the discrepancies will reflect errors in gene calling and <a name="annotations23">annotations</a>. In the cases of more distant genomes, it becomes possible to infer metabolic differences from the discrepancies.</span><br />
     </ul>
     <li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/the-issue-of-retrieving-functional-coupling-data/" target="_blank">The Issue of Retrieving Functional Coupling Data</a><ul>
      <li><span style="font-style: italic;">As we proceed to improve <a name="annotations24">annotations</a> of gene function, precision of metabolic models, and estimates of regulatory architecture, we will need access to the clues that support estimates of &quot;functional coupling&quot;. In this short example, we allow the user to designate a genome within the set stored in the CS (presumably a prokaryotic genome), and we retrieve data relating to co-occurrence (in close proximity on multiple genomes) and co-expression (when we have data available ).</span><br />
     </ul>
     <li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/" target="_blank">Mapping Functions Between Sources of Annotations: Take 2</a><ul>
      <li><span style="font-style: italic;">In another tutorial we outlined an approach to establishing exchangable <a name="annotations25">annotations</a> based on exemplars. The approach we illustrate here involves a set of perl programs that invoke KBase command-line tools (rather than using the KBase <a name="API26">API</a> directly). In our view this is a convenient paradigm for problems for which performance is not the big issue.<br /></span><br />
     </ul>
    </ul>
  </ul>
</ol>
