<h1>Accessing Protein Family Data from the Central Store (CS)</h1>
<h2>Introduction</h2>
 The KBase will contain <a name="protein families369">protein families</a>
 from a number of sources. During the first load in February, 2012, FIGfams were loaded. In the second load scheduled for May, 2012, the TIGRfams/Equivalogs will be loaded. <p> It is worth reflecting a bit on what is meant by a <a name="protein family370">protein family</a>, why they are constructed, and how they are used. 
<h2>What is a Protein Family?</h2>
 The basic common notion shared by most groups building <a name="protein families371">protein families</a>
 is that a <a name="protein family372">protein family</a>
 is a set of similar proteins that derived from a common ancestor. As the creators of PIR put it <blockquote><i> Protein <a name="family373">family</a> members are homologous (sharing common ancestry) and homeomorphic (sharing full-length sequence similarity with common domain architecture). </i></blockquote>
 Superfamilies are then used to group <a name="families374">families</a>
 in which the members share ancestry, but differing domain architectures. Subfamilies are often thought of as a set of sequences (within a <a name="family375">family</a>
) that share the same function. That is, within a subfamily, the functions of the members are considered identical (although there is always a significant element of imprecision due to the difficulty of defining what is meant by saying that two proteins have identical function). <p> In this tutorial, we will focus on FIGfams, the <a name="protein families376">protein families</a> developed within the <a href="http://www.theseed.org/wiki/Main_Page" name="SEED377" target="blank_">SEED Project</a>. In following tutorials we will focus on other sources of <a name="protein families378">protein families</a> and how to compare them. 
<p> Within the FIGfams, a <a name="protein family379">protein family</a> is thought of as a set of <i>isofunctional homologs</i>. That is, they are <i>subfamilies </i> in the established vernacular. 
<h2>What Are FIGfams Used for?</h2>
 FIGfams, like most other collections of <a name="protein families380">protein families</a>
, were created to support automated projection of function. However, the way they support automated projection is quite unique. It is based on the use of <i>signature kmers</i>
. A <i>signature kmer</i>
 is a short peptide that occurs only in <a name="protein families381">protein families</a>
 that have the same function (normally, a single <a name="family382">family</a>
). That is, if a new protein includes a signature kmer, this amounts to evidence that the function associated with the kmer can be projected. This much is common to many approaches to function projection. What made FIGfams somewhat unique is the collection of all signature kmers (the vast majority of which occur just once in one protein in one <a name="family383">family</a>
) and the use of millions of such signatures to project function. To make this work properly, you need a large and cleanly annotated (at least consistently annotated) collection of <a name="protein families384">protein families</a>
. <p> FIGfams are not manually curated. They are derived from a collection of <a href="http://www.theseed.org/wiki/Glossary#Subsystem" name="subsystems385" target="blasn_">subsystems</a>, which are manually curated. Further, some <a name="protein families386">protein families</a> arise through automated processes that examine sets of closely related genomes, forming <a name="families387">families</a> from genes based on similarity and preserved chromosomal context. The original collection was described in <a href="http://nar.oxfordjournals.org/content/37/20/6643.long" name="protein families388" target="blank_"> FIGfams: yet another set of protein families</a>. Their use and technology for construction has evolved since then. Release57 contains 268,358 <a name="families389">families</a>. 
<h2>Accessing the FIGfams from within KBase</h2>
 Suppose that you just wanted a list of the FIGfam <a name="families390">families</a>
 in KBase. You can do this with <br />
<pre>                <a href="http://kbase.us/developer-zone/api-documentation/cdm-entity-relationship-command-scripts/#query_entity_Family" target='_blank'>query_entity_Family</a> -like &#39;id,FIG%&#39; | \<br />                sort |                                \<br />                <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_functions" target='_blank'>protein_families_to_functions</a> |       \<br />                <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_fids" target='_blank'>protein_families_to_fids</a> -c 1 |       \<br />                <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_functions" target='_blank'>fids_to_functions</a> |                   \<br />                <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_protein_sequences" target='_blank'>fids_to_protein_sequences</a> -c 3 -fasta 1 -fc 1,4 &gt; all.figfams.fasta<br /><br /></pre>
 This little <a name="pipeline391">pipeline</a>
 is worth examining in detail: <ul><li>The first line extracts from the CS the ids of the FIGfams. It uses one of the query_entity operators. These can be quite useful. The <i>-like &#39;id,FIG%&#39;</i> says &quot;get the <a name="Family392">Family</a> entities in which the id field begins with FIG&quot;. That is the &quot;%&quot; will match any substring. The <a name="sort393">sort</a> that follows simply makes the output appear sorted on FIGfam ids. <li>The last four lines may be thought of as a little <a name="pipeline394">pipeline</a> that takes in a set of FIGfam ids and produces a fasta file for those <a name="families395">families</a>. First, the <a name="family396">family</a> functions are appended. Then, the <a name="families397">families</a> are expanded to show the fids within each <a name="family398">family</a>, and the functions of each fid get appended. <li> Finally, the fids are used to look up their protein sequences. The arguments have the following meanings: <ol><li><i>-c 3</i> says that the fids are taken from the third column of the input (at this point, the columns are [FIGfam-id,FIGfam-function,fid-in-FIGfam,fid-function]). <li><i>-fasta 1</i> says &quot;give me the output as a fasta file, rather than as a tab-separated table&quot;. <li><i>-fc 1,4</i> says &quot;make the comment field of the generated fasta file from columns 1 and 4 (i.e., the FIGfam-id and the fid-function)&quot;. If you wanted the <a name="protein family399">protein family</a> function instead (it should agree with the fid function, but sometimes does not), what would you use? </ol> The output of this last line is a fasta file. </ul>
 The <a name="pipeline400">pipeline</a>
 produces output like the following: <br />
<pre>&gt;kb|g.3.peg.3387 FIG00000001; Cysteine desulfurase<br />MFAPVYLDHNASAPLLDAVREAMLPWLGARFGNASSRHEYGRAARRAIDDARARVAAAVGAHPTEVVFTSGGSEANNLFIKGAAARMKPGLVQVSAIEHP\<br />CVREPARQLVRAGWTLAEMPADADGRVELDALDVRAQLVSVMAANNETGVLQDISAFATAAKRQGAWLHTDAVQALGKIPVDFRALGVAAMTLSAHKIGG\<br />PQGAGALVLDKRVDLEPLIAGGGHERGLRSGTENIAAIVGFGVACEEAVAALGARREAMQAQRALIERCVAEQGGSVFGAGAERLPNTCYFALPGIDGET\<br />LVGKLDRAGFAVASGAACSSASPEPSRTLTAMGVAADLARGAVRVSVGASTTMQDVNDFCGALARVADELRGLRAMAAA<br />&gt;kb|g.3.peg.3358 FIG00000001; Cysteine desulfurase (EC 2.8.1.7)<br />MSGLSSPLPPGAPDESILSRLASELFAALPGQTPPSFDAPLSAAPTPSPQAPALPLPADAPSSPAFAAAPVAPQHALSLGASSPALVPHSSAGNGLPDAA\<br />VDAVPVGAPRAGGVQGVPEAYAAALPVVGHASPQAGGASVPSSPYYFLGEASALQSAPGGASQGAVSVPDDDRVSAQSFGLPGGDALRALLAAPSREPAL\<br />PVSAAPAASDYYFVQPQSVPTVGHAGRVPNADAGRHPPFDVNAVRRDFPILAERVNGKPLVWFDNAATTQKPQAVIDRLAYFYAHENSNIHRAAHELAAR\<br />ATDAYEAARQKVARFIGAGSVEEIIFVRGATEAINLVAKSWGAKHIGAGDEIIVSNLEHHANIVPWQQLAAEKGAKIRVIPVDDSGQVLLEEYKRLLNEK\<br />TKLVAVTQVSNALGTVVPVQEIVDLANRAGARTLVDGAQSVSHMRVNVRALDTDFFVFSGHKIFGPTGIGVVYGKRDVLEDMPPWQGGGNMIQDVTFERT\<br />IFHPPPARFEAGTGNIADAVGLGAALDYVDRVGIENIARYEHDLLVYATHAIRDIPGVRLIGTAADKASVLSFVLKGYSTDEVGRALNQEGIAVRTGHHC\<br />AQPILRRFGVETTVRPSLAFYNTCEEVDRFATVIRRLAGQRKV<br />.<br />.<br />.<br /><br /></pre>
 From the very start, this looks a bit questionable, does it not? The differences in length, and the fact that the fids do not have completely identical functions reflects the current situation (and, it is a bit upsetting). There is much work to be done. <p> Suppose that you wished to get the members of a specific <a name="protein family401">protein family</a> for which you know the ID. From the above example, it should be clear that something like <br />
<pre>        echo FIG01300397 | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_fids" target='_blank'>protein_families_to_fids</a><br /><br /></pre>
 would get you a two column table containing the <a name="family402">family</a>
 name in the first column and the features in the second column. If you wanted to get a FASTA formatted file containing the members in the <a name="family403">family</a>
, you might use something like <br />
<pre>        echo FIG01300397 | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_fids" target='_blank'>protein_families_to_fids</a> | <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_protein_sequences" target='_blank'>fids_to_protein_sequences</a> -fasta 1 <br /><br /></pre>
 For many purposes it is better to weed out very close sequences, constructing a set of <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#representative" target='_blank'>representative</a> sequence. Thus, <br />
<pre>        echo FIG01300397 | \<br />        <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#protein_families_to_fids" target='_blank'>protein_families_to_fids</a> | \<br />        <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_protein_sequences" target='_blank'>fids_to_protein_sequences</a> -fasta 1 | \<br />        <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#representative_sequences" target='_blank'>representative_sequences</a> -s 0.5 &gt; <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#representative" target='_blank'>representative</a>.set.fasta<br /><br /></pre>
 This would take a cluster of sequences that were grouped using a 50% identity threshold and eliminate all but a single <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#representative" target='_blank'>representative</a>. You have the ability to set numerous parameters that impact the choice of <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#representative" target='_blank'>representative</a> sequences: <ul><li>You can cluster based on a number of distinct ways to measure similarity. You can use percent identity by using <i>-m identity_fraction</i>. You can measure similarity by asking what fraction of the aligned amino acids have a positive Blossum score by using <i>-m positive_fraction</i>. Finally, <i>-m score_per_position</i> will use an estimate of bit-score to measure similarity (and the values here range from 0 to 2). <li>Having chosen a notion of similarity, you can set a cutoff for grouping using <i>-s X</i> where <i>X</i> is a fraction from 0-1 for the first two options of similarity or a value between 0-2 for the bit-score based notion. <li>If you want to get FASTA files for each subgroup (each containing at least one <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#representative" target='_blank'>representative</a> sequence), use <i>-d Directory</i>. This will cause the directory to be built, if necessary, and a file per cluster will get written into the directory. The program will order the naming of the clusters by size (largest clusters getting the smallest ids). <li>To get just a listing of the ids that make up each cluster, use <i>-f file.of.clusters</i> which causes the file <i>file.of.clusters</i> as an example) to get built. Each line in the file will display the ids that make up a cluster. <li>The precise algorithm used to construct the clusters is beyond the scope of this tutorial. However, it is impacted by the order in which sequences are &quot;examined&quot;. It tends to work better when the sequences are ordered from longest to shortest in that groups are represented by longer sequences (tending to avoid representing a group with a partial sequence). To get this behavior, we recommend using <i>-b</i>. The user can preorder the sequences, and we recommend experimenting if you want to really see the impact. </ul>
 There is one final parameter that needs to be mentioned that allows you to start with an initial set of representatives that will be extended. This is useful when you are periodically extending <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#representative" target='_blank'>representative</a> sets. To get this behavior, use <i>-e RepsToExtend</i>
 where <i>RepsToExtend</i>
 is a file containing the existing representatives (in FASTA format). <h2>Coming from Roles in Models Instead</h2>
 Rather than starting with a <a name="protein family404">protein family</a>
, you might wish to see the set of <a name="families405">families</a>
 that correspond to a specific role. You might try something like <br />
<pre>        <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#all_roles_used_in_models" target='_blank'>all_roles_used_in_models</a> | grep &#39;(EC 2.7.1.11)&#39; <br /><br /></pre>
 which gives you <br />
<pre>6-phosphofructokinase (EC 2.7.1.11)<br />6-phosphofructokinase class II (EC 2.7.1.11)<br /><br /></pre>
 Then, you can use something like <br />
<pre>       echo &#39;6-phosphofructokinase (EC 2.7.1.11)&#39; | \<br />       <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#roles_to_fids" target='_blank'>roles_to_fids</a> | \<br />       <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#fids_to_protein_sequences" target='_blank'>fids_to_protein_sequences</a> -fasta 1 | \<br />       <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#representative_sequences" target='_blank'>representative_sequences</a> &gt; 2.7.1.11.fasta<br /><br /></pre>
 This would produce a set of <a href="http://kbase.us/developer-zone/api-documentation/cdm-command-line-scripts/#representative" target='_blank'>representative</a> sequences that could be used as a basis for constructing searchs for the specific function (in the case <i>6-phosphofructokinase (EC 2.7.1.11))</i>
. That is, you do not have to start with a <a name="protein family406">protein family</a>
; you can use functional roles as a path to fids and the corresponding sequences. 