<html>
  <head>
  </head>
  <body>
    <h1>Extending the CS-Commands with Operators</h1> The <a name="CS-API406">CS-API</a> allows a fairly rich set of command to access data from the CS. We believe that, if we supplement this <a name="API407">API</a> with just a few routines and then construct the corresponding commands, we will produce an extremely effective technology for serious work. In this conception, one constructs pipelines (or, manually invokes commands) that roughly correspond to three stages of computation: <ol>
      <li>During the first stage, one constructs either a tab-separated table populated with data from the CS or a fasta file of sequence data (either <a name="DNA408">DNA</a> or protein). <li> During the second stage, one invokes computational operators. <li>In the last stage, output is cast in one of a variety of forms. </ol>
    <p> In reality, the stages are not always clearly delimited or consecutive. <p> In our introduction to this style we will focus on a single operator, <b><a name="corresponds409"><a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#corresponds" target='_blank'>corresponds</a></a></b>, which is not part of the CS-commands. It is not used to extract data from the CS, but it is an extremely convenient tool for extending the tab-separated tables often produced by CS-commands. <p>
    <h2>The <i><a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#corresponds" target='_blank'>corresponds</a></i> Operator</h2> The <i><a name="corresponds410"><a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#corresponds" target='_blank'>corresponds</a></a></i> operator takes as input a file containing a tab-separated table in which a column contains <i>fids</i>, along with a single genome ID as a command-line argument. <p> The operator attempts to locate, for each input fid, a corresponding fid in the designated genome. When it succeeds, an output line is written containing the corresponding input line with extra columns containing details of the mapping and the corresponding fid. When a corresponding fid cannot be located, the input line is written to STDERR. Thus, <br /><pre>
       echo &#39;kb|g.2223&#39; | <a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#genomes_to_fids" target='_blank'>genomes_to_fids</a> | <a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#corresponds" target='_blank'>corresponds</a> -g &#39;kb|g.2814&#39; -a 0.6
<br /></pre> would attempt to map the fids in genome <i>kb|g.2223</i> to the fids in genome <i>kb|g.2814</i>. The result would be a sequence of tuples in what is called the &quot;abbreviated format&quot;. These would look like <br /><pre>
kb|g.2223       kb|g.2223.peg.2139      0.755   kb|g.2814.peg.5135
kb|g.2223       kb|g.2223.peg.3355      0.808   kb|g.2814.peg.5519
kb|g.2223       kb|g.2223.peg.4625      0.860   kb|g.2814.peg.5256
kb|g.2223       kb|g.2223.peg.4076      0.926   kb|g.2814.peg.5445
kb|g.2223       kb|g.2223.peg.4434      0.934   kb|g.2814.peg.5197
kb|g.2223       kb|g.2223.peg.4500      0.896   kb|g.2814.peg.5449
kb|g.2223       kb|g.2223.peg.4277      0.860   kb|g.2814.peg.5127
kb|g.2223       kb|g.2223.peg.4258      0.813   kb|g.2814.peg.5329
kb|g.2223       kb|g.2223.peg.758       0.812   kb|g.2814.peg.5559
.
.
.
<br /></pre> Here, in the abbreviated form, two columns were added by the <i>correspomnds</i> command: <ol>
        <li> The first added column contains a score that goes from 0 to 1 and reflects the reliability of the conjectured correspondence. It is computed as a function of the percent identity of the match and the number of genes that appear to be conserved in the contexts of the corresponding genes. The value given in the <i>-a 0.6</i> argument indicates that correspondences are shown only if the computed score is greater than or equal to 0.6. <li>The second added field is the corresponding gene detected in the genome specified using the <i>-g &#39;kb|g.2814&#39;</i> argument. </ol> If you wished to see a richer set of data for each match, you could use <br /><pre>
       echo &#39;kb|g.2223&#39; | <a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#genomes_to_fids" target='_blank'>genomes_to_fids</a> | <a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#corresponds" target='_blank'>corresponds</a> -g &#39;kb|g.2814&#39;
<br /></pre> which gives lines like <br /><pre>
kb|g.2223       kb|g.2223.peg.2139      98.88   4       1       267     267     1       267     267     0.755   kb|g.2814.peg.5135
kb|g.2223       kb|g.2223.peg.3355      98.36   5       1       304     304     1       304     304     0.808   kb|g.2814.peg.5519
kb|g.2223       kb|g.2223.peg.4625      100.00  6       1       266     266     1       266     266     0.860   kb|g.2814.peg.5256
kb|g.2223       kb|g.2223.peg.4076      96.07   8       78      255     256     1       178     179     0.926   kb|g.2814.peg.5445
kb|g.2223       kb|g.2223.peg.4434      99.00   8       1       299     299     108     406     406     0.934   kb|g.2814.peg.5197
kb|g.2223       kb|g.2223.peg.4500      98.41   7       1       251     251     1       251     251     0.896   kb|g.2814.peg.5449
kb|g.2223       kb|g.2223.peg.4277      100.00  6       1       240     240     1       240     240     0.860   kb|g.2814.peg.5127
kb|g.2223       kb|g.2223.peg.4258      100.00  5       1       155     155     1       155     155     0.813   kb|g.2814.peg.5329
kb|g.2223       kb|g.2223.peg.758       99.59   5       4       246     246     1       243     243     0.812   kb|g.2814.peg.5559
kb|g.2223       kb|g.2223.peg.632       37.33   0       1       149     160     1       144     392     0.178   kb|g.2814.peg.4418
kb|g.2223       kb|g.2223.peg.349       27.96   2       20      709     1134    93      828     833     0.440   kb|g.2814.peg.5035
kb|g.2223       kb|g.2223.peg.463       43.73   0       96      894     921     41      808     821     0.191   kb|g.2814.peg.5308
kb|g.2223       kb|g.2223.peg.1760      27.23   2       30      251     253     64      285     438     0.439   kb|g.2814.peg.5443
kb|g.2223       kb|g.2223.peg.1379      33.33   2       6       442     443     5       445     447     0.449   kb|g.2814.peg.5000
kb|g.2223       kb|g.2223.peg.1793      28.09   1       65      355     360     39      319     331     0.330   kb|g.2814.peg.5076
<br /></pre> Here 12 fields were added to each input line (due to not using the <i>-a</i> argument): <ol>
        <li>the percent identity for the match (which was computed using <i>blast</i>, <li>the number of genes that appear to be conserved in the contexts of these corresponding genes (maximum value of 10), <li>beginning coordinate of match in id1, <li>end coordinate of match in id1, <li>length of id1, <li>beginning coordinate of match in id2, <li>end coordinate of match in id2, <li>length of id2, <li>the score discussed above, and <li>the corresponding fid (id2). </ol>
    <p> The <i><a name="corresponds411"><a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#corresponds" target='_blank'>corresponds</a></a></i> operator works best for mapping prokaryotic features to fids in a prokaryotic genome, or for mapping eukaryotic fids to genes in eukaryotic genomes. Basically, the operator tries to construct correspondences between genes in two distinct genomes, and it makes sense to do this only between genomes of the same type. The tools for constructing a mapping between genes in two prokaryotic genomes are quite different than those for building mappings between eukaryotic genomes. The implementation of this operator is under frequent improvement as new tools for mapping neighborhoods become available. <p> The implementation of operators like <i><a name="corresponds412"><a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#corresponds" target='_blank'>corresponds</a></a></i> requires construction of maps between fids from pairs of genomes. Constructing these maps can be quite time consuming, so KBase includes a server that constructs them and saves the results. Hundreds of thousands of mappings between genomes have already been computed, but the collection certainly does not include a comprehensive collection. Thus, invoking <i><a name="corresponds413"><a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#corresponds" target='_blank'>corresponds</a></a></i> may induce a long delay (if one or more maps need to be computed) or a relatively short delay (if the maps exist and have been cached). <p> Normally, the input to <i><a name="corresponds414"><a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#corresponds" target='_blank'>corresponds</a></a></i> is a set of fids from a single genome. However, there is nothing to prevent a user from inputting fids from a large number of genomes. You should realize that each distinct input genome requires either the compututation or retrieval of a separate mapping between two genomes. <p> Finally, it should be noted that numerous tools have been built to compute correspondences between fids in different genome. Superior solutions can often be achieved in special cases. It is intended that the <i><a name="corresponds415"><a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#corresponds" target='_blank'>corresponds</a></a></i> operator be built on a growing set of these existing tools. Default settings should cause invocation of an appropriate tool, and command-line arguments should allow a user to select desired computations of correspondence. <h2>The <i><a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#close_genomes" target='_blank'>close_genomes</a></i> Operator</h2> The <i><a name="close_genomes416"><a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#close_genomes" target='_blank'>close_genomes</a></a></i> operator is used to locate a set of close genomes. This can be useful for many purposes. To illustrate, suppose that we wished to annotate a newly-sequenced <i>Staphylococcus</i> genome. One might, as a first step, request a set of 10 close genomes, and then request correspondences from fids in these 10 genomes to fids in the new genome. Once these correspondences have been computed, one could easily formulate a voting strategy to estimate function assignments based on the correspondences. <p> As a second example, suppose that we wished to project atomic <a name="regulons417">regulons</a> from one genome to others that are &quot;close&quot;. We could use the <i><a name="close_genomes418"><a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#close_genomes" target='_blank'>close_genomes</a></a></i> operator to identify the closest genomes, compute the set of fids that we wish to look for in each genome, and then use the <i><a name="corresponds419"><a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#corresponds" target='_blank'>corresponds</a></a></i> operator to create mappings to the fids that might implement atomic <a name="regulons420">regulons</a> in each of the close genomes. <p> One usually invokes the <i><a name="close_genomes421"><a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#close_genomes" target='_blank'>close_genomes</a></a></i> operator using something like <br /><pre>
          echo &#39;kb|g.0&#39; | <a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#close_genomes" target='_blank'>close_genomes</a> -n 10
<br /></pre> which would produce 10 genomes that are viewed (under some criterium) as &quot;close&quot;. The command is actually more general; it takes a tab-separated table containing a column of genomes as input and adds &quot;close genomes&quot; as a single added column. <p> Like the <i><a name="corresponds422"><a href="http://kbase.us/developer-zone/api-documentation/cdmi-api/#corresponds" target='_blank'>corresponds</a></a></i> operator, there are many different ways to compute the desired output. It is a very active research topic. We offer a reasonable choice by default, and we will gradually enrich the set of options one can specify to effect the way in which close genomes are determined. </body>
</html>
