<h1>Index of Terms in KBase Tutorials</h1>
<b>API</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#API82">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#API220">Summary</a>
</ul>
<li>Tutorial: <i>Comparing the Functional Roles in Two Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/comparing-the-functional-roles-in-two-genomes/#API413">A Short Program for Extracting Data from the CS</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#API603">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#API617">Updating or Defining a CS-API "Well-Trodden Path" Routine</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#API2">Here</a>
</ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#API507">Introduction</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#API524">Introduction</a>
</ul>
</ul>
<b>Atomic Regulons</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Atomic Regulons264">Introduction</a>
</ul>
</ul>
<b>Bio::KBase</b>
<ul>
<li>Tutorial: <i>Developing Software in the KBase Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/developing-software-in-the-kbase-environment/#Bio::KBase719">The KBase Module</a>
</ul>
</ul>
<b>CDM</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#CDM76">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#CDM85">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#CDM224">Introduction</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#CDM327">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#CDM332">Navigating the Diagram</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#CDM601">Introduction</a>
</ul>
</ul>
<b>CPAN</b>
<ul>
<li>Tutorial: <i>Getting Started with the KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/getting-started-with-the-kbase/#CPAN65">CPAN installation</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/getting-started-with-the-kbase/#CPAN72">Prepackaged Installations</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/getting-started-with-the-kbase/#CPAN74">Where to find software</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#CPAN14">Here</a>
</ul>
</ul>
<b>CS-API</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#CS-API256">Introduction</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#CS-API600">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#CS-API619">Updating or Defining a CS-API "Well-Trodden Path" Routine</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#CS-API5">Here</a>
</ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#CS-API353">Finding identical genomes</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#CS-API354">A Small Perl Program to Invoke the CS-API Routines</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#CS-API361">Correspondence Between Genes from Two Identical Genomes</a>
</ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#CS-API506">Introduction</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#CS-API557">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>Central Data Model</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Central Data Model78">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Central Data Model86">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#Central Data Model223">Introduction</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#Central Data Model326">Introduction</a>
</ul>
</ul>
<b>Central Store</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#Central Store734">Associating Features with Protein Sequences</a>
</ul>
<li>Tutorial: <i>Annotate a Genome Using KBase Tools </i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#Central Store313">Select a genome</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Central Store81">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Central Store87">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#Central Store221">Introduction</a>
</ul>
<li>Tutorial: <i>Publishing a New Central Store Software Revision</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/publishing-a-new-central-store-software-revision/#Central Store712">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/publishing-a-new-central-store-software-revision/#Central Store713">Introduction</a>
</ul>
<li>Tutorial: <i>Extracting Genomes, Features, and Sequences from the KBase CS</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/extracting-genomes-features-and-sequences-from-the-kbase-cs/#Central Store345">Here</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#Central Store1">Here</a>
</ul>
</ul>
<b>Co-occurring</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Co-occurring270">Introduction</a>
</ul>
</ul>
<b>Coexpressed</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Coexpressed266">Introduction</a>
</ul>
</ul>
<b>Concerns</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Concerns93">How to Work with the CDM Entity-Relationship Model</a>
</ul>
</ul>
<b>Contig</b>
<ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#Contig363">A Short Program: Two Genes Correspond IFF Locations Are Identical </a>
</ul>
</ul>
<b>ContigSequence</b>
<ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#ContigSequence364">A Short Program: Two Genes Correspond IFF Locations Are Identical </a>
</ul>
</ul>
<b>DNA</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#DNA259">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#DNA139">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Extracting Genomes, Features, and Sequences from the KBase CS</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/extracting-genomes-features-and-sequences-from-the-kbase-cs/#DNA347">Extracting the Genomic Contigs in FASTA Format</a>
<li><a href="http://kbase.us/developer-zone/tutorials/extracting-genomes-features-and-sequences-from-the-kbase-cs/#DNA349">Getting the Sequences of Features</a>
</ul>
<li>Tutorial: <i>Constructing RAST2 in the IRIS Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#DNA40"> Tools for Analysis </a>
</ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#DNA508">Introduction</a>
</ul>
</ul>
<b>E.coli</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#E.coli123">How to Work with the CDM Entity-Relationship Model</a>
</ul>
</ul>
<b>ER-Model</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#ER-Model80">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#ER-Model88">How to Work with the CDM Entity-Relationship Model</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#ER-Model128">Extracting Fields from a Set of Instances of an Entity</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#ER-Model330">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#ER-Model343">Navigating the Diagram</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#ER-Model602">Introduction</a>
</ul>
</ul>
<b>Entity-Relationship Model</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Entity-Relationship Model77">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Entity-Relationship Model219">Summary</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#Entity-Relationship Model329">Introduction</a>
</ul>
</ul>
<b>Exemplars</b>
<ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#Exemplars8">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#Exemplars489">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#Exemplars455">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary of Function</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-of-function/#Exemplars421">A Three-Part Tutorial</a>
</ul>
</ul>
<b>Family</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Family268">Introduction</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#Family389">Accessing the FIGfams from within KBase</a>
</ul>
</ul>
<b>Feature</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#Feature731">Getting the Primary Function Associated with a Feature</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#Feature733">Associating Features with Protein Sequences</a>
</ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Feature258">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Feature83">How to Work with the CDM Entity-Relationship Model</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Feature135">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#Feature341">Navigating the Diagram</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#Feature440">The Concept of Exemplars</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#Feature456">Here</a>
</ul>
</ul>
<b>Genome</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Genome257">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Genome120">How to Work with the CDM Entity-Relationship Model</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Genome129">Extracting Fields from a Set of Instances of an Entity</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#Genome333">Navigating the Diagram</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#Genome562">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>Identical Genomes</b>
<ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#Identical Genomes7">Here</a>
</ul>
</ul>
<b>IsCoregulatedWith</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#IsCoregulatedWith84">How to Work with the CDM Entity-Relationship Model</a>
</ul>
</ul>
<b>IsOwnerOf</b>
<ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#IsOwnerOf337">Navigating the Diagram</a>
</ul>
</ul>
<b>IsProteinFor</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#IsProteinFor109">How to Work with the CDM Entity-Relationship Model</a>
</ul>
</ul>
<b>MicrobesOnLine</b>
<ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#MicrobesOnLine12">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#MicrobesOnLine494">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#MicrobesOnLine458">Creating a Translation Table</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#MicrobesOnLine527">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>Model</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Model125">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#Model425">Here</a>
</ul>
</ul>
<b>OTU</b>
<ul>
<li>Tutorial: <i>Computing Pan Genomes for OTU's</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/computing-pan-genomes-for-otus/#OTU727">Invoking CS Command scripts from Perl</a>
<li><a href="http://kbase.us/developer-zone/tutorials/computing-pan-genomes-for-otus/#OTU728"> Example 2</a>
</ul>
</ul>
<b>Pearson Correlation Coefficient</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Pearson Correlation Coefficient265">Introduction</a>
</ul>
<li>Tutorial: <i>The Issue of Retrieving Functional Coupling Data</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/the-issue-of-retrieving-functional-coupling-data/#Pearson Correlation Coefficient418">Discussion</a>
</ul>
</ul>
<b>Protein Sequence</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Protein Sequence267">Introduction</a>
</ul>
</ul>
<b>ProteinSequence</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#ProteinSequence735">Associating Features with Protein Sequences</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#ProteinSequence92">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#ProteinSequence439">The Concept of Exemplars</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#ProteinSequence530">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>Publication</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Publication89">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#Publication444">The Concept of Exemplars</a>
</ul>
</ul>
<b>Reaction</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Reaction124">How to Work with the CDM Entity-Relationship Model</a>
</ul>
</ul>
<b>RegPrecise</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#RegPrecise152">Scripts Handling Popular Operations</a>
</ul>
</ul>
<b>Role</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Role271">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#Role158">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#Role561">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>SEED</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#SEED737">Associating Assignments with Features and Protein Sequences</a>
</ul>
<li>Tutorial: <i>Getting Started with the KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/getting-started-with-the-kbase/#SEED73">Prepackaged Installations</a>
</ul>
<li>Tutorial: <i>Publishing a New Central Store Software Revision</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/publishing-a-new-central-store-software-revision/#SEED714">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/publishing-a-new-central-store-software-revision/#SEED717">Installing the new release into the KBase git</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#SEED598">Introduction</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#SEED15">Here</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#SEED374">What is a Protein Family?</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#SEED424">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#SEED490">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#SEED495">Begin with the SEED Translation</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#SEED505">Create a Table of Data Relating to the MOL Genomes</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#SEED459">Creating a Translation Table</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#SEED480">Generating the SEED Translations</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#SEED526">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>annotations</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#annotations730">Introduction</a>
</ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#annotations283">Introduction</a>
</ul>
<li>Tutorial: <i>Annotate a Genome Using KBase Tools </i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#annotations314">Annotate a genome object</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#annotations319">Get roles that might impact metabolic models</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#annotations322">Get roles that might have been missed</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#annotations134">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Comparing the Functional Roles in Two Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/comparing-the-functional-roles-in-two-genomes/#annotations410">Introduction</a>
</ul>
<li>Tutorial: <i>Constructing RAST2 in the IRIS Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#annotations31">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#annotations37">The Concept of the Genome Typed Object</a>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#annotations38"> Tools for Analysis </a>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#annotations51">Summary</a>
</ul>
<li>Tutorial: <i>Looking for Features with Identical Sequence that have Different Annotations</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/looking-for-features-with-identical-sequence-that-have-different-annotations/#annotations405">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/looking-for-features-with-identical-sequence-that-have-different-annotations/#annotations408">A Program to Look Up Functions Associated with Feature IDs</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#annotations10">Here</a>
</ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#annotations359">Controlling the Output </a>
</ul>
<li>Tutorial: <i>The Issue of Retrieving Functional Coupling Data</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/the-issue-of-retrieving-functional-coupling-data/#annotations416">Introduction</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#annotations454">The Concept of Exemplars</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#annotations493">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#annotations457">Creating a Translation Table</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#annotations487">Generating the SEED Translations</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#annotations523">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#annotations525">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>chromosome</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#chromosome285">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#chromosome138">Scripts Handling Popular Operations</a>
</ul>
</ul>
<b>close_genomes</b>
<ul>
<li>Tutorial: <i>Constructing RAST2 in the IRIS Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#close_genomes42"> Tools for Analysis </a>
</ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#close_genomes516">The close_genomes Operator</a>
</ul>
</ul>
<b>co-regulated</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#co-regulated151">Scripts Handling Popular Operations</a>
</ul>
</ul>
<b>conserved neighborhood</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#conserved neighborhood286">Introduction</a>
</ul>
</ul>
<b>contig</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#contig143">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Extracting Genomes, Features, and Sequences from the KBase CS</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/extracting-genomes-features-and-sequences-from-the-kbase-cs/#contig348">Getting the Features for a Genome</a>
</ul>
</ul>
<b>corresponds</b>
<ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#corresponds362">A Short Program: Two Genes Correspond IFF Locations Are Identical </a>
</ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#corresponds509">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#corresponds510">The corresponds Operator</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#corresponds519">The close_genomes Operator</a>
</ul>
</ul>
<b>cut</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#cut247">The Abstract View</a>
</ul>
<li>Tutorial: <i>Getting Started with the KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/getting-started-with-the-kbase/#cut70">CPAN installation</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises using the DOE KBase IRIS interface</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/some-basic-exercises-using-the-doe-kbase-iris-interface/#cut63">IRIS Unix command summary;</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#cut618">Updating or Defining a CS-API "Well-Trodden Path" Routine</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#cut437">The Concept of Exemplars</a>
</ul>
</ul>
<b>cvs</b>
<ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#cvs629">Updating or Defining a CS Command-Line Script</a>
</ul>
</ul>
<b>families</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#families280">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#families145">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#families371">What is a Protein Family?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#families384">What Are FIGfams Used for?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#families387">Accessing the FIGfams from within KBase</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#families402">Coming from Roles in Models Instead</a>
</ul>
</ul>
<b>family</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#family146">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Developing Software in the KBase Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/developing-software-in-the-kbase-environment/#family718">A Note on Languages</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#family370">What is a Protein Family?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#family379">What Are FIGfams Used for?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#family393">Accessing the FIGfams from within KBase</a>
</ul>
</ul>
<b>generator</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#generator234">The Abstract View</a>
</ul>
</ul>
<b>generators</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#generators237">The Abstract View</a>
</ul>
</ul>
<b>grep</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#grep122">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#grep227">The Basic Philosophy of the CS Command-line Tools</a>
</ul>
<li>Tutorial: <i>Getting Started with the KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/getting-started-with-the-kbase/#grep71">CPAN installation</a>
</ul>
<li>Tutorial: <i>Extracting Genomes, Features, and Sequences from the KBase CS</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/extracting-genomes-features-and-sequences-from-the-kbase-cs/#grep346">Exploring the Collection of Genomes in the CS</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises using the DOE KBase IRIS interface</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/some-basic-exercises-using-the-doe-kbase-iris-interface/#grep53">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/some-basic-exercises-using-the-doe-kbase-iris-interface/#grep60">Additional Features of IRIS</a>
<li><a href="http://kbase.us/developer-zone/tutorials/some-basic-exercises-using-the-doe-kbase-iris-interface/#grep62">IRIS Unix command summary;</a>
</ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#grep352">Finding identical genomes</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#grep564">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>literature</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#literature281">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#literature142">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#literature432">The Concept of Exemplars</a>
</ul>
</ul>
<b>md5</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#md5296">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#md5147">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#md5248">The Abstract View</a>
</ul>
<li>Tutorial: <i>Looking for Features with Identical Sequence that have Different Annotations</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/looking-for-features-with-identical-sequence-that-have-different-annotations/#md5407">A Program to Look Up Functions Associated with Feature IDs</a>
</ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#md5351">Finding identical genomes</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#md5355">A Small Perl Program to Invoke the CS-API Routines</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#md5360">Correspondence Between Genes from Two Identical Genomes</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#md5365">A Short Program: Two Genes Correspond IFF Locations Are Identical </a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#md5438">The Concept of Exemplars</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#md5498">Begin with the SEED Translation</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#md5501">Create a Table of Data Relating to the MOL Genomes</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#md5464">Creating a Translation Table</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#md5479">Generating the SEED Translations</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#md5529">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>metabolic</b>
<ul>
<li>Tutorial: <i>Annotate a Genome Using KBase Tools </i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#metabolic312">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#metabolic315">Build a metabolic reconstruction from an annotated genome</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#metabolic320">Get roles that might impact metabolic models</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#metabolic323">Create an initial metabolic model</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#metabolic325">Export the model to external tools</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#metabolic184">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#metabolic222">Introduction</a>
</ul>
<li>Tutorial: <i>Comparing the Functional Roles in Two Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/comparing-the-functional-roles-in-two-genomes/#metabolic409">Introduction</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#metabolic344">Navigating the Diagram</a>
</ul>
<li>Tutorial: <i>Constructing RAST2 in the IRIS Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#metabolic36">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#metabolic50"> Tools for Analysis </a>
</ul>
<li>Tutorial: <i>Reconstruction and Analysis of Metabolic Models in KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/reconstruction-and-analysis-of-metabolic-models-in-kbase/#metabolic705">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/reconstruction-and-analysis-of-metabolic-models-in-kbase/#metabolic711">Summary</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#metabolic4">Here</a>
</ul>
<li>Tutorial: <i>The Issue of Retrieving Functional Coupling Data</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/the-issue-of-retrieving-functional-coupling-data/#metabolic417">Introduction</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#metabolic423">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#metabolic429">Limiting the Scope</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#metabolic537">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>multifunctional genes</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#multifunctional genes156">Scripts Handling Popular Operations</a>
</ul>
</ul>
<b>peg</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#peg732">Getting the Primary Function Associated with a Feature</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#peg736">Associating Assignments with Features and Protein Sequences</a>
</ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#peg287">Introduction</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#peg231"> A Short Exercise</a>
</ul>
<li>Tutorial: <i>Comparing the Functional Roles in Two Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/comparing-the-functional-roles-in-two-genomes/#peg414">A Short Program for Extracting Data from the CS</a>
</ul>
<li>Tutorial: <i>Looking for Features with Identical Sequence that have Different Annotations</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/looking-for-features-with-identical-sequence-that-have-different-annotations/#peg404">Here</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#peg20">Here</a>
</ul>
<li>Tutorial: <i>The Issue of Retrieving Functional Coupling Data</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/the-issue-of-retrieving-functional-coupling-data/#peg419">Discussion</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#peg447">The Concept of Exemplars</a>
</ul>
</ul>
<b>pipeline</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#pipeline226">The Basic Philosophy of the CS Command-line Tools</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#pipeline232"> A Short Exercise</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#pipeline233">The Abstract View</a>
</ul>
<li>Tutorial: <i>Constructing RAST2 in the IRIS Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#pipeline49"> Tools for Analysis </a>
</ul>
<li>Tutorial: <i>Some Basic Exercises using the DOE KBase IRIS interface</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/some-basic-exercises-using-the-doe-kbase-iris-interface/#pipeline52">Introduction</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#pipeline388">Accessing the FIGfams from within KBase</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#pipeline560">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>protein families</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#protein families279">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#protein families144">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Constructing RAST2 in the IRIS Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#protein families41"> Tools for Analysis </a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein families366">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein families368">What is a Protein Family?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein families377">What Are FIGfams Used for?</a>
</ul>
</ul>
<b>protein family</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#protein family175">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein family367">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein family369">What is a Protein Family?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein family396">Accessing the FIGfams from within KBase</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein family401">Coming from Roles in Models Instead</a>
</ul>
</ul>
<b>regulon</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#regulon295">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#regulon130">Scripts Handling Popular Operations</a>
</ul>
</ul>
<b>regulons</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#regulons293">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#regulons133">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#regulons517">The close_genomes Operator</a>
</ul>
</ul>
<b>sort</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#sort249">The Abstract View</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises using the DOE KBase IRIS interface</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/some-basic-exercises-using-the-doe-kbase-iris-interface/#sort61">IRIS Unix command summary;</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#sort390">Accessing the FIGfams from within KBase</a>
</ul>
<li>Tutorial: <i>The Issue of Retrieving Functional Coupling Data</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/the-issue-of-retrieving-functional-coupling-data/#sort420">Discussion</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#sort499">Begin with the SEED Translation</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#sort478">Generating the SEED Translations</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#sort566">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>subsystem</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#subsystem275">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#subsystem157">Scripts Handling Popular Operations</a>
</ul>
</ul>
<b>subsystems</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#subsystems273">Introduction</a>
</ul>
<li>Tutorial: <i>Annotate a Genome Using KBase Tools </i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#subsystems316">Build a metabolic reconstruction from an annotated genome</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-kbase/#subsystems159">Scripts Handling Popular Operations</a>
</ul>
<li>Tutorial: <i>Constructing RAST2 in the IRIS Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/constructing-rast2-in-the-iris-environment/#subsystems33">Here</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#subsystems382">What Are FIGfams Used for?</a>
</ul>
</ul>
<b>wc</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#wc251">The Abstract View</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises using the DOE KBase IRIS interface</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/some-basic-exercises-using-the-doe-kbase-iris-interface/#wc64">IRIS Unix command summary;</a>
</ul>
</ul>
