<h1>Index of Terms in KBase Tutorials</h1>
<b>API</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#API56">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#API199">Summary</a>
</ul>
<li>Tutorial: <i>Comparing the Functional Roles in Two Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/comparing-the-functional-roles-in-two-genomes/#API392">A Short Program for Extracting Data from the CS</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#API582">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#API596">Updating or Defining a CS-API "Well-Trodden Path" Routine</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#API2">Here</a>
</ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#API486">Introduction</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#API503">Introduction</a>
</ul>
</ul>
<b>Atomic Regulons</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Atomic Regulons243">Introduction</a>
</ul>
</ul>
<b>Bio::KBase</b>
<ul>
<li>Tutorial: <i>Developing Software in the KBase Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/developing-software-in-the-kbase-environment/#Bio::KBase698">The KBase Module</a>
</ul>
</ul>
<b>CDM</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#CDM58">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#CDM62">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#CDM203">Introduction</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#CDM306">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#CDM311">Navigating the Diagram</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#CDM580">Introduction</a>
</ul>
</ul>
<b>CPAN</b>
<ul>
<li>Tutorial: <i>Getting Started with the KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/getting-started-with-the-kbase/#CPAN44">CPAN installation</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/getting-started-with-the-kbase/#CPAN51">Prepackaged Installations</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/getting-started-with-the-kbase/#CPAN53">Where to find software</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#CPAN14">Here</a>
</ul>
</ul>
<b>CS-API</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#CS-API235">Introduction</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#CS-API579">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#CS-API598">Updating or Defining a CS-API "Well-Trodden Path" Routine</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#CS-API5">Here</a>
</ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#CS-API332">Finding identical genomes</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#CS-API333">A Small Perl Program to Invoke the CS-API Routines</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#CS-API340">Correspondence Between Genes from Two Identical Genomes</a>
</ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#CS-API485">Introduction</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#CS-API536">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>Central Data Model</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#Central Data Model57">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#Central Data Model63">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#Central Data Model202">Introduction</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#Central Data Model305">Introduction</a>
</ul>
</ul>
<b>Central Store</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#Central Store713">Associating Features with Protein Sequences</a>
</ul>
<li>Tutorial: <i>Annotate a Genome Using KBase Tools </i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#Central Store292">Select a genome</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#Central Store55">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#Central Store64">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#Central Store200">Introduction</a>
</ul>
<li>Tutorial: <i>Publishing a New Central Store Software Revision</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/publishing-a-new-central-store-software-revision/#Central Store691">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/publishing-a-new-central-store-software-revision/#Central Store692">Introduction</a>
</ul>
<li>Tutorial: <i>Extracting Genomes, Features, and Sequences from the KBase CS</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/extracting-genomes-features-and-sequences-from-the-kbase-cs/#Central Store324">Here</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#Central Store1">Here</a>
</ul>
</ul>
<b>Co-occurring</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Co-occurring249">Introduction</a>
</ul>
</ul>
<b>Coexpressed</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Coexpressed245">Introduction</a>
</ul>
</ul>
<b>Concerns</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#Concerns70">How to Work with the CDM Entity-Relationship Model</a>
</ul>
</ul>
<b>Contig</b>
<ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#Contig342">A Short Program: Two Genes Correspond IFF Locations Are Identical </a>
</ul>
</ul>
<b>ContigSequence</b>
<ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#ContigSequence343">A Short Program: Two Genes Correspond IFF Locations Are Identical </a>
</ul>
</ul>
<b>DNA</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#DNA238">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#DNA116">"Well-Trodden Paths"</a>
</ul>
<li>Tutorial: <i>Extracting Genomes, Features, and Sequences from the KBase CS</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/extracting-genomes-features-and-sequences-from-the-kbase-cs/#DNA326">Extracting the Genomic Contigs in FASTA Format</a>
<li><a href="http://kbase.us/developer-zone/tutorials/extracting-genomes-features-and-sequences-from-the-kbase-cs/#DNA328">Getting the Sequences of Features</a>
</ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#DNA487">Introduction</a>
</ul>
</ul>
<b>E.coli</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#E.coli100">How to Work with the CDM Entity-Relationship Model</a>
</ul>
</ul>
<b>ER-Model</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#ER-Model59">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#ER-Model65">How to Work with the CDM Entity-Relationship Model</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#ER-Model105">Extracting Fields from a Set of Instances of an Entity</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#ER-Model309">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#ER-Model322">Navigating the Diagram</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#ER-Model581">Introduction</a>
</ul>
</ul>
<b>Entity-Relationship Model</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#Entity-Relationship Model198">Summary</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#Entity-Relationship Model308">Introduction</a>
</ul>
</ul>
<b>Exemplars</b>
<ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#Exemplars8">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#Exemplars468">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#Exemplars434">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary of Function</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-of-function/#Exemplars400">A Three-Part Tutorial</a>
</ul>
</ul>
<b>Family</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Family247">Introduction</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#Family368">Accessing the FIGfams from within KBase</a>
</ul>
</ul>
<b>Feature</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#Feature710">Getting the Primary Function Associated with a Feature</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#Feature712">Associating Features with Protein Sequences</a>
</ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Feature237">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#Feature60">How to Work with the CDM Entity-Relationship Model</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#Feature112">"Well-Trodden Paths"</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#Feature320">Navigating the Diagram</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#Feature419">The Concept of Exemplars</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#Feature435">Here</a>
</ul>
</ul>
<b>Genome</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Genome236">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#Genome97">How to Work with the CDM Entity-Relationship Model</a>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#Genome106">Extracting Fields from a Set of Instances of an Entity</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#Genome312">Navigating the Diagram</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#Genome541">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>Identical Genomes</b>
<ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#Identical Genomes7">Here</a>
</ul>
</ul>
<b>IsCoregulatedWith</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#IsCoregulatedWith61">How to Work with the CDM Entity-Relationship Model</a>
</ul>
</ul>
<b>IsOwnerOf</b>
<ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#IsOwnerOf316">Navigating the Diagram</a>
</ul>
</ul>
<b>IsProteinFor</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#IsProteinFor86">How to Work with the CDM Entity-Relationship Model</a>
</ul>
</ul>
<b>MicrobesOnLine</b>
<ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#MicrobesOnLine12">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#MicrobesOnLine473">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#MicrobesOnLine437">Creating a Translation Table</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#MicrobesOnLine506">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>Model</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#Model102">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#Model404">Here</a>
</ul>
</ul>
<b>OTU</b>
<ul>
<li>Tutorial: <i>Computing Pan Genomes for OTU's</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/computing-pan-genomes-for-otus/#OTU706">Invoking CS Command scripts from Perl</a>
<li><a href="http://kbase.us/developer-zone/tutorials/computing-pan-genomes-for-otus/#OTU707"> Example 2</a>
</ul>
</ul>
<b>Pearson Correlation Coefficient</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Pearson Correlation Coefficient244">Introduction</a>
</ul>
<li>Tutorial: <i>The Issue of Retrieving Functional Coupling Data</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/the-issue-of-retrieving-functional-coupling-data/#Pearson Correlation Coefficient397">Discussion</a>
</ul>
</ul>
<b>Protein Sequence</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Protein Sequence246">Introduction</a>
</ul>
</ul>
<b>ProteinSequence</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#ProteinSequence714">Associating Features with Protein Sequences</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#ProteinSequence69">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#ProteinSequence418">The Concept of Exemplars</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#ProteinSequence509">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>Publication</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#Publication66">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#Publication423">The Concept of Exemplars</a>
</ul>
</ul>
<b>Reaction</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#Reaction101">How to Work with the CDM Entity-Relationship Model</a>
</ul>
</ul>
<b>RegPrecise</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#RegPrecise129">"Well-Trodden Paths"</a>
</ul>
</ul>
<b>Role</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#Role250">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#Role135">"Well-Trodden Paths"</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#Role540">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>SEED</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#SEED716">Associating Assignments with Features and Protein Sequences</a>
</ul>
<li>Tutorial: <i>Getting Started with the KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/getting-started-with-the-kbase/#SEED52">Prepackaged Installations</a>
</ul>
<li>Tutorial: <i>Publishing a New Central Store Software Revision</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/publishing-a-new-central-store-software-revision/#SEED693">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/publishing-a-new-central-store-software-revision/#SEED696">Installing the new release into the KBase git</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#SEED577">Introduction</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#SEED15">Here</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#SEED353">What is a Protein Family?</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#SEED403">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#SEED469">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#SEED474">Begin with the SEED Translation</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#SEED484">Create a Table of Data Relating to the MOL Genomes</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#SEED438">Creating a Translation Table</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#SEED459">Generating the SEED Translations</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#SEED505">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>annotations</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#annotations709">Introduction</a>
</ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#annotations262">Introduction</a>
</ul>
<li>Tutorial: <i>Annotate a Genome Using KBase Tools </i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#annotations293">Annotate a genome object</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#annotations298">Get roles that might impact metabolic models</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#annotations301">Get roles that might have been missed</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#annotations111">"Well-Trodden Paths"</a>
</ul>
<li>Tutorial: <i>Comparing the Functional Roles in Two Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/comparing-the-functional-roles-in-two-genomes/#annotations389">Introduction</a>
</ul>
<li>Tutorial: <i>Looking for Features with Identical Sequence that have Different Annotations</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/looking-for-features-with-identical-sequence-that-have-different-annotations/#annotations384">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/looking-for-features-with-identical-sequence-that-have-different-annotations/#annotations387">A Program to Look Up Functions Associated with Feature IDs</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#annotations10">Here</a>
</ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#annotations338">Controlling the Output </a>
</ul>
<li>Tutorial: <i>The Issue of Retrieving Functional Coupling Data</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/the-issue-of-retrieving-functional-coupling-data/#annotations395">Introduction</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#annotations433">The Concept of Exemplars</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#annotations472">Here</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#annotations436">Creating a Translation Table</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#annotations466">Generating the SEED Translations</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#annotations502">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#annotations504">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>chromosome</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#chromosome264">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#chromosome115">"Well-Trodden Paths"</a>
</ul>
</ul>
<b>close_genomes</b>
<ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#close_genomes495">The close_genomes Operator</a>
</ul>
</ul>
<b>co-regulated</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#co-regulated128">"Well-Trodden Paths"</a>
</ul>
</ul>
<b>conserved neighborhood</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#conserved neighborhood265">Introduction</a>
</ul>
</ul>
<b>contig</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#contig120">"Well-Trodden Paths"</a>
</ul>
<li>Tutorial: <i>Extracting Genomes, Features, and Sequences from the KBase CS</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/extracting-genomes-features-and-sequences-from-the-kbase-cs/#contig327">Getting the Features for a Genome</a>
</ul>
</ul>
<b>corresponds</b>
<ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#corresponds341">A Short Program: Two Genes Correspond IFF Locations Are Identical </a>
</ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#corresponds488">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#corresponds489">The corresponds Operator</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#corresponds498">The close_genomes Operator</a>
</ul>
</ul>
<b>cut</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#cut226">The Abstract View</a>
</ul>
<li>Tutorial: <i>Getting Started with the KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/getting-started-with-the-kbase/#cut49">CPAN installation</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises using the DOE KBase IRIS interface</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/some-basic-exercises-using-the-doe-kbase-iris-interface/#cut42">IRIS Unix command summary;</a>
</ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#cut597">Updating or Defining a CS-API "Well-Trodden Path" Routine</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#cut416">The Concept of Exemplars</a>
</ul>
</ul>
<b>cvs</b>
<ul>
<li>Tutorial: <i>KBase Development in the SEED Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/kbase-development-in-the-seed-environment/#cvs608">Updating or Defining a CS Command-Line Script</a>
</ul>
</ul>
<b>families</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#families259">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#families122">"Well-Trodden Paths"</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#families350">What is a Protein Family?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#families363">What Are FIGfams Used for?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#families366">Accessing the FIGfams from within KBase</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#families381">Coming from Roles in Models Instead</a>
</ul>
</ul>
<b>family</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#family123">"Well-Trodden Paths"</a>
</ul>
<li>Tutorial: <i>Developing Software in the KBase Environment</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/developing-software-in-the-kbase-environment/#family697">A Note on Languages</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#family349">What is a Protein Family?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#family358">What Are FIGfams Used for?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#family372">Accessing the FIGfams from within KBase</a>
</ul>
</ul>
<b>generator</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#generator213">The Abstract View</a>
</ul>
</ul>
<b>generators</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#generators216">The Abstract View</a>
</ul>
</ul>
<b>grep</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#grep99">How to Work with the CDM Entity-Relationship Model</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#grep206">The Basic Philosophy of the CS Command-line Tools</a>
</ul>
<li>Tutorial: <i>Getting Started with the KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/getting-started-with-the-kbase/#grep50">CPAN installation</a>
</ul>
<li>Tutorial: <i>Extracting Genomes, Features, and Sequences from the KBase CS</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/extracting-genomes-features-and-sequences-from-the-kbase-cs/#grep325">Exploring the Collection of Genomes in the CS</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises using the DOE KBase IRIS interface</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/some-basic-exercises-using-the-doe-kbase-iris-interface/#grep32">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/some-basic-exercises-using-the-doe-kbase-iris-interface/#grep39">Additional Features of IRIS</a>
<li><a href="http://kbase.us/developer-zone/tutorials/some-basic-exercises-using-the-doe-kbase-iris-interface/#grep41">IRIS Unix command summary;</a>
</ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#grep331">Finding identical genomes</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#grep543">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>literature</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#literature260">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#literature119">"Well-Trodden Paths"</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#literature411">The Concept of Exemplars</a>
</ul>
</ul>
<b>md5</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#md5275">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#md5124">"Well-Trodden Paths"</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#md5227">The Abstract View</a>
</ul>
<li>Tutorial: <i>Looking for Features with Identical Sequence that have Different Annotations</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/looking-for-features-with-identical-sequence-that-have-different-annotations/#md5386">A Program to Look Up Functions Associated with Feature IDs</a>
</ul>
<li>Tutorial: <i>Processing Identical Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#md5330">Finding identical genomes</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#md5334">A Small Perl Program to Invoke the CS-API Routines</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#md5339">Correspondence Between Genes from Two Identical Genomes</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/processing-identical-genomes/#md5344">A Short Program: Two Genes Correspond IFF Locations Are Identical </a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#md5417">The Concept of Exemplars</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#md5477">Begin with the SEED Translation</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#md5480">Create a Table of Data Relating to the MOL Genomes</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#md5443">Creating a Translation Table</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#md5458">Generating the SEED Translations</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#md5508">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>metabolic</b>
<ul>
<li>Tutorial: <i>Annotate a Genome Using KBase Tools </i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#metabolic291">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#metabolic294">Build a metabolic reconstruction from an annotated genome</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#metabolic299">Get roles that might impact metabolic models</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#metabolic302">Create an initial metabolic model</a>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#metabolic304">Export the model to external tools</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#metabolic163">"Well-Trodden Paths"</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#metabolic201">Introduction</a>
</ul>
<li>Tutorial: <i>Comparing the Functional Roles in Two Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/comparing-the-functional-roles-in-two-genomes/#metabolic388">Introduction</a>
</ul>
<li>Tutorial: <i>The CDM Represented as an Entity-Relationship Model</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/the-cdm-represented-as-an-entity-relationship-model/#metabolic323">Navigating the Diagram</a>
</ul>
<li>Tutorial: <i>Reconstruction and Analysis of Metabolic Models in KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/reconstruction-and-analysis-of-metabolic-models-in-kbase/#metabolic684">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/reconstruction-and-analysis-of-metabolic-models-in-kbase/#metabolic690">Summary</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#metabolic4">Here</a>
</ul>
<li>Tutorial: <i>The Issue of Retrieving Functional Coupling Data</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/the-issue-of-retrieving-functional-coupling-data/#metabolic396">Introduction</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#metabolic402">Here</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#metabolic408">Limiting the Scope</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#metabolic516">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>multifunctional genes</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#multifunctional genes133">"Well-Trodden Paths"</a>
</ul>
</ul>
<b>peg</b>
<ul>
<li>Tutorial: <i>Accessing Annotations from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#peg711">Getting the Primary Function Associated with a Feature</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-annotations-from-the-central-store-cs/#peg715">Associating Assignments with Features and Protein Sequences</a>
</ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#peg266">Introduction</a>
</ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#peg210"> A Short Exercise</a>
</ul>
<li>Tutorial: <i>Comparing the Functional Roles in Two Genomes</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/comparing-the-functional-roles-in-two-genomes/#peg393">A Short Program for Extracting Data from the CS</a>
</ul>
<li>Tutorial: <i>Looking for Features with Identical Sequence that have Different Annotations</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/looking-for-features-with-identical-sequence-that-have-different-annotations/#peg383">Here</a>
</ul>
<li>Tutorial: <i>Tutorials to Help you Get Started</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/tutorials-to-help-you-get-started/#peg20">Here</a>
</ul>
<li>Tutorial: <i>The Issue of Retrieving Functional Coupling Data</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/the-issue-of-retrieving-functional-coupling-data/#peg398">Discussion</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 1: Defining Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-1-defining-exemplars/#peg426">The Concept of Exemplars</a>
</ul>
</ul>
<b>pipeline</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#pipeline205">The Basic Philosophy of the CS Command-line Tools</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#pipeline211"> A Short Exercise</a>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#pipeline212">The Abstract View</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises using the DOE KBase IRIS interface</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/some-basic-exercises-using-the-doe-kbase-iris-interface/#pipeline31">Introduction</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#pipeline367">Accessing the FIGfams from within KBase</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#pipeline539">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>protein families</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#protein families258">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#protein families121">"Well-Trodden Paths"</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein families345">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein families347">What is a Protein Family?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein families356">What Are FIGfams Used for?</a>
</ul>
</ul>
<b>protein family</b>
<ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#protein family154">"Well-Trodden Paths"</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein family346">Introduction</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein family348">What is a Protein Family?</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein family375">Accessing the FIGfams from within KBase</a>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#protein family380">Coming from Roles in Models Instead</a>
</ul>
</ul>
<b>regulon</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#regulon274">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#regulon107">"Well-Trodden Paths"</a>
</ul>
</ul>
<b>regulons</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#regulons272">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#regulons110">"Well-Trodden Paths"</a>
</ul>
<li>Tutorial: <i>Extending the CS-Commands with Operators</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/extending-the-cs-commands-with-operators/#regulons496">The close_genomes Operator</a>
</ul>
</ul>
<b>sort</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#sort228">The Abstract View</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises using the DOE KBase IRIS interface</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/some-basic-exercises-using-the-doe-kbase-iris-interface/#sort40">IRIS Unix command summary;</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#sort369">Accessing the FIGfams from within KBase</a>
</ul>
<li>Tutorial: <i>The Issue of Retrieving Functional Coupling Data</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/example-tasks/the-issue-of-retrieving-functional-coupling-data/#sort399">Discussion</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 3: Making the MOL Translation to the Abstract Function Vocabulary</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-3-making-the-mol-translation-to-the-abstract-function-vocabulary/#sort478">Begin with the SEED Translation</a>
</ul>
<li>Tutorial: <i>Towards a Controlled Vocabulary Part 2: Mapping to Exemplars</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/towards-a-controlled-vocabulary-part-2-mapping-to-exemplars/#sort457">Generating the SEED Translations</a>
</ul>
<li>Tutorial: <i>Mapping Functions Between Sources of Annotations: Take 2</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/command-line-applications/mapping-functions-between-sources-of-annotations-take-2/#sort545">"Clean md5s," "Matched Assertions," and Building a Mapping</a>
</ul>
</ul>
<b>subsystem</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#subsystem254">Introduction</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#subsystem134">"Well-Trodden Paths"</a>
</ul>
</ul>
<b>subsystems</b>
<ul>
<li>Tutorial: <i>Extracting Data from the CS Using the CS-API: Some Typical Examples</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/apis/getting-started-with-kbase-apis/extracting-data-from-the-cs-using-the-cs-api-some-typical-examples/#subsystems252">Introduction</a>
</ul>
<li>Tutorial: <i>Annotate a Genome Using KBase Tools </i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/annotate-a-genome-using-kbase-tools-/#subsystems295">Build a metabolic reconstruction from an annotated genome</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises Using the DOE KBase</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/getting-started/some-basic-exercises-using-the-doe-kbase/#subsystems136">"Well-Trodden Paths"</a>
</ul>
<li>Tutorial: <i>Accessing Protein Family Data from the Central Store (CS)</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/accessing-protein-family-data-from-the-central-store-cs/#subsystems361">What Are FIGfams Used for?</a>
</ul>
</ul>
<b>wc</b>
<ul>
<li>Tutorial: <i>Getting What You Need from the CS Using Command-Line Scripts</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/command-line-scripts/accessing-central-store-data/getting-what-you-need-from-the-cs-using-command-line-scripts/#wc230">The Abstract View</a>
</ul>
<li>Tutorial: <i>Some Basic Exercises using the DOE KBase IRIS interface</i>
<ul>
<li><a href="http://kbase.us/developer-zone/tutorials/some-basic-exercises-using-the-doe-kbase-iris-interface/#wc43">IRIS Unix command summary;</a>
</ul>
</ul>
