<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>aliases_to_fids</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>aliases_to_fids [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>aliases_to_fids is used to search for the feature ids that a set of given aliases maps to.</p>

<p>Example:</p>

<pre>    aliases_to_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each input line, there can be many output lines, one per feature. The feature id is added to the end of the line.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: aliases_to_fids [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_roles_used_in_models</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_roles_used_in_models &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The all_roles_used_in_models allows a user to access the set of roles that are included in current models.
Note that there are far fewer roles used in models than in the complete set of functional roles.
Hence,
the returned set represents the minimal set we need to clean up in order to properly support modeling.</p>

<p>Example:</p>

<pre>    all_roles_used_in_models &#62; output</pre>

<p>This is a pipe command. The is no input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_roles_used_in_models &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_roles_used_in_models_and_subsystems</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_roles_used_in_models_and_subsystems &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The all_roles_used_in_models_and_subsystems allows a user to access the set of roles that are included in current models and subsystems.
This is important.
There are far fewer roles used in models than overall.
Hence,
the returned set represents the minimal set we need to clean up in order to properly support modeling.</p>

<p>Example:</p>

<pre>    all_roles_used_in_models_and_subsystems &#62; output</pre>

<p>This is a pipe command. The is no input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The output is a file in which each line contains just a role used in the construction of models or occurring in a subsystem.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_roles_used_in_models_and_subsystems &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>alleles_to_bp_locs</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>alleles_to_bp_locs [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This command can be used to find the contig and position corresponding to each of a set of alleles.
The command takes in a table with a designated column containing allele IDs.
It adds a column containing the location corresponding to the allele</p>

<p>Example:</p>

<pre>    alleles_to_bp_locs [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the identifier.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: alleles_to_bp_locs [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>alleles_to_traits</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>alleles_to_traits [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Example:</p>

<pre>    alleles_to_traits [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the identifier.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: alleles_to_traits [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>atomic_regulons_to_fids</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>atomic_regulons_to_fids [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Example:</p>

<pre>    atomic_regulons_to_fids [arguments] &#60; input &#62; output</pre>

<p>Atomic_regulons are sets of fids that are believed to have identical expression profiles. The estmates were made using expression data, operon estimates, and subsystems.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: atomic_regulons_to_fids [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>close_genomes</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>close_genomes [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Example:</p>

<pre>    close_genomes [arguments] &#60; input &#62; output</pre>

<p>This is a strange command. It has two quite distinct uses:</p>

<pre>    1. it can be used to find genomes close to existing genomes (stored in either
       the KBase CS or the PubSEED).  
    2. Alternatively, it can be used to compute close genomes for a new genome
       encoded in a JSON file.</pre>

<p>The second use will be performed iff the</p>

<pre>     -g Encoded_JSON_directory</pre>

<p>is used. In that case, the updated genome directory will be written to STDOUT.</p>

<p>If the input is to be one or more genomes from the CS, then the standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the identifier.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: close_genomes [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input
    --n integer
    --g string</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>co_occurrence_evidence</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>co_occurrence_evidence [-c column] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>co-occurence_evidence is used to retrieve the detailed pairs of genes that go into the computation of co-occurence scores.
The scores reflect an estimate of the number of distinct OTUs that contain an instance of a co-occuring pair.
This routine returns as evidence a list of all the pairs that went into the computation.</p>

<p>The input to the computation is table in which one of the columns contains co-occurring pairs.
For example,</p>

<pre>     echo &#39;kb|g.0.peg.101:kb|g.0.peg.263&#39; | co_occurrence_evidence </pre>

<p>takes in a table with 1 column and one row. The input row is a single pair of related fids joined using a &#39;:&#39;.</p>

<p>The output is a table with an added (usually huge) column containing the evidence that the pair of fids are,in fact, related. The evidence is a comma-separated list of pairs (each pair being two fids joined with &#39;:&#39;). Thus the command above produces</p>

<pre>           kb|g.0.peg.101:kb|g.0.peg.263        kb|g.0.peg.101:kb|g.0.peg.263,kb|g.10.peg.3334:kb|g.10.peg.3978,...</pre>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: co_occurrence_evidence [-c column] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<p>The standard output is a tab-delimited file. It consists of the input file with an extra column (containing large evidence strings) added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="POD_ERRORS"
>POD ERRORS</a></h2>

<p>Hey! <b>The above document had some coding errors, which are explained below:</b></p>

<dl>
<dt><a name="Around_line_48:"
>Around line 48:</a></dt>

<dd>
<p>Unknown directive: =head</p>
</dd>
</dl>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>complexes_to_complex_data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>complexes_to_complex_data [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Example:</p>

<pre>    complexes_to_complex_data [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: complexes_to_complex_data [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>complexes_to_roles</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>complexes_to_roles [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Complexes catalyze reactions and are made up of one or more roles.
The roles that make up a complex can be &#34;optional&#34;,
or not.</p>

<p>This command takes as input a table with a column composed of complex ids and adds one column (containing a role that is part of the complex).
The &#34;optional&#34; field is not returned,
so you do not get back a field indicating whether or not the role is optional.
If you need that information,
we recommend that you use</p>

<pre>    get_relationship_IsTriggeredBy -rel optional -to id</pre>

<p>This would tack on both the optional flag and the role.</p>

<p>Example:</p>

<pre>    complexes_to_roles [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: complexes_to_roles [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>contigs_to_lengths</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>contigs_to_lengths [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>In some cases,
one wishes to know just the lengths of the contigs,
rather than their actual DNA sequence (e.g.,
suppose that you wished to know if a gene boundary occured within 100 bp of the end of the contig).
To avoid requiring a user to access the entire DNA sequence,
we offer the ability to retrieve just the contig lengths.
Input to the routine is a list of contig IDs.
The routine returns a mapping from contig IDs to lengths</p>

<p>Example:</p>

<pre>    contigs_to_lengths [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: contigs_to_lengths [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>contigs_to_md5s</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>contigs_to_md5s [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>contigs_to_md5s can be used to acquire MD5 values for each of a list of contigs.
The quickest way to determine whether two contigs are identical is to compare their associated MD5 values,
eliminating the need to retrieve the sequence of each and compare them.</p>

<p>The routine takes as input a list of contig IDs.
The output is a mapping from contig ID to MD5 value.</p>

<p>Example:</p>

<pre>    contigs_to_md5s [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: contigs_to_md5s [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>contigs_to_sequences</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>contigs_to_sequences [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>contigs_to_sequences is used to access the DNA sequence associated with each of a set of input contigs.
It takes as input a set of contig IDs (from which the genome can be determined) and produces a mapping from the input IDs to the returned DNA sequence in each case.</p>

<p>Normally,
people want to get back a fasta file,
so that is the default (the input table is essentially discarded,
keeping only the fasta file).
The alternative to to add a column to the table (containing the contig sequence).
The &#34;-fasta=0&#34; argument should be used,
if you do not want fasta output.</p>

<p>Example:</p>

<pre>    contigs_to_sequences [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: contigs_to_sequences [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    --fasta integer
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>corresponds</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>corresponds [-a cutoff] genome1 genome2 &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This command produces a correspondence between genes in genome=1 and genome-2.
It is often the case that you wish to know which gene in genome X &#34;corresponds&#34; to a specific gene in genome Y.
Frequently the term &#34;ortholog&#34; is used,
although not always accurately.
Here we offer a basic tool that produces a reasonable estimate for the set of genes that form a 1-1 correspondence.</p>

<p>Example:</p>

<pre>    corresponds [-a cutoff] genome1 genome2 &#62; output</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>corresponds [-a cutoff] genome1 genome2 &#62; output</p>

<pre>    -a Cutoff
        This requests an abbreviated format in which only two columns are written:
        the projection score and the fid in the target genome.  If this is not
        used 10 column are produced.  The Cutoff specifies a minimum projection score.

    genome1
    genome2

        Input genomes. These are either filenames which are expected to contain genome
        typed-objects (as created by annotate_genome, for instances) or they are
        KBase genome IDs.</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="OUTPUT_FORMAT"
>OUTPUT FORMAT</a></h2>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added. If the abbreviated format is requested, two columns get added (sc and the fid projected to). If the abbreviated format is not requested, ten columns will be added</p>

<dl>
<dt><a name="1_-_percent-identity:_the_average_percent_identity_over_the_range_of_the_match"
>1 - percent-identity: the average percent identity over the range of the match</a></dt>

<dd>
<dt><a 
>2 - the number of genes in the five adjacent genes to the left and the five adjacent genes to the right (i.e., the number of genes in the &#34;chromosomal context&#34;) that correspond to genes in genome-2 which occur within the chromosomal context of id2</a></dt>

<dd>
<dt><a name="3_-_b1,_the_start_of_the_match_in_the_protein_encoded_by_the_gene_in_genome_-1"
>3 - b1, the start of the match in the protein encoded by the gene in genome -1</a></dt>

<dd>
<dt><a name="4_-_e1,_the_end_of_the_match_in_the_protein_encoded_by_the_gene_in_genome-1"
>4 - e1, the end of the match in the protein encoded by the gene in genome-1</a></dt>

<dd>
<dt><a name="5_-_ln1,_the_length_of_the_protein_encoded_by_the_gene_in_genome-1"
>5 - ln1, the length of the protein encoded by the gene in genome-1</a></dt>

<dd>
<dt><a name="6_-_b2"
>6 - b2</a></dt>

<dd>
<dt><a name="7_-_e2"
>7 - e2</a></dt>

<dd>
<dt><a name="8_-_ln2"
>8 - ln2</a></dt>

<dd>
<dt><a name="9_-_a_score_between_0_and_1_that_reflects_the_reliability_of_the_projection"
>9 - a score between 0 and 1 that reflects the reliability of the projection</a></dt>

<dd>
<dt><a name="10_-_id2,_the_corresponding_gene_in_genome-2"
>10 - id2, the corresponding gene in genome-2</a></dt>

<dd>
<p>Input lines that cannot be extended are written to stderr.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>dump_relationship</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>dump_relationship relname &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Dump all the from/to data for the given relname.
This command talks directly to the Sapling database and thus must run within the KB infrastructure (in other words,
this is not a general purpose command).</p>

<p>Example:</p>

<pre>   dump_relationship relname &#62; output</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: dump_relationship relname &#62; output</p>

<dl>
<dt><a name="relname"
>relname</a></dt>

<dd>
<p>Name of the relationship to dump.</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>Tab delimited file containing to and from links for the given relationship.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>equiv_sequence_assertions</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>equiv_sequence_assertions [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Different groups have made assertions of function for numerous protein sequences.
The equiv_sequence_assertions allows the user to gather function assertions from all of the sources.
Each assertion includes a field indicating whether the person making the assertion viewed themself as an &#34;expert&#34;.
The routine gathers assertions for all proteins having identical protein sequence.</p>

<p>Example:</p>

<pre>    equiv_sequence_assertions [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain a protein identifier (i.e., an md5 value). If another column contains the md5s use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the protein identifiers.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: equiv_sequence_assertions [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>ermodel_to_text</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>ermodel_to_text [--decamelize] &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Given the XML description of an Entity-Relationship model on stdin,
dump the description of the model as an easy-to-parse tab-delimited text file.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: ermodel_to_text [--decamelize] &#62; output</p>

<pre>    --decamelize

        If set, convert all entity, relationship, and field names to a decamelized form.</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="OUTPUT_FORMAT"
>OUTPUT FORMAT</a></h2>

<p>The output is a tab-delimited file. It contains blocks of data delimited by lines containing only the string &#34;//&#34;.</p>

<p>Each block of data describes either an entity or a relationship. A single line defines the entity or relationship name and the associated metadata. The remaining lines in the block define the field data associated with the entity or relationship.</p>

<p>An entity line contains the following fields:</p>

<dl>
<dt><a name="The_string_&#34;entity&#34;"
>The string &#34;entity&#34;</a></dt>

<dd>
<dt><a name="The_entity_name"
>The entity name</a></dt>
</dl>

<p>A relationship line contains the following fields:</p>

<dl>
<dt><a name="The_string_&#34;relationship&#34;"
>The string &#34;relationship&#34;</a></dt>

<dd>
<dt><a name="The_relationship_name"
>The relationship name</a></dt>

<dd>
<dt><a name="The_name_of_the_&#34;from&#34;_entity."
>The name of the &#34;from&#34; entity.</a></dt>

<dd>
<dt><a name="The_name_of_the_&#34;to&#34;_entity."
>The name of the &#34;to&#34; entity.</a></dt>

<dd>
<dt><a name="The_arity_of_the_relationship_(1M_or_MM)."
>The arity of the relationship (1M or MM).</a></dt>

<dd>
<dt><a name="The_name_of_the_converse_relationship."
>The name of the converse relationship.</a></dt>
</dl>

<p>Each field description line contains the following fields:</p>

<dl>
<dt><a name="The_name_of_the_field"
>The name of the field</a></dt>

<dd>
<dt><a name="The_field_type"
>The field type</a></dt>

<dd>
<dt><a name="The_name_of_a_relation,_if_the_field_data_is_stored_in_a_separate_relation."
>The name of a relation, if the field data is stored in a separate relation.</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>external_ids_to_fids</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>external_ids_to_fids [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>external_ids_to_fids is used to search for the feature ids that a set of given external_ids maps to.</p>

<p>Example:</p>

<pre>    external_ids_to_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each input line, there can be many output lines, one per feature. The feature id is added to the end of the line.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: external_ids_to_fids [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>fids_to_annotations</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>fids_to_annotations [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This routine takes as input a list of fids.
It retrieves the existing annotations for each fid,
including the text of the annotation,
who made the annotation and when (as seconds from the epoch).</p>

<p>Example:</p>

<pre>    fids_to_annotations [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: fids_to_annotations [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>fids_to_atomic_regulons</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>fids_to_atomic_regulons [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The fids_to_atomic_regulons allows one to map fids into regulons that contain the fids.
Normally a fid will be in at most one regulon,
but we support multiple regulons.</p>

<p>Example:</p>

<pre>    fids_to_atomic_regulons [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: fids_to_atomic_regulons [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>fids_to_co_occurring_fids</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>fids_to_co_occurring_fids [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>One of the most powerful clues to function relates to conserved clusters of genes on the chromosome (in prokaryotic genomes).
We have attempted to record pairs of genes that tend to occur close to one another on the chromosome.
To meaningfully do this,
we need to construct similarity-based mappings between genes in distinct genomes.
We have constructed such mappings for many (but not all) genomes maintained in the KBase CS.
The prokaryotic geneomes in the CS are grouped into OTUs by ribosomal RNA (genomes within a single OTU have SSU rRNA that is greater than 97% identical).
If two genes occur close to one another (i.e.,
corresponding genes occur close to one another),
then we assign a score,
which is the number of distinct OTUs in which such clustering is detected.
This allows one to normalize for situations in which hundreds of corresponding genes are detected,
but they all come from very closely related genomes.</p>

<p>The significance of the score relates to the number of genomes in the database.
We recommend that you take the time to look at a set of scored pairs and determine approximately what percentage appear to be actually related for a few cutoff values.</p>

<p>Example:</p>

<pre>    fids_to_co_occurring_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: fids_to_co_occurring_fids [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>fids_to_coexpressed_fids</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>fids_to_coexpressed_fids [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The routine fids_to_coexpressed_fids returns (for each input fid) a list of features that appear to be coexpressed.
That is,
for an input fid,
we determine the set of fids from the same genome that have Pearson Correlation Coefficients (based on normalized expression data) greater than 0.5 or less than -0.5.</p>

<p>Example:</p>

<pre>    fids_to_coexpressed_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: fids_to_coexpressed_fids [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>fids_to_dna_sequences</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>fids_to_dna_sequences [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>fids_to_dna_sequences allows the user to look up the DNA sequences corresponding to each of a set of fids.</p>

<p>Example:</p>

<pre>    fids_to_dna_sequences [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: fids_to_dna_sequences [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    --fasta integer
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>fids_to_feature_data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>fids_to_feature_data [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Example:</p>

<pre>    fids_to_feature_data [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: fids_to_feature_data [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>fids_to_functions</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>fids_to_functions [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This routine takes as input a list of fids and returns a mapping from the fids to their assigned functions.</p>

<p>Example:</p>

<pre>    fids_to_functions [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: fids_to_functions [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>fids_to_genomes</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>fids_to_genomes [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Example:</p>

<pre>    fids_to_genomes [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: fids_to_genomes [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>fids_to_literature</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>fids_to_literature [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>We try to associate features and publications,
when the publications constitute supporting evidence of the function.
We connect a paper to a feature when we believe that an &#34;expert&#34; has asserted that the function of the feature is basically what we have associated with the feature.
Thus,
we might attach a paper reporting the crystal structure of a protein,
even though the paper is clearly not the paper responsible for the original characterization.
Our position in this matter is somewhat controversial,
but we are seeking to characterize some assertions as relatively solid,
and this strategy seems to support that goal.
Please note that we certainly wish we could also capture original publications,
and when experts can provide those connections,
we hope that they will help record the associations.</p>

<p>Example:</p>

<pre>    fids_to_literature [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: fids_to_literature [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>fids_to_locations</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>fids_to_locations [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A &#34;location&#34; is a sequence of &#34;regions&#34;.
A region is a contiguous set of bases in a contig.
We work with locations in both the string form and as structures.
fids_to_locations takes as input a list of fids.
For each fid,
a structured location is returned.
The location is a list of regions; a region is given as a pointer to a list containing</p>

<pre>             the contig,
             the beginning base in the contig (from 1).
             the strand (+ or -), and
             the length</pre>

<p>Note that specifying a region using these 4 values allows you to represent a single base-pair region on either strand unambiguously (which giving begin/end pairs does not achieve).</p>

<p>Example:</p>

<pre>    fids_to_locations [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: fids_to_locations [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>fids_to_protein_families</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>fids_to_protein_families [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>KBase supports the creation and maintence of protein families.
Each family is intended to contain a set of isofunctional homologs.
Currently,
the families are collections of translations of features,
rather than of just protein sequences (represented by md5s,
for example).
fids_to_protein_families supports access to the features that have been grouped into a family.
Ideally,
each feature in a family would have the same assigned function.
This is not always true,
but probably should be.</p>

<p>Example:</p>

<pre>    fids_to_protein_families [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: fids_to_protein_families [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>fids_to_protein_sequences</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>fids_to_protein_sequences [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>fids_to_protein_sequences allows the user to look up the amino acid sequences corresponding to each of a set of fids.
You can also get the sequence from proteins (i.e.,
md5 values).
This routine saves you having to look up the md5 sequence and then accessing the protein string in a separate call.</p>

<p>Example:</p>

<pre>    fids_to_protein_sequences [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: fids_to_protein_sequences [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    --fasta integer
    --fc string
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>fids_to_proteins</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>fids_to_proteins [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Example:</p>

<pre>    fids_to_proteins [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: fids_to_proteins [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>fids_to_regulon_data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>fids_to_regulon_data [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Example:</p>

<pre>    fids_to_regulon_data [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: fids_to_regulon_data [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    --fields string
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>fids_to_roles</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>fids_to_roles [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Given a feature,
one can get the set of roles it implements using fid_to_roles.
Remember,
a protein can be multifunctional -- implementing several roles.
This can occur due to fusions or to broad specificity of substrate.</p>

<p>Example:</p>

<pre>    fids_to_roles [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: fids_to_roles [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>fids_to_subsystem_data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>fids_to_subsystem_data [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Example:</p>

<pre>    fids_to_subsystem_data [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: fids_to_subsystem_data [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>fids_to_subsystems</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>fids_to_subsystems [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>fids in subsystems normally have somewhat more reliable assigned functions than those not in subsystems.
Hence,
it is common to ask &#34;Is this protein-encoding gene included in any subsystems?&#34; fids_to_subsystems can be used to see which subsystems contain a fid (or,
you can submit as input a set of fids and get the subsystems for each).</p>

<p>Example:</p>

<pre>    fids_to_subsystems [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: fids_to_subsystems [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>genome_to_intergenic_regions</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>genome_to_intergenic_regions -g GenomeID [-pad=num] &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>genome_to_intergenic_regions is used to get the intergenic regions in a specified genome (specified using the -g argument).
The command allows you to take &#34;padding DNA&#34; from the bounding fids using a -pad argument.</p>

<p>Example:</p>

<pre>    genome_to_intergenic_regions -g &#39;kb|g.0&#39; -pad=10 &#62; output</pre>

<p>This is a pipe command in the sense that it produces a tab-separated list of lines. The columns will correspond to [Fid1,Fid2,padded-intergenic-region]. Overlapping Fids produce no lines in the output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: genome_to_intergenic_regions -g GenomeID [-pad=num] &#62; output</p>

<pre>    -g GenomeID

        This is required and specifies a KBase genome

    -pad PadSize [defaults to 0]

        If the intergenic regions are to be padded with characters from the Fids,
        this specifies the size.</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="OUTPUT_FORMAT"
>OUTPUT FORMAT</a></h2>

<p>The standard output is a tab-delimited file. It consists of 3 columns: [Fid1,Fid2,LocationOfIntergenicRegion]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>genomes_to_contigs</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>genomes_to_contigs [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The routine genomes_to_contigs can be used to retrieve the IDs of the contigs associated with each of a list of input genomes.
The routine constructs a mapping from genome ID to the list of contigs included in the genome.</p>

<p>Example:</p>

<pre>    genomes_to_contigs [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each line of input, there can be multiple output lines, one per contig. The contig id is added to the end of each line.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: genomes_to_contigs [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>genomes_to_fids</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>genomes_to_fids [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>genomes_to_fids is used to get the fids included in specific genomes.
It is often the case that you want just one or two types of fids -- hence,
the types_of_fids argument.</p>

<p>Example:</p>

<pre>    genomes_to_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each input line, ther can be many output lines, one per feature. The feature id is added to the end of the line.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: genomes_to_fids [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>genomes_to_genome_data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>genomes_to_genome_data [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Example:</p>

<pre>    genomes_to_genome_data [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: genomes_to_genome_data [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>genomes_to_md5s</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>genomes_to_md5s [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The routine genomes_to_md5s can be used to look up the MD5 value associated with each of a set of genomes.
The MD5 values are computed when the genome is loaded,
so this routine just retrieves the precomputed values.</p>

<p>Note that the MD5 value of a genome is independent of the contig names and case of the DNA sequences that make up the genome.</p>

<p>Example:</p>

<pre>    genomes_to_md5s [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: genomes_to_md5s [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>genomes_to_subsystems</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>genomes_to_subsystems [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A user can invoke genomes_to_subsystems to rerieve the names of the subsystems relevant to each genome.
The input is a list of genomes.
The output is a mapping from genome to a list of 2-tuples,
where each 2-tuple give a variant code and a subsystem name.
Variant codes of -1 (or *-1) amount to assertions that the genome contains no active variant.
A variant code of 0 means &#34;work in progress&#34;,
and presence or absence of the subsystem in the genome should be undetermined.</p>

<p>Example:</p>

<pre>    genomes_to_subsystems [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: genomes_to_subsystems [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>genomes_to_taxonomies</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>genomes_to_taxonomies [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The routine genomes_to_taxonomies can be used to retrieve taxonomic information for each of a list of input genomes.
For each genome in the input list of genomes,
a list of taxonomic groups is returned.
KBase will use the groups maintained by NCBI.
For an NCBI taxonomic string like</p>

<pre>     cellular organisms;
     Bacteria;
     Proteobacteria;
     Gammaproteobacteria;
     Enterobacteriales;
     Enterobacteriaceae;
     Escherichia;
     Escherichia coli</pre>

<p>associated with the strain &#39;Escherichia coli 1412&#39;, this routine would return a list of these taxonomic groups:</p>

<pre>     [&#39;Bacteria&#39;,
      &#39;Proteobacteria&#39;,
      &#39;Gammaproteobacteria&#39;,
      &#39;Enterobacteriales&#39;,
      &#39;Enterobacteriaceae&#39;,
      &#39;Escherichia&#39;,
      &#39;Escherichia coli&#39;,
      &#39;Escherichia coli 1412&#39;
     ]</pre>

<p>That is, the initial &#34;cellular organisms&#34; has been deleted, and the strain ID has been added as the last &#34;grouping&#34;.</p>

<p>The output is a mapping from genome IDs to lists of the form shown above.</p>

<p>Example:</p>

<pre>    genomes_to_taxonomies [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: genomes_to_taxonomies [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>locations_to_dna_sequences</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>locations_to_dna_sequences [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>locations_to_dna_sequences takes as input a list of locations (each in the form of a list of regions).
The routine constructs 2-tuples composed of</p>

<pre>     [the input location,the dna string]</pre>

<p>The returned DNA string is formed by concatenating the DNA for each of the regions that make up the location.</p>

<p>Example:</p>

<pre>    locations_to_dna_sequences [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: locations_to_dna_sequences [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>locations_to_fids</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>locations_to_fids [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>It is frequently the case that one wishes to look up the genes that occur in a given region of a contig.
Location_to_fids can be used to extract such sets of genes for each region in the input set of regions.
We define a gene as &#34;occuring&#34; in a region if the location of the gene overlaps the designated region.</p>

<p>This command takes as input a table in which one column is a location.
It adds an extra column which will contain a feature that overlaps the location.
if n features overlap a location,
then that input line will generate n output lines (each with an extra column giving one of the overlapping fids).</p>

<p>Example:</p>

<pre>    locations_to_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the location. If another column contains the location use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the location.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: locations_to_fids [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>md5s_to_genomes</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>md5s_to_genomes [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>md5s to genomes is used to get the genomes associated with each of a list of input md5 values.</p>

<pre>           The routine takes as input a list of MD5 values.  It constructs a mapping from each input
           MD5 value to a list of genomes that share the same MD5 value.

           The MD5 value for a genome is independent of the names of contigs and the case of the DNA sequence
           data.</pre>

<p>Example:</p>

<pre>    md5s_to_genomes [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: md5s_to_genomes [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>otu_members</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>otu_members [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>We deine the notion of OTU loosely.
We try to keep genomes with 97% identical SSUrRNAs within the same OTU.
We pick a representative genome for each OTU,
and we try to keep all genomes that have 97% identical SSU rRNAs in the same OTU.</p>

<p>This command is intended to add a column containing the genomes from the same OTU.</p>

<p>Example:</p>

<pre>    otu_members [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain a genome identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: otu_members [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>otus_to_representatives</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>otus_to_representatives [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>We deine the notion of OTU loosely.
We try to keep genomes with 97% identical SSUrRNAs within the same OTU.
We pick a representative genome for each OTU,
and we try to keep all genomes that have 97% identical SSU rRNAs in the same OTU.</p>

<p>This command is intended to add a column containing the representative genome from the given OTU.</p>

<p>Example:</p>

<pre>    otus_to_representatives [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain a genome identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: otus_to_representatives [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>ous_with_trait</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>ous_with_trait -g Genome -mtype MeasurementType -min MinVal -max MaxVal &#60; traits_input &#62; output_with_ous</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Fetches &#34;Observational Units&#34; having a specified trait.
An &#34;Observational Unit&#34; is an individual plant that 1) is part of an experiment or study,
2) has measured traits,
and 3) is assayed for the purpose of determining alleles.</p>

<p>Example:</p>

<pre>    ous_with_trait -g Genome -mtype MeasurementType -min MinVal -max MaxVal &#60; traits_input &#62; output_with_ous</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the identifier.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: ous_with_trait -g Genome -mtype MeasurementType -min MinVal -max MaxVal &#60; traits_input &#62; output_with_ous</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input
    -g genome     We are extracting ous for which this is the reference genome
    -mtype Type   Measurement Type
    -min MinVal   Minimum value of measurement
    -max MaxVal   Maximum value of measurement</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="OUTPUT_FORMAT"
>OUTPUT FORMAT</a></h2>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added (the measurement vlue and an ou with that value).</p>

<p>Input lines that cannot be extended are written to stderr.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>protein_families_to_co_occurring_families</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>protein_families_to_co_occurring_families [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Since we accumulate data relating to the co-occurrence (i.e.,
chromosomal clustering) of genes in prokaryotic genomes,
we can note which pairs of genes tend to co-occur.
From this data,
one can compute the protein families that tend to co-occur (i.e.,
tend to cluster on the chromosome).
This allows one to formulate conjectures for unclustered pairs,
based on clustered pairs from the same protein_families.</p>

<p>Example:</p>

<pre>    protein_families_to_co_occurring_families [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: protein_families_to_co_occurring_families [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>protein_families_to_fids</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>protein_families_to_fids [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>protein_families_to_fids can be used to access the set of fids represented by each of a set of protein_families.
We define protein_families as sets of fids (rather than sets of MD5s.
This may,
or may not,
be a mistake.</p>

<p>Example:</p>

<pre>    protein_families_to_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each input line, there are multiple output lines, one for each fid in the family, the fid is added to the end of each output line.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: protein_families_to_fids [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>protein_families_to_functions</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>protein_families_to_functions [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>protein_families_to_functions can be used to extract the set of functions assigned to the fids that make up the family.
Each input protein_family is mapped to a family function.</p>

<p>Example:</p>

<pre>    protein_families_to_functions [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: protein_families_to_functions [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>protein_families_to_proteins</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>protein_families_to_proteins [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>protein_families_to_proteins can be used to access the set of proteins (i.e.,
the set of MD5 values) represented by each of a set of protein_families.
We define protein_families as sets of fids (rather than sets of MD5s.
This may,
or may not,
be a mistake.</p>

<p>Example:</p>

<pre>    protein_families_to_proteins [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: protein_families_to_proteins [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>proteins_to_fids</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>proteins_to_fids [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>proteins_to_fids takes as input a list of proteins (i.e.,
a list of md5s) and returns for each a set of protein-encoding fids that have the designated sequence as their translation.
That is,
for each sequence,
the returned fids will be the entire set (within Kbase) that have the sequence as a translation.</p>

<p>Example:</p>

<pre>    proteins_to_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each input line, there are multiple output lines, one for each fid the protein maps to. The fid is added to the end of each line.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: proteins_to_fids [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>proteins_to_functions</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>proteins_to_functions [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The routine proteins_to_functions allows users to access functions associated with specific protein sequences.
The input proteins are given as a list of MD5 values (these MD5 values each correspond to a specific protein sequence).
For each input MD5 value,
a list of [feature-id,function] pairs is constructed and returned.
Note that there are many cases in which a single protein sequence corresponds to the translation associated with multiple protein-encoding genes,
and each may have distinct functions (an undesirable situation,
we grant).</p>

<p>This function allows you to access all of the functions assigned (by all annotation groups represented in Kbase) to each of a set of sequences.</p>

<p>Example:</p>

<pre>    proteins_to_functions [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each input line there are multiple output lines, one for each fid the protein maps to. Two columns are added to each output line, a fid and a function.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: proteins_to_functions [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>proteins_to_literature</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>proteins_to_literature [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The routine proteins_to_literature can be used to extract the list of papers we have associated with specific protein sequences.
The user should note that in many cases the association of a paper with a protein sequence is not precise.
That is,
the paper may actually describe a closely-related protein (that may not yet even be in a sequenced genome).
Annotators attempt to use best judgement when associating literature and proteins.
Publication references include [pubmed ID,URL for the paper,
title of the paper].
In some cases,
the URL and title are omitted.
In theory,
we can extract them from PubMed and we will attempt to do so.</p>

<p>Example:</p>

<pre>    proteins_to_literature [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: proteins_to_literature [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>proteins_to_protein_families</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>proteins_to_protein_families [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Protein families contain a set of isofunctional homologs.
proteins_to_protein_families can be used to look up is used to get the set of protein_families containing a specified protein.
For performance reasons,
you can submit a batch of proteins (i.e.,
a list of proteins),
and for each input protein,
you get back a set (possibly empty) of protein_families.
Specific collections of families (e.g.,
FIGfams) usually require that a protein be in at most one family.
However,
we will be integrating protein families from a number of sources,
and so a protein can be in multiple families.</p>

<p>Example:</p>

<pre>    proteins_to_protein_families [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the protein.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each protein, the family it belongs to is added at the end of the input line.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: proteins_to_protein_families [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>proteins_to_roles</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>proteins_to_roles [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The routine proteins_to_roles allows a user to gather the set of functional roles that are associated with specifc protein sequences.
A single protein sequence (designated by an MD5 value) may have numerous associated functions,
since functions are treated as an attribute of the feature,
and multiple features may have precisely the same translation.
In our experience,
it is not uncommon,
even for the best annotation teams,
to assign distinct functions (and,
hence,
functional roles) to identical protein sequences.</p>

<p>For each input MD5 value,
this routine gathers the set of features (fids) that share the same sequence,
collects the associated functions,
expands these into functional roles (for multi-functional proteins),
and returns the set of roles that results.</p>

<p>Note that,
if the user wishes to see the specific features that have the assigned functional roles,
they should use proteins_to_functions instead (it returns the fids associated with each assigned function).</p>

<p>Example:</p>

<pre>    proteins_to_roles [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each input line, there can be multiple output lines, one for each role the protein can map to. The role is added to the end of each line.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: proteins_to_roles [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>proteins_to_sequences</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>proteins_to_sequences [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>proteins_to_sequences allows the user to look up the amino acid sequences corresponding to each of a set of proteins (represented as MD5 hash values) This command allows you to get back formatted fasta files.</p>

<p>Example:</p>

<pre>    proteins_to_sequences [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Command-Line_Options"
>Command-Line Options</a></h3>

<dl>
<dt><a name="-c_Column"
>-c Column</a></dt>

<dd>
<p>This is used only if the column containing the subsystem is not the last column.</p>

<dt><a name="-i_InputFile_[_use_InputFile,_rather_than_stdin_]"
>-i InputFile [ use InputFile, rather than stdin ]</a></dt>

<dd>
<dt><a name="-fasta"
>-fasta</a></dt>

<dd>
<p>This is used to request a fasta output file (dropping all of the other columns in the input lines). It defaults to outputing just a fasta entry.</p>

<dt><a name="-fc_Columns_[_construct_comment_for_fasta_from_these_columns_]"
>-fc Columns [ construct comment for fasta from these columns ]</a></dt>

<dd>
<p>This is used to ask for &#34;fasta comments&#34; formed from one or more columns (comma-separated)</p>
</dd>
</dl>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Output_Format"
>Output Format</a></h3>

<p>The standard output is jsut a fasta file with the sequence. You can also get a tab-delimited file by using -fasta=0. The tab-delimited format consists of the input file with an extra column of sequence added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>reaction_strings</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>reaction_strings [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Reaction_strings are text strings that represent (albeit crudely) the details of Reactions.</p>

<p>Example:</p>

<pre>    reaction_strings [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain a reaction ID. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: reaction_strings [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>reactions_to_complexes</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>reactions_to_complexes [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Reactions are thought of as being either spontaneous or implemented by one or more Complexes.
Complexes connect to Roles.
Hence,
the connection of fids or roles to reactions goes through Complexes.</p>

<p>Complexes are best thought of as things like the ribosome or ATP synthase,
but when a single protein is not grouped with others,
we still think of it as a complex made up of a single member.</p>

<p>Example:</p>

<pre>    reactions_to_complexes [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the reaction identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: reactions_to_complexes [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>region_to_alleles</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>region_to_alleles [-c Column] [-d regionSize] &#60; input &#62; output region_to_alleles [-c Column] [-d regionSize] -i input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Example:</p>

<pre>    region_to_alleles [-d 10000] &#60; input &#62; output</pre>

<p>Here the region will be 20kb centered on the input positions.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the identifier.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: region_to_alleles [-c Column] [-d regionSize] &#60; input &#62; output Usage: region_to_alleles [-c Column] [-d regionSize] -i input &#62; output</p>

<pre>    -d Distance   Width of region in base-pairs on either side of feature (Default=5000)
    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="OUTPUT_FORMAT"
>OUTPUT FORMAT</a></h2>

<p>The standard output is a tab-delimited file. It consists of the input file with extra columns added.</p>

<p>Input lines that cannot be extended are written to stderr.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>region_to_fids</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>region_to_fids [arguments] &#60;input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This command is used to take positions in contigs and to find fids that ocur within a specified distance of each position.
It differs from locations_to_fids in which locations (list of regions) are explicitly given.</p>

<p>Example:</p>

<pre>    region_to_fids -d 10000 &#60; input &#62; output [gets fids within 10kb of the positions]</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the position. If another column contains the positions use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the identifier.</p>

<p>Positions are represented as Contig_Pos or Contig_Pos+1.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: region_to_fids [arguments] &#60;input &#62; output</p>

<pre>    -d Distance   All feature-IDs within this distance of a query will be reported (Default=5000)
    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>regulons_to_fids</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>regulons_to_fids [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The CS contains two similar notions:</p>

<pre>    1. atomic regulons, which are sets of fids thought to have exactly the
       same expression profiles (derived from operons, subsystems, and expression data)

    2. regulons which are thought of as a set of genes impacted by a transcription factor.
       This second notion is built up looking for binding-site/transcription-factor pairs 
       and reconciling the insights with operons and subsystems.</pre>

<p>Clearly the two notions are closely related.</p>

<p>This command takes as input a table in which one column is composed of regulon ids, and it adds a column corresponding to the fids that make up the regulon.</p>

<p>Example:</p>

<pre>    regulons_to_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the regulon identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the regulon ID.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: regulons_to_fids [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>representative</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>representative [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Genomes within KBase have been grouped into OTUs.
Usually,
members of the same OTU have SSU RNAs that are 97% or more identical,
and usually two genomes with 97% identical SSU RNAs are part of the same OTU.
However,
we do not guarantee either assertion,
and there are some genomes that have not been placed into OTUs (usually due to a missing or truncated rRNA).</p>

<p>Example:</p>

<pre>    representative [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the genome identifer. If another column contains the genome identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the genome ID..</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: representative [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>representative_sequences</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>representative_sequences [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Example:</p>

<pre>    representative_sequences [arguments] &#60; input &#62; output</pre>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: representative_sequences [arguments] &#60; input &#62; output</p>

<pre>    --e string
    --o string
    -c num        Select the identifier from column num
    --d string
    --f string
    --m integer
    --s value</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>roles_to_complexes</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>roles_to_complexes [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>roles_to_complexes allows a user to connect Roles to Complexes,
from there,
the connection exists to Reactions (although in the actual ER-model model,
the connection from Complex to Reaction goes through ReactionComplex).
Since Roles also connect to fids,
the connection between fids and Reactions is induced.</p>

<p>Example:</p>

<pre>    roles_to_complexes [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output. For each line of input there may be multiple lines of output, one per complex associated with the role. Two columns are appended to the input line, the optional flag, and the complex id.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: roles_to_complexes [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>roles_to_fids</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>roles_to_fids [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<pre>            You often wish to find the fids in one or more genomes that
            implement specific functional roles.  To do this, you can use
            roles_to_fids.</pre>

<p>Example:</p>

<pre>    roles_to_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: roles_to_fids [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>roles_to_protein_families</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>roles_to_protein_families [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>roles_to_protein_families can be used to locate the protein families containing features that have assigned functions implying that they implement designated roles.
Note that for any input role (given as a role description),
you may have a set of distinct protein_families returned.</p>

<p>Example:</p>

<pre>    roles_to_protein_families [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: roles_to_protein_families [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>roles_to_proteins</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>roles_to_proteins [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>roles_to_proteins can be used to extract the set of proteins (designated by MD5 values) that currently are believed to implement a given role.
Note that the proteins may be multifunctional,
meaning that they may be implementing other roles,
as well.</p>

<p>Example:</p>

<pre>    roles_to_proteins [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: roles_to_proteins [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>roles_to_subsystems</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>roles_to_subsystems [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>roles_to_subsystems can be used to access the set of subsystems that include specific roles.
The input is a list of roles (i.e.,
role descriptions),
and a mapping is returned as a hash with key role description and values composed of sets of susbsystem names.</p>

<p>Example:</p>

<pre>    roles_to_subsystems [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: roles_to_subsystems [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>subsystems_to_fids</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>subsystems_to_fids [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This command gives to access to all of the rows in a subsystem.
It does not support selecting just rows for specific genomes (although the underlying API routine subsystems_to_fids does).
This command gives you all of the</p>

<pre>    [genome,variant-code,fids]</pre>

<p>for an input subsystem.</p>

<p>Example:</p>

<pre>    subsystems_to_fids [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain a subssytem name. If another column contains the subsystem use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: subsystems_to_fids [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>subsystems_to_genomes</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>subsystems_to_genomes [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This command takes as input a table with a column containing subsystem names.
An extra column is appended to the table containing genome IDs (for those genomes included in the subsystem with an active variant code (not matching /\*?(o|-1)/).</p>

<p>Example:</p>

<pre>    subsystems_to_genomes [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the subsystem name. If another column contains the subsystem name use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: subsystems_to_genomes [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>subsystems_to_roles</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>subsystems_to_roles [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The command subsystem_to_roles takes as input a table with a column containing subsystem IDs.
The table is extended with an extra column set to roles that make up the subsystems.
For each input line (designating a subsystem),
a line will be added to the output for each role that occurs in the subsystem.</p>

<p>You can ask for auxiliary roles,
but the default is to leave them out.
If you want them,
use &#34;-aux=1&#34;.
Example:</p>

<pre>    subsystems_to_roles [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the subsystem name. If another column contains the subsystem, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: subsystems_to_roles [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>subsystems_to_spreadsheets</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>subsystems_to_spreadsheets [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The subsystem_to_spreadsheets command allows the user to output the entire spreadsheet for a set of input subsystems.
The input is a table with a column containing subsystem names.
The output is a table with 4 appended columns</p>

<pre>     [Genome,variant-code,role,fid]</pre>

<p>Example:</p>

<pre>    subsystems_to_spreadsheets [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the subsystem name. If another column contains the subsystem use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: subsystems_to_spreadsheets [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>text_search</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>text_search [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>text_search performs a search against a full-text index maintained for the CDMI.
The parameter &#34;input&#34; is the text string to be searched for.
The parameter &#34;entities&#34; defines the entities to be searched.
If the list is empty,
all indexed entities will be searched.
The &#34;start&#34; and &#34;count&#34; parameters limit the results to &#34;count&#34; hits starting at &#34;start&#34;.</p>

<p>Example:</p>

<pre>    text_search [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the subsystem.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: text_search [arguments] &#60; input &#62; output</p>

<pre>    --start string
    --count string
    --entity string</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>traits_to_alleles</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>traits_to_alleles [arguments] &#60; input &#62; output</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Example:</p>

<pre>    traits_to_alleles [arguments] &#60; input &#62; output</pre>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the identifer. If another column contains the identifier use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the identifier.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: traits_to_alleles [arguments] &#60; input &#62; output</p>

<pre>    -c num        Select the identifier from column num
    -i filename   Use filename rather than stdin for input</pre>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
