<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Alignment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Alignment [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Alignment entity.</p>

<p>An alignment arranges a group of sequences so that they match.
Each alignment is associated with a phylogenetic tree that describes how the sequences developed and their evolutionary distance.</p>

<p>Example:</p>

<pre>    all_entities_Alignment -a </pre>

<p>would retrieve all entities of type Alignment and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Alignment entity has the following relationship links:</p>

<dl>
<dt><a name="HasAlignmentAttribute_AlignmentAttribute"
>HasAlignmentAttribute AlignmentAttribute</a></dt>

<dd>
<dt><a name="IncludesAlignmentRow_AlignmentRow"
>IncludesAlignmentRow AlignmentRow</a></dt>

<dd>
<dt><a name="IsModificationOfAlignment_Alignment"
>IsModificationOfAlignment Alignment</a></dt>

<dd>
<dt><a name="IsModifiedToBuildAlignment_Alignment"
>IsModifiedToBuildAlignment Alignment</a></dt>

<dd>
<dt><a name="IsSupersededByAlignment_Alignment"
>IsSupersededByAlignment Alignment</a></dt>

<dd>
<dt><a name="IsUsedToBuildTree_Tree"
>IsUsedToBuildTree Tree</a></dt>

<dd>
<dt><a name="SupersedesAlignment_Alignment"
>SupersedesAlignment Alignment</a></dt>

<dd>
<dt><a name="WasAlignedBy_Source"
>WasAlignedBy Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Alignment [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<p>number of rows in the alignment</p>

<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<p>number of columns in the alignment</p>

<dt><a name="status"
>status</a></dt>

<dd>
<p>status of the alignment, currently either [i]active[/i], [i]superseded[/i], or [i]bad[/i]</p>

<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<p>TRUE if the rows of the alignment map to multiple sequences, FALSE if they map to single sequences</p>

<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<p>type of sequence being aligned, currently either [i]Protein[/i], [i]DNA[/i], [i]RNA[/i], or [i]Mixed[/i]</p>

<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<p>date and time the alignment was loaded</p>

<dt><a name="method"
>method</a></dt>

<dd>
<p>name of the primary software package or script used to construct the alignment</p>

<dt><a name="parameters"
>parameters</a></dt>

<dd>
<p>non-default parameters used as input to the software package or script indicated in the method attribute</p>

<dt><a name="protocol"
>protocol</a></dt>

<dd>
<p>description of the steps taken to construct the alignment, or a reference to an external pipeline</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>ID of this alignment in the source database</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_AlignmentAttribute</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_AlignmentAttribute [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the AlignmentAttribute entity.</p>

<p>This entity represents an attribute type that can be assigned to an alignment.
The attribute values are stored in the relationships to the target.
The key is the attribute name.</p>

<p>Example:</p>

<pre>    all_entities_AlignmentAttribute -a </pre>

<p>would retrieve all entities of type AlignmentAttribute and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AlignmentAttribute entity has the following relationship links:</p>

<dl>
<dt><a name="DescribesAlignment_Alignment"
>DescribesAlignment Alignment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_AlignmentAttribute [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_AlignmentRow</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_AlignmentRow [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the AlignmentRow entity.</p>

<p>This entity represents a single row of an alignment.
In general,
this corresponds to a sequence,
but in a concatenated alignment multiple sequences may be represented here.</p>

<p>Example:</p>

<pre>    all_entities_AlignmentRow -a </pre>

<p>would retrieve all entities of type AlignmentRow and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AlignmentRow entity has the following relationship links:</p>

<dl>
<dt><a name="ContainsAlignedDNA_ContigSequence"
>ContainsAlignedDNA ContigSequence</a></dt>

<dd>
<dt><a name="ContainsAlignedProtein_ProteinSequence"
>ContainsAlignedProtein ProteinSequence</a></dt>

<dd>
<dt><a name="IsAlignmentRowIn_Alignment"
>IsAlignmentRowIn Alignment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_AlignmentRow [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="row_number"
>row_number</a></dt>

<dd>
<p>1-based ordinal number of this row in the alignment</p>

<dt><a name="row_id"
>row_id</a></dt>

<dd>
<p>identifier for this row in the FASTA file for the alignment</p>

<dt><a name="row_description"
>row_description</a></dt>

<dd>
<p>description of this row in the FASTA file for the alignment</p>

<dt><a name="n_components"
>n_components</a></dt>

<dd>
<p>number of components that make up this alignment row; for a single-sequence alignment this is always &#34;1&#34;</p>

<dt><a name="beg_pos_aln"
>beg_pos_aln</a></dt>

<dd>
<p>the 1-based column index in the alignment where this sequence row begins</p>

<dt><a name="end_pos_aln"
>end_pos_aln</a></dt>

<dd>
<p>the 1-based column index in the alignment where this sequence row ends</p>

<dt><a name="md5_of_ungapped_sequence"
>md5_of_ungapped_sequence</a></dt>

<dd>
<p>the MD5 of this row&#39;s sequence after gaps have been removed; for DNA and RNA sequences, the [b]U[/b] codes are also normalized to [b]T[/b] before the MD5 is computed</p>

<dt><a name="sequence"
>sequence</a></dt>

<dd>
<p>sequence for this alignment row (with indels)</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_AlleleFrequency</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_AlleleFrequency [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the AlleleFrequency entity.</p>

<p>An allele frequency represents a summary of the major and minor allele frequencies for a position on a chromosome.</p>

<p>Example:</p>

<pre>    all_entities_AlleleFrequency -a </pre>

<p>would retrieve all entities of type AlleleFrequency and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AlleleFrequency entity has the following relationship links:</p>

<dl>
<dt><a name="Summarizes_Contig"
>Summarizes Contig</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_AlleleFrequency [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>identifier for this allele in the original (source) database</p>

<dt><a name="position"
>position</a></dt>

<dd>
<p>Specific position on the contig where the allele occurs</p>

<dt><a name="minor_AF"
>minor_AF</a></dt>

<dd>
<p>Minor allele frequency. Floating point number from 0.0 to 0.5.</p>

<dt><a name="minor_allele"
>minor_allele</a></dt>

<dd>
<p>Text letter representation of the minor allele. Valid values are A, C, G, and T.</p>

<dt><a name="major_AF"
>major_AF</a></dt>

<dd>
<p>Major allele frequency. Floating point number less than or equal to 1.0.</p>

<dt><a name="major_allele"
>major_allele</a></dt>

<dd>
<p>Text letter representation of the major allele. Valid values are A, C, G, and T.</p>

<dt><a name="obs_unit_count"
>obs_unit_count</a></dt>

<dd>
<p>Number of observational units used to compute the allele frequencies. Indicates the quality of the analysis.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Annotation</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Annotation [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Annotation entity.</p>

<p>An annotation is a comment attached to a feature.
Annotations are used to track the history of a feature&#39;s functional assignments and any related issues.
The key is the feature ID followed by a colon and a complemented ten-digit sequence number.</p>

<p>Example:</p>

<pre>    all_entities_Annotation -a </pre>

<p>would retrieve all entities of type Annotation and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Annotation entity has the following relationship links:</p>

<dl>
<dt><a name="Annotates_Feature"
>Annotates Feature</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Annotation [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="annotator"
>annotator</a></dt>

<dd>
<p>name of the annotator who made the comment</p>

<dt><a name="comment"
>comment</a></dt>

<dd>
<p>text of the annotation</p>

<dt><a name="annotation_time"
>annotation_time</a></dt>

<dd>
<p>date and time at which the annotation was made</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Assay</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Assay [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Assay entity.</p>

<p>An assay is an experimental design for determining alleles at specific chromosome positions.</p>

<p>Example:</p>

<pre>    all_entities_Assay -a </pre>

<p>would retrieve all entities of type Assay and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Assay entity has the following relationship links:</p>

<dl>
<dt><a name="IsAssayOf_StudyExperiment"
>IsAssayOf StudyExperiment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Assay [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>identifier for this assay in the original (source) database</p>

<dt><a name="assay_type"
>assay_type</a></dt>

<dd>
<p>Text description of the type of assay (e.g., SNP, length, sequence, categorical, array, short read, SSR marker, AFLP marker)</p>

<dt><a name="assay_type_id"
>assay_type_id</a></dt>

<dd>
<p>source ID associated with the assay type (informational)</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_AtomicRegulon</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_AtomicRegulon [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the AtomicRegulon entity.</p>

<p>An atomic regulon is an indivisible group of coregulated features on a single genome.
Atomic regulons are constructed so that a given feature can only belong to one.
Because of this,
the expression levels for atomic regulons represent in some sense the state of a cell.</p>

<p>Example:</p>

<pre>    all_entities_AtomicRegulon -a </pre>

<p>would retrieve all entities of type AtomicRegulon and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AtomicRegulon entity has the following relationship links:</p>

<dl>
<dt><a name="IsAffectedIn_Experiment"
>IsAffectedIn Experiment</a></dt>

<dd>
<dt><a name="IsFormedOf_Feature"
>IsFormedOf Feature</a></dt>

<dd>
<dt><a name="ReflectsStateOf_Genome"
>ReflectsStateOf Genome</a></dt>

<dd>
<dt><a name="WasGeneratedFrom_ProbeSet"
>WasGeneratedFrom ProbeSet</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_AtomicRegulon [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Attribute</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Attribute [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Attribute entity.</p>

<p>An attribute describes a category of condition or characteristic for an experiment.
The goals of the experiment can be inferred from its values for all the attributes of interest.</p>

<p>Example:</p>

<pre>    all_entities_Attribute -a </pre>

<p>would retrieve all entities of type Attribute and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Attribute entity has the following relationship links:</p>

<dl>
<dt><a name="HasValueIn_Experiment"
>HasValueIn Experiment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Attribute [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="description"
>description</a></dt>

<dd>
<p>Descriptive text indicating the nature and use of this attribute.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Biomass</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Biomass [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Biomass entity.</p>

<p>A biomass is a collection of compounds in a specific ratio and in specific compartments that are necessary for a cell to function properly.
The prediction of biomasses is key to the functioning of the model.
Each biomass belongs to a specific model.</p>

<p>Example:</p>

<pre>    all_entities_Biomass -a </pre>

<p>would retrieve all entities of type Biomass and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Biomass entity has the following relationship links:</p>

<dl>
<dt><a name="IsComprisedOf_CompoundInstance"
>IsComprisedOf CompoundInstance</a></dt>

<dd>
<dt><a name="IsManagedBy_Model"
>IsManagedBy Model</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Biomass [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<p>last modification date of the biomass data</p>

<dt><a name="name"
>name</a></dt>

<dd>
<p>descriptive name for this biomass</p>

<dt><a name="dna"
>dna</a></dt>

<dd>
<p>portion of a gram of this biomass (expressed as a fraction of 1.0) that is DNA</p>

<dt><a name="protein"
>protein</a></dt>

<dd>
<p>portion of a gram of this biomass (expressed as a fraction of 1.0) that is protein</p>

<dt><a name="cell_wall"
>cell_wall</a></dt>

<dd>
<p>portion of a gram of this biomass (expressed as a fraction of 1.0) that is cell wall</p>

<dt><a name="lipid"
>lipid</a></dt>

<dd>
<p>portion of a gram of this biomass (expressed as a fraction of 1.0) that is lipid but is not part of the cell wall</p>

<dt><a name="cofactor"
>cofactor</a></dt>

<dd>
<p>portion of a gram of this biomass (expressed as a fraction of 1.0) that function as cofactors</p>

<dt><a name="energy"
>energy</a></dt>

<dd>
<p>number of ATP molecules hydrolized per gram of this biomass</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_CodonUsage</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_CodonUsage [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the CodonUsage entity.</p>

<p>This entity contains information about the codon usage frequency in a particular genome with respect to a particular type of analysis (e.g.
high-expression genes,
modal,
mean,
etc.).</p>

<p>Example:</p>

<pre>    all_entities_CodonUsage -a </pre>

<p>would retrieve all entities of type CodonUsage and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The CodonUsage entity has the following relationship links:</p>

<dl>
<dt><a name="AreCodonsFor_Genome"
>AreCodonsFor Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_CodonUsage [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="frequencies"
>frequencies</a></dt>

<dd>
<p>A packed-string representation of the codon usage frequencies. These are not global frequencies, but rather frequenicy of use relative to other codons that produce the same amino acid.</p>

<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<p>Genetic code used for these codons.</p>

<dt><a name="type"
>type</a></dt>

<dd>
<p>Type of frequency analysis: average, modal, high-expression, or non-native.</p>

<dt><a name="subtype"
>subtype</a></dt>

<dd>
<p>Specific nature of the codon usage with respect to the given type, generally indicative of how the frequencies were computed.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Complex</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Complex [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Complex entity.</p>

<p>A complex is a set of chemical reactions that act in concert to effect a role.</p>

<p>Example:</p>

<pre>    all_entities_Complex -a </pre>

<p>would retrieve all entities of type Complex and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Complex entity has the following relationship links:</p>

<dl>
<dt><a name="HasStep_Reaction"
>HasStep Reaction</a></dt>

<dd>
<dt><a name="IsTriggeredBy_Role"
>IsTriggeredBy Role</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Complex [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<p>name of this complex. Not all complexes have names.</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>ID of this complex in the source from which it was added.</p>

<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<p>date and time of the last change to this complex&#39;s definition</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Compound</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Compound [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Compound entity.</p>

<p>A compound is a chemical that participates in a reaction.
Both ligands and reaction components are treated as compounds.</p>

<p>Example:</p>

<pre>    all_entities_Compound -a </pre>

<p>would retrieve all entities of type Compound and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Compound entity has the following relationship links:</p>

<dl>
<dt><a name="ComponentOf_Compound"
>ComponentOf Compound</a></dt>

<dd>
<dt><a name="CompoundMeasuredBy_Measurement"
>CompoundMeasuredBy Measurement</a></dt>

<dd>
<dt><a name="ConsistsOfCompounds_Compound"
>ConsistsOfCompounds Compound</a></dt>

<dd>
<dt><a name="IncludedIn_Environment"
>IncludedIn Environment</a></dt>

<dd>
<dt><a name="IsPresentIn_Media"
>IsPresentIn Media</a></dt>

<dd>
<dt><a name="IsShownOn_Diagram"
>IsShownOn Diagram</a></dt>

<dd>
<dt><a name="IsTerminusFor_Scenario"
>IsTerminusFor Scenario</a></dt>

<dd>
<dt><a name="ParticipatesAs_LocalizedCompound"
>ParticipatesAs LocalizedCompound</a></dt>

<dd>
<dt><a name="UsesAliasForCompound_Source"
>UsesAliasForCompound Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Compound [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="label"
>label</a></dt>

<dd>
<p>primary name of the compound, for use in displaying reactions</p>

<dt><a name="abbr"
>abbr</a></dt>

<dd>
<p>shortened abbreviation for the compound name</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>common modeling ID of this compound</p>

<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<p>TRUE if this compound is found in most reactions, else FALSE</p>

<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<p>date and time of the last modification to the compound definition</p>

<dt><a name="mass"
>mass</a></dt>

<dd>
<p>pH-neutral atomic mass of the compound</p>

<dt><a name="formula"
>formula</a></dt>

<dd>
<p>a pH-neutral formula for the compound</p>

<dt><a name="charge"
>charge</a></dt>

<dd>
<p>computed charge of the compound in a pH-neutral solution</p>

<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<p>the pH 7 reference Gibbs free-energy of formation for this compound as calculated by the group contribution method (units are kcal/mol)</p>

<dt><a name="deltaG_error"
>deltaG_error</a></dt>

<dd>
<p>the uncertainty in the [b]deltaG[/b] value (units are kcal/mol)</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_CompoundInstance</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_CompoundInstance [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the CompoundInstance entity.</p>

<p>A Compound Instance represents the occurrence of a particular compound in a location in a model.</p>

<p>Example:</p>

<pre>    all_entities_CompoundInstance -a </pre>

<p>would retrieve all entities of type CompoundInstance and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The CompoundInstance entity has the following relationship links:</p>

<dl>
<dt><a name="Comprises_Biomass"
>Comprises Biomass</a></dt>

<dd>
<dt><a name="HasRealLocationIn_LocationInstance"
>HasRealLocationIn LocationInstance</a></dt>

<dd>
<dt><a name="IsReagentIn_ReactionInstance"
>IsReagentIn ReactionInstance</a></dt>

<dd>
<dt><a name="IsUsageOf_LocalizedCompound"
>IsUsageOf LocalizedCompound</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_CompoundInstance [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="charge"
>charge</a></dt>

<dd>
<p>computed charge based on the location instance pH and similar constraints</p>

<dt><a name="formula"
>formula</a></dt>

<dd>
<p>computed chemical formula for this compound based on the location instance pH and similar constraints</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Contig</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Contig [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Contig entity.</p>

<p>A contig is thought of as composing a part of the DNA associated with a specific genome.
It is represented as an ID (including the genome ID) and a ContigSequence.
We do not think of strings of DNA from,
say,
a metgenomic sample as &#34;contigs&#34;,
since there is no associated genome (these would be considered ContigSequences).
This use of the term &#34;ContigSequence&#34;,
rather than just &#34;DNA sequence&#34;,
may turn out to be a bad idea.
For now,
you should just realize that a Contig has an associated genome,
but a ContigSequence does not.</p>

<p>Example:</p>

<pre>    all_entities_Contig -a </pre>

<p>would retrieve all entities of type Contig and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Contig entity has the following relationship links:</p>

<dl>
<dt><a name="HasAsSequence_ContigSequence"
>HasAsSequence ContigSequence</a></dt>

<dd>
<dt><a name="HasVariationIn_ObservationalUnit"
>HasVariationIn ObservationalUnit</a></dt>

<dd>
<dt><a name="IsComponentOf_Genome"
>IsComponentOf Genome</a></dt>

<dd>
<dt><a name="IsImpactedBy_Trait"
>IsImpactedBy Trait</a></dt>

<dd>
<dt><a name="IsLocusFor_Feature"
>IsLocusFor Feature</a></dt>

<dd>
<dt><a name="IsSummarizedBy_AlleleFrequency"
>IsSummarizedBy AlleleFrequency</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Contig [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>ID of this contig from the core (source) database</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_ContigChunk</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_ContigChunk [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the ContigChunk entity.</p>

<p>ContigChunks are strings of DNA thought of as being a string in a 4-character alphabet with an associated ID.
We allow a broader alphabet that includes U (for RNA) and the standard ambiguity characters.</p>

<p>Example:</p>

<pre>    all_entities_ContigChunk -a </pre>

<p>would retrieve all entities of type ContigChunk and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ContigChunk entity has the following relationship links:</p>

<dl>
<dt><a name="IsSectionOf_ContigSequence"
>IsSectionOf ContigSequence</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_ContigChunk [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="sequence"
>sequence</a></dt>

<dd>
<p>base pairs that make up this sequence</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_ContigSequence</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_ContigSequence [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the ContigSequence entity.</p>

<p>ContigSequences are strings of DNA.
Contigs have an associated genome,
but ContigSequences do not.
We can think of random samples of DNA as a set of ContigSequences.
There are no length constraints imposed on ContigSequences -- they can be either very short or very long.
The basic unit of data that is moved to/from the database is the ContigChunk,
from which ContigSequences are formed.
The key of a ContigSequence is the sequence&#39;s MD5 identifier.</p>

<p>Example:</p>

<pre>    all_entities_ContigSequence -a </pre>

<p>would retrieve all entities of type ContigSequence and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ContigSequence entity has the following relationship links:</p>

<dl>
<dt><a name="HasSection_ContigChunk"
>HasSection ContigChunk</a></dt>

<dd>
<dt><a name="IsAlignedDNAComponentOf_AlignmentRow"
>IsAlignedDNAComponentOf AlignmentRow</a></dt>

<dd>
<dt><a name="IsSequenceOf_Contig"
>IsSequenceOf Contig</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_ContigSequence [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="length"
>length</a></dt>

<dd>
<p>number of base pairs in the contig</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_CoregulatedSet</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_CoregulatedSet [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the CoregulatedSet entity.</p>

<p>We need to represent sets of genes that are coregulated via some regulatory mechanism.
In particular,
we wish to represent genes that are coregulated using transcription binding sites and corresponding transcription regulatory proteins.
We represent a coregulated set (which may,
or may not,
be considered a regulon) using CoregulatedSet.</p>

<p>Example:</p>

<pre>    all_entities_CoregulatedSet -a </pre>

<p>would retrieve all entities of type CoregulatedSet and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The CoregulatedSet entity has the following relationship links:</p>

<dl>
<dt><a name="IsControlledUsing_Feature"
>IsControlledUsing Feature</a></dt>

<dd>
<dt><a name="IsRegulatedSetOf_Feature"
>IsRegulatedSetOf Feature</a></dt>

<dd>
<dt><a name="WasFormulatedBy_Source"
>WasFormulatedBy Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_CoregulatedSet [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>original ID of this coregulated set in the source (core) database</p>

<dt><a name="binding_location"
>binding_location</a></dt>

<dd>
<p>binding location for this set&#39;s transcription factor; there may be none of these or there may be more than one</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Diagram</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Diagram [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Diagram entity.</p>

<p>A functional diagram describes a network of chemical reactions,
often comprising a single subsystem.</p>

<p>Example:</p>

<pre>    all_entities_Diagram -a </pre>

<p>would retrieve all entities of type Diagram and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Diagram entity has the following relationship links:</p>

<dl>
<dt><a name="Displays_Reaction"
>Displays Reaction</a></dt>

<dd>
<dt><a name="IncludesPartOf_Scenario"
>IncludesPartOf Scenario</a></dt>

<dd>
<dt><a name="IsRelevantFor_Subsystem"
>IsRelevantFor Subsystem</a></dt>

<dd>
<dt><a name="Shows_Compound"
>Shows Compound</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Diagram [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<p>descriptive name of this diagram</p>

<dt><a name="content"
>content</a></dt>

<dd>
<p>content of the diagram, in PNG format</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_EcNumber</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_EcNumber [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the EcNumber entity.</p>

<p>EC numbers are assigned by the Enzyme Commission,
and consist of four numbers separated by periods,
each indicating a successively smaller cateogry of enzymes.</p>

<p>Example:</p>

<pre>    all_entities_EcNumber -a </pre>

<p>would retrieve all entities of type EcNumber and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The EcNumber entity has the following relationship links:</p>

<dl>
<dt><a name="IsConsistentWith_Role"
>IsConsistentWith Role</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_EcNumber [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="obsolete"
>obsolete</a></dt>

<dd>
<p>This boolean indicates when an EC number is obsolete.</p>

<dt><a name="replacedby"
>replacedby</a></dt>

<dd>
<p>When an obsolete EC number is replaced with another EC number, this string will hold the name of the replacement EC number.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Environment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Environment [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Environment entity.</p>

<p>An Environment is a set of conditions for microbial growth,
including temperature,
aerobicity,
media,
and supplementary conditions.</p>

<p>Example:</p>

<pre>    all_entities_Environment -a </pre>

<p>would retrieve all entities of type Environment and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Environment entity has the following relationship links:</p>

<dl>
<dt><a name="HasMedia_Media"
>HasMedia Media</a></dt>

<dd>
<dt><a name="HasParameter_Parameter"
>HasParameter Parameter</a></dt>

<dd>
<dt><a name="IncludesAdditionalCompounds_Compound"
>IncludesAdditionalCompounds Compound</a></dt>

<dd>
<dt><a name="IsContextOf_ExperimentalUnit"
>IsContextOf ExperimentalUnit</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Environment [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="temperature"
>temperature</a></dt>

<dd>
<p>The temperature in Kelvin.</p>

<dt><a name="description"
>description</a></dt>

<dd>
<p>A description of the environment.</p>

<dt><a name="oxygenConcentration"
>oxygenConcentration</a></dt>

<dd>
<p>The oxygen concentration in the environment in Molar (mol/L). A value of -1 indicates that there is oxygen in the environment but the concentration is not known, (e.g. an open air shake flask experiment).</p>

<dt><a name="pH"
>pH</a></dt>

<dd>
<p>The pH of the media used in the environment.</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the environment used by the data source.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Experiment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Experiment [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Experiment entity.</p>

<p>An experiment is a combination of conditions for which gene expression information is desired.
The result of the experiment is a set of expression levels for features under the given conditions.</p>

<p>Example:</p>

<pre>    all_entities_Experiment -a </pre>

<p>would retrieve all entities of type Experiment and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Experiment entity has the following relationship links:</p>

<dl>
<dt><a name="AffectsLevelOf_AtomicRegulon"
>AffectsLevelOf AtomicRegulon</a></dt>

<dd>
<dt><a name="HasResultsFor_ProbeSet"
>HasResultsFor ProbeSet</a></dt>

<dd>
<dt><a name="HasValueFor_Attribute"
>HasValueFor Attribute</a></dt>

<dd>
<dt><a name="IndicatesSignalFor_Feature"
>IndicatesSignalFor Feature</a></dt>

<dd>
<dt><a name="OperatesIn_Media"
>OperatesIn Media</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Experiment [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source"
>source</a></dt>

<dd>
<p>Publication or lab relevant to this experiment.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_ExperimentalUnit</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_ExperimentalUnit [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the ExperimentalUnit entity.</p>

<p>An ExperimentalUnit is a subset of an experiment consisting of a Strain,
an Environment,
and one or more Measurements on that strain in the specified environment.
ExperimentalUnits belong to a single experiment.</p>

<p>Example:</p>

<pre>    all_entities_ExperimentalUnit -a </pre>

<p>would retrieve all entities of type ExperimentalUnit and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ExperimentalUnit entity has the following relationship links:</p>

<dl>
<dt><a name="GroupedBy_ExperimentalUnitGroup"
>GroupedBy ExperimentalUnitGroup</a></dt>

<dd>
<dt><a name="HasEnvironment_Environment"
>HasEnvironment Environment</a></dt>

<dd>
<dt><a name="HasMeasurement_Measurement"
>HasMeasurement Measurement</a></dt>

<dd>
<dt><a name="IncludesStrain_Strain"
>IncludesStrain Strain</a></dt>

<dd>
<dt><a name="IsExperimentalUnitOf_ExperimentMeta"
>IsExperimentalUnitOf ExperimentMeta</a></dt>

<dd>
<dt><a name="IsTimepointOf_TimeSeries"
>IsTimepointOf TimeSeries</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_ExperimentalUnit [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the experimental unit used by the data source.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_ExperimentalUnitGroup</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_ExperimentalUnitGroup [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the ExperimentalUnitGroup entity.</p>

<p>An ExperimentalUnitGroup allows for grouping related experimental units and their measurements - for instance measurements that were in the same plate.</p>

<p>Example:</p>

<pre>    all_entities_ExperimentalUnitGroup -a </pre>

<p>would retrieve all entities of type ExperimentalUnitGroup and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ExperimentalUnitGroup entity has the following relationship links:</p>

<dl>
<dt><a name="ContainsExperimentalUnit_ExperimentalUnit"
>ContainsExperimentalUnit ExperimentalUnit</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_ExperimentalUnitGroup [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the experimental unit group used by the data source.</p>

<dt><a name="name"
>name</a></dt>

<dd>
<p>The name of this group, if any.</p>

<dt><a name="comments"
>comments</a></dt>

<dd>
<p>Any comments about this group.</p>

<dt><a name="groupType"
>groupType</a></dt>

<dd>
<p>The type of this grouping, for example &#39;24 well plate&#39;, &#39;96 well plate&#39;, &#39;384 well plate&#39;, &#39;microarray&#39;.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_ExperimentMeta</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_ExperimentMeta [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the ExperimentMeta entity.</p>

<p>An Experiment consists of (potentially) multiple strains,
environments,
and measurements on those strains and environments.</p>

<p>Example:</p>

<pre>    all_entities_ExperimentMeta -a </pre>

<p>would retrieve all entities of type ExperimentMeta and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ExperimentMeta entity has the following relationship links:</p>

<dl>
<dt><a name="ExperimentPublishedIn_Publication"
>ExperimentPublishedIn Publication</a></dt>

<dd>
<dt><a name="HasExperimentalUnit_ExperimentalUnit"
>HasExperimentalUnit ExperimentalUnit</a></dt>

<dd>
<dt><a name="PerformedBy_Person"
>PerformedBy Person</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_ExperimentMeta [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="title"
>title</a></dt>

<dd>
<p>Title of the experiment.</p>

<dt><a name="description"
>description</a></dt>

<dd>
<p>Description of the experiment including the experimental plan, general results, and conclusions, if possible.</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the experiment used by the data source.</p>

<dt><a name="startDate"
>startDate</a></dt>

<dd>
<p>The date this experiment was started.</p>

<dt><a name="comments"
>comments</a></dt>

<dd>
<p>Any data describing the experiment that is not covered by the description field.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Family</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Family [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Family entity.</p>

<p>The Kbase will support the maintenance of protein families (as sets of Features with associated translations).
We are initially only supporting the notion of a family as composed of a set of isofunctional homologs.
That is,
the families we initially support should be thought of as containing protein-encoding genes whose associated sequences all implement the same function (we do understand that the notion of &#34;function&#34; is somewhat ambiguous,
so let us sweep this under the rug by calling a functional role a &#34;primitive concept&#34;).
We currently support families in which the members are protein sequences as well.
Identical protein sequences as products of translating distinct genes may or may not have identical functions.
This may be justified,
since in a very,
very,
very few cases identical proteins do,
in fact,
have distinct functions.</p>

<p>Example:</p>

<pre>    all_entities_Family -a </pre>

<p>would retrieve all entities of type Family and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Family entity has the following relationship links:</p>

<dl>
<dt><a name="HasMember_Feature"
>HasMember Feature</a></dt>

<dd>
<dt><a name="HasProteinMember_ProteinSequence"
>HasProteinMember ProteinSequence</a></dt>

<dd>
<dt><a name="IsCoupledTo_Family"
>IsCoupledTo Family</a></dt>

<dd>
<dt><a name="IsCoupledWith_Family"
>IsCoupledWith Family</a></dt>

<dd>
<dt><a name="IsFamilyFor_Role"
>IsFamilyFor Role</a></dt>

<dd>
<dt><a name="IsRepresentedIn_Genome"
>IsRepresentedIn Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Family [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="type"
>type</a></dt>

<dd>
<p>type of protein family (e.g. FIGfam, equivalog)</p>

<dt><a name="release"
>release</a></dt>

<dd>
<p>release number / subtype of protein family</p>

<dt><a name="family_function"
>family_function</a></dt>

<dd>
<p>optional free-form description of the family. For function-based families, this would be the functional role for the family members.</p>

<dt><a name="alignment"
>alignment</a></dt>

<dd>
<p>FASTA-formatted alignment of the family&#39;s protein sequences</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Feature</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Feature [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Feature entity.</p>

<p>A feature (sometimes also called a gene) is a part of a genome that is of special interest.
Features may be spread across multiple DNA sequences (contigs) of a genome,
but never across more than one genome.
Each feature in the database has a unique ID that functions as its ID in this table.
Normally a Feature is just a single contigous region on a contig.
Features have types,
and an appropriate choice of available types allows the support of protein-encoding genes,
exons,
RNA genes,
binding sites,
pathogenicity islands,
or whatever.</p>

<p>Example:</p>

<pre>    all_entities_Feature -a </pre>

<p>would retrieve all entities of type Feature and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Feature entity has the following relationship links:</p>

<dl>
<dt><a name="Controls_CoregulatedSet"
>Controls CoregulatedSet</a></dt>

<dd>
<dt><a name="Encompasses_Feature"
>Encompasses Feature</a></dt>

<dd>
<dt><a name="FeatureMeasuredBy_Measurement"
>FeatureMeasuredBy Measurement</a></dt>

<dd>
<dt><a name="HasCoregulationWith_Feature"
>HasCoregulationWith Feature</a></dt>

<dd>
<dt><a name="HasFunctional_Role"
>HasFunctional Role</a></dt>

<dd>
<dt><a name="HasIndicatedSignalFrom_Experiment"
>HasIndicatedSignalFrom Experiment</a></dt>

<dd>
<dt><a name="HasLevelsFrom_ProbeSet"
>HasLevelsFrom ProbeSet</a></dt>

<dd>
<dt><a name="ImplementsReaction_ReactionInstance"
>ImplementsReaction ReactionInstance</a></dt>

<dd>
<dt><a name="IsAnnotatedBy_Annotation"
>IsAnnotatedBy Annotation</a></dt>

<dd>
<dt><a name="IsContainedIn_SSCell"
>IsContainedIn SSCell</a></dt>

<dd>
<dt><a name="IsCoregulatedWith_Feature"
>IsCoregulatedWith Feature</a></dt>

<dd>
<dt><a name="IsEncompassedIn_Feature"
>IsEncompassedIn Feature</a></dt>

<dd>
<dt><a name="IsExemplarOf_Role"
>IsExemplarOf Role</a></dt>

<dd>
<dt><a name="IsFormedInto_AtomicRegulon"
>IsFormedInto AtomicRegulon</a></dt>

<dd>
<dt><a name="IsInPair_Pairing"
>IsInPair Pairing</a></dt>

<dd>
<dt><a name="IsLocatedIn_Contig"
>IsLocatedIn Contig</a></dt>

<dd>
<dt><a name="IsMemberOf_Family"
>IsMemberOf Family</a></dt>

<dd>
<dt><a name="IsOwnedBy_Genome"
>IsOwnedBy Genome</a></dt>

<dd>
<dt><a name="IsRegulatedIn_CoregulatedSet"
>IsRegulatedIn CoregulatedSet</a></dt>

<dd>
<dt><a name="KnockedOutIn_Strain"
>KnockedOutIn Strain</a></dt>

<dd>
<dt><a name="Produces_ProteinSequence"
>Produces ProteinSequence</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Feature [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<p>Code indicating the type of this feature. Among the codes currently supported are &#34;peg&#34; for a protein encoding gene, &#34;bs&#34; for a binding site, &#34;opr&#34; for an operon, and so forth.</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>ID for this feature in its original source (core) database</p>

<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<p>Number of base pairs in this feature.</p>

<dt><a name="function"
>function</a></dt>

<dd>
<p>Functional assignment for this feature. This will often indicate the feature&#39;s functional role or roles, and may also have comments.</p>

<dt><a name="alias"
>alias</a></dt>

<dd>
<p>alternative identifier for the feature. These are highly unstructured, and frequently non-unique.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Genome</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Genome [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Genome entity.</p>

<p>The Kbase houses a large and growing set of genomes.
We often have multiple genomes that have identical DNA.
These usually have distinct gene calls and annotations,
but not always.
We consider the Kbase to be a framework for managing hundreds of thousands of genomes and offering the tools needed to support compartive analysis on large sets of genomes,
some of which are virtually identical.</p>

<p>Example:</p>

<pre>    all_entities_Genome -a </pre>

<p>would retrieve all entities of type Genome and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Genome entity has the following relationship links:</p>

<dl>
<dt><a name="GenomeParentOf_Strain"
>GenomeParentOf Strain</a></dt>

<dd>
<dt><a name="HadResultsProducedBy_ProbeSet"
>HadResultsProducedBy ProbeSet</a></dt>

<dd>
<dt><a name="HasRepresentativeOf_Family"
>HasRepresentativeOf Family</a></dt>

<dd>
<dt><a name="IsCollectedInto_OTU"
>IsCollectedInto OTU</a></dt>

<dd>
<dt><a name="IsComposedOf_Contig"
>IsComposedOf Contig</a></dt>

<dd>
<dt><a name="IsConfiguredBy_AtomicRegulon"
>IsConfiguredBy AtomicRegulon</a></dt>

<dd>
<dt><a name="IsInTaxa_TaxonomicGrouping"
>IsInTaxa TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsModeledBy_Model"
>IsModeledBy Model</a></dt>

<dd>
<dt><a name="IsOwnerOf_Feature"
>IsOwnerOf Feature</a></dt>

<dd>
<dt><a name="IsReferencedBy_ObservationalUnit"
>IsReferencedBy ObservationalUnit</a></dt>

<dd>
<dt><a name="Uses_SSRow"
>Uses SSRow</a></dt>

<dd>
<dt><a name="UsesCodons_CodonUsage"
>UsesCodons CodonUsage</a></dt>

<dd>
<dt><a name="WasSubmittedBy_Source"
>WasSubmittedBy Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Genome [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<p>Number of protein encoding genes for this genome.</p>

<dt><a name="rnas"
>rnas</a></dt>

<dd>
<p>Number of RNA features found for this organism.</p>

<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<p>Full genus/species/strain name of the genome sequence.</p>

<dt><a name="complete"
>complete</a></dt>

<dd>
<p>TRUE if the genome sequence is complete, else FALSE</p>

<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<p>TRUE if this is a prokaryotic genome sequence, else FALSE</p>

<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<p>Number of base pairs in the genome sequence.</p>

<dt><a name="contigs"
>contigs</a></dt>

<dd>
<p>Number of contigs for this genome sequence.</p>

<dt><a name="domain"
>domain</a></dt>

<dd>
<p>Domain for this organism (Archaea, Bacteria, Eukaryota, Virus, Plasmid, or Environmental Sample).</p>

<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<p>Genetic code number used for protein translation on most of this genome sequence&#39;s contigs.</p>

<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<p>Percent GC content present in the genome sequence&#39;s DNA.</p>

<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<p>zero or more strings describing phenotypic information about this genome sequence</p>

<dt><a name="md5"
>md5</a></dt>

<dd>
<p>MD5 identifier describing the genome&#39;s DNA sequence</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>identifier assigned to this genome by the original source</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Locality</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Locality [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Locality entity.</p>

<p>A locality is a geographic location.</p>

<p>Example:</p>

<pre>    all_entities_Locality -a </pre>

<p>would retrieve all entities of type Locality and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Locality entity has the following relationship links:</p>

<dl>
<dt><a name="HasUnits_ObservationalUnit"
>HasUnits ObservationalUnit</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Locality [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<p>Name or description of the location used as a collection site.</p>

<dt><a name="city"
>city</a></dt>

<dd>
<p>City of the collecting site.</p>

<dt><a name="state"
>state</a></dt>

<dd>
<p>State or province of the collecting site.</p>

<dt><a name="country"
>country</a></dt>

<dd>
<p>Country of the collecting site.</p>

<dt><a name="origcty"
>origcty</a></dt>

<dd>
<p>3-letter ISO 3166-1 extended country code for the country of origin.</p>

<dt><a name="elevation"
>elevation</a></dt>

<dd>
<p>Elevation of the collecting site, expressed in meters above sea level. Negative values are allowed.</p>

<dt><a name="latitude"
>latitude</a></dt>

<dd>
<p>Latitude of the collecting site, recorded as a decimal number. North latitudes are positive values and south latitudes are negative numbers.</p>

<dt><a name="longitude"
>longitude</a></dt>

<dd>
<p>Longitude of the collecting site, recorded as a decimal number. West longitudes are positive values and east longitudes are negative numbers.</p>

<dt><a name="lo_accession"
>lo_accession</a></dt>

<dd>
<p>gazeteer ontology term ID</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_LocalizedCompound</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_LocalizedCompound [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the LocalizedCompound entity.</p>

<p>This entity represents a compound occurring in a specific location.
A reaction always involves localized compounds.
If a reaction occurs entirely in a single location,
it will frequently only be represented by the cytoplasmic versions of the compounds; however,
a transport always uses specifically located compounds.</p>

<p>Example:</p>

<pre>    all_entities_LocalizedCompound -a </pre>

<p>would retrieve all entities of type LocalizedCompound and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The LocalizedCompound entity has the following relationship links:</p>

<dl>
<dt><a name="HasUsage_CompoundInstance"
>HasUsage CompoundInstance</a></dt>

<dd>
<dt><a name="IsInvolvedIn_Reaction"
>IsInvolvedIn Reaction</a></dt>

<dd>
<dt><a name="IsParticipationOf_Compound"
>IsParticipationOf Compound</a></dt>

<dd>
<dt><a name="ParticipatesAt_Location"
>ParticipatesAt Location</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_LocalizedCompound [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Location</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Location [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Location entity.</p>

<p>A location is a region of the cell where reaction compounds originate from or are transported to (e.g.
cell wall,
extracellular,
cytoplasm).</p>

<p>Example:</p>

<pre>    all_entities_Location -a </pre>

<p>would retrieve all entities of type Location and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Location entity has the following relationship links:</p>

<dl>
<dt><a name="IsInstantiatedBy_LocationInstance"
>IsInstantiatedBy LocationInstance</a></dt>

<dd>
<dt><a name="IsParticipatingAt_LocalizedCompound"
>IsParticipatingAt LocalizedCompound</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Location [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<p>date and time of the last modification to the compartment&#39;s definition</p>

<dt><a name="name"
>name</a></dt>

<dd>
<p>common name for the location</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>ID from the source of this location</p>

<dt><a name="abbr"
>abbr</a></dt>

<dd>
<p>an abbreviation (usually a single letter) for the location.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_LocationInstance</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_LocationInstance [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the LocationInstance entity.</p>

<p>The Location Instance represents a region of a cell (e.g.
cell wall,
cytoplasm) as it appears in a specific model.</p>

<p>Example:</p>

<pre>    all_entities_LocationInstance -a </pre>

<p>would retrieve all entities of type LocationInstance and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The LocationInstance entity has the following relationship links:</p>

<dl>
<dt><a name="IsDivisionOf_Model"
>IsDivisionOf Model</a></dt>

<dd>
<dt><a name="IsInstanceOf_Location"
>IsInstanceOf Location</a></dt>

<dd>
<dt><a name="IsRealLocationOf_CompoundInstance"
>IsRealLocationOf CompoundInstance</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_LocationInstance [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="index"
>index</a></dt>

<dd>
<p>number used to distinguish between different instances of the same type of location in a single model. Within a model, any two instances of the same location must have difference compartment index values.</p>

<dt><a name="label"
>label</a></dt>

<dd>
<p>description used to differentiate between instances of the same location in a single model</p>

<dt><a name="pH"
>pH</a></dt>

<dd>
<p>pH of the cell region, which is used to determine compound charge and pH gradient across cell membranes</p>

<dt><a name="potential"
>potential</a></dt>

<dd>
<p>electrochemical potential of the cell region, which is used to determine the electrochemical gradient across cell membranes</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Measurement</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Measurement [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Measurement entity.</p>

<p>A Measurement is a value generated by performing a protocol to evaluate a value on an ExperimentalUnit - e.g.
a strain in an environment.</p>

<p>Example:</p>

<pre>    all_entities_Measurement -a </pre>

<p>would retrieve all entities of type Measurement and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Measurement entity has the following relationship links:</p>

<dl>
<dt><a name="IsDefinedBy_MeasurementDescription"
>IsDefinedBy MeasurementDescription</a></dt>

<dd>
<dt><a name="IsMeasureOf_ExperimentalUnit"
>IsMeasureOf ExperimentalUnit</a></dt>

<dd>
<dt><a name="MeasuresCompound_Compound"
>MeasuresCompound Compound</a></dt>

<dd>
<dt><a name="MeasuresFeature_Feature"
>MeasuresFeature Feature</a></dt>

<dd>
<dt><a name="WasMeasuredBy_Protocol"
>WasMeasuredBy Protocol</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Measurement [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the measurement used by the data source.</p>

<dt><a name="value"
>value</a></dt>

<dd>
<p>The value of the measurement.</p>

<dt><a name="mean"
>mean</a></dt>

<dd>
<p>The mean of multiple replicates if they are included in the measurement.</p>

<dt><a name="median"
>median</a></dt>

<dd>
<p>The median of multiple replicates if they are included in the measurement.</p>

<dt><a name="stddev"
>stddev</a></dt>

<dd>
<p>The standard deviation of multiple replicates if they are included in the measurement.</p>

<dt><a name="N"
>N</a></dt>

<dd>
<p>The number of replicates if they are included in the measurement.</p>

<dt><a name="p_value"
>p_value</a></dt>

<dd>
<p>The p-value of multiple replicates if they are included in the measurement. The exact meaning of the p-value is specified in the MeasurementDescription object for this measurement.</p>

<dt><a name="Z_score"
>Z_score</a></dt>

<dd>
<p>The Z-score of multiple replicates if they are included in the measurement. The exact meaning of the p-value is specified in the MeasurementDescription object for this measurement.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_MeasurementDescription</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_MeasurementDescription [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the MeasurementDescription entity.</p>

<p>A MeasurementDescription provides information about a measurement value.</p>

<p>Example:</p>

<pre>    all_entities_MeasurementDescription -a </pre>

<p>would retrieve all entities of type MeasurementDescription and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The MeasurementDescription entity has the following relationship links:</p>

<dl>
<dt><a name="DescribesMeasurement_Measurement"
>DescribesMeasurement Measurement</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_MeasurementDescription [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<p>The name of the measurement.</p>

<dt><a name="description"
>description</a></dt>

<dd>
<p>The description of the measurement, how it is measured, and what the measurement statistics mean.</p>

<dt><a name="unitOfMeasure"
>unitOfMeasure</a></dt>

<dd>
<p>The units of the measurement.</p>

<dt><a name="category"
>category</a></dt>

<dd>
<p>The category the measurement fits into, for example phenotype, experimental input, environment.</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the measurement description used by the data source.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Media</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Media [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Media entity.</p>

<p>A media describes the chemical content of the solution in which cells are grown in an experiment or for the purposes of a model.
The key is the common media name.
The nature of the media is described by its relationship to its constituent compounds.</p>

<p>Example:</p>

<pre>    all_entities_Media -a </pre>

<p>would retrieve all entities of type Media and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Media entity has the following relationship links:</p>

<dl>
<dt><a name="HasPresenceOf_Compound"
>HasPresenceOf Compound</a></dt>

<dd>
<dt><a name="IsUtilizedIn_Experiment"
>IsUtilizedIn Experiment</a></dt>

<dd>
<dt><a name="UsedIn_Environment"
>UsedIn Environment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Media [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<p>date and time of the last modification to the media&#39;s definition</p>

<dt><a name="name"
>name</a></dt>

<dd>
<p>descriptive name of the media</p>

<dt><a name="is_minimal"
>is_minimal</a></dt>

<dd>
<p>TRUE if this is a minimal media, else FALSE</p>

<dt><a name="description"
>description</a></dt>

<dd>
<p>description of the media condition</p>

<dt><a name="solid"
>solid</a></dt>

<dd>
<p>Whether the media is solid (True) or liquid (False).</p>

<dt><a name="is_defined"
>is_defined</a></dt>

<dd>
<p>TRUE if this media condition is defined (all components explicitly known)</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the media used by the data source.</p>

<dt><a name="type"
>type</a></dt>

<dd>
<p>The general category of the media.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Model</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Model [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Model entity.</p>

<p>A model specifies a relationship between sets of features and reactions in a cell.
It is used to simulate cell growth and gene knockouts to validate annotations.</p>

<p>Example:</p>

<pre>    all_entities_Model -a </pre>

<p>would retrieve all entities of type Model and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Model entity has the following relationship links:</p>

<dl>
<dt><a name="HasRequirementOf_ReactionInstance"
>HasRequirementOf ReactionInstance</a></dt>

<dd>
<dt><a name="IsDividedInto_LocationInstance"
>IsDividedInto LocationInstance</a></dt>

<dd>
<dt><a name="Manages_Biomass"
>Manages Biomass</a></dt>

<dd>
<dt><a name="Models_Genome"
>Models Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Model [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<p>date and time of the last change to the model data</p>

<dt><a name="name"
>name</a></dt>

<dd>
<p>descriptive name of the model</p>

<dt><a name="version"
>version</a></dt>

<dd>
<p>revision number of the model</p>

<dt><a name="type"
>type</a></dt>

<dd>
<p>string indicating where the model came from (e.g. single genome, multiple genome, or community model)</p>

<dt><a name="status"
>status</a></dt>

<dd>
<p>indicator of whether the model is stable, under construction, or under reconstruction</p>

<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<p>number of reactions in the model</p>

<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<p>number of compounds in the model</p>

<dt><a name="annotation_count"
>annotation_count</a></dt>

<dd>
<p>number of features associated with one or more reactions in the model</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_ObservationalUnit</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_ObservationalUnit [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the ObservationalUnit entity.</p>

<p>An ObservationalUnit is an individual plant that 1) is part of an experiment or study,
2) has measured traits,
and 3) is assayed for the purpose of determining alleles.</p>

<p>Example:</p>

<pre>    all_entities_ObservationalUnit -a </pre>

<p>would retrieve all entities of type ObservationalUnit and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ObservationalUnit entity has the following relationship links:</p>

<dl>
<dt><a name="DefinedBy_TaxonomicGrouping"
>DefinedBy TaxonomicGrouping</a></dt>

<dd>
<dt><a name="HasTrait_Trait"
>HasTrait Trait</a></dt>

<dd>
<dt><a name="IsLocated_Locality"
>IsLocated Locality</a></dt>

<dd>
<dt><a name="IsPartOf_StudyExperiment"
>IsPartOf StudyExperiment</a></dt>

<dd>
<dt><a name="IsVariedIn_Contig"
>IsVariedIn Contig</a></dt>

<dd>
<dt><a name="UsesReference_Genome"
>UsesReference Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_ObservationalUnit [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<p>Name/ID by which the observational unit may be known by the originator and is used in queries.</p>

<dt><a name="source_name2"
>source_name2</a></dt>

<dd>
<p>Secondary name/ID by which the observational unit may be known and is queried.</p>

<dt><a name="plant_id"
>plant_id</a></dt>

<dd>
<p>ID of the plant that was tested to produce this observational unit. Observational units with the same plant ID are different assays of a single physical organism.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_OTU</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_OTU [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the OTU entity.</p>

<p>An OTU (Organism Taxonomic Unit) is a named group of related genomes.</p>

<p>Example:</p>

<pre>    all_entities_OTU -a </pre>

<p>would retrieve all entities of type OTU and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The OTU entity has the following relationship links:</p>

<dl>
<dt><a name="IsCollectionOf_Genome"
>IsCollectionOf Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_OTU [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Pairing</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Pairing [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Pairing entity.</p>

<p>A pairing indicates that two features are found close together in a genome.
Not all possible pairings are stored in the database; only those that are considered for some reason to be significant for annotation purposes.The key of the pairing is the concatenation of the feature IDs in alphabetical order with an intervening colon.</p>

<p>Example:</p>

<pre>    all_entities_Pairing -a </pre>

<p>would retrieve all entities of type Pairing and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Pairing entity has the following relationship links:</p>

<dl>
<dt><a name="Determines_PairSet"
>Determines PairSet</a></dt>

<dd>
<dt><a name="IsPairOf_Feature"
>IsPairOf Feature</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Pairing [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_PairSet</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_PairSet [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the PairSet entity.</p>

<p>A PairSet is a precompute set of pairs or genes.
Each pair occurs close to one another of the chromosome.
We believe that all of the first members of the pairs correspond to one another (are quite similar),
as do all of the second members of the pairs.
These pairs (from prokaryotic genomes) offer one of the most powerful clues relating to uncharacterized genes/peroteins.</p>

<p>Example:</p>

<pre>    all_entities_PairSet -a </pre>

<p>would retrieve all entities of type PairSet and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The PairSet entity has the following relationship links:</p>

<dl>
<dt><a name="IsDeterminedBy_Pairing"
>IsDeterminedBy Pairing</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_PairSet [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="score"
>score</a></dt>

<dd>
<p>Score for this evidence set. The score indicates the number of significantly different genomes represented by the pairings.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Parameter</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Parameter [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Parameter entity.</p>

<p>A parameter is the name of some quantity that has a value.</p>

<p>Example:</p>

<pre>    all_entities_Parameter -a </pre>

<p>would retrieve all entities of type Parameter and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Parameter entity has the following relationship links:</p>

<dl>
<dt><a name="OfEnvironment_Environment"
>OfEnvironment Environment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Parameter [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Person</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Person [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Person entity.</p>

<p>A person represents a human affiliated in some way with Kbase.</p>

<p>Example:</p>

<pre>    all_entities_Person -a </pre>

<p>would retrieve all entities of type Person and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Person entity has the following relationship links:</p>

<dl>
<dt><a name="PerformedExperiment_ExperimentMeta"
>PerformedExperiment ExperimentMeta</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Person [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="firstName"
>firstName</a></dt>

<dd>
<p>The given name of the person.</p>

<dt><a name="lastName"
>lastName</a></dt>

<dd>
<p>The surname of the person.</p>

<dt><a name="contactEmail"
>contactEmail</a></dt>

<dd>
<p>Email address of the person.</p>

<dt><a name="institution"
>institution</a></dt>

<dd>
<p>The institution where the person works.</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the person used by the data source.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_ProbeSet</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_ProbeSet [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the ProbeSet entity.</p>

<p>A probe set is a device containing multiple probe sequences for use in gene expression experiments.</p>

<p>Example:</p>

<pre>    all_entities_ProbeSet -a </pre>

<p>would retrieve all entities of type ProbeSet and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ProbeSet entity has the following relationship links:</p>

<dl>
<dt><a name="GeneratedLevelsFor_AtomicRegulon"
>GeneratedLevelsFor AtomicRegulon</a></dt>

<dd>
<dt><a name="HasResultsIn_Experiment"
>HasResultsIn Experiment</a></dt>

<dd>
<dt><a name="IndicatedLevelsFor_Feature"
>IndicatedLevelsFor Feature</a></dt>

<dd>
<dt><a name="ProducedResultsFor_Genome"
>ProducedResultsFor Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_ProbeSet [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_ProteinSequence</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_ProteinSequence [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the ProteinSequence entity.</p>

<p>We use the concept of ProteinSequence as an amino acid string with an associated MD5 value.
It is easy to access the set of Features that relate to a ProteinSequence.
While function is still associated with Features (and may be for some time),
publications are associated with ProteinSequences (and the inferred impact on Features is through the relationship connecting ProteinSequences to Features).</p>

<p>Example:</p>

<pre>    all_entities_ProteinSequence -a </pre>

<p>would retrieve all entities of type ProteinSequence and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ProteinSequence entity has the following relationship links:</p>

<dl>
<dt><a name="HasAssertedFunctionFrom_Source"
>HasAssertedFunctionFrom Source</a></dt>

<dd>
<dt><a name="IsATopicOf_Publication"
>IsATopicOf Publication</a></dt>

<dd>
<dt><a name="IsAlignedProteinComponentOf_AlignmentRow"
>IsAlignedProteinComponentOf AlignmentRow</a></dt>

<dd>
<dt><a name="IsProteinFor_Feature"
>IsProteinFor Feature</a></dt>

<dd>
<dt><a name="IsProteinMemberOf_Family"
>IsProteinMemberOf Family</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_ProteinSequence [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="sequence"
>sequence</a></dt>

<dd>
<p>The sequence contains the letters corresponding to the protein&#39;s amino acids.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Protocol</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Protocol [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Protocol entity.</p>

<p>A Protocol is a step by step set of instructions for performing a part of an experiment.</p>

<p>Example:</p>

<pre>    all_entities_Protocol -a </pre>

<p>would retrieve all entities of type Protocol and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Protocol entity has the following relationship links:</p>

<dl>
<dt><a name="IsMeasurementMethodOf_Measurement"
>IsMeasurementMethodOf Measurement</a></dt>

<dd>
<dt><a name="ProtocolPublishedIn_Publication"
>ProtocolPublishedIn Publication</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Protocol [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<p>The name of the protocol.</p>

<dt><a name="description"
>description</a></dt>

<dd>
<p>The step by step instructions for performing the experiment, including measurement details, materials, and equipment. A researcher should be able to reproduce the experimental results with this information.</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the protocol used by the data source.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Publication</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Publication [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Publication entity.</p>

<p>Experimenters attach publications to experiments and protocols.
Annotators attach publications to ProteinSequences.
The attached publications give an ID (usually a DOI or Pubmed ID),
a URL to the paper (when we have it),
and a title (when we have it).
Pubmed IDs are given unmodified.
DOI IDs are prefixed with [b]doi:[/b],
e.g.
[i]doi:1002385[/i].</p>

<p>Example:</p>

<pre>    all_entities_Publication -a </pre>

<p>would retrieve all entities of type Publication and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Publication entity has the following relationship links:</p>

<dl>
<dt><a name="Concerns_ProteinSequence"
>Concerns ProteinSequence</a></dt>

<dd>
<dt><a name="PublishedExperiment_ExperimentMeta"
>PublishedExperiment ExperimentMeta</a></dt>

<dd>
<dt><a name="PublishedProtocol_Protocol"
>PublishedProtocol Protocol</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Publication [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="title"
>title</a></dt>

<dd>
<p>title of the article, or (unknown) if the title is not known</p>

<dt><a name="link"
>link</a></dt>

<dd>
<p>URL of the article, DOI preferred</p>

<dt><a name="pubdate"
>pubdate</a></dt>

<dd>
<p>publication date of the article</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Reaction</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Reaction [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Reaction entity.</p>

<p>A reaction is a chemical process that converts one set of compounds (substrate) to another set (products).</p>

<p>Example:</p>

<pre>    all_entities_Reaction -a </pre>

<p>would retrieve all entities of type Reaction and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Reaction entity has the following relationship links:</p>

<dl>
<dt><a name="Involves_LocalizedCompound"
>Involves LocalizedCompound</a></dt>

<dd>
<dt><a name="IsDisplayedOn_Diagram"
>IsDisplayedOn Diagram</a></dt>

<dd>
<dt><a name="IsExecutedAs_ReactionInstance"
>IsExecutedAs ReactionInstance</a></dt>

<dd>
<dt><a name="IsStepOf_Complex"
>IsStepOf Complex</a></dt>

<dd>
<dt><a name="ParticipatesIn_Scenario"
>ParticipatesIn Scenario</a></dt>

<dd>
<dt><a name="UsesAliasForReaction_Source"
>UsesAliasForReaction Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Reaction [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<p>date and time of the last modification to this reaction&#39;s definition</p>

<dt><a name="name"
>name</a></dt>

<dd>
<p>descriptive name of this reaction</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>ID of this reaction in the resource from which it was added</p>

<dt><a name="abbr"
>abbr</a></dt>

<dd>
<p>abbreviated name of this reaction</p>

<dt><a name="direction"
>direction</a></dt>

<dd>
<p>direction of this reaction (&#62; for forward-only, &#60; for backward-only, = for bidirectional)</p>

<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<p>Gibbs free-energy change for the reaction calculated using the group contribution method (units are kcal/mol)</p>

<dt><a name="deltaG_error"
>deltaG_error</a></dt>

<dd>
<p>uncertainty in the [b]deltaG[/b] value (units are kcal/mol)</p>

<dt><a name="thermodynamic_reversibility"
>thermodynamic_reversibility</a></dt>

<dd>
<p>computed reversibility of this reaction in a pH-neutral environment</p>

<dt><a name="default_protons"
>default_protons</a></dt>

<dd>
<p>number of protons absorbed by this reaction in a pH-neutral environment</p>

<dt><a name="status"
>status</a></dt>

<dd>
<p>string indicating additional information about this reaction, generally indicating whether the reaction is balanced and/or lumped</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_ReactionInstance</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_ReactionInstance [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the ReactionInstance entity.</p>

<p>A reaction instance describes the specific implementation of a reaction in a model.</p>

<p>Example:</p>

<pre>    all_entities_ReactionInstance -a </pre>

<p>would retrieve all entities of type ReactionInstance and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ReactionInstance entity has the following relationship links:</p>

<dl>
<dt><a name="ImplementedBasedOn_Feature"
>ImplementedBasedOn Feature</a></dt>

<dd>
<dt><a name="IsARequirementOf_Model"
>IsARequirementOf Model</a></dt>

<dd>
<dt><a name="IsExecutionOf_Reaction"
>IsExecutionOf Reaction</a></dt>

<dd>
<dt><a name="Targets_CompoundInstance"
>Targets CompoundInstance</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_ReactionInstance [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="direction"
>direction</a></dt>

<dd>
<p>reaction directionality (&#62; for forward, &#60; for backward, = for bidirectional) with respect to this model</p>

<dt><a name="protons"
>protons</a></dt>

<dd>
<p>number of protons produced by this reaction when proceeding in the forward direction. If this is a transport reaction, these protons end up in the reaction instance&#39;s main location. If the number is negative, then the protons are consumed by the reaction rather than being produced.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Role</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Role [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Role entity.</p>

<p>A role describes a biological function that may be fulfilled by a feature.
One of the main goals of the database is to assign features to roles.
Most roles are effected by the construction of proteins.
Some,
however,
deal with functional regulation and message transmission.</p>

<p>Example:</p>

<pre>    all_entities_Role -a </pre>

<p>would retrieve all entities of type Role and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Role entity has the following relationship links:</p>

<dl>
<dt><a name="DeterminesFunctionOf_Family"
>DeterminesFunctionOf Family</a></dt>

<dd>
<dt><a name="HasAsExemplar_Feature"
>HasAsExemplar Feature</a></dt>

<dd>
<dt><a name="IsConsistentTo_EcNumber"
>IsConsistentTo EcNumber</a></dt>

<dd>
<dt><a name="IsFunctionalIn_Feature"
>IsFunctionalIn Feature</a></dt>

<dd>
<dt><a name="IsIncludedIn_Subsystem"
>IsIncludedIn Subsystem</a></dt>

<dd>
<dt><a name="IsRoleOf_SSCell"
>IsRoleOf SSCell</a></dt>

<dd>
<dt><a name="Triggers_Complex"
>Triggers Complex</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Role [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="hypothetical"
>hypothetical</a></dt>

<dd>
<p>TRUE if a role is hypothetical, else FALSE</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Scenario</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Scenario [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Scenario entity.</p>

<p>A scenario is a partial instance of a subsystem with a defined set of reactions.
Each scenario converts input compounds to output compounds using reactions.
The scenario may use all of the reactions controlled by a subsystem or only some,
and may also incorporate additional reactions.
Because scenario names are not unique,
the actual scenario ID is a number.</p>

<p>Example:</p>

<pre>    all_entities_Scenario -a </pre>

<p>would retrieve all entities of type Scenario and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Scenario entity has the following relationship links:</p>

<dl>
<dt><a name="HasAsTerminus_Compound"
>HasAsTerminus Compound</a></dt>

<dd>
<dt><a name="HasParticipant_Reaction"
>HasParticipant Reaction</a></dt>

<dd>
<dt><a name="Overlaps_Diagram"
>Overlaps Diagram</a></dt>

<dd>
<dt><a name="Validates_Subsystem"
>Validates Subsystem</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Scenario [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="common_name"
>common_name</a></dt>

<dd>
<p>Common name of the scenario. The name, rather than the ID number, is usually displayed everywhere.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Source</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Source [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Source entity.</p>

<p>A source is a user or organization that is permitted to assign its own identifiers or to submit bioinformatic objects to the database.</p>

<p>Example:</p>

<pre>    all_entities_Source -a </pre>

<p>would retrieve all entities of type Source and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Source entity has the following relationship links:</p>

<dl>
<dt><a name="Aligned_Alignment"
>Aligned Alignment</a></dt>

<dd>
<dt><a name="AssertsFunctionFor_ProteinSequence"
>AssertsFunctionFor ProteinSequence</a></dt>

<dd>
<dt><a name="Formulated_CoregulatedSet"
>Formulated CoregulatedSet</a></dt>

<dd>
<dt><a name="HasCompoundAliasFrom_Compound"
>HasCompoundAliasFrom Compound</a></dt>

<dd>
<dt><a name="HasReactionAliasFrom_Reaction"
>HasReactionAliasFrom Reaction</a></dt>

<dd>
<dt><a name="Provided_Subsystem"
>Provided Subsystem</a></dt>

<dd>
<dt><a name="Submitted_Genome"
>Submitted Genome</a></dt>

<dd>
<dt><a name="Treed_Tree"
>Treed Tree</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Source [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_SSCell</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_SSCell [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the SSCell entity.</p>

<p>An SSCell (SpreadSheet Cell) represents a role as it occurs in a subsystem spreadsheet row.
The key is a colon-delimited triple containing an MD5 hash of the subsystem ID followed by a genome ID (with optional region string) and a role abbreviation.</p>

<p>Example:</p>

<pre>    all_entities_SSCell -a </pre>

<p>would retrieve all entities of type SSCell and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SSCell entity has the following relationship links:</p>

<dl>
<dt><a name="Contains_Feature"
>Contains Feature</a></dt>

<dd>
<dt><a name="HasRole_Role"
>HasRole Role</a></dt>

<dd>
<dt><a name="IsRoleFor_SSRow"
>IsRoleFor SSRow</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_SSCell [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_SSRow</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_SSRow [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the SSRow entity.</p>

<p>An SSRow (that is,
a row in a subsystem spreadsheet) represents a collection of functional roles present in the Features of a single Genome.
The roles are part of a designated subsystem,
and the features associated with each role are included in the row.
That is,
a row amounts to an instance of a subsystem as it exists in a specific,
designated genome.</p>

<p>Example:</p>

<pre>    all_entities_SSRow -a </pre>

<p>would retrieve all entities of type SSRow and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SSRow entity has the following relationship links:</p>

<dl>
<dt><a name="Implements_Variant"
>Implements Variant</a></dt>

<dd>
<dt><a name="IsRowOf_SSCell"
>IsRowOf SSCell</a></dt>

<dd>
<dt><a name="IsUsedBy_Genome"
>IsUsedBy Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_SSRow [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="curated"
>curated</a></dt>

<dd>
<p>This flag is TRUE if the assignment of the molecular machine has been curated, and FALSE if it was made by an automated program.</p>

<dt><a name="region"
>region</a></dt>

<dd>
<p>Region in the genome for which the row is relevant. Normally, this is an empty string, indicating that the machine covers the whole genome. If a subsystem has multiple rows for a genome, this contains a location string describing the region occupied by this particular row.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Strain</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Strain [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Strain entity.</p>

<p>This entity represents an organism derived from a genome or another organism with one or more modifications to the organism&#39;s genome.</p>

<p>Example:</p>

<pre>    all_entities_Strain -a </pre>

<p>would retrieve all entities of type Strain and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Strain entity has the following relationship links:</p>

<dl>
<dt><a name="DerivedFromGenome_Genome"
>DerivedFromGenome Genome</a></dt>

<dd>
<dt><a name="DerivedFromStrain_Strain"
>DerivedFromStrain Strain</a></dt>

<dd>
<dt><a name="EvaluatedIn_ExperimentalUnit"
>EvaluatedIn ExperimentalUnit</a></dt>

<dd>
<dt><a name="HasKnockoutIn_Feature"
>HasKnockoutIn Feature</a></dt>

<dd>
<dt><a name="StrainParentOf_Strain"
>StrainParentOf Strain</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Strain [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<p>The common or laboratory name of the strain, e.g. DH5a or JMP1004.</p>

<dt><a name="description"
>description</a></dt>

<dd>
<p>A description of the strain, e.g. knockout/modification methods, resulting phenotypes, etc.</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the strain used by the data source.</p>

<dt><a name="aggregateData"
>aggregateData</a></dt>

<dd>
<p>Denotes whether this entity represents a physical strain (False) or aggregate data calculated from one or more strains (True).</p>

<dt><a name="wildtype"
>wildtype</a></dt>

<dd>
<p>Denotes this strain is presumably identical to the parent genome.</p>

<dt><a name="referenceStrain"
>referenceStrain</a></dt>

<dd>
<p>Denotes whether this strain is a reference strain; e.g. it is identical to the genome it&#39;s related to (True) or not (False). In contrast to wildtype, a referenceStrain is abstract and does not physically exist and is used for data that refers to a genome but not a particular strain. There should only exist one reference strain per genome and all reference strains are wildtype.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_StudyExperiment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_StudyExperiment [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the StudyExperiment entity.</p>

<p>An Experiment is a collection of observational units with one originator that are part of a specific study.
An experiment may be conducted at more than one location and in more than one season or year.</p>

<p>Example:</p>

<pre>    all_entities_StudyExperiment -a </pre>

<p>would retrieve all entities of type StudyExperiment and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The StudyExperiment entity has the following relationship links:</p>

<dl>
<dt><a name="IncludesPart_ObservationalUnit"
>IncludesPart ObservationalUnit</a></dt>

<dd>
<dt><a name="IsAssayedBy_Assay"
>IsAssayedBy Assay</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_StudyExperiment [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<p>Name/ID by which the experiment is known at the source.</p>

<dt><a name="design"
>design</a></dt>

<dd>
<p>Design of the experiment including the numbers and types of observational units, traits, replicates, sampling plan, and analysis that are planned.</p>

<dt><a name="originator"
>originator</a></dt>

<dd>
<p>Name of the individual or program that are the originators of the experiment.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Subsystem</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Subsystem [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Subsystem entity.</p>

<p>A subsystem is a set of functional roles that have been annotated simultaneously (e.g.,
the roles present in a specific pathway),
with an associated subsystem spreadsheet which encodes the fids in each genome that implement the functional roles in the subsystem.</p>

<p>Example:</p>

<pre>    all_entities_Subsystem -a </pre>

<p>would retrieve all entities of type Subsystem and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Subsystem entity has the following relationship links:</p>

<dl>
<dt><a name="Describes_Variant"
>Describes Variant</a></dt>

<dd>
<dt><a name="Includes_Role"
>Includes Role</a></dt>

<dd>
<dt><a name="IsInClass_SubsystemClass"
>IsInClass SubsystemClass</a></dt>

<dd>
<dt><a name="IsRelevantTo_Diagram"
>IsRelevantTo Diagram</a></dt>

<dd>
<dt><a name="IsSubInstanceOf_Scenario"
>IsSubInstanceOf Scenario</a></dt>

<dd>
<dt><a name="WasProvidedBy_Source"
>WasProvidedBy Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Subsystem [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="version"
>version</a></dt>

<dd>
<p>version number for the subsystem. This value is incremented each time the subsystem is backed up.</p>

<dt><a name="curator"
>curator</a></dt>

<dd>
<p>name of the person currently in charge of the subsystem</p>

<dt><a name="notes"
>notes</a></dt>

<dd>
<p>descriptive notes about the subsystem</p>

<dt><a name="description"
>description</a></dt>

<dd>
<p>description of the subsystem&#39;s function in the cell</p>

<dt><a name="usable"
>usable</a></dt>

<dd>
<p>TRUE if this is a usable subsystem, else FALSE. An unusable subsystem is one that is experimental or is of such low quality that it can negatively affect analysis.</p>

<dt><a name="private"
>private</a></dt>

<dd>
<p>TRUE if this is a private subsystem, else FALSE. A private subsystem has valid data, but is not considered ready for general distribution.</p>

<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<p>TRUE if this is a clustering-based subsystem, else FALSE. A clustering-based subsystem is one in which there is functional-coupling evidence that genes belong together, but we do not yet know what they do.</p>

<dt><a name="experimental"
>experimental</a></dt>

<dd>
<p>TRUE if this is an experimental subsystem, else FALSE. An experimental subsystem is designed for investigation and is not yet ready to be used in comparative analysis and annotation.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_SubsystemClass</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_SubsystemClass [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the SubsystemClass entity.</p>

<p>Subsystem classes impose a hierarchical organization on the subsystems.</p>

<p>Example:</p>

<pre>    all_entities_SubsystemClass -a </pre>

<p>would retrieve all entities of type SubsystemClass and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SubsystemClass entity has the following relationship links:</p>

<dl>
<dt><a name="IsClassFor_Subsystem"
>IsClassFor Subsystem</a></dt>

<dd>
<dt><a name="IsSubclassOf_SubsystemClass"
>IsSubclassOf SubsystemClass</a></dt>

<dd>
<dt><a name="IsSuperclassOf_SubsystemClass"
>IsSuperclassOf SubsystemClass</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_SubsystemClass [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_TaxonomicGrouping</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_TaxonomicGrouping [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the TaxonomicGrouping entity.</p>

<p>We associate with most genomes a &#34;taxonomy&#34; based on the NCBI taxonomy.
This includes,
for each genome,
a list of ever larger taxonomic groups.
The groups are stored as instances of this entity,
and chained together by the IsGroupFor relationship.</p>

<p>Example:</p>

<pre>    all_entities_TaxonomicGrouping -a </pre>

<p>would retrieve all entities of type TaxonomicGrouping and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The TaxonomicGrouping entity has the following relationship links:</p>

<dl>
<dt><a name="IsGroupFor_TaxonomicGrouping"
>IsGroupFor TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsInGroup_TaxonomicGrouping"
>IsInGroup TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsRepresentedBy_ObservationalUnit"
>IsRepresentedBy ObservationalUnit</a></dt>

<dd>
<dt><a name="IsTaxonomyOf_Genome"
>IsTaxonomyOf Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_TaxonomicGrouping [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="domain"
>domain</a></dt>

<dd>
<p>TRUE if this is a domain grouping, else FALSE.</p>

<dt><a name="hidden"
>hidden</a></dt>

<dd>
<p>TRUE if this is a hidden grouping, else FALSE. Hidden groupings are not typically shown in a lineage list.</p>

<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<p>Primary scientific name for this grouping. This is the name used when displaying a taxonomy.</p>

<dt><a name="alias"
>alias</a></dt>

<dd>
<p>Alternate name for this grouping. A grouping may have many alternate names. The scientific name should also be in this list.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_TimeSeries</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_TimeSeries [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the TimeSeries entity.</p>

<p>A TimeSeries provides a means to specify a series of experimental data over time by ordering multiple ExperimentalUnits.</p>

<p>Example:</p>

<pre>    all_entities_TimeSeries -a </pre>

<p>would retrieve all entities of type TimeSeries and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The TimeSeries entity has the following relationship links:</p>

<dl>
<dt><a name="OrdersExperimentalUnit_ExperimentalUnit"
>OrdersExperimentalUnit ExperimentalUnit</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_TimeSeries [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the time series used by the data source.</p>

<dt><a name="name"
>name</a></dt>

<dd>
<p>The name of this time series, if any.</p>

<dt><a name="comments"
>comments</a></dt>

<dd>
<p>Any comments regarding this time series.</p>

<dt><a name="timeUnits"
>timeUnits</a></dt>

<dd>
<p>The units of time for this time series, e.g. &#39;seconds&#39;, &#39;hours&#39;, or more abstractly, &#39;number of times culture grown to saturation.&#39;</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Trait</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Trait [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Trait entity.</p>

<p>A Trait is a phenotypic quality that can be measured or observed for an observational unit.
Examples include height,
sugar content,
color,
or cold tolerance.</p>

<p>Example:</p>

<pre>    all_entities_Trait -a </pre>

<p>would retrieve all entities of type Trait and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Trait entity has the following relationship links:</p>

<dl>
<dt><a name="Impacts_Contig"
>Impacts Contig</a></dt>

<dd>
<dt><a name="Measures_ObservationalUnit"
>Measures ObservationalUnit</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Trait [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="trait_name"
>trait_name</a></dt>

<dd>
<p>Text name or description of the trait</p>

<dt><a name="unit_of_measure"
>unit_of_measure</a></dt>

<dd>
<p>The units of measure used when determining this trait. If multiple units of measure are applicable, each has its own row in the database.</p>

<dt><a name="TO_ID"
>TO_ID</a></dt>

<dd>
<p>Trait Ontology term ID (http://www.gramene.org/plant-ontology/)</p>

<dt><a name="protocol"
>protocol</a></dt>

<dd>
<p>A thorough description of how the trait was collected, and if a rating, the minimum and maximum values</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Tree</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Tree [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Tree entity.</p>

<p>A tree describes how the sequences in an alignment relate to each other.
Most trees are phylogenetic,
but some may be based on taxonomy or gene content.</p>

<p>Example:</p>

<pre>    all_entities_Tree -a </pre>

<p>would retrieve all entities of type Tree and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Tree entity has the following relationship links:</p>

<dl>
<dt><a name="HasNodeAttribute_TreeNodeAttribute"
>HasNodeAttribute TreeNodeAttribute</a></dt>

<dd>
<dt><a name="HasTreeAttribute_TreeAttribute"
>HasTreeAttribute TreeAttribute</a></dt>

<dd>
<dt><a name="IsBuiltFromAlignment_Alignment"
>IsBuiltFromAlignment Alignment</a></dt>

<dd>
<dt><a name="IsModificationOfTree_Tree"
>IsModificationOfTree Tree</a></dt>

<dd>
<dt><a name="IsModifiedToBuildTree_Tree"
>IsModifiedToBuildTree Tree</a></dt>

<dd>
<dt><a name="IsSupersededByTree_Tree"
>IsSupersededByTree Tree</a></dt>

<dd>
<dt><a name="IsTreeFrom_Source"
>IsTreeFrom Source</a></dt>

<dd>
<dt><a name="SupersedesTree_Tree"
>SupersedesTree Tree</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Tree [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="status"
>status</a></dt>

<dd>
<p>status of the tree, currently either [i]active[/i], [i]superseded[/i], or [i]bad[/i]</p>

<dt><a name="data_type"
>data_type</a></dt>

<dd>
<p>type of data the tree was built from, usually [i]sequence_alignment[/i]</p>

<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<p>date and time the tree was loaded</p>

<dt><a name="method"
>method</a></dt>

<dd>
<p>name of the primary software package or script used to construct the tree</p>

<dt><a name="parameters"
>parameters</a></dt>

<dd>
<p>non-default parameters used as input to the software package or script indicated in the method attribute</p>

<dt><a name="protocol"
>protocol</a></dt>

<dd>
<p>description of the steps taken to construct the tree, or a reference to an external pipeline</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>ID of this tree in the source database</p>

<dt><a name="newick"
>newick</a></dt>

<dd>
<p>NEWICK format string containing the structure of the tree</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_TreeAttribute</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_TreeAttribute [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the TreeAttribute entity.</p>

<p>This entity represents an attribute type that can be assigned to a tree.
The attribute values are stored in the relationships to the target.
The key is the attribute name.</p>

<p>Example:</p>

<pre>    all_entities_TreeAttribute -a </pre>

<p>would retrieve all entities of type TreeAttribute and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The TreeAttribute entity has the following relationship links:</p>

<dl>
<dt><a name="DescribesTree_Tree"
>DescribesTree Tree</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_TreeAttribute [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_TreeNodeAttribute</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_TreeNodeAttribute [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the TreeNodeAttribute entity.</p>

<p>This entity represents an attribute type that can be assigned to a node.
The attribute values are stored in the relationships to the target.
The key is the attribute name.</p>

<p>Example:</p>

<pre>    all_entities_TreeNodeAttribute -a </pre>

<p>would retrieve all entities of type TreeNodeAttribute and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The TreeNodeAttribute entity has the following relationship links:</p>

<dl>
<dt><a name="DescribesTreeNode_Tree"
>DescribesTreeNode Tree</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_TreeNodeAttribute [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>all_entities_Variant</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>all_entities_Variant [-a] [--fields fieldlist] &#62; entity-data</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Return all instances of the Variant entity.</p>

<p>Each subsystem may include the designation of distinct variants.
Thus,
there may be three closely-related,
but distinguishable forms of histidine degradation.
Each form would be called a &#34;variant&#34;,
with an associated code,
and all genomes implementing a specific variant can easily be accessed.</p>

<p>Example:</p>

<pre>    all_entities_Variant -a </pre>

<p>would retrieve all entities of type Variant and include all fields in the entities in the output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Variant entity has the following relationship links:</p>

<dl>
<dt><a name="IsDescribedBy_Subsystem"
>IsDescribedBy Subsystem</a></dt>

<dd>
<dt><a name="IsImplementedBy_SSRow"
>IsImplementedBy SSRow</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: all_entities_Variant [arguments] &#62; entity.data</p>

<pre>    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="role_rule"
>role_rule</a></dt>

<dd>
<p>a space-delimited list of role IDs, in alphabetical order, that represents a possible list of non-auxiliary roles applicable to this variant. The roles are identified by their abbreviations. A variant may have multiple role rules.</p>

<dt><a name="code"
>code</a></dt>

<dd>
<p>the variant code all by itself</p>

<dt><a name="type"
>type</a></dt>

<dd>
<p>variant type indicating the quality of the subsystem support. A type of &#34;vacant&#34; means that the subsystem does not appear to be implemented by the variant. A type of &#34;incomplete&#34; means that the subsystem appears to be missing many reactions. In all other cases, the type is &#34;normal&#34;.</p>

<dt><a name="comment"
>comment</a></dt>

<dd>
<p>commentary text about the variant</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Alignment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Alignment [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>An alignment arranges a group of sequences so that they match.
Each alignment is associated with a phylogenetic tree that describes how the sequences developed and their evolutionary distance.</p>

<p>Example:</p>

<pre>    get_entity_Alignment -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Alignment entity has the following relationship links:</p>

<dl>
<dt><a name="HasAlignmentAttribute_AlignmentAttribute"
>HasAlignmentAttribute AlignmentAttribute</a></dt>

<dd>
<dt><a name="IncludesAlignmentRow_AlignmentRow"
>IncludesAlignmentRow AlignmentRow</a></dt>

<dd>
<dt><a name="IsModificationOfAlignment_Alignment"
>IsModificationOfAlignment Alignment</a></dt>

<dd>
<dt><a name="IsModifiedToBuildAlignment_Alignment"
>IsModifiedToBuildAlignment Alignment</a></dt>

<dd>
<dt><a name="IsSupersededByAlignment_Alignment"
>IsSupersededByAlignment Alignment</a></dt>

<dd>
<dt><a name="IsUsedToBuildTree_Tree"
>IsUsedToBuildTree Tree</a></dt>

<dd>
<dt><a name="SupersedesAlignment_Alignment"
>SupersedesAlignment Alignment</a></dt>

<dd>
<dt><a name="WasAlignedBy_Source"
>WasAlignedBy Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Alignment [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<p>number of rows in the alignment</p>

<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<p>number of columns in the alignment</p>

<dt><a name="status"
>status</a></dt>

<dd>
<p>status of the alignment, currently either [i]active[/i], [i]superseded[/i], or [i]bad[/i]</p>

<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<p>TRUE if the rows of the alignment map to multiple sequences, FALSE if they map to single sequences</p>

<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<p>type of sequence being aligned, currently either [i]Protein[/i], [i]DNA[/i], [i]RNA[/i], or [i]Mixed[/i]</p>

<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<p>date and time the alignment was loaded</p>

<dt><a name="method"
>method</a></dt>

<dd>
<p>name of the primary software package or script used to construct the alignment</p>

<dt><a name="parameters"
>parameters</a></dt>

<dd>
<p>non-default parameters used as input to the software package or script indicated in the method attribute</p>

<dt><a name="protocol"
>protocol</a></dt>

<dd>
<p>description of the steps taken to construct the alignment, or a reference to an external pipeline</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>ID of this alignment in the source database</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_AlignmentAttribute</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_AlignmentAttribute [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This entity represents an attribute type that can be assigned to an alignment.
The attribute values are stored in the relationships to the target.
The key is the attribute name.</p>

<p>Example:</p>

<pre>    get_entity_AlignmentAttribute -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AlignmentAttribute entity has the following relationship links:</p>

<dl>
<dt><a name="DescribesAlignment_Alignment"
>DescribesAlignment Alignment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_AlignmentAttribute [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_AlignmentRow</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_AlignmentRow [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This entity represents a single row of an alignment.
In general,
this corresponds to a sequence,
but in a concatenated alignment multiple sequences may be represented here.</p>

<p>Example:</p>

<pre>    get_entity_AlignmentRow -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AlignmentRow entity has the following relationship links:</p>

<dl>
<dt><a name="ContainsAlignedDNA_ContigSequence"
>ContainsAlignedDNA ContigSequence</a></dt>

<dd>
<dt><a name="ContainsAlignedProtein_ProteinSequence"
>ContainsAlignedProtein ProteinSequence</a></dt>

<dd>
<dt><a name="IsAlignmentRowIn_Alignment"
>IsAlignmentRowIn Alignment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_AlignmentRow [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="row_number"
>row_number</a></dt>

<dd>
<p>1-based ordinal number of this row in the alignment</p>

<dt><a name="row_id"
>row_id</a></dt>

<dd>
<p>identifier for this row in the FASTA file for the alignment</p>

<dt><a name="row_description"
>row_description</a></dt>

<dd>
<p>description of this row in the FASTA file for the alignment</p>

<dt><a name="n_components"
>n_components</a></dt>

<dd>
<p>number of components that make up this alignment row; for a single-sequence alignment this is always &#34;1&#34;</p>

<dt><a name="beg_pos_aln"
>beg_pos_aln</a></dt>

<dd>
<p>the 1-based column index in the alignment where this sequence row begins</p>

<dt><a name="end_pos_aln"
>end_pos_aln</a></dt>

<dd>
<p>the 1-based column index in the alignment where this sequence row ends</p>

<dt><a name="md5_of_ungapped_sequence"
>md5_of_ungapped_sequence</a></dt>

<dd>
<p>the MD5 of this row&#39;s sequence after gaps have been removed; for DNA and RNA sequences, the [b]U[/b] codes are also normalized to [b]T[/b] before the MD5 is computed</p>

<dt><a name="sequence"
>sequence</a></dt>

<dd>
<p>sequence for this alignment row (with indels)</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_AlleleFrequency</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_AlleleFrequency [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>An allele frequency represents a summary of the major and minor allele frequencies for a position on a chromosome.</p>

<p>Example:</p>

<pre>    get_entity_AlleleFrequency -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AlleleFrequency entity has the following relationship links:</p>

<dl>
<dt><a name="Summarizes_Contig"
>Summarizes Contig</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_AlleleFrequency [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>identifier for this allele in the original (source) database</p>

<dt><a name="position"
>position</a></dt>

<dd>
<p>Specific position on the contig where the allele occurs</p>

<dt><a name="minor_AF"
>minor_AF</a></dt>

<dd>
<p>Minor allele frequency. Floating point number from 0.0 to 0.5.</p>

<dt><a name="minor_allele"
>minor_allele</a></dt>

<dd>
<p>Text letter representation of the minor allele. Valid values are A, C, G, and T.</p>

<dt><a name="major_AF"
>major_AF</a></dt>

<dd>
<p>Major allele frequency. Floating point number less than or equal to 1.0.</p>

<dt><a name="major_allele"
>major_allele</a></dt>

<dd>
<p>Text letter representation of the major allele. Valid values are A, C, G, and T.</p>

<dt><a name="obs_unit_count"
>obs_unit_count</a></dt>

<dd>
<p>Number of observational units used to compute the allele frequencies. Indicates the quality of the analysis.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Annotation</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Annotation [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>An annotation is a comment attached to a feature.
Annotations are used to track the history of a feature&#39;s functional assignments and any related issues.
The key is the feature ID followed by a colon and a complemented ten-digit sequence number.</p>

<p>Example:</p>

<pre>    get_entity_Annotation -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Annotation entity has the following relationship links:</p>

<dl>
<dt><a name="Annotates_Feature"
>Annotates Feature</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Annotation [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="annotator"
>annotator</a></dt>

<dd>
<p>name of the annotator who made the comment</p>

<dt><a name="comment"
>comment</a></dt>

<dd>
<p>text of the annotation</p>

<dt><a name="annotation_time"
>annotation_time</a></dt>

<dd>
<p>date and time at which the annotation was made</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Assay</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Assay [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>An assay is an experimental design for determining alleles at specific chromosome positions.</p>

<p>Example:</p>

<pre>    get_entity_Assay -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Assay entity has the following relationship links:</p>

<dl>
<dt><a name="IsAssayOf_StudyExperiment"
>IsAssayOf StudyExperiment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Assay [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>identifier for this assay in the original (source) database</p>

<dt><a name="assay_type"
>assay_type</a></dt>

<dd>
<p>Text description of the type of assay (e.g., SNP, length, sequence, categorical, array, short read, SSR marker, AFLP marker)</p>

<dt><a name="assay_type_id"
>assay_type_id</a></dt>

<dd>
<p>source ID associated with the assay type (informational)</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_AtomicRegulon</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_AtomicRegulon [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>An atomic regulon is an indivisible group of coregulated features on a single genome.
Atomic regulons are constructed so that a given feature can only belong to one.
Because of this,
the expression levels for atomic regulons represent in some sense the state of a cell.</p>

<p>Example:</p>

<pre>    get_entity_AtomicRegulon -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AtomicRegulon entity has the following relationship links:</p>

<dl>
<dt><a name="IsAffectedIn_Experiment"
>IsAffectedIn Experiment</a></dt>

<dd>
<dt><a name="IsFormedOf_Feature"
>IsFormedOf Feature</a></dt>

<dd>
<dt><a name="ReflectsStateOf_Genome"
>ReflectsStateOf Genome</a></dt>

<dd>
<dt><a name="WasGeneratedFrom_ProbeSet"
>WasGeneratedFrom ProbeSet</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_AtomicRegulon [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Attribute</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Attribute [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>An attribute describes a category of condition or characteristic for an experiment.
The goals of the experiment can be inferred from its values for all the attributes of interest.</p>

<p>Example:</p>

<pre>    get_entity_Attribute -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Attribute entity has the following relationship links:</p>

<dl>
<dt><a name="HasValueIn_Experiment"
>HasValueIn Experiment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Attribute [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="description"
>description</a></dt>

<dd>
<p>Descriptive text indicating the nature and use of this attribute.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Biomass</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Biomass [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A biomass is a collection of compounds in a specific ratio and in specific compartments that are necessary for a cell to function properly.
The prediction of biomasses is key to the functioning of the model.
Each biomass belongs to a specific model.</p>

<p>Example:</p>

<pre>    get_entity_Biomass -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Biomass entity has the following relationship links:</p>

<dl>
<dt><a name="IsComprisedOf_CompoundInstance"
>IsComprisedOf CompoundInstance</a></dt>

<dd>
<dt><a name="IsManagedBy_Model"
>IsManagedBy Model</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Biomass [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<p>last modification date of the biomass data</p>

<dt><a name="name"
>name</a></dt>

<dd>
<p>descriptive name for this biomass</p>

<dt><a name="dna"
>dna</a></dt>

<dd>
<p>portion of a gram of this biomass (expressed as a fraction of 1.0) that is DNA</p>

<dt><a name="protein"
>protein</a></dt>

<dd>
<p>portion of a gram of this biomass (expressed as a fraction of 1.0) that is protein</p>

<dt><a name="cell_wall"
>cell_wall</a></dt>

<dd>
<p>portion of a gram of this biomass (expressed as a fraction of 1.0) that is cell wall</p>

<dt><a name="lipid"
>lipid</a></dt>

<dd>
<p>portion of a gram of this biomass (expressed as a fraction of 1.0) that is lipid but is not part of the cell wall</p>

<dt><a name="cofactor"
>cofactor</a></dt>

<dd>
<p>portion of a gram of this biomass (expressed as a fraction of 1.0) that function as cofactors</p>

<dt><a name="energy"
>energy</a></dt>

<dd>
<p>number of ATP molecules hydrolized per gram of this biomass</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_CodonUsage</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_CodonUsage [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This entity contains information about the codon usage frequency in a particular genome with respect to a particular type of analysis (e.g.
high-expression genes,
modal,
mean,
etc.).</p>

<p>Example:</p>

<pre>    get_entity_CodonUsage -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The CodonUsage entity has the following relationship links:</p>

<dl>
<dt><a name="AreCodonsFor_Genome"
>AreCodonsFor Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_CodonUsage [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="frequencies"
>frequencies</a></dt>

<dd>
<p>A packed-string representation of the codon usage frequencies. These are not global frequencies, but rather frequenicy of use relative to other codons that produce the same amino acid.</p>

<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<p>Genetic code used for these codons.</p>

<dt><a name="type"
>type</a></dt>

<dd>
<p>Type of frequency analysis: average, modal, high-expression, or non-native.</p>

<dt><a name="subtype"
>subtype</a></dt>

<dd>
<p>Specific nature of the codon usage with respect to the given type, generally indicative of how the frequencies were computed.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Complex</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Complex [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A complex is a set of chemical reactions that act in concert to effect a role.</p>

<p>Example:</p>

<pre>    get_entity_Complex -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Complex entity has the following relationship links:</p>

<dl>
<dt><a name="HasStep_Reaction"
>HasStep Reaction</a></dt>

<dd>
<dt><a name="IsTriggeredBy_Role"
>IsTriggeredBy Role</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Complex [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<p>name of this complex. Not all complexes have names.</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>ID of this complex in the source from which it was added.</p>

<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<p>date and time of the last change to this complex&#39;s definition</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Compound</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Compound [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A compound is a chemical that participates in a reaction.
Both ligands and reaction components are treated as compounds.</p>

<p>Example:</p>

<pre>    get_entity_Compound -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Compound entity has the following relationship links:</p>

<dl>
<dt><a name="ComponentOf_Compound"
>ComponentOf Compound</a></dt>

<dd>
<dt><a name="CompoundMeasuredBy_Measurement"
>CompoundMeasuredBy Measurement</a></dt>

<dd>
<dt><a name="ConsistsOfCompounds_Compound"
>ConsistsOfCompounds Compound</a></dt>

<dd>
<dt><a name="IncludedIn_Environment"
>IncludedIn Environment</a></dt>

<dd>
<dt><a name="IsPresentIn_Media"
>IsPresentIn Media</a></dt>

<dd>
<dt><a name="IsShownOn_Diagram"
>IsShownOn Diagram</a></dt>

<dd>
<dt><a name="IsTerminusFor_Scenario"
>IsTerminusFor Scenario</a></dt>

<dd>
<dt><a name="ParticipatesAs_LocalizedCompound"
>ParticipatesAs LocalizedCompound</a></dt>

<dd>
<dt><a name="UsesAliasForCompound_Source"
>UsesAliasForCompound Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Compound [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="label"
>label</a></dt>

<dd>
<p>primary name of the compound, for use in displaying reactions</p>

<dt><a name="abbr"
>abbr</a></dt>

<dd>
<p>shortened abbreviation for the compound name</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>common modeling ID of this compound</p>

<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<p>TRUE if this compound is found in most reactions, else FALSE</p>

<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<p>date and time of the last modification to the compound definition</p>

<dt><a name="mass"
>mass</a></dt>

<dd>
<p>pH-neutral atomic mass of the compound</p>

<dt><a name="formula"
>formula</a></dt>

<dd>
<p>a pH-neutral formula for the compound</p>

<dt><a name="charge"
>charge</a></dt>

<dd>
<p>computed charge of the compound in a pH-neutral solution</p>

<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<p>the pH 7 reference Gibbs free-energy of formation for this compound as calculated by the group contribution method (units are kcal/mol)</p>

<dt><a name="deltaG_error"
>deltaG_error</a></dt>

<dd>
<p>the uncertainty in the [b]deltaG[/b] value (units are kcal/mol)</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_CompoundInstance</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_CompoundInstance [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A Compound Instance represents the occurrence of a particular compound in a location in a model.</p>

<p>Example:</p>

<pre>    get_entity_CompoundInstance -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The CompoundInstance entity has the following relationship links:</p>

<dl>
<dt><a name="Comprises_Biomass"
>Comprises Biomass</a></dt>

<dd>
<dt><a name="HasRealLocationIn_LocationInstance"
>HasRealLocationIn LocationInstance</a></dt>

<dd>
<dt><a name="IsReagentIn_ReactionInstance"
>IsReagentIn ReactionInstance</a></dt>

<dd>
<dt><a name="IsUsageOf_LocalizedCompound"
>IsUsageOf LocalizedCompound</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_CompoundInstance [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="charge"
>charge</a></dt>

<dd>
<p>computed charge based on the location instance pH and similar constraints</p>

<dt><a name="formula"
>formula</a></dt>

<dd>
<p>computed chemical formula for this compound based on the location instance pH and similar constraints</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Contig</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Contig [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A contig is thought of as composing a part of the DNA associated with a specific genome.
It is represented as an ID (including the genome ID) and a ContigSequence.
We do not think of strings of DNA from,
say,
a metgenomic sample as &#34;contigs&#34;,
since there is no associated genome (these would be considered ContigSequences).
This use of the term &#34;ContigSequence&#34;,
rather than just &#34;DNA sequence&#34;,
may turn out to be a bad idea.
For now,
you should just realize that a Contig has an associated genome,
but a ContigSequence does not.</p>

<p>Example:</p>

<pre>    get_entity_Contig -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Contig entity has the following relationship links:</p>

<dl>
<dt><a name="HasAsSequence_ContigSequence"
>HasAsSequence ContigSequence</a></dt>

<dd>
<dt><a name="HasVariationIn_ObservationalUnit"
>HasVariationIn ObservationalUnit</a></dt>

<dd>
<dt><a name="IsComponentOf_Genome"
>IsComponentOf Genome</a></dt>

<dd>
<dt><a name="IsImpactedBy_Trait"
>IsImpactedBy Trait</a></dt>

<dd>
<dt><a name="IsLocusFor_Feature"
>IsLocusFor Feature</a></dt>

<dd>
<dt><a name="IsSummarizedBy_AlleleFrequency"
>IsSummarizedBy AlleleFrequency</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Contig [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>ID of this contig from the core (source) database</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_ContigChunk</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_ContigChunk [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>ContigChunks are strings of DNA thought of as being a string in a 4-character alphabet with an associated ID.
We allow a broader alphabet that includes U (for RNA) and the standard ambiguity characters.</p>

<p>Example:</p>

<pre>    get_entity_ContigChunk -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ContigChunk entity has the following relationship links:</p>

<dl>
<dt><a name="IsSectionOf_ContigSequence"
>IsSectionOf ContigSequence</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_ContigChunk [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="sequence"
>sequence</a></dt>

<dd>
<p>base pairs that make up this sequence</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_ContigSequence</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_ContigSequence [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>ContigSequences are strings of DNA.
Contigs have an associated genome,
but ContigSequences do not.
We can think of random samples of DNA as a set of ContigSequences.
There are no length constraints imposed on ContigSequences -- they can be either very short or very long.
The basic unit of data that is moved to/from the database is the ContigChunk,
from which ContigSequences are formed.
The key of a ContigSequence is the sequence&#39;s MD5 identifier.</p>

<p>Example:</p>

<pre>    get_entity_ContigSequence -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ContigSequence entity has the following relationship links:</p>

<dl>
<dt><a name="HasSection_ContigChunk"
>HasSection ContigChunk</a></dt>

<dd>
<dt><a name="IsAlignedDNAComponentOf_AlignmentRow"
>IsAlignedDNAComponentOf AlignmentRow</a></dt>

<dd>
<dt><a name="IsSequenceOf_Contig"
>IsSequenceOf Contig</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_ContigSequence [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="length"
>length</a></dt>

<dd>
<p>number of base pairs in the contig</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_CoregulatedSet</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_CoregulatedSet [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>We need to represent sets of genes that are coregulated via some regulatory mechanism.
In particular,
we wish to represent genes that are coregulated using transcription binding sites and corresponding transcription regulatory proteins.
We represent a coregulated set (which may,
or may not,
be considered a regulon) using CoregulatedSet.</p>

<p>Example:</p>

<pre>    get_entity_CoregulatedSet -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The CoregulatedSet entity has the following relationship links:</p>

<dl>
<dt><a name="IsControlledUsing_Feature"
>IsControlledUsing Feature</a></dt>

<dd>
<dt><a name="IsRegulatedSetOf_Feature"
>IsRegulatedSetOf Feature</a></dt>

<dd>
<dt><a name="WasFormulatedBy_Source"
>WasFormulatedBy Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_CoregulatedSet [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>original ID of this coregulated set in the source (core) database</p>

<dt><a name="binding_location"
>binding_location</a></dt>

<dd>
<p>binding location for this set&#39;s transcription factor; there may be none of these or there may be more than one</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Diagram</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Diagram [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A functional diagram describes a network of chemical reactions,
often comprising a single subsystem.</p>

<p>Example:</p>

<pre>    get_entity_Diagram -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Diagram entity has the following relationship links:</p>

<dl>
<dt><a name="Displays_Reaction"
>Displays Reaction</a></dt>

<dd>
<dt><a name="IncludesPartOf_Scenario"
>IncludesPartOf Scenario</a></dt>

<dd>
<dt><a name="IsRelevantFor_Subsystem"
>IsRelevantFor Subsystem</a></dt>

<dd>
<dt><a name="Shows_Compound"
>Shows Compound</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Diagram [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<p>descriptive name of this diagram</p>

<dt><a name="content"
>content</a></dt>

<dd>
<p>content of the diagram, in PNG format</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_EcNumber</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_EcNumber [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>EC numbers are assigned by the Enzyme Commission,
and consist of four numbers separated by periods,
each indicating a successively smaller cateogry of enzymes.</p>

<p>Example:</p>

<pre>    get_entity_EcNumber -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The EcNumber entity has the following relationship links:</p>

<dl>
<dt><a name="IsConsistentWith_Role"
>IsConsistentWith Role</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_EcNumber [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="obsolete"
>obsolete</a></dt>

<dd>
<p>This boolean indicates when an EC number is obsolete.</p>

<dt><a name="replacedby"
>replacedby</a></dt>

<dd>
<p>When an obsolete EC number is replaced with another EC number, this string will hold the name of the replacement EC number.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Environment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Environment [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>An Environment is a set of conditions for microbial growth,
including temperature,
aerobicity,
media,
and supplementary conditions.</p>

<p>Example:</p>

<pre>    get_entity_Environment -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Environment entity has the following relationship links:</p>

<dl>
<dt><a name="HasMedia_Media"
>HasMedia Media</a></dt>

<dd>
<dt><a name="HasParameter_Parameter"
>HasParameter Parameter</a></dt>

<dd>
<dt><a name="IncludesAdditionalCompounds_Compound"
>IncludesAdditionalCompounds Compound</a></dt>

<dd>
<dt><a name="IsContextOf_ExperimentalUnit"
>IsContextOf ExperimentalUnit</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Environment [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="temperature"
>temperature</a></dt>

<dd>
<p>The temperature in Kelvin.</p>

<dt><a name="description"
>description</a></dt>

<dd>
<p>A description of the environment.</p>

<dt><a name="oxygenConcentration"
>oxygenConcentration</a></dt>

<dd>
<p>The oxygen concentration in the environment in Molar (mol/L). A value of -1 indicates that there is oxygen in the environment but the concentration is not known, (e.g. an open air shake flask experiment).</p>

<dt><a name="pH"
>pH</a></dt>

<dd>
<p>The pH of the media used in the environment.</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the environment used by the data source.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Experiment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Experiment [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>An experiment is a combination of conditions for which gene expression information is desired.
The result of the experiment is a set of expression levels for features under the given conditions.</p>

<p>Example:</p>

<pre>    get_entity_Experiment -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Experiment entity has the following relationship links:</p>

<dl>
<dt><a name="AffectsLevelOf_AtomicRegulon"
>AffectsLevelOf AtomicRegulon</a></dt>

<dd>
<dt><a name="HasResultsFor_ProbeSet"
>HasResultsFor ProbeSet</a></dt>

<dd>
<dt><a name="HasValueFor_Attribute"
>HasValueFor Attribute</a></dt>

<dd>
<dt><a name="IndicatesSignalFor_Feature"
>IndicatesSignalFor Feature</a></dt>

<dd>
<dt><a name="OperatesIn_Media"
>OperatesIn Media</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Experiment [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source"
>source</a></dt>

<dd>
<p>Publication or lab relevant to this experiment.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_ExperimentalUnit</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_ExperimentalUnit [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>An ExperimentalUnit is a subset of an experiment consisting of a Strain,
an Environment,
and one or more Measurements on that strain in the specified environment.
ExperimentalUnits belong to a single experiment.</p>

<p>Example:</p>

<pre>    get_entity_ExperimentalUnit -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ExperimentalUnit entity has the following relationship links:</p>

<dl>
<dt><a name="GroupedBy_ExperimentalUnitGroup"
>GroupedBy ExperimentalUnitGroup</a></dt>

<dd>
<dt><a name="HasEnvironment_Environment"
>HasEnvironment Environment</a></dt>

<dd>
<dt><a name="HasMeasurement_Measurement"
>HasMeasurement Measurement</a></dt>

<dd>
<dt><a name="IncludesStrain_Strain"
>IncludesStrain Strain</a></dt>

<dd>
<dt><a name="IsExperimentalUnitOf_ExperimentMeta"
>IsExperimentalUnitOf ExperimentMeta</a></dt>

<dd>
<dt><a name="IsTimepointOf_TimeSeries"
>IsTimepointOf TimeSeries</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_ExperimentalUnit [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the experimental unit used by the data source.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_ExperimentalUnitGroup</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_ExperimentalUnitGroup [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>An ExperimentalUnitGroup allows for grouping related experimental units and their measurements - for instance measurements that were in the same plate.</p>

<p>Example:</p>

<pre>    get_entity_ExperimentalUnitGroup -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ExperimentalUnitGroup entity has the following relationship links:</p>

<dl>
<dt><a name="ContainsExperimentalUnit_ExperimentalUnit"
>ContainsExperimentalUnit ExperimentalUnit</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_ExperimentalUnitGroup [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the experimental unit group used by the data source.</p>

<dt><a name="name"
>name</a></dt>

<dd>
<p>The name of this group, if any.</p>

<dt><a name="comments"
>comments</a></dt>

<dd>
<p>Any comments about this group.</p>

<dt><a name="groupType"
>groupType</a></dt>

<dd>
<p>The type of this grouping, for example &#39;24 well plate&#39;, &#39;96 well plate&#39;, &#39;384 well plate&#39;, &#39;microarray&#39;.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_ExperimentMeta</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_ExperimentMeta [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>An Experiment consists of (potentially) multiple strains,
environments,
and measurements on those strains and environments.</p>

<p>Example:</p>

<pre>    get_entity_ExperimentMeta -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ExperimentMeta entity has the following relationship links:</p>

<dl>
<dt><a name="ExperimentPublishedIn_Publication"
>ExperimentPublishedIn Publication</a></dt>

<dd>
<dt><a name="HasExperimentalUnit_ExperimentalUnit"
>HasExperimentalUnit ExperimentalUnit</a></dt>

<dd>
<dt><a name="PerformedBy_Person"
>PerformedBy Person</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_ExperimentMeta [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="title"
>title</a></dt>

<dd>
<p>Title of the experiment.</p>

<dt><a name="description"
>description</a></dt>

<dd>
<p>Description of the experiment including the experimental plan, general results, and conclusions, if possible.</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the experiment used by the data source.</p>

<dt><a name="startDate"
>startDate</a></dt>

<dd>
<p>The date this experiment was started.</p>

<dt><a name="comments"
>comments</a></dt>

<dd>
<p>Any data describing the experiment that is not covered by the description field.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Family</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Family [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The Kbase will support the maintenance of protein families (as sets of Features with associated translations).
We are initially only supporting the notion of a family as composed of a set of isofunctional homologs.
That is,
the families we initially support should be thought of as containing protein-encoding genes whose associated sequences all implement the same function (we do understand that the notion of &#34;function&#34; is somewhat ambiguous,
so let us sweep this under the rug by calling a functional role a &#34;primitive concept&#34;).
We currently support families in which the members are protein sequences as well.
Identical protein sequences as products of translating distinct genes may or may not have identical functions.
This may be justified,
since in a very,
very,
very few cases identical proteins do,
in fact,
have distinct functions.</p>

<p>Example:</p>

<pre>    get_entity_Family -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Family entity has the following relationship links:</p>

<dl>
<dt><a name="HasMember_Feature"
>HasMember Feature</a></dt>

<dd>
<dt><a name="HasProteinMember_ProteinSequence"
>HasProteinMember ProteinSequence</a></dt>

<dd>
<dt><a name="IsCoupledTo_Family"
>IsCoupledTo Family</a></dt>

<dd>
<dt><a name="IsCoupledWith_Family"
>IsCoupledWith Family</a></dt>

<dd>
<dt><a name="IsFamilyFor_Role"
>IsFamilyFor Role</a></dt>

<dd>
<dt><a name="IsRepresentedIn_Genome"
>IsRepresentedIn Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Family [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="type"
>type</a></dt>

<dd>
<p>type of protein family (e.g. FIGfam, equivalog)</p>

<dt><a name="release"
>release</a></dt>

<dd>
<p>release number / subtype of protein family</p>

<dt><a name="family_function"
>family_function</a></dt>

<dd>
<p>optional free-form description of the family. For function-based families, this would be the functional role for the family members.</p>

<dt><a name="alignment"
>alignment</a></dt>

<dd>
<p>FASTA-formatted alignment of the family&#39;s protein sequences</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Feature</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Feature [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A feature (sometimes also called a gene) is a part of a genome that is of special interest.
Features may be spread across multiple DNA sequences (contigs) of a genome,
but never across more than one genome.
Each feature in the database has a unique ID that functions as its ID in this table.
Normally a Feature is just a single contigous region on a contig.
Features have types,
and an appropriate choice of available types allows the support of protein-encoding genes,
exons,
RNA genes,
binding sites,
pathogenicity islands,
or whatever.</p>

<p>Example:</p>

<pre>    get_entity_Feature -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Feature entity has the following relationship links:</p>

<dl>
<dt><a name="Controls_CoregulatedSet"
>Controls CoregulatedSet</a></dt>

<dd>
<dt><a name="Encompasses_Feature"
>Encompasses Feature</a></dt>

<dd>
<dt><a name="FeatureMeasuredBy_Measurement"
>FeatureMeasuredBy Measurement</a></dt>

<dd>
<dt><a name="HasCoregulationWith_Feature"
>HasCoregulationWith Feature</a></dt>

<dd>
<dt><a name="HasFunctional_Role"
>HasFunctional Role</a></dt>

<dd>
<dt><a name="HasIndicatedSignalFrom_Experiment"
>HasIndicatedSignalFrom Experiment</a></dt>

<dd>
<dt><a name="HasLevelsFrom_ProbeSet"
>HasLevelsFrom ProbeSet</a></dt>

<dd>
<dt><a name="ImplementsReaction_ReactionInstance"
>ImplementsReaction ReactionInstance</a></dt>

<dd>
<dt><a name="IsAnnotatedBy_Annotation"
>IsAnnotatedBy Annotation</a></dt>

<dd>
<dt><a name="IsContainedIn_SSCell"
>IsContainedIn SSCell</a></dt>

<dd>
<dt><a name="IsCoregulatedWith_Feature"
>IsCoregulatedWith Feature</a></dt>

<dd>
<dt><a name="IsEncompassedIn_Feature"
>IsEncompassedIn Feature</a></dt>

<dd>
<dt><a name="IsExemplarOf_Role"
>IsExemplarOf Role</a></dt>

<dd>
<dt><a name="IsFormedInto_AtomicRegulon"
>IsFormedInto AtomicRegulon</a></dt>

<dd>
<dt><a name="IsInPair_Pairing"
>IsInPair Pairing</a></dt>

<dd>
<dt><a name="IsLocatedIn_Contig"
>IsLocatedIn Contig</a></dt>

<dd>
<dt><a name="IsMemberOf_Family"
>IsMemberOf Family</a></dt>

<dd>
<dt><a name="IsOwnedBy_Genome"
>IsOwnedBy Genome</a></dt>

<dd>
<dt><a name="IsRegulatedIn_CoregulatedSet"
>IsRegulatedIn CoregulatedSet</a></dt>

<dd>
<dt><a name="KnockedOutIn_Strain"
>KnockedOutIn Strain</a></dt>

<dd>
<dt><a name="Produces_ProteinSequence"
>Produces ProteinSequence</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Feature [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<p>Code indicating the type of this feature. Among the codes currently supported are &#34;peg&#34; for a protein encoding gene, &#34;bs&#34; for a binding site, &#34;opr&#34; for an operon, and so forth.</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>ID for this feature in its original source (core) database</p>

<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<p>Number of base pairs in this feature.</p>

<dt><a name="function"
>function</a></dt>

<dd>
<p>Functional assignment for this feature. This will often indicate the feature&#39;s functional role or roles, and may also have comments.</p>

<dt><a name="alias"
>alias</a></dt>

<dd>
<p>alternative identifier for the feature. These are highly unstructured, and frequently non-unique.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Genome</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Genome [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The Kbase houses a large and growing set of genomes.
We often have multiple genomes that have identical DNA.
These usually have distinct gene calls and annotations,
but not always.
We consider the Kbase to be a framework for managing hundreds of thousands of genomes and offering the tools needed to support compartive analysis on large sets of genomes,
some of which are virtually identical.</p>

<p>Example:</p>

<pre>    get_entity_Genome -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Genome entity has the following relationship links:</p>

<dl>
<dt><a name="GenomeParentOf_Strain"
>GenomeParentOf Strain</a></dt>

<dd>
<dt><a name="HadResultsProducedBy_ProbeSet"
>HadResultsProducedBy ProbeSet</a></dt>

<dd>
<dt><a name="HasRepresentativeOf_Family"
>HasRepresentativeOf Family</a></dt>

<dd>
<dt><a name="IsCollectedInto_OTU"
>IsCollectedInto OTU</a></dt>

<dd>
<dt><a name="IsComposedOf_Contig"
>IsComposedOf Contig</a></dt>

<dd>
<dt><a name="IsConfiguredBy_AtomicRegulon"
>IsConfiguredBy AtomicRegulon</a></dt>

<dd>
<dt><a name="IsInTaxa_TaxonomicGrouping"
>IsInTaxa TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsModeledBy_Model"
>IsModeledBy Model</a></dt>

<dd>
<dt><a name="IsOwnerOf_Feature"
>IsOwnerOf Feature</a></dt>

<dd>
<dt><a name="IsReferencedBy_ObservationalUnit"
>IsReferencedBy ObservationalUnit</a></dt>

<dd>
<dt><a name="Uses_SSRow"
>Uses SSRow</a></dt>

<dd>
<dt><a name="UsesCodons_CodonUsage"
>UsesCodons CodonUsage</a></dt>

<dd>
<dt><a name="WasSubmittedBy_Source"
>WasSubmittedBy Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Genome [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<p>Number of protein encoding genes for this genome.</p>

<dt><a name="rnas"
>rnas</a></dt>

<dd>
<p>Number of RNA features found for this organism.</p>

<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<p>Full genus/species/strain name of the genome sequence.</p>

<dt><a name="complete"
>complete</a></dt>

<dd>
<p>TRUE if the genome sequence is complete, else FALSE</p>

<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<p>TRUE if this is a prokaryotic genome sequence, else FALSE</p>

<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<p>Number of base pairs in the genome sequence.</p>

<dt><a name="contigs"
>contigs</a></dt>

<dd>
<p>Number of contigs for this genome sequence.</p>

<dt><a name="domain"
>domain</a></dt>

<dd>
<p>Domain for this organism (Archaea, Bacteria, Eukaryota, Virus, Plasmid, or Environmental Sample).</p>

<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<p>Genetic code number used for protein translation on most of this genome sequence&#39;s contigs.</p>

<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<p>Percent GC content present in the genome sequence&#39;s DNA.</p>

<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<p>zero or more strings describing phenotypic information about this genome sequence</p>

<dt><a name="md5"
>md5</a></dt>

<dd>
<p>MD5 identifier describing the genome&#39;s DNA sequence</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>identifier assigned to this genome by the original source</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Locality</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Locality [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A locality is a geographic location.</p>

<p>Example:</p>

<pre>    get_entity_Locality -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Locality entity has the following relationship links:</p>

<dl>
<dt><a name="HasUnits_ObservationalUnit"
>HasUnits ObservationalUnit</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Locality [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<p>Name or description of the location used as a collection site.</p>

<dt><a name="city"
>city</a></dt>

<dd>
<p>City of the collecting site.</p>

<dt><a name="state"
>state</a></dt>

<dd>
<p>State or province of the collecting site.</p>

<dt><a name="country"
>country</a></dt>

<dd>
<p>Country of the collecting site.</p>

<dt><a name="origcty"
>origcty</a></dt>

<dd>
<p>3-letter ISO 3166-1 extended country code for the country of origin.</p>

<dt><a name="elevation"
>elevation</a></dt>

<dd>
<p>Elevation of the collecting site, expressed in meters above sea level. Negative values are allowed.</p>

<dt><a name="latitude"
>latitude</a></dt>

<dd>
<p>Latitude of the collecting site, recorded as a decimal number. North latitudes are positive values and south latitudes are negative numbers.</p>

<dt><a name="longitude"
>longitude</a></dt>

<dd>
<p>Longitude of the collecting site, recorded as a decimal number. West longitudes are positive values and east longitudes are negative numbers.</p>

<dt><a name="lo_accession"
>lo_accession</a></dt>

<dd>
<p>gazeteer ontology term ID</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_LocalizedCompound</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_LocalizedCompound [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This entity represents a compound occurring in a specific location.
A reaction always involves localized compounds.
If a reaction occurs entirely in a single location,
it will frequently only be represented by the cytoplasmic versions of the compounds; however,
a transport always uses specifically located compounds.</p>

<p>Example:</p>

<pre>    get_entity_LocalizedCompound -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The LocalizedCompound entity has the following relationship links:</p>

<dl>
<dt><a name="HasUsage_CompoundInstance"
>HasUsage CompoundInstance</a></dt>

<dd>
<dt><a name="IsInvolvedIn_Reaction"
>IsInvolvedIn Reaction</a></dt>

<dd>
<dt><a name="IsParticipationOf_Compound"
>IsParticipationOf Compound</a></dt>

<dd>
<dt><a name="ParticipatesAt_Location"
>ParticipatesAt Location</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_LocalizedCompound [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Location</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Location [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A location is a region of the cell where reaction compounds originate from or are transported to (e.g.
cell wall,
extracellular,
cytoplasm).</p>

<p>Example:</p>

<pre>    get_entity_Location -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Location entity has the following relationship links:</p>

<dl>
<dt><a name="IsInstantiatedBy_LocationInstance"
>IsInstantiatedBy LocationInstance</a></dt>

<dd>
<dt><a name="IsParticipatingAt_LocalizedCompound"
>IsParticipatingAt LocalizedCompound</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Location [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<p>date and time of the last modification to the compartment&#39;s definition</p>

<dt><a name="name"
>name</a></dt>

<dd>
<p>common name for the location</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>ID from the source of this location</p>

<dt><a name="abbr"
>abbr</a></dt>

<dd>
<p>an abbreviation (usually a single letter) for the location.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_LocationInstance</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_LocationInstance [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The Location Instance represents a region of a cell (e.g.
cell wall,
cytoplasm) as it appears in a specific model.</p>

<p>Example:</p>

<pre>    get_entity_LocationInstance -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The LocationInstance entity has the following relationship links:</p>

<dl>
<dt><a name="IsDivisionOf_Model"
>IsDivisionOf Model</a></dt>

<dd>
<dt><a name="IsInstanceOf_Location"
>IsInstanceOf Location</a></dt>

<dd>
<dt><a name="IsRealLocationOf_CompoundInstance"
>IsRealLocationOf CompoundInstance</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_LocationInstance [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="index"
>index</a></dt>

<dd>
<p>number used to distinguish between different instances of the same type of location in a single model. Within a model, any two instances of the same location must have difference compartment index values.</p>

<dt><a name="label"
>label</a></dt>

<dd>
<p>description used to differentiate between instances of the same location in a single model</p>

<dt><a name="pH"
>pH</a></dt>

<dd>
<p>pH of the cell region, which is used to determine compound charge and pH gradient across cell membranes</p>

<dt><a name="potential"
>potential</a></dt>

<dd>
<p>electrochemical potential of the cell region, which is used to determine the electrochemical gradient across cell membranes</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Measurement</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Measurement [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A Measurement is a value generated by performing a protocol to evaluate a value on an ExperimentalUnit - e.g.
a strain in an environment.</p>

<p>Example:</p>

<pre>    get_entity_Measurement -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Measurement entity has the following relationship links:</p>

<dl>
<dt><a name="IsDefinedBy_MeasurementDescription"
>IsDefinedBy MeasurementDescription</a></dt>

<dd>
<dt><a name="IsMeasureOf_ExperimentalUnit"
>IsMeasureOf ExperimentalUnit</a></dt>

<dd>
<dt><a name="MeasuresCompound_Compound"
>MeasuresCompound Compound</a></dt>

<dd>
<dt><a name="MeasuresFeature_Feature"
>MeasuresFeature Feature</a></dt>

<dd>
<dt><a name="WasMeasuredBy_Protocol"
>WasMeasuredBy Protocol</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Measurement [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the measurement used by the data source.</p>

<dt><a name="value"
>value</a></dt>

<dd>
<p>The value of the measurement.</p>

<dt><a name="mean"
>mean</a></dt>

<dd>
<p>The mean of multiple replicates if they are included in the measurement.</p>

<dt><a name="median"
>median</a></dt>

<dd>
<p>The median of multiple replicates if they are included in the measurement.</p>

<dt><a name="stddev"
>stddev</a></dt>

<dd>
<p>The standard deviation of multiple replicates if they are included in the measurement.</p>

<dt><a name="N"
>N</a></dt>

<dd>
<p>The number of replicates if they are included in the measurement.</p>

<dt><a name="p_value"
>p_value</a></dt>

<dd>
<p>The p-value of multiple replicates if they are included in the measurement. The exact meaning of the p-value is specified in the MeasurementDescription object for this measurement.</p>

<dt><a name="Z_score"
>Z_score</a></dt>

<dd>
<p>The Z-score of multiple replicates if they are included in the measurement. The exact meaning of the p-value is specified in the MeasurementDescription object for this measurement.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_MeasurementDescription</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_MeasurementDescription [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A MeasurementDescription provides information about a measurement value.</p>

<p>Example:</p>

<pre>    get_entity_MeasurementDescription -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The MeasurementDescription entity has the following relationship links:</p>

<dl>
<dt><a name="DescribesMeasurement_Measurement"
>DescribesMeasurement Measurement</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_MeasurementDescription [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<p>The name of the measurement.</p>

<dt><a name="description"
>description</a></dt>

<dd>
<p>The description of the measurement, how it is measured, and what the measurement statistics mean.</p>

<dt><a name="unitOfMeasure"
>unitOfMeasure</a></dt>

<dd>
<p>The units of the measurement.</p>

<dt><a name="category"
>category</a></dt>

<dd>
<p>The category the measurement fits into, for example phenotype, experimental input, environment.</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the measurement description used by the data source.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Media</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Media [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A media describes the chemical content of the solution in which cells are grown in an experiment or for the purposes of a model.
The key is the common media name.
The nature of the media is described by its relationship to its constituent compounds.</p>

<p>Example:</p>

<pre>    get_entity_Media -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Media entity has the following relationship links:</p>

<dl>
<dt><a name="HasPresenceOf_Compound"
>HasPresenceOf Compound</a></dt>

<dd>
<dt><a name="IsUtilizedIn_Experiment"
>IsUtilizedIn Experiment</a></dt>

<dd>
<dt><a name="UsedIn_Environment"
>UsedIn Environment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Media [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<p>date and time of the last modification to the media&#39;s definition</p>

<dt><a name="name"
>name</a></dt>

<dd>
<p>descriptive name of the media</p>

<dt><a name="is_minimal"
>is_minimal</a></dt>

<dd>
<p>TRUE if this is a minimal media, else FALSE</p>

<dt><a name="description"
>description</a></dt>

<dd>
<p>description of the media condition</p>

<dt><a name="solid"
>solid</a></dt>

<dd>
<p>Whether the media is solid (True) or liquid (False).</p>

<dt><a name="is_defined"
>is_defined</a></dt>

<dd>
<p>TRUE if this media condition is defined (all components explicitly known)</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the media used by the data source.</p>

<dt><a name="type"
>type</a></dt>

<dd>
<p>The general category of the media.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Model</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Model [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A model specifies a relationship between sets of features and reactions in a cell.
It is used to simulate cell growth and gene knockouts to validate annotations.</p>

<p>Example:</p>

<pre>    get_entity_Model -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Model entity has the following relationship links:</p>

<dl>
<dt><a name="HasRequirementOf_ReactionInstance"
>HasRequirementOf ReactionInstance</a></dt>

<dd>
<dt><a name="IsDividedInto_LocationInstance"
>IsDividedInto LocationInstance</a></dt>

<dd>
<dt><a name="Manages_Biomass"
>Manages Biomass</a></dt>

<dd>
<dt><a name="Models_Genome"
>Models Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Model [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<p>date and time of the last change to the model data</p>

<dt><a name="name"
>name</a></dt>

<dd>
<p>descriptive name of the model</p>

<dt><a name="version"
>version</a></dt>

<dd>
<p>revision number of the model</p>

<dt><a name="type"
>type</a></dt>

<dd>
<p>string indicating where the model came from (e.g. single genome, multiple genome, or community model)</p>

<dt><a name="status"
>status</a></dt>

<dd>
<p>indicator of whether the model is stable, under construction, or under reconstruction</p>

<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<p>number of reactions in the model</p>

<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<p>number of compounds in the model</p>

<dt><a name="annotation_count"
>annotation_count</a></dt>

<dd>
<p>number of features associated with one or more reactions in the model</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_ObservationalUnit</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_ObservationalUnit [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>An ObservationalUnit is an individual plant that 1) is part of an experiment or study,
2) has measured traits,
and 3) is assayed for the purpose of determining alleles.</p>

<p>Example:</p>

<pre>    get_entity_ObservationalUnit -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ObservationalUnit entity has the following relationship links:</p>

<dl>
<dt><a name="DefinedBy_TaxonomicGrouping"
>DefinedBy TaxonomicGrouping</a></dt>

<dd>
<dt><a name="HasTrait_Trait"
>HasTrait Trait</a></dt>

<dd>
<dt><a name="IsLocated_Locality"
>IsLocated Locality</a></dt>

<dd>
<dt><a name="IsPartOf_StudyExperiment"
>IsPartOf StudyExperiment</a></dt>

<dd>
<dt><a name="IsVariedIn_Contig"
>IsVariedIn Contig</a></dt>

<dd>
<dt><a name="UsesReference_Genome"
>UsesReference Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_ObservationalUnit [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<p>Name/ID by which the observational unit may be known by the originator and is used in queries.</p>

<dt><a name="source_name2"
>source_name2</a></dt>

<dd>
<p>Secondary name/ID by which the observational unit may be known and is queried.</p>

<dt><a name="plant_id"
>plant_id</a></dt>

<dd>
<p>ID of the plant that was tested to produce this observational unit. Observational units with the same plant ID are different assays of a single physical organism.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_OTU</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_OTU [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>An OTU (Organism Taxonomic Unit) is a named group of related genomes.</p>

<p>Example:</p>

<pre>    get_entity_OTU -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The OTU entity has the following relationship links:</p>

<dl>
<dt><a name="IsCollectionOf_Genome"
>IsCollectionOf Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_OTU [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Pairing</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Pairing [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A pairing indicates that two features are found close together in a genome.
Not all possible pairings are stored in the database; only those that are considered for some reason to be significant for annotation purposes.The key of the pairing is the concatenation of the feature IDs in alphabetical order with an intervening colon.</p>

<p>Example:</p>

<pre>    get_entity_Pairing -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Pairing entity has the following relationship links:</p>

<dl>
<dt><a name="Determines_PairSet"
>Determines PairSet</a></dt>

<dd>
<dt><a name="IsPairOf_Feature"
>IsPairOf Feature</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Pairing [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_PairSet</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_PairSet [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A PairSet is a precompute set of pairs or genes.
Each pair occurs close to one another of the chromosome.
We believe that all of the first members of the pairs correspond to one another (are quite similar),
as do all of the second members of the pairs.
These pairs (from prokaryotic genomes) offer one of the most powerful clues relating to uncharacterized genes/peroteins.</p>

<p>Example:</p>

<pre>    get_entity_PairSet -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The PairSet entity has the following relationship links:</p>

<dl>
<dt><a name="IsDeterminedBy_Pairing"
>IsDeterminedBy Pairing</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_PairSet [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="score"
>score</a></dt>

<dd>
<p>Score for this evidence set. The score indicates the number of significantly different genomes represented by the pairings.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Parameter</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Parameter [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A parameter is the name of some quantity that has a value.</p>

<p>Example:</p>

<pre>    get_entity_Parameter -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Parameter entity has the following relationship links:</p>

<dl>
<dt><a name="OfEnvironment_Environment"
>OfEnvironment Environment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Parameter [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Person</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Person [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A person represents a human affiliated in some way with Kbase.</p>

<p>Example:</p>

<pre>    get_entity_Person -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Person entity has the following relationship links:</p>

<dl>
<dt><a name="PerformedExperiment_ExperimentMeta"
>PerformedExperiment ExperimentMeta</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Person [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="firstName"
>firstName</a></dt>

<dd>
<p>The given name of the person.</p>

<dt><a name="lastName"
>lastName</a></dt>

<dd>
<p>The surname of the person.</p>

<dt><a name="contactEmail"
>contactEmail</a></dt>

<dd>
<p>Email address of the person.</p>

<dt><a name="institution"
>institution</a></dt>

<dd>
<p>The institution where the person works.</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the person used by the data source.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_ProbeSet</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_ProbeSet [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A probe set is a device containing multiple probe sequences for use in gene expression experiments.</p>

<p>Example:</p>

<pre>    get_entity_ProbeSet -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ProbeSet entity has the following relationship links:</p>

<dl>
<dt><a name="GeneratedLevelsFor_AtomicRegulon"
>GeneratedLevelsFor AtomicRegulon</a></dt>

<dd>
<dt><a name="HasResultsIn_Experiment"
>HasResultsIn Experiment</a></dt>

<dd>
<dt><a name="IndicatedLevelsFor_Feature"
>IndicatedLevelsFor Feature</a></dt>

<dd>
<dt><a name="ProducedResultsFor_Genome"
>ProducedResultsFor Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_ProbeSet [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_ProteinSequence</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_ProteinSequence [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>We use the concept of ProteinSequence as an amino acid string with an associated MD5 value.
It is easy to access the set of Features that relate to a ProteinSequence.
While function is still associated with Features (and may be for some time),
publications are associated with ProteinSequences (and the inferred impact on Features is through the relationship connecting ProteinSequences to Features).</p>

<p>Example:</p>

<pre>    get_entity_ProteinSequence -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ProteinSequence entity has the following relationship links:</p>

<dl>
<dt><a name="HasAssertedFunctionFrom_Source"
>HasAssertedFunctionFrom Source</a></dt>

<dd>
<dt><a name="IsATopicOf_Publication"
>IsATopicOf Publication</a></dt>

<dd>
<dt><a name="IsAlignedProteinComponentOf_AlignmentRow"
>IsAlignedProteinComponentOf AlignmentRow</a></dt>

<dd>
<dt><a name="IsProteinFor_Feature"
>IsProteinFor Feature</a></dt>

<dd>
<dt><a name="IsProteinMemberOf_Family"
>IsProteinMemberOf Family</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_ProteinSequence [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="sequence"
>sequence</a></dt>

<dd>
<p>The sequence contains the letters corresponding to the protein&#39;s amino acids.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Protocol</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Protocol [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A Protocol is a step by step set of instructions for performing a part of an experiment.</p>

<p>Example:</p>

<pre>    get_entity_Protocol -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Protocol entity has the following relationship links:</p>

<dl>
<dt><a name="IsMeasurementMethodOf_Measurement"
>IsMeasurementMethodOf Measurement</a></dt>

<dd>
<dt><a name="ProtocolPublishedIn_Publication"
>ProtocolPublishedIn Publication</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Protocol [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<p>The name of the protocol.</p>

<dt><a name="description"
>description</a></dt>

<dd>
<p>The step by step instructions for performing the experiment, including measurement details, materials, and equipment. A researcher should be able to reproduce the experimental results with this information.</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the protocol used by the data source.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Publication</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Publication [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Experimenters attach publications to experiments and protocols.
Annotators attach publications to ProteinSequences.
The attached publications give an ID (usually a DOI or Pubmed ID),
a URL to the paper (when we have it),
and a title (when we have it).
Pubmed IDs are given unmodified.
DOI IDs are prefixed with [b]doi:[/b],
e.g.
[i]doi:1002385[/i].</p>

<p>Example:</p>

<pre>    get_entity_Publication -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Publication entity has the following relationship links:</p>

<dl>
<dt><a name="Concerns_ProteinSequence"
>Concerns ProteinSequence</a></dt>

<dd>
<dt><a name="PublishedExperiment_ExperimentMeta"
>PublishedExperiment ExperimentMeta</a></dt>

<dd>
<dt><a name="PublishedProtocol_Protocol"
>PublishedProtocol Protocol</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Publication [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="title"
>title</a></dt>

<dd>
<p>title of the article, or (unknown) if the title is not known</p>

<dt><a name="link"
>link</a></dt>

<dd>
<p>URL of the article, DOI preferred</p>

<dt><a name="pubdate"
>pubdate</a></dt>

<dd>
<p>publication date of the article</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Reaction</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Reaction [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A reaction is a chemical process that converts one set of compounds (substrate) to another set (products).</p>

<p>Example:</p>

<pre>    get_entity_Reaction -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Reaction entity has the following relationship links:</p>

<dl>
<dt><a name="Involves_LocalizedCompound"
>Involves LocalizedCompound</a></dt>

<dd>
<dt><a name="IsDisplayedOn_Diagram"
>IsDisplayedOn Diagram</a></dt>

<dd>
<dt><a name="IsExecutedAs_ReactionInstance"
>IsExecutedAs ReactionInstance</a></dt>

<dd>
<dt><a name="IsStepOf_Complex"
>IsStepOf Complex</a></dt>

<dd>
<dt><a name="ParticipatesIn_Scenario"
>ParticipatesIn Scenario</a></dt>

<dd>
<dt><a name="UsesAliasForReaction_Source"
>UsesAliasForReaction Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Reaction [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<p>date and time of the last modification to this reaction&#39;s definition</p>

<dt><a name="name"
>name</a></dt>

<dd>
<p>descriptive name of this reaction</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>ID of this reaction in the resource from which it was added</p>

<dt><a name="abbr"
>abbr</a></dt>

<dd>
<p>abbreviated name of this reaction</p>

<dt><a name="direction"
>direction</a></dt>

<dd>
<p>direction of this reaction (&#62; for forward-only, &#60; for backward-only, = for bidirectional)</p>

<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<p>Gibbs free-energy change for the reaction calculated using the group contribution method (units are kcal/mol)</p>

<dt><a name="deltaG_error"
>deltaG_error</a></dt>

<dd>
<p>uncertainty in the [b]deltaG[/b] value (units are kcal/mol)</p>

<dt><a name="thermodynamic_reversibility"
>thermodynamic_reversibility</a></dt>

<dd>
<p>computed reversibility of this reaction in a pH-neutral environment</p>

<dt><a name="default_protons"
>default_protons</a></dt>

<dd>
<p>number of protons absorbed by this reaction in a pH-neutral environment</p>

<dt><a name="status"
>status</a></dt>

<dd>
<p>string indicating additional information about this reaction, generally indicating whether the reaction is balanced and/or lumped</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_ReactionInstance</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_ReactionInstance [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A reaction instance describes the specific implementation of a reaction in a model.</p>

<p>Example:</p>

<pre>    get_entity_ReactionInstance -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ReactionInstance entity has the following relationship links:</p>

<dl>
<dt><a name="ImplementedBasedOn_Feature"
>ImplementedBasedOn Feature</a></dt>

<dd>
<dt><a name="IsARequirementOf_Model"
>IsARequirementOf Model</a></dt>

<dd>
<dt><a name="IsExecutionOf_Reaction"
>IsExecutionOf Reaction</a></dt>

<dd>
<dt><a name="Targets_CompoundInstance"
>Targets CompoundInstance</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_ReactionInstance [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="direction"
>direction</a></dt>

<dd>
<p>reaction directionality (&#62; for forward, &#60; for backward, = for bidirectional) with respect to this model</p>

<dt><a name="protons"
>protons</a></dt>

<dd>
<p>number of protons produced by this reaction when proceeding in the forward direction. If this is a transport reaction, these protons end up in the reaction instance&#39;s main location. If the number is negative, then the protons are consumed by the reaction rather than being produced.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Role</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Role [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A role describes a biological function that may be fulfilled by a feature.
One of the main goals of the database is to assign features to roles.
Most roles are effected by the construction of proteins.
Some,
however,
deal with functional regulation and message transmission.</p>

<p>Example:</p>

<pre>    get_entity_Role -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Role entity has the following relationship links:</p>

<dl>
<dt><a name="DeterminesFunctionOf_Family"
>DeterminesFunctionOf Family</a></dt>

<dd>
<dt><a name="HasAsExemplar_Feature"
>HasAsExemplar Feature</a></dt>

<dd>
<dt><a name="IsConsistentTo_EcNumber"
>IsConsistentTo EcNumber</a></dt>

<dd>
<dt><a name="IsFunctionalIn_Feature"
>IsFunctionalIn Feature</a></dt>

<dd>
<dt><a name="IsIncludedIn_Subsystem"
>IsIncludedIn Subsystem</a></dt>

<dd>
<dt><a name="IsRoleOf_SSCell"
>IsRoleOf SSCell</a></dt>

<dd>
<dt><a name="Triggers_Complex"
>Triggers Complex</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Role [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="hypothetical"
>hypothetical</a></dt>

<dd>
<p>TRUE if a role is hypothetical, else FALSE</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Scenario</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Scenario [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A scenario is a partial instance of a subsystem with a defined set of reactions.
Each scenario converts input compounds to output compounds using reactions.
The scenario may use all of the reactions controlled by a subsystem or only some,
and may also incorporate additional reactions.
Because scenario names are not unique,
the actual scenario ID is a number.</p>

<p>Example:</p>

<pre>    get_entity_Scenario -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Scenario entity has the following relationship links:</p>

<dl>
<dt><a name="HasAsTerminus_Compound"
>HasAsTerminus Compound</a></dt>

<dd>
<dt><a name="HasParticipant_Reaction"
>HasParticipant Reaction</a></dt>

<dd>
<dt><a name="Overlaps_Diagram"
>Overlaps Diagram</a></dt>

<dd>
<dt><a name="Validates_Subsystem"
>Validates Subsystem</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Scenario [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="common_name"
>common_name</a></dt>

<dd>
<p>Common name of the scenario. The name, rather than the ID number, is usually displayed everywhere.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Source</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Source [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A source is a user or organization that is permitted to assign its own identifiers or to submit bioinformatic objects to the database.</p>

<p>Example:</p>

<pre>    get_entity_Source -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Source entity has the following relationship links:</p>

<dl>
<dt><a name="Aligned_Alignment"
>Aligned Alignment</a></dt>

<dd>
<dt><a name="AssertsFunctionFor_ProteinSequence"
>AssertsFunctionFor ProteinSequence</a></dt>

<dd>
<dt><a name="Formulated_CoregulatedSet"
>Formulated CoregulatedSet</a></dt>

<dd>
<dt><a name="HasCompoundAliasFrom_Compound"
>HasCompoundAliasFrom Compound</a></dt>

<dd>
<dt><a name="HasReactionAliasFrom_Reaction"
>HasReactionAliasFrom Reaction</a></dt>

<dd>
<dt><a name="Provided_Subsystem"
>Provided Subsystem</a></dt>

<dd>
<dt><a name="Submitted_Genome"
>Submitted Genome</a></dt>

<dd>
<dt><a name="Treed_Tree"
>Treed Tree</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Source [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_SSCell</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_SSCell [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>An SSCell (SpreadSheet Cell) represents a role as it occurs in a subsystem spreadsheet row.
The key is a colon-delimited triple containing an MD5 hash of the subsystem ID followed by a genome ID (with optional region string) and a role abbreviation.</p>

<p>Example:</p>

<pre>    get_entity_SSCell -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SSCell entity has the following relationship links:</p>

<dl>
<dt><a name="Contains_Feature"
>Contains Feature</a></dt>

<dd>
<dt><a name="HasRole_Role"
>HasRole Role</a></dt>

<dd>
<dt><a name="IsRoleFor_SSRow"
>IsRoleFor SSRow</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_SSCell [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_SSRow</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_SSRow [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>An SSRow (that is,
a row in a subsystem spreadsheet) represents a collection of functional roles present in the Features of a single Genome.
The roles are part of a designated subsystem,
and the features associated with each role are included in the row.
That is,
a row amounts to an instance of a subsystem as it exists in a specific,
designated genome.</p>

<p>Example:</p>

<pre>    get_entity_SSRow -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SSRow entity has the following relationship links:</p>

<dl>
<dt><a name="Implements_Variant"
>Implements Variant</a></dt>

<dd>
<dt><a name="IsRowOf_SSCell"
>IsRowOf SSCell</a></dt>

<dd>
<dt><a name="IsUsedBy_Genome"
>IsUsedBy Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_SSRow [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="curated"
>curated</a></dt>

<dd>
<p>This flag is TRUE if the assignment of the molecular machine has been curated, and FALSE if it was made by an automated program.</p>

<dt><a name="region"
>region</a></dt>

<dd>
<p>Region in the genome for which the row is relevant. Normally, this is an empty string, indicating that the machine covers the whole genome. If a subsystem has multiple rows for a genome, this contains a location string describing the region occupied by this particular row.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Strain</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Strain [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This entity represents an organism derived from a genome or another organism with one or more modifications to the organism&#39;s genome.</p>

<p>Example:</p>

<pre>    get_entity_Strain -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Strain entity has the following relationship links:</p>

<dl>
<dt><a name="DerivedFromGenome_Genome"
>DerivedFromGenome Genome</a></dt>

<dd>
<dt><a name="DerivedFromStrain_Strain"
>DerivedFromStrain Strain</a></dt>

<dd>
<dt><a name="EvaluatedIn_ExperimentalUnit"
>EvaluatedIn ExperimentalUnit</a></dt>

<dd>
<dt><a name="HasKnockoutIn_Feature"
>HasKnockoutIn Feature</a></dt>

<dd>
<dt><a name="StrainParentOf_Strain"
>StrainParentOf Strain</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Strain [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<p>The common or laboratory name of the strain, e.g. DH5a or JMP1004.</p>

<dt><a name="description"
>description</a></dt>

<dd>
<p>A description of the strain, e.g. knockout/modification methods, resulting phenotypes, etc.</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the strain used by the data source.</p>

<dt><a name="aggregateData"
>aggregateData</a></dt>

<dd>
<p>Denotes whether this entity represents a physical strain (False) or aggregate data calculated from one or more strains (True).</p>

<dt><a name="wildtype"
>wildtype</a></dt>

<dd>
<p>Denotes this strain is presumably identical to the parent genome.</p>

<dt><a name="referenceStrain"
>referenceStrain</a></dt>

<dd>
<p>Denotes whether this strain is a reference strain; e.g. it is identical to the genome it&#39;s related to (True) or not (False). In contrast to wildtype, a referenceStrain is abstract and does not physically exist and is used for data that refers to a genome but not a particular strain. There should only exist one reference strain per genome and all reference strains are wildtype.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_StudyExperiment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_StudyExperiment [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>An Experiment is a collection of observational units with one originator that are part of a specific study.
An experiment may be conducted at more than one location and in more than one season or year.</p>

<p>Example:</p>

<pre>    get_entity_StudyExperiment -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The StudyExperiment entity has the following relationship links:</p>

<dl>
<dt><a name="IncludesPart_ObservationalUnit"
>IncludesPart ObservationalUnit</a></dt>

<dd>
<dt><a name="IsAssayedBy_Assay"
>IsAssayedBy Assay</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_StudyExperiment [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<p>Name/ID by which the experiment is known at the source.</p>

<dt><a name="design"
>design</a></dt>

<dd>
<p>Design of the experiment including the numbers and types of observational units, traits, replicates, sampling plan, and analysis that are planned.</p>

<dt><a name="originator"
>originator</a></dt>

<dd>
<p>Name of the individual or program that are the originators of the experiment.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Subsystem</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Subsystem [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A subsystem is a set of functional roles that have been annotated simultaneously (e.g.,
the roles present in a specific pathway),
with an associated subsystem spreadsheet which encodes the fids in each genome that implement the functional roles in the subsystem.</p>

<p>Example:</p>

<pre>    get_entity_Subsystem -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Subsystem entity has the following relationship links:</p>

<dl>
<dt><a name="Describes_Variant"
>Describes Variant</a></dt>

<dd>
<dt><a name="Includes_Role"
>Includes Role</a></dt>

<dd>
<dt><a name="IsInClass_SubsystemClass"
>IsInClass SubsystemClass</a></dt>

<dd>
<dt><a name="IsRelevantTo_Diagram"
>IsRelevantTo Diagram</a></dt>

<dd>
<dt><a name="IsSubInstanceOf_Scenario"
>IsSubInstanceOf Scenario</a></dt>

<dd>
<dt><a name="WasProvidedBy_Source"
>WasProvidedBy Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Subsystem [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="version"
>version</a></dt>

<dd>
<p>version number for the subsystem. This value is incremented each time the subsystem is backed up.</p>

<dt><a name="curator"
>curator</a></dt>

<dd>
<p>name of the person currently in charge of the subsystem</p>

<dt><a name="notes"
>notes</a></dt>

<dd>
<p>descriptive notes about the subsystem</p>

<dt><a name="description"
>description</a></dt>

<dd>
<p>description of the subsystem&#39;s function in the cell</p>

<dt><a name="usable"
>usable</a></dt>

<dd>
<p>TRUE if this is a usable subsystem, else FALSE. An unusable subsystem is one that is experimental or is of such low quality that it can negatively affect analysis.</p>

<dt><a name="private"
>private</a></dt>

<dd>
<p>TRUE if this is a private subsystem, else FALSE. A private subsystem has valid data, but is not considered ready for general distribution.</p>

<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<p>TRUE if this is a clustering-based subsystem, else FALSE. A clustering-based subsystem is one in which there is functional-coupling evidence that genes belong together, but we do not yet know what they do.</p>

<dt><a name="experimental"
>experimental</a></dt>

<dd>
<p>TRUE if this is an experimental subsystem, else FALSE. An experimental subsystem is designed for investigation and is not yet ready to be used in comparative analysis and annotation.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_SubsystemClass</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_SubsystemClass [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Subsystem classes impose a hierarchical organization on the subsystems.</p>

<p>Example:</p>

<pre>    get_entity_SubsystemClass -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SubsystemClass entity has the following relationship links:</p>

<dl>
<dt><a name="IsClassFor_Subsystem"
>IsClassFor Subsystem</a></dt>

<dd>
<dt><a name="IsSubclassOf_SubsystemClass"
>IsSubclassOf SubsystemClass</a></dt>

<dd>
<dt><a name="IsSuperclassOf_SubsystemClass"
>IsSuperclassOf SubsystemClass</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_SubsystemClass [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_TaxonomicGrouping</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_TaxonomicGrouping [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>We associate with most genomes a &#34;taxonomy&#34; based on the NCBI taxonomy.
This includes,
for each genome,
a list of ever larger taxonomic groups.
The groups are stored as instances of this entity,
and chained together by the IsGroupFor relationship.</p>

<p>Example:</p>

<pre>    get_entity_TaxonomicGrouping -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The TaxonomicGrouping entity has the following relationship links:</p>

<dl>
<dt><a name="IsGroupFor_TaxonomicGrouping"
>IsGroupFor TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsInGroup_TaxonomicGrouping"
>IsInGroup TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsRepresentedBy_ObservationalUnit"
>IsRepresentedBy ObservationalUnit</a></dt>

<dd>
<dt><a name="IsTaxonomyOf_Genome"
>IsTaxonomyOf Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_TaxonomicGrouping [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="domain"
>domain</a></dt>

<dd>
<p>TRUE if this is a domain grouping, else FALSE.</p>

<dt><a name="hidden"
>hidden</a></dt>

<dd>
<p>TRUE if this is a hidden grouping, else FALSE. Hidden groupings are not typically shown in a lineage list.</p>

<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<p>Primary scientific name for this grouping. This is the name used when displaying a taxonomy.</p>

<dt><a name="alias"
>alias</a></dt>

<dd>
<p>Alternate name for this grouping. A grouping may have many alternate names. The scientific name should also be in this list.</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_TimeSeries</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_TimeSeries [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A TimeSeries provides a means to specify a series of experimental data over time by ordering multiple ExperimentalUnits.</p>

<p>Example:</p>

<pre>    get_entity_TimeSeries -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The TimeSeries entity has the following relationship links:</p>

<dl>
<dt><a name="OrdersExperimentalUnit_ExperimentalUnit"
>OrdersExperimentalUnit ExperimentalUnit</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_TimeSeries [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>The ID of the time series used by the data source.</p>

<dt><a name="name"
>name</a></dt>

<dd>
<p>The name of this time series, if any.</p>

<dt><a name="comments"
>comments</a></dt>

<dd>
<p>Any comments regarding this time series.</p>

<dt><a name="timeUnits"
>timeUnits</a></dt>

<dd>
<p>The units of time for this time series, e.g. &#39;seconds&#39;, &#39;hours&#39;, or more abstractly, &#39;number of times culture grown to saturation.&#39;</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Trait</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Trait [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A Trait is a phenotypic quality that can be measured or observed for an observational unit.
Examples include height,
sugar content,
color,
or cold tolerance.</p>

<p>Example:</p>

<pre>    get_entity_Trait -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Trait entity has the following relationship links:</p>

<dl>
<dt><a name="Impacts_Contig"
>Impacts Contig</a></dt>

<dd>
<dt><a name="Measures_ObservationalUnit"
>Measures ObservationalUnit</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Trait [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="trait_name"
>trait_name</a></dt>

<dd>
<p>Text name or description of the trait</p>

<dt><a name="unit_of_measure"
>unit_of_measure</a></dt>

<dd>
<p>The units of measure used when determining this trait. If multiple units of measure are applicable, each has its own row in the database.</p>

<dt><a name="TO_ID"
>TO_ID</a></dt>

<dd>
<p>Trait Ontology term ID (http://www.gramene.org/plant-ontology/)</p>

<dt><a name="protocol"
>protocol</a></dt>

<dd>
<p>A thorough description of how the trait was collected, and if a rating, the minimum and maximum values</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Tree</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Tree [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A tree describes how the sequences in an alignment relate to each other.
Most trees are phylogenetic,
but some may be based on taxonomy or gene content.</p>

<p>Example:</p>

<pre>    get_entity_Tree -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Tree entity has the following relationship links:</p>

<dl>
<dt><a name="HasNodeAttribute_TreeNodeAttribute"
>HasNodeAttribute TreeNodeAttribute</a></dt>

<dd>
<dt><a name="HasTreeAttribute_TreeAttribute"
>HasTreeAttribute TreeAttribute</a></dt>

<dd>
<dt><a name="IsBuiltFromAlignment_Alignment"
>IsBuiltFromAlignment Alignment</a></dt>

<dd>
<dt><a name="IsModificationOfTree_Tree"
>IsModificationOfTree Tree</a></dt>

<dd>
<dt><a name="IsModifiedToBuildTree_Tree"
>IsModifiedToBuildTree Tree</a></dt>

<dd>
<dt><a name="IsSupersededByTree_Tree"
>IsSupersededByTree Tree</a></dt>

<dd>
<dt><a name="IsTreeFrom_Source"
>IsTreeFrom Source</a></dt>

<dd>
<dt><a name="SupersedesTree_Tree"
>SupersedesTree Tree</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Tree [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="status"
>status</a></dt>

<dd>
<p>status of the tree, currently either [i]active[/i], [i]superseded[/i], or [i]bad[/i]</p>

<dt><a name="data_type"
>data_type</a></dt>

<dd>
<p>type of data the tree was built from, usually [i]sequence_alignment[/i]</p>

<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<p>date and time the tree was loaded</p>

<dt><a name="method"
>method</a></dt>

<dd>
<p>name of the primary software package or script used to construct the tree</p>

<dt><a name="parameters"
>parameters</a></dt>

<dd>
<p>non-default parameters used as input to the software package or script indicated in the method attribute</p>

<dt><a name="protocol"
>protocol</a></dt>

<dd>
<p>description of the steps taken to construct the tree, or a reference to an external pipeline</p>

<dt><a name="source_id"
>source_id</a></dt>

<dd>
<p>ID of this tree in the source database</p>

<dt><a name="newick"
>newick</a></dt>

<dd>
<p>NEWICK format string containing the structure of the tree</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_TreeAttribute</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_TreeAttribute [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This entity represents an attribute type that can be assigned to a tree.
The attribute values are stored in the relationships to the target.
The key is the attribute name.</p>

<p>Example:</p>

<pre>    get_entity_TreeAttribute -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The TreeAttribute entity has the following relationship links:</p>

<dl>
<dt><a name="DescribesTree_Tree"
>DescribesTree Tree</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_TreeAttribute [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_TreeNodeAttribute</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_TreeNodeAttribute [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This entity represents an attribute type that can be assigned to a node.
The attribute values are stored in the relationships to the target.
The key is the attribute name.</p>

<p>Example:</p>

<pre>    get_entity_TreeNodeAttribute -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The TreeNodeAttribute entity has the following relationship links:</p>

<dl>
<dt><a name="DescribesTreeNode_Tree"
>DescribesTreeNode Tree</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_TreeNodeAttribute [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_entity_Variant</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_entity_Variant [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Each subsystem may include the designation of distinct variants.
Thus,
there may be three closely-related,
but distinguishable forms of histidine degradation.
Each form would be called a &#34;variant&#34;,
with an associated code,
and all genomes implementing a specific variant can easily be accessed.</p>

<p>Example:</p>

<pre>    get_entity_Variant -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the entity.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Variant entity has the following relationship links:</p>

<dl>
<dt><a name="IsDescribedBy_Subsystem"
>IsDescribedBy Subsystem</a></dt>

<dd>
<dt><a name="IsImplementedBy_SSRow"
>IsImplementedBy SSRow</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_entity_Variant [arguments] &#60; ids &#62; table.with.fields.added</p>

<pre>    -a              Return all available fields.
    -c num          Select the identifier from column num.
    -i filename     Use filename rather than stdin for input.
    --fields list   Choose a set of fields to return. List is a comma-separated list of strings.
    -a              Return all available fields.
    --show-fields   List the available fields.</pre>

<p>The following fields are available:</p>

<dl>
<dt><a name="role_rule"
>role_rule</a></dt>

<dd>
<p>a space-delimited list of role IDs, in alphabetical order, that represents a possible list of non-auxiliary roles applicable to this variant. The roles are identified by their abbreviations. A variant may have multiple role rules.</p>

<dt><a name="code"
>code</a></dt>

<dd>
<p>the variant code all by itself</p>

<dt><a name="type"
>type</a></dt>

<dd>
<p>variant type indicating the quality of the subsystem support. A type of &#34;vacant&#34; means that the subsystem does not appear to be implemented by the variant. A type of &#34;incomplete&#34; means that the subsystem appears to be missing many reactions. In all other cases, the type is &#34;normal&#34;.</p>

<dt><a name="comment"
>comment</a></dt>

<dd>
<p>commentary text about the variant</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_AffectsLevelOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_AffectsLevelOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship indicates the expression level of an atomic regulon for a given experiment.</p>

<p>Example:</p>

<pre>    get_relationship_AffectsLevelOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_AffectsLevelOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="level"
>level</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the AtomicRegulon entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Aligned</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Aligned [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an alignment to the database from which it was generated.</p>

<p>Example:</p>

<pre>    get_relationship_Aligned -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Aligned [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Alignment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Annotates</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Annotates [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a feature to its annotations.
A feature may have multiple annotations,
but an annotation belongs to only one feature.</p>

<p>Example:</p>

<pre>    get_relationship_Annotates -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Annotates [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Annotation entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="annotator"
>annotator</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>

<dd>
<dt><a name="annotation_time"
>annotation_time</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_AreCodonsFor</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_AreCodonsFor [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a genome to the various codon usage records for it.</p>

<p>Example:</p>

<pre>    get_relationship_AreCodonsFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_AreCodonsFor [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the CodonUsage entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="frequencies"
>frequencies</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="subtype"
>subtype</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_AssertsFunctionFor</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_AssertsFunctionFor [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Sources (users) can make assertions about protein sequence function.
The assertion is associated with an external identifier.</p>

<p>Example:</p>

<pre>    get_relationship_AssertsFunctionFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_AssertsFunctionFor [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="external_id"
>external_id</a></dt>

<dd>
<dt><a name="organism"
>organism</a></dt>

<dd>
<dt><a name="gi_number"
>gi_number</a></dt>

<dd>
<dt><a name="release_date"
>release_date</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_ComponentOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_ComponentOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship defines the subcompounds that make up a compound.
For example,
CoCl2-6H2O is made up of 1 Co2+,
2 Cl-,
and 6 H2O.</p>

<p>Example:</p>

<pre>    get_relationship_ComponentOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_ComponentOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="molar_ratio"
>molar_ratio</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_CompoundMeasuredBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_CompoundMeasuredBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Denotes the compound that a measurement quantifies.</p>

<p>Example:</p>

<pre>    get_relationship_CompoundMeasuredBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_CompoundMeasuredBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Measurement entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>

<dd>
<dt><a name="mean"
>mean</a></dt>

<dd>
<dt><a name="median"
>median</a></dt>

<dd>
<dt><a name="stddev"
>stddev</a></dt>

<dd>
<dt><a name="N"
>N</a></dt>

<dd>
<dt><a name="p_value"
>p_value</a></dt>

<dd>
<dt><a name="Z_score"
>Z_score</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Comprises</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Comprises [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a biomass composition reaction to the compounds specified as contained in the biomass.</p>

<p>Example:</p>

<pre>    get_relationship_Comprises -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Comprises [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the CompoundInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="coefficient"
>coefficient</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Biomass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="dna"
>dna</a></dt>

<dd>
<dt><a name="protein"
>protein</a></dt>

<dd>
<dt><a name="cell_wall"
>cell_wall</a></dt>

<dd>
<dt><a name="lipid"
>lipid</a></dt>

<dd>
<dt><a name="cofactor"
>cofactor</a></dt>

<dd>
<dt><a name="energy"
>energy</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Concerns</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Concerns [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a publication to the protein sequences it describes.</p>

<p>Example:</p>

<pre>    get_relationship_Concerns -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Concerns [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Publication entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="link"
>link</a></dt>

<dd>
<dt><a name="pubdate"
>pubdate</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_ConsistsOfCompounds</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_ConsistsOfCompounds [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship defines the subcompounds that make up a compound.
For example,
CoCl2-6H2O is made up of 1 Co2+,
2 Cl-,
and 6 H2O.</p>

<p>Example:</p>

<pre>    get_relationship_ConsistsOfCompounds -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_ConsistsOfCompounds [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="molar_ratio"
>molar_ratio</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Contains</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Contains [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a subsystem spreadsheet cell to the features that occur in it.
A feature may occur in many machine roles and a machine role may contain many features.
The subsystem annotation process is essentially the maintenance of this relationship.</p>

<p>Example:</p>

<pre>    get_relationship_Contains -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Contains [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSCell entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_ContainsAlignedDNA</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_ContainsAlignedDNA [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a nucleotide alignment row to the contig sequences from which its components are formed.</p>

<p>Example:</p>

<pre>    get_relationship_ContainsAlignedDNA -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_ContainsAlignedDNA [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the AlignmentRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="row_number"
>row_number</a></dt>

<dd>
<dt><a name="row_id"
>row_id</a></dt>

<dd>
<dt><a name="row_description"
>row_description</a></dt>

<dd>
<dt><a name="n_components"
>n_components</a></dt>

<dd>
<dt><a name="beg_pos_aln"
>beg_pos_aln</a></dt>

<dd>
<dt><a name="end_pos_aln"
>end_pos_aln</a></dt>

<dd>
<dt><a name="md5_of_ungapped_sequence"
>md5_of_ungapped_sequence</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="index_in_concatenation"
>index_in_concatenation</a></dt>

<dd>
<dt><a name="beg_pos_in_parent"
>beg_pos_in_parent</a></dt>

<dd>
<dt><a name="end_pos_in_parent"
>end_pos_in_parent</a></dt>

<dd>
<dt><a name="parent_seq_len"
>parent_seq_len</a></dt>

<dd>
<dt><a name="beg_pos_aln"
>beg_pos_aln</a></dt>

<dd>
<dt><a name="end_pos_aln"
>end_pos_aln</a></dt>

<dd>
<dt><a name="kb_feature_id"
>kb_feature_id</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ContigSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="length"
>length</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_ContainsAlignedProtein</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_ContainsAlignedProtein [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a protein alignment row to the protein sequences from which its components are formed.</p>

<p>Example:</p>

<pre>    get_relationship_ContainsAlignedProtein -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_ContainsAlignedProtein [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the AlignmentRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="row_number"
>row_number</a></dt>

<dd>
<dt><a name="row_id"
>row_id</a></dt>

<dd>
<dt><a name="row_description"
>row_description</a></dt>

<dd>
<dt><a name="n_components"
>n_components</a></dt>

<dd>
<dt><a name="beg_pos_aln"
>beg_pos_aln</a></dt>

<dd>
<dt><a name="end_pos_aln"
>end_pos_aln</a></dt>

<dd>
<dt><a name="md5_of_ungapped_sequence"
>md5_of_ungapped_sequence</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="index_in_concatenation"
>index_in_concatenation</a></dt>

<dd>
<dt><a name="beg_pos_in_parent"
>beg_pos_in_parent</a></dt>

<dd>
<dt><a name="end_pos_in_parent"
>end_pos_in_parent</a></dt>

<dd>
<dt><a name="parent_seq_len"
>parent_seq_len</a></dt>

<dd>
<dt><a name="beg_pos_aln"
>beg_pos_aln</a></dt>

<dd>
<dt><a name="end_pos_aln"
>end_pos_aln</a></dt>

<dd>
<dt><a name="kb_feature_id"
>kb_feature_id</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_ContainsExperimentalUnit</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_ContainsExperimentalUnit [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Experimental units may be collected into groups,
such as assay plates.
This relationship describes which experimenal units belong to which groups.</p>

<p>Example:</p>

<pre>    get_relationship_ContainsExperimentalUnit -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_ContainsExperimentalUnit [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentalUnitGroup entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="comments"
>comments</a></dt>

<dd>
<dt><a name="groupType"
>groupType</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="location"
>location</a></dt>

<dd>
<dt><a name="groupMeta"
>groupMeta</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Controls</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Controls [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a coregulated set to the features that are used as its transcription factors.</p>

<p>Example:</p>

<pre>    get_relationship_Controls -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Controls [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the CoregulatedSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="binding_location"
>binding_location</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_DefinedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_DefinedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship associates observational units with a genus,
species,
strain,
and/or variety that was the source material.</p>

<p>Example:</p>

<pre>    get_relationship_DefinedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_DefinedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ObservationalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="source_name2"
>source_name2</a></dt>

<dd>
<dt><a name="plant_id"
>plant_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the TaxonomicGrouping entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_DerivedFromGenome</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_DerivedFromGenome [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The GenomeParentOf relationship specifies the direct child strains of a specific genome.</p>

<p>Example:</p>

<pre>    get_relationship_DerivedFromGenome -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_DerivedFromGenome [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Strain entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="aggregateData"
>aggregateData</a></dt>

<dd>
<dt><a name="wildtype"
>wildtype</a></dt>

<dd>
<dt><a name="referenceStrain"
>referenceStrain</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_DerivedFromStrain</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_DerivedFromStrain [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The recursive StrainParentOf relationship organizes derived organisms into a tree based on parent/child relationships.</p>

<p>Example:</p>

<pre>    get_relationship_DerivedFromStrain -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_DerivedFromStrain [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Strain entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="aggregateData"
>aggregateData</a></dt>

<dd>
<dt><a name="wildtype"
>wildtype</a></dt>

<dd>
<dt><a name="referenceStrain"
>referenceStrain</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Strain entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="aggregateData"
>aggregateData</a></dt>

<dd>
<dt><a name="wildtype"
>wildtype</a></dt>

<dd>
<dt><a name="referenceStrain"
>referenceStrain</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Describes</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Describes [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a subsystem to the individual variants used to implement it.
Each variant contains a slightly different subset of the roles in the parent subsystem.</p>

<p>Example:</p>

<pre>    get_relationship_Describes -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Describes [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Variant entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="role_rule"
>role_rule</a></dt>

<dd>
<dt><a name="code"
>code</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_DescribesAlignment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_DescribesAlignment [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an alignment to its free-form attributes.</p>

<p>Example:</p>

<pre>    get_relationship_DescribesAlignment -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_DescribesAlignment [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the AlignmentAttribute entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Alignment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_DescribesMeasurement</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_DescribesMeasurement [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The DescribesMeasurement relationship specifies a description for a particular measurement.</p>

<p>Example:</p>

<pre>    get_relationship_DescribesMeasurement -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_DescribesMeasurement [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the MeasurementDescription entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="unitOfMeasure"
>unitOfMeasure</a></dt>

<dd>
<dt><a name="category"
>category</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Measurement entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>

<dd>
<dt><a name="mean"
>mean</a></dt>

<dd>
<dt><a name="median"
>median</a></dt>

<dd>
<dt><a name="stddev"
>stddev</a></dt>

<dd>
<dt><a name="N"
>N</a></dt>

<dd>
<dt><a name="p_value"
>p_value</a></dt>

<dd>
<dt><a name="Z_score"
>Z_score</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_DescribesTree</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_DescribesTree [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a tree to its free-form attributes.</p>

<p>Example:</p>

<pre>    get_relationship_DescribesTree -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_DescribesTree [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the TreeAttribute entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Tree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="data_type"
>data_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="newick"
>newick</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_DescribesTreeNode</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_DescribesTreeNode [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an tree to the free-form attributes of its nodes.</p>

<p>Example:</p>

<pre>    get_relationship_DescribesTreeNode -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_DescribesTreeNode [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the TreeNodeAttribute entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>

<dd>
<dt><a name="node_id"
>node_id</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Tree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="data_type"
>data_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="newick"
>newick</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Determines</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Determines [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A functional coupling evidence set exists because it has pairings in it,
and this relationship connects the evidence set to its constituent pairings.
A pairing cam belong to multiple evidence sets.</p>

<p>Example:</p>

<pre>    get_relationship_Determines -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Determines [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Pairing entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="inverted"
>inverted</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the PairSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="score"
>score</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_DeterminesFunctionOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_DeterminesFunctionOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an isofunctional family to the roles that make up its assigned function.</p>

<p>Example:</p>

<pre>    get_relationship_DeterminesFunctionOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_DeterminesFunctionOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="release"
>release</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>

<dd>
<dt><a name="alignment"
>alignment</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Displays</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Displays [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a diagram to its reactions.
A diagram shows multiple reactions,
and a reaction can be on many diagrams.</p>

<p>Example:</p>

<pre>    get_relationship_Displays -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Displays [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Diagram entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="location"
>location</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>

<dd>
<dt><a name="thermodynamic_reversibility"
>thermodynamic_reversibility</a></dt>

<dd>
<dt><a name="default_protons"
>default_protons</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Encompasses</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Encompasses [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a feature to a related feature; for example,
it would connect a gene to its constituent splice variants,
and the splice variants to their exons.</p>

<p>Example:</p>

<pre>    get_relationship_Encompasses -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Encompasses [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_EvaluatedIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_EvaluatedIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The EvaluatedIn relationship specifies the experimental units performed on a particular strain.</p>

<p>Example:</p>

<pre>    get_relationship_EvaluatedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_EvaluatedIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Strain entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="aggregateData"
>aggregateData</a></dt>

<dd>
<dt><a name="wildtype"
>wildtype</a></dt>

<dd>
<dt><a name="referenceStrain"
>referenceStrain</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_ExperimentPublishedIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_ExperimentPublishedIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The PublishedExperiment relationship describes where a particular experiment was published.</p>

<p>Example:</p>

<pre>    get_relationship_ExperimentPublishedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_ExperimentPublishedIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentMeta entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="startDate"
>startDate</a></dt>

<dd>
<dt><a name="comments"
>comments</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Publication entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="link"
>link</a></dt>

<dd>
<dt><a name="pubdate"
>pubdate</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_FeatureMeasuredBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_FeatureMeasuredBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Denotes the feature that a measurement quantifies.</p>

<p>Example:</p>

<pre>    get_relationship_FeatureMeasuredBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_FeatureMeasuredBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Measurement entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>

<dd>
<dt><a name="mean"
>mean</a></dt>

<dd>
<dt><a name="median"
>median</a></dt>

<dd>
<dt><a name="stddev"
>stddev</a></dt>

<dd>
<dt><a name="N"
>N</a></dt>

<dd>
<dt><a name="p_value"
>p_value</a></dt>

<dd>
<dt><a name="Z_score"
>Z_score</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Formulated</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Formulated [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a coregulated set to the source organization that originally computed it.</p>

<p>Example:</p>

<pre>    get_relationship_Formulated -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Formulated [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the CoregulatedSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="binding_location"
>binding_location</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_GeneratedLevelsFor</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_GeneratedLevelsFor [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an atomic regulon to a probe set from which experimental data was produced for its features.
It contains a vector of the expression levels.</p>

<p>Example:</p>

<pre>    get_relationship_GeneratedLevelsFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_GeneratedLevelsFor [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProbeSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="level_vector"
>level_vector</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the AtomicRegulon entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_GenomeParentOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_GenomeParentOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The GenomeParentOf relationship specifies the direct child strains of a specific genome.</p>

<p>Example:</p>

<pre>    get_relationship_GenomeParentOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_GenomeParentOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Strain entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="aggregateData"
>aggregateData</a></dt>

<dd>
<dt><a name="wildtype"
>wildtype</a></dt>

<dd>
<dt><a name="referenceStrain"
>referenceStrain</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_GroupedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_GroupedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Experimental units may be collected into groups,
such as assay plates.
This relationship describes which experimenal units belong to which groups.</p>

<p>Example:</p>

<pre>    get_relationship_GroupedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_GroupedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="location"
>location</a></dt>

<dd>
<dt><a name="groupMeta"
>groupMeta</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentalUnitGroup entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="comments"
>comments</a></dt>

<dd>
<dt><a name="groupType"
>groupType</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HadResultsProducedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HadResultsProducedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a probe set to a genome for which it was used to produce experimental results.
In general,
a probe set is used for only one genome and vice versa,
but this is not a requirement.</p>

<p>Example:</p>

<pre>    get_relationship_HadResultsProducedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HadResultsProducedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProbeSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasAlignmentAttribute</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasAlignmentAttribute [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an alignment to its free-form attributes.</p>

<p>Example:</p>

<pre>    get_relationship_HasAlignmentAttribute -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasAlignmentAttribute [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Alignment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the AlignmentAttribute entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasAsExemplar</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasAsExemplar [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship links a role to a feature that provides a typical example of how the role is implemented.</p>

<p>Example:</p>

<pre>    get_relationship_HasAsExemplar -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasAsExemplar [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasAsSequence</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasAsSequence [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a Contig as it occurs in a genome to the Contig Sequence that represents the physical DNA base pairs.
A contig sequence may represent many contigs,
but each contig has only one sequence.</p>

<p>Example:</p>

<pre>    get_relationship_HasAsSequence -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasAsSequence [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ContigSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="length"
>length</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasAssertedFunctionFrom</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasAssertedFunctionFrom [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Sources (users) can make assertions about protein sequence function.
The assertion is associated with an external identifier.</p>

<p>Example:</p>

<pre>    get_relationship_HasAssertedFunctionFrom -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasAssertedFunctionFrom [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="external_id"
>external_id</a></dt>

<dd>
<dt><a name="organism"
>organism</a></dt>

<dd>
<dt><a name="gi_number"
>gi_number</a></dt>

<dd>
<dt><a name="release_date"
>release_date</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasAsTerminus</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasAsTerminus [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A terminus for a scenario is a compound that acts as its input or output.
A compound can be the terminus for many scenarios,
and a scenario will have many termini.
The relationship attributes indicate whether the compound is an input to the scenario or an output.
In some cases,
there may be multiple alternative output groups.
This is also indicated by the attributes.</p>

<p>Example:</p>

<pre>    get_relationship_HasAsTerminus -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasAsTerminus [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Scenario entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="common_name"
>common_name</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="group_number"
>group_number</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasCompoundAliasFrom</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasCompoundAliasFrom [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a source (database or organization) with the compounds for which it has assigned names (aliases).
The alias itself is stored as intersection data.</p>

<p>Example:</p>

<pre>    get_relationship_HasCompoundAliasFrom -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasCompoundAliasFrom [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasCoregulationWith</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasCoregulationWith [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a feature with another feature in the same genome with which it appears to be coregulated as a result of expression data analysis.</p>

<p>Example:</p>

<pre>    get_relationship_HasCoregulationWith -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasCoregulationWith [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="coefficient"
>coefficient</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasEnvironment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasEnvironment [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The IsContextOf relationship describes the enviroment a subexperiment defined by an ExperimentalUnit was performed in.</p>

<p>Example:</p>

<pre>    get_relationship_HasEnvironment -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasEnvironment [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Environment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="temperature"
>temperature</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="oxygenConcentration"
>oxygenConcentration</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasExperimentalUnit</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasExperimentalUnit [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The HasExperimentalUnit relationship describes which ExperimentalUnits are part of a Experiment.</p>

<p>Example:</p>

<pre>    get_relationship_HasExperimentalUnit -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasExperimentalUnit [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentMeta entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="startDate"
>startDate</a></dt>

<dd>
<dt><a name="comments"
>comments</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasFunctional</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasFunctional [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a role with the features in which it plays a functional part.</p>

<p>Example:</p>

<pre>    get_relationship_HasFunctional -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasFunctional [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasIndicatedSignalFrom</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasIndicatedSignalFrom [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an experiment to a feature.
The feature expression levels inferred from the experimental results are stored here.</p>

<p>Example:</p>

<pre>    get_relationship_HasIndicatedSignalFrom -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasIndicatedSignalFrom [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="rma_value"
>rma_value</a></dt>

<dd>
<dt><a name="level"
>level</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasKnockoutIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasKnockoutIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The HasKnockoutIn relationship specifies the gene knockouts in a particular strain.</p>

<p>Example:</p>

<pre>    get_relationship_HasKnockoutIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasKnockoutIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Strain entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="aggregateData"
>aggregateData</a></dt>

<dd>
<dt><a name="wildtype"
>wildtype</a></dt>

<dd>
<dt><a name="referenceStrain"
>referenceStrain</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasLevelsFrom</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasLevelsFrom [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a feature to a probe set from which experimental data was produced for the feature.
It contains a vector of the expression levels.</p>

<p>Example:</p>

<pre>    get_relationship_HasLevelsFrom -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasLevelsFrom [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="level_vector"
>level_vector</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProbeSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasMeasurement</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasMeasurement [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The HasMeasurement relationship specifies a measurement(s) performed on a particular experimental unit.</p>

<p>Example:</p>

<pre>    get_relationship_HasMeasurement -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasMeasurement [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Measurement entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>

<dd>
<dt><a name="mean"
>mean</a></dt>

<dd>
<dt><a name="median"
>median</a></dt>

<dd>
<dt><a name="stddev"
>stddev</a></dt>

<dd>
<dt><a name="N"
>N</a></dt>

<dd>
<dt><a name="p_value"
>p_value</a></dt>

<dd>
<dt><a name="Z_score"
>Z_score</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasMedia</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasMedia [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The UsedIn relationship defines which media is used by an Environment.</p>

<p>Example:</p>

<pre>    get_relationship_HasMedia -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasMedia [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Environment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="temperature"
>temperature</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="oxygenConcentration"
>oxygenConcentration</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Media entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="is_minimal"
>is_minimal</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="solid"
>solid</a></dt>

<dd>
<dt><a name="is_defined"
>is_defined</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasMember</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasMember [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects each feature family to its constituent features.
A family always has many features,
and a single feature can be found in many families.</p>

<p>Example:</p>

<pre>    get_relationship_HasMember -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasMember [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="release"
>release</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>

<dd>
<dt><a name="alignment"
>alignment</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasNodeAttribute</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasNodeAttribute [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an tree to the free-form attributes of its nodes.</p>

<p>Example:</p>

<pre>    get_relationship_HasNodeAttribute -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasNodeAttribute [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Tree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="data_type"
>data_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="newick"
>newick</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>

<dd>
<dt><a name="node_id"
>node_id</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the TreeNodeAttribute entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasParameter</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasParameter [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship denotes which parameters each environment has,
as well as the value of the parameter.</p>

<p>Example:</p>

<pre>    get_relationship_HasParameter -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasParameter [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Environment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="temperature"
>temperature</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="oxygenConcentration"
>oxygenConcentration</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Parameter entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasParticipant</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasParticipant [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A scenario consists of many participant reactions that convert the input compounds to output compounds.
A single reaction may participate in many scenarios.</p>

<p>Example:</p>

<pre>    get_relationship_HasParticipant -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasParticipant [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Scenario entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="common_name"
>common_name</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>

<dd>
<dt><a name="thermodynamic_reversibility"
>thermodynamic_reversibility</a></dt>

<dd>
<dt><a name="default_protons"
>default_protons</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasPresenceOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasPresenceOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a media to the compounds that occur in it.
The intersection data describes how much of each compound can be found.</p>

<p>Example:</p>

<pre>    get_relationship_HasPresenceOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasPresenceOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Media entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="is_minimal"
>is_minimal</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="solid"
>solid</a></dt>

<dd>
<dt><a name="is_defined"
>is_defined</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="concentration"
>concentration</a></dt>

<dd>
<dt><a name="units"
>units</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasProteinMember</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasProteinMember [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects each feature family to its constituent protein sequences.
A family always has many protein sequences,
and a single sequence can be found in many families.</p>

<p>Example:</p>

<pre>    get_relationship_HasProteinMember -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasProteinMember [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="release"
>release</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>

<dd>
<dt><a name="alignment"
>alignment</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasReactionAliasFrom</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasReactionAliasFrom [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a source (database or organization) with the reactions for which it has assigned names (aliases).
The alias itself is stored as intersection data.</p>

<p>Example:</p>

<pre>    get_relationship_HasReactionAliasFrom -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasReactionAliasFrom [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>

<dd>
<dt><a name="thermodynamic_reversibility"
>thermodynamic_reversibility</a></dt>

<dd>
<dt><a name="default_protons"
>default_protons</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasRealLocationIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasRealLocationIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a specific instance of a compound in a model to the specific instance of the model subcellular location where the compound exists.</p>

<p>Example:</p>

<pre>    get_relationship_HasRealLocationIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasRealLocationIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the CompoundInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the LocationInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="index"
>index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasRepresentativeOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasRepresentativeOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a genome to the FIGfam protein families for which it has representative proteins.
This information can be computed from other relationships,
but it is provided explicitly to allow fast access to a genome&#39;s FIGfam profile.</p>

<p>Example:</p>

<pre>    get_relationship_HasRepresentativeOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasRepresentativeOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="release"
>release</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>

<dd>
<dt><a name="alignment"
>alignment</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasRequirementOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasRequirementOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a model to the instances of reactions that represent how the reactions occur in the model.</p>

<p>Example:</p>

<pre>    get_relationship_HasRequirementOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasRequirementOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Model entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ReactionInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="protons"
>protons</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasResultsFor</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasResultsFor [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a probe set to the experiments that were applied to it.</p>

<p>Example:</p>

<pre>    get_relationship_HasResultsFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasResultsFor [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProbeSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasResultsIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasResultsIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a probe set to the experiments that were applied to it.</p>

<p>Example:</p>

<pre>    get_relationship_HasResultsIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasResultsIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProbeSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasRole</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasRole [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a role to the machine roles that represent its appearance in a molecular machine.
A machine role has exactly one associated role,
but a role may be represented by many machine roles.</p>

<p>Example:</p>

<pre>    get_relationship_HasRole -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasRole [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSCell entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasSection</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasSection [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a contig&#39;s sequence to its DNA sequences.</p>

<p>Example:</p>

<pre>    get_relationship_HasSection -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasSection [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ContigSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="length"
>length</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ContigChunk entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasStep</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasStep [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a complex to the reactions it catalyzes.</p>

<p>Example:</p>

<pre>    get_relationship_HasStep -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasStep [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Complex entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>

<dd>
<dt><a name="thermodynamic_reversibility"
>thermodynamic_reversibility</a></dt>

<dd>
<dt><a name="default_protons"
>default_protons</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasTrait</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasTrait [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship contains the measurement values of a trait on a specific observational Unit</p>

<p>Example:</p>

<pre>    get_relationship_HasTrait -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasTrait [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ObservationalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="source_name2"
>source_name2</a></dt>

<dd>
<dt><a name="plant_id"
>plant_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>

<dd>
<dt><a name="statistic_type"
>statistic_type</a></dt>

<dd>
<dt><a name="measure_id"
>measure_id</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Trait entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="trait_name"
>trait_name</a></dt>

<dd>
<dt><a name="unit_of_measure"
>unit_of_measure</a></dt>

<dd>
<dt><a name="TO_ID"
>TO_ID</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasTreeAttribute</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasTreeAttribute [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a tree to its free-form attributes.</p>

<p>Example:</p>

<pre>    get_relationship_HasTreeAttribute -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasTreeAttribute [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Tree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="data_type"
>data_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="newick"
>newick</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the TreeAttribute entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasUnits</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasUnits [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship associates observational units with the geographic location where the unit is planted.</p>

<p>Example:</p>

<pre>    get_relationship_HasUnits -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasUnits [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Locality entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="city"
>city</a></dt>

<dd>
<dt><a name="state"
>state</a></dt>

<dd>
<dt><a name="country"
>country</a></dt>

<dd>
<dt><a name="origcty"
>origcty</a></dt>

<dd>
<dt><a name="elevation"
>elevation</a></dt>

<dd>
<dt><a name="latitude"
>latitude</a></dt>

<dd>
<dt><a name="longitude"
>longitude</a></dt>

<dd>
<dt><a name="lo_accession"
>lo_accession</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ObservationalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="source_name2"
>source_name2</a></dt>

<dd>
<dt><a name="plant_id"
>plant_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasUsage</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasUsage [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a specific compound in a model to the localized compound to which it corresponds.</p>

<p>Example:</p>

<pre>    get_relationship_HasUsage -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasUsage [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the LocalizedCompound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the CompoundInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasValueFor</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasValueFor [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an experiment to its attributes.
The attribute values are stored here.</p>

<p>Example:</p>

<pre>    get_relationship_HasValueFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasValueFor [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Attribute entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasValueIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasValueIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an experiment to its attributes.
The attribute values are stored here.</p>

<p>Example:</p>

<pre>    get_relationship_HasValueIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasValueIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Attribute entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_HasVariationIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_HasVariationIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship defines an observational unit&#39;s DNA variation from a contig in the reference genome.</p>

<p>Example:</p>

<pre>    get_relationship_HasVariationIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_HasVariationIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="position"
>position</a></dt>

<dd>
<dt><a name="len"
>len</a></dt>

<dd>
<dt><a name="data"
>data</a></dt>

<dd>
<dt><a name="data2"
>data2</a></dt>

<dd>
<dt><a name="quality"
>quality</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ObservationalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="source_name2"
>source_name2</a></dt>

<dd>
<dt><a name="plant_id"
>plant_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Impacts</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Impacts [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship contains the best scoring statistical correlations between measured traits and the responsible alleles.</p>

<p>Example:</p>

<pre>    get_relationship_Impacts -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Impacts [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Trait entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="trait_name"
>trait_name</a></dt>

<dd>
<dt><a name="unit_of_measure"
>unit_of_measure</a></dt>

<dd>
<dt><a name="TO_ID"
>TO_ID</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="rank"
>rank</a></dt>

<dd>
<dt><a name="pvalue"
>pvalue</a></dt>

<dd>
<dt><a name="position"
>position</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_ImplementedBasedOn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_ImplementedBasedOn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects features to reaction instances that exist because the feature is included in a model.</p>

<p>Example:</p>

<pre>    get_relationship_ImplementedBasedOn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_ImplementedBasedOn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ReactionInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="protons"
>protons</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Implements</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Implements [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a variant to the physical machines that implement it in the genomes.
A variant is implemented by many machines,
but a machine belongs to only one variant.</p>

<p>Example:</p>

<pre>    get_relationship_Implements -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Implements [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="curated"
>curated</a></dt>

<dd>
<dt><a name="region"
>region</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Variant entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="role_rule"
>role_rule</a></dt>

<dd>
<dt><a name="code"
>code</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_ImplementsReaction</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_ImplementsReaction [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects features to reaction instances that exist because the feature is included in a model.</p>

<p>Example:</p>

<pre>    get_relationship_ImplementsReaction -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_ImplementsReaction [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ReactionInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="protons"
>protons</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IncludedIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IncludedIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a environment to the compounds that occur in it.
The intersection data describes how much of each compound can be found.</p>

<p>Example:</p>

<pre>    get_relationship_IncludedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IncludedIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="concentration"
>concentration</a></dt>

<dd>
<dt><a name="units"
>units</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Environment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="temperature"
>temperature</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="oxygenConcentration"
>oxygenConcentration</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Includes</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Includes [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A subsystem is defined by its roles.
The subsystem&#39;s variants contain slightly different sets of roles,
but all of the roles in a variant must be connected to the parent subsystem by this relationship.
A subsystem always has at least one role,
and a role always belongs to at least one subsystem.</p>

<p>Example:</p>

<pre>    get_relationship_Includes -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Includes [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>

<dd>
<dt><a name="abbreviation"
>abbreviation</a></dt>

<dd>
<dt><a name="auxiliary"
>auxiliary</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IncludesAdditionalCompounds</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IncludesAdditionalCompounds [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a environment to the compounds that occur in it.
The intersection data describes how much of each compound can be found.</p>

<p>Example:</p>

<pre>    get_relationship_IncludesAdditionalCompounds -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IncludesAdditionalCompounds [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Environment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="temperature"
>temperature</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="oxygenConcentration"
>oxygenConcentration</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="concentration"
>concentration</a></dt>

<dd>
<dt><a name="units"
>units</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IncludesAlignmentRow</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IncludesAlignmentRow [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an alignment to its component rows.</p>

<p>Example:</p>

<pre>    get_relationship_IncludesAlignmentRow -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IncludesAlignmentRow [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Alignment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the AlignmentRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="row_number"
>row_number</a></dt>

<dd>
<dt><a name="row_id"
>row_id</a></dt>

<dd>
<dt><a name="row_description"
>row_description</a></dt>

<dd>
<dt><a name="n_components"
>n_components</a></dt>

<dd>
<dt><a name="beg_pos_aln"
>beg_pos_aln</a></dt>

<dd>
<dt><a name="end_pos_aln"
>end_pos_aln</a></dt>

<dd>
<dt><a name="md5_of_ungapped_sequence"
>md5_of_ungapped_sequence</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IncludesPart</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IncludesPart [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship associates observational units with the experiments that generated the data on them.</p>

<p>Example:</p>

<pre>    get_relationship_IncludesPart -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IncludesPart [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the StudyExperiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="design"
>design</a></dt>

<dd>
<dt><a name="originator"
>originator</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ObservationalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="source_name2"
>source_name2</a></dt>

<dd>
<dt><a name="plant_id"
>plant_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IncludesPartOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IncludesPartOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A Scenario overlaps a diagram when the diagram displays a portion of the reactions that make up the scenario.
A scenario may overlap many diagrams,
and a diagram may be include portions of many scenarios.</p>

<p>Example:</p>

<pre>    get_relationship_IncludesPartOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IncludesPartOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Diagram entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Scenario entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="common_name"
>common_name</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IncludesStrain</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IncludesStrain [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The EvaluatedIn relationship specifies the experimental units performed on a particular strain.</p>

<p>Example:</p>

<pre>    get_relationship_IncludesStrain -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IncludesStrain [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Strain entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="aggregateData"
>aggregateData</a></dt>

<dd>
<dt><a name="wildtype"
>wildtype</a></dt>

<dd>
<dt><a name="referenceStrain"
>referenceStrain</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IndicatedLevelsFor</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IndicatedLevelsFor [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a feature to a probe set from which experimental data was produced for the feature.
It contains a vector of the expression levels.</p>

<p>Example:</p>

<pre>    get_relationship_IndicatedLevelsFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IndicatedLevelsFor [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProbeSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="level_vector"
>level_vector</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IndicatesSignalFor</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IndicatesSignalFor [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an experiment to a feature.
The feature expression levels inferred from the experimental results are stored here.</p>

<p>Example:</p>

<pre>    get_relationship_IndicatesSignalFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IndicatesSignalFor [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="rma_value"
>rma_value</a></dt>

<dd>
<dt><a name="level"
>level</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Involves</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Involves [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a reaction to the specific localized compounds that participate in it.</p>

<p>Example:</p>

<pre>    get_relationship_Involves -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Involves [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>

<dd>
<dt><a name="thermodynamic_reversibility"
>thermodynamic_reversibility</a></dt>

<dd>
<dt><a name="default_protons"
>default_protons</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="coefficient"
>coefficient</a></dt>

<dd>
<dt><a name="cofactor"
>cofactor</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the LocalizedCompound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsAffectedIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsAffectedIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship indicates the expression level of an atomic regulon for a given experiment.</p>

<p>Example:</p>

<pre>    get_relationship_IsAffectedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsAffectedIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the AtomicRegulon entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="level"
>level</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsAlignedDNAComponentOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsAlignedDNAComponentOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a nucleotide alignment row to the contig sequences from which its components are formed.</p>

<p>Example:</p>

<pre>    get_relationship_IsAlignedDNAComponentOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsAlignedDNAComponentOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ContigSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="length"
>length</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="index_in_concatenation"
>index_in_concatenation</a></dt>

<dd>
<dt><a name="beg_pos_in_parent"
>beg_pos_in_parent</a></dt>

<dd>
<dt><a name="end_pos_in_parent"
>end_pos_in_parent</a></dt>

<dd>
<dt><a name="parent_seq_len"
>parent_seq_len</a></dt>

<dd>
<dt><a name="beg_pos_aln"
>beg_pos_aln</a></dt>

<dd>
<dt><a name="end_pos_aln"
>end_pos_aln</a></dt>

<dd>
<dt><a name="kb_feature_id"
>kb_feature_id</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the AlignmentRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="row_number"
>row_number</a></dt>

<dd>
<dt><a name="row_id"
>row_id</a></dt>

<dd>
<dt><a name="row_description"
>row_description</a></dt>

<dd>
<dt><a name="n_components"
>n_components</a></dt>

<dd>
<dt><a name="beg_pos_aln"
>beg_pos_aln</a></dt>

<dd>
<dt><a name="end_pos_aln"
>end_pos_aln</a></dt>

<dd>
<dt><a name="md5_of_ungapped_sequence"
>md5_of_ungapped_sequence</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsAlignedProteinComponentOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsAlignedProteinComponentOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a protein alignment row to the protein sequences from which its components are formed.</p>

<p>Example:</p>

<pre>    get_relationship_IsAlignedProteinComponentOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsAlignedProteinComponentOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="index_in_concatenation"
>index_in_concatenation</a></dt>

<dd>
<dt><a name="beg_pos_in_parent"
>beg_pos_in_parent</a></dt>

<dd>
<dt><a name="end_pos_in_parent"
>end_pos_in_parent</a></dt>

<dd>
<dt><a name="parent_seq_len"
>parent_seq_len</a></dt>

<dd>
<dt><a name="beg_pos_aln"
>beg_pos_aln</a></dt>

<dd>
<dt><a name="end_pos_aln"
>end_pos_aln</a></dt>

<dd>
<dt><a name="kb_feature_id"
>kb_feature_id</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the AlignmentRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="row_number"
>row_number</a></dt>

<dd>
<dt><a name="row_id"
>row_id</a></dt>

<dd>
<dt><a name="row_description"
>row_description</a></dt>

<dd>
<dt><a name="n_components"
>n_components</a></dt>

<dd>
<dt><a name="beg_pos_aln"
>beg_pos_aln</a></dt>

<dd>
<dt><a name="end_pos_aln"
>end_pos_aln</a></dt>

<dd>
<dt><a name="md5_of_ungapped_sequence"
>md5_of_ungapped_sequence</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsAlignmentRowIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsAlignmentRowIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an alignment to its component rows.</p>

<p>Example:</p>

<pre>    get_relationship_IsAlignmentRowIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsAlignmentRowIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the AlignmentRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="row_number"
>row_number</a></dt>

<dd>
<dt><a name="row_id"
>row_id</a></dt>

<dd>
<dt><a name="row_description"
>row_description</a></dt>

<dd>
<dt><a name="n_components"
>n_components</a></dt>

<dd>
<dt><a name="beg_pos_aln"
>beg_pos_aln</a></dt>

<dd>
<dt><a name="end_pos_aln"
>end_pos_aln</a></dt>

<dd>
<dt><a name="md5_of_ungapped_sequence"
>md5_of_ungapped_sequence</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Alignment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsAnnotatedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsAnnotatedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a feature to its annotations.
A feature may have multiple annotations,
but an annotation belongs to only one feature.</p>

<p>Example:</p>

<pre>    get_relationship_IsAnnotatedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsAnnotatedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Annotation entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="annotator"
>annotator</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>

<dd>
<dt><a name="annotation_time"
>annotation_time</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsARequirementOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsARequirementOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a model to the instances of reactions that represent how the reactions occur in the model.</p>

<p>Example:</p>

<pre>    get_relationship_IsARequirementOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsARequirementOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ReactionInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="protons"
>protons</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Model entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsAssayedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsAssayedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship associates each assay with the relevant experiments.</p>

<p>Example:</p>

<pre>    get_relationship_IsAssayedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsAssayedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the StudyExperiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="design"
>design</a></dt>

<dd>
<dt><a name="originator"
>originator</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Assay entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="assay_type"
>assay_type</a></dt>

<dd>
<dt><a name="assay_type_id"
>assay_type_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsAssayOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsAssayOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship associates each assay with the relevant experiments.</p>

<p>Example:</p>

<pre>    get_relationship_IsAssayOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsAssayOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Assay entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="assay_type"
>assay_type</a></dt>

<dd>
<dt><a name="assay_type_id"
>assay_type_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the StudyExperiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="design"
>design</a></dt>

<dd>
<dt><a name="originator"
>originator</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsATopicOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsATopicOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a publication to the protein sequences it describes.</p>

<p>Example:</p>

<pre>    get_relationship_IsATopicOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsATopicOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Publication entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="link"
>link</a></dt>

<dd>
<dt><a name="pubdate"
>pubdate</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsBuiltFromAlignment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsBuiltFromAlignment [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects each tree to the alignment from which it is built.
There is at most one.</p>

<p>Example:</p>

<pre>    get_relationship_IsBuiltFromAlignment -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsBuiltFromAlignment [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Tree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="data_type"
>data_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="newick"
>newick</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Alignment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsClassFor</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsClassFor [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects each subsystem class with the subsystems that belong to it.
A class can contain many subsystems,
but a subsystem is only in one class.
Some subsystems are not in any class,
but this is usually a temporary condition.</p>

<p>Example:</p>

<pre>    get_relationship_IsClassFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsClassFor [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the SubsystemClass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsCollectedInto</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsCollectedInto [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A genome belongs to only one genome set.
For each set,
this relationship marks the genome to be used as its representative.</p>

<p>Example:</p>

<pre>    get_relationship_IsCollectedInto -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsCollectedInto [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="representative"
>representative</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the OTU entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsCollectionOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsCollectionOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A genome belongs to only one genome set.
For each set,
this relationship marks the genome to be used as its representative.</p>

<p>Example:</p>

<pre>    get_relationship_IsCollectionOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsCollectionOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the OTU entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="representative"
>representative</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsComponentOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsComponentOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a genome to its constituent contigs.
Unlike contig sequences,
a contig belongs to only one genome.</p>

<p>Example:</p>

<pre>    get_relationship_IsComponentOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsComponentOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsComposedOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsComposedOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a genome to its constituent contigs.
Unlike contig sequences,
a contig belongs to only one genome.</p>

<p>Example:</p>

<pre>    get_relationship_IsComposedOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsComposedOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsComprisedOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsComprisedOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a biomass composition reaction to the compounds specified as contained in the biomass.</p>

<p>Example:</p>

<pre>    get_relationship_IsComprisedOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsComprisedOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Biomass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="dna"
>dna</a></dt>

<dd>
<dt><a name="protein"
>protein</a></dt>

<dd>
<dt><a name="cell_wall"
>cell_wall</a></dt>

<dd>
<dt><a name="lipid"
>lipid</a></dt>

<dd>
<dt><a name="cofactor"
>cofactor</a></dt>

<dd>
<dt><a name="energy"
>energy</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="coefficient"
>coefficient</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the CompoundInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsConfiguredBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsConfiguredBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a genome to the atomic regulons that describe its state.</p>

<p>Example:</p>

<pre>    get_relationship_IsConfiguredBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsConfiguredBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the AtomicRegulon entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsConsistentTo</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsConsistentTo [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a functional role to the EC numbers consistent with the chemistry described in the role.</p>

<p>Example:</p>

<pre>    get_relationship_IsConsistentTo -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsConsistentTo [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the EcNumber entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="obsolete"
>obsolete</a></dt>

<dd>
<dt><a name="replacedby"
>replacedby</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsConsistentWith</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsConsistentWith [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a functional role to the EC numbers consistent with the chemistry described in the role.</p>

<p>Example:</p>

<pre>    get_relationship_IsConsistentWith -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsConsistentWith [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the EcNumber entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="obsolete"
>obsolete</a></dt>

<dd>
<dt><a name="replacedby"
>replacedby</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsContainedIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsContainedIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a subsystem spreadsheet cell to the features that occur in it.
A feature may occur in many machine roles and a machine role may contain many features.
The subsystem annotation process is essentially the maintenance of this relationship.</p>

<p>Example:</p>

<pre>    get_relationship_IsContainedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsContainedIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSCell entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsContextOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsContextOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The IsContextOf relationship describes the enviroment a subexperiment defined by an ExperimentalUnit was performed in.</p>

<p>Example:</p>

<pre>    get_relationship_IsContextOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsContextOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Environment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="temperature"
>temperature</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="oxygenConcentration"
>oxygenConcentration</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsControlledUsing</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsControlledUsing [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a coregulated set to the features that are used as its transcription factors.</p>

<p>Example:</p>

<pre>    get_relationship_IsControlledUsing -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsControlledUsing [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the CoregulatedSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="binding_location"
>binding_location</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsCoregulatedWith</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsCoregulatedWith [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a feature with another feature in the same genome with which it appears to be coregulated as a result of expression data analysis.</p>

<p>Example:</p>

<pre>    get_relationship_IsCoregulatedWith -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsCoregulatedWith [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="coefficient"
>coefficient</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsCoupledTo</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsCoupledTo [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects two FIGfams that we believe to be related either because their members occur in proximity on chromosomes or because the members are expressed together.
Such a relationship is evidence the functions of the FIGfams are themselves related.
This relationship is commutative; only the instance in which the first FIGfam has a lower ID than the second is stored.</p>

<p>Example:</p>

<pre>    get_relationship_IsCoupledTo -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsCoupledTo [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="release"
>release</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>

<dd>
<dt><a name="alignment"
>alignment</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="co_occurrence_evidence"
>co_occurrence_evidence</a></dt>

<dd>
<dt><a name="co_expression_evidence"
>co_expression_evidence</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="release"
>release</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>

<dd>
<dt><a name="alignment"
>alignment</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsCoupledWith</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsCoupledWith [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects two FIGfams that we believe to be related either because their members occur in proximity on chromosomes or because the members are expressed together.
Such a relationship is evidence the functions of the FIGfams are themselves related.
This relationship is commutative; only the instance in which the first FIGfam has a lower ID than the second is stored.</p>

<p>Example:</p>

<pre>    get_relationship_IsCoupledWith -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsCoupledWith [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="release"
>release</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>

<dd>
<dt><a name="alignment"
>alignment</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="co_occurrence_evidence"
>co_occurrence_evidence</a></dt>

<dd>
<dt><a name="co_expression_evidence"
>co_expression_evidence</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="release"
>release</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>

<dd>
<dt><a name="alignment"
>alignment</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsDefinedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsDefinedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The DescribesMeasurement relationship specifies a description for a particular measurement.</p>

<p>Example:</p>

<pre>    get_relationship_IsDefinedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsDefinedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Measurement entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>

<dd>
<dt><a name="mean"
>mean</a></dt>

<dd>
<dt><a name="median"
>median</a></dt>

<dd>
<dt><a name="stddev"
>stddev</a></dt>

<dd>
<dt><a name="N"
>N</a></dt>

<dd>
<dt><a name="p_value"
>p_value</a></dt>

<dd>
<dt><a name="Z_score"
>Z_score</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the MeasurementDescription entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="unitOfMeasure"
>unitOfMeasure</a></dt>

<dd>
<dt><a name="category"
>category</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsDescribedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsDescribedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a subsystem to the individual variants used to implement it.
Each variant contains a slightly different subset of the roles in the parent subsystem.</p>

<p>Example:</p>

<pre>    get_relationship_IsDescribedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsDescribedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Variant entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="role_rule"
>role_rule</a></dt>

<dd>
<dt><a name="code"
>code</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsDeterminedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsDeterminedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A functional coupling evidence set exists because it has pairings in it,
and this relationship connects the evidence set to its constituent pairings.
A pairing cam belong to multiple evidence sets.</p>

<p>Example:</p>

<pre>    get_relationship_IsDeterminedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsDeterminedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the PairSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="score"
>score</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="inverted"
>inverted</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Pairing entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsDisplayedOn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsDisplayedOn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a diagram to its reactions.
A diagram shows multiple reactions,
and a reaction can be on many diagrams.</p>

<p>Example:</p>

<pre>    get_relationship_IsDisplayedOn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsDisplayedOn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>

<dd>
<dt><a name="thermodynamic_reversibility"
>thermodynamic_reversibility</a></dt>

<dd>
<dt><a name="default_protons"
>default_protons</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="location"
>location</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Diagram entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsDividedInto</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsDividedInto [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a model to its instances of subcellular locations that participate in the model.</p>

<p>Example:</p>

<pre>    get_relationship_IsDividedInto -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsDividedInto [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Model entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the LocationInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="index"
>index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsDivisionOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsDivisionOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a model to its instances of subcellular locations that participate in the model.</p>

<p>Example:</p>

<pre>    get_relationship_IsDivisionOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsDivisionOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the LocationInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="index"
>index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Model entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsEncompassedIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsEncompassedIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a feature to a related feature; for example,
it would connect a gene to its constituent splice variants,
and the splice variants to their exons.</p>

<p>Example:</p>

<pre>    get_relationship_IsEncompassedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsEncompassedIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsExecutedAs</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsExecutedAs [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship links a reaction to the way it is used in a model.</p>

<p>Example:</p>

<pre>    get_relationship_IsExecutedAs -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsExecutedAs [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>

<dd>
<dt><a name="thermodynamic_reversibility"
>thermodynamic_reversibility</a></dt>

<dd>
<dt><a name="default_protons"
>default_protons</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ReactionInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="protons"
>protons</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsExecutionOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsExecutionOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship links a reaction to the way it is used in a model.</p>

<p>Example:</p>

<pre>    get_relationship_IsExecutionOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsExecutionOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ReactionInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="protons"
>protons</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>

<dd>
<dt><a name="thermodynamic_reversibility"
>thermodynamic_reversibility</a></dt>

<dd>
<dt><a name="default_protons"
>default_protons</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsExemplarOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsExemplarOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship links a role to a feature that provides a typical example of how the role is implemented.</p>

<p>Example:</p>

<pre>    get_relationship_IsExemplarOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsExemplarOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsExperimentalUnitOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsExperimentalUnitOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The HasExperimentalUnit relationship describes which ExperimentalUnits are part of a Experiment.</p>

<p>Example:</p>

<pre>    get_relationship_IsExperimentalUnitOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsExperimentalUnitOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentMeta entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="startDate"
>startDate</a></dt>

<dd>
<dt><a name="comments"
>comments</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsFamilyFor</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsFamilyFor [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an isofunctional family to the roles that make up its assigned function.</p>

<p>Example:</p>

<pre>    get_relationship_IsFamilyFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsFamilyFor [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="release"
>release</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>

<dd>
<dt><a name="alignment"
>alignment</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsFormedInto</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsFormedInto [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects each feature to the atomic regulon to which it belongs.</p>

<p>Example:</p>

<pre>    get_relationship_IsFormedInto -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsFormedInto [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the AtomicRegulon entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsFormedOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsFormedOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects each feature to the atomic regulon to which it belongs.</p>

<p>Example:</p>

<pre>    get_relationship_IsFormedOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsFormedOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the AtomicRegulon entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsFunctionalIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsFunctionalIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a role with the features in which it plays a functional part.</p>

<p>Example:</p>

<pre>    get_relationship_IsFunctionalIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsFunctionalIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsGroupFor</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsGroupFor [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The recursive IsGroupFor relationship organizes taxonomic groupings into a hierarchy based on the standard organism taxonomy.</p>

<p>Example:</p>

<pre>    get_relationship_IsGroupFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsGroupFor [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the TaxonomicGrouping entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the TaxonomicGrouping entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsImpactedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsImpactedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship contains the best scoring statistical correlations between measured traits and the responsible alleles.</p>

<p>Example:</p>

<pre>    get_relationship_IsImpactedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsImpactedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="rank"
>rank</a></dt>

<dd>
<dt><a name="pvalue"
>pvalue</a></dt>

<dd>
<dt><a name="position"
>position</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Trait entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="trait_name"
>trait_name</a></dt>

<dd>
<dt><a name="unit_of_measure"
>unit_of_measure</a></dt>

<dd>
<dt><a name="TO_ID"
>TO_ID</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsImplementedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsImplementedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a variant to the physical machines that implement it in the genomes.
A variant is implemented by many machines,
but a machine belongs to only one variant.</p>

<p>Example:</p>

<pre>    get_relationship_IsImplementedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsImplementedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Variant entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="role_rule"
>role_rule</a></dt>

<dd>
<dt><a name="code"
>code</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="curated"
>curated</a></dt>

<dd>
<dt><a name="region"
>region</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsInClass</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsInClass [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects each subsystem class with the subsystems that belong to it.
A class can contain many subsystems,
but a subsystem is only in one class.
Some subsystems are not in any class,
but this is usually a temporary condition.</p>

<p>Example:</p>

<pre>    get_relationship_IsInClass -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsInClass [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SubsystemClass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsIncludedIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsIncludedIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A subsystem is defined by its roles.
The subsystem&#39;s variants contain slightly different sets of roles,
but all of the roles in a variant must be connected to the parent subsystem by this relationship.
A subsystem always has at least one role,
and a role always belongs to at least one subsystem.</p>

<p>Example:</p>

<pre>    get_relationship_IsIncludedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsIncludedIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>

<dd>
<dt><a name="abbreviation"
>abbreviation</a></dt>

<dd>
<dt><a name="auxiliary"
>auxiliary</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsInGroup</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsInGroup [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The recursive IsGroupFor relationship organizes taxonomic groupings into a hierarchy based on the standard organism taxonomy.</p>

<p>Example:</p>

<pre>    get_relationship_IsInGroup -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsInGroup [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the TaxonomicGrouping entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the TaxonomicGrouping entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsInPair</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsInPair [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A pairing contains exactly two protein sequences.
A protein sequence can belong to multiple pairings.
When going from a protein sequence to its pairings,
they are presented in alphabetical order by sequence key.</p>

<p>Example:</p>

<pre>    get_relationship_IsInPair -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsInPair [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Pairing entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsInstanceOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsInstanceOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a subcellular location to the instances of that location that occur in models.</p>

<p>Example:</p>

<pre>    get_relationship_IsInstanceOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsInstanceOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the LocationInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="index"
>index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Location entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsInstantiatedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsInstantiatedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a subcellular location to the instances of that location that occur in models.</p>

<p>Example:</p>

<pre>    get_relationship_IsInstantiatedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsInstantiatedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Location entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the LocationInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="index"
>index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsInTaxa</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsInTaxa [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A genome is assigned to a particular point in the taxonomy tree,
but not necessarily to a leaf node.
In some cases,
the exact species and strain is not available when inserting the genome,
so it is placed at the lowest node that probably contains the actual genome.</p>

<p>Example:</p>

<pre>    get_relationship_IsInTaxa -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsInTaxa [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the TaxonomicGrouping entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsInvolvedIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsInvolvedIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a reaction to the specific localized compounds that participate in it.</p>

<p>Example:</p>

<pre>    get_relationship_IsInvolvedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsInvolvedIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the LocalizedCompound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="coefficient"
>coefficient</a></dt>

<dd>
<dt><a name="cofactor"
>cofactor</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>

<dd>
<dt><a name="thermodynamic_reversibility"
>thermodynamic_reversibility</a></dt>

<dd>
<dt><a name="default_protons"
>default_protons</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsLocated</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsLocated [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship associates observational units with the geographic location where the unit is planted.</p>

<p>Example:</p>

<pre>    get_relationship_IsLocated -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsLocated [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ObservationalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="source_name2"
>source_name2</a></dt>

<dd>
<dt><a name="plant_id"
>plant_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Locality entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="city"
>city</a></dt>

<dd>
<dt><a name="state"
>state</a></dt>

<dd>
<dt><a name="country"
>country</a></dt>

<dd>
<dt><a name="origcty"
>origcty</a></dt>

<dd>
<dt><a name="elevation"
>elevation</a></dt>

<dd>
<dt><a name="latitude"
>latitude</a></dt>

<dd>
<dt><a name="longitude"
>longitude</a></dt>

<dd>
<dt><a name="lo_accession"
>lo_accession</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsLocatedIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsLocatedIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A feature is a set of DNA sequence fragments,
the location of which are specified by the fields of this relationship.
Most features are a single contiguous fragment,
so they are located in only one DNA sequence; however,
for search optimization reasons,
fragments have a maximum length,
so even a single contiguous feature may participate in this relationship multiple times.
Thus,
it is better to use the CDMI API methods to get feature positions and sequences as those methods rejoin the fragements for contiguous features.
A few features belong to multiple DNA sequences.
In that case,
however,
all the DNA sequences belong to the same genome.
A DNA sequence itself will frequently have thousands of features connected to it.</p>

<p>Example:</p>

<pre>    get_relationship_IsLocatedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsLocatedIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="ordinal"
>ordinal</a></dt>

<dd>
<dt><a name="begin"
>begin</a></dt>

<dd>
<dt><a name="len"
>len</a></dt>

<dd>
<dt><a name="dir"
>dir</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsLocusFor</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsLocusFor [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A feature is a set of DNA sequence fragments,
the location of which are specified by the fields of this relationship.
Most features are a single contiguous fragment,
so they are located in only one DNA sequence; however,
for search optimization reasons,
fragments have a maximum length,
so even a single contiguous feature may participate in this relationship multiple times.
Thus,
it is better to use the CDMI API methods to get feature positions and sequences as those methods rejoin the fragements for contiguous features.
A few features belong to multiple DNA sequences.
In that case,
however,
all the DNA sequences belong to the same genome.
A DNA sequence itself will frequently have thousands of features connected to it.</p>

<p>Example:</p>

<pre>    get_relationship_IsLocusFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsLocusFor [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="ordinal"
>ordinal</a></dt>

<dd>
<dt><a name="begin"
>begin</a></dt>

<dd>
<dt><a name="len"
>len</a></dt>

<dd>
<dt><a name="dir"
>dir</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsManagedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsManagedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a model to its associated biomass composition reactions.</p>

<p>Example:</p>

<pre>    get_relationship_IsManagedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsManagedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Biomass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="dna"
>dna</a></dt>

<dd>
<dt><a name="protein"
>protein</a></dt>

<dd>
<dt><a name="cell_wall"
>cell_wall</a></dt>

<dd>
<dt><a name="lipid"
>lipid</a></dt>

<dd>
<dt><a name="cofactor"
>cofactor</a></dt>

<dd>
<dt><a name="energy"
>energy</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Model entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsMeasurementMethodOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsMeasurementMethodOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The IsMeasurementMethodOf relationship describes which protocol was used to take a measurement.</p>

<p>Example:</p>

<pre>    get_relationship_IsMeasurementMethodOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsMeasurementMethodOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Protocol entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Measurement entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>

<dd>
<dt><a name="mean"
>mean</a></dt>

<dd>
<dt><a name="median"
>median</a></dt>

<dd>
<dt><a name="stddev"
>stddev</a></dt>

<dd>
<dt><a name="N"
>N</a></dt>

<dd>
<dt><a name="p_value"
>p_value</a></dt>

<dd>
<dt><a name="Z_score"
>Z_score</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsMeasureOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsMeasureOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The HasMeasurement relationship specifies a measurement(s) performed on a particular experimental unit.</p>

<p>Example:</p>

<pre>    get_relationship_IsMeasureOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsMeasureOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Measurement entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>

<dd>
<dt><a name="mean"
>mean</a></dt>

<dd>
<dt><a name="median"
>median</a></dt>

<dd>
<dt><a name="stddev"
>stddev</a></dt>

<dd>
<dt><a name="N"
>N</a></dt>

<dd>
<dt><a name="p_value"
>p_value</a></dt>

<dd>
<dt><a name="Z_score"
>Z_score</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsMemberOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsMemberOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects each feature family to its constituent features.
A family always has many features,
and a single feature can be found in many families.</p>

<p>Example:</p>

<pre>    get_relationship_IsMemberOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsMemberOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="release"
>release</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>

<dd>
<dt><a name="alignment"
>alignment</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsModeledBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsModeledBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A genome can be modeled by many different models,
but a model belongs to only one genome.</p>

<p>Example:</p>

<pre>    get_relationship_IsModeledBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsModeledBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Model entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsModificationOfAlignment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsModificationOfAlignment [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Relates an alignment to other alignments built from it.</p>

<p>Example:</p>

<pre>    get_relationship_IsModificationOfAlignment -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsModificationOfAlignment [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Alignment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="modification_type"
>modification_type</a></dt>

<dd>
<dt><a name="modification_value"
>modification_value</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Alignment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsModificationOfTree</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsModificationOfTree [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Relates a tree to other trees built from it.</p>

<p>Example:</p>

<pre>    get_relationship_IsModificationOfTree -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsModificationOfTree [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Tree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="data_type"
>data_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="newick"
>newick</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="modification_type"
>modification_type</a></dt>

<dd>
<dt><a name="modification_value"
>modification_value</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Tree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="data_type"
>data_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="newick"
>newick</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsModifiedToBuildAlignment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsModifiedToBuildAlignment [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Relates an alignment to other alignments built from it.</p>

<p>Example:</p>

<pre>    get_relationship_IsModifiedToBuildAlignment -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsModifiedToBuildAlignment [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Alignment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="modification_type"
>modification_type</a></dt>

<dd>
<dt><a name="modification_value"
>modification_value</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Alignment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsModifiedToBuildTree</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsModifiedToBuildTree [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Relates a tree to other trees built from it.</p>

<p>Example:</p>

<pre>    get_relationship_IsModifiedToBuildTree -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsModifiedToBuildTree [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Tree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="data_type"
>data_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="newick"
>newick</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="modification_type"
>modification_type</a></dt>

<dd>
<dt><a name="modification_value"
>modification_value</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Tree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="data_type"
>data_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="newick"
>newick</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsOwnedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsOwnedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a genome to the features it contains.
Though technically redundant (the information is available from the feature&#39;s contigs),
it simplifies the extremely common process of finding all features for a genome.</p>

<p>Example:</p>

<pre>    get_relationship_IsOwnedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsOwnedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsOwnerOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsOwnerOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a genome to the features it contains.
Though technically redundant (the information is available from the feature&#39;s contigs),
it simplifies the extremely common process of finding all features for a genome.</p>

<p>Example:</p>

<pre>    get_relationship_IsOwnerOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsOwnerOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsPairOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsPairOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A pairing contains exactly two protein sequences.
A protein sequence can belong to multiple pairings.
When going from a protein sequence to its pairings,
they are presented in alphabetical order by sequence key.</p>

<p>Example:</p>

<pre>    get_relationship_IsPairOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsPairOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Pairing entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsParticipatingAt</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsParticipatingAt [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a localized compound to the location in which it occurs during one or more reactions.</p>

<p>Example:</p>

<pre>    get_relationship_IsParticipatingAt -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsParticipatingAt [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Location entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the LocalizedCompound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsParticipationOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsParticipationOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a generic compound to a specific compound where subceullar location has been specified.</p>

<p>Example:</p>

<pre>    get_relationship_IsParticipationOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsParticipationOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the LocalizedCompound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsPartOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsPartOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship associates observational units with the experiments that generated the data on them.</p>

<p>Example:</p>

<pre>    get_relationship_IsPartOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsPartOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ObservationalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="source_name2"
>source_name2</a></dt>

<dd>
<dt><a name="plant_id"
>plant_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the StudyExperiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="design"
>design</a></dt>

<dd>
<dt><a name="originator"
>originator</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsPresentIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsPresentIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a media to the compounds that occur in it.
The intersection data describes how much of each compound can be found.</p>

<p>Example:</p>

<pre>    get_relationship_IsPresentIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsPresentIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="concentration"
>concentration</a></dt>

<dd>
<dt><a name="units"
>units</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Media entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="is_minimal"
>is_minimal</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="solid"
>solid</a></dt>

<dd>
<dt><a name="is_defined"
>is_defined</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsProteinFor</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsProteinFor [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a peg feature to the protein sequence it produces (if any).
Only peg features participate in this relationship.
A single protein sequence will frequently be produced by many features.</p>

<p>Example:</p>

<pre>    get_relationship_IsProteinFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsProteinFor [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsProteinMemberOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsProteinMemberOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects each feature family to its constituent protein sequences.
A family always has many protein sequences,
and a single sequence can be found in many families.</p>

<p>Example:</p>

<pre>    get_relationship_IsProteinMemberOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsProteinMemberOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="release"
>release</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>

<dd>
<dt><a name="alignment"
>alignment</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsReagentIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsReagentIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a compound instance to the reaction instance in which it is transformed.</p>

<p>Example:</p>

<pre>    get_relationship_IsReagentIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsReagentIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the CompoundInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="coefficient"
>coefficient</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ReactionInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="protons"
>protons</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsRealLocationOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsRealLocationOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a specific instance of a compound in a model to the specific instance of the model subcellular location where the compound exists.</p>

<p>Example:</p>

<pre>    get_relationship_IsRealLocationOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsRealLocationOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the LocationInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="index"
>index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the CompoundInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsReferencedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsReferencedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship associates each observational unit with the reference genome that it will be compared to.
All variations will be differences between the observational unit and the reference.</p>

<p>Example:</p>

<pre>    get_relationship_IsReferencedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsReferencedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ObservationalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="source_name2"
>source_name2</a></dt>

<dd>
<dt><a name="plant_id"
>plant_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsRegulatedIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsRegulatedIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a feature to the set of coregulated features.</p>

<p>Example:</p>

<pre>    get_relationship_IsRegulatedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsRegulatedIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the CoregulatedSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="binding_location"
>binding_location</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsRegulatedSetOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsRegulatedSetOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a feature to the set of coregulated features.</p>

<p>Example:</p>

<pre>    get_relationship_IsRegulatedSetOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsRegulatedSetOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the CoregulatedSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="binding_location"
>binding_location</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsRelevantFor</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsRelevantFor [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a diagram to the subsystems that are depicted on it.
Only diagrams which are useful in curating or annotation the subsystem are specified in this relationship.</p>

<p>Example:</p>

<pre>    get_relationship_IsRelevantFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsRelevantFor [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Diagram entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsRelevantTo</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsRelevantTo [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a diagram to the subsystems that are depicted on it.
Only diagrams which are useful in curating or annotation the subsystem are specified in this relationship.</p>

<p>Example:</p>

<pre>    get_relationship_IsRelevantTo -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsRelevantTo [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Diagram entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsRepresentedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsRepresentedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship associates observational units with a genus,
species,
strain,
and/or variety that was the source material.</p>

<p>Example:</p>

<pre>    get_relationship_IsRepresentedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsRepresentedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the TaxonomicGrouping entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ObservationalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="source_name2"
>source_name2</a></dt>

<dd>
<dt><a name="plant_id"
>plant_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsRepresentedIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsRepresentedIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a genome to the FIGfam protein families for which it has representative proteins.
This information can be computed from other relationships,
but it is provided explicitly to allow fast access to a genome&#39;s FIGfam profile.</p>

<p>Example:</p>

<pre>    get_relationship_IsRepresentedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsRepresentedIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Family entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="release"
>release</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>

<dd>
<dt><a name="alignment"
>alignment</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsRoleFor</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsRoleFor [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a subsystem spreadsheet row to its constituent spreadsheet cells.</p>

<p>Example:</p>

<pre>    get_relationship_IsRoleFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsRoleFor [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSCell entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="curated"
>curated</a></dt>

<dd>
<dt><a name="region"
>region</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsRoleOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsRoleOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a role to the machine roles that represent its appearance in a molecular machine.
A machine role has exactly one associated role,
but a role may be represented by many machine roles.</p>

<p>Example:</p>

<pre>    get_relationship_IsRoleOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsRoleOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSCell entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsRowOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsRowOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a subsystem spreadsheet row to its constituent spreadsheet cells.</p>

<p>Example:</p>

<pre>    get_relationship_IsRowOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsRowOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="curated"
>curated</a></dt>

<dd>
<dt><a name="region"
>region</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSCell entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsSectionOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsSectionOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a contig&#39;s sequence to its DNA sequences.</p>

<p>Example:</p>

<pre>    get_relationship_IsSectionOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsSectionOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ContigChunk entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ContigSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="length"
>length</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsSequenceOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsSequenceOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a Contig as it occurs in a genome to the Contig Sequence that represents the physical DNA base pairs.
A contig sequence may represent many contigs,
but each contig has only one sequence.</p>

<p>Example:</p>

<pre>    get_relationship_IsSequenceOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsSequenceOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ContigSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="length"
>length</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsShownOn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsShownOn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship indicates that a compound appears on a particular diagram.
The same compound can appear on many diagrams,
and a diagram always contains many compounds.</p>

<p>Example:</p>

<pre>    get_relationship_IsShownOn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsShownOn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="location"
>location</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Diagram entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsStepOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsStepOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a complex to the reactions it catalyzes.</p>

<p>Example:</p>

<pre>    get_relationship_IsStepOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsStepOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>

<dd>
<dt><a name="thermodynamic_reversibility"
>thermodynamic_reversibility</a></dt>

<dd>
<dt><a name="default_protons"
>default_protons</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Complex entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsSubclassOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsSubclassOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This is a recursive relationship that imposes a hierarchy on the subsystem classes.</p>

<p>Example:</p>

<pre>    get_relationship_IsSubclassOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsSubclassOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the SubsystemClass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SubsystemClass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsSubInstanceOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsSubInstanceOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a scenario to its subsystem it validates.
A scenario belongs to exactly one subsystem,
but a subsystem may have multiple scenarios.</p>

<p>Example:</p>

<pre>    get_relationship_IsSubInstanceOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsSubInstanceOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Scenario entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="common_name"
>common_name</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsSummarizedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsSummarizedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship describes the statistical frequencies of the most common alleles in various positions on the reference contig.</p>

<p>Example:</p>

<pre>    get_relationship_IsSummarizedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsSummarizedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="position"
>position</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the AlleleFrequency entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="position"
>position</a></dt>

<dd>
<dt><a name="minor_AF"
>minor_AF</a></dt>

<dd>
<dt><a name="minor_allele"
>minor_allele</a></dt>

<dd>
<dt><a name="major_AF"
>major_AF</a></dt>

<dd>
<dt><a name="major_allele"
>major_allele</a></dt>

<dd>
<dt><a name="obs_unit_count"
>obs_unit_count</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsSuperclassOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsSuperclassOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This is a recursive relationship that imposes a hierarchy on the subsystem classes.</p>

<p>Example:</p>

<pre>    get_relationship_IsSuperclassOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsSuperclassOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the SubsystemClass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SubsystemClass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsSupersededByAlignment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsSupersededByAlignment [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an alignment to the alignments it replaces.</p>

<p>Example:</p>

<pre>    get_relationship_IsSupersededByAlignment -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsSupersededByAlignment [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Alignment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="successor_type"
>successor_type</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Alignment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsSupersededByTree</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsSupersededByTree [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a tree to the trees it replaces.</p>

<p>Example:</p>

<pre>    get_relationship_IsSupersededByTree -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsSupersededByTree [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Tree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="data_type"
>data_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="newick"
>newick</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="successor_type"
>successor_type</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Tree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="data_type"
>data_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="newick"
>newick</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsTaxonomyOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsTaxonomyOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A genome is assigned to a particular point in the taxonomy tree,
but not necessarily to a leaf node.
In some cases,
the exact species and strain is not available when inserting the genome,
so it is placed at the lowest node that probably contains the actual genome.</p>

<p>Example:</p>

<pre>    get_relationship_IsTaxonomyOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsTaxonomyOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the TaxonomicGrouping entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsTerminusFor</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsTerminusFor [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A terminus for a scenario is a compound that acts as its input or output.
A compound can be the terminus for many scenarios,
and a scenario will have many termini.
The relationship attributes indicate whether the compound is an input to the scenario or an output.
In some cases,
there may be multiple alternative output groups.
This is also indicated by the attributes.</p>

<p>Example:</p>

<pre>    get_relationship_IsTerminusFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsTerminusFor [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="group_number"
>group_number</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Scenario entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="common_name"
>common_name</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsTimepointOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsTimepointOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Experimental units may be ordered into time series.
This relationship describes which experimenal units belong to which time series.</p>

<p>Example:</p>

<pre>    get_relationship_IsTimepointOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsTimepointOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="time"
>time</a></dt>

<dd>
<dt><a name="timeMeta"
>timeMeta</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the TimeSeries entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="comments"
>comments</a></dt>

<dd>
<dt><a name="timeUnits"
>timeUnits</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsTreeFrom</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsTreeFrom [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a tree to the source database from which it was generated.</p>

<p>Example:</p>

<pre>    get_relationship_IsTreeFrom -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsTreeFrom [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Tree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="data_type"
>data_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="newick"
>newick</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsTriggeredBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsTriggeredBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This connects a complex to the roles that work together to form the complex.</p>

<p>Example:</p>

<pre>    get_relationship_IsTriggeredBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsTriggeredBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Complex entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="optional"
>optional</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="triggering"
>triggering</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsUsageOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsUsageOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a specific compound in a model to the localized compound to which it corresponds.</p>

<p>Example:</p>

<pre>    get_relationship_IsUsageOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsUsageOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the CompoundInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the LocalizedCompound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsUsedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsUsedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a genome to the machines that form its metabolic pathways.
A genome can use many machines,
but a machine is used by exactly one genome.</p>

<p>Example:</p>

<pre>    get_relationship_IsUsedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsUsedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="curated"
>curated</a></dt>

<dd>
<dt><a name="region"
>region</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsUsedToBuildTree</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsUsedToBuildTree [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects each tree to the alignment from which it is built.
There is at most one.</p>

<p>Example:</p>

<pre>    get_relationship_IsUsedToBuildTree -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsUsedToBuildTree [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Alignment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Tree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="data_type"
>data_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="newick"
>newick</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsUtilizedIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsUtilizedIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an experiment to the media in which the experiment took place.</p>

<p>Example:</p>

<pre>    get_relationship_IsUtilizedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsUtilizedIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Media entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="is_minimal"
>is_minimal</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="solid"
>solid</a></dt>

<dd>
<dt><a name="is_defined"
>is_defined</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_IsVariedIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_IsVariedIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship defines an observational unit&#39;s DNA variation from a contig in the reference genome.</p>

<p>Example:</p>

<pre>    get_relationship_IsVariedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_IsVariedIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ObservationalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="source_name2"
>source_name2</a></dt>

<dd>
<dt><a name="plant_id"
>plant_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="position"
>position</a></dt>

<dd>
<dt><a name="len"
>len</a></dt>

<dd>
<dt><a name="data"
>data</a></dt>

<dd>
<dt><a name="data2"
>data2</a></dt>

<dd>
<dt><a name="quality"
>quality</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_KnockedOutIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_KnockedOutIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The HasKnockoutIn relationship specifies the gene knockouts in a particular strain.</p>

<p>Example:</p>

<pre>    get_relationship_KnockedOutIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_KnockedOutIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Strain entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="aggregateData"
>aggregateData</a></dt>

<dd>
<dt><a name="wildtype"
>wildtype</a></dt>

<dd>
<dt><a name="referenceStrain"
>referenceStrain</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Manages</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Manages [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a model to its associated biomass composition reactions.</p>

<p>Example:</p>

<pre>    get_relationship_Manages -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Manages [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Model entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Biomass entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="dna"
>dna</a></dt>

<dd>
<dt><a name="protein"
>protein</a></dt>

<dd>
<dt><a name="cell_wall"
>cell_wall</a></dt>

<dd>
<dt><a name="lipid"
>lipid</a></dt>

<dd>
<dt><a name="cofactor"
>cofactor</a></dt>

<dd>
<dt><a name="energy"
>energy</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Measures</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Measures [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship contains the measurement values of a trait on a specific observational Unit</p>

<p>Example:</p>

<pre>    get_relationship_Measures -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Measures [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Trait entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="trait_name"
>trait_name</a></dt>

<dd>
<dt><a name="unit_of_measure"
>unit_of_measure</a></dt>

<dd>
<dt><a name="TO_ID"
>TO_ID</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>

<dd>
<dt><a name="statistic_type"
>statistic_type</a></dt>

<dd>
<dt><a name="measure_id"
>measure_id</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ObservationalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="source_name2"
>source_name2</a></dt>

<dd>
<dt><a name="plant_id"
>plant_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_MeasuresCompound</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_MeasuresCompound [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Denotes the compound that a measurement quantifies.</p>

<p>Example:</p>

<pre>    get_relationship_MeasuresCompound -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_MeasuresCompound [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Measurement entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>

<dd>
<dt><a name="mean"
>mean</a></dt>

<dd>
<dt><a name="median"
>median</a></dt>

<dd>
<dt><a name="stddev"
>stddev</a></dt>

<dd>
<dt><a name="N"
>N</a></dt>

<dd>
<dt><a name="p_value"
>p_value</a></dt>

<dd>
<dt><a name="Z_score"
>Z_score</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_MeasuresFeature</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_MeasuresFeature [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Denotes the feature that a measurement quantifies.</p>

<p>Example:</p>

<pre>    get_relationship_MeasuresFeature -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_MeasuresFeature [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Measurement entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>

<dd>
<dt><a name="mean"
>mean</a></dt>

<dd>
<dt><a name="median"
>median</a></dt>

<dd>
<dt><a name="stddev"
>stddev</a></dt>

<dd>
<dt><a name="N"
>N</a></dt>

<dd>
<dt><a name="p_value"
>p_value</a></dt>

<dd>
<dt><a name="Z_score"
>Z_score</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Models</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Models [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A genome can be modeled by many different models,
but a model belongs to only one genome.</p>

<p>Example:</p>

<pre>    get_relationship_Models -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Models [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Model entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_OfEnvironment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_OfEnvironment [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship denotes which parameters each environment has,
as well as the value of the parameter.</p>

<p>Example:</p>

<pre>    get_relationship_OfEnvironment -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_OfEnvironment [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Parameter entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Environment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="temperature"
>temperature</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="oxygenConcentration"
>oxygenConcentration</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_OperatesIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_OperatesIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an experiment to the media in which the experiment took place.</p>

<p>Example:</p>

<pre>    get_relationship_OperatesIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_OperatesIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Experiment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source"
>source</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Media entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="is_minimal"
>is_minimal</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="solid"
>solid</a></dt>

<dd>
<dt><a name="is_defined"
>is_defined</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_OrdersExperimentalUnit</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_OrdersExperimentalUnit [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Experimental units may be ordered into time series.
This relationship describes which experimenal units belong to which time series.</p>

<p>Example:</p>

<pre>    get_relationship_OrdersExperimentalUnit -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_OrdersExperimentalUnit [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the TimeSeries entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="comments"
>comments</a></dt>

<dd>
<dt><a name="timeUnits"
>timeUnits</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="time"
>time</a></dt>

<dd>
<dt><a name="timeMeta"
>timeMeta</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Overlaps</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Overlaps [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A Scenario overlaps a diagram when the diagram displays a portion of the reactions that make up the scenario.
A scenario may overlap many diagrams,
and a diagram may be include portions of many scenarios.</p>

<p>Example:</p>

<pre>    get_relationship_Overlaps -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Overlaps [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Scenario entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="common_name"
>common_name</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Diagram entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_ParticipatesAs</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_ParticipatesAs [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a generic compound to a specific compound where subceullar location has been specified.</p>

<p>Example:</p>

<pre>    get_relationship_ParticipatesAs -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_ParticipatesAs [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the LocalizedCompound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_ParticipatesAt</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_ParticipatesAt [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a localized compound to the location in which it occurs during one or more reactions.</p>

<p>Example:</p>

<pre>    get_relationship_ParticipatesAt -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_ParticipatesAt [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the LocalizedCompound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Location entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_ParticipatesIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_ParticipatesIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>A scenario consists of many participant reactions that convert the input compounds to output compounds.
A single reaction may participate in many scenarios.</p>

<p>Example:</p>

<pre>    get_relationship_ParticipatesIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_ParticipatesIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>

<dd>
<dt><a name="thermodynamic_reversibility"
>thermodynamic_reversibility</a></dt>

<dd>
<dt><a name="default_protons"
>default_protons</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Scenario entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="common_name"
>common_name</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_PerformedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_PerformedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Denotes that a Person was associated with a Experiment in some role.</p>

<p>Example:</p>

<pre>    get_relationship_PerformedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_PerformedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentMeta entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="startDate"
>startDate</a></dt>

<dd>
<dt><a name="comments"
>comments</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="role"
>role</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Person entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="firstName"
>firstName</a></dt>

<dd>
<dt><a name="lastName"
>lastName</a></dt>

<dd>
<dt><a name="contactEmail"
>contactEmail</a></dt>

<dd>
<dt><a name="institution"
>institution</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_PerformedExperiment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_PerformedExperiment [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Denotes that a Person was associated with a Experiment in some role.</p>

<p>Example:</p>

<pre>    get_relationship_PerformedExperiment -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_PerformedExperiment [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Person entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="firstName"
>firstName</a></dt>

<dd>
<dt><a name="lastName"
>lastName</a></dt>

<dd>
<dt><a name="contactEmail"
>contactEmail</a></dt>

<dd>
<dt><a name="institution"
>institution</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="role"
>role</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentMeta entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="startDate"
>startDate</a></dt>

<dd>
<dt><a name="comments"
>comments</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_ProducedResultsFor</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_ProducedResultsFor [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a probe set to a genome for which it was used to produce experimental results.
In general,
a probe set is used for only one genome and vice versa,
but this is not a requirement.</p>

<p>Example:</p>

<pre>    get_relationship_ProducedResultsFor -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_ProducedResultsFor [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProbeSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Produces</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Produces [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a peg feature to the protein sequence it produces (if any).
Only peg features participate in this relationship.
A single protein sequence will frequently be produced by many features.</p>

<p>Example:</p>

<pre>    get_relationship_Produces -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Produces [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Feature entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProteinSequence entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_ProtocolPublishedIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_ProtocolPublishedIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The ProtocolPublishedIn relationship describes where a particular protocol was published.</p>

<p>Example:</p>

<pre>    get_relationship_ProtocolPublishedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_ProtocolPublishedIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Protocol entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Publication entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="link"
>link</a></dt>

<dd>
<dt><a name="pubdate"
>pubdate</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Provided</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Provided [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a source (core) database to the subsystems it submitted to the knowledge base.</p>

<p>Example:</p>

<pre>    get_relationship_Provided -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Provided [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_PublishedExperiment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_PublishedExperiment [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The PublishedExperiment relationship describes where a particular experiment was published.</p>

<p>Example:</p>

<pre>    get_relationship_PublishedExperiment -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_PublishedExperiment [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Publication entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="link"
>link</a></dt>

<dd>
<dt><a name="pubdate"
>pubdate</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ExperimentMeta entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="startDate"
>startDate</a></dt>

<dd>
<dt><a name="comments"
>comments</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_PublishedProtocol</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_PublishedProtocol [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The ProtocolPublishedIn relationship describes where a particular protocol was published.</p>

<p>Example:</p>

<pre>    get_relationship_PublishedProtocol -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_PublishedProtocol [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Publication entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="link"
>link</a></dt>

<dd>
<dt><a name="pubdate"
>pubdate</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Protocol entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_ReflectsStateOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_ReflectsStateOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a genome to the atomic regulons that describe its state.</p>

<p>Example:</p>

<pre>    get_relationship_ReflectsStateOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_ReflectsStateOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the AtomicRegulon entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Shows</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Shows [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship indicates that a compound appears on a particular diagram.
The same compound can appear on many diagrams,
and a diagram always contains many compounds.</p>

<p>Example:</p>

<pre>    get_relationship_Shows -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Shows [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Diagram entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="location"
>location</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_StrainParentOf</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_StrainParentOf [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The recursive StrainParentOf relationship organizes derived organisms into a tree based on parent/child relationships.</p>

<p>Example:</p>

<pre>    get_relationship_StrainParentOf -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_StrainParentOf [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Strain entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="aggregateData"
>aggregateData</a></dt>

<dd>
<dt><a name="wildtype"
>wildtype</a></dt>

<dd>
<dt><a name="referenceStrain"
>referenceStrain</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Strain entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="aggregateData"
>aggregateData</a></dt>

<dd>
<dt><a name="wildtype"
>wildtype</a></dt>

<dd>
<dt><a name="referenceStrain"
>referenceStrain</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Submitted</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Submitted [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a genome to the core database from which it was loaded.</p>

<p>Example:</p>

<pre>    get_relationship_Submitted -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Submitted [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Summarizes</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Summarizes [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship describes the statistical frequencies of the most common alleles in various positions on the reference contig.</p>

<p>Example:</p>

<pre>    get_relationship_Summarizes -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Summarizes [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the AlleleFrequency entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="position"
>position</a></dt>

<dd>
<dt><a name="minor_AF"
>minor_AF</a></dt>

<dd>
<dt><a name="minor_allele"
>minor_allele</a></dt>

<dd>
<dt><a name="major_AF"
>major_AF</a></dt>

<dd>
<dt><a name="major_allele"
>major_allele</a></dt>

<dd>
<dt><a name="obs_unit_count"
>obs_unit_count</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="position"
>position</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Contig entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_SupersedesAlignment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_SupersedesAlignment [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an alignment to the alignments it replaces.</p>

<p>Example:</p>

<pre>    get_relationship_SupersedesAlignment -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_SupersedesAlignment [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Alignment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="successor_type"
>successor_type</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Alignment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_SupersedesTree</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_SupersedesTree [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a tree to the trees it replaces.</p>

<p>Example:</p>

<pre>    get_relationship_SupersedesTree -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_SupersedesTree [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Tree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="data_type"
>data_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="newick"
>newick</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="successor_type"
>successor_type</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Tree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="data_type"
>data_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="newick"
>newick</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Targets</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Targets [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a compound instance to the reaction instance in which it is transformed.</p>

<p>Example:</p>

<pre>    get_relationship_Targets -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Targets [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ReactionInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="protons"
>protons</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="coefficient"
>coefficient</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the CompoundInstance entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Treed</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Treed [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a tree to the source database from which it was generated.</p>

<p>Example:</p>

<pre>    get_relationship_Treed -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Treed [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Tree entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="data_type"
>data_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="newick"
>newick</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Triggers</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Triggers [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This connects a complex to the roles that work together to form the complex.</p>

<p>Example:</p>

<pre>    get_relationship_Triggers -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Triggers [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Role entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="optional"
>optional</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="triggering"
>triggering</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Complex entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_UsedIn</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_UsedIn [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The UsedIn relationship defines which media is used by an Environment.</p>

<p>Example:</p>

<pre>    get_relationship_UsedIn -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_UsedIn [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Media entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="is_minimal"
>is_minimal</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="solid"
>solid</a></dt>

<dd>
<dt><a name="is_defined"
>is_defined</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Environment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="temperature"
>temperature</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="oxygenConcentration"
>oxygenConcentration</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Uses</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Uses [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a genome to the machines that form its metabolic pathways.
A genome can use many machines,
but a machine is used by exactly one genome.</p>

<p>Example:</p>

<pre>    get_relationship_Uses -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Uses [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the SSRow entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="curated"
>curated</a></dt>

<dd>
<dt><a name="region"
>region</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_UsesAliasForCompound</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_UsesAliasForCompound [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a source (database or organization) with the compounds for which it has assigned names (aliases).
The alias itself is stored as intersection data.</p>

<p>Example:</p>

<pre>    get_relationship_UsesAliasForCompound -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_UsesAliasForCompound [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Compound entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_UsesAliasForReaction</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_UsesAliasForReaction [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a source (database or organization) with the reactions for which it has assigned names (aliases).
The alias itself is stored as intersection data.</p>

<p>Example:</p>

<pre>    get_relationship_UsesAliasForReaction -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_UsesAliasForReaction [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Reaction entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>

<dd>
<dt><a name="thermodynamic_reversibility"
>thermodynamic_reversibility</a></dt>

<dd>
<dt><a name="default_protons"
>default_protons</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_UsesCodons</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_UsesCodons [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a genome to the various codon usage records for it.</p>

<p>Example:</p>

<pre>    get_relationship_UsesCodons -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_UsesCodons [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the CodonUsage entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="frequencies"
>frequencies</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="subtype"
>subtype</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_UsesReference</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_UsesReference [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship associates each observational unit with the reference genome that it will be compared to.
All variations will be differences between the observational unit and the reference.</p>

<p>Example:</p>

<pre>    get_relationship_UsesReference -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_UsesReference [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the ObservationalUnit entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="source_name2"
>source_name2</a></dt>

<dd>
<dt><a name="plant_id"
>plant_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_Validates</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_Validates [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a scenario to its subsystem it validates.
A scenario belongs to exactly one subsystem,
but a subsystem may have multiple scenarios.</p>

<p>Example:</p>

<pre>    get_relationship_Validates -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_Validates [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Scenario entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="common_name"
>common_name</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_WasAlignedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_WasAlignedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an alignment to the database from which it was generated.</p>

<p>Example:</p>

<pre>    get_relationship_WasAlignedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_WasAlignedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Alignment entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_WasFormulatedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_WasFormulatedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a coregulated set to the source organization that originally computed it.</p>

<p>Example:</p>

<pre>    get_relationship_WasFormulatedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_WasFormulatedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the CoregulatedSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="binding_location"
>binding_location</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_WasGeneratedFrom</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_WasGeneratedFrom [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects an atomic regulon to a probe set from which experimental data was produced for its features.
It contains a vector of the expression levels.</p>

<p>Example:</p>

<pre>    get_relationship_WasGeneratedFrom -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_WasGeneratedFrom [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the AtomicRegulon entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>

<dd>
<dt><a name="level_vector"
>level_vector</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the ProbeSet entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_WasMeasuredBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_WasMeasuredBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>The IsMeasurementMethodOf relationship describes which protocol was used to take a measurement.</p>

<p>Example:</p>

<pre>    get_relationship_WasMeasuredBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_WasMeasuredBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Measurement entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>

<dd>
<dt><a name="mean"
>mean</a></dt>

<dd>
<dt><a name="median"
>median</a></dt>

<dd>
<dt><a name="stddev"
>stddev</a></dt>

<dd>
<dt><a name="N"
>N</a></dt>

<dd>
<dt><a name="p_value"
>p_value</a></dt>

<dd>
<dt><a name="Z_score"
>Z_score</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Protocol entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_WasProvidedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_WasProvidedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a source (core) database to the subsystems it submitted to the knowledge base.</p>

<p>Example:</p>

<pre>    get_relationship_WasProvidedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_WasProvidedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Subsystem entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>get_relationship_WasSubmittedBy</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>get_relationship_WasSubmittedBy [-c N] [-a] [--fields field-list] &#60; ids &#62; table.with.fields.added</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>This relationship connects a genome to the core database from which it was loaded.</p>

<p>Example:</p>

<pre>    get_relationship_WasSubmittedBy -a &#60; ids &#62; table.with.fields.added</pre>

<p>would read in a file of ids and add a column for each field in the relationship.</p>

<p>The standard input should be a tab-separated table (i.e., each line is a tab-separated set of fields). Normally, the last field in each line would contain the id. If some other column contains the id, use</p>

<pre>    -c N</pre>

<p>where N is the column (from 1) that contains the id.</p>

<p>This is a pipe command. The input is taken from the standard input, and the output is to the standard output.</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>Usage: get_relationship_WasSubmittedBy [arguments] &#60; ids &#62; table.with.fields.added</p>

<dl>
<dt><a name="-c_num"
>-c num</a></dt>

<dd>
<p>Select the identifier from column num</p>

<dt><a name="-from_field-list"
>-from field-list</a></dt>

<dd>
<p>Choose a set of fields from the Genome entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>

<dd>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>

<dt><a name="-rel_field-list"
>-rel field-list</a></dt>

<dd>
<p>Choose a set of fields from the relationship to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="from_link"
>from_link</a></dt>

<dd>
<dt><a name="to_link"
>to_link</a></dt>
</dl>

<dt><a name="-to_field-list"
>-to field-list</a></dt>

<dd>
<p>Choose a set of fields from the Source entity to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="id"
>id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Alignment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Alignment [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Alignment.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>An alignment arranges a group of sequences so that they match.
Each alignment is associated with a phylogenetic tree that describes how the sequences developed and their evolutionary distance.</p>

<p>Example:</p>

<pre>    query_entity_Alignment -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Alignment entity has the following relationship links:</p>

<dl>
<dt><a name="HasAlignmentAttribute_AlignmentAttribute"
>HasAlignmentAttribute AlignmentAttribute</a></dt>

<dd>
<dt><a name="IncludesAlignmentRow_AlignmentRow"
>IncludesAlignmentRow AlignmentRow</a></dt>

<dd>
<dt><a name="IsModificationOfAlignment_Alignment"
>IsModificationOfAlignment Alignment</a></dt>

<dd>
<dt><a name="IsModifiedToBuildAlignment_Alignment"
>IsModifiedToBuildAlignment Alignment</a></dt>

<dd>
<dt><a name="IsSupersededByAlignment_Alignment"
>IsSupersededByAlignment Alignment</a></dt>

<dd>
<dt><a name="IsUsedToBuildTree_Tree"
>IsUsedToBuildTree Tree</a></dt>

<dd>
<dt><a name="SupersedesAlignment_Alignment"
>SupersedesAlignment Alignment</a></dt>

<dd>
<dt><a name="WasAlignedBy_Source"
>WasAlignedBy Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Alignment [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="n_rows"
>n_rows</a></dt>

<dd>
<dt><a name="n_cols"
>n_cols</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="is_concatenation"
>is_concatenation</a></dt>

<dd>
<dt><a name="sequence_type"
>sequence_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_AlignmentAttribute</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_AlignmentAttribute [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity AlignmentAttribute.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>This entity represents an attribute type that can be assigned to an alignment.
The attribute values are stored in the relationships to the target.
The key is the attribute name.</p>

<p>Example:</p>

<pre>    query_entity_AlignmentAttribute -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AlignmentAttribute entity has the following relationship links:</p>

<dl>
<dt><a name="DescribesAlignment_Alignment"
>DescribesAlignment Alignment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_AlignmentAttribute [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_AlignmentRow</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_AlignmentRow [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity AlignmentRow.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>This entity represents a single row of an alignment.
In general,
this corresponds to a sequence,
but in a concatenated alignment multiple sequences may be represented here.</p>

<p>Example:</p>

<pre>    query_entity_AlignmentRow -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AlignmentRow entity has the following relationship links:</p>

<dl>
<dt><a name="ContainsAlignedDNA_ContigSequence"
>ContainsAlignedDNA ContigSequence</a></dt>

<dd>
<dt><a name="ContainsAlignedProtein_ProteinSequence"
>ContainsAlignedProtein ProteinSequence</a></dt>

<dd>
<dt><a name="IsAlignmentRowIn_Alignment"
>IsAlignmentRowIn Alignment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_AlignmentRow [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="row_number"
>row_number</a></dt>

<dd>
<dt><a name="row_id"
>row_id</a></dt>

<dd>
<dt><a name="row_description"
>row_description</a></dt>

<dd>
<dt><a name="n_components"
>n_components</a></dt>

<dd>
<dt><a name="beg_pos_aln"
>beg_pos_aln</a></dt>

<dd>
<dt><a name="end_pos_aln"
>end_pos_aln</a></dt>

<dd>
<dt><a name="md5_of_ungapped_sequence"
>md5_of_ungapped_sequence</a></dt>

<dd>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_AlleleFrequency</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_AlleleFrequency [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity AlleleFrequency.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>An allele frequency represents a summary of the major and minor allele frequencies for a position on a chromosome.</p>

<p>Example:</p>

<pre>    query_entity_AlleleFrequency -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AlleleFrequency entity has the following relationship links:</p>

<dl>
<dt><a name="Summarizes_Contig"
>Summarizes Contig</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_AlleleFrequency [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="position"
>position</a></dt>

<dd>
<dt><a name="minor_AF"
>minor_AF</a></dt>

<dd>
<dt><a name="minor_allele"
>minor_allele</a></dt>

<dd>
<dt><a name="major_AF"
>major_AF</a></dt>

<dd>
<dt><a name="major_allele"
>major_allele</a></dt>

<dd>
<dt><a name="obs_unit_count"
>obs_unit_count</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Annotation</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Annotation [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Annotation.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>An annotation is a comment attached to a feature.
Annotations are used to track the history of a feature&#39;s functional assignments and any related issues.
The key is the feature ID followed by a colon and a complemented ten-digit sequence number.</p>

<p>Example:</p>

<pre>    query_entity_Annotation -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Annotation entity has the following relationship links:</p>

<dl>
<dt><a name="Annotates_Feature"
>Annotates Feature</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Annotation [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="annotator"
>annotator</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>

<dd>
<dt><a name="annotation_time"
>annotation_time</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Assay</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Assay [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Assay.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>An assay is an experimental design for determining alleles at specific chromosome positions.</p>

<p>Example:</p>

<pre>    query_entity_Assay -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Assay entity has the following relationship links:</p>

<dl>
<dt><a name="IsAssayOf_StudyExperiment"
>IsAssayOf StudyExperiment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Assay [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="assay_type"
>assay_type</a></dt>

<dd>
<dt><a name="assay_type_id"
>assay_type_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_AtomicRegulon</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_AtomicRegulon [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity AtomicRegulon.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>An atomic regulon is an indivisible group of coregulated features on a single genome.
Atomic regulons are constructed so that a given feature can only belong to one.
Because of this,
the expression levels for atomic regulons represent in some sense the state of a cell.</p>

<p>Example:</p>

<pre>    query_entity_AtomicRegulon -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The AtomicRegulon entity has the following relationship links:</p>

<dl>
<dt><a name="IsAffectedIn_Experiment"
>IsAffectedIn Experiment</a></dt>

<dd>
<dt><a name="IsFormedOf_Feature"
>IsFormedOf Feature</a></dt>

<dd>
<dt><a name="ReflectsStateOf_Genome"
>ReflectsStateOf Genome</a></dt>

<dd>
<dt><a name="WasGeneratedFrom_ProbeSet"
>WasGeneratedFrom ProbeSet</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_AtomicRegulon [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Attribute</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Attribute [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Attribute.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>An attribute describes a category of condition or characteristic for an experiment.
The goals of the experiment can be inferred from its values for all the attributes of interest.</p>

<p>Example:</p>

<pre>    query_entity_Attribute -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Attribute entity has the following relationship links:</p>

<dl>
<dt><a name="HasValueIn_Experiment"
>HasValueIn Experiment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Attribute [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="description"
>description</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Biomass</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Biomass [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Biomass.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A biomass is a collection of compounds in a specific ratio and in specific compartments that are necessary for a cell to function properly.
The prediction of biomasses is key to the functioning of the model.
Each biomass belongs to a specific model.</p>

<p>Example:</p>

<pre>    query_entity_Biomass -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Biomass entity has the following relationship links:</p>

<dl>
<dt><a name="IsComprisedOf_CompoundInstance"
>IsComprisedOf CompoundInstance</a></dt>

<dd>
<dt><a name="IsManagedBy_Model"
>IsManagedBy Model</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Biomass [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="dna"
>dna</a></dt>

<dd>
<dt><a name="protein"
>protein</a></dt>

<dd>
<dt><a name="cell_wall"
>cell_wall</a></dt>

<dd>
<dt><a name="lipid"
>lipid</a></dt>

<dd>
<dt><a name="cofactor"
>cofactor</a></dt>

<dd>
<dt><a name="energy"
>energy</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_CodonUsage</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_CodonUsage [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity CodonUsage.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>This entity contains information about the codon usage frequency in a particular genome with respect to a particular type of analysis (e.g.
high-expression genes,
modal,
mean,
etc.).</p>

<p>Example:</p>

<pre>    query_entity_CodonUsage -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The CodonUsage entity has the following relationship links:</p>

<dl>
<dt><a name="AreCodonsFor_Genome"
>AreCodonsFor Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_CodonUsage [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="frequencies"
>frequencies</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="subtype"
>subtype</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Complex</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Complex [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Complex.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A complex is a set of chemical reactions that act in concert to effect a role.</p>

<p>Example:</p>

<pre>    query_entity_Complex -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Complex entity has the following relationship links:</p>

<dl>
<dt><a name="HasStep_Reaction"
>HasStep Reaction</a></dt>

<dd>
<dt><a name="IsTriggeredBy_Role"
>IsTriggeredBy Role</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Complex [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Compound</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Compound [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Compound.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A compound is a chemical that participates in a reaction.
Both ligands and reaction components are treated as compounds.</p>

<p>Example:</p>

<pre>    query_entity_Compound -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Compound entity has the following relationship links:</p>

<dl>
<dt><a name="ComponentOf_Compound"
>ComponentOf Compound</a></dt>

<dd>
<dt><a name="CompoundMeasuredBy_Measurement"
>CompoundMeasuredBy Measurement</a></dt>

<dd>
<dt><a name="ConsistsOfCompounds_Compound"
>ConsistsOfCompounds Compound</a></dt>

<dd>
<dt><a name="IncludedIn_Environment"
>IncludedIn Environment</a></dt>

<dd>
<dt><a name="IsPresentIn_Media"
>IsPresentIn Media</a></dt>

<dd>
<dt><a name="IsShownOn_Diagram"
>IsShownOn Diagram</a></dt>

<dd>
<dt><a name="IsTerminusFor_Scenario"
>IsTerminusFor Scenario</a></dt>

<dd>
<dt><a name="ParticipatesAs_LocalizedCompound"
>ParticipatesAs LocalizedCompound</a></dt>

<dd>
<dt><a name="UsesAliasForCompound_Source"
>UsesAliasForCompound Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Compound [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="ubiquitous"
>ubiquitous</a></dt>

<dd>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="mass"
>mass</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>

<dd>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_CompoundInstance</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_CompoundInstance [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity CompoundInstance.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A Compound Instance represents the occurrence of a particular compound in a location in a model.</p>

<p>Example:</p>

<pre>    query_entity_CompoundInstance -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The CompoundInstance entity has the following relationship links:</p>

<dl>
<dt><a name="Comprises_Biomass"
>Comprises Biomass</a></dt>

<dd>
<dt><a name="HasRealLocationIn_LocationInstance"
>HasRealLocationIn LocationInstance</a></dt>

<dd>
<dt><a name="IsReagentIn_ReactionInstance"
>IsReagentIn ReactionInstance</a></dt>

<dd>
<dt><a name="IsUsageOf_LocalizedCompound"
>IsUsageOf LocalizedCompound</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_CompoundInstance [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="charge"
>charge</a></dt>

<dd>
<dt><a name="formula"
>formula</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Contig</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Contig [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Contig.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A contig is thought of as composing a part of the DNA associated with a specific genome.
It is represented as an ID (including the genome ID) and a ContigSequence.
We do not think of strings of DNA from,
say,
a metgenomic sample as &#34;contigs&#34;,
since there is no associated genome (these would be considered ContigSequences).
This use of the term &#34;ContigSequence&#34;,
rather than just &#34;DNA sequence&#34;,
may turn out to be a bad idea.
For now,
you should just realize that a Contig has an associated genome,
but a ContigSequence does not.</p>

<p>Example:</p>

<pre>    query_entity_Contig -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Contig entity has the following relationship links:</p>

<dl>
<dt><a name="HasAsSequence_ContigSequence"
>HasAsSequence ContigSequence</a></dt>

<dd>
<dt><a name="HasVariationIn_ObservationalUnit"
>HasVariationIn ObservationalUnit</a></dt>

<dd>
<dt><a name="IsComponentOf_Genome"
>IsComponentOf Genome</a></dt>

<dd>
<dt><a name="IsImpactedBy_Trait"
>IsImpactedBy Trait</a></dt>

<dd>
<dt><a name="IsLocusFor_Feature"
>IsLocusFor Feature</a></dt>

<dd>
<dt><a name="IsSummarizedBy_AlleleFrequency"
>IsSummarizedBy AlleleFrequency</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Contig [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_ContigChunk</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_ContigChunk [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity ContigChunk.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>ContigChunks are strings of DNA thought of as being a string in a 4-character alphabet with an associated ID.
We allow a broader alphabet that includes U (for RNA) and the standard ambiguity characters.</p>

<p>Example:</p>

<pre>    query_entity_ContigChunk -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ContigChunk entity has the following relationship links:</p>

<dl>
<dt><a name="IsSectionOf_ContigSequence"
>IsSectionOf ContigSequence</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_ContigChunk [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_ContigSequence</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_ContigSequence [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity ContigSequence.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>ContigSequences are strings of DNA.
Contigs have an associated genome,
but ContigSequences do not.
We can think of random samples of DNA as a set of ContigSequences.
There are no length constraints imposed on ContigSequences -- they can be either very short or very long.
The basic unit of data that is moved to/from the database is the ContigChunk,
from which ContigSequences are formed.
The key of a ContigSequence is the sequence&#39;s MD5 identifier.</p>

<p>Example:</p>

<pre>    query_entity_ContigSequence -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ContigSequence entity has the following relationship links:</p>

<dl>
<dt><a name="HasSection_ContigChunk"
>HasSection ContigChunk</a></dt>

<dd>
<dt><a name="IsAlignedDNAComponentOf_AlignmentRow"
>IsAlignedDNAComponentOf AlignmentRow</a></dt>

<dd>
<dt><a name="IsSequenceOf_Contig"
>IsSequenceOf Contig</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_ContigSequence [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="length"
>length</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_CoregulatedSet</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_CoregulatedSet [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity CoregulatedSet.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>We need to represent sets of genes that are coregulated via some regulatory mechanism.
In particular,
we wish to represent genes that are coregulated using transcription binding sites and corresponding transcription regulatory proteins.
We represent a coregulated set (which may,
or may not,
be considered a regulon) using CoregulatedSet.</p>

<p>Example:</p>

<pre>    query_entity_CoregulatedSet -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The CoregulatedSet entity has the following relationship links:</p>

<dl>
<dt><a name="IsControlledUsing_Feature"
>IsControlledUsing Feature</a></dt>

<dd>
<dt><a name="IsRegulatedSetOf_Feature"
>IsRegulatedSetOf Feature</a></dt>

<dd>
<dt><a name="WasFormulatedBy_Source"
>WasFormulatedBy Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_CoregulatedSet [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="binding_location"
>binding_location</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Diagram</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Diagram [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Diagram.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A functional diagram describes a network of chemical reactions,
often comprising a single subsystem.</p>

<p>Example:</p>

<pre>    query_entity_Diagram -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Diagram entity has the following relationship links:</p>

<dl>
<dt><a name="Displays_Reaction"
>Displays Reaction</a></dt>

<dd>
<dt><a name="IncludesPartOf_Scenario"
>IncludesPartOf Scenario</a></dt>

<dd>
<dt><a name="IsRelevantFor_Subsystem"
>IsRelevantFor Subsystem</a></dt>

<dd>
<dt><a name="Shows_Compound"
>Shows Compound</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Diagram [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="content"
>content</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_EcNumber</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_EcNumber [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity EcNumber.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>EC numbers are assigned by the Enzyme Commission,
and consist of four numbers separated by periods,
each indicating a successively smaller cateogry of enzymes.</p>

<p>Example:</p>

<pre>    query_entity_EcNumber -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The EcNumber entity has the following relationship links:</p>

<dl>
<dt><a name="IsConsistentWith_Role"
>IsConsistentWith Role</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_EcNumber [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="obsolete"
>obsolete</a></dt>

<dd>
<dt><a name="replacedby"
>replacedby</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Environment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Environment [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Environment.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>An Environment is a set of conditions for microbial growth,
including temperature,
aerobicity,
media,
and supplementary conditions.</p>

<p>Example:</p>

<pre>    query_entity_Environment -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Environment entity has the following relationship links:</p>

<dl>
<dt><a name="HasMedia_Media"
>HasMedia Media</a></dt>

<dd>
<dt><a name="HasParameter_Parameter"
>HasParameter Parameter</a></dt>

<dd>
<dt><a name="IncludesAdditionalCompounds_Compound"
>IncludesAdditionalCompounds Compound</a></dt>

<dd>
<dt><a name="IsContextOf_ExperimentalUnit"
>IsContextOf ExperimentalUnit</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Environment [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="temperature"
>temperature</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="oxygenConcentration"
>oxygenConcentration</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Experiment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Experiment [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Experiment.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>An experiment is a combination of conditions for which gene expression information is desired.
The result of the experiment is a set of expression levels for features under the given conditions.</p>

<p>Example:</p>

<pre>    query_entity_Experiment -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Experiment entity has the following relationship links:</p>

<dl>
<dt><a name="AffectsLevelOf_AtomicRegulon"
>AffectsLevelOf AtomicRegulon</a></dt>

<dd>
<dt><a name="HasResultsFor_ProbeSet"
>HasResultsFor ProbeSet</a></dt>

<dd>
<dt><a name="HasValueFor_Attribute"
>HasValueFor Attribute</a></dt>

<dd>
<dt><a name="IndicatesSignalFor_Feature"
>IndicatesSignalFor Feature</a></dt>

<dd>
<dt><a name="OperatesIn_Media"
>OperatesIn Media</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Experiment [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source"
>source</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_ExperimentalUnit</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_ExperimentalUnit [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity ExperimentalUnit.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>An ExperimentalUnit is a subset of an experiment consisting of a Strain,
an Environment,
and one or more Measurements on that strain in the specified environment.
ExperimentalUnits belong to a single experiment.</p>

<p>Example:</p>

<pre>    query_entity_ExperimentalUnit -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ExperimentalUnit entity has the following relationship links:</p>

<dl>
<dt><a name="GroupedBy_ExperimentalUnitGroup"
>GroupedBy ExperimentalUnitGroup</a></dt>

<dd>
<dt><a name="HasEnvironment_Environment"
>HasEnvironment Environment</a></dt>

<dd>
<dt><a name="HasMeasurement_Measurement"
>HasMeasurement Measurement</a></dt>

<dd>
<dt><a name="IncludesStrain_Strain"
>IncludesStrain Strain</a></dt>

<dd>
<dt><a name="IsExperimentalUnitOf_ExperimentMeta"
>IsExperimentalUnitOf ExperimentMeta</a></dt>

<dd>
<dt><a name="IsTimepointOf_TimeSeries"
>IsTimepointOf TimeSeries</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_ExperimentalUnit [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_ExperimentalUnitGroup</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_ExperimentalUnitGroup [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity ExperimentalUnitGroup.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>An ExperimentalUnitGroup allows for grouping related experimental units and their measurements - for instance measurements that were in the same plate.</p>

<p>Example:</p>

<pre>    query_entity_ExperimentalUnitGroup -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ExperimentalUnitGroup entity has the following relationship links:</p>

<dl>
<dt><a name="ContainsExperimentalUnit_ExperimentalUnit"
>ContainsExperimentalUnit ExperimentalUnit</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_ExperimentalUnitGroup [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="comments"
>comments</a></dt>

<dd>
<dt><a name="groupType"
>groupType</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_ExperimentMeta</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_ExperimentMeta [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity ExperimentMeta.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>An Experiment consists of (potentially) multiple strains,
environments,
and measurements on those strains and environments.</p>

<p>Example:</p>

<pre>    query_entity_ExperimentMeta -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ExperimentMeta entity has the following relationship links:</p>

<dl>
<dt><a name="ExperimentPublishedIn_Publication"
>ExperimentPublishedIn Publication</a></dt>

<dd>
<dt><a name="HasExperimentalUnit_ExperimentalUnit"
>HasExperimentalUnit ExperimentalUnit</a></dt>

<dd>
<dt><a name="PerformedBy_Person"
>PerformedBy Person</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_ExperimentMeta [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="startDate"
>startDate</a></dt>

<dd>
<dt><a name="comments"
>comments</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Family</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Family [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Family.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>The Kbase will support the maintenance of protein families (as sets of Features with associated translations).
We are initially only supporting the notion of a family as composed of a set of isofunctional homologs.
That is,
the families we initially support should be thought of as containing protein-encoding genes whose associated sequences all implement the same function (we do understand that the notion of &#34;function&#34; is somewhat ambiguous,
so let us sweep this under the rug by calling a functional role a &#34;primitive concept&#34;).
We currently support families in which the members are protein sequences as well.
Identical protein sequences as products of translating distinct genes may or may not have identical functions.
This may be justified,
since in a very,
very,
very few cases identical proteins do,
in fact,
have distinct functions.</p>

<p>Example:</p>

<pre>    query_entity_Family -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Family entity has the following relationship links:</p>

<dl>
<dt><a name="HasMember_Feature"
>HasMember Feature</a></dt>

<dd>
<dt><a name="HasProteinMember_ProteinSequence"
>HasProteinMember ProteinSequence</a></dt>

<dd>
<dt><a name="IsCoupledTo_Family"
>IsCoupledTo Family</a></dt>

<dd>
<dt><a name="IsCoupledWith_Family"
>IsCoupledWith Family</a></dt>

<dd>
<dt><a name="IsFamilyFor_Role"
>IsFamilyFor Role</a></dt>

<dd>
<dt><a name="IsRepresentedIn_Genome"
>IsRepresentedIn Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Family [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="release"
>release</a></dt>

<dd>
<dt><a name="family_function"
>family_function</a></dt>

<dd>
<dt><a name="alignment"
>alignment</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Feature</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Feature [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Feature.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A feature (sometimes also called a gene) is a part of a genome that is of special interest.
Features may be spread across multiple DNA sequences (contigs) of a genome,
but never across more than one genome.
Each feature in the database has a unique ID that functions as its ID in this table.
Normally a Feature is just a single contigous region on a contig.
Features have types,
and an appropriate choice of available types allows the support of protein-encoding genes,
exons,
RNA genes,
binding sites,
pathogenicity islands,
or whatever.</p>

<p>Example:</p>

<pre>    query_entity_Feature -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Feature entity has the following relationship links:</p>

<dl>
<dt><a name="Controls_CoregulatedSet"
>Controls CoregulatedSet</a></dt>

<dd>
<dt><a name="Encompasses_Feature"
>Encompasses Feature</a></dt>

<dd>
<dt><a name="FeatureMeasuredBy_Measurement"
>FeatureMeasuredBy Measurement</a></dt>

<dd>
<dt><a name="HasCoregulationWith_Feature"
>HasCoregulationWith Feature</a></dt>

<dd>
<dt><a name="HasFunctional_Role"
>HasFunctional Role</a></dt>

<dd>
<dt><a name="HasIndicatedSignalFrom_Experiment"
>HasIndicatedSignalFrom Experiment</a></dt>

<dd>
<dt><a name="HasLevelsFrom_ProbeSet"
>HasLevelsFrom ProbeSet</a></dt>

<dd>
<dt><a name="ImplementsReaction_ReactionInstance"
>ImplementsReaction ReactionInstance</a></dt>

<dd>
<dt><a name="IsAnnotatedBy_Annotation"
>IsAnnotatedBy Annotation</a></dt>

<dd>
<dt><a name="IsContainedIn_SSCell"
>IsContainedIn SSCell</a></dt>

<dd>
<dt><a name="IsCoregulatedWith_Feature"
>IsCoregulatedWith Feature</a></dt>

<dd>
<dt><a name="IsEncompassedIn_Feature"
>IsEncompassedIn Feature</a></dt>

<dd>
<dt><a name="IsExemplarOf_Role"
>IsExemplarOf Role</a></dt>

<dd>
<dt><a name="IsFormedInto_AtomicRegulon"
>IsFormedInto AtomicRegulon</a></dt>

<dd>
<dt><a name="IsInPair_Pairing"
>IsInPair Pairing</a></dt>

<dd>
<dt><a name="IsLocatedIn_Contig"
>IsLocatedIn Contig</a></dt>

<dd>
<dt><a name="IsMemberOf_Family"
>IsMemberOf Family</a></dt>

<dd>
<dt><a name="IsOwnedBy_Genome"
>IsOwnedBy Genome</a></dt>

<dd>
<dt><a name="IsRegulatedIn_CoregulatedSet"
>IsRegulatedIn CoregulatedSet</a></dt>

<dd>
<dt><a name="KnockedOutIn_Strain"
>KnockedOutIn Strain</a></dt>

<dd>
<dt><a name="Produces_ProteinSequence"
>Produces ProteinSequence</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Feature [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="feature_type"
>feature_type</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="sequence_length"
>sequence_length</a></dt>

<dd>
<dt><a name="function"
>function</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Genome</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Genome [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Genome.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>The Kbase houses a large and growing set of genomes.
We often have multiple genomes that have identical DNA.
These usually have distinct gene calls and annotations,
but not always.
We consider the Kbase to be a framework for managing hundreds of thousands of genomes and offering the tools needed to support compartive analysis on large sets of genomes,
some of which are virtually identical.</p>

<p>Example:</p>

<pre>    query_entity_Genome -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Genome entity has the following relationship links:</p>

<dl>
<dt><a name="GenomeParentOf_Strain"
>GenomeParentOf Strain</a></dt>

<dd>
<dt><a name="HadResultsProducedBy_ProbeSet"
>HadResultsProducedBy ProbeSet</a></dt>

<dd>
<dt><a name="HasRepresentativeOf_Family"
>HasRepresentativeOf Family</a></dt>

<dd>
<dt><a name="IsCollectedInto_OTU"
>IsCollectedInto OTU</a></dt>

<dd>
<dt><a name="IsComposedOf_Contig"
>IsComposedOf Contig</a></dt>

<dd>
<dt><a name="IsConfiguredBy_AtomicRegulon"
>IsConfiguredBy AtomicRegulon</a></dt>

<dd>
<dt><a name="IsInTaxa_TaxonomicGrouping"
>IsInTaxa TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsModeledBy_Model"
>IsModeledBy Model</a></dt>

<dd>
<dt><a name="IsOwnerOf_Feature"
>IsOwnerOf Feature</a></dt>

<dd>
<dt><a name="IsReferencedBy_ObservationalUnit"
>IsReferencedBy ObservationalUnit</a></dt>

<dd>
<dt><a name="Uses_SSRow"
>Uses SSRow</a></dt>

<dd>
<dt><a name="UsesCodons_CodonUsage"
>UsesCodons CodonUsage</a></dt>

<dd>
<dt><a name="WasSubmittedBy_Source"
>WasSubmittedBy Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Genome [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="pegs"
>pegs</a></dt>

<dd>
<dt><a name="rnas"
>rnas</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="complete"
>complete</a></dt>

<dd>
<dt><a name="prokaryotic"
>prokaryotic</a></dt>

<dd>
<dt><a name="dna_size"
>dna_size</a></dt>

<dd>
<dt><a name="contigs"
>contigs</a></dt>

<dd>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="genetic_code"
>genetic_code</a></dt>

<dd>
<dt><a name="gc_content"
>gc_content</a></dt>

<dd>
<dt><a name="phenotype"
>phenotype</a></dt>

<dd>
<dt><a name="md5"
>md5</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Locality</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Locality [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Locality.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A locality is a geographic location.</p>

<p>Example:</p>

<pre>    query_entity_Locality -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Locality entity has the following relationship links:</p>

<dl>
<dt><a name="HasUnits_ObservationalUnit"
>HasUnits ObservationalUnit</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Locality [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="city"
>city</a></dt>

<dd>
<dt><a name="state"
>state</a></dt>

<dd>
<dt><a name="country"
>country</a></dt>

<dd>
<dt><a name="origcty"
>origcty</a></dt>

<dd>
<dt><a name="elevation"
>elevation</a></dt>

<dd>
<dt><a name="latitude"
>latitude</a></dt>

<dd>
<dt><a name="longitude"
>longitude</a></dt>

<dd>
<dt><a name="lo_accession"
>lo_accession</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_LocalizedCompound</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_LocalizedCompound [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity LocalizedCompound.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>This entity represents a compound occurring in a specific location.
A reaction always involves localized compounds.
If a reaction occurs entirely in a single location,
it will frequently only be represented by the cytoplasmic versions of the compounds; however,
a transport always uses specifically located compounds.</p>

<p>Example:</p>

<pre>    query_entity_LocalizedCompound -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The LocalizedCompound entity has the following relationship links:</p>

<dl>
<dt><a name="HasUsage_CompoundInstance"
>HasUsage CompoundInstance</a></dt>

<dd>
<dt><a name="IsInvolvedIn_Reaction"
>IsInvolvedIn Reaction</a></dt>

<dd>
<dt><a name="IsParticipationOf_Compound"
>IsParticipationOf Compound</a></dt>

<dd>
<dt><a name="ParticipatesAt_Location"
>ParticipatesAt Location</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_LocalizedCompound [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Location</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Location [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Location.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A location is a region of the cell where reaction compounds originate from or are transported to (e.g.
cell wall,
extracellular,
cytoplasm).</p>

<p>Example:</p>

<pre>    query_entity_Location -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Location entity has the following relationship links:</p>

<dl>
<dt><a name="IsInstantiatedBy_LocationInstance"
>IsInstantiatedBy LocationInstance</a></dt>

<dd>
<dt><a name="IsParticipatingAt_LocalizedCompound"
>IsParticipatingAt LocalizedCompound</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Location [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_LocationInstance</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_LocationInstance [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity LocationInstance.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>The Location Instance represents a region of a cell (e.g.
cell wall,
cytoplasm) as it appears in a specific model.</p>

<p>Example:</p>

<pre>    query_entity_LocationInstance -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The LocationInstance entity has the following relationship links:</p>

<dl>
<dt><a name="IsDivisionOf_Model"
>IsDivisionOf Model</a></dt>

<dd>
<dt><a name="IsInstanceOf_Location"
>IsInstanceOf Location</a></dt>

<dd>
<dt><a name="IsRealLocationOf_CompoundInstance"
>IsRealLocationOf CompoundInstance</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_LocationInstance [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="index"
>index</a></dt>

<dd>
<dt><a name="label"
>label</a></dt>

<dd>
<dt><a name="pH"
>pH</a></dt>

<dd>
<dt><a name="potential"
>potential</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Measurement</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Measurement [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Measurement.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A Measurement is a value generated by performing a protocol to evaluate a value on an ExperimentalUnit - e.g.
a strain in an environment.</p>

<p>Example:</p>

<pre>    query_entity_Measurement -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Measurement entity has the following relationship links:</p>

<dl>
<dt><a name="IsDefinedBy_MeasurementDescription"
>IsDefinedBy MeasurementDescription</a></dt>

<dd>
<dt><a name="IsMeasureOf_ExperimentalUnit"
>IsMeasureOf ExperimentalUnit</a></dt>

<dd>
<dt><a name="MeasuresCompound_Compound"
>MeasuresCompound Compound</a></dt>

<dd>
<dt><a name="MeasuresFeature_Feature"
>MeasuresFeature Feature</a></dt>

<dd>
<dt><a name="WasMeasuredBy_Protocol"
>WasMeasuredBy Protocol</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Measurement [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="value"
>value</a></dt>

<dd>
<dt><a name="mean"
>mean</a></dt>

<dd>
<dt><a name="median"
>median</a></dt>

<dd>
<dt><a name="stddev"
>stddev</a></dt>

<dd>
<dt><a name="N"
>N</a></dt>

<dd>
<dt><a name="p_value"
>p_value</a></dt>

<dd>
<dt><a name="Z_score"
>Z_score</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_MeasurementDescription</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_MeasurementDescription [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity MeasurementDescription.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A MeasurementDescription provides information about a measurement value.</p>

<p>Example:</p>

<pre>    query_entity_MeasurementDescription -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The MeasurementDescription entity has the following relationship links:</p>

<dl>
<dt><a name="DescribesMeasurement_Measurement"
>DescribesMeasurement Measurement</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_MeasurementDescription [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="unitOfMeasure"
>unitOfMeasure</a></dt>

<dd>
<dt><a name="category"
>category</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Media</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Media [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Media.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A media describes the chemical content of the solution in which cells are grown in an experiment or for the purposes of a model.
The key is the common media name.
The nature of the media is described by its relationship to its constituent compounds.</p>

<p>Example:</p>

<pre>    query_entity_Media -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Media entity has the following relationship links:</p>

<dl>
<dt><a name="HasPresenceOf_Compound"
>HasPresenceOf Compound</a></dt>

<dd>
<dt><a name="IsUtilizedIn_Experiment"
>IsUtilizedIn Experiment</a></dt>

<dd>
<dt><a name="UsedIn_Environment"
>UsedIn Environment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Media [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="is_minimal"
>is_minimal</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="solid"
>solid</a></dt>

<dd>
<dt><a name="is_defined"
>is_defined</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Model</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Model [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Model.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A model specifies a relationship between sets of features and reactions in a cell.
It is used to simulate cell growth and gene knockouts to validate annotations.</p>

<p>Example:</p>

<pre>    query_entity_Model -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Model entity has the following relationship links:</p>

<dl>
<dt><a name="HasRequirementOf_ReactionInstance"
>HasRequirementOf ReactionInstance</a></dt>

<dd>
<dt><a name="IsDividedInto_LocationInstance"
>IsDividedInto LocationInstance</a></dt>

<dd>
<dt><a name="Manages_Biomass"
>Manages Biomass</a></dt>

<dd>
<dt><a name="Models_Genome"
>Models Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Model [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="reaction_count"
>reaction_count</a></dt>

<dd>
<dt><a name="compound_count"
>compound_count</a></dt>

<dd>
<dt><a name="annotation_count"
>annotation_count</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_ObservationalUnit</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_ObservationalUnit [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity ObservationalUnit.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>An ObservationalUnit is an individual plant that 1) is part of an experiment or study,
2) has measured traits,
and 3) is assayed for the purpose of determining alleles.</p>

<p>Example:</p>

<pre>    query_entity_ObservationalUnit -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ObservationalUnit entity has the following relationship links:</p>

<dl>
<dt><a name="DefinedBy_TaxonomicGrouping"
>DefinedBy TaxonomicGrouping</a></dt>

<dd>
<dt><a name="HasTrait_Trait"
>HasTrait Trait</a></dt>

<dd>
<dt><a name="IsLocated_Locality"
>IsLocated Locality</a></dt>

<dd>
<dt><a name="IsPartOf_StudyExperiment"
>IsPartOf StudyExperiment</a></dt>

<dd>
<dt><a name="IsVariedIn_Contig"
>IsVariedIn Contig</a></dt>

<dd>
<dt><a name="UsesReference_Genome"
>UsesReference Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_ObservationalUnit [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="source_name2"
>source_name2</a></dt>

<dd>
<dt><a name="plant_id"
>plant_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_OTU</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_OTU [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity OTU.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>An OTU (Organism Taxonomic Unit) is a named group of related genomes.</p>

<p>Example:</p>

<pre>    query_entity_OTU -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The OTU entity has the following relationship links:</p>

<dl>
<dt><a name="IsCollectionOf_Genome"
>IsCollectionOf Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_OTU [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Pairing</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Pairing [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Pairing.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A pairing indicates that two features are found close together in a genome.
Not all possible pairings are stored in the database; only those that are considered for some reason to be significant for annotation purposes.The key of the pairing is the concatenation of the feature IDs in alphabetical order with an intervening colon.</p>

<p>Example:</p>

<pre>    query_entity_Pairing -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Pairing entity has the following relationship links:</p>

<dl>
<dt><a name="Determines_PairSet"
>Determines PairSet</a></dt>

<dd>
<dt><a name="IsPairOf_Feature"
>IsPairOf Feature</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Pairing [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_PairSet</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_PairSet [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity PairSet.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A PairSet is a precompute set of pairs or genes.
Each pair occurs close to one another of the chromosome.
We believe that all of the first members of the pairs correspond to one another (are quite similar),
as do all of the second members of the pairs.
These pairs (from prokaryotic genomes) offer one of the most powerful clues relating to uncharacterized genes/peroteins.</p>

<p>Example:</p>

<pre>    query_entity_PairSet -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The PairSet entity has the following relationship links:</p>

<dl>
<dt><a name="IsDeterminedBy_Pairing"
>IsDeterminedBy Pairing</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_PairSet [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="score"
>score</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Parameter</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Parameter [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Parameter.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A parameter is the name of some quantity that has a value.</p>

<p>Example:</p>

<pre>    query_entity_Parameter -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Parameter entity has the following relationship links:</p>

<dl>
<dt><a name="OfEnvironment_Environment"
>OfEnvironment Environment</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Parameter [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Person</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Person [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Person.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A person represents a human affiliated in some way with Kbase.</p>

<p>Example:</p>

<pre>    query_entity_Person -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Person entity has the following relationship links:</p>

<dl>
<dt><a name="PerformedExperiment_ExperimentMeta"
>PerformedExperiment ExperimentMeta</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Person [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="firstName"
>firstName</a></dt>

<dd>
<dt><a name="lastName"
>lastName</a></dt>

<dd>
<dt><a name="contactEmail"
>contactEmail</a></dt>

<dd>
<dt><a name="institution"
>institution</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_ProbeSet</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_ProbeSet [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity ProbeSet.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A probe set is a device containing multiple probe sequences for use in gene expression experiments.</p>

<p>Example:</p>

<pre>    query_entity_ProbeSet -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ProbeSet entity has the following relationship links:</p>

<dl>
<dt><a name="GeneratedLevelsFor_AtomicRegulon"
>GeneratedLevelsFor AtomicRegulon</a></dt>

<dd>
<dt><a name="HasResultsIn_Experiment"
>HasResultsIn Experiment</a></dt>

<dd>
<dt><a name="IndicatedLevelsFor_Feature"
>IndicatedLevelsFor Feature</a></dt>

<dd>
<dt><a name="ProducedResultsFor_Genome"
>ProducedResultsFor Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_ProbeSet [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_ProteinSequence</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_ProteinSequence [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity ProteinSequence.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>We use the concept of ProteinSequence as an amino acid string with an associated MD5 value.
It is easy to access the set of Features that relate to a ProteinSequence.
While function is still associated with Features (and may be for some time),
publications are associated with ProteinSequences (and the inferred impact on Features is through the relationship connecting ProteinSequences to Features).</p>

<p>Example:</p>

<pre>    query_entity_ProteinSequence -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ProteinSequence entity has the following relationship links:</p>

<dl>
<dt><a name="HasAssertedFunctionFrom_Source"
>HasAssertedFunctionFrom Source</a></dt>

<dd>
<dt><a name="IsATopicOf_Publication"
>IsATopicOf Publication</a></dt>

<dd>
<dt><a name="IsAlignedProteinComponentOf_AlignmentRow"
>IsAlignedProteinComponentOf AlignmentRow</a></dt>

<dd>
<dt><a name="IsProteinFor_Feature"
>IsProteinFor Feature</a></dt>

<dd>
<dt><a name="IsProteinMemberOf_Family"
>IsProteinMemberOf Family</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_ProteinSequence [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="sequence"
>sequence</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Protocol</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Protocol [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Protocol.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A Protocol is a step by step set of instructions for performing a part of an experiment.</p>

<p>Example:</p>

<pre>    query_entity_Protocol -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Protocol entity has the following relationship links:</p>

<dl>
<dt><a name="IsMeasurementMethodOf_Measurement"
>IsMeasurementMethodOf Measurement</a></dt>

<dd>
<dt><a name="ProtocolPublishedIn_Publication"
>ProtocolPublishedIn Publication</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Protocol [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Publication</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Publication [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Publication.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>Experimenters attach publications to experiments and protocols.
Annotators attach publications to ProteinSequences.
The attached publications give an ID (usually a DOI or Pubmed ID),
a URL to the paper (when we have it),
and a title (when we have it).
Pubmed IDs are given unmodified.
DOI IDs are prefixed with [b]doi:[/b],
e.g.
[i]doi:1002385[/i].</p>

<p>Example:</p>

<pre>    query_entity_Publication -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Publication entity has the following relationship links:</p>

<dl>
<dt><a name="Concerns_ProteinSequence"
>Concerns ProteinSequence</a></dt>

<dd>
<dt><a name="PublishedExperiment_ExperimentMeta"
>PublishedExperiment ExperimentMeta</a></dt>

<dd>
<dt><a name="PublishedProtocol_Protocol"
>PublishedProtocol Protocol</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Publication [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="title"
>title</a></dt>

<dd>
<dt><a name="link"
>link</a></dt>

<dd>
<dt><a name="pubdate"
>pubdate</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Reaction</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Reaction [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Reaction.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A reaction is a chemical process that converts one set of compounds (substrate) to another set (products).</p>

<p>Example:</p>

<pre>    query_entity_Reaction -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Reaction entity has the following relationship links:</p>

<dl>
<dt><a name="Involves_LocalizedCompound"
>Involves LocalizedCompound</a></dt>

<dd>
<dt><a name="IsDisplayedOn_Diagram"
>IsDisplayedOn Diagram</a></dt>

<dd>
<dt><a name="IsExecutedAs_ReactionInstance"
>IsExecutedAs ReactionInstance</a></dt>

<dd>
<dt><a name="IsStepOf_Complex"
>IsStepOf Complex</a></dt>

<dd>
<dt><a name="ParticipatesIn_Scenario"
>ParticipatesIn Scenario</a></dt>

<dd>
<dt><a name="UsesAliasForReaction_Source"
>UsesAliasForReaction Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Reaction [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="mod_date"
>mod_date</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="abbr"
>abbr</a></dt>

<dd>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="deltaG"
>deltaG</a></dt>

<dd>
<dt><a name="deltaG_error"
>deltaG_error</a></dt>

<dd>
<dt><a name="thermodynamic_reversibility"
>thermodynamic_reversibility</a></dt>

<dd>
<dt><a name="default_protons"
>default_protons</a></dt>

<dd>
<dt><a name="status"
>status</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_ReactionInstance</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_ReactionInstance [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity ReactionInstance.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A reaction instance describes the specific implementation of a reaction in a model.</p>

<p>Example:</p>

<pre>    query_entity_ReactionInstance -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The ReactionInstance entity has the following relationship links:</p>

<dl>
<dt><a name="ImplementedBasedOn_Feature"
>ImplementedBasedOn Feature</a></dt>

<dd>
<dt><a name="IsARequirementOf_Model"
>IsARequirementOf Model</a></dt>

<dd>
<dt><a name="IsExecutionOf_Reaction"
>IsExecutionOf Reaction</a></dt>

<dd>
<dt><a name="Targets_CompoundInstance"
>Targets CompoundInstance</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_ReactionInstance [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="direction"
>direction</a></dt>

<dd>
<dt><a name="protons"
>protons</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Role</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Role [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Role.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A role describes a biological function that may be fulfilled by a feature.
One of the main goals of the database is to assign features to roles.
Most roles are effected by the construction of proteins.
Some,
however,
deal with functional regulation and message transmission.</p>

<p>Example:</p>

<pre>    query_entity_Role -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Role entity has the following relationship links:</p>

<dl>
<dt><a name="DeterminesFunctionOf_Family"
>DeterminesFunctionOf Family</a></dt>

<dd>
<dt><a name="HasAsExemplar_Feature"
>HasAsExemplar Feature</a></dt>

<dd>
<dt><a name="IsConsistentTo_EcNumber"
>IsConsistentTo EcNumber</a></dt>

<dd>
<dt><a name="IsFunctionalIn_Feature"
>IsFunctionalIn Feature</a></dt>

<dd>
<dt><a name="IsIncludedIn_Subsystem"
>IsIncludedIn Subsystem</a></dt>

<dd>
<dt><a name="IsRoleOf_SSCell"
>IsRoleOf SSCell</a></dt>

<dd>
<dt><a name="Triggers_Complex"
>Triggers Complex</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Role [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="hypothetical"
>hypothetical</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Scenario</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Scenario [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Scenario.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A scenario is a partial instance of a subsystem with a defined set of reactions.
Each scenario converts input compounds to output compounds using reactions.
The scenario may use all of the reactions controlled by a subsystem or only some,
and may also incorporate additional reactions.
Because scenario names are not unique,
the actual scenario ID is a number.</p>

<p>Example:</p>

<pre>    query_entity_Scenario -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Scenario entity has the following relationship links:</p>

<dl>
<dt><a name="HasAsTerminus_Compound"
>HasAsTerminus Compound</a></dt>

<dd>
<dt><a name="HasParticipant_Reaction"
>HasParticipant Reaction</a></dt>

<dd>
<dt><a name="Overlaps_Diagram"
>Overlaps Diagram</a></dt>

<dd>
<dt><a name="Validates_Subsystem"
>Validates Subsystem</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Scenario [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="common_name"
>common_name</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Source</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Source [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Source.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A source is a user or organization that is permitted to assign its own identifiers or to submit bioinformatic objects to the database.</p>

<p>Example:</p>

<pre>    query_entity_Source -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Source entity has the following relationship links:</p>

<dl>
<dt><a name="Aligned_Alignment"
>Aligned Alignment</a></dt>

<dd>
<dt><a name="AssertsFunctionFor_ProteinSequence"
>AssertsFunctionFor ProteinSequence</a></dt>

<dd>
<dt><a name="Formulated_CoregulatedSet"
>Formulated CoregulatedSet</a></dt>

<dd>
<dt><a name="HasCompoundAliasFrom_Compound"
>HasCompoundAliasFrom Compound</a></dt>

<dd>
<dt><a name="HasReactionAliasFrom_Reaction"
>HasReactionAliasFrom Reaction</a></dt>

<dd>
<dt><a name="Provided_Subsystem"
>Provided Subsystem</a></dt>

<dd>
<dt><a name="Submitted_Genome"
>Submitted Genome</a></dt>

<dd>
<dt><a name="Treed_Tree"
>Treed Tree</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Source [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_SSCell</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_SSCell [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity SSCell.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>An SSCell (SpreadSheet Cell) represents a role as it occurs in a subsystem spreadsheet row.
The key is a colon-delimited triple containing an MD5 hash of the subsystem ID followed by a genome ID (with optional region string) and a role abbreviation.</p>

<p>Example:</p>

<pre>    query_entity_SSCell -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SSCell entity has the following relationship links:</p>

<dl>
<dt><a name="Contains_Feature"
>Contains Feature</a></dt>

<dd>
<dt><a name="HasRole_Role"
>HasRole Role</a></dt>

<dd>
<dt><a name="IsRoleFor_SSRow"
>IsRoleFor SSRow</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_SSCell [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_SSRow</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_SSRow [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity SSRow.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>An SSRow (that is,
a row in a subsystem spreadsheet) represents a collection of functional roles present in the Features of a single Genome.
The roles are part of a designated subsystem,
and the features associated with each role are included in the row.
That is,
a row amounts to an instance of a subsystem as it exists in a specific,
designated genome.</p>

<p>Example:</p>

<pre>    query_entity_SSRow -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SSRow entity has the following relationship links:</p>

<dl>
<dt><a name="Implements_Variant"
>Implements Variant</a></dt>

<dd>
<dt><a name="IsRowOf_SSCell"
>IsRowOf SSCell</a></dt>

<dd>
<dt><a name="IsUsedBy_Genome"
>IsUsedBy Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_SSRow [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="curated"
>curated</a></dt>

<dd>
<dt><a name="region"
>region</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Strain</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Strain [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Strain.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>This entity represents an organism derived from a genome or another organism with one or more modifications to the organism&#39;s genome.</p>

<p>Example:</p>

<pre>    query_entity_Strain -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Strain entity has the following relationship links:</p>

<dl>
<dt><a name="DerivedFromGenome_Genome"
>DerivedFromGenome Genome</a></dt>

<dd>
<dt><a name="DerivedFromStrain_Strain"
>DerivedFromStrain Strain</a></dt>

<dd>
<dt><a name="EvaluatedIn_ExperimentalUnit"
>EvaluatedIn ExperimentalUnit</a></dt>

<dd>
<dt><a name="HasKnockoutIn_Feature"
>HasKnockoutIn Feature</a></dt>

<dd>
<dt><a name="StrainParentOf_Strain"
>StrainParentOf Strain</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Strain [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="aggregateData"
>aggregateData</a></dt>

<dd>
<dt><a name="wildtype"
>wildtype</a></dt>

<dd>
<dt><a name="referenceStrain"
>referenceStrain</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_StudyExperiment</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_StudyExperiment [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity StudyExperiment.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>An Experiment is a collection of observational units with one originator that are part of a specific study.
An experiment may be conducted at more than one location and in more than one season or year.</p>

<p>Example:</p>

<pre>    query_entity_StudyExperiment -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The StudyExperiment entity has the following relationship links:</p>

<dl>
<dt><a name="IncludesPart_ObservationalUnit"
>IncludesPart ObservationalUnit</a></dt>

<dd>
<dt><a name="IsAssayedBy_Assay"
>IsAssayedBy Assay</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_StudyExperiment [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source_name"
>source_name</a></dt>

<dd>
<dt><a name="design"
>design</a></dt>

<dd>
<dt><a name="originator"
>originator</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Subsystem</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Subsystem [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Subsystem.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A subsystem is a set of functional roles that have been annotated simultaneously (e.g.,
the roles present in a specific pathway),
with an associated subsystem spreadsheet which encodes the fids in each genome that implement the functional roles in the subsystem.</p>

<p>Example:</p>

<pre>    query_entity_Subsystem -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Subsystem entity has the following relationship links:</p>

<dl>
<dt><a name="Describes_Variant"
>Describes Variant</a></dt>

<dd>
<dt><a name="Includes_Role"
>Includes Role</a></dt>

<dd>
<dt><a name="IsInClass_SubsystemClass"
>IsInClass SubsystemClass</a></dt>

<dd>
<dt><a name="IsRelevantTo_Diagram"
>IsRelevantTo Diagram</a></dt>

<dd>
<dt><a name="IsSubInstanceOf_Scenario"
>IsSubInstanceOf Scenario</a></dt>

<dd>
<dt><a name="WasProvidedBy_Source"
>WasProvidedBy Source</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Subsystem [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="version"
>version</a></dt>

<dd>
<dt><a name="curator"
>curator</a></dt>

<dd>
<dt><a name="notes"
>notes</a></dt>

<dd>
<dt><a name="description"
>description</a></dt>

<dd>
<dt><a name="usable"
>usable</a></dt>

<dd>
<dt><a name="private"
>private</a></dt>

<dd>
<dt><a name="cluster_based"
>cluster_based</a></dt>

<dd>
<dt><a name="experimental"
>experimental</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_SubsystemClass</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_SubsystemClass [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity SubsystemClass.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>Subsystem classes impose a hierarchical organization on the subsystems.</p>

<p>Example:</p>

<pre>    query_entity_SubsystemClass -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The SubsystemClass entity has the following relationship links:</p>

<dl>
<dt><a name="IsClassFor_Subsystem"
>IsClassFor Subsystem</a></dt>

<dd>
<dt><a name="IsSubclassOf_SubsystemClass"
>IsSubclassOf SubsystemClass</a></dt>

<dd>
<dt><a name="IsSuperclassOf_SubsystemClass"
>IsSuperclassOf SubsystemClass</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_SubsystemClass [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_TaxonomicGrouping</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_TaxonomicGrouping [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity TaxonomicGrouping.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>We associate with most genomes a &#34;taxonomy&#34; based on the NCBI taxonomy.
This includes,
for each genome,
a list of ever larger taxonomic groups.
The groups are stored as instances of this entity,
and chained together by the IsGroupFor relationship.</p>

<p>Example:</p>

<pre>    query_entity_TaxonomicGrouping -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The TaxonomicGrouping entity has the following relationship links:</p>

<dl>
<dt><a name="IsGroupFor_TaxonomicGrouping"
>IsGroupFor TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsInGroup_TaxonomicGrouping"
>IsInGroup TaxonomicGrouping</a></dt>

<dd>
<dt><a name="IsRepresentedBy_ObservationalUnit"
>IsRepresentedBy ObservationalUnit</a></dt>

<dd>
<dt><a name="IsTaxonomyOf_Genome"
>IsTaxonomyOf Genome</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_TaxonomicGrouping [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="domain"
>domain</a></dt>

<dd>
<dt><a name="hidden"
>hidden</a></dt>

<dd>
<dt><a name="scientific_name"
>scientific_name</a></dt>

<dd>
<dt><a name="alias"
>alias</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_TimeSeries</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_TimeSeries [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity TimeSeries.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A TimeSeries provides a means to specify a series of experimental data over time by ordering multiple ExperimentalUnits.</p>

<p>Example:</p>

<pre>    query_entity_TimeSeries -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The TimeSeries entity has the following relationship links:</p>

<dl>
<dt><a name="OrdersExperimentalUnit_ExperimentalUnit"
>OrdersExperimentalUnit ExperimentalUnit</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_TimeSeries [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="name"
>name</a></dt>

<dd>
<dt><a name="comments"
>comments</a></dt>

<dd>
<dt><a name="timeUnits"
>timeUnits</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Trait</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Trait [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Trait.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A Trait is a phenotypic quality that can be measured or observed for an observational unit.
Examples include height,
sugar content,
color,
or cold tolerance.</p>

<p>Example:</p>

<pre>    query_entity_Trait -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Trait entity has the following relationship links:</p>

<dl>
<dt><a name="Impacts_Contig"
>Impacts Contig</a></dt>

<dd>
<dt><a name="Measures_ObservationalUnit"
>Measures ObservationalUnit</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Trait [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="trait_name"
>trait_name</a></dt>

<dd>
<dt><a name="unit_of_measure"
>unit_of_measure</a></dt>

<dd>
<dt><a name="TO_ID"
>TO_ID</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Tree</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Tree [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Tree.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>A tree describes how the sequences in an alignment relate to each other.
Most trees are phylogenetic,
but some may be based on taxonomy or gene content.</p>

<p>Example:</p>

<pre>    query_entity_Tree -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Tree entity has the following relationship links:</p>

<dl>
<dt><a name="HasNodeAttribute_TreeNodeAttribute"
>HasNodeAttribute TreeNodeAttribute</a></dt>

<dd>
<dt><a name="HasTreeAttribute_TreeAttribute"
>HasTreeAttribute TreeAttribute</a></dt>

<dd>
<dt><a name="IsBuiltFromAlignment_Alignment"
>IsBuiltFromAlignment Alignment</a></dt>

<dd>
<dt><a name="IsModificationOfTree_Tree"
>IsModificationOfTree Tree</a></dt>

<dd>
<dt><a name="IsModifiedToBuildTree_Tree"
>IsModifiedToBuildTree Tree</a></dt>

<dd>
<dt><a name="IsSupersededByTree_Tree"
>IsSupersededByTree Tree</a></dt>

<dd>
<dt><a name="IsTreeFrom_Source"
>IsTreeFrom Source</a></dt>

<dd>
<dt><a name="SupersedesTree_Tree"
>SupersedesTree Tree</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Tree [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="status"
>status</a></dt>

<dd>
<dt><a name="data_type"
>data_type</a></dt>

<dd>
<dt><a name="timestamp"
>timestamp</a></dt>

<dd>
<dt><a name="method"
>method</a></dt>

<dd>
<dt><a name="parameters"
>parameters</a></dt>

<dd>
<dt><a name="protocol"
>protocol</a></dt>

<dd>
<dt><a name="source_id"
>source_id</a></dt>

<dd>
<dt><a name="newick"
>newick</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_TreeAttribute</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_TreeAttribute [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity TreeAttribute.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>This entity represents an attribute type that can be assigned to a tree.
The attribute values are stored in the relationships to the target.
The key is the attribute name.</p>

<p>Example:</p>

<pre>    query_entity_TreeAttribute -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The TreeAttribute entity has the following relationship links:</p>

<dl>
<dt><a name="DescribesTree_Tree"
>DescribesTree Tree</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_TreeAttribute [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_TreeNodeAttribute</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_TreeNodeAttribute [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity TreeNodeAttribute.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>This entity represents an attribute type that can be assigned to a node.
The attribute values are stored in the relationships to the target.
The key is the attribute name.</p>

<p>Example:</p>

<pre>    query_entity_TreeNodeAttribute -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The TreeNodeAttribute entity has the following relationship links:</p>

<dl>
<dt><a name="DescribesTreeNode_Tree"
>DescribesTreeNode Tree</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_TreeNodeAttribute [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
<a name='___top' class='dummyTopAnchor' ></a>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="NAME"
>NAME</a></h2>

<p>query_entity_Variant</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="SYNOPSIS"
>SYNOPSIS</a></h2>

<p>query_entity_Variant [--is field,value] [--like field,value] [--op operator,field,value]</p>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="DESCRIPTION"
>DESCRIPTION</a></h2>

<p>Query the entity Variant.
Results are limited using one or more of the query flags:</p>

<dl>
<dt><a name="the_--is_flag_to_match_for_exact_values;"
>the <code>--is</code> flag to match for exact values;</a></dt>

<dd>
<dt><a name="the_--like_flag_for_SQL_LIKE_searches,_or"
>the <code>--like</code> flag for SQL LIKE searches,
or</a></dt>

<dd>
<dt><a name="the_--op_flag_for_making_other_comparisons."
>the <code>--op</code> flag for making other comparisons.</a></dt>
</dl>

<p>Each subsystem may include the designation of distinct variants.
Thus,
there may be three closely-related,
but distinguishable forms of histidine degradation.
Each form would be called a &#34;variant&#34;,
with an associated code,
and all genomes implementing a specific variant can easily be accessed.</p>

<p>Example:</p>

<pre>    query_entity_Variant -is id,exact-match-value -a &#62; records</pre>

<h3><a class='u' href='#___top' title='click to go to top of document'
name="Related_entities"
>Related entities</a></h3>

<p>The Variant entity has the following relationship links:</p>

<dl>
<dt><a name="IsDescribedBy_Subsystem"
>IsDescribedBy Subsystem</a></dt>

<dd>
<dt><a name="IsImplementedBy_SSRow"
>IsImplementedBy SSRow</a></dt>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="COMMAND-LINE_OPTIONS"
>COMMAND-LINE OPTIONS</a></h2>

<p>query_entity_Variant [arguments] &#62; records</p>

<dl>
<dt><a name="--is_field,value"
>--is field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field has the given value.</p>

<dt><a name="--like_field,value"
>--like field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is LIKE (in the sql sense) the given value.</p>

<dt><a name="--op_operator,field,value"
>--op operator,field,value</a></dt>

<dd>
<p>Limit the results to entities where the given field is related to the given value based on the given operator.</p>

<p>The operators supported are as follows. We provide text based alternatives to the comparison operators so that extra quoting is not required to keep the command-line shell from confusing them with shell I/O redirection operators.</p>

<dl>
<dt><a name="&#60;_or_lt"
>&#60; or lt</a></dt>

<dd>
<dt><a name="&#62;_or_gt"
>&#62; or gt</a></dt>

<dd>
<dt><a name="&#60;=_or_le"
>&#60;= or le</a></dt>

<dd>
<dt><a name="&#62;=_or_ge"
>&#62;= or ge</a></dt>

<dd>
<dt><a name="="
>=</a></dt>

<dd>
<dt><a name="LIKE"
>LIKE</a></dt>
</dl>

<dt><a name="--a"
>--a</a></dt>

<dd>
<p>Return all fields.</p>

<dt><a name="--show-fields"
>--show-fields</a></dt>

<dd>
<p>Display a list of the fields available for use.</p>

<dt><a name="--fields_field-list"
>--fields field-list</a></dt>

<dd>
<p>Choose a set of fields to return. Field-list is a comma-separated list of strings. The following fields are available:</p>

<dl>
<dt><a name="role_rule"
>role_rule</a></dt>

<dd>
<dt><a name="code"
>code</a></dt>

<dd>
<dt><a name="type"
>type</a></dt>

<dd>
<dt><a name="comment"
>comment</a></dt>
</dl>
</dd>
</dl>

<h2><a class='u' href='#___top' title='click to go to top of document'
name="AUTHORS"
>AUTHORS</a></h2>

<p><a href="http://www.theseed.org" class="podlinkurl"
>The SEED Project</a></p>
<hr>
