<h1>
	Using the Assembly Service Command-Line Client 
</h1>
<p>

	<strong>
		Required Prerequisite Activities: 
	</strong>
	
 <a href="/get-started/iris-quick-start">Get Started with IRIS</a> </p>
	
<p>
	KBase command-line interfaces use Linux commands. To familiarize yourself, check out these quick guides: 
	<a href="http://www.debian.org/doc/manuals/refcard/refcard">
		Reference card 
	</a>
	, 
	<a href="http://mally.stanford.edu/~sr/computing/basic-unix.html">
		Beginners Guide 
	</a>
</p>
<h2>
	Introduction 
</h2>
<p>
	The Assembly Service currently is only available through the KBase Download. This service is at an early phase, and supports the following capabilities: 
</p>
<ul>
	<li>
		Velvet assembly: Velvet is a popular de Bruijn graph based assembler for de novo genomic and transcriptome sequence assembly. It was designed for short read sequencing technologies. 
	</li>
	<li>
		Kiki assembly: is a de novo genome and metagenome assembler developed at Argonne. It uses a hybrid of overlap-layout-consensus and greedy extension approaches. Kiki is fast and can work on distributed-memory machines. 
	</li>
	<li>
		A5: A5 is an assembly pipeline optimized for de novo assembly of microbial genomes. It is known for generating large scaffolds. 
	</li>
	<li>
		IDBA: IDBA is de Bruijn graph based assembler that iterates from small to large overlaps to reduce the false positive nodes in the assembly graph and improve assembly quality. 
	</li>
</ul>
<p>
	Here we will walk through a couple examples. 
</p>
<h2>
	Logging In 
</h2>
<p>
	Because assembly computation does not provide immediate results, users will need to login using their KBase account. 
</p>
<p>
	To login at the command-line, use the ar-login command: 
</p>
<pre class="literal-block">
<code title="iris-tutorials">ar-login</code>
and to logout, <code title="iris-tutorials">ar-logout</code>
</pre> File formats supported include fasta and fastq. Compressed files are not yet supported, but we anticipate it will next release (August 2013). 
<h2>
	Assembling Single-End Reads 
</h2>
<p>
	Here we will use a sample metagenome file called 
	<a class="reference external" href="http://www.mcs.anl.gov/~fangfang/test/se.fastq">
		se.fastq 
	</a>
</p>
<pre class="literal-block">
wget http://www.mcs.anl.gov/~fangfang/test/se.fastq
</pre> 
<h3>
	1. Submit the file to be assembled with velvet: 
</h3>
<pre class="literal-block">
<code title="iris-tutorials">ar-run</code> -f se.fastq -a velvet -m &quot;Description of job&quot;
</pre> 
<p>
	The -a sets the tool to use for assembly. This example shows the use of velvet. Users can set "-a" to velvet, kiki, a5, or idba 
</p>
<h3>
	2. Check and monitor the status of your jobs: 
</h3>
<pre class="literal-block">
<code title="iris-tutorials">ar-stat</code> -w
</pre> 
<p>
	Output: 
</p>
<pre class="literal-block">
+--------+---------+-----------------------------------------+----------+-------------+
| Job ID | Data ID |                  Status                 | Run time | Description |
+--------+---------+-----------------------------------------+----------+-------------+
|   83   |    40   |           pipeline [success]            | 0:22:29  |     None    |
|   92   |    40   |              Running: [4%]              | 0:00:22  |     None    |
|   93   |    41   |           pipeline [success]            | 0:00:03  |     None    |
|   94   |    44   |           pipeline [success]            | 0:22:08  |     None    |
+--------+---------+-----------------------------------------+----------+-------------+
</pre> 
<h3>
	3. Get assembly data upon completion: 
</h3>
<pre class="literal-block">
<code title="iris-tutorials">ar-get</code> -j 94
</pre> 
<p>
	The ar-get command retrieves the job status, "-j" sets what job to look up and the number (in this case 94) is the Job ID. 
</p>
<p>
	The assembly service will download 3 files 
</p>
<ul class="simple">
	<li>
		<strong>
			JOB_ID_report.txt 
		</strong>
		: a general log of commands run and files used 
	</li>
	<li>
		<strong>
			JOB_ID_assemblies.tar.gz 
		</strong>
		: a compressed archive of assembler data 
	</li>
	<li>
		<strong>
			JOB_ID_analysis.tar.gz 
		</strong>
		: assembly analysis performed via 
		<a class="reference external" href="http://bioinf.spbau.ru/quast">
			QUAST 
		</a>
	</li>
	</ul>
	<p>
		Follow the QUAST link to learn more about output file contents. 
	</p>

<h2>
	Assembling Pair-End Reads 
</h2>
<p>
	We will walk through the creation and comparison of two assemblies using Kiki and Velvet. Here we will use sample paired end files called 
	<a class="reference external" href="http://www.mcs.anl.gov/~fangfang/test/pe_1.fastq">
		pe_1.fastq 
	</a>
	and 
	<a class="reference external" href="http://www.mcs.anl.gov/~fangfang/test/pe_2.fastq">
		pe_2.fastq 
	</a>
</p>
<pre class="literal-block">
wget http://www.mcs.anl.gov/~fangfang/test/pe_1.fastq
wget http://www.mcs.anl.gov/~fangfang/test/pe_2.fastq
</pre> 
<h3>
	1. Submit the file to be assembled with kiki and velvet: 
</h3>
<pre class="literal-block">
<code title="iris-tutorials">ar-run</code> --pair pe_1.fastq pe_2.fastq -a kiki velvet -m &quot;Description of job&quot;
</pre> 
<h3>
	2. Check and monitor the status of your jobs: 
</h3>
<pre class="literal-block">
<code title="iris-tutorials">ar-stat</code> -w
</pre> 
<p>
	Output: 
</p>
<pre class="literal-block">
+--------+---------+-----------------------------------------+----------+-------------+
| Job ID | Data ID |                  Status                 | Run time | Description |
+--------+---------+-----------------------------------------+----------+-------------+
|   83   |    40   |           pipeline [success]            | 0:22:29  |     None    |
|   92   |    40   |              Running: [4%]              | 0:00:22  |     None    |
|   93   |    41   |           pipeline [success]            | 0:00:03  |     None    |
|   94   |    44   |           pipeline [success]            | 0:22:08  |     None    |
|   95   |    44   |              Running: [50%]             | 0:00:08  |     Test    |
+--------+---------+-----------------------------------------+----------+-------------+
</pre> 
<h3>
	3. Get assembly data upon completion: 
</h3>
<pre class="literal-block">
<code title="iris-tutorials">ar-get</code> -j 95
</pre> 
<p>
	Once again, the assembly service will download 3 files: 
</p>
<ul class="simple">
	<li>
		<em>
			JOB_ID_report.txt 
		</em>
		: a general log of commands run and files used 
	</li>
	<li>
		<em>
			JOB_ID_assemblies.tar.gz 
		</em>
		: a compressed archive of assembler data 
	</li>
	<li>
		<em>
			JOB_ID_analysis.tar.gz 
		</em>
		: assembly analysis performed via 
		<a class="reference external" href="http://bioinf.spbau.ru/quast">
			QUAST 
		</a>
	</li>
</ul>
<h3>
	4. Look at QUAST analysis by extracting the analysis archive: 
</h3>
<pre class="literal-block">
mkdir results
tar -xf 95_analysis.tar.gz -C results/
cat results/report.txt
</pre> 
<p>
	<code>
		mkdir
	</code>
	makes a directory, in this case, called results. The .tar.gz file is an archived and compressed file. 
	<code>
		tar -xf
	</code>
	uncompresses and extracts the files; the "-C" option moves the extracted files into your new directory "results". 
</p>
<p>
	Output: 
</p>
<pre class="literal-block">
Assembly                   P1_Ki    P2_Vt
# contigs (&gt;= 0 bp)        6579     12317
# contigs (&gt;= 1000 bp)     309      190
Total length (&gt;= 0 bp)     2698557  2573302
Total length (&gt;= 1000 bp)  1246799  273827
# contigs                  730      978
Largest contig             107620   3356
Total length               1534546  800695
GC (%)                     40.84    41.95
N50                        4999     806
N75                        1275     617
# N's per 100 kbp          0.00     0.00
</pre> 
<p>
	Or open 
	<strong>
		report.html 
	</strong>
	for visualizations: 
</p>
<img src="/files/5113/7972/2716/quast.png" alt="quast image" style="width:100%;" /> 