<!DOCTYPE html>
<html><head>
    
<meta http-equiv="content-type" content="text/html; charset=UTF-8">
<title>KBase :: MatR</title>
<meta name="description" content="This is an analysis client built as an extension package to the popular R programming language for statistics. Facilities are implemented to download annotation data from KBase. Easy access to lightly structured metadata, authentication for access to private data, and customized versions of relevant statistical and visualization tools are provided.">
<meta name="generator" content="concrete5 - 5.5.1">

<script src="KBase%20_%20MatR_files/ga.js" async="" type="text/javascript"></script><script type="text/javascript">
var CCM_DISPATCHER_FILENAME = '/index.php';
var CCM_CID = 572;
var CCM_EDIT_MODE = false;
var CCM_ARRANGE_MODE = false;
var CCM_IMAGE_PATH = "/concrete/images";
var CCM_TOOLS_PATH = "/index.php/tools/required";
var CCM_REL = "";

</script>

	<link rel="shortcut icon" href="http://kbase.us/files/2613/5006/8591/favicon.ico" type="image/x-icon">
	<link rel="icon" href="http://kbase.us/files/2613/5006/8591/favicon.ico" type="image/x-icon">

<link rel="stylesheet" type="text/css" href="KBase%20_%20MatR_files/ccm.css">
<script type="text/javascript" src="KBase%20_%20MatR_files/jquery.js"></script>
<script type="text/javascript" src="KBase%20_%20MatR_files/ccm.js"></script>
    <link rel="stylesheet" type="text/css" href="KBase%20_%20MatR_files/stylesheet.css">

    <script src="KBase%20_%20MatR_files/nav.js"></script>
  </head>
  <body>
    <script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-38443357-1']);
  _gaq.push(['_setDomainName', 'kbase.us']);
  _gaq.push(['_setAllowLinker', true]);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>    
<div class="bannerArea">
  <div class="container clearfix">
    <div id="search">
      <div class="search-box">
        <form action="/search-results" method="get" class="ccm-search-block-form">
          <input name="query" placeholder="Search this site" autocomplete="off" class="ccm-search-block-text" type="text">
          <input name="start" placeholder="Search this site" autocomplete="off" value="0" class="ccm-search-block-text" type="hidden">           
          <img class="search-icon" type="image" src="KBase%20_%20MatR_files/search_icon.png">            
        </form>
      </div>
      <!--<div class="search-suggestions dropdown-menu"></div>-->
    </div>
    <div class="toplogo"><a href="http://kbase.us/" border="0"><img src="KBase%20_%20MatR_files/Kbase_Logo.png" alt="K Base Logo" height="86" width="250"></a></div>
  </div>
</div>

<div class="top-nav-area">
  <div class="container clearfix"> 
    <div class="top-nav">
      <ul id="MenuBar1"><li style="text-shadow: 0px 2px 3px rgb(34, 34, 34); border-width: 4px 4px 0px; border-style: solid; border-color: rgb(68, 68, 68); -moz-border-top-colors: none; -moz-border-right-colors: none; -moz-border-bottom-colors: none; -moz-border-left-colors: none; border-image: none; background: none repeat scroll 0% 0% rgb(68, 68, 68);" class="clearfix"><span style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" class="">About KBase<div style="visibility: visible;" class="arrow-down"></div></span><ul><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/about/about/" target="_self" class="">Overview</a></li><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/about/the-team/" target="_self" class="">The Team</a></li><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/about/training/" target="_self" class="">Training</a></li><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/about/resources/" target="_self" class="">Resources</a></li><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/about/general-faqs/" target="_self" class="">General FAQs</a></li><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/about/acknowledgements/" target="_self" class="">Acknowledgements</a></li></ul></li><li style="text-shadow: 0px 2px 3px rgb(34, 34, 34); border-width: 4px 4px 0px; border-style: solid; border-color: rgb(68, 68, 68); -moz-border-top-colors: none; -moz-border-right-colors: none; -moz-border-bottom-colors: none; -moz-border-left-colors: none; border-image: none; background: none repeat scroll 0% 0% rgb(68, 68, 68); color: rgb(51, 51, 51);"><span style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" class="">News<div style="visibility: visible;" class="arrow-down"></div></span><ul style="visibility: hidden;"><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/news/blog/" target="_self" class="">Blog</a></li><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/news/upcoming-events/" target="_self" class="">Calendar</a></li><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/news/press/" target="_self" class="">Press</a></li></ul></li><li style="text-shadow: 0px 2px 3px rgb(34, 34, 34); border-width: 4px 4px 0px; border-style: solid; border-color: rgb(68, 68, 68); -moz-border-top-colors: none; -moz-border-right-colors: none; -moz-border-bottom-colors: none; -moz-border-left-colors: none; border-image: none; background: none repeat scroll 0% 0% rgb(68, 68, 68);"><span style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" class="nav-path-selected">For Users<div style="visibility: visible;" class="arrow-down"></div></span><ul><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/for-users/user-home/" target="_self" class="">User Home</a></li><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/for-users/about/" target="_self" class="">About KBase</a></li><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/for-users/get-started/" target="_self" class="">Get Started</a></li><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/for-users/tutorials/" target="_self" class="nav-path-selected">Tutorials</a></li><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/for-users/requirements/" target="_self" class="">Requirements</a></li><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/for-users/user-guide/" target="_self" class="">User Guide</a></li><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/for-users/glimpse-of-the-future/" target="_self" class="">Glimpse the Future</a></li></ul></li><li style="text-shadow: 0px 2px 3px rgb(34, 34, 34); border-width: 4px 4px 0px; border-style: solid; border-color: rgb(68, 68, 68); -moz-border-top-colors: none; -moz-border-right-colors: none; -moz-border-bottom-colors: none; -moz-border-left-colors: none; border-image: none; background: none repeat scroll 0% 0% rgb(68, 68, 68);"><span style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" class="">For Developers<div style="visibility: visible;" class="arrow-down"></div></span><ul><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/developer-zone/overview/" target="_self" class="">Overview</a></li><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/developer-zone/tutorials/" target="_self" class="">Tutorials</a></li><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/developer-zone/api-documentation/" target="_self" class="">Documentation</a></li><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/developer-zone/services/" target="_self" class="">Services</a></li><li><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/developer-zone/downloads/" target="_self" class="">Downloads</a></li></ul></li><li style="text-shadow: 0px 2px 3px rgb(34, 34, 34); border-width: 4px 4px 0px; border-style: solid; border-color: rgb(68, 68, 68); -moz-border-top-colors: none; -moz-border-right-colors: none; -moz-border-bottom-colors: none; -moz-border-left-colors: none; border-image: none; background: none repeat scroll 0% 0% rgb(68, 68, 68);"><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/labs/" target="_self" class="">KBase Labs</a></li><li style="text-shadow: 0px 2px 3px rgb(34, 34, 34); border-width: 4px 4px 0px; border-style: solid; border-color: rgb(68, 68, 68); -moz-border-top-colors: none; -moz-border-right-colors: none; -moz-border-bottom-colors: none; -moz-border-left-colors: none; border-image: none; background: none repeat scroll 0% 0% rgb(68, 68, 68);"><a style="color: rgb(234, 255, 237); text-shadow: 0px 2px 3px rgb(34, 34, 34);" href="http://kbase.us/contact-us/" target="_self" class="">Contact Us</a></li></ul>    </div>
  </div>
</div>

<!--
<div class="login-container">
  <div id="login" class="hide">
    <button class="btn btn-small btn-primary">login<span class="caret"></span></button>
  </div>
  <div id="logout" class="hide">
    <button class="btn btn-small btn-primary">logout</button>
  </div>
          <p id="login-state"></p>
</div>
-->  

    <div class="contentArea">
      <div class="container">
       <h1>
        
<div id="HTMLBlock2074" class="HTMLBlock">
<h2>User Tutorials</h2></div>      </h1>

      <div class="content-doc-left">
       <ul id="MenuBar1"><li class="clearfix"><a href="http://kbase.us/for-users/user-home/" target="_self" class="">User Home</a></li><li><a href="http://kbase.us/for-users/about/" target="_self" class="">About KBase</a></li><li><a href="http://kbase.us/for-users/get-started/" target="_self" class="">Get Started</a></li><li><span class="nav-path-selected">Tutorials<div style="visibility: visible;" class="arrow-down"></div></span><ul><li><a href="http://kbase.us/for-users/tutorials/installing-the-kbase-app-dmg/" target="_self" class="">Installing the KBase.app DMG</a></li><li><a href="http://kbase.us/for-users/tutorials/basics-in-retrieving-data-from-kbase/" target="_self" class="">Basics in Retrieving Data from KBase</a></li><li><a href="http://kbase.us/for-users/tutorials/advanced-methods-for-retrieving-data/" target="_self" class="">Advanced Methods for Retrieving Data</a></li><li><a href="http://kbase.us/for-users/tutorials/advanced-methods-for-retrieving-data-ii/" target="_self" class="">Advanced Methods for Retrieving Data II</a></li><li><a href="http://kbase.us/for-users/tutorials/metabolic-model-reconstruction-and-analysis/" target="_self" class="">Metabolic Model Reconstruction and Analysis</a></li><li><a href="http://kbase.us/for-users/tutorials/navigating-iris/" target="_self" class="">Navigating IRIS</a></li><li><a href="http://kbase.us/for-users/tutorials/annotating-a-genome/" target="_self" class="">Annotating a Genome</a></li><li><a href="http://kbase.us/for-users/tutorials/assembly-service/" target="_self" class="">Assembly Service</a></li><li><a href="http://kbase.us/for-users/tutorials/fba-modeling/" target="_self" class="">FBA Modeling</a></li><li><a href="http://kbase.us/for-users/tutorials/prom-service/" target="_self" class="">Prom Service</a></li><li><a href="http://kbase.us/for-users/tutorials/regulation-service/" target="_self" class="">Regulation Service</a></li><li><a href="http://kbase.us/for-users/tutorials/sequence-quality-control-qc-service/" target="_self" class="">Sequence Quality Control (QC) Service</a></li><li><a href="http://kbase.us/for-users/tutorials/phispy-service/" target="_self" class="">Phispy Service</a></li><li><a href="http://kbase.us/for-users/tutorials/ontology-service/" target="_self" class="">Ontology Service</a></li><li><a href="http://kbase.us/for-users/tutorials/matr/" target="_self" class="nav-selected nav-path-selected">MatR</a></li><li><a href="http://kbase.us/for-users/tutorials/tree-service/" target="_self" class="">Tree Service</a></li><li><a href="http://kbase.us/for-users/tutorials/plant-expression-service/" target="_self" class="">Plant Expression Service</a></li><li><a href="http://kbase.us/for-users/tutorials/genotyping-service/" target="_self" class="">Genotyping Service</a></li><li><a href="http://kbase.us/for-users/tutorials/genotype-phenotype-service/" target="_self" class="">Genotype-Phenotype Service</a></li><li><a href="http://kbase.us/for-users/tutorials/experiment-service/" target="_self" class="">Experiment Service</a></li><li><a href="http://kbase.us/for-users/tutorials/kbase-labs/" target="_self" class="">KBase Labs</a></li></ul></li><li><a href="http://kbase.us/for-users/requirements/" target="_self" class="">Requirements</a></li><li><a href="http://kbase.us/for-users/user-guide/" target="_self" class="">User Guide</a></li><li><a href="http://kbase.us/for-users/glimpse-of-the-future/" target="_self" class="">Glimpse the Future</a></li></ul>      </div>

      <div class="content-doc-right">
        <h1>MatR</h1>




<title>Using the matR Package</title>



<p>
<b>Purpose:</b> Learn what the matR package is, what it can do, how to install it, and where to find more information.
</p>

<p>
<b>Required Prerequisite Activities:</b> A current installation of the R language (&gt;= 2.15) is required.
</p>

<p>
<b>Suggested Prerequisite Activities:</b> Gain some familiarity with R.  Online tutorials can be found at the <a href="http://www.r-project.org/">R Language homepage</a>.  </p>

<h1>Introduction</h1>
<p>
matR is a client for analysis of metagenomic annotations built as an 
extension package to the popular R programming language for statistics. 
 It features facilities to build and examine metagenome collections, 
access to metadata, authentication for access to private data, 
customized analysis and visualization tools, and easy integration with 
the wide world of R extension packages.
</p>

<p>
<b>Related Tutorials:</b>
<p>
The following page presents a typical matR based analysis.  
The individual steps are highlighted below, you can proceed through them
sequentially, or view each separately:  
</p>
</p>


<p>
<b>Suggested Prerequisite Activities:</b> Gain some familiarity with R.  Online tutorials can be found at the <a href="http://www.r-project.org/">R Language homepage</a>.  </p>






<h1>Installation</h1>
At the R prompt, enter: 
<pre>&gt; install.packages ("matR", type = "source",
    repo = "http://dunkirk.mcs.anl.gov/~braithwaite/matR")
</pre>
And then, enter:
<pre>&gt; library (matR)
</pre>
to load the package, and finally enter:
<pre>&gt; dependencies ()
</pre>
to install supporting packages.  Note that in the technical sense there 
are no dependencies, but without supporting packages, many features will
 be unusable.

<h1>Example</h1>
We use a prepackaged collection of 24 metagenomes, fifteen from a fresh water sample, the others from a hot spring.
<pre>&gt; Waters
&gt; samples (Waters)
</pre>
We look at a principal coordinates analysis with samples colored by group:
<pre>&gt; groups (Waters) &lt;- c (rep (1, 15), rep (2, 9))
&gt; col &lt;- groups (Waters)
&gt; levels (col) &lt;- c ("blue", "red")
&gt; col &lt;- as.character (col)
&gt; pco (Waters, comp = c (1,2), main = "fresh vs. spring water samples", col = col)
</pre>
We can also look at principal coordinates in three dimensions, or a heatmap:
<pre>&gt; pco (Waters, comp = c (1,2,3), main = "fresh vs. spring water samples", color = col)
&gt; heatmap (Waters)
</pre>
A statistical test such as Kruskal-Wallis can help identify the most significant rows (annotations) and sharpen the picture:
<pre>&gt; results &lt;- sigtest (Waters, test = "Kruskal-Wallis")
</pre>
We can browse the test results:
<pre>&gt; head (results$sd)
&gt; head (results$mean)
&gt; head (results$stat)
</pre>
Using p-values, we check how many annotations are retained, at various significance levels:
<pre>&gt; sum (results$stat$p.value &lt; 0.05)
&gt; sum (results$stat$p.value &lt; 0.005)
&gt; sum (results$stat$p.value &lt; 0.0005)
</pre>
It is easy to subselect, leaving abundance data for the eighteen 
functional annotations of highest significance in differentiating the 
two groups:
<pre>&gt; subm &lt;- Waters$normed [results$stat$p.value &lt; 0.0005, ]
&gt; dimnames (subm)
&gt; subm
</pre>
A heatmap restricted to these functions shows clear distinctions between
 fresh and spring water, but also a curious aberration in the ninth 
spring water sample:
<pre>&gt; heatmap (Waters, rows = (results$stat$p.value &lt; 0.0005))
</pre>
We can scrutinize available metadata for an explanation, and after 
noticing the various projects contained in the collection, it seems 
natural to compare sample environments:
<pre>&gt; metadata (Waters) ["project.name"]
&gt; metadata (Waters) ["4443745.3"]
&gt; metadata (Waters) ["env_package"]
&gt; metadata (Waters) ["sample.data"]
&gt; metadata (Waters) ["temperature"]
</pre>
Nothing turns up, so we turn to library information:
<pre>&gt; metadata (Waters) ["library"]
</pre>
There, we discover that the ninth spring water sample, unlike every 
other in the collection and even in its own project, underwent Amplicon 
sequencing:
<pre>&gt; metadata (Waters) ["library.type"]
</pre>
At this point we may wish to continue analysis with subcollections, using WGS metagenomes, only:
<pre>&gt; Waters2 &lt;- Waters [1:23]
</pre>
Or we may, to isolate a comparison of different sequencing technologies,
 select metagenomes only from the project that includes the Amplicon 
metagenome:
<pre>&gt; Waters3 &lt;- Waters [19:24]
</pre>
For instance, we can redo the eighteen-row heatmap from above, with the Amplicon metagenome omitted:
<pre>&gt; groups (Waters2) &lt;- c (rep (1, 15), rep (2, 8))
&gt; results2 &lt;- sigtest (Waters2, test = "Kruskal-Wallis")
&gt; heatmap (Waters2, rows = (results2$stat$p.value &lt; 0.0005))
</pre>
Significance testing could also be redone with annotations from a higher (or lower) level of the Subsystems hierarchy:
<pre>&gt; Waters$lev2 &lt;- c (entry = "normed", level = "level2")
&gt; results.lev2 &lt;- sigtest (Waters, view = "lev2", test = "Kruskal-Wallis")
&gt; heatmap (Waters, view = "lev2", rows = (results.lev2$stat$p.value &lt; 0.005))
&gt; rownames (Waters$lev2) [results.lev2$stat$p.value &lt; 0.005]
</pre>
We added a new view to the collection, and proceeded as before.



<h1>Functionality</h1>
This tutorial provides an overview of functionality, by example.

<h2>Collections</h2>
Metagenome collections are constructed using metagenome IDs.  For example:
<pre>&gt; ids &lt;- c ("4441679.3", "4441680.3", "4441682.3", "4441695.3", "4441696.3", "4440463.3", "4440464.3")
&gt; cc &lt;- collection (ids)
</pre>
This creates a collection object containing metadata and abundance data 
for the identified metagenomes.  Various functions apply to collections:
<pre>&gt; samples (cc)        # show ids of constituent metagenomes
&gt; names (cc)          # show names of metagenomes
&gt; views (cc)          # show the matrix views that the collection contains (see below)
&gt; viewnames (cc)      # show just the names of the views
&gt; metadata (cc)       # access metadata of the collection
</pre>
Views of collections are accessed with "$".   For example, this command 
shows raw abundance counts of functional annotations at level 3 of the 
Subsystems hierarchy:
<pre>&gt; cc$count
</pre>
while this shows the same data, but normalized to lie between 0 and 1:
<pre>&gt; cc$normed
</pre>
Generally, the purpose of views is to show different aspects of the same
 selection of metagenomes.  Views can be specified when a collection is 
constructed:
<pre>&gt; dd &lt;- collection (ids, kegg = c (source = "KEGG"))
</pre>
and can also be added to an existing collection:
<pre>&gt; dd$cog &lt;- c (source = "COG")
</pre>
Subsets can be taken of collections, as of regular vector objects.  For instance, a subcollection is created by
<pre>&gt; ee &lt;- cc [1:4]
</pre>
that contains only the first four metagenomes.


<h2>Metadata</h2>
Metadata is a list of named elements with a hierarchical organization.  
To select elements, an arbitrary number of index vectors may be 
specified.  For instance, we can use one index (of length one) to get 
all metadata from one metagenome:
<pre>&gt; mm &lt;- metadata (cc)
&gt; mm ["4440464.3"]
</pre>
Or we can use two indices, each of length one, to get certain elements from all metagenomes:
<pre>&gt; mm ["latitude", "longitude"]
</pre>
The following example uses one index of length two, to get the entire environmental package from one metagenome:
<pre>&gt; mm [c ("4440464.3", "env_package.data")]
</pre>
And this example uses three indices, all of length two, to select a miscellany of elements:
<pre>&gt; mm [c ("0464", "PI"), c ("0464","seq"), c ("0464","biome")]
</pre>
These are just examples to show the flexibility of indexing metadata.  
One other convenient feature creates a data.frame with one row per 
metagenome, with this syntax:
<pre>&gt; mm ["host_common_name", "disease", ".age", bygroup = TRUE]
</pre>
If a requested metadata field is missing from a metagenome, NA is placed instead.

<h2>Basic Analysis</h2>
A very large library of statistical tools is available in the R 
language.   matR provides convenient wrappings of some useful procedures
 and visualizations.  The following examples use prepackaged datasets.
<p>
<!--<img alt="" src="boxplot.svg"><br>-->
Boxplots usefully summarize diversity, with this command:
</p><pre>&gt; render (Coral, view="count")
&gt; render (Coral, view="normed")
</pre>
<!--<img alt="" src="pco.svg"><br>-->
A principal components analysis is plotted in two or three dimensions, and grouped by color, this way:
<pre>&gt; g &lt;- factor (metadata (Hospital) ["ventilation_type"])
&gt; col &lt;- g
&gt; levels (col) &lt;- c ("red", "blue", "orange")
&gt; col &lt;- as.character (col)
&gt; pco (Hospital, comp = c (1,2), col = col, labels = g)
&gt; P &lt;- pco (Hospital, comp = c (1,2,3), color = col, labels = g)
</pre>
The numerical results of the analysis are stored in the object P.
<p>
<!--<img alt="" src="heatmap1.svg"><br>-->
A heatmap gives a quick visual summary of annotations, as well suggesting further investigations:
</p><pre>&gt; heatmap (Guts, cexCol = 0.9)
</pre>
Significance testing of annotations is the starting point for detailed 
analysis.  The following routine calculates the significance of each 
annotation in establishing a specified grouping of samples, by the 
Kruskal-Wallis test.  Then, a heatmap is made of only the rows passing a
 specified significance threshold.
<!--<img alt="" src="heatmap2.svg"><br><br>-->
<pre>&gt; groups(Waters) &lt;- c (rep (1, 15), rep (2, 9))
&gt; results &lt;- sigtest (Waters, test = "Kruskal-Wallis")
&gt; heatmap (Waters, rows = (results$stat$p.value &lt; 0.005), main = "annotations at p &lt; 0.005")
</pre>
Such analyses can be fruitfully iterated.

<h2>Integration</h2>
matR integrates easily with other R software.  Collection views are 
directly extracted as ordinary matrices with the $ operator, as above.  
Metadata is an S3-classed character vector.  Visualization routines 
accept all standard graphical parameters.  Shortcuts are provided to 
write collection data and analyses to text files.

<h1>Further Resources</h1>
Once the package is loaded in an R session, use:
<pre>&gt; demo (package = "matR")
</pre>
for a list of short tutorial demos.  Run these with the regular R command demo() or the enhanced version demo2(), for instance:
<pre>&gt; demo2 ("metadata")
</pre>
To list prepackaged datasets, use:
<pre>&gt; data (package = "matR")
</pre>



      </div>
    
      <div style="clear:both;"></div>
      </div>

    </div>

    
<div id="search-results"></div>

<div id="footer">
  <div class="container">
    <div class="footer-column">
      <ul>
<li><a href="https://gologin.kbase.us/SignUp">Sign up for a KBase account</a></li>
</ul>    </div>

    <div class="footer-column">
      <ul>
<li><a href="http://kbase.us/about/acknowledgements/">Acknowledgements</a></li>
</ul>    </div>        

    <div class="footer-column">
      <p style="font-size: .9em;">KBase is sponsored by the&nbsp;<a href="http://science.energy.gov/ber/">U.S. Department of Energy's Office of Biological and Environmental Research</a></p>
<p><img src="KBase%20_%20MatR_files/doe_sc_logo.png" alt="U.S. Department of Energy -- Office of Science" height="28" width="140"></p>    </div>
  <div style="clear:both;"></div> 
  </div>
</div>

      

</body></html>