<div>
<h1>Retrieving Tree Structure and Alignment Data</h1>
<p>With the same 'get-tree' and 'get-alignment' commands, you can retrieve the tree structure in
<a href="http://en.wikipedia.org/wiki/FASTA_format">newick</a> format or an alignment in
<a href="http://en.wikipedia.org/wiki/FASTA_format">fasta</a> format.  Run the same command, but
omit the '-m' option:</p>
<pre>
    > <code title = "iris-tutorials">tree-get-tree</code> 'kb|tree.991753'
    (((((20604894_1_491_700:0.6297,((17850078_1_500_699:0.05393,20109977_1_501_700:0.0718)0:0.03445,
    (19471596_1_492_701:1.54275,20480750_1_492_701:0.00541)0.959:1.10426)1:0.83977)0.993:0.99986,(19
    514631_1_3_290:1.12331,20800851_1_1_97:3.85784)0.733:1.55142)0.32:0.11495,6396335_1_23_358:2.704
    04)0.854:0.32054,(((((8501581_1_263_414:0,20544729_1_221_372:0):1.31674,13411475_1_261_406:1.287
    29)0.622:0.38827,((2636527_1_236_543:0,13325328_1_140_447:0,13333732_1_182_489:0):2.45305,((7281
    613_1_33_184:0.00016,(3178202_1_194_390:0,13792085_1_179_375:0):0.60306)0.687:0.71617,((20720986
    ....
    > <code title = "iris-tutorials">tree-get-alignment</code> 'kb|aln.991753'
    >100153_1_279_577
    ------------------------------------------------------------------------------------VDAVNQIAQ ...
    >100162_1_219_588
    ---------AYMGRLAEGDYSQEPPMRERKDEVGEMAAAVSVFRAAAIERRQAVQLEKDR----EAAAREEAFVAAEAEARGRRSFVVDALDE ...
    >10122599_1_5_300
    --------------------------------------------------------------------------------------------- ...
    >10122879_1_680_1064
    --------------------------RKRQEAINAELEAINRRLEAEMRHAAEAEREKRRALEEAAEHDRRAREQLQQQQQEYLEQSVLRILE ...
    >10124179_1_205_610
    SVVRPVQELEAAAYRISEGDLETTVAVRTEDEVGKLARAFNQMVASMREALDA-----LQEEKAGAEQKVREAVAEIQAQQRYLAESVEHMLT ...
    ....
</pre>
</div>
<div>
<p>Trees and alignments in KBase are stored with each leaf node or alignment row labeled by an ID that is
unique only within the given tree or alignment.  This is necessary because some leaves in a tree may be mapped to
different regions of a single protein sequence (so protein MD5s cannot uniquely identifiy leaves), while in other
cases a single leaf may be representative of a set of concatenated protein sequences.  Therefore in general, the raw
node labels will not directly map to protein or feature IDs in KBase.</p>
<p>For gene trees built from protein sequence alignments that have only a single KBase protein sequence per alignment row,
it is possible to retrieve a tree or an alignment with node labels replaced with KBase protein sequence identifiers.</p>
<pre>
    > <code title = "iris-tutorials">tree-get-tree</code> -p 'kb|tree.991753'
    (((((32079382d2a3f6f5f2988f97233c92cd:0.6297,((edf756abf2c4186fc34a9c94de377d20:0.05393,5babb472e
    d7ae9399f8f5103f51b91ee:0.0718)0:0.03445,(d6369e4d35cdbd6ddf1116d311fd7a59:1.54275,ef323273867b73
    b39ab5fedae251f47f:0.00541)0.959:1.10426)1:0.83977)0.993:0.99986,(49cdbb61b099a7cae3b4fc46dd34f44
    b:1.12331,98b5ce0f9772ff421f34d6d50bf92323:3.85784)0.733:1.55142)0.32:0.11495,caf6be0a8bf38e96b62
    61d382413fb09:2.70404)0.854:0.32054,(((((049c37f0c44a9df0d16ff4bd76737dfd:0,7ac4a6cb10d15f1a2f468
    2d89f10f340:0):1.31674,c419f8f2cd68d25ee71d4899e9fe535d:1.28729)0.622:0.38827,((f26b99054a24ecd26
    c6d00e695dea3a:0,7527128b4098231d0a8ea0cd188d342e:0,4696ad83733c59849baf1d79cb6134e6:0):2.45305,(
    ....
    > <code title = "iris-tutorials">tree-get-alignment</code> -p 'kb|aln.991753'
    >480553824a94dd69d05e2976c1727771
    ------------------------------------------------------------------------------------VDAVNQIAQ ...
    >8cd02ced30b18167e19d9202c58d492a
    ---------AYMGRLAEGDYSQEPPMRERKDEVGEMAAAVSVFRAAAIERRQAVQLEKDR----EAAAREEAFVAAEAEARGRRSFVVDALDE ...
    ....
</pre>
</div>
<div>
<p>There is not a strict one-to-one mapping between protein sequences and genomic features; some features may share identical protein
sequences.  Therefore it is not always possible to map an alignment row or leaf node to a <i>single</i> KBase feature.
For some trees, however, a cannonical KBase feature was used to identify the protein sequence to align.  For these
trees, leaf labels can be replaced with feature-ids using the '-f' flag with the following command.</p>
<pre>
    > <code title = "iris-tutorials">tree-get-tree</code> -f 'kb|tree.991753'
    (((((kb|g.3596.peg.2220:0.6297,((kb|g.2711.peg.128:0.05393,kb|g.3446.peg.2305:0.0718)0:0.03445,(k
    b|g.3214.peg.1063:1.54275,kb|g.3557.peg.680:0.00541)0.959:1.10426)1:0.83977)0.993:0.99986,(kb|g.3
    227.peg.1561:1.12331,kb|g.3666.peg.2259:3.85784)0.733:1.55142)0.32:0.11495,kb|g.2160.peg.2450:2.7
    0404)0.854:0.32054,(((((:0,kb|g.3574.peg.1137:0):1.31674,kb|g.1155.peg.3402:1.28729)0.622:0.38827
    ,((kb|g.21423.CDS.6883:0,kb|g.1143.peg.6798:0,kb|g.1144.peg.5065:0):2.45305,((kb|g.1465.peg.4094:
    0.00016,(:0,kb|g.1297.peg.1348:0):0.60306)0.687:0.71617,((kb|g.3639.peg.5961:0,kb|g.22013.CDS.347
    7:0):0.53728,((kb|g.2550.peg.2415:0,kb|g.2551.peg.2233:0):0.43479,(kb|g.2550.peg.1805:0,kb|g.2551
    .peg.2376:0):0.32658)0.971:0.42267)0.976:0.59761)0.763:0.60071)0.441:0.14829)0.605:0.13015,(((((k
    ....
    > <code title = "iris-tutorials">tree-get-alignment</code> -f 'kb|aln.991753'
    >
    ------------------------------------------------------------------------------------VDAVNQIAQ ...
    >kb|g.195.peg.2959
    ---------AYMGRLAEGDYSQEPPMRERKDEVGEMAAAVSVFRAAAIERRQAVQLEKDR----EAAAREEAFVAAEAEARGRRSFVVDALDE ...
    >kb|g.1913.peg.2473
    --------------------------------------------------------------------------------------------- ...
    ....
</pre>
</div>
<div>
<p>Note in the output above that if a given tree or alignment does not have a canonical feature ID, then
that ID is left blank.  Keep in mind also that these feaures may not be the only features in KBase that map
to a particular leaf!  Therefore you may have used the 'tree-find-tree-ids' method to identify a tree containing
that feature's protein sequnce, but later find that the 'tree-get-tree' with the '-f' option returns a tree that
does not include your specific feature id.  Therefore, it is often safer to work with protein sequence identifiers
and only map to feature IDs for display.</p>
</div>
