<div>
<h1>RegPrecise</h1>
<strong>Required Prerequisite Activities:</strong> 
<a href="/for-users/tutorials/installing-the-kbase-app-dmg/">Download KBase: Installing the KBase Application DMG</a>, OR <a href="/for-users/tutorials/navigating-iris/">Get Started with Iris: Navigating Iris</a> </p>
<p>KBase Interfaces, Iris or the Command-line, use Linux commands. To familiarize yourself, check out these quick guides: <a href= "https://www.cfa.harvard.edu/~jbattat/computer/linuxReferenceCard.pdf"> Reference card</a>, <a href= "http://mally.stanford.edu/~sr/computing/basic-unix.html">Beginners Guide</a></p>

</div>
<div><h2> Introduction</h2>
<p>This service can be used to construct a regulatory model for a
genome by
projecting the
regulatory models from a collection of analyzed genomes to a new
genome.
</p>
</div><div><h2>Constructing the model</h2>
<h3>1. Lookup genome id</h3>
<p> The first step is to select the genome that you wish to analyze and
construct the regulatory model. The genome must be preloaded into the
KBase infrastucture. Use the KBase command <i>all_entities_genome</i> to
list the genomes that are avalable,
using grep (or other commands) to narrow down the search.
</p>
<pre>  <code title = "iris-tutorials">all_entities_Genome</code> -f scientific_name | grep -i 'Shewanella sp MR-4'<br>  kb|g.19865	Shewanella sp MR-4<br></pre>
<p>
For this tutorial we will use the genome id 'kb|g.19865'
</p>
</div>
<div>
<h3>2. Display the available model collections</h3>
<p>
To construct a model for a new genome, you must select the regulatory
model collection you wish to use as a reference. This is a manual
process based on taxonomic distance
and collection quality. To get started and in order to look up all of the
available model collections, iterate through them, and print the
collection names and their properties.
</p>
<pre>  ## Get all the regulome model collections<br>  <code title = "iris-tutorials">getRegulomeModelCollections</code><br><br>  ## Iterate through the collection<br>  ##   print the collection name and properties<br></pre>
<h3>3. Select the model collection for your genome</h3>
<p>
Examine the list of model collections and find the collection that is
most closely related to your genome. You should also consider the
quality of the model by looking at the number of regulons and regulated
genes in each model. Model collections
with more regulons will generally work better.
</p>
</div>
<div>
<h3>4. Build the regulome model</h3>
<p>
Now that you have selected your genome id and your regulon collection
id, you
can build the regulatory model for the genome by projecting the regulon
model
from the selected collection to the new genome. This process will
typically take
5 to 10 minutes to complete, so the build command will just return a
process id
as the job is launched. You should then periodically check on the
status of that
process every 10 or 15s until it finishes.
</p>
<pre>  processState = buildRegulomeModel<br><br>  while (processState-&gt;!done)<br>  do<br>    sleep(15s)<br>  done<br></pre>
<br>
</div>
<div>
<h3>5. Examine the regulome model</h3>
<p>
After the regulome model has been constructed on the new genome, you
can inspect it
or compare it to other models. Biologically, the regulon that regulates
the largest
number of genes is the global regulon for the species. This can be
easily discovered
by computing some basic summary statistics on the new model
</p>
<pre>  list<regulonmodelstat> stats = getRegulonModelStats(regulomeModelId)<br><br>  $global_regulon_id = undef;<br>  $global_regulon_cnt = 0;<br><br>  Iterate through the stats<br>    Print the regulonModelId, regulatedOperonCount, regulatedGeneCount<br>    if (regulatedGeneCount &gt; global_regulon_cnt)<br>       $global_regulon_id  = regulonModelId<br>       $global_regulon_cnt = regulatedGeneCount<br>       <br><br>  print "The master regulon is $global_regulon_id and it regulates $global_regulon_cnt genes"<br><br></regulonmodelstat></pre>
</div>
